---
title: Topography consistently drives intra- and interspecific leaf trait variation within Neotropical tree species
output:
  bookdown::word_document2: default
  bookdown::html_document2:
    number_sections: false
    toc: true
    toc_float: yes
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/oikos.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(ggfortify)
library(broom)
library(rstan)
library(bayesplot)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, 
               fig.height = 8, fig.width = 8,
               cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Paracou/"
```

```{r data}
# load("../functional_save/Individuals.Rdata")
# Individuals0 <- Individuals
# sdDBH <- sd(Individuals$DBH)
# Individuals <- Individuals %>%
#   rename(LMA = invSLA) %>%
#   group_by(Species) %>%
#   filter(n() > 20) %>%
#   ungroup() %>%
#   group_by(SpeciesLong) %>%
#   mutate(TWIs = mean(TWI, na.rm = T)) %>%
#   mutate(TWIis = TWI - TWIs) %>%
#   ungroup() %>%
#   mutate_at(vars("LMA", "LDMC", "LT", "invLA", "CC", "DBH", "TWIs", "TWIis"),
#             funs(./sd(., na.rm = T))) %>%
#   group_by(Genus) %>%
#   mutate(SpeciesNum = as.numeric(as.factor(Species))) %>%
#   ungroup()
# species <- Individuals %>%
#   dplyr::select(SpeciesLong, Genus, Species, SpeciesNum) %>%
#   unique()
# load("../functional_save/InterIntraTWI.Rdata")
# traits <- c("LMA", "LDMC", "LT", "invLA", "CC")
# complexes <- c("Symphonia", "Eschweilera")
# models <- sapply(complexes, function(complex) paste0(complex, "-", traits))
# S_sqrt_trans <- function() scales::trans_new("S_sqrt",
#                                              function(x) sign(x)*sqrt(abs(x)),
#                                              function(x) x^2*sign(x))
# save(list = ls(), file = "draftData.Rdata")
load("draftData.Rdata")
```

```{r reviewersComments, eval=F}
src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(CensusYear == 2015) %>% 
  group_by(Plot) %>% 
  summarise(N = n(), Area = PlotArea) %>% 
  mutate(N = N/Area) %>% 
  ungroup() %>% 
  summarise(mean = mean(N), min = min(N), max = max(N))
paracou <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
    tbl("Paracou") %>% 
  filter(CensusYear == 2017) %>% 
  collect() %>% 
  mutate(SpeciesLong = paste(Genus, Species))
paracou <- paracou %>%
   dplyr::select(SpeciesLong, idTree, Xutm, Yutm) %>%
   unique() %>% 
  na.omit()
coordinates(paracou) <- ~Xutm + Yutm
proj4string(paracou) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
paracou <- spTransform(paracou, CRSobj = crs)
twi <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "TWI_1m.tif"))
twi <- projectRaster(twi, crs = crs)
r2 <- twi
r2[] <- 0
counts<- table(cellFromXY(twi, paracou))
r2[as.numeric(names(counts))] <- counts
data.frame(r2 = values(r2)) %>% 
  filter(r2 > 0) %>% 
  group_by(r2) %>% 
  summarise(N = n())
```

# Abstract
 
Theories on the maintenance of biodiversity in species-rich communities have mainly addressed coexistence at the species level. This approach, however, has some limitations as individuals within species vary in their genotypes, their phenotypes,  and their performance. A growing body of research has suggested that intraspecific trait variability is likely to impact community assembly. 
To gain insights into the structure of intraspecific trait variation in tropical tree species, we examined leaf functional trait variation with topography in a hyperdiverse tropical forest of the Guiana Shield. We collected leaf functional traits from 766 trees belonging to two genera and five species in permanent plots encompassing a diversity of  topographic positions. We tested the role of topography on leaf functional trait variation with a hierarchical Bayesian model, controlling for individual tree diameter effect. 

Here, we show that, mirroring what has been previously observed among species and communities, individual leaf traits covary from acquisitive to conservative strategy within species. Moreover, increasing topographic position resulted in a shift of leaf traits from acquisitive to conservative strategies both across and within species. Our results suggest that intraspecific trait variability widens species’ niches and,  if supported by fitness homeostasis, promote species coexistence.

## Keywords

intraspecific variability; leaf traits; species coexistence; tropical forests ; Paracou

\newpage 

# Introduction

The ecological processes underlying species coexistence in highly diverse communities such as tropical forests have been an active field of research for more than forty years [@connell_diversity_1978; @wright2002plant]. Niche theory posits that species can coexist because they occupy different niches, thus limiting competitive exclusion [@lortie_rethinking_2004-1; @weiher_assembly_1995], whereas neutral theory shows that even functionally equivalent species can coexist because of stochastic life, death, reproduction and dispersal dynamics [@Hubbell2001]. Functional traits reflect fundamental trade-offs in species co-occurrences along environmental gradients [@Wright2002] and shape the structure [@Paine2011] and dynamics [@Herault2018] of communities in conjunction with their environment [@kraft_functional_2008]. Trait-based studies therefore contribute to testing theories of community assembly, and thus to improving our knowledge on the maintenance of biodiversity [@wright2002plant], providing crucial information  for community resilience in the face of increasing local and global environmental change [@Sakschewski2016]. 

Functional traits have been defined as phenotypic traits that impact fitness indirectly through their effect on individual performance, which is defined as the ability to recruit, grow, survive, and reproduce in a given environment [@violle_let_2007]. Functional traits covary among species [@diaz_global_2015] and communities [@Bruelheide2018] along distinct economics spectra of leaf [@wright_worldwide_2004; @osnas_global_2013; but see @lloyd_photosynthetically_2013] and wood [@chave_towards_2009]. The leaf economics spectrum opposes acquisitive ecological strategies, with high photosynthetic carbon assimilation, to conservative ones with high investment in leaf defense and durability. For instance, specific leaf mass per area (i.e. fresh weight divided by area) varies among species along gradients of soil fertility and exposure to light and represents a trade-off between resource acquisition by photosynthesis and investment in leaf defense and durability [@Evans2001a; @hodgson_is_2011].  The wood economics spectrum opposes slow growing to fast growing species [@chave_towards_2009]. Nevertheless, some authors advocate for a unique plant economics spectrum [@reich_world-wide_2014].

In community ecology, the species has long been the fundamental unit of trait characterization, neglecting intraspecific variability [@mcgill_rebuilding_2006]. Trait-based community ecology rests on the expectation that functional traits determine demographic outcomes. However, it has been shown that traits have limited power to predict demographic rates at the species level [@Paine2015b; @Yang2018a]. Accounting for individual fitness can shed new light on processes of community assembly because changes in individual fitness may promote long-term coexistence of species [@Clark2010]. This highlights the need to consider within-species variation to investigate ecological processes, for example by examining the variation in individual performance, phenotypic traits, and genes [@Albert2010; @Violle2012]. Phenotypic variation within species is shaped by genetic heritage, through genotypes, and by both the abiotic and biotic environment, including its spatial and temporal heterogeneity [@Whitlock2007], and is modulated by ontogeny [@Wright2002a]. Describing the relative roles of these sources of variation among individuals within and among species should thus contribute to a better understanding of the ecological and evolutionary processes that underlie the maintenance of biological variation within communities at both inter- and intraspecific level.

Large intraspecific variability has been documented in tree allometric relationships [@Vieilledent2010] and in leaf trait values [@Hulshof2010; @Messier2010], and marked differences in variability have been detected among species within communities [@Siefert2015a]. Several studies have investigated the role of intraspecific variability in community assembly and evidenced effects of environmental filtering [@Messier2010; @Paine2011], with shifts in trait values following environmental and resource gradients [@jung_intraspecific_2010; @Siefert2016]. Finally, inter- and intraspecific variability joint answers to environmental gradient remain under-documented, but @Kichenin2013 highlighted both positive and negative covariation of inter- and intraspecific response to topography effect on plant functional traits.

Finally, several authors suggested that intraspecific variability can both promote or hinder species coexistence [@Turcotte2016]. Indeed, species coexistence will depend on intraspecific competition being stronger than interspecific competition [@Chesson2000]. Species coexistence can be promoted when increasing niche differentiation or decreasing fitness differences contribute to overcoming competitive exclusion [@Turcotte2016]. For instance, [@Perez-Ramos2019] found that intraspecific variability in root density resulted  in niche differentiation in annual plants. Additionally, intraspecific variability can contribute to fitness homeostasis, defined as the ability to maintain fitness across heterogeneous environments [Jack-of-all-trades strategy, @Richards2006]. Conversely, intraspecific trait variability may hinder species coexistence when it increases fitness differences. Consequently, intraspecific variability can affect species coexistence through its effects on the species’ niche and individual fitness [@Turcotte2016], or more globally, through its effects on population demography [@Clark2010].

In the present study, we assessed variation in leaf functional trait values of individual tree, among and within species, and addressed the effects of abiotic environment on this variation. We controlled for the effect of tree diameter, as a proxy of tree size and access to light that are both known for their effect on intraspecific trait variation [@Chazdon1993; @Koch2004; @WOODRUFF2007]. We measured leaf traits on a large number of individuals (766 trees > 10 cm in diameter at breast height) belonging to five Neotropical tree species from two dominant genera in a highly diverse tropical forest site located within the Guiana Shield in the Amazon Basin. The site encompasses a diversity of micro-habitats through hydrologic and topographic variation ranging from seasonally flooded bottomlands to drier plateaus [@Allie2015; @ferry2010higher]. Combining tree inventories, LiDAR-derived topographic data, and leaf functional traits, we used multivariate approaches and Bayesian modelling to address the following questions: (1) how do traits covary among individuals within tropical tree species and (2) how do abiotic environment influence individual leaf trait values among and within species? Based on conservation of functional strategies both within plant communities and at the global level [@Bruelheide2018; @diaz_global_2015], we expected trait covariation to be maintained at the within-species level [but see @Messier2017]. We hypothesized that abiotic environment shaped trait variation both among and within species in interaction with tree diameter and access to light [@roggy_links_2005; @Sciences2009], without presuming the sign of the relation between inter- and intraspecific effects [@Kichenin2013].

# Material and Methods

## Study site

The study was conducted in the Guiana Shield, at the Paracou field station in the coastal region of French Guiana, northern South America. The site is characterised by an average annual rainfall of 3041 mm and a mean air temperature of 25.71 °C [@Aguilos2018]. Old tropical forest with an exceptional richness (over 200 woody species per hectare) dwells in this lowland area characterised by heterogeneous microtopographic conditions with numerous small hills generally not exceeding 45 m in elevation [@Gourlet-Fleury2004].

The site comprises 16 permanent plots (fifteen 6.25 ha plots plus one 25 ha plot) which have been censused (diameter at breast height >10 cm) every 1-2 years since 1984. Nine of the plots were intentionally manipulated in 1986 with a range of disturbance intensities that created a variety of biotic environments [details on the experiment in @Herault2018].

## Plant material

Two sampling campaigns were led in 2017 and 2018 during the dry season, between September and December. Sampling was made for two species of *Symphonia* (Clusiaceae) in 2017 and for three species of *Eschweilera* (Lecythidaceae) in 2018  (Table \@ref(tab:headcount)). All three Eschweilera species belong to the Parvifolia clade [@Huang2015; @Mori2016] and represent the most abundant of eleven sympatric species in the clade recorded in the 2017 inventory. The three *Eschweilera* species and the two *Symphonia* species are Amazonian hyperdominant tree species [@TerSteege2013]) and are among the most abundant species in the Paracou station (e.g. *Eschweilera sagotiana* was the second most abundant tree species in the 2017 inventory in Paracou).

The genus *Symphonia* includes the well-recognised species *Symphonia globulifera* L.f (Clusiaceae) with two morphotypes, *S. globulifera sensu stricto* (*S. globulifera* hereafter) and *Symphonia sp.1*, occurring in sympatry but in differentiated habitats, with *S. globulifera* preferentially growing in valley bottoms and *S. sp1* preferentially exploiting a variety of drier habitats. Similarly, *Eschweilera sagotiana* and *E. coriacea* have been shown to exhibit a niche differentiation for topographic position [@Allie2015]. The genus *Symphonia* and the *Parvifolia* clade within the genus *Eschweilera* have been both highlighted as species complexes with low phylogenetic resolution and high levels of plastid DNA sharing among sister species [@baraloto_using_2012-1; @Caron2019; @Gonzalez2009; @Torroba-Balmori2017].

For each individual tree (see sample sizes in Table \@ref(tab:headcount)), five mature and healthy leaves were sampled at the top of the crown using a slingshot sampling device (Bigshot), and kept in humidified ziplock bags with CO2-enriched air in darkness until measurement within the next 6 hours following a standard protocol [@Perez-Harguindeguy2013]. Access to light for each sampled tree was assessed using the Dawkins index [@Dawkins1958].

## Trait measurements

For each leaf, we measured five leaf functional traits related to resource investment strategies [@diaz_plant_1998; @wright_worldwide_2004, Table \@ref(tab:Trait)]: leaf mass per area (i.e. fresh weight divided by the leaf area, LMA, $g.cm^{-2}$), leaf dry matter content (i.e. leaf dry weight divided by the fresh weight, LDMC, $g.g^{-1}$), leaf thickness (LT, $\mu m$), leaf area (LA, $cm^2$), and leaf chlorophyll content (CC, $g.cm^{-2}$). The latter three traits were assessed on fresh leaves.  The petiole was removed for all trait measurements. Leaf fresh weight was measured with an analytical balance (Denver instruments, New York, USA), leaf thickness with a micrometer (Motionics, Austin, USA), leaf area was quantified using the ImageJ software [@Schneider2012] on scanned images of fresh leaves, and leaf chlorophyll content was determined with a SPAD-502  instrument (Konica-Minolta, Osaka, Japan) converted with an allometric model [@Coste2010]. Leaf thickness and chlorophyll content measurements were repeated 3 times per leaf to account for intra-leaf variation. Leaves were then vouchered and oven-dried for at least 48 hours at 80°C before measurement of dry weight.

## Phenotype descriptors

The topographic wetness index (TWI) was selected among several abiotic descriptors (supplementary material S1) as a proxy of water accumulation. Waterlogging and topography have been highlighted as crucial for forest dynamics [@ferry2010higher] and species habitat preference [@Allie2015] at the Paracou study site. TWI was derived from a 1-m resolution digital elevation model using SAGA-GIS [@Conrad2015] based on a LiDAR campaign done in 2015. In order to distinguish inter- from intraspecific abiotic effects, we split  TWI into $TWI_s$ and $TWI_{i|s}$. $TWI_s$ was the species $s$ mean value across all the individual within the species, and represented interspecific abiotic effect. $TWI_{i|s}$ was the individual $i$ value knowing the species $s$ mean value ($TWI_{i|s} = TWI_i -TWI_s$), and represented intraspecific abiotic effect.

The diameter at breast height (DBH, cm) was chosen to control for tree size [@OBrien1995; @Zhang2004], and access to light (evidenced with Dawkins index, supplementary material S2). DBH values of sampled individuals were retrieved from the 2017 inventory of the Paracou permanent plots (supplementary material S3).

## Analysis

We first explored covariation in leaf functional traits across individuals using within-species principal component analyses [wPCA, @DOLEDEC1994], to unravel covariation across traits independently of among-species differences.

Then we used Bayesian modelling to test the role of topography on leaf functional trait variation among and within species, controlling for diameter effect, which effect on intraspecific trait values through ontogeny and access to light has been already shown [@roggy_links_2005; @Sciences2009; @Spasojevic2014]. Individual ($i$) leaf trait ($t$) $Trait_{t,s,i}$ for species $s \in [1;S]$ is explained by its $DBH_i$, using an equation with a Michaelis Menten form (more likely than a linear form, Supplementary Material S4), in interaction with an additive linear form of abiotic inter- ($TWI_s$) and intraspecific ($TWI_{i|s}$) effects:

$$Trait_{t,s,i} \sim \mathcal N(\frac{DBH_i}{\beta_{DBH_{t,s}}+DBH_i}.(\alpha_{t,s}+\beta_{TWI_t}.TWI_s+\gamma_{TWI_{t,s}}.TWI_{i|s}),\sigma_t^2)$$
$$\alpha_{t,s} \sim \mathcal N(\alpha_t,\sigma_{\alpha}^2)$$

where $\alpha_s$ represents the maximum trait value expected at maximum DBH in species $s$ and $\beta_{DBH_s}$ is the DBH of species $s$ for which the trait reaches half of $\alpha_s$. $\alpha_s$ is integrated as a random effect centered on $\alpha$, the mean maximum trait value expected among species, of variance $\sigma_{\alpha}^2$. Maximum trait value is also modulated by $\beta_{TWI}$ and $\gamma_{TWI_{s}}$ the slope of TWI effects on trait among species and within the species $s$, respectively. Contrary to other traits, LA showed a decrease with increasing DBH so we used $\frac{1}{LA}$ instead of LA to allow for a positive covariation with DBH.

Traits, DBH, $TWI_s$ and $TWI_{i|s}$ were all normalised to ease model inference and enable comparison of among traits and among covariates. A Bayesian method was used to build models. We inferred parameters for each leaf trait t and each genus separately, resulting in 10 models, using stan language [@Carpenter2017] and rstan package [@Team2018] in the R environment [@RCoreTeam2018].

# Results

*Eschweilera* species had on average larger, but thinner leaves with a higher dry matter content than *Symphonia* species (Table \@ref(tab:Trait)). Almost seventy percent of the within species variance in leaf trait values was explained by the first plane of the within-species Principal Component Analysis (wPCA, Fig. \@ref(fig:PCA)). Globally, the first plane highlighted the covariation of the five leaf traits on the first axis bearing 46% of the total trait variation. This axis opposed large leaves (i.e. high LA) to high values for all other leaf traits: LT, LDMC, CC and LMA.

Taking into account the strong and expected effect of tree diameter (DBH, Table \@ref(tab:parameters) and supplementary material S5), the parameterized models showed both inter- and intraspecific effect of topography (TWI) on leaf functional trait values, validating our hypothesis. The fitted models had substantial residual variations with $\sigma_t^2$ around 0.8 (Table \@ref(tab:parameters)). TWI had a positive effect on LA and a negative effect on LMA, LDMC and CC  both among and within *Symphonia* species, i.e. *Symphonia* trees in wetter habitat had larger and lighter leaves with less chlorophyll than those in drier habitat within *S. globulifera* and within *S. sp1* and between *S. globulifera* and *S. sp1* (Fig. \@ref(fig:TWInNCI) and Table \@ref(tab:parameters)). TWI had a negative effect on LT both among and within *Eschweilera* species; whereas TWI had a positive effect on CC within *Eschweilera* species but not among species. Moreover, whereas TWI had a positive effect on LA and a negative effect on LT among *Eschweilera* species, only *E. decolorans* showed a negative response of individual LMA and LDMC to TWI within species. In a nutshell, five out of ten models highlighted positive correlation of inter- and intraspecific effects of TWI on leaf functional traits, while two showed only interspecific effect of TWI and two showed only intraspecific effect of TWI for all or one species. 

# Discussion

Theories on the coexistence of species in species-rich communities have generally disregarded within-species variation despite obvious variability in genotypes, phenotypes, and performance. Here, we show that individual leaf traits covary from acquisitive to conservative strategy within species, mirroring what has been previously observed among species and communities [@Bruelheide2018; @wright_worldwide_2004]. Moreover, increasing topographic position resulted in a shift of leaf traits from acquisitive to conservative strategies both across and within species. Our results suggest that intraspecific trait variability widens species’ niches and, if supported by fitness homeostasis, may promote species coexistence.

## Leaf traits covariation within species follows the leaf economics spectrum

All sampled leaf functional traits show substantial variation among  individuals within species along the leaf economics spectrum [@wright_worldwide_2004], suggesting a conservation of functional strategies previously acknowledged among-species level [@wright_worldwide_2004] to the within-species level. The first wPCA axis revealed a covariation of thick (LT) leaves rich in dry matter (LDMC) and chlorophyll content (CC) with high LMA in opposition to large (LA) leaves within species. LMA, LDMC, and LT covariation across species has already been linked to a trade-off between resource acquisition by photosynthesis and investment in leaf defense and durability [@baraloto_decoupled_2010; @Evans2001a; @hodgson_is_2011; @Vile2005]. Moreover, LA was opposed to LMA, LDMC, and LT within species. The functional role of LA remains a subject of debate [@Nicotra2011] especially regarding its contribution to common leaf functional trade-offs between species [@Ackerly2002]. Despite LA debated role between species, our results suggest that within species LA contribute to higher leaf acquisition strategy through photosynthesis. Finally, CC was positively correlated with LDMC and to a lesser extent with LMA and LT in our study within species, as already highlighted by @baraloto_decoupled_2010 among species, but these relationships have so far remained unexplained.

## Intraspecific trait variability widens species niche

Our results suggest that the response of trait values to abiotic environment observed among species at the community level is conserved within species. Topographic wetness index (TWI) characterises topographic position where water accumulates. Thus, TWI represent markers of habitat differentiation with wetter habitats associated with more luminous environment with increased fertility and faster demographic turnover (i.e. tree recruitment, growth, and mortality) in seasonally flooded bottomlands as opposed to nutrient-poor habitats with slower turnover at higher topographic position [@Allie2015; @ferry2010higher]. In our study, TWI had a negative effect on LDMC and LMA, and a positive effect on LA both among and within species, e.g. sites with higher water availability were associated both with trees and species that had larger leaves with lower mass and dry matter content. 

Consequently, higher topographic positioning (i.e. decreasing TWI) resulted in a shift from acquisitive to conservative functional strategy both within and among species in our study. Similar results have been found among species, with species showing more conservative strategies when growing in drier or less fertile habitats: (i) on higher topographic position, e.g. ridge-tops [@kraft_functional_2008; @Mendez-Toribio2017], (ii) with decreasing climatic water availability [@Gotsch2010a], and (iii) with decreased precipitation and phosphorous (Cunningham et al. 2007). Conversely, the increased fertility and water availability and faster turnover of lower elevation habitats can explain that individuals growing there show a more acquisitive strategy [@hodgson_is_2011].

In conclusion, the among-species relationship between functional trait variance and the abiotic environment heterogeneity is mirrored within-species, where we find  the same signals between intraspecific trait variation and the abiotic environment. Thus, intraspecific trait variability is widening species’ niches allowing individuals to grow in environmental conditions where their species average functional trait values would not have allowed them to survive [@Violle2012]. If species niche widening is combined with fitness homeostasis [@Richards2006], intraspecific trait variation may thus contribute to species coexistence. Further studies on the effect of intraspecific trait variability on individual fitness are needed to correctly assess its effect on species coexistence [@Turcotte2016]. In addition to the need of studying jointly species niche and individual fitness, our results on intraspecific trait variation further highlight the importance of investigating the relative contributions of genetically determined local adaptation and phenotypic plasticity  of individuals to specific environmental conditions [@BenitoGarzon2011]. Ongoing genetic data production on the studied species and individuals will  allow us to explore this question in future contributions.

# Data accessibility

Functional traits data have been submitted to TRY initiative [@Kattge2011] under the name ParacouITV. DBH, TWI and spatial positions of individuals were extracted from the Paracou Station database, for which access is modulated by the scientific director of the station (https://paracou.cirad.fr).

\newpage 

# References

<div id="refs"></div>

\newpage 

## Table captions

__Table \@ref(tab:headcount)__ Number of individuals sampled per genus and species.

__Table \@ref(tab:Trait)__ Functional traits measured, with trait unit, abbreviation, and mean and standard error per species.

__Table \@ref(tab:parameters)__ Model estimated parameters for genera, species and traits. $\alpha$ is the species intercept, whereas $\beta_{DBH}$ is the value of DBH for which the trait accounts for half of its plateau value and $\beta_{TWI}$ and $\gamma_{TWI}$ are the slope of TWI inter- and intraspecific effects, respectively. See table \@ref(tab:Trait) for abbreviation of traits.

## Figure captions

__Figure \@ref(fig:PCA)__ Within-species Principal Component Analysis (wPCA) of leaf traits across individuals  in five Neotropical tree species. Circle colors indicate the species, whereas size of circles indicates individual diameter at breast height. See table \@ref(tab:Trait) for abbreviation of traits.

__Figure \@ref(fig:TWInNCI)__ Inter- and intraspecific effect of topographic wetness index (TWI) on each trait for both *Symphonia* and *Eschweilera.* The effect of TWI was estimated as the posterior distribution of the slope parameters $\beta$ and $\gamma$ representing respectively inter- and intraspecific effects (see Materials and Methods) using Bayesian inference. Circles represent the mean estimate, thick lines the 50% confidence interval, and thin lines the 95% confidence interval, and colour the corresponding genus for the interspecific effect or species for the intraspecific effect. See table \@ref(tab:Trait) for abbreviation of traits.

\newpage 

<!-- tables -->

```{r headcount}
Individuals %>% 
  mutate(Species = gsub("_", " ", Species)) %>% 
  group_by(Genus, Species) %>% 
  summarise(N = n(), 
            TWI = paste0(round(mean(TWI), 2), " [",
                         round(quantile(TWI, 0.05), 2), "- ",
                         round(quantile(TWI, 0.95), 2), "]")
  ) %>% 
  arrange(desc(N)) %>% 
  kable(caption = "Number of individuals sampled per genus and species.",
        escape = F, format = "pandoc") %>% 
  kable_styling(full_width = F)
```

```{r Trait}
if(knitr:::is_latex_output()){   
  Individuals0 %>% 
  group_by(Species) %>% 
  filter(n() > 20) %>% 
  ungroup() %>% 
  mutate(LMA = (1/SLA)*10^4) %>% 
  select(Species, LMA, LDMC, LT, LA, CC) %>% 
  reshape2::melt(id.var = "Species", variable.name = "trait") %>% 
  filter(!is.na(value)) %>% 
  group_by(Species, trait) %>% 
  summarise(Value = paste0("$",round(mean(value), 2), "$\n$\\pm",
                           round(sd(value), 2), "$"))  %>% 
  reshape2::dcast(trait ~ Species) %>% 
  mutate(Trait = c("Leaf Mass per Area", "Leaf Dry Matter Content",
                   "Leaf Thickness", "Leaf Area", "Chlorophyll Content")) %>% 
  mutate(Unit = c("$g.m^{-2}$", "$g.g^{-1}$", "$\\mu m$", "$cm^2$", "$g.cm^{-2}$")) %>% 
  mutate_all(linebreak) %>% 
  select(Trait, Unit, trait, globulifera, sp.1, sagotiana, coriacea, decolorans) %>% 
  kable(col.names = c("Trait", "Unit", "Abbreviation", "$S. globulifera$", "$S. sp.1$",
                      "$E. sagotiana$", "$E. coriacea$", "$E. decolorans$"), 
        caption = "Functional traits measured, with trait unit, abbreviation, and mean and standard error per species.",
        escape = F, format = "pandoc") %>% 
  kable_styling(full_width = T)
} else {  
  
  Individuals0 %>% 
  group_by(Species) %>% 
  filter(n() > 20) %>% 
  ungroup() %>% 
  mutate(LMA = (1/SLA)*10^4) %>% 
  select(Species, LMA, LDMC, LT, LA, CC) %>% 
  reshape2::melt(id.var = "Species", variable.name = "trait") %>% 
  filter(!is.na(value)) %>% 
  group_by(Species, trait) %>% 
  summarise(Value = paste0("$",round(mean(value), 2), "\\\\\\pm",
                           round(sd(value), 2), "$"))  %>% 
    reshape2::dcast(trait ~ Species) %>% 
  mutate(Trait = c("Leaf Mass per Area", "Leaf Dry Matter Content",
                   "Leaf Thickness", "Leaf Area", "Chlorophyll Content")) %>% 
  mutate(Unit = c("$g.m^{-2}$", "$g.g^{-1}$", "$\\mu m$", "$cm^2$", "$g.cm^{-2}$")) %>% 
  select(Trait, Unit, trait, globulifera, sp.1, sagotiana, coriacea, decolorans) %>% 
  kable(col.names = c("Trait", "Unit", "Abbreviation", "$S. globulifera$", "$S. sp.1$",
                      "$E. sagotiana$", "$E. coriacea$", "$E. decolorans$"), 
        caption = "Functional traits measured, with trait unit, abbreviation, and mean and standard error per species.",
        escape = F, format = "pandoc") %>% 
  kable_styling(full_width = T)
}
```

```{r parameters}
lapply(fits, function(fit)
  broom::tidyMCMC(fit, pars = c("alpha", "betaDBH", 
                                "betaTWI", "gammaTWI", 
                                "sigma_a", "sigma_p", 
                                "sigma_t", "lp__"), 
                  droppars = NULL, rhat = T)) %>%
  bind_rows(.id = "model") %>% 
  separate(model, c("Genus", "trait"), "-") %>% 
  mutate(SpeciesNum = gsub("([[:alpha:]])", "", term)) %>% 
  dplyr::select(Genus, trait, term, SpeciesNum, estimate, std.error, rhat) %>% 
  mutate(SpeciesNum = gsub("([[:punct:]])", "", SpeciesNum)) %>% 
  mutate(SpeciesNum = as.numeric(SpeciesNum)) %>% 
  left_join(species) %>% 
  mutate(term = gsub("([[:digit:]])", "", term)) %>% 
  mutate(term = gsub("([[:punct:]])", "", term)) %>% 
  mutate(term = recode_factor(term, 
                              alpha = "$\\alpha$",
                              betaDBH = "$\\beta_{DBH}$", 
                              betaTWI = "$\\beta_{TWI}$", 
                              gammaTWI = "$\\gamma_{TWI}$",
                              sigmaa = "$\\sigma^2_{Species}$",
                              sigmap = "$\\sigma^2_{Plot}$",
                              sigmat = "$\\sigma^2_{Trait}$",
                              lp = "log-likelihood")) %>% 
  mutate(trait = recode_factor(trait, 
                              LMA = "$LMA$",
                              LDMC = "$LDMC$", 
                              LT = "$LT$", 
                              invLA = "$\\frac{1}{LA}$",
                              CC = "$CC$")) %>% 
  mutate(SpeciesLong = ifelse(is.na(SpeciesLong), "All", SpeciesLong)) %>% 
  reshape2::dcast(Genus + term + SpeciesLong ~ trait, value.var = "estimate") %>% 
  rename(Parameter = term, Species = SpeciesLong) %>% 
  kable(caption = "Model estimated parameters for genera, species and traits. $\\alpha$ is the species intercept, whereas $\\beta_{DBH}$ is the value of DBH for which the trait accounts for half of its plateau value and $\\beta_{TWI}$ and $\\gamma_{TWI}$ are the slope of TWI inter- and intraspecific effects, respectively. See table \\@ref(tab:Trait) for abbreviation of traits.",
        escape = F, digits = 3, format = "pandoc") %>% 
  kable_styling(full_width = F) %>% 
  column_spec(1, bold = T) %>%
  collapse_rows(columns = 1:2)
```

<!-- figures -->

\newpage 

```{r PCA, fig.cap="Within-species Principal Component Analysis (wPCA) of leaf traits across individuals  in five Neotropical tree species. Circle colors indicate the species, whereas size of circles indicates individual diameter at breast height. See table \\@ref(tab:Trait) for abbreviation of traits.", eval=T}
data <- drop_na(Individuals, LMA, LDMC, LT, LA, CC)
wPCA <- ade4::withinpca(select(data, LMA, LDMC, LT, LA, CC), as.factor(data$Species), 
                  scannf = F, nf = 5)
class(wPCA) <- c("pca", "dudi")
# ind
ind.coord <- wPCA$li
dataind <- wPCA$tab
dataind <- t(apply(dataind, 1, function(x){x*wPCA$norm} ))
dataind <- t(apply(dataind, 1, function(x){x+wPCA$cent}))
eigenvalues <- dataind[1:ncol(ind.coord)]
pca.center <- rep(0, ncol(dataind))
pca.scale <- rep(1, ncol(dataind))
getdistance <- function(ind_row, center, scale){
  return(sum(((ind_row-center)/scale)^2))
}
d2 <- apply(dataind, 1,getdistance, pca.center, pca.scale)
cos2 <- function(ind.coord, d2){return(ind.coord^2/d2)}
ind.cos2 <- apply(ind.coord, 2, cos2, d2)
contrib <- function(ind.coord, eigenvalues, n.ind){
  100*(1/n.ind)*(ind.coord^2/eigenvalues)
}
ind.contrib <- t(apply(ind.coord, 1, contrib,  eigenvalues, nrow(ind.coord)))
colnames(ind.coord) <- colnames(ind.cos2) <-
  colnames(ind.contrib) <- paste0("Dim.", 1:ncol(ind.coord)) 
rnames <- rownames(ind.coord)
if(is.null(rnames)) rnames <- as.character(1:nrow(ind.coord))
rownames(ind.coord) <- rownames(ind.cos2) <- rownames(ind.contrib) <- rnames
pca.ind = list(coord = ind.coord,  cos2 = ind.cos2, contrib = ind.contrib)
# ind/var prep
var <- factoextra::facto_summarize(wPCA, element = "var", 
                                   result = c("coord", "contrib", "cos2"), axes = 1:2)
colnames(var)[2:3] <-  c("x", "y")
ind <- data.frame(pca.ind$coord[, 1:2, drop=FALSE])
colnames(ind)<- c("x", "y")
ind <- cbind(ind, data)
# graph
ggplot(ind, aes(x, y, colour = SpeciesLong)) +
  coord_equal() +
  geom_hline(aes(yintercept = 0), col = 'black', linetype = "dotted") +
  geom_vline(aes(xintercept = 0), col = 'black', linetype = "dotted") +
  geom_point(aes(size = DBH), alpha = 0.5) +
  geom_segment(data = var, aes(x = 0, y = 0, xend = (x*5),
                               yend = (y*5)), arrow = arrow(length = unit(1/2, "picas")),
               color = "black") +
  ggrepel::geom_text_repel(data = var, aes(x = x*5, y = y*5, label = name),
                           box.padding = 0.8, segment.alpha = 0, color = "black") +
  scale_colour_brewer("", palette = "Set1") +
  xlab(paste0("within PCA 1 (", round(wPCA$eig/sum(wPCA$eig)*100)[1], "%)")) +
  ylab(paste0("within PCA 2 (", round(wPCA$eig/sum(wPCA$eig)*100)[2], "%)"))
```

\newpage 

```{r TWInNCI, fig.cap="Inter- and intraspecific effect of topographic wetness index (TWI) on each trait for both *Symphonia* and *Eschweilera.* The effect of TWI was estimated as the posterior distribution of the slope parameters $\\beta$ and $\\gamma$ representing respectively inter- and intraspecific effects (see Materials and Methods) using Bayesian inference. Circles represent the mean estimate, thick lines the 50% confidence interval, and thin lines the 95% confidence interval, and colour the corresponding genus for the interspecific effect or species for the intraspecific effect. See table \\@ref(tab:Trait) for abbreviation of traits."}
lapply(fits, function(fit) 
  mcmc_intervals_data(as.array(fit,
                               pars = c("betaTWI", "gammaTWI")),
                      prob_outer = 0.8)) %>% 
  bind_rows(.id = "model") %>% 
  separate(model, c("Genus", "trait"), "-") %>%
  mutate(SpeciesNum = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(SpeciesNum = gsub("([[:alpha:]])", "", SpeciesNum)) %>% 
  mutate(variable = ifelse(grepl("TWI", parameter), "TWI", "NCI")) %>% 
  mutate(parameter = ifelse(grepl("beta", parameter), "beta", "gamma")) %>% 
  mutate(SpeciesNum = as.numeric(SpeciesNum)) %>% 
  left_join(species) %>% 
  mutate(Genus = ifelse(is.na(Genus), Genus, Genus),
         Species = ifelse(is.na(Species), Genus, Species),
         SpeciesLong = ifelse(is.na(SpeciesLong), Genus, SpeciesLong)) %>% 
  mutate(trait = recode_factor(trait, invLA = "frac(1, LA)")) %>% 
  mutate(SpeciesLong = fct_relevel(SpeciesLong, "E decolorans", "E sagotiana",
                                   "E coriacea", "Eschweilera",
                                   "S globulifera", "S sp.1", "Symphonia")) %>% 
  ggplot(aes(x = SpeciesLong,
             xend = SpeciesLong,
             color = SpeciesLong, fill = SpeciesLong)) +
  geom_hline(yintercept = 0, color = "gray90", size = 1) +
  geom_hline(yintercept = c(-0.25, 0.25), 
             color = "gray90", size = 0.5, lty = "dashed") +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  coord_flip() +
  facet_grid(trait ~ Genus, labeller = label_parsed, 
              scales = "free_y") +
  xaxis_title(F) +
  ylab("TWI effect") +
  scale_y_continuous(trans="S_sqrt") +
  scale_color_discrete(guide = "none") +
  scale_fill_discrete(guide = "none")
```
