```{r setup_functional, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(parallel)
library(Biostrings)
library(tidyverse)
library(kableExtra)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "../../../data/Eschweilera_Genomic/Oscar_Transcriptome/"
```

# Fuctional region selection

@Vargas2019 used 8 individuals of *Lecythidaceae* including 2 *Eschweilera coriacea* and 1 *Eschweilera sagotiana* for transcriptomic analysis. RNA reads were assembled with `Trinity` into transcripts. We will proceed with following steps:

1. __*E. coriacea* alignment on *E. sagotiana*__: `BLAT` will be used to reciprocally align *E. coriacea* and *E. sagotiana* individuals, and we will keep transcript reciprocally matching only one transcripts from the other species to avoid paralogs and have robust targets among species.
1. __Open Reading Frames (ORFs) detection__: `transdecoder` will be used to detect Open Reading Frames (ORFs) in previously selected scaffolds within *E. sagotiana*.
1. __Selection__: Selection will be operated depending on the number, size, coverage, ORF and annotation of transcripts.

## *E. coriacea* alignment on *E. sagotiana*

After alignement we obtained the following statistics :

* 2 954 134 match
* 19 973 reciprocal single-match
* 9 474 *E. sagotiana* transcripts with reciprocal single-match
* 5 637 *E. sagotiana* transcripts with reciprocal single-match  with *E. coriacea* **1**
* 3 837 *E. sagotiana* transcripts with reciprocal single-match  with *E. coriacea* **2**

We will use the 9 474  *E. sagotiana* transcripts with reciprocal single-match on one or both *E. coriacea*. We obtained relativelly small transcripts (mean length < 250 bp) but with high coverage (mean close to 100%, Fig. \@ref(fig:alignmentFilteringFig)).

```{bash trscAlign, eval=F, echo=T}
#!/bin/bash
#SBATCH --time=36:00:00
#SBATCH -J trscAlign
#SBATCH -o trscAlign_output.out
#SBATCH -e trscAlign_error.out
#SBATCH --mem=20G
#SBATCH --cpus-per-task=1
#SBATCH --mail-type=BEGIN,END,FAIL

# Environment
module purge
module load bioinfo/blatSuite.36

# variables
cd ../../../data/Eschweilera_Genomic/Oscar_Transcriptome/
Escor1=Escoria1L.trinity.Trinity.fasta
Escor2=Escoria2L.trinity.Trinity.fasta
Essagot=EssagoL.trinity.Trinity.fasta
# blat=~/Tools/blatSrc/bin/blat

# alignment
blat $Escor1 $Essagot EssagotonEscor1.psl
blat $Escor2 $Essagot EssagotonEscor2.psl
blat $Essagot $Escor1 Escor1onEssagot.psl
blat $Essagot $Escor2 Escor2onEssagot.psl
blat $Escor1 $Escor2 Escor2onEscor1.psl
blat $Escor2 $Escor1 Escor1onEscor2.psl
mkdir Alignment
mv *.psl Alignment
```

```{r alignment}
alignment <- lapply(list.files(file.path(path, "Alignment"), full.names = T), 
                    function(file)
  read_tsv(file.path(path, file), skip = 4, 
           col_names = c("matches", "misMatches", "repMatches", "nCount", 
                         "qNumInsert", "qBaseInsert",
                         "tNumInsert", "tBaseInsert", "strand", 
                         "qName", "qSize", "qStart", "qEnd", 
                         "tName", "tSize", "tStart", "tEnd", 
                         "blockCount", "blockSizes", "qStarts", "tStarts")) %>% # psl3 generics
  mutate(query = file)) %>% 
  bind_rows() %>% 
  mutate(query = gsub("../../../data/Eschweilera_Genomic/Oscar_Transcriptome//Alignment/", 
                      "", query)) %>% 
  mutate(query = gsub(".psl", "", query)) %>% 
  separate(query, c("query", "reference"), "on") %>% 
  mutate(alnSize = matches + misMatches + repMatches) %>% 
  select(query, reference, alnSize,
         qName, qSize, qStart, qEnd,
         tName, tSize, tStart, tEnd)
```

```{r alignmentFiltering}
# alignmentFiltered <- alignment %>% 
#   mutate(sagotName = ifelse(reference == "Essagot", tName, qName)) %>% 
#   group_by(query, reference, qName) %>% 
#   mutate(Nq = n()) %>% 
#   group_by(query, reference, tName) %>% 
#   mutate(Nt = n()) %>% 
#   filter(Nq == 1, Nt == 1) %>% 
#   group_by(query, sagotName) %>% 
#   summarise(Nmatch = n()) %>%
#   spread(query, Nmatch) %>% 
#   filter((Escor1 == 1 & Essagot1 == 1) | (Escor2 == 1 & Essagot2 == 1))
alignmentFiltered <- alignment %>% 
  group_by(query, reference, qName) %>% 
  mutate(Nq = n()) %>% 
  group_by(query, reference, tName) %>% 
  mutate(Nt = n()) %>% 
  filter(Nq == 1, Nt == 1) %>% 
  ungroup() %>% 
  filter(qName %in% tName & tName %in% qName) %>% 
  filter(reference == "Essagot") 
write_tsv(data.frame(trsc = unique(alignmentFiltered$tName)),
          path = file.path(path, "Essago.filtered.list"),
          col_names = F)
```

```{r alignmentFilteringFig, fig.cap="E. sagotiana transcripts distribution by size and alignment coverage."}
alignmentFiltered %>%
  group_by(tName, tSize) %>%
  summarise(tStart = min(tStart), tEnd = max(tEnd)) %>% 
  mutate(alnSize = tEnd - tStart) %>% 
  mutate(coverage = alnSize/tSize*100) %>% 
  ggplot(aes(tSize, coverage)) +
  geom_point() +
  geom_hline(yintercept = 50, col = "red") +
  geom_vline(xintercept = 500, col = "red") +
  xlab("E. sagotiana transcript size (bp)") +
  ylab("Alignement coverage on transcript (%)")
```

```{bash alignmentFilteringFasta, eval=F, echo=T}
cd ../../../data/Eschweilera_Genomic/Oscar_Transcriptome/
ref=EssagoL.trinity.Trinity.fasta
ids=Essago.filtered.list
~/Tools/SeqFilter/bin/SeqFilter $ref --ids $ids -o Essago.filtered.fasta
```

## Open Reading Frames (ORFs) detection

We detected 1530 ORFs on 1530 unique transcripts with `transdecoder` in previously selected transcripts. Only, 469 included a start codon (Fig. \@ref(fig:orfFilterGraph)), which was too low and not retained as a filtering criterion.

```{bash ORFdetectionCluster, eval=F, echo=T}
#!/bin/bash
#SBATCH --time=48:00:00
#SBATCH -J transdecoder
#SBATCH -o transdecoder.out
#SBATCH -e transdecoder.out
#SBATCH --mem=20G
#SBATCH --cpus-per-task=4
#SBATCH --mail-type=BEGIN,END,FAIL

# Environment
module purge
module load bioinfo/TransDecoder-v5.0.2
module load bioinfo/hmmer-3.1b2
module load bioinfo/ncbi-blast-2.7.1+

# script
mkdir TransDecoder
cp Essago.filtered.fasta TransDecoder/
cd TransDecoder
TransDecoder.LongOrfs -t Essago.filtered.fasta
hmmscan --cpu 4 --domtblout pfam.domtblout ../Pfam/Pfam-A.hmm Essago.filtered.fasta.transdecoder_dir/longest_orfs.pep
blastp -query Essago.filtered.fasta.transdecoder_dir/longest_orfs.pep -db /bank/ebi/uniprot/current/fasta/uniprot_sprot.fasta -max_target_seqs 1 -outfmt 6 -evalue 1e-5 -num_threads 10 > blastp.outfmt6
# TransDecoder.Predict -t Essago.filtered.fasta --retain_pfam_hits pfam.domtblout --retain_blastp_hits blastp.outfmt6
TransDecoder.Predict -t Essago.filtered.fasta
rm Essago.filtered.fasta
awk 'BEGIN{RS=">";OFS="\t"}NR>1{print $1,$4,$5,$6,$7,$8$9$10$11$12}' Essago.filtered.fasta.transdecoder.pep > Essago.filtered.fasta.transdecoder.pep.tsv
```

```{r orf}
orf <- read_tsv(file.path(path, "TransDecoder", "Essago.filtered.fasta.transdecoder.pep.tsv"), 
         col_names = c("orf", "type", "length", "strandScore", 
                      "transcriptPositionStrand", "peptide")) %>% 
  mutate(type = gsub("type:", "", type)) %>% 
  mutate(length = gsub("len:", "", length)) %>% 
  separate(strandScore, c("strand", "score"), "=") %>% 
  mutate(strand = gsub("(", "", strand, fixed = T)) %>% 
  mutate(strand = gsub("),score", "", strand)) %>% 
  separate(transcriptPositionStrand, c("transcript", "position"), ":") %>% 
  separate(position, c("trscStart", "trscStop"), "-") %>% 
  mutate(trscStop = gsub("[[:punct:]]", "", trscStop)) %>% 
  mutate(left = substr(peptide, 1, 1)) %>% 
  mutate(right = substr(peptide, nchar(peptide), nchar(peptide))) %>% 
  group_by(transcript) %>% 
  arrange(transcript, trscStart) %>% 
  mutate(overlap = ifelse(trscStart < lag(trscStop), 1, 0)) %>% 
  mutate(overlap = ifelse(is.na(overlap), 0, overlap)) %>% 
  ungroup()
write_tsv(data.frame(trsc = unique(orf$transcript)),
          path = file.path(path, "Essago.filteredORF.list"),
            col_names = F)
```

```{r orfFilterGraph, fig.cap="Open Reading Frames left and right peptides."}
orf %>% 
  select(orf, strand, left, right) %>% 
  reshape2::melt(id.vars = c("orf", "strand"),
                 variable.name = "side",
                 value.name = "peptide") %>% 
  ggplot(aes(peptide, fill = (peptide %in% c("M", "*")))) +
  geom_bar() +
  facet_grid(strand~side, labeller = "label_both") +
  scale_fill_discrete(guide = F)
```

```{bash orfFilteringFasta, eval=F, echo=T}
cd ../../../data/Eschweilera_Genomic/Oscar_Transcriptome/
ref=Essago.filtered.fasta
ids=Essago.filteredORF.list
~/Tools/SeqFilter/bin/SeqFilter $ref --ids $ids -o Essago.filteredORF.fasta
```

## Selection

We thus selected all transcripts to reach required number and length of sequences.

```{r functional}
targets <- readDNAStringSet(file.path(path, "Essago.filteredORF.fasta"))
data.frame(n = length(targets),
           width = sum(width(targets))/10^6,
           Nperc = sum(letterFrequency(targets, "N"))/sum(width(targets))) %>% 
    kable(caption = "Selected funcional targets.",
        col.names = c("N", "Width (Mbp)", "Mask (%N)"),
        format.args = list(big.mark = " "))
```

```{bash arborSciencefunctional, eval=F, echo=T}
cd ../../../data/Eschweilera_Genomic/Oscar_Transcriptome/
cp Essago.filteredORF.fasta functional.targets.20190503.fasta
```