```{r setup_functional, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(parallel)
library(Biostrings)
library(tidyverse)
library(kableExtra)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "../../../data/Eschweilera_Genomic/Oscar_Transcriptome/"
```

# Fuctional region selection

@Vargas2019 used 8 individuals of *Lecythidaceae* including 2 *Eschweilera coriacea* and 1 *Eschweilera sagotiana* for transcriptomic analysis. RNA reads were assembled with `Trinity` into transcripts. We will proceed with following steps:

1. __*E. coriacea* alignment on *E. sagotiana*__: `BLAT` will be used to align *E. coriacea* transcripts alignment on *E. sagotiana*, and we will keep transcript matching only one transcripts from the other species to avoid paralogs and have robust targets among species.
1. __Open Reading Frames (ORFs) detection__: `transdecoder` will be used to detect Open Reading Frames (ORFs) in previously selected scaffolds within *E. sagotiana*. ORFs with a starting codon (Methyonin M) and genes without overlapping genes will be kept.
1. __Annotation ?__:
1. __Candidates genes ?__:
1. __Selection__:

## *E. coriacea* alignment on *E. sagotiana*

```{bash trscAlign, eval=F, echo=T}
#!/bin/bash
#SBATCH --time=36:00:00
#SBATCH -J trscAlign
#SBATCH -o trscAlign_output.out
#SBATCH -e trscAlign_error.out
#SBATCH --mem=20G
#SBATCH --cpus-per-task=1
#SBATCH --mail-type=BEGIN,END,FAIL

# Environment
module purge
module load bioinfo/blatSuite.36

# variables
cd ../../../data/Eschweilera_Genomic/Oscar_Transcriptome/
Escor1=Escoria1L.trinity.Trinity.fasta
Escor2=Escoria2L.trinity.Trinity.fasta
Essagot=EssagoL.trinity.Trinity.fasta
# blat=~/Tools/blatSrc/bin/blat

# alignment
blat $Escor1 $Essagot alignment_Escor1.psl
blat $Escor2 $Essagot alignment_Escor2.psl
```

## Open Reading Frames (ORFs) detection

## Annotation ?

## Candidates genes ?

## Selection
