```{r setup_neutral, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(parallel)
library(Biostrings)
library(tidyverse)
library(kableExtra)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "../../../data/Eschweilera_Genomic/"
```

# Neutral region selection

Myriam *et al, unpublished* used 6 individuals of several species from the *Eschweilera* clade *Parvifolia*. Thomas Folituu is assembling ddRAD-seq reads with `iPyrad`. We will thus use consensus sequences to design targets among species.

1. __Consens sequence alignment__: `BLAT` will be used to align concensus sequences, after deNovo assembly through `iPyrad` from ddRAD-seq, on transcripts from @Vargas2019. We will keep consensus scaffolds matching no transcripts as neutral sequences. *We still risk to select for sequences with paralogs genes ?*
1. __Selection__:

## Consens sequence alignment

## Selection
