```{r setup_denovo, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(parallel)
library(tidyverse)
library(rstan)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = F, cache.lazy = F)
pathCluster <- "~/Remotes/genotoul/work/PhD/data/Eschweilera_Paracou"
path <- "../../../data/Eschweilera_Paracou/"
```

```{r denovoLib}
individuals <- googlesheets::gs_title("Parvicapture") %>% 
  googlesheets::gs_read("Extraction") %>% 
  filter(!is.na(Seq)) %>% 
  dplyr::select(IdGenetic, Genus, Species, Plot, SubPlot, TreeFieldNum) %>% 
  unique() %>% 
  mutate(cat = recode(Species, "wachenheimii" = "other", "grandiflora_form2" = "other",
                      "simiorum" = "outgroup", "collina"  = "other", "persistens" = "outgroup",
                      "squamata" = "other", "pedicellata" = "other", "grandiflora" = "other",
                      "micrantha" = "other", "parviflora" = "other", 
                      "congestiflora" = "outgroup"))
```

# *de novo*

In this chapter, we produced *de novo* snps with `ipyrad`.

## Test 

We failed to produce a correct dataset with the home-made pipeline used for *Symphonia* (`baw-mem` mapping and SNP-call with `GATK` plus others). We consequently used `ipyrad` with restrictive thresholds (clustering across of 0.95 instead of 0.8) and *denovo* assembly to avoid issues of paralogs both in the reads and the original reference. We made a succesful test with 27 samples (Fig. \@ref(fig:genotypetest)).

```{r genotypetest, fig.cap="PCA for SNPs obtained with the `ipyrad` test."}
  read_delim(file.path(path, "Sequences", "denovo_test",
                                "denovo.eigenvec"),
             delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  separate(Sample, paste0("X", 1:7), "[:punct:]", remove = F) %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  dplyr::select(Sample, IdGenetic, Lane, PCA1, PCA2) %>% 
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = cat, 
             label = Sample)) +
  geom_line(aes(group = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  scale_color_discrete("") +
  theme(legend.position = "bottom") +
  ggtitle("Denovo test")
```

## Libraries

To ease *denovo* assembly we filtered reads corresponding to targets by a first mapping wiht `bwa-mem` on the reference with the scripts below.

```{bash raw2filtered, eval=F, echo=T}
#!/bin/bash    

# variable
file=$1
echo "File: $file"
raw=$2
echo "Raw: $raw"
reference=$3
echo "Reference: $reference"

# modules
cd runs/$reference
module load bioinfo/Trimmomatic-0.36
module load bioinfo/bwa-0.7.15
module load bioinfo/picard-2.14.1
module load bioinfo/samtools-1.4
module load bioinfo/bedtools2-2.29.0

# trimming
echo "Step: Trimming"
java -jar $TRIM_HOME/trimmomatic.jar PE \
  ../../$raw/"$file"_R1_001.fastq \
  ../../$raw/"$file"_R2_001.fastq \
  trimmed/paired/"$file"_R1_paired.fq.gz \
  trimmed/unpaired/"$file"_R1_unpaired.fq.gz \
  trimmed/paired/"$file"_R2_paired.fq.gz \
  trimmed/unpaired/"$file"_R2_unpaired.fq.gz \
  ILLUMINACLIP:$TRIM_HOME/adapters/TruSeq3-PE.fa:2:30:10:2:keepBothReads \
  SLIDINGWINDOW:4:20

# mapping
echo "Step: mapping"
bwa mem -M -R "@RG\tID:$file\tSM:$file\tPL:HiSeq4K" \
  -t 1 \
  ../../reference/$reference.fasta \
  trimmed/paired/"$file"_R1_paired.fq.gz 	\
  trimmed/paired/"$file"_R2_paired.fq.gz \
  > mapped/sam/$file.sam
rm trimmed/paired/"$file"_R1_paired.fq.gz trimmed/paired/"$file"_R2_paired.fq.gz
rm trimmed/unpaired/"$file"_R1_unpaired.fq.gz trimmed/unpaired/"$file"_R2_unpaired.fq.gz
java -Xmx4g -jar $PICARD SortSam \
  I=mapped/sam/$file.sam \
  O=mapped/bam/$file.bam \
  SORT_ORDER=coordinate 
samtools index mapped/bam/$file.bam
rm mapped/sam/$file.sam
samtools view -u -f 1 -F 12 \
  mapped/bam/$file.bam > mapped/bam.paired/$file.bam
samtools index mapped/bam.paired/$file.bam
rm mapped/bam/$file.bam mapped/bam/$file.bam.bai
samtools sort -n  \
  mapped/bam.paired/$file.bam > mapped/bam.paired.sorted/$file.bam
rm mapped/bam.paired/$file.bam mapped/bam.paired/$file.bam.bai

# extracting reads
bedtools bamtofastq -i mapped/bam.paired.sorted/$file.bam \
  -fq filtered.reads/"$file"_R1_001.fastq -fq2 filtered.reads/"$file"_R2_001.fastq
rm mapped/bam.paired.sorted/$file.bam
```

```{bash denovoLibSh, eval=F, echo=T}
raw=../raw
reference=../reference/reference.fasta
mkdir trimmed trimmed/paired trimmed/unpaired
mkdir mapped mapped/sam mapped/bam mapped/bam.paired mapped/bam.paired.sorted
mkdir filtered.reads
mkdir out
for lib in $(cat libraries.txt); do echo "bash raw2filtered.sh $lib $raw $reference"; done > runFiltered.sh
sarray -J haplo -o out/%j.out -e out/%j.err \
  -t 10:00:00 --mem=8G --cpus-per-task=2 --mail-type=BEGIN,END,FAIL runFiltered.sh
rm -r trimmed mapped 
rm out/* runFiltered.sh
```

## Reference

But the obtained dataset was too big to be run with `ipyrad` on `genologin` (see discussion on `giter/ipyrad`). So we split the 3730 target sequences from the reference in 38 sub-references to separate the work in 38 jobs.

```{bash, eval=F, echo=T}
mkdir reference
cut ../../reference/reference.fasta.fai -f1 > reference.sequence.list
split -l 100 -d reference.sequence.list reference.sequence_ --additional-suffix=.list
rm reference.sequence.list
module load bioinfo/seqtk-1.2
reference=../../reference/reference.fasta
for list in $(ls *.list); do seqtk subseq $reference $list > ${list%.*}.fasta ; done
module load bioinfo/bwa-0.7.15
for ref in $(ls *.fasta); do bwa index $ref ; done
```


## *de novo*

We used `ipyrad` on the 38 reference parts. We still have issue for one (part 34). 

```{bash, eval=F, echo=T}
mkdir runs
for ref in $(ls reference/*.list); do mkdir runs/$(basename ${ref%.*}) ; done
for ref in $(ls runs); 
  do 
  mkdir runs/$ref/trimmed runs/$ref/trimmed/paired runs/$ref/trimmed/unpaired
  mkdir runs/$ref/mapped runs/$ref/mapped/sam runs/$ref/mapped/bam runs/$ref/mapped/bam.paired runs/$ref/mapped/bam.paired.sorted
  mkdir runs/$ref/filtered.reads
done
for ref in $(ls runs); do for lib in $(cat libraries.txt); do echo "bash raw2filtered2.sh $lib filtered.reads $ref"; done > runFiltered_${ref//[^0-9]/} ; done
for runs in $(ls runFiltered_*); do echo "cat haplo${runs//[^0-9]/} ; sarray  --wait -J haplo${runs//[^0-9]/} -o out/haplo${runs//[^0-9]/}.%j.out -e out/haplo${runs//[^0-9]/}.%j.err -t 10:00:00 --mem=4G --cpus-per-task=1 --mail-type=BEGIN,END,FAIL $runs ;"; done > runFiltered_all.sh # we may number outputs in out/ and remove them as soon as they end to avoid an excessive number of files and mail might be removed and kept only in the general sbatch
for ref in $(ls runs); do rm -r runs/$ref/trimmed runs/$ref/mapped ; done
for ref in $(ls runs); do cp params.txt runs/$ref/ ; done
for ref in $(ls runs); do echo "echo denovo${ref//[^0-9]/} ; cd runs/$ref ; module load system/Miniconda3-4.4.10 ; module load bioinfo/ipyrad-0.7.28 ; ipyrad -p params.txt -s 1234567 -c 4 ;"; done > runDenovo.sh
sarray -J denovo2 -o out/denovo2.%j.out -e out/denovo2.%j.err -t 48:00:00 --mem=40G --cpus-per-task=4 --mail-type=BEGIN,END,FAIL runDenovo2.sh ;
for ref in $(ls runs); do printf "echo denovo${ref//[^0-9]/}\ncd runs/$ref\nipyrad -p params.txt -s 1234567 -c 10\nwait\ncd ../../\n\n"; done > runDenovo.sh
mkdir vcfs
for ref in $(ls runs); do awk '{gsub(/locus/, "ref""'"${ref//[^0-9]/}"'""_locus"); print}' runs/$ref/denovo_outfiles/denovo.vcf > vcfs/ref"${ref//[^0-9]/}".vcf ; done
find -name 'ref*' -size 0 -delete # waiting for all files
```

## Raw

We assembled back locally the obtained vcfs with `plink`. Due to too many contigs for `plink` we temporarily renames all contigs as one pseudo-contig, but they information is kept in the SNP name. **We obtained 704 485 SNPs over 286 libraries.**

```{bash rawSH, eval=F, echo=T}
plink=~/Tools/plink_high_contig/plink # too many contigs
mkdir beds
for vcf in $(ls vcfs); 
  do 
  $plink \
  --vcf vcfs/$vcf \
  --const-fid \
  --allow-extra-chr \
  --set-missing-var-ids @_snp# \
  --mind 0.999 \
  --make-bed \
  --out beds/${vcf%.*} ; 
done
mkdir bed00
for bim in $(ls beds/*.bim); do mv $bim "$bim"0 ; done
for bim in $(ls beds/*.bim0); do awk '$1="1"' $bim > ${bim%.*}.bim ; done

mv beds/*00* bed00
ls beds/*.bed | awk '{gsub(/.bed/, ""); print}' > bedfiles.txt
mkdir raw
$plink \
  --bfile bed00/ref00 \
  --allow-extra-chr \
  --merge-list bedfiles.txt \
  --mind 0.993 \
  --make-bed \
  --out raw/raw \
  --missing --het --freqx --pca var-wts
rm -r beds bed00
rm bedfiles.txt
```

```{r scfRename, eval=F, echo=T}
read_tsv(file.path(path, "Sequences", "denovo", "raw", "raw.bim0"), col_names = c("fake", "snp", "posM", "pos", "A1", "A2")) %>% 
  separate(snp, c("ref", "locus", "locusN", "snpN"), sep = "_", remove = F) %>% 
  mutate(scf = paste0(ref, "_", "locus", locusN)) %>% 
  dplyr::select(scf, snp, posM, A1, A2) %>% 
  write_tsv(file.path(path, "Sequences", "denovo", "raw", "raw.bim"), col_names = F)
```

```{r rawPCA, fig.cap="PCA for raw SNPs."}
read_delim(file.path(path, "Sequences", "denovo", "raw", "raw.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  separate(Sample, paste0("X", 1:7), "[:punct:]", remove = F) %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  dplyr::select(Sample, IdGenetic, Lane, PCA1, PCA2) %>% 
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = cat, 
             label = Sample)) +
  geom_line(aes(group = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  scale_color_discrete("") +
  theme(legend.position = "bottom") +
  ggtitle("Raw data (`--mind = 0.993`)")
```
