```{r setup_sequencesTest, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(parallel)
library(tidyverse)
library(rstan)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
pathCluster <- "~/Remotes/genotoul/work/PhD/data/Eschweilera_Paracou"
```


```{r }
googlesheets::gs_title("Parvicapture") %>% 
  googlesheets::gs_read("Extraction") %>% 
  dplyr::select(IdGenetic, Genus, Species, Plot, SubPlot, TreeFieldNum) %>% 
  unique() %>% 
  na.omit() %>% 
  filter(Species %in% c("sagotiana", "coriacea", "decolorans", "persistens", "simiorum")) %>% 
  group_by(Species) %>% 
  sample_n(10, replace = T) %>% 
  ungroup() %>% 
  dplyr::select(IdGenetic) %>% 
  unique() %>% 
  write_tsv(path = file.path(pathCluster, "Sequences", "individuals_test.txt"), col_names = F)

individuals <- googlesheets::gs_title("Parvicapture") %>% 
  googlesheets::gs_read("Extraction") %>% 
  na.omit(Seq) %>% 
  dplyr::select(IdGenetic, Genus, Species, Plot, SubPlot, TreeFieldNum) %>% 
  unique()
a <- read_tsv(file.path(pathCluster, "Sequences", "unmerged", "libraries", "libraries.txt"),
         col_names = "Library") %>% 
  separate(Library, paste0("X", 1:7), "[:punct:]", remove = F, convert = T) %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5, Label = X3) %>% 
  dplyr::select(Library, IdGenetic, Lane, Label)
b <- googlesheets::gs_title("Parvicapture") %>% 
              googlesheets::gs_read("Extraction") %>% 
              filter(!is.na(Seq)) %>% 
              mutate(Pool = c(rep(LETTERS[1:3], each = ceiling(4*96/3))[1:(4*96)], rep("D", 534-4*96))) %>% 
  mutate(Lane = Seq) %>% 
  mutate(Label = 1:nrow(.)) # IGA label
libraries <- left_join(a, b, by = c("IdGenetic", "Lane", "Label"))
```

# Sequences Test

## Haplotyping

```{bash raw2haplo, eval=F, echo=T}
#!/bin/bash    

# variable
file=$1
echo "File: $file"
raw=$2
echo "Reference: $raw"
reference=$3
echo "Reference: $reference"

# modules
module load bioinfo/Trimmomatic-0.36
module load bioinfo/bwa-0.7.15
module load bioinfo/picard-2.14.1
module load bioinfo/samtools-1.4
module load bioinfo/gatk-4.1.2.0

# trimming
echo "Step: Trimming"
java -jar $TRIM_HOME/trimmomatic.jar PE \
  $raw/"$file"_R1_001.fastq.gz \
  $raw/"$file"_R2_001.fastq.gz \
  trimmed/paired/"$file"_R1_paired.fq.gz \
  trimmed/unpaired/"$file"_R1_unpaired.fq.gz \
  trimmed/paired/"$file"_R2_paired.fq.gz \
  trimmed/unpaired/"$file"_R2_unpaired.fq.gz \
  ILLUMINACLIP:$TRIM_HOME/adapters/TruSeq3-PE.fa:2:30:10:2:keepBothReads \
  SLIDINGWINDOW:4:20

# mapping
echo "Step: mapping"
bwa mem -M -R "@RG\tID:$file\tSM:$file\tPL:HiSeq4K" \
  -t 2 \
  $reference \
  trimmed/paired/"$file"_R1_paired.fq.gz 	\
  trimmed/paired/"$file"_R2_paired.fq.gz \
  > mapped/sam/$file.sam
rm trimmed/paired/"$file"_R1_paired.fq.gz trimmed/paired/"$file"_R2_paired.fq.gz
java -Xmx4g -jar $PICARD SortSam \
  I=mapped/sam/$file.sam \
  O=mapped/bam/$file.bam \
  SORT_ORDER=coordinate 
rm mapped/sam/$file.sam
samtools index mapped/bam/$file.bam

# haplotyping
echo "Step: haplotyping"
gatk  --java-options "-Xmx20G" \
  HaplotypeCaller \
  -R $reference \
  -I mapped/bam/$file.bam \
  -O haplotype/$file.g.vcf.gz  \
  -ERC GVCF
rm mapped/bam/$file.bam
```

## Genotyping

```{bash haplo2geno, eval=F, echo=T}
#!/bin/bash    

# variable
file=$1
echo "File: $file"
reference=$2
echo "Reference: $reference"

# modules
module load bioinfo/gatk-4.1.2.0

# GenomicsDB
echo "Step: DB"
gatk --java-options "-Xmx20g -Xms20g" \
  GenomicsDBImport \
  --genomicsdb-workspace-path db/"${file%.*}".DB \
  -L reference.sequences.lists/$file \
  --sample-name-map sample_map.txt \
  --batch-size 10 --consolidate

## genotyping
echo "Step: genotyping"
gatk --java-options "-Xmx20g" \
  GenotypeGVCFs \
  -R $reference \
  -L reference.sequences.lists/$file \
  -V gendb://db/${file%.*}.DB -O genotype/${file%.*}.vcf.gz
rm -r db/${file%.*}.DB
```

## Data production

```{bash parvicapture, eval=F, echo=T}
#!/bin/bash
#SBATCH --time=48:00:00
#SBATCH -J parvi
#SBATCH -o parvi.out
#SBATCH -e parvi.err
#SBATCH --mem=20G
#SBATCH --cpus-per-task=8
#SBATCH --mail-type=BEGIN,END,FAIL

# variable
raw=../raw
echo "Raw: $raw"
reference=../reference/reference.fasta
echo "Reference: $reference"

# modules
module load bioinfo/bcftools-1.8
module load bioinfo/gatk-4.1.2.0
module load bioinfo/plink_high_contig_20190905
module load bioinfo/plink2_high_contig_20190905

# init
echo "Step: init"
mkdir trimmed trimmed/paired trimmed/unpaired
mkdir mapped mapped/sam mapped/bam
mkdir haplotype
ls $raw | sed -e 's/_R[12]_001.fastq.gz//' | sort | uniq > libraries.txt

# haplotyping
echo "Step: haplotyping"
mkdir outHaplo
for lib in $(cat libraries.txt); do echo "bash raw2haplo.sh $lib $raw $reference"; done > runHaplo.sh
sarray --wait -J haplo -o outHaplo/%j.out -e outHaplo/%j.err \
  -t 48:00:00 --mem=10G --cpus-per-task=2 --mail-type=BEGIN,END,FAIL runHaplo.sh
wait
rm -r trimmed mapped

## genotyping
echo "Step: genotyping"
for file in $(ls haplotype/*.g.vcf.gz) ; do echo -e $(basename $file | sed -e 's/.g.vcf.gz//')"\t"$file ; done > sample_map.txt
mkdir reference.sequences.lists
cut $reference.fai -f1 > reference.sequences.lists/reference.sequences.list
cd reference.sequences.lists
split -l 100 -d reference.sequences.list reference.sequences_ --additional-suffix=.list
rm reference.sequences.list
cd ..
mkdir db genotype outGeno
for file in $(ls reference.sequences.lists/); do echo "bash haplo2geno.sh $file $reference"; done > runGeno.sh
sarray --wait -J genotype -o outGeno/%j.out -e outGeno/%j.err -t 48:00:00 \
  --mem=25G --mail-type=BEGIN,END,FAIL runGeno.sh
wait
echo -e '#!/bin/bash\n#SBATCH --time=48:00:00\n#SBATCH -J gather\n#SBATCH -o gather.out\n#SBATCH -e gather.err\n#SBATCH --mem=20G\n#SBATCH --cpus-per-task=1\n#SBATCH --mail-type=BEGIN,END,FAIL\nmodule load bioinfo/picard-2.14.1\njava -Xmx20g -jar $PICARD GatherVcfs \' > gather.sh
for file in $(ls genotype/*.gz)
do
	echo -e '\tI='$file' \' >> gather.sh
done
echo -e '\tO=parvicapture.raw.vcf.gz\n' >> gather.sh
sbatch --wait gather.sh
wait
rm -r reference.sequences.lists gvcf db genotype

## filtering
echo "Step: filtering"
bcftools view --max-alleles 2 --threads 8 parvicapture.raw.vcf.gz | bgzip -c --threads 8 > parvicapture.biallelic.vcf.gz
gatk IndexFeatureFile -F parvicapture.biallelic.vcf.gz
gatk VariantFiltration \
  -V parvicapture.biallelic.vcf.gz \
  --filter-expression "QUAL < 30.0 || QD < 2.0 || FS > 60.0 || SOR > 3.0" \
  --filter-name "FAIL" \
  -O parvicapture.biallelic.annot.vcf.gz
gatk SelectVariants \
  -select-type SNP \
  -V parvicapture.biallelic.annot.vcf.gz \
  --exclude-filtered \
  -O parvicapture.all.biallelic.snp.filtered.vcf.gz
rm parvicapture.biallelic.vcf.gz parvicapture.biallelic.vcf.gz.tbi parvicapture.biallelic.annot.vcf.gz*
gatk IndexFeatureFile -F parvicapture.all.biallelic.snp.filtered.vcf.gz
```

```{bash unmerged, eval=F, echo=T}
folder=unmerged
mkdir $folder
cp raw2haplo.sh haplo2geno.sh parvicapture.sh $folder/
cd $folder
sbatch parvicapture.sh
```

## All

```{bash allSH, eval=F, echo=T}
module load bioinfo/plink_high_contig_20190905
module load bioinfo/plink2_high_contig_20190905
mkdir filtered
plink2 --threads 8 --memory 20000 \
  --vcf parvicapture.all.biallelic.snp.filtered.vcf.gz \
  --allow-extra-chr \
  --mind 0.95 --geno 0.15 \
  --make-bed --out filtered/parvicapture.biallelic.snp.filtered.nonmissing
plink --bfile filtered/parvicapture.biallelic.snp.filtered.nonmissing \
  --allow-extra-chr --missing --het --freqx --pca \
  --out filtered/parvicapture.biallelic.snp.filtered.nonmissing
```

```{r all}
g.pca <- read_delim(file.path(pathCluster, "Sequences", "unmerged", "all",
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  separate(Sample, paste0("X", 1:7), "[:punct:]") %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  dplyr::select(IdGenetic, Lane, PCA1, PCA2) %>% 
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = paste(IdGenetic, Lane))) +
  geom_line(aes(group = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  scale_color_discrete("Species") +
  ggtitle("All --mind 0.95 --geno 0.15")
plotly::ggplotly(g.pca)
```

## Lanes

```{bash lanesSH, eval=F, echo=T}
mkdir lanes
lanes=(L1 L2)
for lane in "${lanes[@]}" ; 
  do 
  grep $lane libraries/libraries.txt > libraries/$lane.txt
  mkdir lanes/$lane
  plink2 --threads 8 --memory 20000 \
    --vcf parvicapture.all.biallelic.snp.filtered.vcf.gz \
    --allow-extra-chr \
    --keep libraries/$lane.txt \
    --mind 0.95 --geno 0.15 \
    --make-bed --out lanes/$lane/parvicapture.biallelic.snp.filtered.nonmissing
  plink --bfile lanes/$lane/parvicapture.biallelic.snp.filtered.nonmissing \
    --allow-extra-chr --missing --het --freqx --pca \
    --out lanes/$lane/parvicapture.biallelic.snp.filtered.nonmissing
done
```

```{r lanes}
g.pca <- lapply(c("L1", "L2"), function(lane)
  read_delim(file.path(pathCluster, "Sequences", "unmerged", "lanes", lane,
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  separate(Sample, paste0("X", 1:7), "[:punct:]") %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  dplyr::select(IdGenetic, Lane, PCA1, PCA2) %>% 
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = paste(IdGenetic, Lane))) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
    scale_color_discrete(lane))
plotly::subplot(lapply(g.pca, plotly::ggplotly), nrows = 2)
```

## Pools

```{r poolsDef, eval=F, echo=T}
lapply(LETTERS[1:4], function(pool)
  write_tsv(filter(libraries, Pool == pool) %>% dplyr::select(Library),
            file.path(pathCluster, "Sequences", "unmerged", "libraries", paste0("Pool", pool, ".txt")),
            col_names = F))
```

```{bash poolSH, eval=F, echo=T}
mkdir pools
pools=(PoolA PoolB PoolC PoolD)
for pool in "${pools[@]}" ; 
  do 
  mkdir pools/$pool
  plink2 --threads 8 --memory 20000 \
    --vcf parvicapture.all.biallelic.snp.filtered.vcf.gz \
    --allow-extra-chr \
    --keep libraries/$pool.txt \
    --mind 0.95 --geno 0.15 \
    --make-bed --out pools/$pool/parvicapture.biallelic.snp.filtered.nonmissing
  plink --bfile pools/$pool/parvicapture.biallelic.snp.filtered.nonmissing \
    --allow-extra-chr --missing --het --freqx --pca \
    --out pools/$pool/parvicapture.biallelic.snp.filtered.nonmissing
done
```

```{r pools}
g.pca <- lapply(c("PoolA", "PoolB", "PoolC", "PoolD"), function(pool)
  read_delim(file.path(pathCluster, "Sequences", "unmerged", "pools", pool,
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  separate(Sample, paste0("X", 1:7), "[:punct:]") %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  dplyr::select(IdGenetic, Lane, PCA1, PCA2) %>% 
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = paste(IdGenetic, Lane))) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  scale_color_discrete(pool))
plotly::subplot(lapply(g.pca, plotly::ggplotly), nrows = 2)
```

## Plates

```{r platesDef, eval=F, echo=T}
lapply(unique(libraries$Plate_extraction), function(plate)
  write_tsv(filter(libraries, Plate_extraction == plate) %>% dplyr::select(Library),
            file.path(pathCluster, "Sequences", "unmerged", "libraries", paste0("Plate", plate, ".txt")),
            col_names = F))
```

```{bash platesSH, eval=F, echo=T}
mkdir plates
plates=(Plate1 Plate2 Plate3 Plate4 Plate5 Plate6)
for plate in "${plates[@]}" ; 
  do 
  mkdir plates/$plate
  plink2 --threads 8 --memory 20000 \
    --vcf parvicapture.all.biallelic.snp.filtered.vcf.gz \
    --allow-extra-chr \
    --keep libraries/$plate.txt \
    --mind 0.95 --geno 0.15 \
    --make-bed --out plates/$plate/parvicapture.biallelic.snp.filtered.nonmissing
  plink --bfile plates/$plate/parvicapture.biallelic.snp.filtered.nonmissing \
    --allow-extra-chr --missing --het --freqx --pca \
    --out plates/$plate/parvicapture.biallelic.snp.filtered.nonmissing
done
```

```{r plates}
g.pca <- lapply(c("Plate1", "Plate2", "Plate3", "Plate4", "Plate5", "Plate6"), function(plate)
  read_delim(file.path(pathCluster, "Sequences", "unmerged", "plates", plate,
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  separate(Sample, paste0("X", 1:7), "[:punct:]") %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  dplyr::select(IdGenetic, Lane, PCA1, PCA2) %>% 
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = paste(IdGenetic, Lane))) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
    scale_color_discrete(plate))
plotly::subplot(lapply(g.pca, plotly::ggplotly), nrows = 3)
```

## Repeats

```{r repeatsDef, eval=F, echo=T}
write_tsv(group_by(libraries, IdGenetic) %>% filter(n() > 1)  %>% ungroup() %>% dplyr::select(Library),
          file.path(pathCluster, "Sequences", "unmerged", "libraries", "Repeated.txt"), col_names = F)
write_tsv(group_by(libraries, IdGenetic) %>% filter(n() < 2) %>% ungroup() %>% dplyr::select(Library),
          file.path(pathCluster, "Sequences", "unmerged", "libraries", "Unrepeated.txt"), col_names = F)
```

```{bash repeatsSH, eval=F, echo=T}
mkdir repeats
repeats=(Repeated Unrepeated)
for repeat in "${repeats[@]}" ; 
  do 
  echo "### $repeat ###"
  mkdir repeats/$repeat
  plink2 --threads 8 --memory 20000 \
    --vcf parvicapture.all.biallelic.snp.filtered.vcf.gz \
    --allow-extra-chr \
    --keep libraries/$repeat.txt \
    --mind 0.95 --geno 0.15 \
    --make-bed --out repeats/$repeat/parvicapture.biallelic.snp.filtered.nonmissing
  plink --bfile repeats/$repeat/parvicapture.biallelic.snp.filtered.nonmissing \
    --allow-extra-chr --missing --het --freqx --pca \
    --out repeats/$repeat/parvicapture.biallelic.snp.filtered.nonmissing
done
```

```{r repeats}
g.pca <- lapply(c("Repeated", "Unrepeated"), function(rep)
  read_delim(file.path(pathCluster, "Sequences", "unmerged", "repeats", rep,
                       "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
             delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
    separate(Sample, paste0("X", 1:7), "[:punct:]") %>% 
    dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
    dplyr::select(IdGenetic, Lane, PCA1, PCA2) %>% 
    left_join(individuals) %>%
    ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = paste(IdGenetic, Lane))) +
    geom_line(aes(group = IdGenetic)) +
    geom_point(size=2, alpha = 0.5) +
    geom_hline(yintercept = 0) +
    geom_vline(xintercept = 0) +
    scale_color_discrete(rep))
plotly::subplot(lapply(g.pca, plotly::ggplotly), nrows = 2)
```

## Reextracted

```{r reextractedDef, eval=F, echo=T}
reextracted <- googlesheets::gs_title("Parvicapture") %>% 
              googlesheets::gs_read("Extraction") %>% 
              filter(is.na(Seq)) %>% 
  dplyr::select(IdGenetic) %>% unlist()
write_tsv(filter(libraries, IdGenetic %in% reextracted) %>% dplyr::select(Library),
          file.path(pathCluster, "Sequences", "unmerged", "libraries", "Reextracted.txt"), col_names = F)
write_tsv(filter(libraries, !(IdGenetic %in% reextracted)) %>% dplyr::select(Library),
          file.path(pathCluster, "Sequences", "unmerged", "libraries", "Unreextracted.txt"), col_names = F)
```

```{bash reextractedSH, eval=F, echo=T}
mkdir reextracted
reextracted=(Reextracted Unreextracted)
for type in "${reextracted[@]}" ; 
  do 
  mkdir reextracted/$type
  plink2 --threads 8 --memory 20000 \
    --vcf parvicapture.all.biallelic.snp.filtered.vcf.gz \
    --allow-extra-chr \
    --keep libraries/$type.txt \
    --mind 0.95 --geno 0.15 \
    --make-bed --out reextracted/$type/parvicapture.biallelic.snp.filtered.nonmissing
  plink --bfile reextracted/$type/parvicapture.biallelic.snp.filtered.nonmissing \
    --allow-extra-chr --missing --het --freqx --pca \
    --out reextracted/$type/parvicapture.biallelic.snp.filtered.nonmissing
done
```

```{r reextracted}
g.pca <- lapply(c("Reextracted", "Unreextracted"), function(rep)
  read_delim(file.path(pathCluster, "Sequences", "unmerged", "reextracted", rep,
                       "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
             delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
    separate(Sample, paste0("X", 1:7), "[:punct:]") %>% 
    dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
    dplyr::select(IdGenetic, Lane, PCA1, PCA2) %>% 
    left_join(individuals) %>%
    ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = paste(IdGenetic, Lane))) +
    geom_line(aes(group = IdGenetic)) +
    geom_point(size=2, alpha = 0.5) +
    geom_hline(yintercept = 0) +
    geom_vline(xintercept = 0) +
    scale_color_discrete(rep))
plotly::subplot(lapply(g.pca, plotly::ggplotly), nrows = 2)
```

## Clean

```{bash cleanSH, eval=F, echo=T}
mkdir clean
cat libraries/Reextracted.txt libraries/Repeated.txt libraries/PoolD.txt | sort | uniq > libraries/unclean.txt
plink2 --threads 8 --memory 20000 \
--vcf parvicapture.all.biallelic.snp.filtered.vcf.gz \
--allow-extra-chr \
--remove libraries/unclean.txt \
--mind 0.95 --geno 0.15 \
--make-bed --out clean/parvicapture.biallelic.snp.filtered.nonmissing
plink --bfile clean/parvicapture.biallelic.snp.filtered.nonmissing \
--allow-extra-chr --missing --het --freqx --pca \
--out clean/parvicapture.biallelic.snp.filtered.nonmissing
```

```{r clean}
g.pca <- read_delim(file.path(pathCluster, "Sequences", "unmerged", "clean",
                       "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
             delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
    separate(Sample, paste0("X", 1:7), "[:punct:]") %>% 
    dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
    dplyr::select(IdGenetic, Lane, PCA1, PCA2) %>% 
    left_join(individuals) %>%
    ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = paste(IdGenetic, Lane))) +
    geom_line(aes(group = IdGenetic)) +
    geom_point(size=2, alpha = 0.5) +
    geom_hline(yintercept = 0) +
    geom_vline(xintercept = 0) +
    scale_color_discrete("Clean")
plotly::ggplotly(g.pca)
```

## Admixture check

```{bash admixtureSHs, eval=F, echo=T}
module load bioinfo/admixture_linux-1.3.0
module load bioinfo/plink-v1.90b5.3
module load bioinfo/admixture_linux-1.3.0
# read_tsv(file.path(pathCluster, "Sequences", "unmerged", "all", "parvicapture.biallelic.snp.filtered.nonmissing.bim"),
#          col_names = F) %>%
#   mutate(X1 = as.numeric(as.factor(X1))) %>%
#   write_tsv(file.path(pathCluster, "Sequences", "unmerged", "all", "admixture",
#                       "parvicapture.biallelic.snp.filtered.nonmissing.bim"),
#             col_names = F)
cp ../parvicapture.biallelic.snp.filtered.nonmissing.bed ../parvicapture.biallelic.snp.filtered.nonmissing.fam ./
k=14 # expected number of species
admixture parvicapture.biallelic.snp.filtered.nonmissing.bed $k | tee log$k.out
k=3 # expected number of species
admixture parvicapture.biallelic.snp.filtered.nonmissing.bed $k | tee log$k.out
```

```{r admixture}
fam <- read_tsv(file.path(pathCluster, "Sequences", "unmerged", "all", "admixture",
                          "parvicapture.biallelic.snp.filtered.nonmissing.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Library = IID) %>% 
  left_join(libraries) %>% 
  mutate(SpeciesLong = paste(Genus, Species))
admix <- pophelper::readQ(list.files(file.path(pathCluster, "Sequences", "unmerged", "all", "admixture"), 
                          full.names = T, pattern = ".Q"), indlabfromfile = F)
# admix <- lapply(admix, "rownames<-", fam$Library)
admix <- lapply(admix, "rownames<-", fam$Label)
p <- pophelper::plotQMultiline(qlist = admix, exportplot = F, returnplot = T, useindlab = T, indlabsize = 5,
                               ordergrp=T, grplab = fam["SpeciesLong"], sortind = "all")
```

```{r admixture3}
gridExtra::grid.arrange(p$plot[[3]][[1]])  
```

```{r admixture14}
p <- pophelper::plotQMultiline(qlist = admix, exportplot = F, returnplot = T, useindlab = F, 
                               ordergrp=T, grplab = fam["SpeciesLong"], sortind = "all")
gridExtra::grid.arrange(p$plot[[1]][[1]]) 
```

```{r plateComparison3}
g <- fam %>% 
  left_join(admix[[3]] %>% 
              rownames_to_column("Library") %>% 
              reshape2::melt("Library", variable.name = "Cluster") %>% 
              group_by(Library) %>% 
              top_n(1, value)) %>% 
    mutate(row = substr(Position_extraction, 1, 1)) %>% 
  mutate(col = substr(Position_extraction, 2, 3)) %>%
  dplyr::rename(Lab = SpeciesLong, Bioinfo = Cluster) %>% 
  dplyr::select(Library, IdGenetic, Lane, Label, Plate_extraction, Position_extraction, row, col,
                Lab, Bioinfo) %>% 
    reshape2::melt(c("Library", "IdGenetic", "Lane", "Label", "Plate_extraction", "Position_extraction", "row", "col"),
                 variable.name = "Step", value.name = "SpeciesLong") %>% 
  ggplot(aes(y = factor(row,levels = rev(LETTERS[1:8])),
             x = factor(col,levels = 1:12), label = Library)) + 
  geom_point(aes(col = SpeciesLong), size = 4)  +
  facet_grid(Plate_extraction ~ Step, labeller = "label_both") +
  theme_bw() +
  labs(x=NULL, y = NULL) +
  viridis::scale_color_viridis(discrete = T)
plotly::ggplotly(g)
```

```{r plateComparison14}
g <- fam %>% 
  left_join(admix[[1]] %>% 
              rownames_to_column("Library") %>% 
              reshape2::melt("Library", variable.name = "Cluster") %>% 
              group_by(Library) %>% 
              top_n(1, value)) %>% 
    mutate(row = substr(Position_extraction, 1, 1)) %>% 
  mutate(col = substr(Position_extraction, 2, 3)) %>%
  dplyr::rename(Lab = SpeciesLong, Bioinfo = Cluster) %>% 
  dplyr::select(Library, IdGenetic, Lane, Label, Plate_extraction, Position_extraction, row, col,
                Lab, Bioinfo) %>% 
    reshape2::melt(c("Library", "IdGenetic", "Lane", "Label", "Plate_extraction", "Position_extraction", "row", "col"),
                 variable.name = "Step", value.name = "SpeciesLong") %>% 
  ggplot(aes(y = factor(row,levels = rev(LETTERS[1:8])),
             x = factor(col,levels = 1:12), label = Library)) + 
  geom_point(aes(col = SpeciesLong), size = 4)  +
  facet_grid(Plate_extraction ~ Step, labeller = "label_both") +
  theme_bw() +
  labs(x=NULL, y = NULL) +
  viridis::scale_color_viridis(discrete = T)
plotly::ggplotly(g)
```

