```{r setup_denovo, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(parallel)
library(tidyverse)
library(rstan)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = F, cache.lazy = F)
pathCluster <- "~/Remotes/genotoul/work/PhD/data/Eschweilera_Paracou"
path <- "../../../data/Eschweilera_Paracou/"
```


# *de novo*

## Libraries

```{r denovoLib}
individuals <- googlesheets::gs_title("Parvicapture") %>% 
  googlesheets::gs_read("Extraction") %>% 
  filter(!is.na(Seq)) %>% 
  dplyr::select(IdGenetic, Genus, Species, Plot, SubPlot, TreeFieldNum) %>% 
  unique() %>% 
  mutate(cat = recode(Species, "wachenheimii" = "other", "grandiflora_form2" = "other",
                      "simiorum" = "outgroup", "collina"  = "other", "persistens" = "outgroup",
                      "squamata" = "other", "pedicellata" = "other", "grandiflora" = "other",
                      "micrantha" = "other", "parviflora" = "other", 
                      "congestiflora" = "outgroup"))
a <- read_tsv(file.path(pathCluster, "Sequences", "denovo", "libraries.txt"),
         col_names = "Library") %>%
  separate(Library, paste0("X", 1:7), "[:punct:]", remove = F, convert = T) %>%
  dplyr::rename(IdGenetic = X4, Lane = X5, Label = X3) %>%
  dplyr::select(Library, IdGenetic, Lane, Label)
b <- googlesheets::gs_title("Parvicapture") %>%
              googlesheets::gs_read("Extraction") %>%
              filter(!is.na(Seq)) %>%
              mutate(Pool = c(rep(LETTERS[1:3], each = ceiling(4*96/3))[1:(4*96)], rep("D", 534-4*96))) %>%
  mutate(Lane = Seq) %>%
  mutate(Label = 1:nrow(.)) # IGA label
libraries <- left_join(a, b, by = c("IdGenetic", "Lane", "Label"))
```

```{bash raw2filtered, eval=F, echo=T}
#!/bin/bash    

# variable
file=$1
echo "File: $file"
raw=$2
echo "Raw: $raw"
reference=$3
echo "Reference: $reference"

# modules
cd runs/$reference
module load bioinfo/Trimmomatic-0.36
module load bioinfo/bwa-0.7.15
module load bioinfo/picard-2.14.1
module load bioinfo/samtools-1.4
module load bioinfo/bedtools2-2.29.0

# trimming
echo "Step: Trimming"
java -jar $TRIM_HOME/trimmomatic.jar PE \
  ../../$raw/"$file"_R1_001.fastq \
  ../../$raw/"$file"_R2_001.fastq \
  trimmed/paired/"$file"_R1_paired.fq.gz \
  trimmed/unpaired/"$file"_R1_unpaired.fq.gz \
  trimmed/paired/"$file"_R2_paired.fq.gz \
  trimmed/unpaired/"$file"_R2_unpaired.fq.gz \
  ILLUMINACLIP:$TRIM_HOME/adapters/TruSeq3-PE.fa:2:30:10:2:keepBothReads \
  SLIDINGWINDOW:4:20

# mapping
echo "Step: mapping"
bwa mem -M -R "@RG\tID:$file\tSM:$file\tPL:HiSeq4K" \
  -t 1 \
  ../../reference/$reference.fasta \
  trimmed/paired/"$file"_R1_paired.fq.gz 	\
  trimmed/paired/"$file"_R2_paired.fq.gz \
  > mapped/sam/$file.sam
rm trimmed/paired/"$file"_R1_paired.fq.gz trimmed/paired/"$file"_R2_paired.fq.gz
rm trimmed/unpaired/"$file"_R1_unpaired.fq.gz trimmed/unpaired/"$file"_R2_unpaired.fq.gz
java -Xmx4g -jar $PICARD SortSam \
  I=mapped/sam/$file.sam \
  O=mapped/bam/$file.bam \
  SORT_ORDER=coordinate 
samtools index mapped/bam/$file.bam
rm mapped/sam/$file.sam
samtools view -u -f 1 -F 12 \
  mapped/bam/$file.bam > mapped/bam.paired/$file.bam
samtools index mapped/bam.paired/$file.bam
rm mapped/bam/$file.bam mapped/bam/$file.bam.bai
samtools sort -n  \
  mapped/bam.paired/$file.bam > mapped/bam.paired.sorted/$file.bam
rm mapped/bam.paired/$file.bam mapped/bam.paired/$file.bam.bai

# extracting reads
bedtools bamtofastq -i mapped/bam.paired.sorted/$file.bam \
  -fq filtered.reads/"$file"_R1_001.fastq -fq2 filtered.reads/"$file"_R2_001.fastq
rm mapped/bam.paired.sorted/$file.bam
```

```{bash denovoLibSh, eval=F, echo=T}
raw=../raw/
reference=../reference/reference.fasta
mkdir trimmed trimmed/paired trimmed/unpaired
mkdir mapped mapped/sam mapped/bam mapped/bam.paired mapped/bam.paired.sorted
mkdir filtered.reads
mkdir out
for lib in $(cat libraries.txt); do echo "bash raw2filtered.sh $lib $raw $reference"; done > runFiltered.sh
sarray -J haplo -o out/%j.out -e out/%j.err \
  -t 10:00:00 --mem=8G --cpus-per-task=2 --mail-type=BEGIN,END,FAIL runFiltered.sh
rm -r trimmed mapped out
sbatch denovo.sh

for lib in $(ls *.R2); do mv $lib ${lib%.*}_R2_001.fastq  ; done
```

## Reference

3730 target sequences in `reference.fasta.fai` that we will split in 37 jobs of ca 37 sequences.

```{bash, eval=F, echo=T}
mkdir reference
cut ../../reference/reference.fasta.fai -f1 > reference.sequence.list
split -l 100 -d reference.sequence.list reference.sequence_ --additional-suffix=.list
rm reference.sequence.list
module load bioinfo/seqtk-1.2
reference=../../reference/reference.fasta
for list in $(ls *.list); do seqtk subseq $reference $list > ${list%.*}.fasta ; done
module load bioinfo/bwa-0.7.15
for ref in $(ls *.fasta); do bwa index $ref ; done
```


## Libraries subset

```{bash, eval=F, echo=T}
mkdir runs
for ref in $(ls reference/*.list); do mkdir runs/$(basename ${ref%.*}) ; done
for ref in $(ls runs); 
  do 
  mkdir runs/$ref/trimmed runs/$ref/trimmed/paired runs/$ref/trimmed/unpaired
  mkdir runs/$ref/mapped runs/$ref/mapped/sam runs/$ref/mapped/bam runs/$ref/mapped/bam.paired runs/$ref/mapped/bam.paired.sorted
  mkdir runs/$ref/filtered.reads
done
for ref in $(ls runs); do for lib in $(cat libraries.txt); do echo "bash raw2filtered2.sh $lib filtered.reads $ref"; done ; done > runFiltered.sh
split -l 1000 -d runFiltered.sh runFiltered_ --additional-suffix=.sh
rm runFiltered_all.sh # very important against infinite loop because if runFilttered-all.sh exists in the next command it will run itself in an inifinite loop until filling disk memory
for runs in $(ls runFiltered_*); do echo "sarray  --wait -J haplo -o out/%j.out -e out/%j.err -t 10:00:00 --mem=4G --cpus-per-task=1 --mail-type=BEGIN,END,FAIL $runs ;"; done > runFiltered_all.sh # we may number outputs in out/ and remove them as soon as they end to avoid an excessive number of files and mail might be removed and kept only in the general sbatch
for ref in $(ls runs); do rm -r runs/$ref/trimmed runs/$ref/mapped ; done
for ref in $(ls runs); do cp params.txt runs/$ref/ ; done
for ref in $(ls runs); do echo "cd runs/$ref ; module load system/Miniconda3-4.4.10 ; module load bioinfo/ipyrad-0.7.28 ; ipyrad -p params.txt -s 1234567 -c 4 ;"; done > runDenovo.sh
# test
head -n 1 runDenovo.sh > runDenovo.test.sh
sarray -J denovo -o out/denovo.%j.out -e out/denovo.%j.err -t 48:00:00 --mem=40G --cpus-per-task=4 --mail-type=BEGIN,END,FAIL runDenovo.test.sh ;
```


## PCA

```{bash pcaSH, eval=F, echo=T}
plink=~/Tools/plink_linux_x86_64_20190617/plink 
$plink \
  --vcf denovo.vcf \
  --const-fid --allow-extra-chr \
  --set-missing-var-ids @_snp# \
  --mind 0.95 \
  --missing --het --freqx --pca var-wts \
  --make-bed --out denovo
```


```{r genotype}
read_delim(file.path(path, "Sequences", "denovo", "denovo_outfiles",
                              "denovo.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  separate(Sample, paste0("X", 1:7), "[:punct:]", remove = F) %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  dplyr::select(Sample, IdGenetic, Lane, PCA1, PCA2) %>% 
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = Species, 
             label = Sample)) +
  geom_line(aes(group = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  scale_color_discrete("Species") +
  theme(legend.position = "bottom")
read_delim(file.path(path, "Sequences", "denovo", "denovo_outfiles",
                              "denovo.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  separate(Sample, paste0("X", 1:7), "[:punct:]", remove = F) %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  dplyr::select(Sample, IdGenetic, Lane, PCA1, PCA2) %>% 
  left_join(individuals) %>%
  filter(PCA1 < 0) %>% 
  ggplot(aes(x = PCA1, y = PCA2, col = Species, 
             label = Sample)) +
  geom_line(aes(group = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  scale_color_discrete("Species") +
  theme(legend.position = "bottom")
```
