```{r setup_snps, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(parallel)
library(tidyverse)
library(rstan)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "../../../data/Eschweilera_Paracou"
pathCluster <- "~/Remotes/genotoul/work/PhD/data/Eschweilera_Paracou"
```

```{r}
clades <- rbind(
  data.frame(Clade = "Parvifolia",
             Species = c("sagotiana", "coriacea", "decolorans", "pedicellata", "wachenheimii", "grandiflora",
                         "grandiflora_form2")),
  data.frame(Clade = "Chartacea",
             Species = c("congestiflora", "simiorum"))
)
individuals <- googlesheets::gs_title("Parvicapture") %>% 
  googlesheets::gs_read("Extraction") %>% 
  dplyr::select(IdGenetic, Genus, Species, Plot, SubPlot, TreeFieldNum) %>% 
  unique() %>% 
  na.omit() %>% 
  left_join(clades)
individuals %>% 
  filter(Clade == "Parvifolia") %>% 
  mutate(FID = 0, IID = paste0(IdGenetic, ".g.vcf"), PAT = 0, MAT = 0, SEX = 0, PHENO = -9) %>% 
  dplyr::select(FID, IID, PAT, MAT, SEX, PHENO) %>% 
  write_tsv(path = file.path(pathCluster, "Sequences", "calling", "Parvifolia.fam"), col_names = F)
individuals %>% 
  filter(Species %in% c("coriacea", "decolorans", "sagotiana")) %>% 
  mutate(FID = 0, IID = paste0(IdGenetic, ".g.vcf"), PAT = 0, MAT = 0, SEX = 0, PHENO = -9) %>% 
  dplyr::select(FID, IID, PAT, MAT, SEX, PHENO) %>% 
  write_tsv(path = file.path(pathCluster, "Sequences", "calling", "3sp.fam"), col_names = F)
```

# SNPs

We filtered the previously produced raw vcf (**491 517** variants over **438** individuals) with several steps:

* __Biallelic__ raw vcf filtering in **299 207** variants over **438** individuals
* __SNP__ biallelic vcf filtering in **255 700** variants over **438** individuals
* __Filters__ biallelic snp vcf filtering in **217 643** variants over **438** individuals
* __Missing__ filtered biallelic snp vcf filtering in **80 686** variants over **435** individuals
* __Parvifolia__ filtered & non missing biallelic snp vcf filtering in **80 686** variants over **X** individuals

## Biallelic

We then used `bcftools` to limit data to biallelic variants (`--max-alleles 2`), resulting in 299 207 biallelic variants.

```{bash biallelic, eval=F, echo=T}
srun --mem=20G --cpus-per-task=8 --pty bash
module load bioinfo/bcftools-1.8
bcftools view --max-alleles 2 --threads 8 parvicapture.raw.vcf.gz | bgzip -c --threads 8 > parvicapture.biallelic.vcf.gz
bcftools stats --threads 8 parvicapture.biallelic.vcf.gz
```

## SNP

We then used `gatk` to limit data to biallelic snps, resulting in 255 700 biallelic snps.

```{bash snp, eval=F, echo=T}
module load bioinfo/gatk-4.1.2.0
gatk IndexFeatureFile -F parvicapture.biallelic.vcf.gz
gatk SelectVariants -V parvicapture.biallelic.vcf.gz -select-type SNP -O parvicapture.all.biallelic.snp.vcf.gz
gatk IndexFeatureFile -F parvicapture.all.biallelic.snp.vcf.gz
module load bioinfo/bcftools-1.8
bcftools stats  --threads 8 parvicapture.all.biallelic.snp.vcf.gz
```

## Filters

We filtered the biallelic snp vcf with following filters (name, filter, description), resulting in 217 643 filtered biallelic snps,  using next histograms to set and test parameters values :

* __Quality ([QUAL](https://software.broadinstitute.org/gatk/documentation/article?id=7258))__ `QUAL < 30`: represents the likelihood of the site to be homozygote across all samples, we filter out variants having a low quality score (\@ref(fig:filtersR))
* __Quality depth ([QD](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.php))__ `QD < 2`: filter out variants with low variant confidence (\@ref(fig:filtersR))
* __Fisher strand bias ([FS](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.php))__ `FS > 60`: filter out variants based on Phred-scaled p-value using Fisher's exact test to detect strand bias (\@ref(fig:filtersR))
* __Strand odd ratio ([SOR](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.php))__ `SOR < 3`: filter out variants based on Phred-scaled p-value used to detect strand bias (\@ref(fig:filtersR))

```{bash filters, eval=F, echo=T}
module load bioinfo/tabix-0.2.5
module load bioinfo/vcftools-0.1.15
vcftools --gzvcf parvicapture.all.biallelic.snp.vcf.gz --missing-indv -c > stats/missing.ind.txt
vcftools --gzvcf parvicapture.all.biallelic.snp.vcf.gz --missing-site -c > stats/missing.site.txt
vcftools --gzvcf parvicapture.all.biallelic.snp.vcf.gz --site-quality -c > stats/QUAL.txt
vcftools --gzvcf parvicapture.all.biallelic.snp.vcf.gz --get-INFO AC --get-INFO AF --get-INFO QD \
  --get-INFO FS --get-INFO SOR -c > stats/INFO.txt
```

```{r filtersR, fig.cap="Quality, quality by depth, fisher strand and strand odd ratios per biallelic SNPs."}
filtering <- left_join(read_tsv(file.path(path, "Sequences", "calling", "stats", "missing.site.txt")),
          read_tsv(file.path(path, "Sequences", "calling", "stats", "QUAL.txt"))) %>%
  left_join(read_tsv(file.path(path, "Sequences", "calling", "stats", "INFO.txt"))) %>%
  mutate(QD = as.numeric(QD)) %>%
  mutate(AF = as.numeric(AF)) %>%
  sample_n(1000) # for tests
QUALthresh <- 30
g.qual <- ggplot(filtering, aes(QUAL)) +
  geom_rect(xmin = 0, xmax = log10(QUALthresh), ymin = 0, ymax = 100, col = "grey", alpha = 0.4) +
  geom_density(fill = "blue", alpha = 0.4, col = NA) +
  geom_vline(xintercept = QUALthresh, col = "red", linetype = "dashed") +
  xlab("Quality") +
  ylab("Biallelic SNPs") +
  scale_x_log10() +
  ggtitle(paste0("QUAL: ", round(nrow(filter(filtering, QUAL > QUALthresh))/nrow(filtering)*100, 2),
                 "% (", nrow(filter(filtering, QUAL > QUALthresh)),
                 " / ", nrow(filtering), ") variant above ", QUALthresh))
QDthresh <- 2
g.qd <- ggplot(filtering, aes(QD)) +
  geom_rect(xmin = 0, xmax = QDthresh, ymin = 0, ymax = 100, col = "grey", alpha = 0.4) +
  geom_density(fill = "blue", alpha = 0.4, col = NA) +
  geom_vline(xintercept = QDthresh, col = "red", linetype = "dashed") +
  xlab("Quality depth") +
  ylab("Biallelic SNPs") +
  ggtitle(paste0("QD: ", round(nrow(filter(filtering, QD > QDthresh))/nrow(filtering)*100, 2),
                 "% (", nrow(filter(filtering, QD > QDthresh)),
                 " / ", nrow(filtering), ") variant above ", QDthresh))
FSthresh <- 60
g.fs <- ggplot(filtering, aes(FS)) +
  geom_rect(xmin = FSthresh, xmax = 1000, ymin = 0, ymax = 100, col = "grey", alpha = 0.4) +
  geom_density(fill = "blue", alpha = 0.4, col = NA) +
  geom_vline(xintercept = FSthresh, col = "red", linetype = "dashed") +
  xlab("Fisher Strand Ratio") +
  ylab("Biallelic SNPs") +
  ggtitle(paste0("FS: ", round(nrow(filter(filtering, FS < FSthresh))/nrow(filtering)*100, 2),
                 "% (", nrow(filter(filtering, FS < FSthresh)),
                 " / ", nrow(filtering), ") variant above ", FSthresh))
SORthresh <- 3
g.sor <- ggplot(filtering, aes(SOR)) +
  geom_rect(xmin = SORthresh, xmax = 100, ymin = 0, ymax = 100, col = "grey", alpha = 0.4) +
  geom_density(fill = "blue", alpha = 0.4, col = NA) +
  geom_vline(xintercept = SORthresh, col = "red", linetype = "dashed") +
  xlab("Strand Odd Ratio") +
  ylab("Biallelic SNPs") +
  ggtitle(paste0("SOR: ", round(nrow(filter(filtering, SOR < SORthresh))/nrow(filtering)*100, 2),
                 "% (", nrow(filter(filtering, SOR < SORthresh)),
                 " / ", nrow(filtering), ") variant below ", SORthresh))
cowplot::plot_grid(g.qual, g.qd, g.fs, g.sor)
```

```{bash filters2, eval=F, echo=T}
module load bioinfo/gatk-4.1.2.0
gatk VariantFiltration \
  -V parvicapture.all.biallelic.snp.vcf.gz \
  --filter-expression "QUAL < 30.0 || QD < 2.0 || FS > 60.0 || SOR > 3.0" \
  --filter-name "FAIL" \
  -O parvicapture.all.biallelic.snp.intermediate.vcf.gz
gatk SelectVariants \
  -V parvicapture.all.biallelic.snp.intermediate.vcf.gz \
  --exclude-filtered \
  -O parvicapture.all.biallelic.snp.filtered.vcf.gz
gatk IndexFeatureFile -F parvicapture.all.biallelic.snp.filtered.vcf.gz
module load bioinfo/bcftools-1.8
bcftools stats --threads 8 parvicapture.all.biallelic.snp.filtered.vcf.gz
```

## Missing data

Missing data filtering is a bit more tricky because missing data of SNPs and individuals are related, *e.g.* removing individuals with a lot of missing data result in the decrease of SNPs. Ideally, we would want to keep all individuals, but this would result in a lot of SNP loss because of least represented individuals. So we need to chose a threshold for missing data for individuals `--mind` and SNPs `--geno`.

```{bash missing, eval=F, echo=T}
module load bioinfo/plink_high_contig_20190905
module load bioinfo/plink2_high_contig_20190905
mkdir filtered
plink2 --threads 8 --memory 20000 \
  --vcf parvicapture.all.biallelic.snp.filtered.vcf.gz \
  --allow-extra-chr \
  --make-bed --out filtered/parvicapture.biallelic.snp.filtered
  --missing --het --freq --pca
```

```{r missingIndividuals, fig.cap="Missing data statistics for filtered biallelic SNP before missing data filtering per individual."}
g.het_imiss <- read_delim(file.path(pathCluster, "Sequences", "merged", "filtered",
                                    "parvicapture.biallelic.snp.filtered.imiss"), delim = " ") %>%
  dplyr::rename_all(funs(gsub(" ", "", .))) %>%
  mutate_at(c("N_MISS", "N_GENO", "F_MISS"), as.numeric) %>%
  mutate(IdGenetic = gsub(".g.vcf", "", IID)) %>%
  mutate(IdGenetic = gsub(" ", "", IdGenetic)) %>%
  left_join(individuals) %>%
  ggplot(aes(F_MISS, fill = paste(Genus, Species))) +
  geom_histogram(position = "dodge") +
  scale_x_log10() +
  scale_fill_discrete("Species", guide = "none") +
  xlab("Proportion of missing SNPs per individual")
g.pca <- read_delim(file.path(pathCluster, "Sequences", "merged", "filtered",
                              "parvicapture.biallelic.snp.filtered.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  mutate(IdGenetic = gsub(".g.vcf", "", Sample)) %>%
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species))) +
  geom_point(size=2) +
  stat_ellipse(level = 0.95, size = 1) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  scale_color_discrete("Species")
cowplot::plot_grid(g.het_imiss, g.pca, nrow = 2, rel_heights = c(1,2))
```


```{r missingSNP, fig.cap="Missing data statistics for filtered biallelic SNP before missing data filtering per SNP."}
read_delim(file.path(pathCluster, "Sequences", "merged", "filtered",
                                "parvicapture.biallelic.snp.filtered.lmiss"), delim = " ") %>%
  dplyr::rename_all(funs(gsub(" ", "", .))) %>%
  dplyr::rename(contig = CHR) %>%
  mutate(type = ifelse(grepl("TRINITY", contig), "functional", "neutral")) %>% 
  mutate(F_MISS = as.numeric(F_MISS)) %>%
  ggplot(aes(F_MISS)) +
  geom_histogram(position = "dodge") +
  facet_wrap(~ type, scales = "free") +
  xlab("Proportion of missing data") +
  ylab("SNP count") +
  scale_y_sqrt() +
  scale_x_sqrt() +
  geom_vline(xintercept = c(0.15, 0.05), col = "red", linetype = "dashed")
```

## Neutral vs functional

```{bash neutral, eval=F, echo=T}
cat filtered/parvicapture.biallelic.snp.filtered.bim | grep nuc > neutral.bim
rm -r neutral
mkdir neutral
plink2 --bfile filtered/parvicapture.biallelic.snp.filtered \
  --allow-extra-chr \
  --extract neutral.bim \
  --mind 0.95 --geno 0.15 \
  --make-bed --out neutral/parvicapture.biallelic.snp.filtered.nonmissing
plink --bfile neutral/parvicapture.biallelic.snp.filtered.nonmissing \
  --allow-extra-chr --missing --het --freqx --pca \
  --out neutral/parvicapture.biallelic.snp.filtered.nonmissing
```

```{r neutralInd, fig.cap="Missing data statistics for filtered biallelic SNP after missing data filtering (95% for individuals and 15% for SNPs) per individual."}
g.pca <- read_delim(file.path(pathCluster, "Sequences", "merged", "neutral",
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  mutate(IdGenetic = gsub(".g.vcf", "", Sample)) %>%
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  scale_color_discrete("Species") +
  ggtitle("Neutral --mind 0.95 --geno 0.15")
plotly::ggplotly(g.pca)
```

```{bash functional, eval=F, echo=T}
cat filtered/parvicapture.biallelic.snp.filtered.bim | grep TRINITY > functional.bim
rm -r functional
mkdir functional
plink2 --bfile filtered/parvicapture.biallelic.snp.filtered \
  --allow-extra-chr \
  --extract functional.bim \
  --mind 0.95 --geno 0.15 \
  --make-bed --out functional/parvicapture.biallelic.snp.filtered.nonmissing
plink --bfile functional/parvicapture.biallelic.snp.filtered.nonmissing \
  --allow-extra-chr --missing --het --freqx --pca \
  --out functional/parvicapture.biallelic.snp.filtered.nonmissing
```

```{r functionalInd, fig.cap="Missing data statistics for filtered biallelic SNP after missing data filtering (95% for individuals and 15% for SNPs) per individual."}
g.pca <- read_delim(file.path(pathCluster, "Sequences", "merged", "functional",
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  mutate(IdGenetic = gsub(".g.vcf", "", Sample)) %>%
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  scale_color_discrete("Species") +
  ggtitle("Functional --mind 0.95 --geno 0.15")
plotly::ggplotly(g.pca)
```

## 1k

```{bash 1k, eval=F, echo=T}
rm -r 1k
mkdir 1k
plink2 --bfile filtered/parvicapture.biallelic.snp.filtered \
  --allow-extra-chr \
  --thin-count 1000 \
  --mind 0.95 --geno 0.15 \
  --make-bed --out 1k/parvicapture.biallelic.snp.filtered.nonmissing
plink --bfile 1k/parvicapture.biallelic.snp.filtered.nonmissing \
  --allow-extra-chr --missing --het --freqx --pca \
  --out 1k/parvicapture.biallelic.snp.filtered.nonmissing
```

```{r 1kInd, fig.cap="Missing data statistics for filtered biallelic SNP after missing data filtering (95% for individuals and 15% for SNPs) per individual."}
g.pca <- read_delim(file.path(pathCluster, "Sequences", "merged", "1k",
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  mutate(IdGenetic = gsub(".g.vcf", "", Sample)) %>%
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  scale_color_discrete("Species") +
  ggtitle("Neutral --mind 0.95 --geno 0.15")
plotly::ggplotly(g.pca)
```

## Unique vs Repeated

```{r repeatedR, echo=T}
googlesheets::gs_title("Parvicapture") %>% 
  googlesheets::gs_read("Extraction") %>% 
  group_by(IdGenetic) %>% 
  filter(n() > 1) %>% 
  dplyr::select(IdGenetic) %>% 
  unique() %>% ungroup() %>% 
  mutate(FID = 0, IID = paste0(IdGenetic, ".g.vcf"), PAT = 0, MAT = 0, SEX = 0, PHENO = -9) %>% 
  dplyr::select(FID, IID, PAT, MAT, SEX, PHENO) %>% 
  write_tsv(path = file.path(pathCluster, "Sequences", "merged", "Repeated.fam"), col_names = F)
googlesheets::gs_title("Parvicapture") %>% 
  googlesheets::gs_read("Extraction") %>% 
  group_by(IdGenetic) %>% 
  filter(n() < 2) %>% 
  dplyr::select(IdGenetic) %>% 
  unique() %>% ungroup() %>% 
  mutate(FID = 0, IID = paste0(IdGenetic, ".g.vcf"), PAT = 0, MAT = 0, SEX = 0, PHENO = -9) %>% 
  dplyr::select(FID, IID, PAT, MAT, SEX, PHENO) %>% 
  write_tsv(path = file.path(pathCluster, "Sequences", "merged", "Unique.fam"), col_names = F)
```


```{bash repeated, eval=F, echo=T}
rm -r repeated
mkdir repeated
plink2 --bfile filtered/parvicapture.biallelic.snp.filtered \
  --allow-extra-chr \
  --keep Repeated.fam \
  --mind 0.95 --geno 0.15 \
  --make-bed --out repeated/parvicapture.biallelic.snp.filtered.nonmissing
plink --bfile repeated/parvicapture.biallelic.snp.filtered.nonmissing \
  --allow-extra-chr --missing --het --freqx --pca \
  --out repeated/parvicapture.biallelic.snp.filtered.nonmissing
```

```{r repeatedInd, fig.cap="Missing data statistics for filtered biallelic SNP after missing data filtering (95% for individuals and 15% for SNPs) per individual."}
g.pca <- read_delim(file.path(pathCluster, "Sequences", "merged", "repeated",
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  mutate(IdGenetic = gsub(".g.vcf", "", Sample)) %>%
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  scale_color_discrete("Species") +
  ggtitle("Repeated --mind 0.95 --geno 0.15")
plotly::ggplotly(g.pca)
```

```{bash unique, eval=F, echo=T}
rm -r unique
mkdir unique
plink2 --bfile filtered/parvicapture.biallelic.snp.filtered \
  --allow-extra-chr \
  --keep Unique.fam \
  --mind 0.95 --geno 0.15 \
  --make-bed --out unique/parvicapture.biallelic.snp.filtered.nonmissing
plink --bfile unique/parvicapture.biallelic.snp.filtered.nonmissing \
  --allow-extra-chr --missing --het --freqx --pca \
  --out unique/parvicapture.biallelic.snp.filtered.nonmissing
```

```{r uniqueInd, fig.cap="Missing data statistics for filtered biallelic SNP after missing data filtering (95% for individuals and 15% for SNPs) per individual."}
g.pca <- read_delim(file.path(pathCluster, "Sequences", "merged", "unique",
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  mutate(IdGenetic = gsub(".g.vcf", "", Sample)) %>%
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  scale_color_discrete("Species") +
  ggtitle("Unique --mind 0.95 --geno 0.15")
plotly::ggplotly(g.pca)
```

## Outliers

```{bash outliers, eval=F, echo=T}
rm -r outliers
mkdir outliers
plink2 --bfile filtered/parvicapture.biallelic.snp.filtered \
  --allow-extra-chr \
  --mind 0.95 --geno 0.15 \
  --make-bed --out outliers/parvicapture.biallelic.snp.filtered.nonmissing
plink --bfile outliers/parvicapture.biallelic.snp.filtered.nonmissing \
  --allow-extra-chr --missing --het --freqx --pca \
  --out outliers/parvicapture.biallelic.snp.filtered.nonmissing
```

```{r outliersInd, fig.cap="Missing data statistics for filtered biallelic SNP after missing data filtering (95% for individuals and 15% for SNPs) per individual."}
g.pca <- read_delim(file.path(pathCluster, "Sequences", "merged", "outliers",
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  mutate(IdGenetic = gsub(".g.vcf", "", Sample)) %>%
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  scale_color_discrete("Species") +
  ggtitle("All --mind 0.95 --geno 0.15")
plotly::ggplotly(g.pca)
```

```{r outliersProperties}
outliers <- filter(g.pca$data, abs(PCA1) > 0.1 | abs(PCA2) > 10^-1)
googlesheets::gs_title("Parvicapture") %>% 
  googlesheets::gs_read("Extraction") %>% 
  filter(IdGenetic %in% gsub(".g.vcf", "", outliers$Sample)) %>% 
  left_join(googlesheets::gs_title("Parvicapture") %>% 
              googlesheets::gs_read("Reamplification") %>% 
              filter(IdGenetic %in% gsub(".g.vcf", "", outliers$Sample))) %>% 
  mutate(Species = paste(Genus, Species)) %>% 
  mutate(Reamplifcation = ifelse(is.na(Reamplifcation), 0, 1)) %>% 
  filter(!is.na(Seq)) %>% 
  dplyr::select(IdGenetic, Plate_extraction, Species, Plot, Repeat, Seq, Reamplifcation) %>% 
  reshape2::melt("IdGenetic") %>% 
  ggplot(aes(value)) + geom_bar() + facet_wrap(~ variable, scales = "free") + coord_flip()
```

## Test

```{bash test, eval=F, echo=T}
rm -r test
mkdir test
plink2 --bfile filtered/parvicapture.biallelic.snp.filtered \
  --allow-extra-chr \
  --mind 0.1 --geno 0.15 \
  --make-bed --out test/parvicapture.biallelic.snp.filtered.nonmissing
plink --bfile test/parvicapture.biallelic.snp.filtered.nonmissing \
  --allow-extra-chr --missing --het --freqx --pca \
  --out test/parvicapture.biallelic.snp.filtered.nonmissing
```

```{r testInd, fig.cap="Missing data statistics for filtered biallelic SNP after missing data filtering (95% for individuals and 15% for SNPs) per individual."}
g.pca <- read_delim(file.path(pathCluster, "Sequences", "merged", "test",
                              "parvicapture.biallelic.snp.filtered.nonmissing.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>%
  mutate(IdGenetic = gsub(".g.vcf", "", Sample)) %>%
  left_join(individuals) %>%
  ggplot(aes(x = PCA1, y = PCA2, col = paste(Genus, Species), label = IdGenetic)) +
  geom_point(size=2, alpha = 0.5) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  scale_color_discrete("Species") +
  ggtitle("All --mind 0.1 --geno 0.15")
plotly::ggplotly(g.pca)
```

## Parvifolia

Finally, we subseted the filtered and biallelic snp from the Parvifolia clade, resulting in **X** remaining individuals.

```{bash parvifolia, eval=F, echo=T}
module load bioinfo/plink_high_contig_20190905
module load bioinfo/plink2_high_contig_20190905
rm -r parvifolia
mkdir parvifolia
plink2 --bfile filtered/parvicapture.biallelic.snp.filtered \
  --allow-extra-chr \
  --keep Parvifolia.fam \
  --mind 0.50 --geno 0.0 \
  --make-bed --out parvifolia/parvicapture.biallelic.snp.filtered.nonmissing
plink --bfile parvifolia/parvicapture.biallelic.snp.filtered.nonmissing \
  --allow-extra-chr --missing --het --freqx --pca \
  --out parvifolia/parvicapture.biallelic.snp.filtered.nonmissing
```