```{r setup_annot, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(Biostrings)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F, eval = F)
path <- "~/Documents/BIOGECO/PhD/data/Eschweilera_Paracou/Sequences/annotation/"
```

# Annotation

We developed and manually tested with few SNPs a method to match transcriptomic data from Tysklind et al., in prep with our genomic data. We found expected proportion of SNPs type, with a lot of putatively hitchhiker SNPs in functional SNPs. Finally, ca 2000 SNPs were matching between the two datasets.

population structure of individuals. Then, we quickly looked at th spatial and environmental distribution of the different gene pools and individual mismatch (before association genomic analyses).

* __definition__ SNP type definition
* __results__ SNP classification, corresponding SNPs and genes, and common SNPs between datasets.

## Definition

```{r trscAlignment, eval=F, echo=T}
gff <- ape::read.gff(file.path(path, "Essago.filtered.fasta.transdecoder.gff3"), na.strings = c(".", "?"), GFF3 = TRUE)
snps <- read_tsv(file.path(path, "parvicapture.biallelic.snp.filtered.nonmissing.bim"),
         col_names = c("scf", "snp", "posM", "pos", "A1", "A2")) %>% 
  mutate(snp = paste0(scf, "_snp", pos)) %>% 
  mutate(type = ifelse(!grepl("TRINITY", scf), "neutral", "transcriptomic")) %>% 
  left_join(dplyr::rename(gff, scf = seqid, subtype = type)) %>% 
  filter(is.na(start) | (pos > start & pos < end))
write_tsv(snps, file.path(path, "snps.annotation"))
write_tsv(filter(snps, type == "neutral") %>% dplyr::select(scf) %>% unique(),  
          file.path(path, "snps.neutral"), col_names = F)
write_tsv(filter(snps, type == "transcriptomic", grepl("UTR", subtype)) %>% dplyr::select(scf) %>% unique(),  
          file.path(path, "snps.utr"), col_names = F)
write_tsv(filter(snps, type == "transcriptomic", !grepl("UTR", subtype)) %>% dplyr::select(scf) %>% unique(),  
          file.path(path, "snps.genic"), col_names = F)
snps %>% 
  dplyr::select(scf, pos, A1, A2) %>% 
  reshape2::melt(id.vars = c("scf", "pos")) %>% 
  dplyr::select(-variable) %>% 
  write_tsv(file.path(path, "snps.tsv"), col_names = F)
```

## Results

We obtained 8% of neutral snps against a79% of genic SNPs and 8% in UTR (Fig. \@ref(fig:snpType)). 

```{r snpType, fig.cap="Filtered snps by type. SNP have been classified between (i) \"purely neutral\", on scaffolds matching no transcript, (ii) \"hitchhiker\" on a scaffolds matching a trasncript but no within the transcript itself, and (iii) functional, when positioned within the matching transcript."}
read_tsv(file.path(path, "snps.annotation")) %>% 
  mutate(type = ifelse(grepl("UTR", subtype), "UTR", type)) %>% 
  mutate(type = ifelse(type == "transcriptomic" & !grepl("UTR", subtype), "genic", type)) %>% 
  dplyr::select(snp, type) %>% 
  unique() %>% 
  group_by(type) %>% 
  summarise(N = n()) %>% 
  mutate(Pct = N/sum(N)*100) %>% 
  ggplot(aes(x = type, y = N, fill = type, label = paste0(round(Pct),"%"))) +
  geom_bar(stat = "identity") +
  geom_text() +
  ylab("count") +
  theme(axis.text.x = element_blank(),
        axis.line.x = element_blank(),
        axis.title.x = element_blank(),
        axis.ticks.x = element_blank()) +
  scale_fill_discrete("SNP type")
```

## Synonymy

```{bash synonymy, eval=F, echo=T}
~/Tools/gffread/gffread Essago.filtered.fasta.transdecoder.gff3 -T -o Essago.filtered.fasta.transdecoder.gtf
perl ~/Tools/SNPdat_package_v1.0.5/SNPdat_v1.0.5.pl \
  -i snps.tsv \
  -g Essago.filtered.fasta.transdecoder.gtf \
  -f reference.fasta \
  -s synonymy.summary \
  -o synonymy.output
```
