```{r setup_envgenom, include=FALSE, eval=T}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(bayesplot)
library(raster)
library(rstan)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F, eval=T)
path <- "~/Documents/BIOGECO/PhD/data/Eschweilera_Paracou/Sequences/genomics"
crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
load("./parvicapture_save/Env.Rdata") 
eschwcol <- c("#1b9e77", "#d95f02", "#7570b3")
names(eschwcol) <- c("E. decolorans cluster", 
                     "E. sagotiana cluster",
                     "E. coriacea cluster")
```

# Environmental genomics

In this chapter, we decomposed environmental variation with population structure and individual kinship.

```{r DataEnvGenom, eval=F}
clades <- rbind(
  data.frame(Clade = "Parvifolia",
             Species = c("sagotiana", "coriacea", "decolorans", "pedicellata", "wachenheimii", "grandiflora",
                         "grandiflora_form2")),
  data.frame(Clade = "Chartacea",
             Species = c("congestiflora", "simiorum"))
)
trees <- googlesheets::gs_title("Parvicapture") %>% 
  googlesheets::gs_read("Extraction") %>% 
  dplyr::select(IdGenetic, Genus, Species, Plot, SubPlot, TreeFieldNum) %>% 
  unique() %>% 
  na.omit() %>% 
  left_join(clades)
trees <- read_delim(file.path(path, "..", "variants", "final", "filtered.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype"), delim = " ") %>%
  mutate(Library = gsub(".g.vcf", "", IID)) %>% 
  separate(Library, paste0("X", 1:7), "[:punct:]", remove = F, convert = T) %>%
  dplyr::rename(IdGenetic = X4, Lane = X5, Label = X3) %>% 
  left_join(trees)
trees <- left_join(trees, 
                   src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                                        "trees", "Paracou.sqlite")) %>% 
                     tbl("Paracou") %>% 
                     filter(CensusYear == 2015) %>% 
                     collect())
trees <- left_join(trees, 
                   read_tsv(file.path(path, "..", "variants", "final", "filtered.kmeans")), 
                   by = c("IID" = "Sample"))
treesXY <- trees
coordinates(treesXY) <- ~Xutm + Yutm
proj4string(treesXY) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
treesXY <- spTransform(treesXY, CRSobj = crs)
wetness <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "TWI_1m.tif"))
dem <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
wetness <- projectRaster(wetness, crs = crs)
trees$TWI <- raster::extract(wetness, treesXY)
relele <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "RelativeElevation_1m.tif"))
projection(relele) <- projection(dem)
trees$RE <- raster::extract(relele, treesXY)
rm(dem, wetness, treesXY)
cl <- parallel::makeCluster(getOption("cl.cores", 4))
parallel::clusterExport(cl, list("trees"))
NC <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
    tbl("Paracou") %>% 
    filter(Plot == local(trees$Plot[ind])) %>% 
    filter(idTree != local(trees$idTree[ind])) %>% 
    mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
    filter(dij < 20) %>% 
    mutate(DBH = CircCorr/pi) %>% 
    group_by(CensusYear) %>% 
    summarise(NCI = sum(DBH*DBH*exp(-0.25*dij))) %>% 
    ungroup() %>% 
    collect() %>% 
    arrange(CensusYear) %>% 
    summarise(idTree = local(trees$idTree[ind]),
              NCImean = mean(NCI),
              NCI0 = first(NCI),
              NCItoday = last(NCI))
})
parallel::stopCluster(cl) ; rm(cl)
NC <- bind_rows(NC)
trees <- cbind(trees, NC[-1])
rm(NC)
save(trees, file = "./parvicapture_save/Env.Rdata")
```

```{r, eval=F}
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                     "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(CensusYear == 2015) %>% 
  filter(Genus == "Eschweilera") %>% 
  filter(Species %in% c("coriacea", "sagotiana", "decolorans")) %>% 
  collect() %>% 
  sample_n(500)
cl <- parallel::makeCluster(getOption("cl.cores", 4))
parallel::clusterExport(cl, list("trees"))
NC <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
    tbl("Paracou") %>% 
    filter(Plot == local(trees$Plot[ind])) %>% 
    filter(idTree != local(trees$idTree[ind])) %>% 
    mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
    filter(dij < 20) %>% 
    mutate(DBH = CircCorr/pi) %>% 
    group_by(CensusYear) %>% 
    summarise(NCI = sum(DBH*DBH*exp(-0.25*dij))) %>% 
    ungroup() %>%
    arrange(CensusYear) %>% 
    summarise(idTree = local(trees$idTree[ind]),
              NCImean = mean(NCI),
              NCImedian = median(NCI),
              NCIvar = var(NCI),
              NCI0 = first(NCI),
              NCItoday = last(NCI)) %>% 
    collect()
})
parallel::stopCluster(cl) ; rm(cl)
NC <- bind_rows(NC)
trees$NCI <- NC$NCI
data.frame(d = as.vector(dist(trees[c("Xutm", "Yutm")]))) %>% 
  ggplot(aes(d, fill = d < 50)) + geom_histogram() +
  scale_x_log10() +
  ggtitle(paste0("d<50m = ", round(nrow(filter(data.frame(d = as.vector(dist(trees[c("Xutm", "Yutm")]))), d < 50))/nrow(data.frame(d = as.vector(dist(trees[c("Xutm", "Yutm")]))))*100,2), "%")) +
  scale_y_log10()
data.frame(d = as.vector(dist(trees[c("Xutm", "Yutm")])),
           nci = as.vector(dist(trees["NCI"]))) %>% 
  filter(d < 100) %>% 
  ggplot(aes(d, nci)) + geom_point()
pgirmess::correlog(data.frame(trees$Xutm, trees$Yutm), trees$NCI, method = "Moran", nbclass = 70) %>%
  as.data.frame() %>%
  filter(dist.class < 10^3) %>%
  ggplot(aes(x = dist.class, y = coef)) + geom_hline(yintercept = 0) +
  geom_point(aes(alpha = p.value < 0.01)) + geom_line() +
  scale_alpha_manual("p < 0.01", values = c(0.3, 1)) + scale_x_log10() +
  geom_vline(xintercept = c(27, 53), col = "lightgrey", linetype = "dashed") +
  geom_vline(xintercept = 150, col = "lightgrey", linetype = "dotted") +
  xlab("Distance (m)") + ylab("Moran\'s I") + scale_color_discrete("Variable")
```

```{r NCI, eval=F}
cl <- parallel::makeCluster(getOption("cl.cores", 4))
parallel::clusterExport(cl, list("trees"))
NC <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
    tbl("Paracou") %>% 
    filter(Plot == local(trees$Plot[ind])) %>% 
    filter(idTree != local(trees$idTree[ind])) %>% 
    mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
    filter(dij < 20) %>% 
    mutate(DBH = CircCorr/pi) %>% 
    group_by(CensusYear) %>% 
    summarise(idTree = local(trees$idTree[ind]),
              NCI = sum(DBH*DBH*exp(-0.25*dij))) %>% 
    ungroup() %>% 
    collect()
})
parallel::stopCluster(cl) ; rm(cl)
NC <- bind_rows(NC)
NC %>% 
  group_by(idTree) %>% 
  arrange(CensusYear) %>% 
  summarise(NCI0 = first(NCI), NCI2017 = last(NCI)) %>% 
  left_join(trees) %>% 
  mutate(DeltaNCI = NCI2017 - NCI0) %>% 
  # filter(DeltaNCI < 0)
  ggplot(aes(DeltaNCI)) + 
  geom_histogram() +
  geom_vline(xintercept = 1, col = "red", linetype = "dashed") +
  ggtitle("Eschweilera", "N = 71/257 (27.6%)")
```

## Genetic variance

We found a significant effect on the genotype on NCI, RE, and TWI. And population had a stronger effect on RE (13%) than TWI (6%). Nevertheless, residual variation remains high (73-88%).

```{r EnvMdata}
ids <- read_tsv(file.path(path, "..", "variants", "final", "filtered.king.id"))
K <- read_tsv(file.path(path, "..", "variants", "final", "filtered.king"),
              col_names = ids$IID) %>% 
  as.data.frame()
row.names(K) <- ids$IID
K <- as.matrix(K) 
K[is.na(K)] <- 0
K[K < 0] <- 0
K <- K*2
K <- as.matrix(Matrix::nearPD(K)$mat)
subtrees <- trees
rownames(subtrees) <- subtrees$IID
subtrees <- subtrees[row.names(K),]
# Removing other species
# K <- K[subtrees$Species %in% c("coriacea", "decolorans", "sagotiana"),
#   subtrees$Species %in% c("coriacea", "decolorans", "sagotiana")]
# subtrees <- subtrees[subtrees$Species %in% c("coriacea", "decolorans", "sagotiana"),]
mdata <- lapply(c("TWI", "RE", "NCImean", "NCI0", "NCItoday"), 
                function(variable) {
  return(list(N = nrow(subtrees),
              P = 3,
              y = as.vector(scale(subtrees[variable], center = F)),
              population = subtrees$cluster,
              K = K))})
names(mdata) <- c("TWI", "RE", "NCImean", "NCI0", "NCItoday")
mdata$RE$y <- mdata$RE$y + 1 # for the lognormal
save(mdata, file = file.path("parvicapture_save", 'dataEnv.Rdata'))   
```

```{bash envGeno2Cluster, eval=F}
vars=(TWI RE NCImean NCI0 NCItoday)
for var in "${vars[@]}" ; do for chain in $(seq 8) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript EnvGeno2.R $chain $var" ; done ; done > EnvGeno2.sh
sarray -J Env2 -o out/%j.Env2.out -e out/%j.Env2.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL EnvGeno2.sh
watch 'tail -n 1 out/*.Env2.out | grep Iteration'
```

```{r envGeno2Tab, fig.cap="Genetic variance of individual growth potential with a lognormal animal model."}
vars <- c("TWI", "RE", "NCImean", "NCI0", "NCItoday")
fitEnv <- vector("list", length(vars))
names(fitEnv) <- vars
for(var in vars){
  for(sim in list.files("parvicapture_save/all/EnvGeno2", 
                        pattern = var, full.names = T)){
    load(sim)
    fitEnv[[var]] <- c(fitEnv[[var]], fit)
  }
}
fitEnv <- lapply(fitEnv, sflist2stanfit)
fitEnvParvi <- fitEnv
save(fitEnvParvi, file = "../../thesis/rdata/fitEnvParvi.Rdata")
rm(fitEnvParvi)
lapply(fitEnv, broom.mixed::tidyMCMC, c("mu", "Vp", "Vg", "Vr"), 
       droppars = NULL, rhat = T) %>% 
  bind_rows(.id = "Variable") %>% 
  separate(term, c("parameter", "population"), convert = T) %>% 
  mutate(population = ifelse(is.na(population), "", population)) %>% 
  mutate(population = recode(population, "3" =  "E. decolorans cluster", 
                          "2" = "E. sagotiana cluster",
                          "1" = "E. coriacea cluster")) %>% 
  kable(caption = "Summary table of the kinship growth model",
        col.names = c("Variable", "Parameter",  "Population",  
                      "Estimate", "$\\sigma$", "$\\hat{R}$"))
```

```{r envGeno2Trace, fig.cap="Traceplot for environmental variables."}
cowplot::plot_grid(plotlist = lapply(fitEnv, mcmc_trace, 
                                     regex_pars = c("Vp", "Vg", "Vr"), 
                                     facet_args = list(nrow = 3)),
                   labels = names(fitEnv))   
```

```{r envgeno2R2, fig.cap="R2 for environmental variable"}
lapply(fitEnv, pars = c("Vp", "Vg", "Vr"), as.data.frame) %>% 
  bind_rows(.id = "model") %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vg, Vr))) %>% 
  mutate(Vexp = sum(c(Vp, Vg))) %>% 
  mutate_at(c("Vp", "Vg", "Vexp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = "model") %>% 
  group_by(model, variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, 
                           "Vexp" = "Marginal", "Vg" = "Genotype", 
                           "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +  
  theme(axis.title.y = element_blank()) +
  facet_wrap(~ model, nrow = 3) +
  coord_flip()   
```

```{r envGeno2VarPart, fig.cap="Genetic variance partitionning for environmental variables."}
lapply(fitEnv, mcmc_intervals_data, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(parameter = recode(parameter, "Vp" = "Population", "Vg" = "Genotype", "Vr" = "Residual")) %>% 
  group_by(variable) %>%
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = variable, fill = parameter)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  facet_wrap(~ variable, scales = "free") +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2))   
```

```{r aTWInNCI2, fig.cap="Multiplicative"}
t <- lapply(fitEnv, function(fit)
  as.data.frame(fit, "alog") %>% 
    summarise_all(median) %>% 
    reshape2::melt(NULL)) %>% 
  bind_rows(.id = "fit") %>% 
  separate(variable, c("X1", "IndNum", "X2"), convert = T) %>% 
  mutate(value = exp(value)) %>% 
  reshape2::dcast(IndNum ~ fit, value.var = "value") %>%
  cbind(mutate(subtrees, DBH = CircCorr/pi) %>% 
          dplyr::select(DBH, cluster, TWI, RE, NCImean, NCI0, NCItoday) %>% 
          dplyr::rename(TWI.var = TWI, RE.var = RE, pop = cluster, NCImean.var = NCImean, 
                        NCI0.var = NCI0, NCItoday.var = NCItoday)) %>% 
  dplyr::rename(TWI.gen = TWI, RE.gen = RE, NCImean.gen = NCImean, 
                NCI0.gen = NCI0, NCItoday.gen = NCItoday) %>% 
  mutate(pop = as.factor(pop))
reshape2::melt(t, id.vars = c("IndNum", "pop", "DBH", "RE.gen", "TWI.gen", 
                              "NCImean.gen", "NCI0.gen", "NCItoday.gen"), 
               variable.name = "variable", value.name = "variable.value") %>% 
  reshape2::melt(id.vars = c("IndNum", "pop", "DBH", "variable", "variable.value"), 
               variable.name = "genetic", value.name = "genetic.value") %>% 
  mutate(variable = gsub(".var", "", variable)) %>% 
  mutate(genetic = gsub(".gen", "", genetic)) %>% 
  filter(variable == genetic) %>% 
  mutate(pop = recode(pop, "3" =  "E. decolorans cluster", 
                      "2" = "E. sagotiana cluster",
                      "1" = "E. coriacea cluster")) %>% 
  ggplot(aes(variable.value, genetic.value, col = pop, size = DBH)) +
  geom_point() +
  facet_wrap(~ variable, scales = "free") +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous("DBH 2017") +
  scale_color_manual("Cluster", values = as.vector(eschwcol)[c(3,2,1)]) +
  xlab("Variable vlaue") + ylab("Genetic multiplicative value")
dplyr::select(t, TWI.gen, RE.gen ,NCImean.gen, NCI0.gen, NCItoday.gen) %>% 
  cor %>% 
  corrplot::corrplot.mixed()
```
