```{r setup_growthgenom, include=FALSE, eval=T}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(bayesplot)
library(rstan)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F, eval = T)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
path <- "~/Documents/BIOGECO/PhD/data/Eschweilera_Paracou/Sequences/genomics"
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
pairs_stan <- function(chain, stan_model, pars) {
  energy <- as.matrix(sapply(get_sampler_params(stan_model, inc_warmup = F), 
                             function(x) x[,"energy__"]))
  pars <- rstan::extract(stan_model, pars = pars, permuted = F)
  df <- data.frame(energy[,chain], pars[,chain,])
  names(df)[1] <- "energy"
  GGally::ggpairs(df, title = paste0("Chain", chain), 
                  lower = list(continuous = GGally::wrap("points", alpha = 0.2)))
}
crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
load("./parvicapture_save/Env2.Rdata") 
eschwcol <- c("#1b9e77", "#d95f02", "#7570b3")
names(eschwcol) <- c("E. decolorans cluster", 
                     "E. sagotiana cluster",
                     "E. coriacea cluster") 
```

# Growth genomics

```{r DataGrowthGenom}
load("./parvicapture_save/Env.Rdata")
env <- trees
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Eschweilera") %>% 
  mutate(DBH = CircCorr/pi) %>% 
  filter(!(CodeMeas %in% c(4))) %>% 
  collect()
trees <- read_delim(file.path(path, "..", "variants", "final", "filtered.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype"), delim = " ") %>%
  mutate(Library = gsub(".g.vcf", "", IID)) %>% 
  separate(Library, paste0("X", 1:7), "[:punct:]", remove = F, convert = T) %>%
  dplyr::rename(IdGenetic = X4, Lane = X5, Label = X3) %>% 
  left_join(googlesheets::gs_title("Parvicapture") %>% 
              googlesheets::gs_read("Extraction") %>% 
              dplyr::select(IdGenetic, Genus, Species, Plot, SubPlot, TreeFieldNum) %>% 
              unique() %>% 
              na.omit()) %>% 
  left_join(trees)
trees0 <- trees
trees <- trees %>% 
  group_by(Library) %>% 
  mutate(Y0 = dplyr::first(CensusYear), DBH0 = dplyr::first(DBH), 
         DBHtoday = dplyr::last(DBH), N = n()) %>%  
  ungroup() %>% 
  dplyr::select(Library, Y0, DBH0, DBHtoday, N) %>% 
  unique() %>% 
  mutate(DBHtoday = ifelse(DBHtoday == DBH0, DBHtoday + 0.1, DBHtoday)) %>% 
  mutate(DBHtoday = ifelse(DBHtoday < DBH0, DBH0 + 0.1, DBHtoday))
trees <- trees %>%
  left_join(env)
rm(env)
save(trees, file = "./parvicapture_save/Env2.Rdata")  
```

```{r, eval=F}
src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus %in% c("Symphonia", "Eschweilera"), 
         Species %in% c("globulifera", "sagotiana", "coriacea", "decolorans")) %>% 
  mutate(DBH = CircCorr/pi) %>% 
  filter(!(CodeMeas %in% c(4))) %>% 
  collect() %>% 
  group_by(idTree) %>% 
  mutate(Y0 = dplyr::first(CensusYear), DBH0 = dplyr::first(DBH), 
         DBHtoday = dplyr::last(DBH), N = n())  %>% 
  ungroup() %>% 
  dplyr::select(Genus, Species, Plot, idTree, Y0, DBH0, DBHtoday, N) %>% 
  unique() %>% 
  mutate(Treatment = ifelse(Plot %in% c(1, 6, 11, 13:16), "C", "T")) %>% 
   ggplot(aes(Treatment, (DBHtoday - DBH0)/(2017-Y0), fill = Genus)) +
  geom_boxplot()
src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                     "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus %in% c("Symphonia", "Eschweilera", "Eperua", "Qualea", "Dicorynia")) %>% 
  mutate(DBH = CircCorr/pi) %>% 
  filter(!(CodeMeas %in% c(4))) %>% 
  collect() %>% 
  group_by(idTree) %>% 
  mutate(Y0 = dplyr::first(CensusYear), DBH0 = dplyr::first(DBH), 
         DBHtoday = dplyr::last(DBH), N = n())  %>% 
  mutate(DeltaDBH = (DBHtoday - DBH0)/(2017-Y0)) %>% 
  group_by(Genus) %>% 
  mutate(m = median(DeltaDBH)) %>% 
  ungroup() %>% 
  dplyr::select(Genus, m, DeltaDBH) %>% 
  unique() %>% 
  mutate(Genus = fct_reorder(Genus, desc(m))) %>%
  ggplot(aes(Genus, , fill = Genus)) +
  geom_boxplot()
```

```{r mdata}
ids <- read_tsv(file.path(path, "..", "variants", "final", "filtered.king.id"))
K <- read_tsv(file.path(path, "..", "variants", "final", "filtered.king"),
              col_names = ids$IID) %>% 
  as.data.frame()
row.names(K) <- ids$IID
K <- as.matrix(K)
K <- K[trees$IID, trees$IID]
K[K < 0] <- 0
K <- K*2
K <- as.matrix(Matrix::nearPD(K)$mat)
subtrees <- trees
rownames(subtrees) <- subtrees$IID
subtrees <- subtrees[row.names(K),]
# trees with no growth data
nogrwoth <- which(is.na(subtrees$DBH0))
K <- K[-nogrwoth,][,-nogrwoth]
subtrees <- subtrees[-nogrwoth,]
# Removing other species
# K <- K[subtrees$Species %in% c("coriacea", "decolorans", "sagotiana"),
#        subtrees$Species %in% c("coriacea", "decolorans", "sagotiana")]
# subtrees <- subtrees[subtrees$Species %in% c("coriacea", "decolorans", "sagotiana"),]
mdata <- list(I = nrow(subtrees),
              Y = 2017 - min(subtrees$Y0) + 1,
              P = max(subtrees$cluster),
              years = min(subtrees$Y0):2017,
              DBH0 = subtrees$DBH0,
              Y0 = subtrees$Y0,
              DBHtoday = subtrees$DBHtoday + 10^-6,
              TWI = as.vector(scale(subtrees$TWI, center = F)),
              NCI = as.vector(scale(subtrees$NCImean, center = F)),
              RE = as.vector(scale(subtrees$RE, center = F)) + 1,
              pop = subtrees$cluster,
              K = K)
save(mdata, file = file.path("parvicapture_save", "dataGrowthPar.Rdata"))  
```

## Gmax

We used the following growth model with a lognormal distribution to estimate individual growth potential:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \theta_{1,p,i}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\theta_{2,p}})}{\theta_{3,p}}]^2)), \sigma_1) \\ 
  \theta_{1,p,i} \sim \mathcal {logN}(log(\theta_{1,p}), \sigma_2) \\ 
  \theta_{2,p} \sim \mathcal {logN}(log(\theta_2),\sigma_3) \\ 
  \theta_{3,p} \sim \mathcal {logN}(log(\theta_3),\sigma_4)
  (\#eq:gmax)
\end{equation} 

We fitted the equivalent model with following priors:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \hat{\theta_{1,p,i}}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\hat{\theta_{2,p}}})}{\hat{\theta_{3,p}}}]^2)), \sigma_1) \\ 
  \hat{\theta_{1,p,i}} = e^{log(\theta_{1,p}) + \sigma_2.\epsilon_{1,i}} \\ 
  \hat{\theta_{2,p}}  = e^{log(\theta_2) + \sigma_3.\epsilon_{2,p}} \\ 
  \hat{\theta_{3,p}}  = e^{log(\theta_3) + \sigma_4.\epsilon_{3,p}} \\  
  \epsilon_{1,i} \sim \mathcal{N}(0,1) \\
  \epsilon_{2,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{3,p} \sim \mathcal{N}(0,1) \\
  ~ \\
  (\theta_{1,p}, \theta_2, \theta_3) \sim \mathcal{logN}^3(log(1),1) \\ 
  (\sigma_1, \sigma_2, \sigma_3, \sigma_4) \sim \mathcal{N}^4_T(0,1) \\ 
    ~ \\
  V_P = Var(log(\mu_p)) \\ 
  V_R=\sigma_2^2
  (\#eq:gmaxstan)
\end{equation} 

```{bash fitGmaxParCluster, eval=F}
for chain in $(seq 8) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript Gmax.R $chain" ; done > Gmax.sh
sarray -J Gmax -o out/%j.Gmax.out -e out/%j.Gmax.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL Gmax.sh
watch 'tail -n 1 out/*.Gmax.out | grep Iteration'
```

```{r gmaxParTab}
fitGmax <- list()
for(sim in list.files("parvicapture_save/3sp/Gmax", full.names = T)){
  load(sim)
  fitGmax <- c(fitGmax, fit)
}
fitGmax <- sflist2stanfit(fitGmax)
broom::tidyMCMC(fitGmax, pars = c("thetap1", "theta2", "theta3",
                                   "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r gmaxParTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmax, pars = c("thetap1", "theta2", "theta3",
                                      "sigma", "lp__")),
           np = nuts_params(fitGmax)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r gmaxParPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmax, c("sigma"))
```

```{r gmaxParEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmax))
```

```{r gmaxParPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "E. coriacea cluster"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[1]"))))/
                 median(as.array(fitGmax, "thetap3[1]")))^2)) +
    stat_function(aes(col = "E. sagotiana cluster"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[2]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[2]"))))/
                 median(as.array(fitGmax, "thetap3[2]")))^2)) +
    stat_function(aes(col = "E. decolorans cluster"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[3]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[3]"))))/
                 median(as.array(fitGmax, "thetap3[3]")))^2)) +
  scale_color_manual("Cluster", values = as.vector(eschwcol)[c(3,1,2)])
```

```{r GmaxR2, fig.cap="R2 for Gmax."}
as.data.frame(fitGmax, c("Vp", "Vr")) %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vr))) %>% 
  mutate_at(c("Vp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = NULL) %>% 
  group_by(variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +
  theme(axis.title.y = element_blank()) +
  coord_flip()
```

```{r GmaxVarPart, fig.cap="Genetic variance partitionning for Gmax."}
 mcmc_intervals_data(fitGmax, regex_pars = c("Vp", "Vr")) %>% 
  mutate(parameter = recode(parameter, "Vp" = "Population", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = parameter)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2)) 
```

## Gmax & Genotype

We used the following growth model with a lognormal distribution to estimate individual growth potential and associated genotypic variation:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \theta_{1,p,i}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\theta_{2,p}})}{\theta_{3,p}}]^2)), \sigma_1) \\ 
  \theta_{1,p,i} \sim \mathcal {logN}(log(\theta_{1,p}.a_{1,i}), \sigma_2) \\ 
  \theta_{2,p} \sim \mathcal {logN}(log(\theta_2),\sigma_3) \\ 
  \theta_{3,p} \sim \mathcal {logN}(log(\theta_3),\sigma_4) \\
  a_{1,i} \sim \mathcal{MVlogN}(log(1), \sigma_5.K)
  (\#eq:gmaxgeno)
\end{equation} 

We fitted the equivalent model with following priors:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \hat{\theta_{1,p,i}}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\hat{\theta_{2,p}}})}{\hat{\theta_{3,p}}}]^2)), \sigma_1) \\ 
  \hat{\theta_{1,p,i}} = e^{log(\theta_{1,p}.\hat{a_{1,i}}) + \sigma_2.\epsilon_{1,i}} \\ 
  \hat{\theta_{2,p}}  = e^{log(\theta_2) + \sigma_3.\epsilon_{2,p}} \\ 
  \hat{\theta_{3,p}}  = e^{log(\theta_3) + \sigma_4.\epsilon_{3,p}} \\  
  \hat{a_{1,i}} = e^{\sigma_5.A.\epsilon_{4,i}} \\
  \epsilon_{1,i} \sim \mathcal{N}(0,1) \\
  \epsilon_{2,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{3,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{4,i} \sim \mathcal{N}(0,1) \\
  ~ \\
  (\theta_{1,p}, \theta_2, \theta_3) \sim \mathcal{logN}^3(log(1),1) \\ 
  (\sigma_1, \sigma_2, \sigma_3, \sigma_4, \sigma_5) \sim \mathcal{N}^5_T(0,1) \\ 
    ~ \\ 
    V_P = Var(log(\mu_p)) \\
    V_G=\sigma_5^2\\
    V_R=\sigma_2^2
  (\#eq:gmaxgenostan)
\end{equation} 

```{bash fitGmaxGenoParCluster, eval=F}
for chain in $(seq 8) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GmaxGeno.R $chain" ; done > GmaxGeno.sh
sarray -J GmaxGeno -o out/%j.GmaxGeno.out -e out/%j.GmaxGeno.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GmaxGeno.sh
watch 'tail -n 1 out/*.GmaxGeno.out | grep Iteration'
```

```{r gmaxgenoParTab}
fitGmaxGeno <- list()
for(sim in list.files("parvicapture_save/all/GmaxGeno", full.names = T)){
  load(sim)
  fitGmaxGeno <- c(fitGmaxGeno, fit)
}
# fitGmaxGeno <- fitGmaxGeno[c(5,8)]
fitGmaxGeno <- sflist2stanfit(fitGmaxGeno)
broom::tidyMCMC(fitGmaxGeno, pars = c("thetap1", "theta2", "theta3", "sigma", "lp__"),
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", 
                      "$\\hat R$", "$N_{eff}$")) 
```

```{r gmaxgenoParTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmaxGeno, 
                    pars = c("thetap1", "theta2", "theta3", "sigma", "lp__")),
           np = nuts_params(fitGmaxGeno)) +
  viridis::scale_color_viridis(discrete = T) 
```

```{r gmaxgenoParPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmaxGeno, c("sigma")) 
```

```{r gmaxgenoParEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmaxGeno)) 
```

```{r gmaxgeno1ParPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "E. coriacea cluster"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[1]"))))/
                 median(as.array(fitGmax, "thetap3[1]")))^2)) +
    stat_function(aes(col = "E. sagotiana cluster"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[2]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[2]"))))/
                 median(as.array(fitGmax, "thetap3[2]")))^2)) +
    stat_function(aes(col = "E. decolorans cluster"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[3]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[3]"))))/
                 median(as.array(fitGmax, "thetap3[3]")))^2)) +
  scale_color_manual("Cluster", values = as.vector(eschwcol)[c(3,1,2)])
```

```{r GmaxGenoR2, fig.cap="R2 for Gmax."}
as.data.frame(fitGmaxGeno, c("Vp", "Vg", "Vr")) %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vg, Vr))) %>% 
  mutate(Vexp = sum(c(Vp, Vg))) %>% 
  mutate_at(c("Vp", "Vg", "Vexp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = NULL) %>% 
  group_by(variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, 
                           "Vexp" = "Marginal", "Vg" = "Genotype", 
                           "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +
  theme(axis.title.y = element_blank()) +
  coord_flip() 
```

```{r GmaxGenoVarPart, fig.cap="Genetic variance partitionning for Gmax."}
 mcmc_intervals_data(fitGmaxGeno, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  mutate(parameter = recode(parameter, "Vp" = "Population", 
                            "Vg" = "Genotype", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = parameter)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2))  
```

```{r VarVsGenetic, fig.cap="Long legend...", fig.height=8, fig.width=8}
fitNCI <- list()
for(sim in list.files("parvicapture_save/all/EnvGeno2", 
                      pattern = "NCImean", full.names = T)){
  load(sim)
  fitNCI <- c(fitNCI, fit)
}
fitNCI <- sflist2stanfit(fitNCI)
fitRE <- list()
for(sim in list.files("parvicapture_save/all/EnvGeno2", 
                      pattern = "RE", full.names = T)){
  load(sim)
  fitRE <- c(fitRE, fit)
}
fitRE <- sflist2stanfit(fitRE)
t <- lapply(c(fitNCI, fitRE, fitGmaxGeno), function(fit){
        d <- as.data.frame(fit, "alog") %>%
    summarise_all(median) %>%
    reshape2::melt(NULL) # median
  # d <- as.data.frame(fit, "alog")[which.max(as.data.frame(fit, "lp__")$lp__),] %>% 
  #   reshape2::melt(NULL) # maximum likelihood
  if(nrow(d) == 257) d <- d[-nogrwoth,]
  d <- mutate(d, IndNum = 1:nrow(d))
  return(d)}) %>% 
  bind_rows(.id = "fit") %>% 
  mutate(fit = recode(fit, "1" = "aNCI", "2" = "aRE", "3" = "aGmax")) %>% 
  group_by(fit) %>% 
  mutate(value = exp(value)) %>% 
  reshape2::dcast(IndNum ~ fit, value.var = "value") %>% 
  cbind(subtrees) %>% 
  mutate(pop = recode(cluster, "1" = "E. coriacea cluster", 
                      "2" = "E. sagotiana cluster",
                      "3" = "E. decolorans cluster")) %>% 
  mutate(NCI = NCImean)
## Up
g.a.NCI.Gmax <- ggplot(t, aes(aGmax, aNCI, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "lm", col = "black") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = as.vector(eschwcol[c(3,1,2)])) +
  theme(axis.title = element_blank()) +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = 0.9)
g.a.RE.Gmax <- ggplot(t, aes(aGmax, aRE, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "lm", col = "darkgrey") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = as.vector(eschwcol[c(3,1,2)])) +
  theme(axis.title = element_blank()) +  ggtitle("Gmax") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = 0.95, col = "darkgrey")
g.a.RE.NCI <- ggplot(t, aes(aNCI, aRE, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "lm", col = "darkgrey") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = as.vector(eschwcol[c(3,1,2)])) +
  theme(axis.title = element_blank()) +  ggtitle("NCI") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.9, label.y = 0.95, col = "darkgrey")
## Diag
g.RE <- ggplot(t, aes(aRE, RE, col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  theme(axis.title.x = element_blank()) +
  scale_color_manual(guide = "none", values = as.vector(eschwcol[c(3,1,2)])) +
  ylab("RE") + ggtitle("RE")
  # annotate("text", x = 0.9, y = 7, label = expression(R[m]^2==0.35)) +
  # annotate("text", x = 0.9, y = 6, label = expression(R[c]^2==0.44))
g.NCI <- ggplot(t, aes(aNCI, NCI/10^4, col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
    theme(axis.title = element_blank()) +
  scale_color_manual(guide = "none", values = as.vector(eschwcol[c(3,1,2)]))
    # annotate("text", x = 0.9, y = 0.7, label = expression(R[m]^2==0.06)) +
  # annotate("text", x = 0.9, y = 0.6, label = expression(R[c]^2==0.24))
g.gmax <- ggplot(t, aes(aGmax, (DBHtoday - DBH0)/(2017-Y0), 
                        col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
    theme(axis.title = element_blank()) +
  scale_color_manual(guide = "none", values = as.vector(eschwcol[c(3,1,2)]))
      # annotate("text", x = 0.5, y = 1.1, label = expression(R[m]^2==0.10)) +
  # annotate("text", x = 0.5, y = 0.9, label = expression(R[c]^2==0.53))
## Down
load(file = "../symcapture/symcapture_save/p16.Rdata")
g.var.DBH.NCI <- ggplot(t, aes(NCI/10^4, (DBHtoday - DBH0)/(2017-Y0), 
                               size = DBHtoday)) +
  geom_hex(data = p16, fill = "lightgrey", alpha = 0.6, bins = 50) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = as.vector(eschwcol[c(3,1,2)])) +
  theme(axis.title = element_blank())
g.var.DBH.RE <- ggplot(t, aes(RE, (DBHtoday - DBH0)/(2017-Y0), 
                        size = DBHtoday)) +
  # geom_hex(data = p16, fill = "lightgrey", alpha = 0.6) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = as.vector(eschwcol[c(3,1,2)])) +
  theme(axis.title.x = element_blank()) +
  ylab("Delta DBH")
g.var.NCI.RE <- ggplot(t, aes(RE, NCI/10^4, size = DBHtoday)) +
  # geom_hex(data = p16, fill = "lightgrey", alpha = 0.6) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = as.vector(eschwcol[c(3,1,2)])) +
  theme(axis.title.x = element_blank()) +
  ylab("NCI")
# all
cowplot::plot_grid(g.RE, g.a.RE.NCI, g.a.RE.Gmax,
                   g.var.NCI.RE, g.NCI, g.a.NCI.Gmax,
                   g.var.DBH.RE, g.var.DBH.NCI, g.gmax,
                   ncol=3, labels = LETTERS[1:9])
# summary(MASS::rlm(formula = aGmax ~ aNCI, data = t))
# summary(robustbase::lmrob(formula = aGmax ~ aNCI, data = t))
```

```{r}
fitNCImean <- list()
for(sim in list.files("parvicapture_save/all/EnvGeno2", 
                      pattern = "NCImean", full.names = T)){
  load(sim)
  fitNCImean <- c(fitNCImean, fit)
}
fitNCImean <- sflist2stanfit(fitNCImean)
fitNCI0 <- list()
for(sim in list.files("parvicapture_save/all/EnvGeno2", 
                      pattern = "NCI0", full.names = T)){
  load(sim)
  fitNCI0 <- c(fitNCI0, fit)
}
fitNCI0 <- sflist2stanfit(fitNCI0)
fitNCItoday <- list()
for(sim in list.files("parvicapture_save/all/EnvGeno2", 
                      pattern = "NCItoday", full.names = T)){
  load(sim)
  fitNCItoday <- c(fitNCItoday, fit)
}
fitNCItoday <- sflist2stanfit(fitNCItoday)
fitRE <- list()
for(sim in list.files("parvicapture_save/all/EnvGeno2", 
                      pattern = "RE", full.names = T)){
  load(sim)
  fitRE <- c(fitRE, fit)
}
fitRE <- sflist2stanfit(fitRE)
t <- lapply(c(fitNCImean, fitNCI0,
              fitNCItoday, fitRE, fitGmaxGeno), function(fit){
  d <- as.data.frame(fit, "alog") %>%
    summarise_all(median) %>%
    reshape2::melt(NULL) # median
  if(nrow(d) == 257) d <- d[-nogrwoth,]
  d <- mutate(d, IndNum = 1:nrow(d))
  return(d)}) %>% 
  bind_rows(.id = "fit") %>% 
  mutate(fit = recode(fit, "1" = "aNCImean", "2" = "aNCI0", "3" = "aNCItoday",
                      "4" = "aRE", "5" = "aGmax")) %>% 
  group_by(fit) %>% 
  mutate(value = exp(value)) %>% 
  reshape2::dcast(IndNum ~ fit, value.var = "value") %>% 
  cbind(subtrees) %>% 
  mutate(pop = recode(cluster, "1" = "E. coriacea cluster", 
                      "2" = "E. sagotiana cluster",
                      "3" = "E. decolorans cluster"))
n <- 5
t %>% 
  dplyr::select(aGmax, aNCImean, aNCI0, aNCItoday, DBHtoday, pop) %>% 
  reshape2::melt(c("aGmax", "DBHtoday", "pop")) %>% 
  group_by(variable) %>% 
  arrange(DBHtoday) %>% 
  mutate(DBHclass = rep(1:n, each = ceiling(n()/n))[1:n()]) %>% 
  ungroup() %>% 
  ggplot(aes(aGmax, value, size = DBHtoday, col = pop)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "lm", se = F) +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(values = as.vector(eschwcol[c(3,1,2)])) +
  theme(axis.title = element_blank()) +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = c(0.8, 0.85, 0.9)) +
  facet_grid(variable ~ DBHclass, scales = "free") +
  theme(legend.position = "bottom")
```


```{r spcorrelo, fig.cap="Spatial autocorrelogram (Moran\'s I) of variables and associated genetic multiplicative values."}
cor <- lapply(c("RE", "NCImean", "aRE", "aNCImean", "aGmax"), function(var) 
  pgirmess::correlog(data.frame(t$Xutm, t$Yutm), t[,var], method = "Moran", nbclass = 70) %>% 
    as.data.frame())
cor %>% 
  bind_rows(.id = "var") %>% 
  mutate(var = recode(var, "1" = "RE", "2" = "NCImean", "3" = "aRE", "4" = "aNCImean", "5" = "aGmax")) %>% 
  filter(dist.class < 10^3) %>% 
  ggplot(aes(x = dist.class, y = coef, col = var)) + geom_hline(yintercept = 0) +
  geom_point(aes(alpha = p.value < 0.01)) + geom_line() +
  scale_alpha_manual("p < 0.01", values = c(0.3, 1)) + scale_x_log10() +
  geom_vline(xintercept = c(27, 53), col = "lightgrey", linetype = "dashed") +
  geom_vline(xintercept = 150, col = "lightgrey", linetype = "dotted") +
  xlab("Distance (m)") + ylab("Moran\'s I") + scale_color_discrete("Variable") 
```


## Gmax, Genotype & Environment

We used the following growth model with a lognormal distribution to estimate individual growth potential and associated genotypic variation:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \theta_{1,p,i}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\theta_{2,p}})}{\theta_{3,p}}]^2)), \sigma_1) \\ 
  \theta_{1,p,i} \sim \mathcal {logN}(log(\theta_{1,p}.a_{1,i}. \beta_1.RE_i.\beta_2.NCI_i), \sigma_2) \\ 
  \theta_{2,p} \sim \mathcal {logN}(log(\theta_2),\sigma_3) \\ 
  \theta_{3,p} \sim \mathcal {logN}(log(\theta_3),\sigma_4) \\
  a_{1,i} \sim \mathcal{MVlogN}(log(1), \sigma_5.K)
  (\#eq:gmaxgenoenv)
\end{equation} 

We fitted the equivalent model with following priors:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \hat{\theta_{1,p,i}}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\hat{\theta_{2,p}}})}{\hat{\theta_{3,p}}}]^2)), \sigma_1) \\ 
  \hat{\theta_{1,p,i}} = e^{log(\theta_{1,p}.\hat{a_{1,i}}. \beta_1.RE_i.\beta_2.NCI_i) + \sigma_2.\epsilon_{1,i}} \\ 
  \hat{\theta_{2,p}}  = e^{log(\theta_2) + \sigma_3.\epsilon_{2,p}} \\ 
  \hat{\theta_{3,p}}  = e^{log(\theta_3) + \sigma_4.\epsilon_{3,p}} \\  
  \hat{a_{1,i}} = e^{\sigma_5.A.\epsilon_{4,i}} \\
  \epsilon_{1,i} \sim \mathcal{N}(0,1) \\
  \epsilon_{2,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{3,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{4,i} \sim \mathcal{N}(0,1) \\
  ~ \\
  (\theta_{1,p}, \theta_2, \theta_3) \sim \mathcal{logN}^3(log(1),1) \\ 
  (\sigma_1, \sigma_2, \sigma_3, \sigma_4, \sigma_5) \sim \mathcal{N}^5_T(0,1) \\ 
    ~ \\ 
    V_P = Var(log(\mu_p)) \\
    V_G=\sigma_5^2\\
    V_{TWI} = Var(log(\beta_1.TWI_i)) \\
    V_{NCI} = Var(log(\beta_2.NCI_i)) \\
    V_R=\sigma_2^2
  (\#eq:gmaxgenoenvstan)
\end{equation} 

```{bash fitGmaxGenoEnvCluster, eval=F}
for chain in $(seq 8) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GmaxGenoEnv.R $chain" ; done > GmaxGenoEnv.sh
sarray -J GmGE -o out/%j.GmGE.out -e out/%j.GmGE.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GmaxGenoEnv.sh
watch 'tail -n 1 out/*.GmGE.out | grep Iteration'
```

```{r gmaxgenoenvTab}
fitGmaxGenoEnv <- list()
for(sim in list.files("parvicapture_save/all/GmaxGenoEnv", full.names = T)){
  load(sim)
  fitGmaxGenoEnv <- c(fitGmaxGenoEnv, fit)
}
fitGmaxGenoEnv <- sflist2stanfit(fitGmaxGenoEnv)
broom::tidyMCMC(fitGmaxGenoEnv, pars = c("thetap1", "beta", 
                                         "theta2", "theta3",
                                         "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))  
```

```{r gmaxgenoenvTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmaxGenoEnv, pars = c("thetap1", "beta",
                                           "theta2", "theta3",
                                          "sigma", "lp__")),
           np = nuts_params(fitGmaxGenoEnv)) +
  viridis::scale_color_viridis(discrete = T)  
```

```{r gmaxgenoenvPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmaxGenoEnv, c("sigma"))  
```

```{r gmaxgenoenvEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmaxGenoEnv))  
```

```{r gmaxgenoenvPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "E. coriacea cluster"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[1]"))))/
                 median(as.array(fitGmax, "thetap3[1]")))^2)) +
    stat_function(aes(col = "E. sagotiana cluster"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[2]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[2]"))))/
                 median(as.array(fitGmax, "thetap3[2]")))^2)) +
    stat_function(aes(col = "E. decolorans cluster"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[3]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[3]"))))/
                 median(as.array(fitGmax, "thetap3[3]")))^2)) +
  scale_color_manual("Cluster", values = as.vector(eschwcol)[c(3,1,2)])
```

```{r GmaxGenoEnvR2, fig.cap="R2 for Gmax."}
as.data.frame(fitGmaxGenoEnv, c("Vp", "Vg", "Vnci", "Vre", "Vr")) %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vg, Vnci, Vre, Vr))) %>% 
  mutate(Vexp = sum(c(Vp, Vg, Vnci, Vre))) %>% 
  mutate_at(c("Vp", "Vg", "Vnci", "Vre", "Vexp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = NULL) %>% 
  group_by(variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, 
                           "Vexp" = "Marginal", "Vg" = "Genotype", 
                           "Vnci" = "NCI", "Vre" = "RE",
                           "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +
  theme(axis.title.y = element_blank()) +
  coord_flip()  
```

```{r gmaxgenoenvVar, fig.cap="Genetic variance partitionning for ontogenetic parameters."}
mcmc_intervals_data(fitGmaxGenoEnv, regex_pars = c("Vp", "Vg", "Vnci", "Vre", "Vr")) %>% 
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", 
                           "Vnci" = "NCI", "Vre" = "RE", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2))   
```
