```{r setupA4, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
library(tidyverse)
library(bayesplot)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, fig.height = 6, fig.width = 6, cache = T)
load("rdata/Ch3SupMat.Rdata")
```

# Appendix 4: Chapter 3 Supplementary Materials

Article title: **Topography drives microgeographic adaptations among closely-related species of two tropical tree species complexes**

Authors: Sylvain Schmitt, Niklas Tysklind, Bruno HÃ©rault, Myriam Heuertz

Article acceptance date: unknown

The following Supporting Information is available for this article:

**Method S[1](#A4methodS1).** Design of the probes set for *Symphonia*.

**Method S[2](#A4methodS2).** Design of the probes set for *Eschweilera*.

**Model S[1](#A4model).**  Stan code for the animal model

**Tab. \@ref(tab:A4kmeansConfusion).** *Eschweilera* botanical species and genetic clusters

**Fig. \@ref(fig:A4symcapture).** Target selection for the capture experiment of *Symphonia*

**Fig. \@ref(fig:A4parvicapture).** Target selection for the capture experiment of *Eschweilera*

**Fig. \@ref(fig:A4libmiss).** SNP abundance per library for *Eschweilera*

**Fig. \@ref(fig:A4snpmiss).** Library abundance per SNP for *Eschweilera*

**Fig. \@ref(fig:A4outgroup).** Outgroup detection for *Eschweilera*

**Fig. \@ref(fig:A4admixtureSympoCV).** Cross-validation for *Symphonia* population structure

**Fig. \@ref(fig:A4admixtureSympo).** *Symphonia* population structure using admixture

**Fig. \@ref(fig:A4introgress).** *Symphonia* population structure using hybrid index 

**Fig. \@ref(fig:A4bayescanOutliers).** Outlier SNPs among *Symphonia* species

\newpage


## Method S1: Design of the probes set for *Symphonia*. {#A4methodS1}

For *Symphonia globulifera*, the genomic and transcriptomic resources used for the design were comprised of a published low-coverage draft genome from Africa [@Olsson2017], an unpublished draft genome from French Guiana [Scotti et al., in prep], an unpublished transcriptome from 20 juveniles from French Guiana [Tysklind et al., in prep], and reduced-representation genomic sequence reads of individuals from French Guiana [Torroba-Balmori et al., unpublished]. 
We aligned genomic reads on the two genome drafts with `bwa` [@Li2009]. 
We kept scaffolds from the two genome drafts with a length superior to 1 kbp and at least one matching alignment with a read with a single match on the genome, and merged the two filtered genome drafts with `quickmerge` [@Chakraborty2016]. 
We aligned transcripts on the new filtered genome draft with `BLAT` [@Kent2002] and selected 533 scaffolds without transcript-match, *i.e.* anonymous scaffolds. 
We masked repetitive regions with `RepeatMasker` [@Smit2015] and selected 533 1-kbp anonymous loci within the 533 previous scaffolds.

Similarly, we filtered transcripts from the 20 juveniles of *Symphonia globulifera* from French Guiana [Tysklind et al., in prep] based on SNP quality, type and frequency. 
We further detected  open reading frames (ORFs) using `transdecoder` [@Haas2013], 
and selected transcripts with non-overlapping  ORFs including a start codon.
We kept ORFs with an alignment on scaffolds from the aforementioned genome draft for *Symphonia* using `BLAT` [@Kent2002],
and masked repetitive regions with `RepeatMasker` [@Smit2015]. 
We selected 1,150 genic loci of 500-bp to 1-kbp, from 100 bp before the start to a maximum of 900 bp after the end of the ORFs, resulting in 1-Mbp genomic loci that included a coding region.

\newpage

## Method S2: Design of the probes set for *Eschweilera*. {#A4methodS2}

For *Eschweilera*, the genomic and transcriptomic resources used for the design were comprised of transcriptomes from  *Eschweilera sagotiana* and *Eschweilera coriacea* [@Vargas2019], and unpublished reduced representation genomic reads (M. Heuertz pers. com.).
We mapped reciprocally *E. coriacea* and *E. sagotiana* transcriptomes using `BLAT` [@Kent2002], and in reciprocal best matches, we kept a single transcript to avoid paralogs and have robust targets among species.
We further detected  open reading frames (ORFs) using `transdecoder` [@Haas2013], 
and selected transcripts with non-overlapping  ORFs including a start codon.
We selected 1,530 transcriptomic loci of 500-bp to 1-kbp, from 100 bp before the start to a maximum of 900 bp after the end of the ORFs, resulting in 0.83-Mbp of transcriptomic loci. 
To build anonymous targets, we built a *de novo* assembly of ddRAD-seq genomic data using `ipyrad` [@Eaton2020], mapped consensus sequences on transcripts using `BLAT` [@Kent2002], and kept consensus sequences with no match on transcripts.
We masked repetitive regions with `RepeatMasker` [@Smit2015] and selected  2.2k anonymous loci resulting in a length 0.52-Mbp.

\newpage

## Model S1: Stan code for the bayesian inference of the animal model. {#A4model}

```{stan A4model, eval=F, echo=T, output.var="m", cache=F}
data {
  int<lower=0>  N ; // # of individuals
  int<lower=0>  P ; // # of populations
  real y[N] ; // phenotype
  int<lower=1, upper=P> population[N] ; // populations
  cov_matrix[N] K ; // kinship covariance matrix
}
transformed data{
  matrix[N, N] A = cholesky_decompose(K) ; // cholesky-decomposed kinship
  real Vy = variance(log(y)) ;
}
parameters {
  vector<lower=0>[P] mu ; // intercept
  vector[N] epsilon ; // genotypic noise
  real<lower=0, upper=sqrt(Vy)> sigma ; // genetic variance
}
transformed parameters {
  real<lower=0> Vp = variance(log(mu[population])) ; // population variance
  real Vr = square(sigma) ;
  real Vg = Vy - Vp - Vr ;
  vector[N] alog = sqrt(Vg)*A*epsilon ;
}
model {
  y ~ lognormal(log(mu[population]) + alog, sigma) ;
  epsilon ~ std_normal() ;
  mu ~ lognormal(0, 1) ;
  sigma ~ normal(0, 1) ;
}
```

\newpage

```{r A4kmeansConfusion}
path <- "~/Documents/BIOGECO/PhD/data/Eschweilera_Paracou/Sequences/genomics"
read_tsv(file.path(path, "..", "variants", "final", "filtered.kmeans")) %>% 
    mutate(cluster = recode(cluster, "3" =  "E. decolorans cluster", 
                          "2" = "E. sagotiana cluster",
                          "1" = "E. coriacea cluster")) %>% 
  separate(Sample, paste0("X", 1:7), "[:punct:]", remove = F, convert = T) %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  left_join(individuals) %>% 
  filter(Species %in% c("coriacea", "decolorans", "sagotiana")) %>%
  reshape2::dcast(Species ~ cluster) %>% 
  mutate_if(is.numeric, funs(round(./sum(.)*100))) %>% 
  mutate(Species = paste0("E. ", Species)) %>% 
  kable(caption = "Confusion matrix (percentage) between genetic cluster and botanical identification for \\emph{Eschweilera} clade \\emph{Parvifolia}.",
        format = "pandoc")
```

\newpage

```{r A4symcapture, fig.cap="Scheme of target selection for the capture experiment of \\emph{Symphonia globulifera} as described in the manuscript.", out.width="\\linewidth", fig.align="center", fig.pos="H"}
include_graphics("../genomics/topography/figures/Symcapture.png")
```

\newpage

```{r A4parvicapture, fig.cap="Scheme of target selection for the capture experiment of \\emph{Eschweilera} clade \\emph{Parvifolia} as described in the manuscript.", out.width="0.5\\linewidth", fig.align="center", fig.pos="H"}
include_graphics("../genomics/topography/figures/Parvicapture.png")
```

\newpage

```{r A4libmiss, fig.cap="SNP abundance per library rank for raw data of \\emph{Eschweilera} clade \\emph{Parvifolia}. Dashed lines represent tested filters.", fig.align="center", fig.pos="H"}
ggplot(libmiss, aes(libraries, snps)) + 
  geom_point() + 
  geom_line() +
  geom_vline(xintercept = Nlibs, linetype = "dashed")  +
  geom_hline(yintercept = libmiss$snps[Nlibs], linetype = "dashed") +
  xlab("number of libraries ranked by number of called SNPs") +
  ylab("number of called SNPs")
```

\newpage

```{r A4snpmiss, fig.cap="Library abundance per SNP rank for library-filtered data of \\emph{Eschweilera} clade \\emph{Parvifolia}. Dashed lines represent tested filters.", fig.align="center", fig.pos="H"}
ggplot(snpmiss, aes(snps, libraries, col = gsub("filtered.", "", filter))) + 
  geom_point() + 
  geom_line() +
  scale_x_sqrt(n.breaks	= 10) +
  scale_y_log10() +
  scale_color_discrete("Libraries filter\n(minimum missing\ndata rate)") +
  geom_hline(yintercept = c(5, 10), linetype = "dashed") +
  geom_vline(aes(xintercept = snps), linetype = "dashed", data = snpmissN) +
  xlab("number of filtered SNPs ranked by number of genotyped libraries") +
  ylab("number of libraries")
```

\newpage

```{r A4outgroup, fig.cap="Outgroup detection with \\emph{Eschweilera} individual clustering in the genomic principal component analysis (gPCA) in two groups using K-means for every filter."}
group_by(pcas, filter) %>% 
  filter(sum(PCA1) != 0) %>% 
  cbind(group_by(pcas, filter) %>% 
          filter(sum(PCA1) != 0) %>% 
          dplyr::select(., PCA1, PCA2) %>% 
          do(hc = cutree(hclust(dist(as.matrix(.))), k = 2)) %>% 
          broom::tidy(hc)) %>% 
  dplyr::rename(cluster = x) %>% 
  group_by(filter, cluster) %>% 
  mutate(N = n()) %>%
  group_by(filter) %>% 
  mutate(Nmin = min(N)) %>% 
  mutate(cluster = ifelse(N == Nmin, "outgroup", "parvifolia")) %>% 
  dplyr::select(-N, -Nmin) %>% 
  ungroup() %>% 
  ggplot(aes(x = PCA1, y = PCA2, col = cluster, 
                 label = Sample)) +
      geom_line(aes(group = IdGenetic)) +
      geom_point(size=2, alpha = 0.5) +
      scale_color_discrete("") +
      theme(legend.position = "bottom") +
      facet_wrap(~ filter, scales = "free") +
      theme(strip.text.x = element_text(size = 6))
```

\newpage

```{r A4admixtureSympoCV, fig.cap="Cross-validation for the clustering of \\emph{Symphonia} individuals using admixture. Y axis indicates cross-validation mean error, suggesting that K = 2 or K = 3 gene pools represent the best solution for genetic structure in \\emph{Symphonia} in Paracou."}
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
read_delim(file.path(path, "admixture", "paracou", "CV.out"), delim = " ", col_names = F) %>% 
  dplyr::select(X3, X4) %>% 
  dplyr::rename(K = X3, CV = X4) %>% 
  mutate(K = gsub("(K=", "", K, fixed = T)) %>% 
  mutate(K = as.numeric(gsub("):", "", K))) %>% 
  ggplot(aes(K, CV)) +
  geom_point() +
  geom_line() +
  geom_vline(xintercept = c(2,3), col = "red", linetype = "dashed") +
  ylab("Cross-validation error")
```

\newpage

```{r A4admixtureSympo, fig.cap="Population structure of \\emph{Symphonia} individuals from K=2 to K=10 using admixture."}
fam <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID))
symcapture.admix <- pophelper::readQ(list.files(file.path(path, "admixture", "paracou"), 
                                     full.names = T, pattern = ".Q"), indlabfromfile=F)
symcapture.admix <- lapply(symcapture.admix, "rownames<-", fam$Ind)
symcapture.admix <- pophelper::alignK(symcapture.admix)
p <- pophelper::plotQ(symcapture.admix[2:10], exportplot = F, returnplot = T, imgoutput = "join",
                      basesize = 11, splab = paste0("K=",2:10),  showindlab = F, useindlab = F, 
                      grplabsize = 4,linesize = 0.8, pointsize = 4, sortind = 'all', sharedindlab = F)
gridExtra::grid.arrange(p$plot[[1]]) 
```

\newpage

```{r A4introgress, fig.cap="Genotypinc constitution of \\emph{Symphonia} using hybrid index. Admixture coefficients (black line) are given with 90% confidence interval (light shade). Admixture coefficients of 10 and 90% are indicated by white stippled lines."}
hi.index %>% 
  mutate_at(c("h", "lower", "upper"), ~ifelse(pair == "pr", 1-., .)) %>% 
  mutate(pair = recode(pair, 
                       sp = "S. sp1 - S. globulifera type Paracou", 
                       sr = "S. sp1 - S. globulifera type Regina",
                       pr = "S. globulifera type Paracou - S. globulifera type Regina")) %>% 
  left_join(symcapture.admix[[3]] %>% 
              rownames_to_column(var = "Ind") %>%
              reshape2::melt(id.vars = "Ind", variable.name = "genepool") %>%
              mutate(genepool = recode(genepool, Cluster1 = "S. sp1", 
                                       Cluster2 = "S. globulifera type Paracou", 
                                       Cluster3 = "S. globulifera type Regina"))) %>%
  arrange(pair, desc(h)) %>%
  mutate(order = 1:nrow(.)) %>% 
  ggplot(aes(reorder(Ind, order), group = NA)) +
  geom_col(aes(y = value, fill = genepool, col = genepool), position = position_stack(reverse = TRUE)) +
  geom_hline(yintercept = c(0.1, 0.9), linetype = "dashed", col = "white") +
  geom_ribbon(aes(ymin = 1 - lower, ymax = 1 - upper), alpha = 0.5, fill = "white") +
  geom_line(aes(y = 1 - h)) +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        axis.title.x = element_blank(), axis.line.x = element_blank()) +
  scale_fill_manual("Genepool", values =  c("#CD2626", "#1E90FF", "#0000EE")) +
  scale_color_manual("Genepool", values =  c("#CD2626", "#1E90FF", "#0000EE")) +
  ylab("hybrid index") +
  facet_wrap(~ pair, nrow = 3) +
  theme(legend.position = "bottom") 
```

\newpage

```{r A4bayescanOutliers, fig.cap="High-differentiation outlier SNPs for \\emph{Symphonia} individuals detected with bayescan. We used the genome-transcriptome alignments built for the design of probes sets (Method S1) to classify called SNPs into (i) anonymous SNPs (on scaffolds matching no transcripts), (ii) putatively-hitchhiker SNPs (close to a transcript or within an intron), and (iii) genic SNPs (within an exon)."}
ggplot(bayescan, aes(snpNb, -log10(qval), alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ylab(expression(-log[10]("q-value"))) +
  xlab("SNP position (contigs are arbitrarily placed)") +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  ggtitle("4 020 significant SNPs (5.7%)", 
          "309 functional (2.5%), 2456 hitchhiker (7.0%), and 1255 neutral (5.4%)")
```

