---
title: Data
author: Sylvain SCHMITT
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    theme: journal
    toc: yes
    toc_depth: 5
    toc_float: yes
csl: /home/sylvain/Documents/Bibliography/csl/mee.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
gdoc_id: 1gc1o_7xY37Z8iXluvRAAX_yuoJYBFyHQGGK6gNwR_Wk
---
   
# Genetic

## INIA Madrid and INRA-Biogeco research (Myriam) 

* ***Symphonia globulifera*** ribosomal (ITS) and chloroplast DNA sequences (11 inds.) and 5 nuclear microsatellite (nSSR) loci, 30 inds., including **Paracou** [@Dick2008].
* ***S. globulifera*** fragmented genome draft (Cameroonian pop) + S. globulifera 20 new SSRs based on transcriptome vs. genome alignment; transcriptome from @Morozova2009 [@Olsson2017]. These new SSRs were tested in **Paracou** (32 inds: 17 S. globulifera, 15 S. sp1; 18 SSRs) and they distinguish *Symphonia globulifera* and *S. sp1* in STRUCTURE or Rst-based differentiation analysis. Microsatellite genotypes are archived in the Dryad Digital Repository with link http://dx.doi.org/10.5061/dryad.78ng1
* ***S. globulifera* genotyping-by-sequencing (GbS)** data, 9 populations including spatially explicit sampling in some pops: inter-population analysis to be included in the thesis of Paloma Torroba-Balmori; data can be reanalysed for intra-population level. This includes **Paracou** (33 *S. globulifera*, 33 *S. sp1*) and **Regina** (10 *S. globulifera*, 10 *S. sp1*) , individuals overlap at least partially with @Olsson2017, and @Torroba-Balmori2017.
* 20 transcriptomes of *Symphonia spp* (18 accessions from Madagascar) and *S. globulifera* (2 accessions from Cameroon) (Illumina data): these are being used by Sanna Olsson for phylogenomics, i.e. speciation history in Madagascar.
* ***S. globulifera* SNPs** (Sequenom, ca. 140 SNPs) developed on transcriptomes: > 600 individuals, including ca. 400 *Symphonia spp* from Madagascar and *S. globulifera* from both continents (**probably Paracou, to verify**). Madagascar data for the thesis of Paloma Torroba-Balmori. No publication plan for outside Madagascar, but will be explored by Master student Janan Gawra to start in February 2018.
* ***S. globulifera* cpDNA dataset**: Sanger sequencing of psba-trnH region in 48 *S. globulifera* and 48 *S. sp1*; re-analysis of SSR data: 3 loci, 148 inds. of which 9 *S. globulifera*, 90 *S. sp1* and 49 indet. from @Degen2004 in **Paracou** [@Torroba-Balmori2017]. Data in the sup mat of the paper, S1_file for SSRs, S3_file for cpDNA (haplotype identification and Genbank accession number)).
* **Lecythidaceae**: ddRAD-Seq in 8 individuals, 2 individuals in each of 4 species (*Eschweilera coriacea*, *E. sagotiana*, *E. decolorans*, *E. congestiflora*), for marker development, development of resequencing markers. Master thesis of Quentin Jehanne 2016, Univ. Bordeaux. 
* 151 resequencing markers, ca. 180 bp each, on **192 individuals of Lecythidaceae** species from the Bertholletia clade. Includes ca. 130 nuclear markers, 10 cpDNA and 10 mitochondrial markers. Analysis currently conducted by Myriam Heuertz and Katharina Budde. Below are results of STRUCTURE cluster analysis for 40 individuals from Paracou and Piste de St. Elie (preliminary; some more individuals available in the complete dataset). It is visible that ***E. coriaceae*** contains four (!) discrete gene pools.  
![STRUCTURE cluster analysis for 40 individuals](image/data1.png)
* **Lecythidaceae transcriptomes** produced for 5 individuals: *Lecythis persistens*, *Eschweilera congestifolia*, *E. coriaceae* (2 inds), *E. sagotiana*. Data with Oscar Vargas at Univ. Michigan for transcriptome assembly.

## Ecofog research (Niklas)

* ***Symphonia globulifera* transcriptome** from Roche 454 sequencing: @Morozova2009.
* **Symphonia* exome capture** in French Guiana (2 pops, Paracou and Regina) and in Cameroon (2 pops, Nkong Mekak and Mbikiliki, 50 inds. per pop, see @Torroba-Balmori2017), ANR-project FLAG coord. by Ivan Scotti, data analysis currently by Caroline Scotti-Saintagne and Hadrien Lalagüe. I don’t know sample sizes in French Guiana.
* ***Symphonia* AFLP and CROPS** (Complexity reduction of polymorhic sequence markers), Caroline Scotti-Saintagne et al.:
![](image/data2.png)
* Lecythidaceae plastid DNA data available from Henri Caron. Myriam Heuertz has raw data. 
* SSR data on Lecythidaceae to be produced at Ecofog on ca. 500 individuals in the COLLEVOL project, funded as annual CEBA project 2016.

##LECYTOMICS ? (Guyamazon project to be deposited end of Sept 2017)

Sequence capture in Piste de St Élie and Paracou ? Objective 2 to be reoriented towards focus on Paracou ecological genomics.

# Phenotype

## BRIDGE

Bridge compile several functional traits for different species from Paracou, Nouragues, Acarouany and BAFOG. Several individuals were used to calculate species functional traits allowing us to get individuals trait values.

* Traits table ?
* Species of interest for the PhD ?
* Individuals in Paracou ?
* Individuals with growth data ?
* Individuals with genetic data ?

## Sébastien Levionnois PhD

Sébastien Levionnois plans to study individuals hydraulic, and more particularly *Symphonia*.

## Louis Santiago

He apparently have also individuals hydraulic traits for some *lecythidaceae*.

## Patrick Heuret

Patrick have apparently collected functional traits data concomitant with genetic data. Patrick’s data is available for the Lecythidaceae from Paracou collected in 2015 by Patrick, Myriam, Niklas, Pascal. It’s 29 individuals belonging to nine species, so there are very probably too few data points per species (max. 4 inds/species).

# Performance

## Paracou

Tree diameter at breast height (dbh) censused for tree above 10 cm dbh every 1 or 2 years for 30 years.

## Guyafor ?

Other plot with tree dbh censused in french Guyana.

## Manaus team ?

The team censused 2 times tree dbh.

# Environment

## Paracou

The site undergone numerous studies and its related to numerous environmental data as topography, soil content, pH...

# References