---
title: Topography drives microgeographic adaptations among closely-related species of two tropical tree species complexes
date: '`r Sys.Date()`'
output:
  bookdown::html_document2:
    number_sections: false  
    toc: true
    toc_float: yes
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
  bookdown::word_document2:
    reference_docx: ./template/template.docx
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/molecol.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

**Running title: Microgeographic adaptations to topography**

1. Sylvain Schmitt
    - Corresponding author
    - sylvain.m.schmitt@gmail.com
    - orcid.org/0000-0001-7759-7106
    - Univ. Bordeaux, INRAE, BIOGECO, 69 route d’Arcachon, 33610 Cestas France
1. Niklas Tysklind
    - Niklas.Tysklind@ecofog.gf
    - INRAE, UMR EcoFoG (Agroparistech, CNRS, Cirad, Université des Antilles, Université de la Guyane), Campus Agronomique, 97310 Kourou, French Guiana
1. Bruno Hérault
    - bruno.herault@cirad.fr
    - orcid.org/0000-0002-6950-7286
    - CIRAD, UPR Forêts et Sociétés, Yamoussoukro, Côte d’Ivoire
    - Forêts et Sociétés, Univ Montpellier, CIRAD, Montpellier, France
    - Institut National Polytechnique Félix Houphouët-Boigny, INP-HB, Yamoussoukro, Côte d’Ivoire
1. Myriam Heuertz
    - heuertzm@gmail.com
    - orcid.org/0000-0002-6322-3645
    - Univ. Bordeaux, INRAE, BIOGECO, 69 route d’Arcachon, 33610 Cestas France

\newpage 

<!-- Formatting -->
<!-- Limit of 8000 words per paper, excluding references, currently . -->
<!-- All manuscripts must have: -->
<!-- Page numbers -->
<!-- Continuous line numbers -->
<!-- Double-spaced text -->
<!-- Single-spacing for: -->
<!-- Table and figure captions -->
<!-- References -->
<!-- Appendices -->
<!-- Supporting/Supplemental information -->
<!-- Side margins 2.5 cm side margins, top and bottom margins 3 cm -->
<!-- clear paragraph delimitations -->
<!-- figures of sufficiently high quality for review -->


# Abstract

<!-- _**About 250 words.**_ -->

Despite the local abundance and regional success of closely-related tree species growing in sympatry in Neotropical species complexes,
little is known of the eco-evolutionary forces responsible for their local coexistence.
Here, we assessed genetic species delimitation and genotypic diversity of closely-related sympatric tree species belonging to two Neotropical species complexes, and addressed their fine-scale spatial adaptation along a topography gradient with underlying edaphic and hydrologic gradients.
Combining tree inventories, LiDAR-derived topographic data, and single nucleotide polymorphisms (SNPs) from gene capture experiments,
we showed that within tree species complexes, closely-related species have different realized optima for topographic niches with species genetic adaptations.
One species complex showed species differentially adapted to water and nutrient distribution with topographic wetness and relative elevation; whereas the second species complex had species differentially adapted to soil chemistry due to hydromorphy.
Our results thus suggest that species adaptations to different characteristics of topography stabilize local coexistence between closely-related species within tree species complexes.

## Keywords

<!-- _**Please provide four to six keywords.**_ -->
syngameon; topographic wetness index; relative elevation; ecological niche; Paracou; tropical forests

\newpage 

# Introduction 

The tremendous biodiversity of tropical forests [@Gaston2000] has intrigued biologists for decades [@connell_diversity_1978].
Lowland Amazonia alone is estimated to shelter around sixteen thousand tree species [@TerSteege2013]. 
But the five thousand species of trees botanically described in Amazonia belong to only height hundred ten genera [@TerSteege2013],
which thus include many species-rich genera. 
Even at the hectare-scale, tropical forests shelter up to several hundred tree species [@Gentry1988], 
including species-rich genera with closely-related species coexisting in sympatry [@Caron2019]. 
Closely-related species are expected to share similar niche and functional strategies due to phylogenetic constraints [@Wiens2010].
These niche and functional similarities are expected to lead to increased competition between closely-related tree species and ultimately to local competitive exclusion of one species by the other [but see competitive combining, @Aarssen1983].
The persistence in sympatry of closely related tree species represents thus an intriguing paradox. 
Despite the local abundance and regional success of closely-related tree species growing in sympatry in the Neotropics [@Gentry1988; @TerSteege2013; @Pinheiro2018],
little is known of the eco-evolutionary forces responsible for their local coexistence.

Genetic studies have shown that species in species-rich genera have a higher level of genetic polymorphism [@Caron2019] 
and hybridize more often between congeners [@Whitney2010] than those in species-poor genera. 
Species-rich genera thus frequently contain species complexes, defined as clades of morphologically similar species and/or species sharing large amounts of genetic variation due to recent common ancestry and/or hybridization [@Pernes1984].
Species-rich genera also frequently contain syngameons, defined as species connected by limited, but recurrent, interspecific gene flow [@Suarez-Gonzalez2018].
Syngameons are evolving at two contrasting taxonomic levels, they maximize each species adaptation to its ecological niche, and they evolve at the syngameon level which benefits all the constituent species. 
As a whole, they might decrease the overall risk of syngameon extinction, while allowing ecological adaptation of their species [@Cannon2015b].
Consequently, syngameon are not necessarily a transitional stage before complete speciation but may represent a successful evolutionary state [@Cannon2019a].
Closely related species from species complexes or syngeamons are especially more susceptible to phylogenetic constraints due to their low genetic differentiation, however against theoretical expectation, they frequently include sympatric species coexisting at local scales [@Caron2019; @Gonzalez2009; Schmitt et al., in prep].

To address the paradox of local persistence in sympatry despite similar expected niches, niche theory explains species local coexistence based on ecological niche differences limiting competitive exclusion [@lortie_rethinking_2004-1; @weiher_assembly_1995].
The heterogeneity of resource distribution in space and time defines fine-scale habitat structure where species can coexist.
In tropical forests, topography explains the fine scale spatial distribution of water and nutrient availability [@ferry2010higher].
Topography drives thus pervasive differentiation in habitat preference among species [@gunatilleke_specieshabitat_2006; @Engelbrecht2007; @kraft_functional_2008; @Allie2015]. 
In particular, soil nutrients, also influenced by topography through hydromorphy, directly influence the spatial distribution of forest tree species [@Levins1971].
Topography drives thus functional responses among and within species [@Schmitt2020].
Topography includes many features, such as topographic wetness [@Schmitt2020] or relative elevation [@Allie2015], with specific roles on nutrient and water availability.
Therefore, topography is a good candidate factor that may explain the coexistence of closely-related tree species in Neotropical species complexes [Schmitt et al., in prep].

Evidence exists for eco-evolutionary processes at microgeographic scale, e.g. within the dispersal neighbourhood of the organism [@Richardson2014].
Closely-related species growing in sympatry in differentiated ecological niches form an adaptive radiation, such as Darwin’s finches in the Galapagos [@Seehausen2004]. 
Evolutionary history behind adaptive radiations falls within a continuum from sympatric ecological speciation to secondary contacts of species ecologically specialised in allopatry or parapatry [@rundell_adaptive_2009].
In particular, species complexes can result from adaptive radiations and species segregation along environmental gradients.
Species complexes may combine and reshuffle genetic features among species in hybrid swarms [@Seehausen2004], 
thus multiplying the number of potential ecological niches and helping species to achieve reproductive isolation [@Runemark2019]. 
Indeed, hybrid or derived species require a certain degree of reproductive isolation from their progenitors to avoid becoming an evolutionary melting pot. 
Nevertheless, species-specific adaptations can be maintained or even maximised under gene flow, 
especially with selective pressures varying in space and or time [@Tigano2016].
For instance, the European white oak forms one of the best known syngameon [@Cannon2019a].
European white oak’s species lack private haplotypes indicating extensive hybridisation [@Petit2002], 
but have a unique ecological niche with tolerance to drought, cold and alkaline soils [@Cannon2019a; @Leroy2019]. 
The coexistence of the different species is partly due to the genes that allow them to survive in different ecological niches [@Leroy2019].
To our knowledge, only few studies evidenced adaptive radiation driven by topography or abiotic habitats for tropical trees [@Fine2004; @Paun2016; @Pillon2014].

In the present study, we assessed genetic species delimitation and genotypic diversity of closely-related sympatric tree species belonging to two Neotropical species complexes. 
We addressed fine-scale spatial adaptation along a topography gradient and identified underlying edaphic and hydrologic gradients. 
Combining tree inventories, LiDAR-derived topographic data, and single nucleotide polymorphisms (SNPs) from gene capture experiments, 
we used population genomics, environmental association analyses and genome wide association studies to address the following questions:

1. How is genetic diversity structured among and within species of Neotropical tree species complexes ?
1. How are water and nutrient availability structured at fine spatial scale by the topographic gradients in which individuals grow?
1. Are tree species and individuals adapted to the fine-scale topographic gradients?

We hypothesized species to be delineated into gene pools corresponding to already described taxonomic morphotypes. 
We expected misidentifications due to absence of true diagnostic markers due to overlapping intraspecific morphological variability, and the difficulty to access reproductive material. 
In this context, molecular data reveals the true evolutionary history as it can reveal cryptic species or lead to the merging of  taxa previously thought to be distinct [@Ewedje2020].
We hypothesized relative elevation and topographic wetness to explain the distribution of water and nutrient availability at fine scale [@ferry2010higher; @Allie2015; Schmitt et al. in prep]. 
We hypothesized species to be delimited along topography with fixed neutral and adaptive variants, 
following previous evidence of topography determining species’ ecological niches [Schmitt et al., in prep]. 
But interspecific gene flow within the species complex could reduce species adaptations to topography and blur evolutionary history [@Tigano2016].

# Material and Methods

## Study site

The study was conducted in the Paracou field station, in the coastal forests of French Guiana.
The site is characterized by an average of 3,041 mm annual rainfall and a mean air temperature of 25.71 °C [@Aguilos2018].
Old tropical forest with an exceptional richness (over 750 woody species) has developed among the succession of small hills of this area, rising to 10–40 m a.s.l. [@Gourlet-Fleury2004].
The site is made of 16 permanent plots (fifteen 6.25 ha plus one 25 ha) which have been censused (DBH>10) every 1-2 years for more than 35 years. Nine of the plots were logged and subjected to human-induced disturbance in 1986 [details on the experiment in @Herault2018].

## Plant material

402 individuals from *Symphonia globulifera* (Clusiaceae) 
and 432 individuals belonging to the clade *Parvifolia* of the genus *Eschweilera* [@Huang2015; @Mori2016], 
two Amazonian hyperdominant tree species [@TerSteege2013], 
were sampled in 2017 and 2018 during the dry season (from September to December) in Paracou (Fig. \@ref(fig:maps)).
*Symphonia globulifera* L.f (Clusiaceae) was previously recognized as composed of two morphotypes in French Guiana [@Sabatier1997].
*S. globulifera sensu stricto* and *Symphonia sp.1* occur in sympatry but in differentiated habitats, 
with *S. globulifera* preferentially growing in valley bottoms with an acquisitive functional strategy 
and *S. sp1* preferentially exploiting a variety of drier habitats with a conservative functional strategy [@Allie2015; Schmitt et al., in prep; @Schmitt2020].
Similarly, *E. sagotiana*, *E. decolorans* , and *E. coriacea* exhibit niche differentiation, different water stress, and functional traits along topography [@Allie2015; Schmitt et al., in prep; @Ferry2007; @Schmitt2020].
The genera *Symphonia*  and *Eschweilera* have been highlighted as species complexes with low phylogenetic resolution and high levels of plastid DNA sharing among sister species [@baraloto_using_2012-1; @Caron2019; @Gonzalez2009; @Torroba-Balmori2017; @Heuertz2020; @Huang2015].
In addition, outgroups were built for *Symphonia* with 13 individuals of *Symphonia globulifera* from Africa (Sao Tome, Gabon, Cameroun, Congo, Benin, Liberia, Ivory Coast, and Gana), 7 *Symphonia globulifera* from South America (Brazil, Costa Rica and Panama), 2 *Symphonia nectarifera* from Madagascar, 2 *Symphonia urophylla* from Madagascar, 5 *Pentadesma butyracea* from Benin and Cameroun and 1 *Pentadesma grandifolia* from Cameroun.
And outgroups were built for *Eschweilera* with individuals from other *Eschweilera* clades in Paracou (*L. persistens*, *E. simiorum*, and *E. chartacea*).
Leaves were collected from the 864 individuals (402 + 30 *Symphonia* + 432 *Eschweilera*)  and  desicated using silica gel. 

## Design of sequence capture 

We designed in *silico* two sets of 20,000 80-mer probes sequences, one for the genome of *Symphonia globulifera*, and a second compatible with both *Eschweilera sagotiana* and *Eschweilera coriacea* (Supplementary Material). 

*Symphonia globulifera*

For *Symphonia globulifera*, we used a published low-coverage draft genome from Africa [@Olsson2017], an unpublished draft genome from French Guiana [Scotti et al., in prep], an unpublished transcriptome from 20 juveniles from French Guiana [Tysklind et al., in prep], and genomic reads of individuals from French Guiana [Torroba-Balmori et al., unpublished].
We aligned genomic reads on the two genome drafts with `bwa` [@Li2009]. 
We kept scaffolds from the two genome drafts with a length superior to 1 kbp and at least one matching alignment with a read with a single match on the genome, and merged the two filtered genome drafts with `quickmerge` [@Chakraborty2016]. 
We aligned transcripts on the new filtered genome draft with `BLAT` [@Kent2002] and selected 533 scaffolds without transcript-match, *i.e.* anonymous scaffolds. 
We masked repetitive regions with `RepeatMasker` [@Smit2015] and selected 533 1-kbp anonymous loci within the 533 previous scaffolds.

Similarly, we filtered transcripts from the 20 juveniles of *Symphonia globulifera* from French Guiana [Tysklind et al., in prep] based on SNP quality, type and frequency. 
We further detected  open reading frames (ORFs) using `transdecoder` [@Haas2013], 
and selected transcripts with non-overlapping  ORFs including a start codon (Methionine). 
We kept ORFs with an alignment on scaffolds from the aforementioned genome draft for *Symphonia* using `BLAT` [@Kent2002],
and masked repetitive regions with `RepeatMasker` [@Smit2015]. 
We selected 1,150 genic loci of 500-bp to 1-kbp, from 100 bp before the start to a maximum of 900 bp after the end of the ORFs, resulting in 1-Mbp of transcriptomic loci. 

*Eschweilera*

For *Eschweilera* , we used transcriptomes from  *Eschweilera sagotiana* and *Eschweilera coriacea* [@Vargas2019], and unpublished genomic reads (M. Heuertz pers. com.).
We mapped reciprocally *E. coriacea* and *E. sagotiana* using `BLAT` [@Kent2002], and kept transcript reciprocally matching only one transcript from the other species to avoid paralogs and have robust targets among species.
We further detected  open reading frames (ORFs) using `transdecoder` [@Haas2013], 
and selected transcripts with non-overlapping  ORFs including a start codon (Methionine). 
We selected 1,530 genic loci of 500-bp to 1-kbp, from 100 bp before the start to a maximum of 900 bp after the end of the ORFs, resulting in 0.83-Mbp of transcriptomic loci. 
We built a *de novo* assembly of ddRAD-seq genomic data using `ipyrad` [@Eaton2020], mapped consensus sequences on transcripts using `BLAT` [@Kent2002], and kept consensus sequences with no match on transcripts.
We masked repetitive regions with `RepeatMasker` [@Smit2015] and selected  2.2k anonymous loci resulting in a length 0.52-Mbp.

## Genomic data production

*Libraries preparations*

Genomic DNA was extracted from 5 mg of dried leaf tissue with a CTAB protocol [@Doyle1987]. 
Libraries were digested with 'Ultra II FS Enzyme Mix' (new England Biolabs Inc, MA, USA) for a target size of 150 bp, and built with the 'NEBNext Ultra II FS DNA Library Prep kit for Illumina'(new England Biolabs Inc, MA, USA).
We amplified and tagged libraries using 5 $\mu L$ of adaptor-ligated DNA, 8.3 $\mu L$ of 'NEBNext Ultra II Q5 Master Mix' (new England Biolabs Inc, MA, USA), 2x 1.6 $\mu L$ of Index Primer i5 and i7 from 'NEBNext Multiplex Oligos for Illumina (Dual Index Primers Set 1 and Set 2)' (new England Biolabs Inc, MA, USA). 
Initial denaturation (98°C for 30 s) was followed by 8 cycles (98°C for 10 s and 65°C for 1 min 30 s) and a final extension (65°C for 5 min). 
We pooled libraries in 4 equimolar multiplex and captured sequences by hybridization for 80 hours.
We used the 20,000 80-mer probes sequences designed for the capture using myBaits Custom 1-20K (Arbor Biosciences, MI, USA) and corresponding myBaits V4 protocol. 
We finally pooled and sequenced the previous 4 multiplexes in 2 lanes using Illumina HiSeq 4000 with pair-end for 2x150bp reads.

*Symphonia SNP call and filtering*

We assessed quality from raw reads using `multiqc` [@Ewels2016] and trimmed them with `trimmomatic` [@Andrews2010]. 
We kept only pair-end reads without adaptors and a phred score above 15 in a sliding window of 4.
70% of trimmed reads mapped off-targets using `bwa` [@Li2009]. 
We thus mapped trimmed reads on the hybrid reference built for target design of sequence capture using `bwa` [@Li2009], `picard` [@BroadInstitute2018], `samtools` [@Li2009a] and `bedtools` [@Quinlan2010].
We called variants for each individual using `HaplotypeCaller`, aggregated variants using `GenomicsDBImport` and jointly-genotyped individuals using `GenotypeGVCFs` all in `GATK4` software [@Freksa1993]. 
We filtered biallelic SNPs with a quality above 30, a quality by depth above 2, a Fisher strand bias below 60 and strand odd ratio above 3 using `GATK4` [@Freksa1993]. 
Finally, we filtered individuals and SNPs for missing data with a maximum of 95% and 15% of missing data per individual and SNP, respectively, using `plink2` [@Chen2019].
We obtained 454,262 biallelic SNPs over 406 individuals.

*Eschweilera SNP call and filtering*

We assessed quality from raw reads using `multiqc` [@Ewels2016] and trimmed them with `trimmomatic` [@Andrews2010]. 
We kept only pair-end reads without adaptors and a phred score above 15 in a sliding window of 4.
We kept only trimmed reads mapped on-targets using `bwa` [@Li2009]. 
Due to paralogs issues in *Eschweilera* clade *Parvifolia* due to a strong signature of a past genome duplication [@Heuertz2020],
we decided to build a *de novo* reference of filtered reads using `ipyrad` with a very strict clustering threshold across individuals of 0.95 [@Eaton2020].
We automatically tested numerous values of missing data and outgroup filtering, and kept strict filters with population structure congruent with botanical identification but keeping a maximum of individuals.
We filtered individuals and SNPs for missing data with a maximum of 99% of missing data per SNP and at least 5 individuals represented per SNP using `plink2` [@Chen2019].
Outgroups were automatically filtered out by clustering individuals in the genomic principal component analysis (gPCA) in two groups using K-means.
We obtained 418,793 biallelic SNPs over 257 individuals.

*Symphonia annotation*

We used the previous alignments (see design of sequence capture) from genome drafts [@Olsson2017; Scotti et al., in prep] with the transcriptome [Tysklind et al., in prep] to classify called SNPs into (i) anonymous SNPs (on scaffolds matching no transcripts), (ii) putatively-hitchhiker SNPs (close to a transcript or within an intron), and (iii) genic SNP (within an exon).
We used transcript annotation of gene ontology to further annotate called SNPs. 
We used long ORFs detection with `TransDecoder` [@Haas2013], their alignment with `blastp` on the `Uniprot` database [@Bateman2019; @Bias1991] and with `hmmscan` on `Pfam` database [@Johnson2010; @El-Gebali2019] in addition to called SNPs position into `SNPdat` [@Doran2013] to assess SNP synonymy.

*Symphonia SNP filtering for quantitative genomics*

This first dataset including 454,262 SNPs was used for population genetic and demographic analyses. 
But, minor alleles and linkage disequilibrium bias the number of fixed loci, increasing false-positives upward in genomic scans [@Foll2008].
So we built a second dataset for quantitative genomics, filtering variants with a minor allele frequency above 5% (18 individuals) and with linkage disequilibrium $r^2<0.99$. 
We further removed admixed individuals from genomic scans (see population genetic analyses for admixed individuals definitions). 
We obtained 70,737 biallelic SNPs over 372 individuals.

## Analyses

### Populations

We investigated population structure using `admixture` [@Alexander2011], 
with 10 repetitions of K genetic groups varying from 1 to 10. 
We assessed the number of gene-pools with cross validation.
We defined as admixed individuals with a membership to gene-pools below 90% and the remaining individuals as *pure-bred*. 
We further investigated admixture with `introgress` R package [@Gompert2010], 
using *pure-bred* individuals as parental populations and all individuals as the hybrid population.
We further removed from subsequent analyses admixed individuals. 
Due to an excess of missing data, `admixture` failed to infer population structure for *Eschweilera*.
We thus used principal component analysis and K-means clustering to detect *Eschweilera* populations.
We assessed the number of gene-pools using the within group sum of squares.
We validated populations structures of *Symphonia* and *Eschweilera* by comparison with botanical identifications.

We used `bayescan` [@Foll2008] to detect fixed SNPs between the different *Symphonia* populations using the 70,737 SNPs filtered for quantitative genomics (see genomic data production).
We considered SNPs with p-value corrected for multiple testing by false discovery rate below 5% as significant outliers.
We tested genic SNP which were outliers for enrichment in gene ontology with the R package `clusterProfiler` [@DeLaCruz2014].

### Topography

We did environmental association analyses [@Rellstab2015a] using general linear mixed models from genome wide association studies (GWAS). 
Counterintuitively, we used the environment as the response variable and genetic structure (populations) and relatedness (kinship matrix) as explanatory variables. 
We assumed that the environment where individuals have grown above 10-cm DBH was strongly correlated to the individual heritable phenotypes [e.g. @Eckert2010]. 

We used two little-correlated variables (Pearson'$r=-0.26$, $p-value<10^{-16}$) representing topography.
We selected the topographic wetness index (TWI) and relative elevation (RE) among several abiotic descriptors as a proxies of water and nutrients availability. 
Waterlogging and topography have been highlighted as crucial for forest dynamics [@ferry2010higher], species-habitats [@Engelbrecht2007], and phenotypic variation of *Symphonia* and *Eschweilera* [@Schmitt2020]. 
TWI and RE were derived from a 1-m resolution digital elevation model using SAGA-GIS [@Conrad2015] based on a LiDAR campaign done in 2015.
We further characterized TWI and RE specificities with analysis of variance (ANOVA) against digital elevation model (DEM) and numerous edaphic and hyrdologic properties (soil organic matter, carbon, nitrogen, exchangeable cation charge [@Soucemarianadin2004; @Roelens2007], water table depth [@Gourlet-Fleury2004] and waterlogging [@Cantet2004]).

We used species and individual kinship in a lognormal Animal model [@Wilson2010] to estimate genetic variance associated to the environment where individuals grow.
We defined species based on the gene-pools identified with `admixture`, removing admixed individuals.
We inferred individual kinship using KING [@Manichaikul2010], as the method is robust to population structure. 
We set to null negative values of kinship as they were confounding with population structure, and we further ensured that the matrix was positive-definitive using `nearPD`  function from the R package `Matrix`.
The environment $y_{s,i}$ where the individual $i$ in species $s$ grow was inferred with a lognormal distribution with following formula:

\begin{equation}
   \begin{aligned}
      y_{s,i} \sim logN(log(\mu_s.a_{i}),\sigma^2_1) \\
      a_{i} \sim MVlogN_N(log(1),\sigma^2_2.K)
   \end{aligned}
   (\#eq:AnimalEnv)
\end{equation}

where $u_s$ is the mean environment of species $s$, $a_i$ is the breeding value of the individual $i$ and $\sigma^2_1$ is the shape parameter of the lognormal. 
Individual breeding values $a_i$ are defined following a multivariate lognormal law $\mathcal{MVlogN}$ of co-shape matrix defined as the product of the kinship matrix $K$ with estimated individual genotypic variation $\sigma^2_2$.
To estimate variances on a normal-scale, we log-transformed species fixed effect, genetic additive values, and we calculated conditional and marginal $R^2$ [@Nakagawa2013].
A Bayesian method was used to infer parameters using `stan` language [@Carpenter2017, see supplementary material] and `rstan` package [@Team2018] in the R environment [@RCoreTeam2020] using No-U-Turn Sampler [NUTS, @Hoffman2014] algorithm, performing better for estimating genetic parameters and breeding values [@Nishio2019].

# Results

*Symphonia globulifera* was previously recognized as composed of two morphotypes in French Guiana [@Sabatier1997]. 
But cross validation of admixture analyses for different levels of clustering K advocated for 3 different gene-pools (Fig. \@ref(fig:symphospecies), see supplementary material).
The three gene-pools corresponded to the previously identified two morphotypes (70-80% of match) *S. globulifera* and *S. sp1*, with *S. globulifera* morphotype itself structured in two gene-pools. 
The two gene-pools composing the *S. globulifera* morphotype corresponded to two locally defined sub-morphotypes called *S. globulifera type Paracou* and *S. globulifera type Régina* (Fig. \@ref(fig:symphomorphotypes), Saint-Omer Cazal, pers. com.). 
*S. sp1* grows preferentially in drier plateaux and slopes (Schmitt et al., submitted) and has a light grey thin and smooth bark associated with smaller leaves, flowers and fruits  (Fig.\@ref(fig:symphomorphotypes); @Sabatier1997). 
*S. globulifera type Paracou* grows in wet habitats of bottomlands, but drier than the last population, and has a dark and intermediate thin and smooth bark compared to the last population (Fig.\@ref(fig:symphomorphotypes); Schmitt, pers. com.). 
*S. globulifera type Regina* grows in the wettest bottomlands of Paracou and has a thick and lashed bark (Fig.\@ref(fig:symphomorphotypes); Schmitt, pers. com.).
We conducted a second blind-identification of every individual in November 2019 to investigate mismatches between gene-pools and morphotypes and the existence of the two morphotypes *S. globulifera type Paracou* and *S. globulifera type Régina*. 
The blind-identification correctly assigned the difference between all *S. globulifera type Paracou* and *S. globulifera type Régina* and reclassified correctly 87% of previous mismatches between gene-pools and previous botanical identification.
Among the three populations, 25 individuals (7%) showed patterns of introgression, 15 between *S. globulifera type Regina* and *S. sp1* and 10 between *S. globulifera type Paracou* and *S. sp1* (confirmed with `introgress` R package [@Gompert2010], see supplementary material). 
Admixed individuals were removed from subsequent analyses.
Last but not least, genomic scans for *Symphonia* revealed fixed markers among populations (5.7% of SNPs) with adaptive markers represented by an enrichment of genes related to “response to water deprivation” ($p=0.01106483$). 
Consequently, *Symphonia globulifera* was locally and genetically structured in three sympatric species (Fig. \@ref(fig:symphospecies)), including thus cryptic species, corresponding to three different morphotypes (Fig. \@ref(fig:symphomorphotypes)). 
*Eschweilera* clade Parvifolia was locally and genetically structured in three genepools corresponding to botanical species *E. coriacea*, *E. sagotiana* and *E. decolorans* (65-80% of match, Fig. \@ref(fig:parvispecies)).

We identified topographic wetness index (TWI) and relative elevation (RE, Fig. \@ref(fig:SpDistVar)) as descriptors of the topographic gradient in which individuals grow.
Topographic wetness index ($TWI$) was related to hydrology (ANOVA with water table depth, $R^2=0.32$, $p<10^{-151}$).
Relative elevation ($RE$) was related to soil chemistry due to hydromorphy (ANOVA with soil carbon, nitrogen, organic matter, exchangeable cation charge and soil color due to hyrdomorphy with $0.23<R^2<0.28$, $p<10^{-30}$).
The spatial distribution of *Symphonia* and *Eschweilera* species suggested that both topographic wetness index ($TWI$) and relative elevation ($RE$) drove individual survival among populations ($0.06<R_m^2<0.39$, Fig. \@ref(fig:SpDistVar)).
The spatial distribution of *Symphonia* species was driven by both topographic wetness index ($TWI$, Fig. \@ref(fig:SpDistVar)C), especially between *S. globulifera Regina* and *S. globulifera Paracou*, and relative elevation ($RE$, Fig. \@ref(fig:SpDistVar)D), especially between *S. sp1* and the two remainings.
The spatial distribution of *Eschweilera*  species was mainly driven by relative elevation ($RE$, Fig. \@ref(fig:SpDistVar)B).
The spatial distribution of *Symphonia* and *Eschweilera* species along topography resulted in intermediate genetic differentiation, $F_{ST}=0.15$ and $F_{ST}=0.35$ between *Symphonia* and *Eschweilera* species respectively

# Discussion

Despite the local abundance and regional success of closely-related tree species growing in sympatry in the Neotropics,
little is known of the eco-evolutionary forces responsible for their local coexistence.
Here, we show that within tree species complexes, closely-related species have different realized optima for topographic niches with species genetic adaptations. 
*Symphonia* species are adapted to water and nutrient distribution with topographic wetness and relative elevation,
growing thus in a broad range of local habitats.
Whereas *Eschweilera* species are differentially adapted to soil chemistry due to hydromorphy, avoiding wettest habitats.
The greater role of topography among *Symhponia* species than among *Eschweilera* species suggests that topography may be the main factor in differentiating niches among *Symhponia* species, while other factors may differentiate niches among *Eschweilera* species.
Consequently, species adaptations to different characteristics of topography stabilize local coexistence between closely-related species within tree species complexes.

## Three species of *Symphonia* adapted to topography coexist in sympatry

We identified three distinct species of *Symphonia globulifera* with different morphology, genepool and topographic niche.
For the study purpose, we named the three species *S. sp1*, *S. globulifera type Paracou* and *S. globulifera type Regina* in reference to already existing morphotypes and local names.
The three species showed a low but significant genetic differentiation ($F_{ST} = 0.15$), already highlighted previously at the local scale between the previously two recognized morphotypes [$F_{ST} = 0.114$ for SSRs; @Olsson2017].
The genetic differentiation was revealed by 5.7% of SNPs being significantly associated with species with an enrichment in genes implicated in response to water deprivation.
Interestingly though, topography has already been highlighted as a driver of the distribution of the two previously recognized morphotypes [@Allie2015; Schmitt et al. submitted], with *S. globulifera* growing in low-elevation and wet bottomlands and *S. sp1* growing in drier slopes and plateaux.
But our study revealed the existence of two *Symphonia globulifera* morphotypes segregated within wet areas with *Symphonia globulifera type Regina* growing in the wettest bottomlands, such as swamp characterized by increased wetness and anoxia.
Similar ecotypic differentiation with swamp have been suggested in Africa [@Budde2013].
Individual topographic conditions were mainly plastic (54-56% of residual variation) but among the remaining genetic variation the majority (80-85%) was due to species differences (Fig. \@ref(fig:SpDistVar)).
Thus, finescale topographic variation between individuals was driven by species adaptations, revealing local adaptation to topography among the three species at the hectare-scale.
Further studies at a broader regional scale are needed, but our results advocate for the definition of three distinct taxonomic species with differences in morphology (Fig. \@ref(fig:symphomorphotypes)), habitats (Fig. \@ref(fig:SpDistVar)), functional traits and growth trajectories (S. Schmitt pers. com.).

## Differentiation of *Symphonia* and *Eschweilera* species along topography: water distribution and soil chemistry

*Symphonia* species were primarily differentially adapted to water and nutrient distribution with topographic wetness and relative elevation; whereas *Eschweilera* species were differentially adapted to soil chemistry due to hydromorphy.
Similarly to *Symphonia* species, at least three distinct species of *Eschweilera* clade *Parvifolia* with different morphology, genepool and topographic niche coexist in sympatry. 
*E. coriacea* preferentially grow in valley bottoms, and *E. sagotiana* and *E. decolorans* preferentially exploit a variety of drier habitats with slopes and plateaus [Fig. \@ref(fig:SpDistVar); @Allie2015; Schmitt et al. submitted].
The lack of power of *Eschweilera* data prevented us from detecting less abundant *Eschweilera* clade *Parvifolia* species (e.g. *E. grandiflora*, *E. whachenheimii*), but our results do not question their botanical definitions.
Topographic conditions of *Eschweilera* individuals were mainly plastic (73-75% of residual variation), but among the remaining genetic variation the majority was due to species differences for relative elevation (52%) while the majority was due to genotypes for topographic wetness index (78%, Fig. \@ref(fig:SpDistVar)).
The results of *Eschweilera* are in contrast to those of *Symphonia*,
with more plasticity and genotypic effects in topographic conditions of *Eschweilera* individuals than *Symphonia* individuals.
Moreover, species topographic differences for *Symphonia* were driven by both topographic wetness, especially between *S. sp1* and others, and relative elevation, especially between the two *S. globulifera*; 
whereas species topographic differences for *Eschweilera* were mainly driven by relative elevation.
Indeed, topographic wetness was more related to hydrology with water table depth; 
whereas relative elevation was more related to soil chemistry.
Thus, *Symphonia* species are able to grow in a wide variety of habitats due to species adaptation to different water and nutrient access;
whereas *Eschweilera* species avoid wettest habitats and grow in a variety of topography due to species adaptation to nutrient access (Fig. \@ref(fig:SpDistVar)).
Topography had globally a greater role among *Symphonia* species than among *Eschweilera* species (less residual variation).
*Eschweilera* species may thus differentiate along other ecological niches or coexist with species competitively combining [@Aarssen1983].

## Species differentiation along topography and coexistence within species complexes

The interaction between topography and genes shapes the microgeographic genetic structure of the three sympatric populations of *Symphonia globulifera* and *Eschweilera* clade *Parvifolia*. 
Species of *Symphonia globulifera* and *Eschweilera* clade *Parvifolia* are locally adapted to fine-scale topography, 
along a gradient from flooded, nutrient-rich bottomlands to drier, poorer slopes and plateaus.
Genomic bases for the microgeographic adaptation of species have been identified for *Symphonia* species,
with in particular a functional response to water distribution identified as a driver of the topographic niches of the species.
Similarly to European white oak, we identified coexisting different species with genes that allow them to survive in different ecological niches despite interspecific hybridization [@Leroy2019]. 
Consequently, we evidenced syngameons of tropical trees with differential adaptations of species to topographic niches stabilizing local coexistence of species. 
If interspecific gene flow benefits all the constituent species, *Symphonia globulifera* and *Eschweilera* clade *Parvifolia* might be a successful evolutionary state as syngamons [@Cannon2019a].
Literature includes numerous examples of closely-related species differentiated along topography [@gunatilleke_specieshabitat_2006; @Engelbrecht2007; @kraft_functional_2008; @Allie2015] or along other ecological niches [@Crystal2003; @Yamasaki2013].
In Paracou, several species complexes have been shown to present differentiated niches along topography [Schmitt et al. in prep].
Combined with our results, this abundant evidence of niche differentiation between closely related species calls into question the more general role of topography in the genetic adaptation and diversification of tropical tree species complexes.
Moreover, this question the role of species complexes and syngameons in the origin and maintenance of Neotropical forest diversity.

# Acknowledgements

We thank the University of Bordeaux for a PhD grant to Sylvain Schmitt. 
We are grateful to Pascal Petronelli and the CIRAD inventory team for their work on tree inventories and botanical identification. 
This study was partially funded by an Investissement d’Avenir grant of the ANR: CEBA (ANR-10-LABEX-0025).

# References

<div id="refs"></div>

# Authors’ contributions

All authors conceived the ideas and designed methodology;
SS, MH, and NT sampled individuals;
SS and MH realized the gene capture experiment;
SS, MH and BH  analysed outputs; 
SS and XX led the writing of the manuscript. 
All authors contributed critically to the drafts and gave final approval for publication.

# Data Accessibility

Authors are required to archive their data in a publicly accessible repository such as Dryad, FigShare, GenBank, etc.  (not a laboratory homepage).

* Upon submission, this statement must be included, but can describe curation plans prior to data having been thus archived.
* Upon acceptance, data must be archived and the Data Accessibility statement completed including database and information such as accession numbers or DOI (as available) for all data from the manuscript.
* Note: if data, scripts, or other artefacts used to generate the analyses presented in the paper are available via a publicly available data repository, authors should include a reference to the location of the material within their paper.


# Tables and Figures

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
library(tidyverse)
library(raster)
library(bayesplot)
library(ggfortify)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, 
               fig.height = 6, fig.width = 6,
               cache = T, cache.lazy = F)
# crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
crs <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
spcol <- c("#1b9e77", "#d95f02", "#7570b3", "#1E90FF", "#0000EE", "#CD2626")
names(spcol) <- c("E. decolorans", "E. sagotiana", "E. coriacea", "S. sp2", "S. sp3", "S. sp1")
```

```{r, eval=F}
load("../symcapture/symcapture_save/Env.Rdata")
symphonia <- trees
load("../parvicapture/parvicapture_save/Env.Rdata")
eschweilera <- trees
bind_rows(
  mutate(symphonia, genus = "Symphonia") %>% 
    dplyr::select("IID", "Ind", "Xutm", "Yutm", "genus", "pop", "TWI", "RE") %>% 
    dplyr::rename(species = pop) %>% 
    mutate(species = gsub("Type", " ", species)) %>% 
    mutate(species = paste0("S. ", species)),
  mutate(eschweilera, genus = "Eschweilera") %>% 
    dplyr::select("IID", "Library", "Xutm", "Yutm", "genus", "cluster", "TWI", "RE") %>% 
    dplyr::rename(Ind = Library, species = cluster) %>% 
    mutate(species = recode(species, "3" =  "E. decolorans", 
                            "2" = "E. sagotiana",
                            "1" = "E. coriacea")) 
) %>% 
  mutate(species = recode(species, "	S. globulifera Paracou" =  "S. sp2",
                          "	S. globulifera Regina" = "S. sp3") ) %>% 
  write_tsv("save/environment.tsv")
```

```{r maps, fig.cap="Map of sampled individuals for species complexes *Symphonia* and *Eschweilera* clade *Parvifolia*. Colours represent species determined based on the genetic structure of capture datasets, which was not necessarily in agreement with original botanical identification. Grey level represents hill shading to visualize topography.", fig.width=6, fig.height=5}
load("../symcapture/symcapture_save/Env.Rdata")
symphonia <- trees
load("../parvicapture/parvicapture_save/Env.Rdata")
eschweilera <- trees
trees <- bind_rows(
  mutate(symphonia, genus = "Symphonia") %>% 
    dplyr::select("Ind", "Xutm", "Yutm", "genus", "pop") %>% 
    dplyr::rename(species = pop) %>% 
    mutate(species = gsub("Type", " ", species))  %>% 
        mutate(species = recode(species, 
                                "globulifera Paracou" =  "sp2", 
                            "globulifera Regina" = "sp3")) %>% 
    mutate(species = paste0("S. ", species)) ,
  mutate(eschweilera, genus = "Eschweilera") %>% 
    dplyr::select("Library", "Xutm", "Yutm", "genus", "cluster") %>% 
    dplyr::rename(Ind = Library, species = cluster) %>% 
    mutate(species = recode(species, "3" =  "E. decolorans", 
                            "2" = "E. sagotiana",
                            "1" = "E. coriacea")) 
)
coordinates(trees) <- ~Xutm + Yutm
proj4string(trees) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
trees <- spTransform(trees, CRSobj = crs)
hills <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                          "topography", "hillshade_1m.tif"))
hills <- crop(hills, trees)
hills <- projectRaster(hills, crs = crs)
hills <- as(aggregate(hills, 4), "SpatialPixelsDataFrame")
hills <- as.data.frame(hills)
colnames(hills) <- c("hills", "Xutm", "Yutm")

limits <- shapefile(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/limits/Plots.shp"))
limits <- crop(limits, trees)

g <- ggplot(hills, aes(x = Xutm, y = Yutm)) +
  geom_tile(aes(fill = hills),  col = NA, alpha = 0.6, data = ) +
  scale_fill_distiller(guide = "none", palette = "Greys", direction = 1) +
  geom_polygon(data = fortify(limits), aes(x = long, y = lat, group = id), colour='black', fill=NA) +
  geom_point(data = as.data.frame(trees), aes(col = species)) +
  scale_color_manual("", values = spcol[c(3, 1, 2, 4, 5, 6)]) +
  facet_wrap(~ genus) +
  coord_equal() +
  cowplot::theme_map() +
  ggspatial::annotation_scale(location = "br", width_hint = 0.5, plot_unit = "m", 
                              data = data.frame(genus = "Symphonia")) +
  ggspatial::annotation_north_arrow(location = "tl", height = unit(1, "cm"), width = unit(1, "cm"),
                                    data = data.frame(genus = "Eschweilera")) +
  theme(axis.title = element_blank(),
        legend.position = "bottom", legend.text = element_text(face = "italic"),
        strip.background = element_blank(), strip.text.x = element_blank())  +
  guides( col = guide_legend(nrow = 2, byrow = T))
ggsave(g, filename = "figures/Figure_1.tiff", device = "tiff", dpi = 600, bg = "white", compression = "lzw")
```

```{r soilpca, fig.cap="Principal component analysis (PCA) of topographic wetness index (TWI), relative elevation (RE) with edaphic and hydrological properties. Edaphic and hydrological properties include organic matter (OM), carbon (C), nitrogen (N), exchangeable cation charge (C) [@Soucemarianadin2004; @Roelens2007], water table depth [WTD in cm, @Gourlet-Fleury2004], hydromorphy and waterlogging [waterlog in 5 classes, @Cantet2004]). Data were obtained through interpolation of soil sample data on TWI and RE from LiDAR data."}
g <- read_tsv("figures/soil.tsv") %>%
  na.omit() %>% 
  dplyr::select(-Refus, -Mg_ech, -K_ech, -Polsen, -S, -TS, -Na_ech, -Al_KCl, - Ca_ech) %>% 
  dplyr::rename(TWI = wetness, RE = relele, OM = MO, "C:N" = C_N, P = PBray2,
                waterlog1 = waterlog_1, waterlog2 = waterlog_2, waterlog3 = waterlog_3,
                waterlog4 = waterlog_4, waterlog5 = waterlog_5, 
                "WTD 0-60" = "wtd_Watertable between 0 and 60 cm",
                "WTD 60-100" = "wtd_Watertable between 60 and 100 cm", 
                "WTD 100+" = "wtd_Watertable deeper than 100 cm") %>% 
  autoplot(prcomp(., scale = T), 
           data = .,
           col = "grey",
           alpha = 0.3,
           loadings = T,
           loadings.colour = c('black', 'black', rep('grey', 15)),
           loadings.label.colour = c('black', 'black', 'blue', 'red', 'red',
                                     'red', 'blue', 'blue', 'red', 'red', 'red',
                                     'red', 'red', 'blue', 'blue', 'blue',
                                     'blue'),
           loadings.label.fontface = c('bold', 'bold', rep('italic', 15)),
           loadings.label.size = c(6, 6, rep(4, 15)),
            loadings.label = T, 
           loadings.label.repel=T) +
  geom_hline(aes(yintercept = 0), col = 'black', linetype = "dotted") +
  geom_vline(aes(xintercept = 0), col = 'black', linetype = "dotted") +
  scale_color_discrete(guide = "none") +
  theme(legend.position = c(1, 1), 
        legend.justification = c(1, 1))
ggsave(g, filename = "figures/Figure_2.tiff", device = "tiff", dpi = 600, bg = "white", compression = "lzw")
```

```{r symphospecies, fig.cap="Admixture plot of *Symphonia* individuals for the best K, K = 3.", fig.width=6, fig.height=3}
g <- cowplot::plot_grid(
  pophelper::readQ(file.path("../../../data/Symphonia_Paracou/Sequences/populationGenomics/",
                           "admixture", "paracou", "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.3.Q"))[[1]] %>%
  dplyr::rename(Ssp1 = Cluster1, SgParacou = Cluster2, SgRegina = Cluster3) %>%
  mutate(Ind = gsub(".g.vcf", "", read_tsv(file.path("../../../data/Symphonia_Paracou/Sequences", "variantCalling", "paracou",
                                                     "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
                                           col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype"))$IID)) %>%
  arrange(desc(Ssp1), desc(SgParacou), desc(SgRegina)) %>%
  mutate(order = 1:nrow(.)) %>%
  reshape2::melt(id.vars = c("order", "Ind"), variable.name = "pop") %>%
  ggplot(aes(reorder(Ind, order), value, fill = pop, col = pop)) +
  geom_col() +
  scale_fill_manual("", values = c("#CD2626", "#1E90FF", "#0000EE"),
                    labels = c("S. sp.1", "S. sp.2", "S. sp.3")) +
  scale_color_manual("", values = c("#CD2626", "#1E90FF", "#0000EE"),
                     labels = c("S. sp.1", "S. sp.2", "S. sp.3")) +
  ylab("Admixture\ncoefficient") +
  theme(axis.title.x = element_blank(), axis.line.x = element_blank(),
        axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        legend.text = element_text(face = "italic"), legend.position = "top"),
  cowplot::ggdraw() + 
    cowplot::draw_image("figures/symphoniamorphotypes.png", scale = 0.9),
  nrow = 2, labels = c("(a)", "(b)")
)
ggsave(g, filename = "figures/Figure_3.tiff", device = "tiff", dpi = 600, bg = "white", width = 6, height = 3, compression = "lzw")
```

```{r parvispecies, fig.cap="Principal component analysis (PCA) of single nucleotide polymorphisms (SNP) from *Eschweilera* clade *Parvifolia.* The colours represent the clusters detected with Kmeans for the best K, K = 3.", fig.height=5, fig.width=5}
vars <- read_delim(file.path("../../../data/Eschweilera_Paracou/Sequences/", "variants", "final", "filtered.eigenval"), 
               delim = " ", col_names = "var")
g <- read_delim(file.path("../../../data/Eschweilera_Paracou/Sequences/", "variants", "final", "filtered.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>% 
  separate(Sample, paste0("X", 1:7), remove = F) %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  left_join(read_tsv(file.path("../../../data/Eschweilera_Paracou/Sequences/", "variants", "final", "filtered.kmeans"))) %>% 
  mutate(species = recode(cluster, "3" =  "E. decolorans", 
                          "2" = "E. sagotiana",
                          "1" = "E. coriacea")) %>% 
  ggplot(aes(x = PCA1, y = PCA2, col = species)) +
  geom_point(size=2, alpha = 0.5) +
  theme(legend.position = "bottom", legend.text = element_text(face = "italic")) +
  xlab(paste0("PCA 1 - ", vars$var[1], "%")) +
  ylab(paste0("PCA 2 - ", vars$var[2], "%")) +
  scale_color_manual("", values = spcol[c(3,1,2)])
ggsave(g, filename = "figures/Figure_4.tiff", device = "tiff", dpi = 600, bg = "white", compression = "lzw")
```

```{r SpDistVar, fig.cap="Species distribution and individual variance partitioning along topography. Each subfigure shows density of individuals per species (density plot) against individual variation (barplot) partitioned with the Animal model into between-species (dark grey), between-genotypes (intermediate grey) and residual (light grey). Species distribution and individual variance partitioning partitioning are given along the topographic wetness index (TWI, left subfigures) and the logarithm of relative elevation (log(RE+1), right subfigures) for the species complexes *Eschweilera* clade *Parvifolia* (top subfigures) and *Symphonia* (bottom subfigures). Colours represent species determined based on the genetic structure of capture datasets, which was not necessarily in agreement with original botanical identification. ", fig.width=11, fig.height=6}
wetness <- raster(file.path("../../../data/Paracou/", "topography", "TWI_1m.tif"))
relele <- raster(file.path("../../../data/Paracou/", "topography", "RelativeElevation_1m.tif"))
dem <- raster(file.path("../../../data/Paracou/", "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
projection(relele) <- projection(dem)
wetness <- projectRaster(wetness, crs = crs)
relele <- projectRaster(relele, crs = crs)
trees$TWI <- raster::extract(wetness, trees)
trees$RE <- raster::extract(relele, trees)
gl.legend <- cowplot::get_legend(trees@data %>%
  mutate(RE = log(RE+1)) %>% 
  reshape2::melt(id.vars = c("Ind", "genus", "species")) %>% 
  mutate(variable = recode(variable, "TWI" = "Topographic wetness index",
                           "RE" = "log(Relative elevation + 1)")) %>% 
  ggplot(aes(value, fill = species, color = species)) +
  geom_density(alpha = 0.1) +
  facet_grid(genus ~ variable, scales = "free") +
  scale_color_manual("Species", values = spcol[c(3, 1, 2, 4, 5, 6)]) +
  scale_fill_manual("Species", values = spcol[c(3, 1, 2, 4, 5, 6)]) +
  theme(legend.position = "bottom", legend.text = element_text(face = "italic")) +
  guides(fill = guide_legend(nrow = 2, byrow = T), col = guide_legend(nrow = 2, byrow = T)))
fitSympho <- list(TWI = list(), RE = list())
for(var in c("TWI", "RE")){
  for(sim in list.files("../symcapture/symcapture_save/EnvGeno", 
                        pattern = var, full.names = T)){
    load(sim)
    fitSympho[[var]] <- c(fitSympho[[var]], fit)
  }
}
fitSympho <- lapply(fitSympho, rstan::sflist2stanfit)
names(fitSympho) <- paste0("Symphonia.", names(fitSympho))
fitParvi <- list(TWI = list(), RE = list())
for(var in c("TWI", "RE")){
  for(sim in list.files("../parvicapture/parvicapture_save/all/EnvGeno2/", 
                        pattern = var, full.names = T)){
    load(sim)
    fitParvi[[var]] <- c(fitParvi[[var]], fit)
  }
}
fitParvi <- lapply(fitParvi, rstan::sflist2stanfit)
names(fitParvi) <- paste0("Eschweilera.", names(fitParvi))
fitEnv <- c(fitSympho, fitParvi)
rm(fitSympho, fitParvi)
gr.legend <- cowplot::get_legend(lapply(fitEnv, mcmc_intervals_data, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(parameter = recode(parameter, "Vp" = "Species", "Vg" = "Genotype", "Vr" = "Residual")) %>% 
  group_by(variable) %>%
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  separate(variable, c("genus", "variable")) %>% 
  group_by(genus, variable) %>%
  mutate(m2 = m/sum(m)) %>% 
  ungroup() %>% 
  mutate(variable = recode_factor(variable, "RE" = "Relative elevation",
                                  "TWI" = "Topographic wetness index")) %>% 
  ggplot(aes(x = variable, fill = parameter)) +
  geom_col(aes(y = m2)) +
  geom_text(aes(y = m2, label = pct), col = "white", position = position_stack(vjust = .5)) +
  facet_wrap(~ genus, scales = "free", nrow = 2) +
  coord_flip() +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(),
        axis.ticks.x = element_blank(), axis.line.y = element_blank(),
        legend.position = "bottom", strip.text = element_text(face = "italic")) +
  scale_fill_grey(expression(sigma^2)) +
  guides(fill = guide_legend(nrow = 3, byrow = T)))
bars <- lapply(fitEnv, mcmc_intervals_data, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(parameter = recode(parameter, "Vp" = "Species", "Vg" = "Genotype", "Vr" = "Residual")) %>% 
  group_by(variable) %>%
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  separate(variable, c("genus", "variable")) %>% 
  group_by(genus, variable) %>%
  mutate(m2 = m/sum(m)) %>% 
  ungroup() %>% 
  mutate(variable = recode_factor(variable, "RE" = "Relative elevation",
                                  "TWI" = "Topographic wetness index")) %>% 
  group_by(genus, variable) %>% 
  do(g = ggplot(data = ., aes(x = variable, fill = parameter)) +
       geom_col(aes(y = m2)) +
       geom_text(aes(y = m2, label = pct), col = "white", position = position_stack(vjust = .5)) +
       theme(axis.title = element_blank(), axis.text = element_blank(), axis.line = element_blank(), 
             axis.ticks = element_blank()) +
       scale_fill_grey(guide = "none"))
densities <- trees@data %>%
  mutate(RE = log(RE+1)) %>% 
  reshape2::melt(id.vars = c("Ind", "genus", "species")) %>% 
  mutate(variable = recode(variable, "TWI" = "Topographic wetness index",
                           "RE" = "log(Relative elevation + 1)")) %>% 
  group_by(genus, variable) %>% 
  do(g = ggplot(data = ., aes(value, fill = species, color = species)) +
       geom_density(alpha = 0.1) +
       facet_grid(genus ~ variable, scales = "free") +
       scale_color_manual("", values = spcol[c(3, 1, 2, 4, 5, 6)]) +
       scale_fill_manual("", values = spcol[c(3, 1, 2, 4, 5, 6)]) +
       theme(legend.position = "none"))
g.white <- qplot(1:10, 1:10, geom="blank") +
  theme(axis.title = element_blank(), axis.line = element_blank(), 
        axis.ticks = element_blank(), axis.text = element_blank())
g <- cowplot::plot_grid(densities$g[[1]], 
                   bars$g[[2]], 
                   densities$g[[2]], 
                   bars$g[[1]],
                   densities$g[[3]], 
                   bars$g[[4]], 
                   densities$g[[4]], 
                   bars$g[[3]],
                   g.white,
                   gl.legend,
                   g.white,
                   gr.legend,
                   nrow = 3, rel_widths = c(3,1,3,1), rel_heights = c(3,3,1),
                   labels = c("A", "", "B", "", "C", "", "D", ""))
ggsave(g, filename = "figures/Figure_5.tiff", device = "tiff", dpi = 600, bg = "white", width = 11, height = 6, compression = "lzw")
```
