---
title: Coordinated answer of individual-tree leaf functional traits to ontogeny and environment between specie of Neotropical complexes (Temporary)
author: Sylvain Schmitt$^{*,1}$, Bruno Herault$^{2,3}$, Emilie Ducouret$^4$, Anne Baranger$^5$, Niklas Tysklind$^6$,  Myriam Heuertz$^5$, Eric Marcon$^7$, Saint Omer Cazal $^6$, Geraldine Derroire$^{4}$ 
date: '`r Sys.Date()`'
output:
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
  bookdown::word_document2: default
  bookdown::html_document2:
    number_sections: false
    toc: true
    toc_float: yes
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/mee.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(ggfortify)
library(broom)
library(rstan)
library(bayesplot)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, 
               fig.height = 8, fig.width = 8,
               cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Paracou/"
```

```{r data}
load("./functional_save/Individuals.Rdata")
Individuals0 <- Individuals
load("./functional_save/CompetitionMatrix.Rdata")
sdDBH <- sd(Individuals$DBH)
Individuals <- Individuals %>% 
  rename(LMA = invSLA) %>% 
  group_by(Species) %>% 
  filter(n() > 20) %>% 
  ungroup() %>% 
  left_join(Competition %>% 
              group_by(idTree) %>% 
              summarise(NCI = (1/mean(1-AreaOutside20))*sum(DBHj^2*exp(-0.25*dij)))) %>%
  mutate(NCI = log(NCI)) %>% 
  mutate_at(vars("LMA", "LDMC", "LT", "invLA", "CC", "DBH", "TWI", "NCI"),
            funs(./sd(., na.rm = T)))
```

```{r model}
load("./functional_save/InteractionReduced.Rdata")
traits <- c("LMA", "LDMC", "LT", "invLA", "CC")
complexes <- c("Symphonia", "Eschweilera")
models <- sapply(complexes, function(complex) paste0(complex, "-", traits))
species <- lapply(models, function(model){
  trait <- strsplit(model, "-")[[1]][2]
  complex <- strsplit(model, "-")[[1]][1]
  Individuals[!is.na(unlist(Individuals[,trait])),] %>% 
    filter(Genus == complex) %>% 
    mutate(species = as.numeric(as.factor(Species))) %>% 
    mutate(SpeciesLong = paste(substr(complex, 1, 1), Species)) %>% 
    select(Genus, Species, SpeciesLong, species) %>% 
    unique()
}) %>% bind_rows() %>% 
  unique() %>% 
  mutate(complex = Genus)
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
```

```{r predicitons}
traits <- c("LMA", "LDMC", "LT", "invLA", "CC")
complexes <- c("Symphonia", "Eschweilera")
models <- sapply(complexes, function(complex) paste0(complex, "-", traits))
preds <- lapply(models, function(model){
  trait <- strsplit(model, "-")[[1]][2]
  complex <- strsplit(model, "-")[[1]][1]
  data_trait <- Individuals[!is.na(unlist(Individuals[,trait])),] %>% 
    filter(Genus == complex)
  
  DBH <- data_trait$DBH
  TWI <- data_trait$TWI
  NCI <- data_trait$NCI
  species <- as.numeric(as.factor(data_trait$Species))
  
  alpha <- as.matrix(fits[[model]], pars = "alpha")
  betaDBH <- as.matrix(fits[[model]], pars = "betaDBH")
  betaTWI <- as.matrix(fits[[model]], pars = "betaTWI")
  betaNCI <- as.matrix(fits[[model]], pars = "betaNCI")
  
  Yp <- do.call("rbind", sapply(1: nrow(alpha), function(n) (alpha[n,species] + betaTWI[n,species] * TWI +  betaNCI[n,species] * NCI) * (DBH / (betaDBH[n,species] + DBH)), simplify = F))
  Yalpha <- do.call("rbind", sapply(1: nrow(alpha), function(n) (alpha[n,species] + betaTWI[n,species] * mean(TWI) +  betaNCI[n,species] * mean(NCI)) * (mean(DBH) / (betaDBH[n,species] + mean(DBH))), simplify = F))
  Ydbh <- do.call("rbind", sapply(1: nrow(alpha), function(n) (alpha[n,species] + betaTWI[n,species] * mean(TWI) +  betaNCI[n,species] * mean(NCI)) * (DBH / (betaDBH[n,species] + DBH)), simplify = F))
  Ytwi <- do.call("rbind", sapply(1: nrow(alpha), function(n) (alpha[n,species] + betaTWI[n,species] * TWI +  betaNCI[n,species] * mean(NCI)) * (mean(DBH) / (betaDBH[n,species] + mean(DBH))), simplify = F))
  Ynci <- do.call("rbind", sapply(1: nrow(alpha), function(n) (alpha[n,species] + betaTWI[n,species] * mean(TWI) +  betaNCI[n,species] * NCI) * (mean(DBH) / (betaDBH[n,species] + mean(DBH))), simplify = F))
  
  data.frame(
    Model = model,
    Trait = trait,
    Complex = complex,
    DBH = data_trait$DBH, 
    TWI = data_trait$TWI, 
    NCI = NCI,
    Genus = data_trait$Genus,
    Species = data_trait$Species,
    Y = unlist(data_trait[trait]),
    Yp = apply(Yp, 2, mean),
    Yp5 = apply(Yp, 2, quantile, probs = 0.05),
    Yp95 = apply(Yp, 2, quantile, probs = 0.95),
    Yalpha = apply(Yalpha, 2, mean),
    Yalpha5 = apply(Yalpha, 2, quantile, probs = 0.05),
    Yalpha95 = apply(Yalpha, 2, quantile, probs = 0.95),
    Ydbh = apply(Ydbh, 2, mean),
    Ydbh5 = apply(Ydbh, 2, quantile, probs = 0.05),
    Ydbh95 = apply(Ydbh, 2, quantile, probs = 0.95),
    Ytwi = apply(Ytwi, 2, mean),
    Ytwi5 = apply(Ytwi, 2, quantile, probs = 0.05),
    Ytwi95 = apply(Ytwi, 2, quantile, probs = 0.95),
    Ynci = apply(Ynci, 2, mean),
    Ynci5 = apply(Ynci, 2, quantile, probs = 0.05),
    Ynci95 = apply(Ynci, 2, quantile, probs = 0.95)
  )}) %>% bind_rows()
```

$^1$ UMR BIOGECO, Université de Bordeaux, 69 route d’Arcachon, F-33612 Cestas cedex France

$^2$ Cirad, Univ Montpellier, UR Forests & Societies, Montpellier, France

$^3$ INP-HB, Institut National Polytechnique Félix Houphouët-Boigny, BP 1093, Yamoussoukro, Ivory Coast

$^4$ Cirad, UMR EcoFoG (Agroparistech, CNRS, Inra, Université des
Antilles, Université de la Guyane), Campus Agronomique, 97310 Kourou, French Guiana

$^5$ UMR BIOGECO, INRA, 69 route d’Arcachon, F-33612 Cestas cedex France

$^6$ INRA, UMR EcoFoG (Agroparistech, CNRS, Université des
Antilles, Université de la Guyane, Cirad), Campus Agronomique, 97310 Kourou, French Guiana

$^7$ Agroparistech, UMR EcoFoG (CNRS, Inra, Université des
Antilles, Université de la Guyane, Cirad), Campus Agronomique, 97310 Kourou, French Guiana

__$^*$ Corresponding author:__ sylvain.schmitt@agroparistech.fr

\newpage 
\newpage 

 # Abstract

> To be written

## Keywords

Intraspecific trait variation; species complexes; ...

\newpage 

# Introduction 

<!-- FT & community assembly / biodiversity -->

Whereas functional traits are more and more used by community ecologists to explain the ecosystem structure [@Paine2011] and response to environment [@kraft_functional_2008]  in order to adapt to increasing change [@Sakschewski2016]; the outstanding biodiversity of tropical forests is still an open question [@wright2002plant], already highlighted 40 years ago [@connell_diversity_1978].

<!-- FT and individual and species ecology - few examples of FT role -->

Functional traits have been defined as phenotypic traits impacting fitness indirectly through their effect on individual performance which can be related to tree abilities to grow, survive and reproduce in a given environment [@violle_let_2007]. Consequently, functional traits expresses fundamental trade-offs determining individual, and species co-occurrences along environmental gradients [@Wright2002]. For instance, specific leaf mass per area (fresh weight divided by area) varies along soil fertility and shade gradients and represents a trade-off between resource acquisition by photosynthesis and investment in leaf defense and durability [@Evans2001; @hodgson_is_2011]. 

<!-- FT covariations -->

In addition, functional traits vary at community [@Bruelheide2018] and species [@diaz_global_2015] levels according to basic leaf [@wright_worldwide_2004; but see @lloyd_photosynthetically_2013; @osnas_global_2013] and wood [@chave_towards_2009] economic spectra; despite some authors advocate for a unique economics spectrum [@reich_world-wide_2014] evidenced in subarctic grasslands [@freschet_evidence_2010]. The leaf economics spectrum opposes acquisitive to conservative ecological strategies, often expressed at the leaf level. The wood economics spectrum opposes slow to fast growing species, expressed at leaf, twig and stem levels [@chave_towards_2009].

<!-- Biodiversity structure from community to individual & Species coexistence theories-->

On the other hand, the outstanding biodiversity of tropical forest can be seen in three nested level: (i) communities, (ii) species, and (iii) individual diversity. Community can be defined as a group of interacting organisms classified into species [ref]. Species have been defined as groups of organisms which are reproducing isolated from other such groups, protecting the integrity of their genotypes [@Mayr1996]. This reflects the biological species concept based on reproductive isolation. Different theories have been developed at the species level to explain community assembly and biodiversity maintenance. Niche theory explains species coexistence depending on niche differences to limit competitive exclusion [@Lortie2004a; @weiher_assembly_1995], while neutral theory shows that maintenance of high diversity is possible even for functionally equivalent species, because of stochastic life, death, reproduction and dispersal dynamics [@Hubbell2001]. But numerous theories exists to explain species coexistence [@wright2002plant].

<!-- Species issue and species complexes, Increased role of individual & Determinant of phenotypîc variation-->

Still, species concepts and definitions have been largely debated [@Mayden1997], especially concerning plants, and many species concepts can be defined (phylogenetic, biological, ecological, morphological or genetic, see @DeQueiroz2007a). Moreover, some closely related species share a large amount of common genetic variation due to a recent common ancestor or hybridization and form a species complex challenging the static vision of species [@Fujita2012]. However, individuals present variation in performance, phenotypes, and genes within the species. These variations result in intraspecific variability, ultimately represented by phenotypic variation among communities and ecosystems. Phenotypic variation will be shaped (i) by genetic heritage, through genotypes, (ii) by the environment (both abiotic and biotic) with spatial and temporal heterogeneity, and (iii) by random stochastic factors [@Whitlock2007].

<!-- ITV evidence -->

But intraspecific trait variability is relatively unexplored [@Hallgrimsson2005] beside being highlighted as large in tree allometric relations [@Vieilledent2010], in leaf traits [@Hulshof2010a; @Messier2010], and within community itself [@Siefert2015a]. Thus several authors advocates for a better understanding of intraspecific variability role in ecology [@Albert2010a; @chave_neutral_2004; @Violle2012]. Beyond the assessment of intraspecific trait variation, some studies investigated the role of intraspecific variability in community assembly and evidenced environmental filtering [@Messier2010; @Paine2011], traits shift along environmental and resources gradient [@Siefert2016; @jung_intraspecific_2010 ], and  niche differentiation [@jung_intraspecific_2010; @Paine2011] at the individual level.

<!-- ITV Theory -->

Effectively, several author suggested that intraspecific variability helped to maintain biodiversity increasing species answer to their environment to pass environmental filtering [@Clark2010], exceeding negative interspecific interactions [@Chesson2000], and increasing species capacity to answer to their neighborhood [@Aarssen1983]. @Laughlin2012 even suggested that intraspecific variability could answer paradox in theories of species coexistence.

<!-- What we did -->

In the present study, we used a trait-based approach to address the role of ontogeny, abiotic environment and tree neighborhood in shaping leaf functional traits of tree individuals belonging to 5 species and 2 genera from two Neotropical species complexes. We took advantage of a large dataset (766 trees > 10 cm in diameter at breast height) from a highly diverse tropical forest plot located within the Guiana Shield from the Amazon Basin. The plots encompass a diversity of micro-habitats through topographic variation between seasonally flooded, bottom land, and dry plateau [@Allie2015] and competition variation between opened gaps and mature forest patches [@ferry2010higher] expected to strongly impacts tree functional traits. Combining tree level, LiDAR-derived topographical data and newly measured functional traits, we used regressions through a Bayesian framework to address the following questions:

<!-- Questions -->

* How traits co-vary between individuals within closely related tropical tree species ? 
* How ontogeny and environment shape the individual trait co-variation across ecological scales ?

<!-- Hypothesis -->

Based on conservation of functional strategies both at the community and species scale [@Bruelheide2018; @diaz_global_2015], we hypothesized trait co-variation to be maintained at the individual level [but see @messier_trait_2016]. In addition, we hypothesized ontogeny in interaction with abiotic environment and biotic interaction to shape individual trait variation without assuming response similarity or differences between species or genus and the remaining variance associated to intra-individual variation.

# Material and Methods

## Study site

The study was conducted in the northernmost part of the Guiana Plateau region, at the Paracou field station. The site was characterized by an average of 3041 mm annual rainfall and a mean air temperature of 25.71 °C. Old tropical forest with an exceptional richness (over 750 woody species) has developed among the succession of small hills of this area, rising to 10–40 m a.s.l. [@Gourlet-Fleury2004].

The site is made of 16 permanent plots (fifteen 6.25 ha plus one 25 ha) which have been censused (DBH>10) every 1-2 years for more than 35 years. Nine of the plots were logged and subjected to human-induced disturbance in 1986.


## Individuals sampling of traits values

Two sampling campaigns were led in 2017 and 2018 during dry season (between September and December). Sampling was made for respectively 14 and 2 species from *Eschweilera* (Lecythidaceae) and *Symphonia* (Clusiaceae) genera present in the Paracou station. Species from *Eschweilera* genus were all included in the *Parvifolia* clade [Heuertz *et al*. unpublished; @Mori2016]. Both *Eschweilera* clade *Parvifolia* and *Symphonia* species are Amazonian hyperdominant tree species [@TerSteege2013] and part of the most abundant trees within Paracou station (*Eschweilera sagotiana* from the clade *Parvifolia* is the most abundant tree species in 2017 for Paracou).

*Symphonia* genus only include the *Symphonia globulifera* L.f (Clusiaceae) species in French Guiana but presents two morphotypes, *S. globulifera* and *S. sp.1*, living in sympatry but apparently in differentiated habitats, with *S. globulifera* preferentially growing in valley bottoms and *S. sp1* preferentially exploiting a variety of drier habitats. Similarly, *Eschweilera* clade *parvifolia* include numerous species, with *E. sagotiana* and *E. coriacea* which have been shown to follow a niche differentiation similar to *S. sp1* and *S. globulifera* respectively [Schmitt *et al*. In prep;  @Allie2015]. In addition, *Symphonia* genus and *Eschweilera* clade Parvifolia have been both highlighted as species complexes with low genetic resolution and high plastid DNA sharing between sister species [Heuertz *et al*. unpublished; @baraloto_using_2012-1; @Gonzalez2009].

Between 232 and 1 individual have been sampled per species, depending on their natural abundance within the Paracou station, for a total of 823 individuals, but species with a minimum of 10 individuals were kept (Table \@ref(tab:headcount)). *Symphonia globulifera* were collected over all plots of the study site to reach targeted number, whereas *Eschweilera* clade *Parvifolia* were sampled only within non-disturbed plots. Anyway disturbance will be accounted through competition measurement and did not show significant effect for most of *Symphonia* functional traits (Student's T test p-value > 0.1 for SLA, LT and LA, and p-value > 0.02 for LDMC and CC). For each individual, 5 mature and healthy leaves at the top of the crown were collected, and kept in humidified ziplock with enriched CO2 air in dark until measurement within the next 6 hours following standard protocol [@Perez-Harguindeguy2013]. In addition, light conditions with Dawkins index [@Dawkins1958] and tree, bark and leaf morphology were assessed for every individuals.

```{r disturbance, eval=F}
Individuals %>% 
  filter(Genus == "Symphonia") %>% 
  mutate(disturbed = ifelse(Plot %in% c(2:5, 7:10, 12), 1, 0)) %>% 
  select(SLA, LDMC, LT, LA, CC, disturbed) %>% 
  reshape2::melt(id.vars = "disturbed",
                 variable.name = "trait") %>% 
  group_by(trait) %>% 
  do(student = t.test(value ~ disturbed, data = .)) %>% 
  broom::tidy(student) %>% 
  select(trait, p.value)
```

## Trait measurements

We calculated 5 functional traits from leaf samples already highlighted in previous studies as phenotypic descriptor of major ecological strategies [@diaz_plant_1998; @wright_worldwide_2004] (Table \@ref(tab:Trait)): leaf mass per area (fresh weight divided by the leaf area, $LMA$, $cm^2.g^{-1}$), leaf dry matter content (leaf dry weight divided by the fresh weight, $LDMC$, $g.g^{-1}$), leaf thickness ($LT$, $\mu m$), leaf area ($LA$, $cm^2$), leaf chlorophyll content ($CC$, $g.cm^{-2}$). Leaf petiole was removed for all measurements. Leaf fresh weight was measured with an analytic balance [brand], leaf fresh thickness with a micrometer [brand], leaf area through the ImageJ software [@Schneider2012] using scanned images of fresh leaves, and leaf chlorophyll content with a SPAD measurement [brand] converted with an allometric model [@Coste2010]. Leaf thickness and chlorophyll content measurement was repeated 3 times per leaf to reduce errors due to intra-leaf variation. Leaves were then kept in herbarium and oven-dried for at least 48 hours at 80°C before measurement of dry weight. Subsequent analysis will use $\frac{1}{LA}$ to ease regressions with functional traits co-varying with diameter at breast height ($DBH$).

## Environmental variables

Three little correlated (Pearson's r < 0.14) ontogenetic and environmental descriptors were chosen to depict individual leaf traits. Diameter at breast height ($DBH$, $cm$) was chosen as an indirect proxy for ontogeny effect [but see @roggy_links_2005] but was shown to be confounded with light access through Dawkins index (see supplementary material S1 [A02-Environment](A02-Environment.html)).

A neighbor crowding index [$NCI$; @Uriarte2004a] was used as a descriptor of biotic competition through tree neighborhood, implying that other biotic effects, as herbivory and pathogens effects, were not directly taken into account. The neighborhood crowding index $NCI_i$ from tree individual $i$ was calculated with the following formula:
$$NCI_i = \sum _{j|\delta_{i,j}<20m} ^{J_i} DBH_j ^\beta e^{-\alpha.\delta_{i,j}}$$
with $DBH_j$ the diameter from neighboring tree $j$ and $\delta_{i,j}$ its distance to individual tree $i$. $NCI_i$ is computed for all neighbors at a distance $\delta_{i,j}$ inferior to 20m as often observed in literature and because NCI effect showed very little effect beyond 20m in preliminary analysis. $\beta$ represents the size effect of neighbors through their diameter and was set to 2 to represent neighbors through their surface. $\alpha$ represents the decrease of neighbors $DBH_j$ effect with distance and was set to 0.25 corresponding to neighbors above 20m having less than 1% effect on $NCI$.

Finally, topographic wetness index ($TWI$) was selected between various abiotic descriptors available (see supplementary material S1 [A02-Environment](A02-Environment.html)) and highlighted water accumulation areas, which is crucial information at local scale when it comes to tropical ecosystems [@ferry2010higher]. Descriptors used were derived from field censuses ($DBH$ and $NCI$) or derived from a 1-m resolution digital elevation model built using LiDAR campaign done in 2015 ($TWI$) using SAGA-GIS [ref].


## Analysis

We first explored leaf functional traits co-variation between individuals with principal component analyses (PCA). Species organisation within the PCA was explored by ANOVA on PCA axis from the first plane with post-hoc groups investigated by Tukey honest significant differences.

Then, we explained leaf trait $Trait$ variation with ontogeny ($DBH$), abiotic environment ($TWI$), biotic interactions ($NCI$) and species ($S$) thanks to a model with fixed species effect including interaction between ontogeny and environment (more likely than an additive form without interactions, see supplementary material S2 [A07-Additive](A07-Additive.html)).

Individual leaf trait $Trait_{s,i}$ for species $s \in [1;S]$ is first explained by its ontogeny confounded with light though individual DBH $DBH_i$ with a Michaelis Menten form (more likely than a linear form, see supplementary material S3 [A05-Ontogeny](A05-Ontogeny.html)):
$$Trait_{s,i} \sim \mathcal{N}(\alpha_{env}.\frac{DBH_i}{{\beta_{DBH}}_{s} + DBH_i}, \sigma)$$
$\alpha_{env}$ is the trait maximum value in the Michaelis Menten and $\beta_{DBH}$ is the value of DBH for which the trait account for half of its plateau value (when DBH tend to its maximum value). With higher $\beta_{DBH}$, traits evolves slower towards its plateau, consequently, ontogeny affects more traits values. We will further use $DBH_{90}$ equal to $9.\beta_{DBH}$ being the DBH value for which the trait account for 90% of its maximum value that we can assume as the DBH for which the individual trait value is close to the maximum ontogenetic value.

Finally $\alpha_{env}$ was represented by an additive linear form of environmental effects represented by $TWI$ for abiotic environment and by $NCI$ for biotic interactions:
$$alpha_{env} = \alpha_{s} + {\beta_{TWI}}_{s}.TWI_i + {\beta_{NCI}}_{s}.NCI_i$$
where $\alpha$ is the intercept, $\beta_{TWI}$ is the slope of TWI effect, e.g. positive $\beta_{TWI}$ represents an increase of the trait value with water accumulation; whereas $\beta_{NCI}$ is the slope of competition effect, e.g. negative $\beta_{NCI}$ highlight a decrease of trait value with competition.

Traits, DBH, TWI and NCI were all reduced for the model inference, in order to ease model inference and later compare strength of effects between traits and between effects. A Bayesian method was used to infer parameters of the model regarding each leaf traits and the two complex, resulting in 10 regressions, using `stan` language [ref] and `rstan` package [ref] in the R environment [ref] (see Supplementary material S4 [A08-Interaction](A08-Interaction.html) for model equations and `stan` code).

Finally, to assess the relative importance of ontogeny and environmental variables we conducted a sensitivity analysis of co-variables. We compared the full model prediction $\hat {Y_{full}}$ against the model prediction with all variables set to their mean except the tested one $\hat {Y_{variable}}$ with the following formula:
$$\Delta(Prediction) = \sqrt{(\hat {Y_{full}} - \hat {Y_{variable}})^2}$$
Thus, the more the model is sensitive to a variable the less the prediction will change resulting in low values of $\Delta(Prediction)$.

# Results

70% of the total variance was conserved on the first plane from the PCA of leaf traits (Fig. \@ref(fig:PCA)). Globally, the first plane highlighted the co-variation of the five leaf traits on the first axis bearing almost 40% of the total trait variation, with big, thick leaves rich in dry matter and chlorophyll content with high LMA. Finally, besides both axes showed significative species segregation ($p-value < 10^{-4}$), the second PCA axis segregated the two complexes (*Symphonia* and *Eschweilera* clade Parvifolia) with little overlap when the first axis better ranked species.

Intercept posteriors and PCA (Fig. \@ref(fig:PCA) and \@ref(fig:Intercept)) highlighted complexes and species differences with $LA$ and $LT$ showing more species differentiation than $LMA$, $LDMC$ and $CC$, for which *Symphonia* species values were equal. **Useless ?**

$DBH$ significantly increased $LMA$, $LT$, and decreased $LA$ of the two species complexes whereas $DBH$ had no effect for *Eschweilera* and a medium positive effect for *Symphonia* on $CC$ and $LDMC$ (Fig. \@ref(fig:DBH90) and \@ref(fig:DBHpred) and Table \@ref(tab:parameters)).

$TWI$ had a significative negative effect on $LMA$, $LDMC$ and $CC$ and positive effect on $LA$ for *Symphonia* complex whereas $TWI$ had significative positive effect for $CC$ and negative effect for $LT$ for *Eschweilera* complex (Fig. \@ref(fig:TWI) and \@ref(fig:DBHpred) and Table \@ref(tab:parameters)). Finally, only *E. decolorans* showed a negative answer of $LMA$ and $LDMC$ to $TWI$.

$NCI$ had globally no effect on leaf trait variation except a negative effect on *Sympohnia* $LA$ (Fig. \@ref(fig:NCI) and \@ref(fig:DBHpred) and Table \@ref(tab:parameters)).

Finally, models showed a strong residual variation (from 0.5 to 1, Table \@ref(tab:parameters)) and were sensitive primarily to $DBH$, and secondly equally sensitive to $TWI$ and $NCI$ at the exception of $CC$ for *Eschweielra* more sensitive to $TWI$ than the two remaining co-variables (Fig. \@ref(fig:sensitivity)). **Useless after posteriors results ?**

# Discussion

## Take home message

> Individual leaf functional traits varies coordinately between species of species complexes mainly with ontogeny in interaction with abiotic environment through light and water accumulation.

## Trait co-variations

**Nothing really new, this paragraph is mainly useful to describe covaraition and functional strategy to be used in next paragraphs. Still the covariation of LA and CC are not well explained in litterature and might be interessant to be discussed.**

First PCA axis highlighted co-variation of big, thick leaves rich in dry matter and chlorophyll content with high LMA similar to the co-variation of leaves with dense tissues and high chlorophyll content with low SLA observed by @Baraloto2010 and interpreted as the leaf economics spectrum [@wright_worldwide_2004]. In addition LMA, LDMC, and LT have been shown to participate to a trade-off between resource acquisition by photosynthesis and investment in leaf defense and durability [@Evans2001; @hodgson_is_2011; @Vile2005]. Moreover LA participation to the leaf economics spectrum is debated [@Ackerly2002] and its role is still unclear [@Nicotra2011]. But @Yang2010 evidenced a negative ratio between leaf area and mass interpreted as an investment in photosynthesis or lifespan for large leaves. We effectively found LA to be orthogonal to LMA but co-varying with LDMC (Fig. \@ref(fig:PCA)), which may illustrate the ambiguous strategy of high value of LA between defense and acquisition. Finally, CC was co-varying along the major axis of functional strategy opposing acquisition to conservation and was correlated to LMA and LT. @Baraloto2010 highlighted CC participation to the LES, interestingly opposed to SLA and the leaf nitrogen content without further interpretation. **...**

## DBH effect

**We should accentuate the fact that our study found a continuous and late effect of DBH on most of functional traits for tree > 10 cm DBH opposed to studies focused on young trees and only using ANOVAs between classes.**

DBH globally increased trait values highlighting a shift of individual leaves toward small and thick leaves with high LMA, and even increased dry matter content and chlorophyll for *Symphonia*. This can be interpreted as a shift from acquisitive to conservative leaves during individual ontogeny confounded with individual access to light as bigger DBH represent both old trees and canopy trees with full light access. Ontogeny has been highlighted as a major driver of individual phenotype through DBH effect on growth [@Herault2011] with similar response between species. But growth response to ontogeny present a hump-shaped form, indicating a maximum growth at intermediate DBH, different from our Michaelis Menten form indicating continuous evolution of functional traits toward a plateau corresponding to trait value at high ontogenetic stage and increased light level. Similar leaf functional trait evolution have also been shown with a log relation [@Niinemets2006]. __TBD ?__ 

Leaves functional traits evolution toward conservative strategy values have been already highlighted numerous times with ontogeny [@Lusk2007;  @Spasojevic2014], light [@Niinemets2006; @Rozendaal2006b] or both [@roggy_links_2005; @Sciences2009; @Dang-Le2013; @Rijkers2000]. The pattern has been highlighted on LMA, LT, LDMC (or similarly LTD), and CC and few times on LA [@Dang-Le2013; @Rozendaal2006b], besides few articles also found little to no effect on LDMC [@roggy_links_2005] and CC [@Rijkers2000; @Rozendaal2006b] as highlighted in our study. __TBD ?__

Furthermore, @Wright2002a and @Coleman1994 advocated for a comprehensive assessment of ontogeny to evaluate functional traits relation between them and with environment. But previously highlighted effect of ontogeny and light on functional traits have been shown to be difficulty disentangled [@Lusk2007; @roggy_links_2005], questioning the concept of sun-shade leaves [@Lusk2007; @Dang-Le2013], but see [@Poorter2012] who highlighted light effect correcting for ontogeny. Still light and ontogeny showed little to no interaction [@roggy_links_2005; @Sciences2009] with a greater effect of ontogeny [@Rijkers2000; @Rozendaal2006b] and reduced effect of light on functional trait values [@Dang-Le2013; @Rozendaal2006b]. Those results together with the fact that we found a continuous and late effect of DBH on most of our traits question studies not correcting for ontogeny and advocating light effects on tree functional traits [@Rozendaal2006b].

LMA increase with ontogeny has been related to water potential decrease with tree height, due to gravity or path length resistance [@Koch2004; @WOODRUFF2007]; whereas LMA increase with light might be the results of a thicker mesophyll with elongated and more layers of palisade parenchyma [@Chazdon1993]. Both phenomenon are also consistent with smaller and thicker leaves with increasing tree size and light availability. In addition, smaller and thicker leaves with increased LMA are thus better protected against herbivory and risk of mechanical damages [@Sterck1992]. On the other hand, smaller trees in shadow environment should gain from acquisitive strategy to increase light capture [@Chazdon1996] and insure increased development [@Herault2011].

## Environment effect

**Besides little, the effect of water and topography within species is similar to the one between species with drought tolerance in drier habitat and acquisition in fertile and lighter bottom lands.**

Both biotic and abiotic local environment had globally little or no effect on leaf trait variation at the intraspecific level. **Environment as a broader effect on species assemblage and community => Messier ?** **Weight of intra-individual variation and the strong uncertainty around parameters and remaining variance, cf following paragraph.**

Nevertheless, water accumulation resulted in bigger leaves with decreased dry matter content and lower LMA for *Symphonia* and *E.decolorans* and thinner leaves for *Eschweilera*, indicating a shift from conservative to acquisitive species as water became available for individuals. 

Effectively, several studies have linked topography and/or water availability with functional traits at different scales highlighting divergence of responses. Whereas LT and LMA were found to increase along a precipitation gradient [@Santiago2004] and LDMC to increase in lower topographic position [@Mendez-Toribio2017] at the community level, revealing more conservation and drought tolerance with increasing water availability ; @kraft_functional_2008 found bigger leaves with decreased LMA in valley bottoms compared to ridge-tops at the species level, similarly to what we found at the individual level.

Moreover, increase in LA with a decrease of LDMC and LMA observed within species with increasing water availability correspond to observed differences of species mean traits in relation to their habitat preferences, i.e. _S. globulifera_ and _E. coriacea_ have respectively bigger leaves with less dry matter and mass per area than _E. sagotiana_ and _S. sp1_ and they preferentially grow in flooded valley bottom than drier slopes and hilltops [@Allie2015; Schmitt et al. unpublished]. This observation combined to our results and @kraft_functional_2008 suggests that interspecific functional answer to topography and water is conserved between individual of the species itself.

Several hypothesis can explain observed trait variation with topographic wetness index. Denser tissues with smaller leaves allow more drought tolerance of individuals with less water availability within species [@Mendez-Toribio2017]. On the opposite, bottom lands are characterized by higher treefall, recruitment and growth rates inducing a strong light gradient combined with a higher fertility [@ferry2010higher] which might explain the functional filtering of acquisitive traits with increasing water availability. This pattern is again also highlighted at the species level by the increased maximum diameter of _S. globulifera_ and _E. coriacea_ respectively compared to _E. sagotiana_ and _S. sp1_ [@Allie2015].

In addition, chlorophyll content showed opposite response to water accumulation between the two complexes **??**. 

Finally, competition only decreased leaf size of *Symphonia* species __...__

## Variation structure

**Traits variation between ecological scale (community, speices, individual) and methodological implications (Fig. \@ref(fig:bootstrap))**

As highlighted by posteriors, once the mean differences between species leaf functional traits taken into account, leaf functional trait variability was primarily due to ontogeny and light access and a little by the interaction with environment mainly represented by abiotic environment through water accumulation. Still the remaining variation remains strong ($\sigma ^2 \in [0.5;1]$) due to measurement errors, seasonal variation (Supp Mat ...) and mainly intra-individual variation **...**

Literature:

* "The questions is also raised as to whether leaf traits correlate more strongly with ontogeny than with not only light intensity but also other environ- mental factors." [@Dang-Le2013]
* LMA & Height within individual [@Koch2004]

# Supplementary materials 

* Supplementary material S1 [A02-Environment](A02-Environment.html)
* Supplementary material S2 [A07-Additive](A07-Additive.html)
* Supplementary material S3 [A05-Ontogeny](A05-Ontogeny.html)
* Supplementary material S4 [A08-Interaction](A08-Interaction.html)


# Acknowledgements

We thanks the university of Bordeaux for a PhD grant to Sylvain Schmitt and acknowledge support of a grant from Investissement d’Avenir grants of the ANR (CEBA:ANR-10-LABX-25-01) and from GUYAMAZON program (LECYTHOMICS project). We are grateful to Pascal Petronelli and the CIRAD inventory team for their work on tree inventories and botanical identification. Special thanks go to Saint-Omer and Josselin Cazal, Ilke Gelaldi, Fabien Lehuede, Adeline Adam, Agathe Benfredj Zaleski, Numa Faucherre, and David Zipper for their assistance during sampling in Paracou station.

# Authors’ contributions

SS, GD, BH, ED, and AB conceived the ideas; SS, GD, BH, and AB designed methodology; SS and GD analysed model outputs; SS, and GD led the writing of the manuscript. All authors contributed critically to the drafts and gave final approval for publication.

# References

<div id="refs"></div>

\newpage 

## Table captions

__Table \@ref(tab:headcount)__ Number of individuals sampled per genus and species.

__Table \@ref(tab:Trait)__ Functional traits measured, with trait unit, abbreviation, headcount, mean and standard error.

__Table \@ref(tab:parameters)__ Model parameters for complex, species and traits.

__Table \@ref(tab:lm)__ Similar linear model results.

## Figure captions

__Figure \@ref(fig:PCA)__ Principal Component Analysis (PCA) of leaf trait  between species and species segregation on the two first axis. Dot and box colors indicates the species complex, whereas dot size indicates individual diameter at breast height. Species segregation has been investigated generally by Anova, **** indicates a $p-value < 0.0001$ and letters indicate post-hoc groups investigated by Tukey Honest Significant Differences. See table \@ref(tab:Trait) for traits abbreviation.

__Figure \@ref(fig:Intercept)__  Species model intercept.

__Figure \@ref(fig:DBH90)__  Species DBH effect ($DBH_{90}$).

__Figure \@ref(fig:DBHpred)__  Model predictions for DBH.

__Figure \@ref(fig:TWI)__  Species TWI effect.

__Figure \@ref(fig:NCI)__  Species NCI effect.

__Figure \@ref(fig:sensitivity)__ Model sensitivity to variables. ...

__Figure \@ref(fig:bootstrap)__  Coefficient of variation of mean estimation depending on the number of individuals sampled of DBH > 30 cm.

\newpage 

<!-- tables -->

```{r headcount}
Individuals %>% 
  mutate(Species = gsub("_", " ", Species)) %>% 
  group_by(Genus, Species) %>% 
  summarise(N = n()) %>% 
  arrange(desc(N)) %>% 
  kable(caption = "Number of individuals sampled per genus and species.",
        escape = F, format = "pandoc") %>% 
  kable_styling(full_width = F)
```

```{r Trait}
Individuals0 %>% 
  mutate(LMA = (1/SLA)*10^4) %>% 
  select(LMA, LDMC, LT, LA, CC) %>% 
  reshape2::melt(variable.name = "trait") %>% 
  filter(!is.na(value)) %>% 
  group_by(trait) %>% 
  summarise(N = n(),
            Value = paste0("$",round(mean(value), 2), "\\pm", 
                           round(sd(value), 2), "$")) %>% 
  mutate(Trait = c("Leaf Mass per Area", "Leaf Dry Matter Content",
                    "Leaf Thickness", "Leaf Area", "Chlorophyll Content")) %>% 
  mutate(Unit = c("$g.m^{-2}$", "$g.g^{-1}$", "$\\mu m$", "$cm^2$", "$g.cm^{-2}$")) %>%
  select(Trait, Unit, trait, N, Value) %>% 
  kable(col.names = c("Trait", "Unit", "Abbreviation", "N", "Mean (sd)"), format = "pandoc",
    caption = "Functional traits measured, with trait unit, abbreviation, headcount, mean and standard error.", escape = F) %>% 
  kable_styling(full_width = F)
```

```{r parameters}
lapply(fits, function(fit)
  broom::tidyMCMC(fit, pars = c("alpha", "betaDBH", "betaTWI",
                                "betaNCI", "sigma"), 
                  droppars = NULL, rhat = T)) %>%
  bind_rows(.id = "model") %>% 
  separate(model, c("complex", "trait"), "-") %>% 
  mutate(species = gsub("([[:alpha:]])", "", term)) %>% 
  select(complex, trait, term, species, estimate, std.error, rhat) %>% 
  mutate(species = gsub("([[:punct:]])", "", species)) %>% 
  mutate(species = as.numeric(species)) %>% 
  left_join(species) %>% 
  mutate(term = gsub("([[:digit:]])", "", term)) %>% 
  mutate(term = gsub("([[:punct:]])", "", term)) %>% 
  mutate(term = recode_factor(term, 
                              alpha = "$\\alpha$",
                              betaDBH = "$\\beta_{DBH}$", 
                              betaTWI = "$\\beta_{TWI}$", 
                              betaNCI = "$\\beta_{Comp}$",
                              sigma = "$\\sigma^2$")) %>% 
  mutate(trait = recode_factor(trait, 
                              LMA = "$LMA$",
                              LDMC = "$LDMC$", 
                              LT = "$LT$", 
                              invLA = "$\\frac{1}{LA}$",
                              CC = "$CC$")) %>% 
  reshape2::dcast(complex + term + SpeciesLong ~ trait, value.var = "estimate") %>% 
  mutate(SpeciesLong = ifelse(is.na(SpeciesLong), "", SpeciesLong)) %>% 
  rename(Parameter = term, Species = SpeciesLong) %>% 
  kable(caption = "Model parameters for complex, species and traits.",
        escape = F, digits = 3, format = "pandoc") %>% 
  kable_styling(full_width = F) %>% 
  column_spec(1, bold = T) %>%
  collapse_rows(columns = 1:2)
```

```{r lm}
Individuals %>%
  select(Genus, Species, DBH, TWI, NCI, LMA, LDMC, LT, invLA, CC) %>%
  reshape2::melt(id.vars = c("Genus", "Species", "DBH", "TWI", "NCI"),
                 variable.name = "Trait") %>%
  group_by(Trait, Genus) %>%
  do(model = lm(value ~ Species + log(DBH) + log(DBH):TWI + log(DBH):NCI, data = .)) %>% 
  broom::tidy(model) %>% 
  ungroup() %>% 
  filter(term != "(Intercept)") %>% 
  mutate(term = gsub("Species", "", term)) %>% 
  mutate(`P-value` = ifelse(p.value < 0.001, "***",
                            ifelse(p.value < 0.01, "**",
                                   ifelse(p.value < 0.05, "*",
                                          ifelse(p.value < 0.1, ".", 
                                                 "n.s."))))) %>% 
    mutate(Trait = recode_factor(Trait, 
                              LMA = "$LMA$",
                              LDMC = "$LDMC$", 
                              LT = "$LT$", 
                              invLA = "$\\frac{1}{LA}$",
                              CC = "$CC$")) %>% 
  mutate(Coefficient = round(estimate,2)) %>% 
  mutate(value = paste(Coefficient, `P-value`)) %>%
  reshape2::dcast(term + Genus ~ Trait, value.var = "value") %>% 
  kable(escape = F, format = "pandoc",
        caption = "Similar linear model results.") %>% 
  kable_styling(full_width = F)
```

<!-- figures -->

```{r PCA, fig.cap='Principal Component Analysis (PCA) of leaf trait  between species and species seggregation on the two first axis. Dot and box colors indicates the species complex, whereas dot size indicates individual diameter at breast height. Species seggregation has been investigated generally by Anova and letters indicate post-hoc groups investigated by Tukey Honest Significant Differences. See table \\@ref(tab:Trait) for traits abbreviation.'}
pca.plot <- drop_na(Individuals, LMA, LDMC, LT, LA, CC) %>% 
  autoplot(princomp(~ LMA + LDMC + LT + LA + CC, data = ., cor = T), 
           data = .,
           colour = "SpeciesLong", alpha = 0.5, size = "DBH",
           loadings.label.size = 6,
           loadings.label.colour = 'black', loadings.label.vjust = 1.1,
           loadings = T, loadings.label = T, loadings.colour = 'black') +
  coord_equal() +
  geom_hline(aes(yintercept = 0), col = 'black', linetype = "dotted") +
  geom_vline(aes(xintercept = 0), col = 'black', linetype = "dotted") +
  scale_size_continuous(guide = "none") +
  scale_colour_brewer("", palette = "Set1") +
  guides(col = guide_legend(nrow=3)) +
  theme(legend.position = "bottom")
tHSD <- pca.plot$data %>%
  select(Comp.1, Comp.2, SpeciesLong) %>%
  group_by(SpeciesLong) %>% 
  filter(n() > 10) %>% 
  reshape2::melt(id.vars = "SpeciesLong", variable.name = "axis") %>% 
  group_by(axis) %>% 
  do(anova = aov(value ~ SpeciesLong, data = .)) %>% 
  mutate(tHSD = TukeyHSD(anova, ordered = F, conf.level = 0.95)) %$%
  lapply(tHSD, function(x) multcompView::multcompLetters(x[,4])) %>% 
  lapply(function(x) data.frame(SpeciesLong = names(x$Letters), group = x$Letters)) %>% 
  bind_rows() %>% 
  mutate(axis = c(rep("Comp.1", nrow(species)), rep("Comp.2", nrow(species))))
axes.boxplots <- pca.plot$data %>%
  select(Comp.1, Comp.2, SpeciesLong) %>%
  reshape2::melt(id.vars = "SpeciesLong", variable.name = "axis") %>%
  left_join(tHSD) %>% 
  ggplot(aes(datatools::reorder_within(SpeciesLong, value, axis),
             value, fill = SpeciesLong, label = group)) +
  geom_boxplot() +
  geom_text(aes(y = max(value)*0.8)) +
  datatools::scale_x_reordered() +
  facet_wrap(~axis, scales = "free", nrow = 2) +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  scale_fill_brewer("", palette = "Set1", guide = "none")
cowplot::plot_grid(pca.plot, axes.boxplots, rel_widths = c(3,2))
```

```{r Intercept, fig.cap="Species model intercept."}
lapply(fits, function(fit) 
  mcmc_intervals_data(as.array(fit,
                               pars = "alpha"))) %>% 
  bind_rows(.id = "model") %>% 
  separate(model, c("complex", "trait"), "-") %>%
  mutate(species = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(species = gsub("([[:alpha:]])", "", species)) %>% 
  mutate(species = as.numeric(species)) %>% 
  left_join(species) %>% 
  mutate(trait = recode_factor(trait, invLA = "frac(1, LA)")) %>% 
  ggplot(aes(x = datatools::reorder_within(SpeciesLong, m, trait), 
             xend = datatools::reorder_within(SpeciesLong, m, trait),
             color = SpeciesLong, fill = SpeciesLong)) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  coord_flip() +
  datatools::scale_x_reordered() +
  facet_grid(trait ~ ., labeller = label_parsed, scales = "free_y") +
  xaxis_title(F) +
  ylab(expression(alpha)) +
  scale_y_continuous(trans="S_sqrt") +
  scale_color_brewer("", palette = "Set1", guide = "none") +
  scale_fill_brewer("", palette = "Set1", guide = "none")
```

```{r DBH90, fig.cap="Species DBH effect ($DBH_{90}$)."}
lapply(fits, function(fit) 
  mcmc_intervals_data(as.array(fit,
                               pars = "betaDBH"))) %>% 
  bind_rows(.id = "model") %>% 
  separate(model, c("complex", "trait"), "-",remove = F) %>%
  mutate(species = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(species = gsub("([[:alpha:]])", "", species)) %>% 
   mutate(species = as.numeric(species)) %>% 
  left_join(species) %>% 
  mutate(trait = recode_factor(trait, invLA = "frac(1, LA)")) %>% 
  mutate_at(vars("ll", "l", "m", "h", "hh"), funs(.*sdDBH*9)) %>% 
  ggplot(aes(x = datatools::reorder_within(SpeciesLong, m, trait), 
             xend = datatools::reorder_within(SpeciesLong, m, trait),
             color = SpeciesLong, fill = SpeciesLong)) +
  geom_hline(yintercept = c(10, 57, 68), color = "gray90", size = 0.5) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  coord_flip() +
  datatools::scale_x_reordered() +
  facet_grid(trait ~ Genus, labeller = label_parsed, 
             scales = "free_y") +
  xaxis_title(F) +
  ylab(expression(DBH[90]~(cm))) +
  scale_y_log10() +
  scale_color_discrete(guide = "none") +
  scale_fill_discrete(guide = "none")
```

```{r DBHpred, fig.cap="Model predictions for DBH."}
preds %>% 
  mutate(Trait = recode_factor(Trait, invLA = "frac(1, LA)")) %>% 
  ggplot(aes(DBH*sdDBH, col = Species)) +
  geom_point(aes(y = Y), alpha = 0.2) +
  geom_ribbon(aes(ymin = Ydbh5, ymax = Ydbh95), alpha = 0.2) +
  geom_line(aes(y = Ydbh)) +
  facet_grid(Trait ~ Complex, scales = "free", labeller = label_parsed) +
  scale_y_log10()
```

```{r TWI, fig.cap="Species TWI effect."}
lapply(fits, function(fit) 
  mcmc_intervals_data(as.array(fit,
                               pars = "betaTWI"))) %>% 
  bind_rows(.id = "model") %>% 
  separate(model, c("complex", "trait"), "-") %>%
  mutate(species = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(species = gsub("([[:alpha:]])", "", species)) %>% 
    mutate(species = as.numeric(species)) %>% 
  left_join(species) %>% 
  mutate(trait = recode_factor(trait, invLA = "frac(1, LA)")) %>% 
  ggplot(aes(x = datatools::reorder_within(SpeciesLong, m, trait), 
             xend = datatools::reorder_within(SpeciesLong, m, trait),
             color = SpeciesLong, fill = SpeciesLong)) +
  geom_hline(yintercept = 0, color = "gray90", size = 1) +
  geom_hline(yintercept = c(-0.25, 0.25), 
             color = "gray90", size = 0.5, lty = "dashed") +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  coord_flip() +
  facet_grid(trait ~ Genus, labeller = label_parsed, 
              scales = "free_y") +
  datatools::scale_x_reordered() +
  xaxis_title(F) +
  ylab(expression(beta[TWI])) +
  scale_y_continuous(trans="S_sqrt") +
  scale_color_discrete(guide = "none") +
  scale_fill_discrete(guide = "none") +
  ylim(-0.5, 0.5)
```

```{r NCI, fig.cap="Species NCI effect."}
lapply(fits, function(fit) 
  mcmc_intervals_data(as.array(fit,
                               pars = "betaNCI"))) %>% 
  bind_rows(.id = "model") %>% 
  separate(model, c("complex", "trait"), "-") %>%
  mutate(species = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(species = gsub("([[:alpha:]])", "", species)) %>% 
    mutate(species = as.numeric(species)) %>% 
  mutate(trait = recode_factor(trait, invLA = "frac(1, LA)")) %>%
  left_join(species) %>% 
  ggplot(aes(x = datatools::reorder_within(SpeciesLong, m, trait), 
             xend = datatools::reorder_within(SpeciesLong, m, trait),
             color = SpeciesLong, fill = SpeciesLong)) +
  geom_hline(yintercept = 0, color = "gray90", size = 1) +
  geom_hline(yintercept = c(-0.25, 0.25), 
             color = "gray90", size = 0.5, lty = "dashed") +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  coord_flip() +
  datatools::scale_x_reordered() +
    facet_grid(trait ~ Genus, labeller = label_parsed, 
              scales = "free") +
  xaxis_title(F) +
  ylab(expression(beta[NCI])) +
  scale_y_continuous(trans="S_sqrt") +
  scale_color_discrete(guide = "none") +
  scale_fill_discrete(guide = "none") +
  ylim(-0.5, 0.5)
```

```{r sensitivity, fig.cap="Model sensitivity to variables. ..."}
preds %>%
  select(Trait, Genus, Yp, Ydbh, Ytwi, Ynci) %>% 
  rename(DBH = Ydbh, TWI = Ytwi, NCI = Ynci) %>% 
  reshape2::melt(id.vars = c("Trait", "Genus", "Yp")) %>% 
  mutate(Delta = sqrt((Yp - value)^2)) %>% 
  mutate(variable = factor(variable, levels = c("NCI", "TWI", "DBH"))) %>%
  ggplot(aes(variable, Delta, fill = variable)) +
  geom_boxplot() +
  facet_grid(Trait ~ Genus, scales = "free") +
  coord_flip() +
  scale_y_sqrt() +
  scale_fill_discrete(guide = "none") +
  xlab("") + ylab(expression(Delta(Prediction)))
```

```{r bootstrap, fig.cap="Coefficient of variation of mean estimation depending on the number of individuals sampled of DBH > 30 cm."}
t <- Individuals0 %>% 
  mutate(LMA = (1/SLA)*10^4) %>% 
  group_by(Species) %>% 
  filter(n() > 10) %>% 
  filter(DBH*sdDBH > 30) %>%
  select(Genus, Species, LMA, LDMC, LT, LA, CC) %>% 
  reshape2::melt(id.vars = c("Genus", "Species"), variable.name = "Trait") %>% 
  na.omit() %>% 
  group_by(Genus, Species, Trait, value) %>% 
  expand(n = 2:10) %>% 
  group_by(Genus, Species, Trait, n) %>% 
  do(cv = sd(apply(replicate(1000, sample(.$value, .$n)), 2, mean))/mean(.$value)*100) %>% 
  broom::tidy(cv) %>% 
  rename(cv = x)
ggplot(t, aes(as.factor(n), cv, fill = Trait)) +
  geom_boxplot() +
  stat_summary(fun.y=mean, geom="line", aes(group=1)) +
  facet_grid(~ Trait) +
  scale_fill_discrete(guide = "none") +
  xlab("Sampled tree number") +
  ylab("Coefficient of variation (%).") +
  scale_y_sqrt()
```

```{r varcomp, fig.cap="Variance decomposition..."}
traitsSymphonia <- googlesheets::gs_title("Measures_Symphonia") %>%
  googlesheets::gs_read("AllTraits") %>%
  mutate(SLA = as.numeric(SLA), LDMC = as.numeric(LDMC)) %>%
  mutate(Genus = "Symphonia") %>%
  rename(Species = Morphotype) %>%
  mutate(Species = ifelse(Species == "Indet.",
                          c("globulifera", "sp.1", "sp.1")[fct_recode(Bark, "globulifera" = "G",
                                     "sp.1" =  "S")], Species))
traitsEschweilera <- googlesheets::gs_title("Measures_Eschweilera") %>%
  googlesheets::gs_read("AllTraits") %>%
  filter(!(Plot == 14 & SubPlot == 1 & TreeFieldNum == 760)) %>%  # outlier
  filter(!(Species %in% c("congestiflora","simiorum","persistens")))
traits <- bind_rows(traitsEschweilera, traitsSymphonia) %>%
  mutate(SpeciesLong = paste(substr(Genus, 1, 1), Species)) %>%
  group_by(idTree, Plot, SubPlot, TreeFieldNum, Genus, Species,
           SpeciesLong, Bark, Dawkins)
traits %>% 
  mutate(LMA = (1/SLA)*10^4) %>% 
  group_by(Species) %>% 
  filter(n() > 10) %>% 
  filter(CC > 0) %>% # CC have an issue for lme
  select(Plot, Genus, Species, idTree, Leaf,
         LMA, LDMC, LT, LA, CC) %>% 
  reshape2::melt(id.vars = c("Plot", "Genus", "Species", "idTree", "Leaf"), 
                 variable.name = "Trait") %>% 
  na.omit() %>% 
  group_by(Trait) %>% 
  do(varcomp = as.vector(ape::varcomp(nlme::lme(log(value) ~ Plot,
                            random=~1|Genus/Species/idTree/Leaf, 
                            data = ., na.action = na.omit), 1))) %>% 
  broom::tidy(varcomp) %>% 
  rename(Variance = x) %>% 
  mutate(Component = c("Genus", "Species", "Tree", "Leaf", "Residual")) %>% 
  ggplot(aes(Trait, Variance, label = round(Variance*100), 
             fill = factor(Component, levels = c("Genus", "Species", 
                                                 "Tree", "Leaf", "Residual")))) +
  geom_bar(stat = "identity") +
  scale_fill_brewer("Component", palette = "Accent") +
  geom_text(position = position_stack(vjust = 0.5))
```
