```{r setup_mapping, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(Biostrings)
library(GenomicAlignments)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/"
```

```{r, eval=F}
libraries <- read_delim(file.path(path, "Sequences", "libraries.uniq.txt"),
           delim = " ", col_names = c("Library"))
mapped <- read_delim(file.path(path, "Sequences", "mapped.txt"),
                     delim = " ", col_names = c("Library")) %>% 
  mutate(Library = gsub("mapping/bam/", "", Library)) %>% 
  mutate(Library = gsub(".bam", "", Library)) %>% 
  mutate(mapped = 1)
libraries %>% 
  left_join(mapped) %>% 
  filter(is.na(mapped)) %>% 
  select(-mapped) %>% 
  write_delim(file.path(path, "Sequences", "unmapped.txt"),
              delim = " ", col_names = F)
```

```{r mapping}

```


# Mapping

We proceeded to read mapping for every libraries on the built reference, already partly annotated:

1. __Repeats merging__: 41 libraries were repeated, we merged their alignment to increase their information before variant calling
1. __Reads mapping__: we mapped every libraries in pair end with `bwa mem` on the hybrid reference from Ivan Scotti et Sana Olson used to built the targets
1. __Reference sequences__: we built bedtools for every alignment in order to list sequences with mathces in the reference to be used to reduce explred reference area in variant calling

## Repeats merging

41 libraries were repeated, we merged their FASTQ to increase their information before variant calling. Meging repeats confirmed the presence of all 430 individuals at the end of the alignment (402 from Paracou, 20 from herbariums, and 8 from BCI, Itubera and La Selva).

```{bash compression, eval=F, echo=T}
#!/bin/bash
#SBATCH --time=36:00:00
#SBATCH -J compression
#SBATCH -o compression_output.out
#SBATCH -e compression_error.out
#SBATCH --mem=4G
#SBATCH --cpus-per-task=1
#SBATCH --mail-type=BEGIN,END,FAIL

folder=trimmed.paired.joined/
name=symcapture.trimmed.paired.joined

module purge
mkdir trimmed.paired
for file in $(ls paired/*)
do
  mv $file trimmed.paired/$(basename $(echo $file | sed -e 's/_[[:alpha:]]*-[[:alpha:]]*-[[:alnum:]]*//'))
done
rm -r paired
cp ../libraries.txt ./
cat libraries.txt | sed -e 's/_[[:alpha:]]*-[[:alpha:]]*-[[:alnum:]]*_L00[56]//' | sort | uniq | sed -e 's/-b//' | sort | uniq > libraries.uniq.txt
mkdir trimmed.paired.joined
for ind in $(cat libraries.uniq.txt)
do
  echo $ind
  cat trimmed.paired/$ind*_R1_paired.fq.gz > trimmed.paired.joined/"$ind"_R1_paired.fq.gz
  cat trimmed.paired/$ind*_R2_paired.fq.gz > trimmed.paired.joined/"$ind"_R2_paired.fq.gz
done
rm -r trimmed.paired
tar -zcvf $name.tar.gz $folder
```

## Reads mapping

We mapped every libraries in pair end with `bwa mem` on the hybrid reference from Ivan Scotti et Sana Olson used to built the targets (parralelizing with 32 alignement with 2 process on 64 cores of 1 node of genologin). We had globally a good mapping with more than 80% of the reads mapped for  98% of the libraries \@ref(fig:mappingStat).

```{bash mappingSH, eval=F, echo=T}
#!/bin/bash
#SBATCH --time=48:00:00
#SBATCH -J mapping
#SBATCH -o mapping_output.out
#SBATCH -e mapping_error.out
#SBATCH --mem=160G
#SBATCH --cpus-per-task=64
#SBATCH --mail-type=BEGIN,END,FAIL

module purge
module load bioinfo/bwa-0.7.15
module load bioinfo/picard-2.14.1
module load bioinfo/samtools-1.4
module load bioinfo/bedtools-2.26.0

task(){
	echo MAPPING "$1"
	bwa mem -M -R "@RG\tID:$1\tSM:$1\tPL:HiSeq4K" \
   		-t 2 \
   		reference/reference.fasta \
   		trimming/trimmed.paired.joined/"$1"_R1_paired.fq.gz \
   		trimming/trimmed.paired.joined/"$1"_R2_paired.fq.gz \
   		> mapping/sam/"$1".sam
   	rm trimming/trimmed.paired.joined/"$1"_R1_paired.fq.gz
   	rm trimming/trimmed.paired.joined/"$1"_R2_paired.fq.gz
	java -Xmx4g -jar $PICARD SortSam \
   		I=mapping/sam/"$1".sam \
   		O=mapping/bam2/"$1".bam
   		SORT_ORDER=coordinate
	rm mapping/sam/"$1".sam
	samtools index mapping/bam2/"$1".bam
}

mkdir mapping/sam
mkdir mapping/bam2
N=32
(
for library in $(cat unmapped.txt)
do 
   ((i=i%N)); ((i++==0)) && wait
   task "$library" &
done
)
rm -r mapping/sam
```

```{r mappingStat, fig.cap="Mapping result"}
read_delim(file.path(path, "Sequences", "mapping", "mapping_stat.txt"),
           delim = " ", col_names = c("Library", "readsMapped", "+", "0", "mapped", "percentage", ":", "N/A")) %>% 
  select(Library, readsMapped, percentage) %>% 
  mutate(Library = gsub("bam/", "", Library)) %>% 
  mutate(Library = gsub("_[[:alpha:]]*-[[:alpha:]]*-[[:alnum:]]*_L00[56].bam$", "", Library)) %>% 
  mutate(percentage = gsub("(", "", percentage, fixed = T)) %>% 
  mutate(percentage = gsub("%", "", percentage, fixed = T)) %>% 
  mutate(percentage = as.numeric(percentage)) %>% 
  mutate(type = ifelse(grepl("P", Library), "Paracou", "Outgroup")) %>% 
  ggplot(aes(percentage, fill = type)) +
  geom_histogram() +
  ggtitle("4 Paracou libraries < 75% coverage (99.2% above)", "10 Paracou libraries < 80% coverage (98% above)")
```

## Reference sequences

We built bedtools for every alignment in order to list sequences with matches in the reference to be used to reduce explred reference area in variant calling. 99.98% of reference sequences have at least one library matching (\@ref(fig:referenceSequences)). Ceonsequently we will use all sequences from the reference in the variant calling, besides few sequences our libraries are under represented, but they will be removed at the SNP filtering.

```{bash referenceSequencesSH, eval=F, echo=T}
#!/bin/bash
#SBATCH --time=36:00:00
#SBATCH -J referenceSequences
#SBATCH -o referenceSequences_output.out
#SBATCH -e referenceSequences_error.out
#SBATCH --mem=80G
#SBATCH --cpus-per-task=40
#SBATCH --mail-type=BEGIN,END,FAIL

module purge
module load bioinfo/bedtools-2.26.0

mkdir bed
N=40 # number of threads
(
for file in $(ls bam/*.bam)
do 
   ((i=i%N)); ((i++==0)) && wait
   bedtools bamtobed -i $file > bed/$(basename "${file%.*}").bed
done
)
(
for file in $(ls bed/*.bed)
do 
   ((i=i%N)); ((i++==0)) && wait
   bedtools merge -i bed/$(basename "${file%.*}").bed > merged.bed/$(basename "${file%.*}").merged.bed
done
)
rm -r bed
```

```{r referenceSequences, fig.cap="Sequences from reference alignment with reads from library."}
# bed <- lapply(list.files(file.path(path, "Sequences", "mapping", "merged.bed")), function(bed)
#   read_tsv(file.path(path, "Sequences", "mapping", "merged.bed", bed),
#            col_names = c("sequence", "start", "stop")) %>% 
#     select(sequence) %>% 
#     unique() %>% 
#     mutate(library = gsub(".merged.bed", "", bed))) %>% 
#   bind_rows()
# save(bed, file = "./symcapture_save/bed.Rdata")
# load(file = "./symcapture_save/bed.Rdata")
# ggplot(bed, aes(library, sequence)) +
#   geom_tile() +
#   coord_flip() +
#   theme(axis.text.x = element_blank(), 
#         axis.line.x =element_blank(), 
#         axis.ticks.x = element_blank(),
#         axis.text.y = element_blank(), 
#         axis.line.y =element_blank(), 
#         axis.ticks.y = element_blank()) +
#   ggtitle(paste(length(unique(bed$sequence)), " with a least one library match (99.98%)")) +
#   ggsave("./symcapture_save/seqAligned.png")
knitr::include_graphics("./symcapture_save/seqAligned.png")
```

