```{r setup_variant, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(Biostrings)
library(GenomicAlignments)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/"
```

```{r variant}
```

# Variant call

We used `GATK` as it has apparently similar performance to other variant callers [@Supernat2018] and was more known by Myriam. For that we used the following pipeline:

1. __Variant calling__ Run the `HaplotypeCaller` on each sample's BAM files to create single-sample gVCFs using the `.g.vcf` extension for the output file.
1. __Data aggregation__ Aggregate the GVCF files and feed in one GVCF with `GenomicsDBImport` to be genotyped
1. __Joint genotyping__ Run `GenotypeGVCFs` on all of them together to create the raw SNP and indel VCFs that are usually emitted by the callers.

## Variant calling

Run the `HaplotypeCaller` on each sample's BAM files to create single-sample gVCFs using the `.g.vcf` extension for the output file.

```{bash HaplotypeCaller, eval=F, echo=T}
#!/bin/bash
#SBATCH --time=48:00:00
#SBATCH -J HaplotypeCaller
#SBATCH -o HaplotypeCaller_output.out
#SBATCH -e HaplotypeCaller_error.out
#SBATCH --mem=256G
#SBATCH --cpus-per-task=64
#SBATCH --mail-type=BEGIN,END,FAIL

module purge
module load bioinfo/gatk-4.0.0.0
module load bioinfo/picard-2.14.1
module load bioinfo/samtools-1.4

# java -Xmx4g -jar $PICARD CreateSequenceDictionary R=reference/reference.fasta O=reference/reference.dict
# samtools faidx reference/reference.fasta
N=64
(
for file in $(ls mapping/bam2/*.bam)
do 
   ((i=i%N)); ((i++==0)) && wait
   gatk --java-options "-Xmx4G" HaplotypeCaller -R reference/reference.fasta -I $file -O variantCalling/gvcf/$(basename "${file%.*}").g.vcf.gz  -ERC GVCF & 
done
)
```

## Data aggregation

Aggregate the GVCF files and feed in one GVCF with `GenomicsDBImport` to be genotyped. **Beware**, `GATK 4.0.0.0` does not deal with multiple intervals when using `GenomicsDBImport`, so we used `GATK 4.1.2.0`. We may divide the step on several intervals if the operation is too memory-consuming.

```{bash combine, eval=F, echo=T}
#!/bin/bash
#SBATCH --time=36:00:00
#SBATCH -J combine
#SBATCH -o combine_output.out
#SBATCH -e combine_error.out
#SBATCH --mem=256G
#SBATCH --cpus-per-task=64
#SBATCH --mail-type=BEGIN,END,FAIL

module purge
module load bioinfo/gatk-4.1.2.0

cut reference/reference.fasta.fai -f1 > reference/reference.sequences.list # to be changed by several lists if needed
# cut ../reference/reference.fasta.fai -f1 | split -l 2600
touch variantCalling/sample_map.txt
for file in $(ls variantCalling/gvcf/*.g.vcf.gz)
do 
   echo -e $(basename "${file%.*}")"\t"$file >> variantCalling/sample_map.txt
done
mkdir variantCalling/tmp
gatk --java-options "-Xmx256g -Xms256g" GenomicsDBImport \
  --genomicsdb-workspace-path variantCalling/symcaptureDB \
  -L reference/reference.sequences.list \
  --sample-name-map variantCalling/sample_map.txt \
  --batch-size 64 \
  --reader-threads 64
  --tmp-dir=variantCalling/tmp
rm -r variantCalling/tmp
```



## Joint genotyping

Run `GenotypeGVCFs` on all of them together to create the raw SNP and indel VCFs that are usually emitted by the callers. If joint genotyping is too slow, we may divide the step on several intervals list and join vcf after.

```{bash genotype, eval=F, echo=T}
#!/bin/bash
#SBATCH --time=36:00:00
#SBATCH -J genotype
#SBATCH -o genotype_output.out
#SBATCH -e genotype_error.out
#SBATCH --mem=256G
#SBATCH --cpus-per-task=64
#SBATCH --mail-type=BEGIN,END,FAIL

module purge
module load bioinfo/gatk-4.1.2.0

mkdir variantCalling/tmp
gatk --java-options "-Xmx256g" GenotypeGVCFs \
 -R reference/reference.fasta \
 -V gendb://variantCalling/symcaptureDB \
 -O variantCalling/symcapture.vcf.gz \
 --tmp-dir=variantCalling/tmp
rm -r variantCalling/tmp

# java -jar picard.jar MergeVcfs \
#          I=input_variant_files.list \
#          O=output_variants.vcf.gz
```
