```{r setup_variantFiltering, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/"
```

# Variant filtering

**!! Work in progress, the current missing data filtering is an example !!** 

We filtered the previously produced raw vcf with several steps:

* __Gather__ raw vcf files gathering in **26 813 513** variants over **432** individuals
* __Biallelic__ raw vcf filtering in **19 242 294** variants over **432** individuals
* __SNP__ biallelic vcf filtering in **17 521 879** variants over **432** individuals
* __Filters__ biallelic snp vcf filtering in **15 531 866** variants over **432** individuals
* __Missing__ filtered biallelic snp vcf filtering in **454 262** variants over **406** individuals
* __Paracou__ filtered & non missing biallelic snp vcf filtering in **454 262** variants over **385** individuals

## Gather

We first gathered all raw vcf. 42 sequences subset lost individuals in genotype call (over 878, 4%), blocking the functionning of `gatk CombineVariants`, so we first removed the variants associated to those subest of reference sequences. This resulted in 26 813 513 variants.

```{bash missingidv, eval=F, echo=T}
mkdir out
mkdir missing_ind
for file in $(ls symcapture.filtered.vcf/*.vcf.gz) ; do file=$(basename $file) ; file=${file%.*} ; echo "module load bioinfo/tabix-0.2.5 ; module load bioinfo/vcftools-0.1.15 ; vcftools --gzvcf symcapture.filtered.vcf/$file.gz --missing-indv -c > missing_ind/$file.missing.txt"; done > missingInd.sh
sarray -J missingInd -o out/%j.missingInd.out -e out/%j.missingInd.err -t 1:00:00 --mail-type=BEGIN,END,FAIL missingInd.sh
for file in $(ls missing_ind/*.missing.txt) ; do awk '{{if (NR!=1) print FILENAME"\t"$0}}' $file ; done > missingInd.txt
rm - r out
rm -r missing_ind
```

```{r missingindR, eval=F}
missing <- read_tsv(file.path(path, "Sequences", "variantCalling", "missingInd.txt"), 
         col_names = c("ref", "sample", "ndata", "ngenotypes", "nmiss", "freq")) %>% 
  mutate(ref = gsub("missing_ind/reference.sequences_", "", ref)) %>% 
  mutate(ref = gsub(".vcf.missing.txt", "", ref)) %>% 
  mutate(sample = gsub(".g.vcf", "", sample)) %>% 
  reshape2::dcast(ref ~ sample, value.var = "freq") %>% 
  reshape2::melt(id.vars = "ref", variable.name = "sample", value.name = "missing") %>% 
  mutate(missing = ifelse(is.na(missing) | is.nan(missing), 1, missing))
read_tsv(file.path(path, "Sequences", "variantCalling", "missingInd.txt"), 
         col_names = c("ref", "sample", "ndata", "ngenotypes", "nmiss", "freq")) %>% 
  mutate(ref = gsub("missing_ind/reference.sequences_", "", ref)) %>% 
  mutate(ref = gsub(".vcf.missing.txt", "", ref)) %>% 
  mutate(sample = gsub(".g.vcf", "", sample)) %>% 
  group_by(ref) %>% 
  summarise(N = n()) %>% 
  filter(N == 432) %>% 
  select(ref) %>% 
  mutate(ref = paste0("symcapture.filtered.vcf/reference.sequences_", ref, ".vcf.gz")) %>% 
  write_tsv(file.path(path, "Sequences", "variantCalling", "nonmissing.list"), col_names = F)
googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop") %>% 
  mutate(Ind = paste0(Ind, ".g.vcf", "")) %>% 
  filter(Pop2 != "O") %>% 
  select(Ind) %>% 
  write_tsv(file.path(path, "Sequences", "variantCalling", "Paracou.ind"), col_names = F)
```

```{bash gather, eval=F, echo=T}
echo -e '#!/bin/bash\n#SBATCH --time=48:00:00\n#SBATCH -J gather\n#SBATCH -o gather.out\n#SBATCH -e gather.err\n#SBATCH --mem=20G\n#SBATCH --cpus-per-task=1\n#SBATCH --mail-type=BEGIN,END,FAIL\nmodule load bioinfo/picard-2.14.1\njava -Xmx20g -jar $PICARD GatherVcfs \' > gather.sh
for file in $(cat nonmissing.list)
do
	echo -e '\tI=symcapture.raw.vcf/'$(basename $file)' \' >> gather.sh
done
echo -e '\tO=symcapture.all.raw.vcf.gz\n' >> gather.sh
zcat symcapture.all.raw.vcf.gz | grep "#contig" | wc -l
```

## Biallelic

We then used `bcftools` to limit data to biallelic variants (`--max-alleles 2`), resulting in 19 242 294 biallelic variants.

```{bash biallelic, eval=F, echo=T}
srun --mem=80G --cpus-per-task=8 --pty bash
module load bioinfo/bcftools-1.8
bcftools view --max-alleles 2 --threads 8 symcapture.all.raw.vcf.gz | bgzip -c --threads 8 > symcapture.all.biallelic.vcf.gz
bcftools stats  --threads 8 symcapture.all.biallelic.vcf.gz
```

## SNP

We then used `gatk` to limit data to biallelic snps, resulting in 17 521 879 biallelic snps.

```{bash snp, eval=F, echo=T}
module load bioinfo/gatk-4.1.2.0
gatk IndexFeatureFile \
  -F symcapture.all.biallelic.vcf.gz
gatk SelectVariants \
  -V symcapture.all.biallelic.vcf.gz \
  -select-type SNP \
  -O symcapture.all.biallelic.snp.vcf.gz
gatk IndexFeatureFile \
  -F symcapture.all.biallelic.snp.vcf.gz
module load bioinfo/bcftools-1.8
bcftools stats  --threads 8 symcapture.all.biallelic.snp.vcf.gz
```

## Filters

We filtered the biallelic snp vcf with following filters (name, filter, description), resulting in 15 531 866 filtered biallelic snps,  using next histograms to set and test parameters values :

* __Quality ([QUAL](https://software.broadinstitute.org/gatk/documentation/article?id=7258))__ `QUAL < 30`: represents the likelihood of the site to be homozygote across all samples, we filter out variants having a low quality score (\@ref(fig:filtersR))
* __Quality depth ([QD](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.php))__ `QD < 2`: filter out variants with low variant confidence (\@ref(fig:filtersR))
* __Fisher strand bias ([FS](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.php))__ `FS > 60`: filter out variants based on Phred-scaled p-value using Fisher's exact test to detect strand bias (\@ref(fig:filtersR))
* __Strand odd ratio ([SOR](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.php))__ `SOR < 3`: filter out variants based on Phred-scaled p-value used to detect strand bias (\@ref(fig:filtersR))

```{bash filters, eval=F, echo=T}
vcftools --gzvcf reference.sequences_00.vcf.gz --missing-indv -c
vcftools --gzvcf reference.sequences_00.vcf.gz --missing-site -c > missing.txt
vcftools --gzvcf reference.sequences_00.vcf.gz --site-quality -c > QUAL.txt
vcftools --gzvcf reference.sequences_00.vcf.gz \
  --get-INFO AC \
  --get-INFO AF \
  --get-INFO QD \
  --get-INFO FS \
  --get-INFO SOR \
  -c > INFO.txt
```

```{r filtersR, fig.cap="Quality, quality by depth, fisher strand and strand odd ratios per biallelic SNPs."}
filtering <- left_join(read_tsv(file.path(path, "Sequences", "variantCalling", "symcapture.vcf.stats", "missing.txt")),
          read_tsv(file.path(path, "Sequences", "variantCalling", "symcapture.vcf.stats", "QUAL.txt"))) %>% 
  left_join(read_tsv(file.path(path, "Sequences", "variantCalling", "symcapture.vcf.stats", "INFO.txt"))) %>% 
  mutate(QD = as.numeric(QD)) %>% 
  mutate(AF = as.numeric(AF)) %>% 
  sample_n(1000) # for tests
QUALthresh <- 30
g.qual <- ggplot(filtering, aes(QUAL)) + 
  geom_rect(xmin = 0, xmax = log10(QUALthresh), ymin = 0, ymax = 100, col = "grey", alpha = 0.4) +
  geom_density(fill = "blue", alpha = 0.4, col = NA) +
  geom_vline(xintercept = QUALthresh, col = "red", linetype = "dashed") +
  xlab("Quality") +
  ylab("Biallelic SNPs") +
  scale_x_log10() +
  ggtitle(paste0("QUAL: ", round(nrow(filter(filtering, QUAL > QUALthresh))/nrow(filtering)*100, 2),
                 "% (", nrow(filter(filtering, QUAL > QUALthresh)), 
                 " / ", nrow(filtering), ") variant above ", QUALthresh))
QDthresh <- 2
g.qd <- ggplot(filtering, aes(QD)) + 
  geom_rect(xmin = 0, xmax = QDthresh, ymin = 0, ymax = 100, col = "grey", alpha = 0.4) +
  geom_density(fill = "blue", alpha = 0.4, col = NA) +
  geom_vline(xintercept = QDthresh, col = "red", linetype = "dashed") +
  xlab("Quality depth") +
  ylab("Biallelic SNPs") +
  ggtitle(paste0("QD: ", round(nrow(filter(filtering, QD > QDthresh))/nrow(filtering)*100, 2),
                 "% (", nrow(filter(filtering, QD > QDthresh)), 
                 " / ", nrow(filtering), ") variant above ", QDthresh))
FSthresh <- 60
g.fs <- ggplot(filtering, aes(FS)) + 
  geom_rect(xmin = FSthresh, xmax = 1000, ymin = 0, ymax = 100, col = "grey", alpha = 0.4) +
  geom_density(fill = "blue", alpha = 0.4, col = NA) +
  geom_vline(xintercept = FSthresh, col = "red", linetype = "dashed") +
  xlab("Fisher Strand Ratio") +
  ylab("Biallelic SNPs") +
  ggtitle(paste0("FS: ", round(nrow(filter(filtering, FS < FSthresh))/nrow(filtering)*100, 2),
                 "% (", nrow(filter(filtering, FS < FSthresh)), 
                 " / ", nrow(filtering), ") variant above ", FSthresh))
SORthresh <- 3
g.sor <- ggplot(filtering, aes(SOR)) + 
  geom_rect(xmin = SORthresh, xmax = 100, ymin = 0, ymax = 100, col = "grey", alpha = 0.4) +
  geom_density(fill = "blue", alpha = 0.4, col = NA) +
  geom_vline(xintercept = SORthresh, col = "red", linetype = "dashed") +
  xlab("Strand Odd Ratio") +
  ylab("Biallelic SNPs") +
  ggtitle(paste0("SOR: ", round(nrow(filter(filtering, SOR < SORthresh))/nrow(filtering)*100, 2),
                 "% (", nrow(filter(filtering, SOR < SORthresh)), 
                 " / ", nrow(filtering), ") variant below ", SORthresh))
cowplot::plot_grid(g.qual, g.qd, g.fs, g.sor)
```

```{bash filters2, eval=F, echo=T}
module load bioinfo/gatk-4.1.2.0
gatk VariantFiltration \
  -V symcapture.all.biallelic.snp.vcf.gz \
  --filter-expression "QUAL < 30.0 || QD < 2.0 || FS > 60.0 || SOR > 3.0" \
  --filter-name "FAIL" \
  -O symcapture.all.biallelic.snp.intermediate.vcf.gz
gatk SelectVariants \
  -V symcapture.all.biallelic.snp.intermediate.vcf.gz \
  --exclude-filtered \
  -O symcapture.all.biallelic.snp.filtered.vcf.gz
module load bioinfo/bcftools-1.8
bcftools stats --threads 8 symcapture.all.biallelic.snp.filtered.vcf.gz
gatk IndexFeatureFile \
  -F symcapture.all.biallelic.snp.filtered.vcf.gz
```

## Missing Paracou

Missing data filtering is a bit more tricky because missing data of SNPs and individuals are related, *e.g.* removing individuals with a lot of missing data result in the decrease of SNPs. Ideally, we would want to keep all individuals, but this would result in a lot of SNP loss because of least represented individuals (Fig. \@ref(fig:missingStatsR)). So we need to chose a threshold for missing data for individuals `--mind` and SNPs `--geno`. **Work in progress !**. For instance with a maximum of 95% of missing data per individual `--mind 0.95` and a maximum of 15% of missing data per SNP `-geno 0.15`, we obtained **454 262** biallelic filtered snps for **406** individuals, with a priori good minor allele frequency ditribution and pricnipal component analysis (Fig \@ref(fig:missingR)).

```{bash missingStats, eval=F, echo=T}
# 13M SNPs on unzipped vcf seems the maximum for plink
module load bioinfo/bcftools-1.8
module load bioinfo/plink2-v2.0_alpha2
module load bioinfo/plink-v1.90b5.3
rm symcapture.all.biallelic.snp.intermediate.vcf*
rm symcapture.all.biallelic.snp.vcf*
rm symcapture.all.biallelic.vcf*
mkdir filtered
cd filtered
zcat symcapture.all.biallelic.snp.filtered.vcf.gz > filtered/symcapture.all.biallelic.snp.filtered.vcf
bcftools view -h symcapture.all.biallelic.snp.filtered.vcf > symcapture.all.biallelic.snp.filtered.vcf.header
wc -l symcapture.all.biallelic.snp.filtered.vcf
split -l 7807350 -d 1 symcapture.all.biallelic.snp.filtered.vcf symcapture.all.biallelic.snp.filtered.vcf.part
rm symcapture.all.biallelic.snp.filtered.vcf
mv symcapture.all.biallelic.snp.filtered.vcf.part00 symcapture.all.biallelic.snp.filtered.vcf.part1
mv symcapture.all.biallelic.snp.filtered.vcf.header symcapture.all.biallelic.snp.filtered.vcf.part2
cat symcapture.all.biallelic.snp.filtered.vcf.part01 >> symcapture.all.biallelic.snp.filtered.vcf.part2
rm symcapture.all.biallelic.snp.filtered.vcf.part01
plink2 \
  --threads 8 \
  --memory 80000 \
  --vcf symcapture.all.biallelic.snp.filtered.vcf.part1 \
  --allow-extra-chr \
  --make-bed --out  symcapture.all.biallelic.snp.filtered.part1
rm symcapture.all.biallelic.snp.filtered.vcf.part1
plink2 \
  --threads 8 \
  --memory 80000 \
  --vcf symcapture.all.biallelic.snp.filtered.vcf.part2 \
  --allow-extra-chr \
  --make-bed --out  symcapture.all.biallelic.snp.filtered.part2
rm symcapture.all.biallelic.snp.filtered.vcf.part2
plink --threads 8 --memory 80000 \
  --bfile symcapture.all.biallelic.snp.filtered.part1 \
  --allow-extra-chr \
  --missing --het --freq  --pca \
  --out  symcapture.all.biallelic.snp.filtered.part1
plink --threads 8 --memory 80000 \
  --bfile symcapture.all.biallelic.snp.filtered.part2 \
  --allow-extra-chr \
  --missing --het --freq  --pca \
  --out  symcapture.all.biallelic.snp.filtered.part2
rm symcapture.all.biallelic.snp.filtered.vcf
```

```{r missingStatsR, fig.cap="Missing data statistis for filtered biallelic SNP before missing data filtering."}
pathCluster <- "/home/sylvain/Remotes/genotoul/work/PhD/data/Symphonia_Paracou/"
  
g.het_imiss <- read_delim(file.path(pathCluster, "Sequences", "variantCalling", "filtered", 
                                    "symcapture.all.biallelic.snp.filtered.part1.imiss"), delim = " ") %>%
  bind_rows(read_delim(file.path(pathCluster, "Sequences", "variantCalling", "filtered", 
                                    "symcapture.all.biallelic.snp.filtered.part2.imiss"), delim = " ")) %>% 
  left_join(read_delim(file.path(pathCluster, "Sequences", "variantCalling", "filtered", 
                                 "symcapture.all.biallelic.snp.filtered.part1.het"), delim = " ") %>% 
              bind_rows(read_delim(file.path(pathCluster, "Sequences", "variantCalling", "filtered", 
                                 "symcapture.all.biallelic.snp.filtered.part2.het"), delim = " "))) %>% 
  dplyr::rename_all(funs(gsub(" ", "", .))) %>% 
  mutate_at(c("N_MISS", "N_GENO", "F_MISS", "O(HOM)", "E(HOM)", "N(NM)", "F"), as.numeric) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(Ind = gsub(" ", "", Ind)) %>% 
  left_join( googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop")) %>% 
  ggplot(aes(F_MISS, `F`, col = Pop2)) +
  geom_point() +
  scale_x_log10() +
  xlab("Proportion of missing SNPs") +
  ylab("Heterozygosity rate")

g.pca <- read_delim(file.path(pathCluster, "Sequences", "variantCalling", "filtered", 
                              "symcapture.all.biallelic.snp.filtered.part1.eigenvec"), 
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>% 
  bind_rows(read_delim(file.path(pathCluster, "Sequences", "variantCalling", "filtered", 
                              "symcapture.all.biallelic.snp.filtered.part2.eigenvec"), 
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20)))) %>% 
  mutate(Ind = gsub(".g.vcf", "", Sample)) %>% 
  left_join( googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop")) %>% 
  filter(Pop2 != "O") %>%
  ggplot(aes(x = PCA1, y = PCA2, col = Pop)) + 
  geom_point(size=2) +
  stat_ellipse(level = 0.95, size = 1) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)

g.lmiss <- read_delim(file.path(pathCluster, "Sequences", "variantCalling", "filtered", 
                                "symcapture.all.biallelic.snp.filtered.part1.lmiss"), delim = " ") %>% 
  bind_rows(read_delim(file.path(pathCluster, "Sequences", "variantCalling", "filtered", 
                                "symcapture.all.biallelic.snp.filtered.part2.lmiss"), delim = " ")) %>% 
  dplyr::rename_all(funs(gsub(" ", "", .))) %>% 
  mutate(F_MISS = as.numeric(F_MISS)) %>% 
  ggplot(aes(F_MISS)) +
  geom_histogram() +
  xlab("Proportion of missing data") +
  ylab("SNP count")

g.maf <- read_delim(file.path(pathCluster, "Sequences", "variantCalling", "filtered", 
                              "symcapture.all.biallelic.snp.filtered.part1.frq"), delim = " ") %>% 
  bind_rows(read_delim(file.path(pathCluster, "Sequences", "variantCalling", "filtered", 
                              "symcapture.all.biallelic.snp.filtered.part2.frq"), delim = " ")) %>% 
  dplyr::rename_all(funs(gsub(" ", "", .))) %>% 
  mutate(MAF = as.numeric(MAF)) %>% 
  ggplot(aes(MAF)) +
  geom_histogram() +
  xlab("Minor allele frequency") +
  ylab("SNP count")

cowplot::plot_grid(g.het_imiss, g.pca, g.lmiss, g.maf)
```

```{bash missing, eval=F, echo=T}
module load bioinfo/bcftools-1.8
module load bioinfo/plink-v1.90b5.3
module load bioinfo/plink2-v2.0_alpha2
mkdir nonmissing
plink2 --threads 8 --memory 80000 \
  --bfile filtered/symcapture.all.biallelic.snp.filtered.part1 \
  --allow-extra-chr \
  --mind 0.95 --geno 0.15 \
  --make-bed --out nonmissing/symcapture.all.biallelic.snp.filtered.nonmissing.part1
plink2 --threads 8 --memory 80000 \
  --bfile filtered/symcapture.all.biallelic.snp.filtered.part2 \
  --allow-extra-chr \
  --mind 0.95 --geno 0.15 \
  --make-bed --out nonmissing/symcapture.all.biallelic.snp.filtered.nonmissing.part2
cd nonmissing
plink --threads 8 --memory 80000 \
  --bfile symcapture.all.biallelic.snp.filtered.nonmissing.part1 \
  --allow-extra-chr --recode-vcf \
  --out symcapture.all.biallelic.snp.filtered.nonmissing.part1
plink --threads 8 --memory 80000 \
  --bfile symcapture.all.biallelic.snp.filtered.nonmissing.part2 \
  --allow-extra-chr --recode-vcf \
  --out symcapture.all.biallelic.snp.filtered.nonmissing.part2
mkdir tmp
mv *.vcf tmp/
rm *
mv tmp/* ./
rm -r tmp
bcftools view -h symcapture.all.biallelic.snp.filtered.nonmissing.part1.vcf | head -n 4 \
  > symcapture.all.biallelic.snp.filtered.nonmissing.vcf
grep "#contig" symcapture.all.biallelic.snp.filtered.nonmissing.part1.vcf \
  >> symcapture.all.biallelic.snp.filtered.nonmissing.vcf
grep "#contig" symcapture.all.biallelic.snp.filtered.nonmissing.part2.vcf \
  >> symcapture.all.biallelic.snp.filtered.nonmissing.vcf
bcftools view -h symcapture.all.biallelic.snp.filtered.nonmissing.part2.vcf | tail -n 5 \
  >> symcapture.all.biallelic.snp.filtered.nonmissing.vcf
sed '/^#/d' symcapture.all.biallelic.snp.filtered.nonmissing.part1.vcf \
  >> symcapture.all.biallelic.snp.filtered.nonmissing.vcf
sed '/^#/d' symcapture.all.biallelic.snp.filtered.nonmissing.part2.vcf \
  >> symcapture.all.biallelic.snp.filtered.nonmissing.vcf
plink --threads 8 --memory 80000 \
  --vcf symcapture.all.biallelic.snp.filtered.nonmissing.vcf \
  --allow-extra-chr --missing --het --freq --pca \
  --make-bed --out symcapture.all.biallelic.snp.filtered.nonmissing
rm *.vcf
```

```{r missingR, fig.cap="Missing data statistis for filtered biallelic SNP after missing data filtering (90% for individuals and 50% for SNPs)."}
pathCluster <- "/home/sylvain/Remotes/genotoul/work/PhD/data/Symphonia_Paracou/"

g.het_imiss <- read_delim(file.path(pathCluster, "Sequences", "variantCalling", "nonmissing", 
                                    "symcapture.all.biallelic.snp.filtered.nonmissing.imiss"), delim = " ") %>% 
  left_join(read_delim(file.path(pathCluster, "Sequences", "variantCalling", "nonmissing", 
                                 "symcapture.all.biallelic.snp.filtered.nonmissing.het"), delim = " ")) %>% 
  dplyr::rename_all(funs(gsub(" ", "", .))) %>% 
  mutate_at(c("N_MISS", "N_GENO", "F_MISS", "O(HOM)", "E(HOM)", "N(NM)", "F"), as.numeric) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(Ind = gsub(" ", "", Ind)) %>% 
  left_join( googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop")) %>% 
  ggplot(aes(F_MISS, `F`, col = Pop2)) +
  geom_point() +
  scale_x_log10() +
  xlab("Proportion of missing SNPs") +
  ylab("Heterozygosity rate")

g.pca <- read_delim(file.path(pathCluster, "Sequences", "variantCalling", "nonmissing", 
                              "symcapture.all.biallelic.snp.filtered.nonmissing.eigenvec"), 
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>% 
  mutate(Ind = gsub(".g.vcf", "", Sample)) %>% 
  left_join( googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop")) %>% 
  filter(Pop2 != "O") %>%
  ggplot(aes(x = PCA1, y = PCA2, col = Pop)) + 
  geom_point(size=2) +
  stat_ellipse(level = 0.95, size = 1) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)

g.lmiss <- read_delim(file.path(pathCluster, "Sequences", "variantCalling", "nonmissing", 
                                "symcapture.all.biallelic.snp.filtered.nonmissing.lmiss"), delim = " ") %>% 
  dplyr::rename_all(funs(gsub(" ", "", .))) %>% 
  mutate(F_MISS = as.numeric(F_MISS)) %>% 
  ggplot(aes(F_MISS)) +
  geom_histogram() +
  xlab("Proportion of missing data") +
  ylab("SNP count")

g.maf <- read_delim(file.path(pathCluster, "Sequences", "variantCalling", "nonmissing", 
                              "symcapture.all.biallelic.snp.filtered.nonmissing.frq"), delim = " ") %>% 
  dplyr::rename_all(funs(gsub(" ", "", .))) %>% 
  mutate(MAF = as.numeric(MAF)) %>% 
  ggplot(aes(MAF)) +
  geom_histogram() +
  xlab("Minor allele frequency") +
  ylab("SNP count")

cowplot::plot_grid(g.het_imiss, g.pca, g.lmiss, g.maf)
```

## Paracou

Finally, we subseted the filtered and biallelic snp from Paracou individuals, resulting in **385** remaining individuals (**17 lost !**).

```{bash paracou, eval=F, echo=T}
module load bioinfo/plink2-v2.0_alpha2
mkdir paracou
plink2 \
  --threads 8 \
  --memory 80000 \
  --bfile nonmissing/symcapture.all.biallelic.snp.filtered.nonmissing \
  --allow-extra-chr \
  --keep Paracou.fam \
  --make-bed --out paracou/symcapture.all.biallelic.snp.filtered.nonmissing.paracou
```
