```{r setup_popgen, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(pophelper)
library(dendextend)
library(ggtree)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
```

```{r popgenR}
googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop") %>% 
  mutate(Ind = paste0(Ind, ".g.vcf", "")) %>% 
  mutate(FID = 0, IID = Ind, CLUSTER = Pop2) %>% 
  select(FID, IID, CLUSTER) %>% 
  filter(CLUSTER %in% c("S", "G")) %>% 
  write_tsv(file.path(path, "paracou.2pop"), col_names = F)
pop <- googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop")
pop <- bind_rows(pop, c(Ind = "P10-3-925", Pop = "SG", Pop2 = "SG")) # to solve
```

# Population genomics

**!! Work in progress, the current analysis are done with a test of missing data filtering !!** 

We will ... :

* __Population structure__ ...

## Metrics

```{bash metrics, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
plink --bfile symcapture.all.biallelic.snp.filtered.nonmissing.paracou \
  --allow-extra-chr \
  --within paracou.2pop \
  --het \
  --hardy \
  --fst \
  --out  symcapture.all.biallelic.snp.filtered.nonmissing.paracou

# 257695 MB RAM detected; reserving 128847 MB for main workspace.
# 454262 variants loaded from .bim file.
# 385 people (0 males, 0 females, 385 ambiguous) loaded from .fam.
# Ambiguous sex IDs written to
# symcapture.all.biallelic.snp.filtered.nonmissing.paracou.nosex .
# --within: 2 clusters loaded, covering a total of 357 people.
# Using 1 thread (no multithreaded calculations invoked).
# Before main variant filters, 385 founders and 0 nonfounders present.
# Calculating allele frequencies... done.
# Total genotyping rate is 0.934092.
# --hardy: Writing Hardy-Weinberg report (founders only) to
# symcapture.all.biallelic.snp.filtered.nonmissing.paracou.hwe ... done.
# 454262 variants and 385 people pass filters and QC.
# Note: No phenotypes present.
# --het: 377743 variants scanned, report written to
# symcapture.all.biallelic.snp.filtered.nonmissing.paracou.het .
# Writing --fst report (2 populations) to
# symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fst ... done.
# 370611 markers with valid Fst estimates (83651 excluded).
# Mean Fst estimate: 0.0122558
# Weighted Fst estimate: 0.0477019
```

```{r metricsR, fig.cap="Heterozygosity per individual and Fst per SNP (density and manhatan plots)."}
pathCluster <- "/home/sylvain/Remotes/genotoul/work/PhD/data/Symphonia_Paracou/Sequences"

g.het <- read_delim(file.path(pathCluster, "variantCalling", "paracou",
                     "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.het"), delim = " ") %>% 
  dplyr::rename_all(funs(gsub(" ", "", .))) %>% 
  mutate(het = as.numeric(`F`)) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(Ind = gsub(" ", "", Ind)) %>% 
  left_join(pop) %>% 
  ggplot(aes(abs(het), fill = Pop2)) +
  geom_histogram(position = "dodge") +
  xlab("Heterozygosity")

g.fst <- read_tsv(file.path(pathCluster, "variantCalling", "paracou",
                   "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fst")) %>% 
  dplyr::rename_all(funs(gsub(" ", "", .))) %>% 
  mutate(FST = as.numeric(FST)) %>% 
  ggplot(aes(FST)) +
  geom_density() +
  scale_x_sqrt() +
  geom_vline(xintercept = 0.015, col = "red", linetype = "dashed") +
  xlab("Fst")

cowplot::plot_grid(g.het, g.fst, nrow = 2)

g.fst$data %>% 
  filter(CHR %in% unique(g.fst$data$CHR)[1:1000]) %>%
  mutate(contig = as.numeric(as.factor(CHR))) %>% 
  filter(!is.nan(FST)) %>% 
  mutate(FST = abs(FST)) %>% 
  qqman::manhattan(chr = "contig", bp = "POS", p = "FST", suggestiveline = 0.5, logp = F,
                   xlab = "First 1k contigs", ylab = "Fst between morphotype")
```

## Structure

```{bash str, eval=F, echo=T}
mkdir structure
mkdir out
for k in $(seq 10) ; do for r in $(seq 10) ; do echo "module load bioinfo/fastStructure-1.0 ; python \$HOME_FAST/structure.py -K $k --input=../variantCalling/paracou/symcapture.all.biallelic.snp.filtered.nonmissing.paracou --format=bed --output=structure/symcapture.paracou.K$k.R$r --full --seed=12345";  done done > structure.sh
sarray -J structure -o out/%j.structure.out -e out/%j.structure.err -t 48:00:00 --mem=8G --mail-type=BEGIN,END,FAIL structure.sh
watch 'echo "$(expr 100 - $(squeue -u sschmitt | wc -l))%"'
rm -r out
scp sschmitt@genologin.toulouse.inra.fr:~/Symcapture/populationGenomics/structure/*.meanQ structure/
scp sschmitt@genologin.toulouse.inra.fr:~/Symcapture/populationGenomics/structure/*.log structure/
```

```{r structureLogs, fig.cap="Structure clustering likelihood"}
lapply(list.files(file.path(path, "structure"), full.names = T, pattern = "log"), 
       function(x)
  read_delim(x, delim = "=", col_names = F) %>% 
    mutate(file = x) %>% 
    filter(X1 == "Marginal Likelihood ")) %>% 
  bind_rows() %>% 
  mutate(file = gsub("/home/sylvain/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics//structure/symcapture.paracou.", "", file)) %>%
  mutate(file = gsub(".10.log", "", file)) %>%
  separate(file, c("K", "R")) %>% 
  mutate(K = as.numeric(gsub("K", "", K))) %>%  
  mutate(R = as.numeric(gsub("R", "", R))) %>% 
  select(-X1) %>% 
  dplyr::rename(MarginalLikelihood = X2) %>% 
  mutate(MarginalLikelihood = as.numeric(MarginalLikelihood)) %>% 
  ggplot(aes(K, MarginalLikelihood)) +
  geom_boxplot(aes(fill = as.factor(K))) +
  geom_line() +
  scale_fill_discrete(guide = "none")
```


```{r structureQ, fig.height=8, fig.width=15, fig.cap="Population structure."}
fam <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID))
symcapture.str <- readQ(list.files(file.path(path, "structure"), 
                                   full.names = T, pattern = "meanQ"), indlabfromfile=F)
symcapture.str <- lapply(symcapture.str, "rownames<-", fam$Ind)
symcapture.str <- sortQ(symcapture.str)
symcapture.str <- alignK(symcapture.str)
symcapture.str <- mergeQ(symcapture.str)
p <- plotQ(symcapture.str[2:10], exportplot = F, returnplot = T, imgoutput = "join", basesize = 11, splab = paste0("K=",2:10),
      showindlab = F, useindlab = F, grplab = data.frame(morphotype = left_join(fam, pop)$Pop2, stringsAsFactors = F),
      grplabsize = 4,linesize = 0.8, pointsize = 4, sortind = "all", sharedindlab = F, ordergrp=T, subsetgrp=c("G", "SG", "S"))
gridExtra::grid.arrange(p$plot[[1]])
p <- plotQMultiline(qlist = symcapture.str[2], exportplot = F, returnplot = T, useindlab = T, ordergrp=T, sortind = "all",
                    subsetgrp=c("G", "SG", "S"), grplab = data.frame(morphotype = left_join(fam, pop)$Pop2, stringsAsFactors = F))
gridExtra::grid.arrange(p$plot[[1]][[1]])
```

## Phylogeny

```{r phylo, fig.height=12, fig.width=12, fig.cap="Individuals phylogeny."}
# srun --mem=80G --cpus-per-task=1 --pty bash
# module load system/R-3.5.2
# module load libraries/gdal-2.3.0
# module load libraries/proj-4.9.3
# R
# library(gdsfmt)
# library(SNPRelate)
# snpgdsBED2GDS("../variantCalling/nonmissing/symcapture.all.biallelic.snp.filtered.nonmissing.bed",
#               "../variantCalling/nonmissing/symcapture.all.biallelic.snp.filtered.nonmissing.fam",
#               "../variantCalling/nonmissing/symcapture.all.biallelic.snp.filtered.nonmissing.bim",
#               "phylogeny/symcapture.all.biallelic.snp.filtered.nonmissing.gds")
# symcapture.geno <- openfn.gds("phylogeny/symcapture.all.biallelic.snp.filtered.nonmissing.gds")
# dissMatrix  <-  snpgdsDiss(symcapture.geno, autosome.only = F, num.thread = 3)
# snpHCluster <-  snpgdsHCluster(dissMatrix, hang = 0.25)
# cutTree <- snpgdsCutTree(snpHCluster, z.threshold = 15, outlier.n = 5, n.perm = 5000,
#                         col.outlier = "red", pch.outlier=4)
# save(cutTree, file = "phylogeny/tree.Rdata")
load(file.path(path, "phylogeny", "tree.Rdata"))
symcapture.phylo <- ape::as.phylo(cutTree$dendrogram)
g <- ggtree(symcapture.phylo, layout='circular')
g + geom_tiplab(data = dplyr::inner_join(mutate(g$data, Ind = gsub(".g.vcf", "", label)), pop), size =1,
               size=2, aes(colour = Pop2, angle = angle, label = Ind)) +
    theme(legend.position = "right", 
        legend.title = element_blank(),
        legend.key = element_blank())
  # ggsave(file.path(path, "phylogeny", "treeAll.png"))
```
