```{r setup_annot, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(Biostrings)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/annotation/"
```

# Annotation

We developed and manually tested with few SNPs a method to match transcriptomic data from Tysklind et al., in prep with our genomic data. We found expected proportion of SNPs type, with a lot of putatively hitchhiker SNPs in functional SNPs. Finally, ca 2000 SNPs were matching between the two datasets.

population structure of individuals. Then, we quickly looked at th spatial and environmental distribution of the different gene pools and individual mismatch (before association genomic analyses).

* __method__ data preparation and methodology to match transcriptomic data from Tysklind et al., in prep with our genomic data
* __results__ SNP classification, corresponding SNPs and genes, and common SNPs between datasets.

## Method

We focused on scaffolds from the hybrid genomic reference which contained a filtered SNP from the final dataset (7551 over 82792). We focused on transcripts longer than 500 bp and with at least one ORF (137 656 over 257 140). We then, used transcripts alignment on scaffolds to align back SNPs on transcripts, (**beware the indexing position depend on the format vcf start with 1 while psl with 0**). We proceeded as follow (not a common procedure but a home-made code to be double-checked):

1. SNP on scaffolds with no transcript match were classified as "purely neutral" 
1. SNP on scaffolds matching a transcript were positioned on the global alignment
1. SNP out the alignment area were classified as "putatively hitchhiking"
1. SNP within the alignment area were further positionned among the different alignment blocks (cf `blat` functionning)
1. SNP out of alignment blocks (intron) were classified as "putatively hitchhiking"
1. SNP in one alignment blocks were classified as "functional"

```{r trscAlignment, eval=F, echo=T}
scf <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                   "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.bim"),
         col_names = c("scf", "snp", "posM", "pos", "A1", "A2")) %>% 
  dplyr::select(scf) %>% 
  unique() %>% 
  unlist()
reference <- readDNAStringSet(file.path(path, "..", "reference",  "reference.fasta"))
writeXStringSet(reference[scf],
                file.path(path, "scaffolds.fasta"))
orf <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("ORF") %>% 
  dplyr::select(transcript_id) %>% 
  dplyr::rename(trsc = transcript_id) %>% 
  collect() %>% 
  unique()
trsc.tbl <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("Transcript") %>% 
  dplyr::select(transcript_id, sequence) %>% 
  dplyr::rename(trsc = transcript_id) %>% 
  collect() %>% 
  filter(trsc %in% orf$trsc)
rm(orf)
trsc <- DNAStringSet(trsc.tbl$sequence)
names(trsc) <- trsc.tbl$trsc  
writeXStringSet(trsc, file.path(path, "transcripts.fasta"))
# blat scaffolds.fasta transcipts.fasta alignment.psl
```

```{r snpAlignment}
snps <- read_tsv(file.path(path, "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.bim"),
         col_names = c("scf", "snp", "posM", "pos", "A1", "A2")) %>% 
  mutate(snp = paste0(scf, "_snp", pos))
alignment <- read_tsv(file.path(path, "alignment.psl"), skip = 5, 
                      col_names = c("matches", "misMatches", "repMatches", "nCount", 
                                    "qNumInsert", "qBaseInsert",
                                    "tNumInsert", "tBaseInsert", "strand", 
                                    "qName", "qSize", "qStart", "qEnd", 
                                    "tName", "tSize", "tStart", "tEnd", 
                                    "blockCount", "blockSizes", "qStarts", "tStarts"), 
                      col_types = cols(blockSizes = col_character())) # psl3 generics
snps <- snps %>% 
  left_join(data.frame(scf = unique(alignment$tName), aligned = 1)) %>% 
  mutate(aligned = ifelse(is.na(aligned), 0, aligned))
snpsFunc <- snps %>% 
  left_join(alignment, by = c("scf" = "tName")) %>% 
  mutate(posScf = pos-1) %>% # PSL 0-indexed !!
  filter(tStart < posScf, tEnd > posScf) %>% 
  separate_rows(blockSizes, qStarts, tStarts, sep = ",", convert = T) %>% 
  filter(tStarts < posScf) %>% 
  mutate(posTrsc = qStarts + posScf - tStarts) %>% 
  mutate(qEnd = qStarts + blockSizes) %>% 
  filter(posTrsc < qEnd) %>% 
  dplyr::rename(trsc = qName) %>% 
  dplyr::select(scf, posScf, snp, trsc, posTrsc)
snpsHitch <- snps %>% 
  left_join(alignment, by = c("scf" = "tName")) %>% 
  mutate(posScf = pos-1) %>% # PSL 0-indexed !!
  filter(!(snp %in% snpsFunc)) %>% 
  dplyr::rename(trsc = qName) %>% 
  dplyr::select(scf, snp, trsc) %>% 
  unique()
snps <- snps %>% 
  left_join(data.frame(snp = unique(snpsFunc$snp), functional = 1)) %>% 
  mutate(functional = ifelse(is.na(functional), 0, functional)) %>% 
  mutate(type = ifelse(aligned == 1 & functional == 0, "hitchhiker", NA)) %>% 
  mutate(type = ifelse(functional == 1, "functional", type)) %>% 
  mutate(type = ifelse(aligned == 0, "neutral", type))
# write_tsv(snps, file.path(path, "snps.annotation"))
# write_tsv(snpsHitch, file.path(path, "snpsHitch.annotation"))
# write_tsv(snpsFunc, file.path(path, "snpsFunc.annotation"))
```

## Results

We obtained as planified a third of neutral snps (33%) against two thirds of non-neutral snps for which the majority was out of transcripts and may behave as hitchhikers (Fig. \@ref(fig:snpType)). Nevertheless, few functional snps matched several transcripts and/or genes due to transcripts isoforms, genes superposition and possibl multimatching errors (Fig. \@ref(fig:snpaAlignedStats)). For the moment we wont further annotate functional SNPs with ORFs and SNPs on transcript alignment, or gene ontology and synonymy. We will only finish annotation for selected SNPs in further analyses. Finally, we compared obtained SNP in genomics in this analysis with transcriptomic SNPs identified by Tysklind et al., in prep (Fig. \@ref(fig:snpValidation))) and validated 2 098 common SNPs.

```{r snpType, fig.cap="Filtered snps by type. SNP have been classified between (i) \"purely neutral\", on scaffolds matching no transcript, (ii) \"hitchhiker\" on a scaffolds matching a trasncript but no within the transcript itself, and (iii) functional, when positioned within the matching transcript."}
snps %>% 
  group_by(type) %>% 
  summarise(N = n()) %>% 
  mutate(Pct = N/sum(N)*100) %>% 
  ggplot(aes(x = type, y = N, fill = type, label = paste0(round(Pct),"%"))) +
  geom_bar(stat = "identity") +
  geom_text() +
  ylab("count") +
  theme(axis.text.x = element_blank(),
        axis.line.x = element_blank(),
        axis.title.x = element_blank(),
        axis.ticks.x = element_blank()) +
  scale_fill_discrete("SNP type")
```


```{r snpaAlignedStats, fig.cap="Transcripts and genes statistics per snp. Few snps have multiple transcripts, corresponding to different isoforms and/or multiple genes corresponding to orf superposition or multimatch error."}
bind_rows(snpsHitch, dplyr::select(snpsFunc, scf, snp, trsc)) %>% 
  dplyr::select(snp, trsc) %>% 
  unique() %>% 
  group_by(snp) %>% 
  summarise(`Number of transcripts` = n()) %>% 
  left_join(bind_rows(snpsHitch, dplyr::select(snpsFunc, scf, snp, trsc)) %>% 
              mutate(gene = gsub("_i[0-9]", "", trsc)) %>%
              dplyr::select(snp, gene) %>% 
              unique() %>% 
              group_by(snp) %>% 
              summarise(`Number of genes` = n())) %>% 
  reshape2::melt(id.vars = "snp") %>% 
  ggplot(aes(value, fill = variable)) +
  geom_histogram(position = "dodge") +
  xlim(0,10) +
  theme(legend.position = c(0.7,0.8))
```

```{r snpValidation, fig.cap="transcriptomic SNPs from Tysklind et al., in prep matching validated by genomic SNPs from the current gene capture."}
snpsNiklas <- read_delim(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts", "k2rt",
                                   "symphonia_juv_fullsample_trinity500_k2rt_type_0a_mainOutput.tsv"),
                         delim = "\t")
snpsFunc %>% 
  inner_join(snpsNiklas %>% 
  dplyr::rename(snpNiklas = SNP_ID, trsc = `#Component_ID`, posTrsc = SNP_position) %>% 
  dplyr::select(trsc, snpNiklas, posTrsc) %>% 
    mutate(posTrsc = as.numeric(posTrsc))) %>% 
  dplyr::select(scf, snp, snpNiklas) %>% 
  unique() %>% 
  group_by(scf) %>% 
  summarise(N = n()) %>% 
  ggplot(aes(N)) +
  geom_histogram() +
  ggtitle("2 098 transcriptomic SNPs from Tysklind et al., in prep",
          "validated by genomic SNPs from the current gene capture") +
  xlab("Number of SNPs per scaffold")
```

```{bash snpLists, eval=F, echo=T}
cut -f2,9 snps.annotation > snps.type
grep neutral snps.type | cut -f1 > snps.neutral
grep hitchhiker snps.type | cut -f1 > snps.hitchhiker
grep functional snps.type | cut -f1 > snps.functional
```

## Synonymy

```{r synonymyPrep, eval=F, echo=T}
snps <- read_tsv(file.path(path, "snpsFunc.annotation")) %>% 
  left_join(read_tsv(file.path(path, "snps.annotation")))
trsc <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                             "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("Transcript") %>% 
  filter(transcript_id %in% local(snps$trsc)) %>% 
  dplyr::rename(trsc = transcript_id) %>% 
  dplyr::select(trsc, sequence) %>% collect()
t <- snps %>% 
  dplyr::select(snp, trsc, posTrsc, A1, A2) %>% 
  reshape2::melt(id.vars = c("snp", "trsc", "posTrsc"), 
                 variable.name = "Allele", value.name = "base") %>% 
  arrange(snp) %>% 
  left_join(trsc) %>% 
  mutate(ref = stringr::str_sub(sequence, posTrsc, posTrsc)) %>%
  group_by(snp) %>% 
  filter(any(base %in% ref))
t %>% 
  filter(base != ref) %>% 
  ungroup() %>% 
  dplyr::select(trsc, posTrsc, base) %>% 
  write_tsv(path = file.path(path, "SNPsOnTrsc.tsv"), col_names = F)
```

```{bash synonymy, eval=F, echo=T}
mkdir TransDecoder
cp transcripts.fasta TransDecoder/
cd TransDecoder/
TransDecoder.LongOrfs -t transcripts.fasta
makeblastdb \
  -in /bank/ebi/uniprot/current/fasta/uniprot_sprot.fasta \
  -dbtype prot -input_type fasta -out uniprot/uniprot_sprot.fa -hash_index
blastp -query transcripts.fasta.transdecoder_dir/longest_orfs.pep \
  -db uniprot/uniprot_sprot.fa \
  -max_target_seqs 1 -outfmt 6 -evalue 1e-5 -num_threads 4 > blastp.outfmt6
hmmscan --cpu 4 --domtblout pfam.domtblout \
	Pfam/Pfam-A.hmm \
	transcripts.fasta.transdecoder_dir/longest_orfs.pep
TransDecoder.Predict -t transcripts.fasta \
	--retain_pfam_hits pfam.domtblout --retain_blastp_hits blastp.outfmt6
~/Tools/gffread/gffread longest_orfs.gff3 -T -o longest_orfs.gtf
perl ~/Tools/SNPdat_package_v1.0.5/SNPdat_v1.0.5.pl \
  -i SNPsOnTrsc.tsv \
  -g longest_orfs.gtf \
  -f transcripts.fasta \
  -s synonymy.summary \
  -o synonymy.output
```

```{r synonymyRes, eval=F, echo=T}
snps <- read_tsv(file.path(path, "snpsFunc.annotation")) %>% 
  left_join(read_tsv(file.path(path, "snps.annotation")))
synonymy <- read_tsv(file.path(path, "synonymy.output"), col_names = T)
ggplot(synonymy, aes(synonymous)) + 
  geom_bar() +
  ggtitle(paste0(round(178908/475272*100), "% of non synonymous SNPs in genic SNPs"))
snps %>% 
  mutate(TRSC = toupper(trsc)) %>% 
  left_join(dplyr::rename(synonymy, TRSC = "Chromosome Number", posTrsc = "SNP Position")) %>% 
  dplyr::select(-TRSC) %>% 
  group_by(snp) %>% 
  summarise(synonymous = as.numeric(any(synonymous == "Y", na.rm = T))) %>% 
  mutate(synonymous = recode(synonymous, "1" = "synonymous", "0" = "anonymous")) %>% 
  write_tsv(path = file.path(path, "snpSynonymy.annotation"), col_names = F)
# grep synonymous snpSynonymy.annotation | cut -f1 > snp.synonymous
# grep anonymous snpSynonymy.annotation | cut -f1 > snp.anonymous
```

## Genomic diversity

```{bash}
plink=~/Tools/plink_linux_x86_64_20190617/plink
pops=(sp1 sp2 sp3)
for pop in "${pops[@]}" ; 
  do 
  $plink \
   --bfile ../paracou/symcapture.all.biallelic.snp.filtered.nonmissing.paracou \
   --allow-extra-chr \
   --set-missing-var-ids @_# \
   --keep $pop.fam \
   --geno 0 \
   --max-mac 1 \
   --make-bed \
   --out $pop
  head $pop.fam -n 1 > $pop.ind1.fam
  $plink \
   --bfile $pop \
   --allow-extra-chr \
   --keep $pop.ind1.fam \
   --make-bed \
   --out $pop.ind1
done
$plink \
 --bfile sp1.ind1 \
 --allow-extra-chr \
 --merge-list merge.list \
 --geno 0 \
 --make-bed \
 --out all.ind1 \
 --recode A
```

```{r}
t <- read_delim("../../../data/Symphonia_Paracou/Sequences/variantCalling/genomicStructure/all.ind1.raw", delim = " ") %>% 
  dplyr::select(-FID, -PAT, -MAT, -SEX, -PHENOTYPE) %>% 
  reshape2::melt("IID") %>% 
  mutate(IID = recode(IID,
                      "P1-1-117.g.vcf" = "sp1",
                      "P1-1-2101.g.vcf" = "sp2",
                      "P10-3-2262.g.vcf" = "sp3")) %>% 
  rename(population = IID, snp = variable, allele = value)
```

