```{r setup_pophist, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(pophelper)
library(dendextend)
library(ggtree)
library(adegenet)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
```

```{r pophist}
googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop") %>% 
  mutate(Ind = paste0(Ind, ".g.vcf", "")) %>% 
  mutate(FID = 0, IID = Ind, CLUSTER = Pop2) %>% 
  dplyr::select(FID, IID, CLUSTER) %>% 
  filter(CLUSTER %in% c("S", "G")) %>% 
  write_tsv(file.path(path, "paracou.2pop"), col_names = F)
pop <- googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop")
pop <- bind_rows(pop, c(Ind = "P10-3-925", Pop = "SG", Pop2 = "SG")) # to solve
```

# Population history

We will ... :

* __xxx__ ...

## Distance-based phylogeny - `SNPRelate`

```{r phyloclust, echo=T, eval=F}
# srun --mem=80G --cpus-per-task=1 --pty bash
# module load system/R-3.5.2
# module load libraries/gdal-2.3.0
# module load libraries/proj-4.9.3
R
library(gdsfmt)
library(SNPRelate)
snpgdsBED2GDS("../variantCalling/treemix/symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bed",
              "../variantCalling/treemix/symcapture.all.biallelic.snp.filtered.nonmissing.treemix.fam",
              "../variantCalling/treemix/symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bim0",
              "phylogeny/symcapture.all.biallelic.snp.filtered.nonmissing.treemix.gds")
symcapture.geno <- openfn.gds("phylogeny/symcapture.all.biallelic.snp.filtered.nonmissing.treemix.gds")
dissMatrix  <-  snpgdsDiss(symcapture.geno, autosome.only = F, num.thread = 3)
snpHCluster <-  snpgdsHCluster(dissMatrix, hang = 0.25)
cutTree <- snpgdsCutTree(snpHCluster, z.threshold = 15, outlier.n = 5, n.perm = 5000,
                        col.outlier = "red", pch.outlier=4)
save(cutTree, file = "phylogeny/tree.Rdata")
```

```{r phylo, fig.cap="Distance-based phylogeny of *Symphonia* and *Pentadesma* populations with `SNPRelate`."}
load(file.path(path, "phylogeny", "tree.Rdata"))
symcapture.phylo <- ape::as.phylo(cutTree$dendrogram)
pop2 <-  read_tsv(file.path(path, "treemix.fam"), 
                 col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype", "pop"))
g <- ggtree(symcapture.phylo, layout='circular')
g <- ggtree(symcapture.phylo, layout = 'daylight')
g +
  geom_hilight_encircle(node=33, fill = RColorBrewer::brewer.pal(8, "Accent")[1]) +
  geom_hilight_encircle(node=35, fill = RColorBrewer::brewer.pal(8, "Accent")[2]) +
  geom_hilight_encircle(node=37, fill = RColorBrewer::brewer.pal(8, "Accent")[3]) +
  geom_hilight_encircle(node=44, fill = RColorBrewer::brewer.pal(8, "Accent")[4]) +
  geom_hilight_encircle(node=46, fill = RColorBrewer::brewer.pal(8, "Accent")[5]) +
  geom_hilight_encircle(node=48, fill = RColorBrewer::brewer.pal(8, "Accent")[6]) +
  geom_hilight_encircle(node=53, fill = RColorBrewer::brewer.pal(8, "Accent")[7]) +
  geom_hilight_encircle(node=54, fill = RColorBrewer::brewer.pal(8, "Accent")[8]) +
  geom_tiplab(data = dplyr::inner_join(g$data, pop2, by = c("label" = "IID")),
              aes(label = pop))
```

## Drift-based phylogeny - `treemix`

```{r treemixR, eval=F}
fam <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID))
symcapture.admix <- readQ(list.files(file.path(path, "admixture", "paracou"), 
                                     full.names = T, pattern = ".Q"), indlabfromfile=F)
symcapture.admix <- lapply(symcapture.admix, "rownames<-", fam$Ind)
fam <- read_tsv(file.path(path, "..", "variantCalling", "nonmissing",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  left_join(pop)
symcapture.admix[[4]] %>% 
  mutate(Ind = row.names(.)) %>% 
  left_join(pop) %>% 
  top_n(5, Cluster1) %>% 
  sample_n(5) %>% 
  mutate(Pop = "S. sp1") %>% 
  bind_rows(symcapture.admix[[4]] %>% 
              mutate(Ind = row.names(.)) %>% 
              left_join(pop) %>% 
              top_n(5, Cluster2) %>% 
              sample_n(5) %>% 
              mutate(Pop = "S. globulifera type Paracou")) %>% 
  bind_rows(symcapture.admix[[4]] %>% 
              mutate(Ind = row.names(.)) %>% 
              left_join(pop) %>% 
              top_n(5, Cluster3) %>% 
              sample_n(5) %>% 
              mutate(Pop = "S. globulifera type Régina")) %>% 
  mutate(IID = paste0(Ind, ".g.vcf")) %>% 
  mutate(FID = 0, FIID = 0, MIID = 0, sex = 0, phenotype = -9) %>% 
  dplyr::select(FID, IID, FIID, MIID, sex, phenotype, Pop) %>% 
  bind_rows(fam %>% 
              filter(Pop2 == "O") %>% 
              select(FID, IID, FIID, MIID, sex, phenotype, Pop)) %>% 
  mutate(Pop = gsub(" ", "_", Pop)) %>% 
  mutate(Pop = recode(Pop, "Costa_Rica" = "CostaRica", "Ivory_Coast" = "IvoryCoast",
                      "P._b_Benin" = "PbBenin", "P._b_Cameroun" = "PbCameroun",
                      "Sao_Tome"	= "SaoTome", "S._globulifera_type_Paracou"	= "SgParacou",
                      "S._globulifera_type_Régina"	= "SgRegina", 
                      "S._n_Madagascar" = "SnMadagascar", "S._sp1" = "Ssp1",
                      "S._u_Madagascar" = "SuMadagascar")) %>% 
  filter(!(IID %in% c("MHDNA0294.g.vcf", "MHDNA0295.g.vcf", "GID1791.g.vcf", "BOD407.g.vcf", "MH2832.g.vcf"))) %>% 
  write_tsv(file.path(path, "treemix.fam"), col_names = F)

symcapture.admix[[4]] %>% 
  mutate(Ind = row.names(.)) %>% 
  left_join(pop) %>% 
  top_n(5, Cluster1) %>% 
  sample_n(5) %>% 
  mutate(Pop = "S. sp1") %>% 
  bind_rows(symcapture.admix[[4]] %>% 
              mutate(Ind = row.names(.)) %>% 
              left_join(pop) %>% 
              top_n(5, Cluster2) %>% 
              sample_n(5) %>% 
              mutate(Pop = "S. globulifera type Paracou")) %>% 
  bind_rows(symcapture.admix[[4]] %>% 
              mutate(Ind = row.names(.)) %>% 
              left_join(pop) %>% 
              top_n(5, Cluster3) %>% 
              sample_n(5) %>% 
              mutate(Pop = "S. globulifera type Régina")) %>% 
  mutate(IID = paste0(Ind, ".g.vcf")) %>% 
  mutate(FID = 0, FIID = 0, MIID = 0, sex = 0, phenotype = -9) %>% 
  dplyr::select(FID, IID, FIID, MIID, sex, phenotype, Pop) %>% 
  bind_rows(fam %>% 
              filter(Pop2 == "O") %>% 
              select(FID, IID, FIID, MIID, sex, phenotype, Pop)) %>% 
  filter(!(IID %in% c("MHDNA0294.g.vcf", "MHDNA0295.g.vcf", "GID1791.g.vcf", 
                      "BOD407.g.vcf", "MH2832.g.vcf"))) %>% 
  mutate(Pop = gsub(" ", "_", Pop)) %>%
  mutate(Pop = recode(Pop, "Costa_Rica" = "CostaRica", "Benin" = "Africa", "Ivory_Coast" = "Africa",
                      "Cameroun" = "Africa", "Congo" = "Africa", "P._b_Cameroun" = "Africa", "Gabon" = "Africa",
                      "Sao_Tome"	= "SaoTome", "S._globulifera_type_Paracou"	= "SgParacou",
                      "S._globulifera_type_Régina"	= "SgRegina", 
                      "S._n_Madagascar" = "Madagascar", "S._sp1" = "Ssp1",
                      "S._u_Madagascar" = "Madagascar")) %>% 
  write_tsv(file.path(path, "treemix2.fam"), col_names = F)
```


```{bash treemix, eval=F, echo=T}
cd ../variantCalling/treemix
module load bioinfo/plink-v1.90b5.3
cut ../treemix2.fam -f1,2,7 > symcapture.all.biallelic.snp.filtered.nonmissing.treemix.pop
cp symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bim symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bim0
awk '{print $1"\t"NR"\t"$3"\t"$4"\t"$5"\t"$6}' symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bim0 > symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bim
  plink --bfile symcapture.all.biallelic.snp.filtered.nonmissing.treemix \
    --allow-extra-chr \
     --freq --missing --within symcapture.all.biallelic.snp.filtered.nonmissing.treemix.pop \
    --out symcapture.all.biallelic.snp.filtered.nonmissing.treemix
gzip symcapture.all.biallelic.snp.filtered.nonmissing.treemix.frq.strat
python plink2treemix.py symcapture.all.biallelic.snp.filtered.nonmissing.treemix.frq.strat.gz treemix.frq.gz 
cp treemix.frq.gz ../../populationGenomics/treemix
cd ../../populationGenomics/treemix
module load bioinfo/treemix-1.13
treemix -i treemix.frq.gz -root Madagascar -o out
for m in $(seq 10) ; do treemix -i treemix.frq.gz -root Madagascar -m $m -g out.vertices.gz out.edges.gz -o out$m
grep Exiting *.llik > migration.llik
```


```{r treemixPlot, fig.cap="Drift-based phylogeny of *Symphonia* and *Pentadesma* populations with `treemix`. Subfigure A. present the tree log-likelihood depending on the number of allowed migration events, suggesting 1 migration event to better represent the tree than none. Subfigure B. represent the standard-error of the covariation between branches. And Subfigre C. represent the phylogeny depending on the drift parameters with a red arrow to represent the migration event from *Sp.1* and Brazil most common ancestors to *S. globulifera* most commaon ancestors."}
g.llik <- read_tsv(file.path(path, "treemix", "migration.likelihood"), col_names = F) %>% 
  separate(X1, c("M", "what", "likelihood"), ":", convert = T) %>% 
  mutate(M = as.numeric(gsub("out", "", gsub(".llik", "", M)))) %>% 
  ggplot(aes(M, likelihood)) +
  geom_point() +
  geom_line() +
  xlab("Migration events") + ylab("Log-likelihhod") +
  geom_vline(xintercept = 1) + labs(tag = "A.")
g.tree <- ggtree(ape::read.tree(file.path(path, "treemix", "out1.treeout")), layout = "slanted") +
  geom_taxalink(13, 15, curvature = 0, col = "red",
                arrow = arrow(angle = 10, length = unit(0.1, "inches"))) +
  geom_tiplab() + theme_tree2() + xlab("Drift parameter") + labs(tag = "B.")
cov.se <- read.table(file.path(path, "treemix", "out1.covse"), row.names = 1)
cov.se[lower.tri(cov.se, diag=T)] <- 0
g.se <- cov.se %>% 
  mutate(P1 = row.names(.)) %>% 
  reshape2::melt(id.vars = "P1", variable.name = "P2", value.name = "cov.se") %>% 
  ggplot(aes(P1, P2, fill = cov.se)) +
  geom_tile() +
  scale_fill_gradient(name = "Covariance\nstandard-error",
                      guide = "none",
                      low = "#FFFFFF",
                      high = "#012345") +
  theme(axis.text.x = element_text(angle = -90),
        axis.title = element_blank()) + labs(tag = "C.")
gridExtra::grid.arrange(g.llik, g.se, g.tree,   
                        heights = c(1, 2),
                        layout_matrix = rbind(c(1, 2), c(3, 3)))
```

## Site Frequency Structure

### Two populations

```{r vcf2sfs2pop, eval=F, echo=T}
source("https://raw.githubusercontent.com/shenglin-liu/vcf2sfs/master/vcf2sfs.r")
path <- "/home/sylvain/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/variantCalling/paracouWeighted/"
vcf <- file.path(path, "symcapture.all.biallelic.snp.filtered.nonmissing.paracouWeighted.vcf")
popmap <- file.path(path, "paracouWeighted.popmap")
pops <- c("sp1", "globulifera")
for(pop in pops){
  vcf2dadi(vcf, popmap, paste0(path, pop, ".dadi"), pop)
  dadi2fsc.1pop(paste0(path, pop, ".dadi"), paste0(path, pop, ".fastsim"),
                pop, fold = T)
}
vcf2dadi(vcf, popmap, paste0(path, "all", ".dadi"), pops)
dadi2fsc.2pop(paste0(path, "all", ".dadi"), paste0(path, "all", ".fastsim"),
              pops, fold = T)
```

```{r sfs1pop2pop, fig.cap="Number of alleles per allele count and population."}
pops <- c("sp1", "globulifera")
lapply(pops, function(pop)
  read_tsv(file.path(path, "..", "variantCalling", "paracouWeighted",
                     paste0(pop, ".fastsim")), skip = 1) %>% gather()) %>% 
  bind_rows() %>% 
  separate(key, c("population", "AlleleCount"), convert = T) %>% 
  mutate(population = gsub("d", "", population)) %>% 
  filter(AlleleCount > 0) %>% 
  ggplot(aes(AlleleCount, value, fill = population)) +
  geom_bar(stat = "identity") +
  facet_wrap(~ population, nrow = 3) +
  scale_fill_discrete(guide = "none") +
  xlim(0,100) +
  xlab("Allele count within population") + ylab("Number of alleles")
```

[Berlin presentation](https://github.com/speciationgenomics/presentations/blob/master/2019-8-Demographic_modeling_JM.pdf)

```{r sfs2pop2pop, fig.cap="Number of alleles per allele count between population."}
g <- read_tsv(file.path(path, "..", "variantCalling", "paracouWeighted",
                        "all.fastsim"), skip = 1) %>%
  reshape2::melt(value.name = "AlleleCount") %>% 
  separate(X1, c("X1", "sp1"), convert = T) %>% 
  separate(variable, c("X2", "globulifera"), convert = T) %>% 
  dplyr::select(-X1, -X2) %>% 
  filter(AlleleCount > 0) %>% 
  ggplot(aes(sp1, globulifera, fill = log(AlleleCount+1))) +
  geom_tile() +
  scale_fill_gradientn(colours = rev(rainbow(10)))
plotly::ggplotly(g)
```

### Three populations

```{r vcf2sfs3pop, eval=F, echo=T}
source("https://raw.githubusercontent.com/shenglin-liu/vcf2sfs/master/vcf2sfs.r")
path <- "/home/sylvain/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/variantCalling/paracouWeighted3pop/"
vcf <- file.path(path, "symcapture.all.biallelic.snp.filtered.nonmissing.paracouWeighted3pop.vcf")
popmap <- file.path(path, "paracouWeighted3pop.popmap")
pops <- c("sp1", "globuliferaTypeParacou", "globuliferaTypeRegina")
for(pop in pops){
  vcf2dadi(vcf, popmap, paste0(path, pop, ".dadi"), pop)
  dadi2fsc.1pop(paste0(path, pop, ".dadi"), paste0(path, pop, ".fastsim"),
                pop, fold = T)
}
popPairs <- combn(pops,2)
for(i in 1:ncol(popPairs)){
  vcf2dadi(vcf, popmap, paste0(path, popPairs[1,i], "_", popPairs[2,i], ".dadi"),
           popPairs[,i])
  dadi2fsc.2pop(paste0(path, popPairs[1,i], "_", popPairs[2,i], ".dadi"), 
                paste0(path, popPairs[1,i], "_", popPairs[2,i], ".fastsim"),
                popPairs[,i], fold = T)
}
```

```{r sfs1pop3pop, fig.cap="Number of alleles per allele count and population."}
pops <- c("sp1", "globuliferaTypeParacou", "globuliferaTypeRegina")
lapply(pops, function(pop)
  read_tsv(file.path(path, "..", "variantCalling", "paracouWeighted3pop",
                     paste0(pop, ".fastsim")), skip = 1) %>% gather()) %>% 
  bind_rows() %>% 
  separate(key, c("population", "AlleleCount"), convert = T) %>% 
  mutate(population = gsub("d", "", population)) %>% 
  filter(AlleleCount > 0) %>% 
  ggplot(aes(AlleleCount, value, fill = population)) +
  geom_bar(stat = "identity") +
  facet_wrap(~ population, nrow = 3) +
  scale_fill_discrete(guide = "none") +
  xlab("Allele count within population") + ylab("Number of alleles")
```

[Berlin presentation](https://github.com/speciationgenomics/presentations/blob/master/2019-8-Demographic_modeling_JM.pdf)

```{r sfs2pop3pop, fig.cap="Number of alleles per allele count between population."}
popPairs <- combn(pops,2)
g <- lapply(1:length(pops), function(i)
  read_tsv(file.path(path, "..", "variantCalling", "paracouWeighted3pop",
                     paste0(popPairs[1,i], "_", popPairs[2,i], ".fastsim")), skip = 1) %>% 
    reshape2::melt(value.name = "AlleleCount")) %>% 
  bind_rows() %>% 
  separate(X1, c("population1", "pop1"), convert = T) %>% 
  separate(variable, c("population2", "pop2"), convert = T) %>% 
  mutate(population1 = gsub("d", "", population1)) %>% 
  mutate(population2 = gsub("d", "", population2)) %>% 
  filter(AlleleCount > 0) %>% 
  ggplot(aes(pop1, pop2, fill = log(AlleleCount+1))) +
  geom_tile() +
  scale_fill_gradientn(colours = rev(rainbow(10))) +
  facet_grid(population1 ~ population2) +
  theme(axis.title = element_blank())
plotly::ggplotly(g)
```

## Conclusion

