```{r setup_pophist, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(pophelper)
library(dendextend)
library(ggtree)
library(adegenet)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
```

```{r pophist}
googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop") %>% 
  mutate(Ind = paste0(Ind, ".g.vcf", "")) %>% 
  mutate(FID = 0, IID = Ind, CLUSTER = Pop2) %>% 
  dplyr::select(FID, IID, CLUSTER) %>% 
  filter(CLUSTER %in% c("S", "G")) %>% 
  write_tsv(file.path(path, "paracou.2pop"), col_names = F)
pop <- googlesheets::gs_title("Symcapture") %>% 
              googlesheets::gs_read("Pop")
pop <- bind_rows(pop, c(Ind = "P10-3-925", Pop = "SG", Pop2 = "SG")) # to solve
```

# Population history

We will ... :

* __Phylogeny__ drift-based phylogeny with `treemix` for neutral and functional SNPs
* __Population diversity__ metrics ($\pi$, $F_{st}$ and Tajima's $D$) for neutral and functional SNPs
* __Site Frequency Structure__ per population and between populations for neutral and functional SNPs
  
## Drift-based Phylogeny - `treemix`

We used representative individuals from *S. sp1*, *S. globulifera type Paracou*, and *S. globulifera type Régina*, in association with South American (La Selva, Baro Colorado Island, and Itubera) and African (Madagascar, Benin, Cameroun, Sao Tome, Congo, Benin, Liberia, Gana and Ivory Coast) *Symphonia* and *Pentadesma* to explore population phylogeny with `treemix` (Fig. \@ref(fig:treemixPlot)). We built phylogeny for the neutral, hitchhiker, and functional SNPs. All datasets suggested 1 migration event to better represent the phylogeny than none. Topology was the same between dataset, functional SNP just show less drift than others.

```{r treemixR, eval=F}
fam <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID))
symcapture.admix <- readQ(list.files(file.path(path, "admixture", "paracou"), 
                                     full.names = T, pattern = ".Q"), indlabfromfile=F)
symcapture.admix <- lapply(symcapture.admix, "rownames<-", fam$Ind)
fam <- read_tsv(file.path(path, "..", "variantCalling", "nonmissing",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  left_join(pop)
symcapture.admix[[4]] %>% 
  mutate(Ind = row.names(.)) %>% 
  left_join(pop) %>% 
  top_n(5, Cluster1) %>% 
  sample_n(5) %>% 
  mutate(Pop = "S. sp1") %>% 
  bind_rows(symcapture.admix[[4]] %>% 
              mutate(Ind = row.names(.)) %>% 
              left_join(pop) %>% 
              top_n(5, Cluster2) %>% 
              sample_n(5) %>% 
              mutate(Pop = "S. globulifera type Paracou")) %>% 
  bind_rows(symcapture.admix[[4]] %>% 
              mutate(Ind = row.names(.)) %>% 
              left_join(pop) %>% 
              top_n(5, Cluster3) %>% 
              sample_n(5) %>% 
              mutate(Pop = "S. globulifera type Régina")) %>% 
  mutate(IID = paste0(Ind, ".g.vcf")) %>% 
  mutate(FID = 0, FIID = 0, MIID = 0, sex = 0, phenotype = -9) %>% 
  dplyr::select(FID, IID, FIID, MIID, sex, phenotype, Pop) %>% 
  bind_rows(fam %>% 
              filter(Pop2 == "O") %>% 
              select(FID, IID, FIID, MIID, sex, phenotype, Pop)) %>% 
  mutate(Pop = gsub(" ", "_", Pop)) %>% 
  mutate(Pop = recode(Pop, "Costa_Rica" = "CostaRica", "Ivory_Coast" = "IvoryCoast",
                      "P._b_Benin" = "PbBenin", "P._b_Cameroun" = "PbCameroun",
                      "Sao_Tome"	= "SaoTome", "S._globulifera_type_Paracou"	= "SgParacou",
                      "S._globulifera_type_Régina"	= "SgRegina", 
                      "S._n_Madagascar" = "SnMadagascar", "S._sp1" = "Ssp1",
                      "S._u_Madagascar" = "SuMadagascar")) %>% 
  filter(!(IID %in% c("MHDNA0294.g.vcf", "MHDNA0295.g.vcf", "GID1791.g.vcf", "BOD407.g.vcf", "MH2832.g.vcf"))) %>% 
  write_tsv(file.path(path, "treemix.fam"), col_names = F)

symcapture.admix[[4]] %>% 
  mutate(Ind = row.names(.)) %>% 
  left_join(pop) %>% 
  top_n(5, Cluster1) %>% 
  sample_n(5) %>% 
  mutate(Pop = "S. sp1") %>% 
  bind_rows(symcapture.admix[[4]] %>% 
              mutate(Ind = row.names(.)) %>% 
              left_join(pop) %>% 
              top_n(5, Cluster2) %>% 
              sample_n(5) %>% 
              mutate(Pop = "S. globulifera type Paracou")) %>% 
  bind_rows(symcapture.admix[[4]] %>% 
              mutate(Ind = row.names(.)) %>% 
              left_join(pop) %>% 
              top_n(5, Cluster3) %>% 
              sample_n(5) %>% 
              mutate(Pop = "S. globulifera type Régina")) %>% 
  mutate(IID = paste0(Ind, ".g.vcf")) %>% 
  mutate(FID = 0, FIID = 0, MIID = 0, sex = 0, phenotype = -9) %>% 
  dplyr::select(FID, IID, FIID, MIID, sex, phenotype, Pop) %>% 
  bind_rows(fam %>% 
              filter(Pop2 == "O") %>% 
              select(FID, IID, FIID, MIID, sex, phenotype, Pop)) %>% 
  filter(!(IID %in% c("MHDNA0294.g.vcf", "MHDNA0295.g.vcf", "GID1791.g.vcf", 
                      "BOD407.g.vcf", "MH2832.g.vcf"))) %>% 
  mutate(Pop = gsub(" ", "_", Pop)) %>%
  mutate(Pop = recode(Pop, "Costa_Rica" = "CostaRica", "Benin" = "Africa", "Ivory_Coast" = "Africa",
                      "Cameroun" = "Africa", "Congo" = "Africa", "P._b_Cameroun" = "Africa", "Gabon" = "Africa",
                      "Sao_Tome"	= "SaoTome", "S._globulifera_type_Paracou"	= "SgParacou",
                      "S._globulifera_type_Régina"	= "SgRegina", 
                      "S._n_Madagascar" = "Madagascar", "S._sp1" = "Ssp1",
                      "S._u_Madagascar" = "Madagascar")) %>% 
  write_tsv(file.path(path, "treemix2.fam"), col_names = F)
```


```{bash treemix, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
cp symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bim symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bim0
awk '{print $1"\t"$1"_snp"$4"\t"$3"\t"$4"\t"$5"\t"$6}' symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bim0 > symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bim
rm symcapture.all.biallelic.snp.filtered.nonmissing.treemix.bim0
plink --bfile all/symcapture.all.biallelic.snp.filtered.nonmissing.treemix \
  --allow-extra-chr \
  --freq --missing --within symcapture.all.biallelic.snp.filtered.nonmissing.treemix.pop \
  --out all/symcapture.all.biallelic.snp.filtered.nonmissing.treemix
plink --bfile all/symcapture.all.biallelic.snp.filtered.nonmissing.treemix \
  --extract snps.functional \
  --allow-extra-chr \
  --freq --missing --within symcapture.all.biallelic.snp.filtered.nonmissing.treemix.pop \
  --out functional/symcapture.all.biallelic.snp.filtered.nonmissing.treemix
plink --bfile all/symcapture.all.biallelic.snp.filtered.nonmissing.treemix \
  --extract snps.hitchhiker \
  --allow-extra-chr \
  --freq --missing --within symcapture.all.biallelic.snp.filtered.nonmissing.treemix.pop \
  --out hitchhiker/symcapture.all.biallelic.snp.filtered.nonmissing.treemix
plink --bfile all/symcapture.all.biallelic.snp.filtered.nonmissing.treemix \
  --extract snps.neutral \
  --allow-extra-chr \
  --freq --missing --within symcapture.all.biallelic.snp.filtered.nonmissing.treemix.pop \
  --out neutral/symcapture.all.biallelic.snp.filtered.nonmissing.treemix
gzip */symcapture.all.biallelic.snp.filtered.nonmissing.treemix.frq.strat
python plink2treemix.py all/symcapture.all.biallelic.snp.filtered.nonmissing.treemix.frq.strat.gz treemix.all.frq.gz 
python plink2treemix.py neutral/symcapture.all.biallelic.snp.filtered.nonmissing.treemix.frq.strat.gz treemix.neutral.frq.gz 
python plink2treemix.py functional/symcapture.all.biallelic.snp.filtered.nonmissing.treemix.frq.strat.gz treemix.functional.frq.gz 
python plink2treemix.py hitchhiker/symcapture.all.biallelic.snp.filtered.nonmissing.treemix.frq.strat.gz treemix.hitchhiker.frq.gz 
cp *.frq.gz ../../populationGenomics/treemix
cd ../../populationGenomics/treemix
module load bioinfo/treemix-1.13
treemix -i treemix.frq.gz -root Madagascar -o out
for m in $(seq 10) ; do treemix -i treemix.frq.gz -root Madagascar -m $m -g out.vertices.gz out.edges.gz -o out$m ; done
grep Exiting *.llik > migration.llik
```

```{r treemixPlot, fig.cap="Drift-based phylogeny of *Symphonia* and *Pentadesma* populations with `treemix` [@Pickrell2012]. Subfigure **A** present the log-likelihood of the phylogeny topology depending on the number of allowed migration events per SNP type, suggesting 1 migration event to better represent the phylogeny topology than none. Others subfigures represent the phylogeny for anonymous (**B**), genic (**C**) and putatively-hitchhiker (**D**) SNPs. The red arrow represents the most likely migration event. Population are named by their localities, including *Symphonia* species only or *Symphonia* and *Pentadesma* species in Africa. At the exception of the three Paracou populations: *S. sp1*, *S. globulifera type Paracou* and *S. globulifera type Regina* respectivelly named Ssp1, SgParacou and SgRegina."}
cols <- c("#984EA3", "#377EB8", "#4DAF4A")
names(cols) <- c("neutral", "functional", "hitchhiker")
cols2 <- cols
names(cols2) <- c("anonymous", "genic", "hitchhiker")
g.llik <- lapply(c("neutral", "hitchhiker", "functional"), function(type)
  read_tsv(file.path(path, "treemix", paste0("migration.", type, ".llik")), 
           col_names = F) %>% 
    mutate(type = type)) %>% 
  bind_rows() %>% 
  separate(X1, c("M", "what", "likelihood"), ":", convert = T) %>% 
  mutate(M = as.numeric(gsub("out", "", gsub(".llik", "", M)))) %>% 
  mutate(type = recode(type, 
                       "functional" = "genic",
                       "neutral" = "anonymous", 
                       "hitchhiker" = "hitchhiker")) %>% 
  ggplot(aes(M, likelihood, col = type)) +
  geom_point() +
  geom_line() +
  xlab("Migration events") + 
  ylab("Log-likelihhod") +
  geom_vline(xintercept = 1, col = "red") + 
  theme(legend.position = c(0.6,0.4)) +
  scale_color_manual(values = cols2)
g.tree <- lapply(c("neutral", "functional", "hitchhiker"),
                 function(type)
                   ggtree::ggtree(ape::read.tree(file.path(path, "treemix", 
                                                           type, "out1.treeout")),
                                  layout = "slanted", col = cols[type]) +
                   ggtree::geom_taxalink(13, 15, curvature = 0, col = "red",
                                         arrow = arrow(angle = 10, 
                                                       length = unit(0.1, "inches"))) +
                   ggtree::geom_tiplab() + 
                   ggtree::theme_tree2() + 
                   xlim(0,0.035) +
                   xlab("Drift parameter"))
cowplot::plot_grid(g.llik, g.tree[[1]], g.tree[[2]], g.tree[[3]], 
                   labels = LETTERS[1:4])
```

## Population diversity

After defining populations based on our 3 gene pools (*S. sp.1*, *S. globulifera type Paracou*, and *S. globulifera type Regina*) with more than 90% of mebership to the gene pool in `admixture`, we used `vcftools` to compute nucleotide diversity $\pi$, population differentiation $F_{st}$, and Tajima's $D$ per SNP type (functional, hitchhiker or neutral).

```{bash populationDiv, eval=F, echo=T}
cp symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop.bim symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop.bim0
awk '{print $1"\t"$1"_snp"$4"\t"$3"\t"$4"\t"$5"\t"$6}' symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop.bim0 > symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop.bim
rm symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop.bim0
module load bioinfo/plink-v1.90b5.3
types=(neutral hitchhiker functional)
for type in "${types[@]}" ; 
  do 
  plink \
    --bfile symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
    --allow-extra-chr \
    --extract snps.$type \
    --recode vcf-iid \
    -out paracou3pop.$type &
done
grep sp1 paracou3pop.popmap | cut -f1 > sp1.list
grep globuliferaTypeParacou paracou3pop.popmap | cut -f1 > globuliferaTypeParacou.list
grep globuliferaTypeRegina paracou3pop.popmap | cut -f1 > globuliferaTypeRegina.list
```

### $\pi$

Nucleotide diversity $\pi$ per site had a mean of 0.05140 across populations and was significantly different between populations (ANOVA, p<2e-16) with *S. globulifera type Regina* being more diverse (Fig. \@ref(fig:piR)). No significant differences existed between SNP types.

```{bash pi, eval=F, echo=T}
types=(neutral hitchhiker functional)
pops=(sp1 globuliferaTypeParacou globuliferaTypeRegina) 
for type in "${types[@]}" ; 
  do 
  for pop in "${pops[@]}" ; 
    do 
    vcftools --vcf paracou3pop.$type.vcf --keep $pop.list --site-pi --out $type.$pop &
  done
done
types=(synonymous anonymous)
pops=(sp1 globuliferaTypeParacou globuliferaTypeRegina) 
for type in "${types[@]}" ; 
  do 
  for pop in "${pops[@]}" ; 
    do 
    vcftools --vcf paracou3pop.$type.vcf --keep $pop.list --window-pi 100 --out $type.$pop &
  done
done
```

```{r piR, fig.cap="Populations $\\pi$ distribution estimated by `vcftools` per site."}
lapply(list.files(file.path(path, "..", "populationGenomics", "diversity"), pattern = ".sites.pi"), function(file)
  read_tsv(file.path(path, "..", "populationGenomics", "diversity", file)) %>% 
    mutate(file = file)) %>% 
  bind_rows() %>% 
  separate(file, c("type", "Population", "sites", "pi"), sep = "[:punct:]") %>% 
  filter(type %in% c("neutral", "hitchiker", "functional")) %>% 
  dplyr::select(-sites, -pi) %>% 
  ggplot(aes(Population, PI, fill = Population)) +
  geom_boxplot() +
  facet_wrap(~ type) +
  scale_fill_discrete(guide = "none") +
  scale_y_log10() +
  ylab(expression(pi)) +
  theme(axis.text.x = element_text(angle = 45),
        axis.line.x = element_blank())
```

```{r piR2, fig.cap="Populations $\\frac{\\pi_a}{\\pi_s}$ distribution estimated by `vcftools` on a 100 bp window."}
lapply(list.files(file.path(path, "..", "populationGenomics", "diversity"), pattern = ".windowed.pi"), function(file)
  read_tsv(file.path(path, "..", "populationGenomics", "diversity", file)) %>% 
    mutate(file = file)) %>% 
  bind_rows() %>% 
  separate(file, c("type", "Population", "sites", "pi"), sep = "[:punct:]") %>% 
  filter(type %in% c("anonymous", "synonymous")) %>% 
  dplyr::select(-sites, -pi) %>% 
  reshape2::dcast(Population + CHROM + BIN_START + BIN_END ~ type, value.var = "PI") %>% 
  na.omit() %>% 
  mutate(piapis = anonymous/synonymous) %>% 
  ggplot(aes(Population, piapis, fill = Population)) +
  geom_boxplot() +
  geom_hline(yintercept = 1, linetype = "dashed") +
  scale_fill_discrete(guide = "none") +
  ylab(expression(frac(pi[a],pi[s]))) +
  scale_y_log10()
```

### $F_{st}$

$F_{st}$ between population was globally low with a mean value of 0.15, still *S. globulifera type Regina* was more differentiated to the two toher gene pools (Fig. \@ref(fig:FstR)) for every SNP type. Nevertheless, functional SNP were significantly less differentiated than hitchhikers and neutral SNPs. 

```{bash Fst, eval=F, echo=T}
types=(neutral hitchhiker functional)
pop1=(sp1 sp1 globuliferaTypeParacou) 
pop2=(globuliferaTypeParacou globuliferaTypeRegina globuliferaTypeRegina) 
for type in "${types[@]}" ; 
  do 
  for i in $(seq 3) ; 
    do 
    vcftools --vcf paracou3pop.$type.vcf \
      --weir-fst-pop "${pop1[i-1]}".list \
      --weir-fst-pop "${pop2[i-1]}".list \
      --out $type."${pop1[i-1]}"_"${pop2[i-1]}" &
  done
done
```

```{r FstR, fig.cap="Between populations Fst estimated by `vcftools`."}
lapply(list.files(file.path(path, "..", "populationGenomics", "diversity"), pattern = ".log"), function(file)
  read_delim(file.path(path, "..", "populationGenomics", "diversity", file), skip = 15, delim = ":",
  col_names = c("variable", "value")) %>% 
    mutate(file = file)) %>% 
  bind_rows() %>% 
  filter(!is.na(value)) %>% 
  mutate(value = as.numeric(value)) %>% 
  mutate(variable = ifelse(grepl("mean", variable), "meanFst", "weightedFst")) %>% 
  separate(file, c("type", "Pop1", "Pop2", "log"), sep = "[:punct:]") %>% 
  dplyr::select(-log) %>% 
  ggplot(aes(Pop1, Pop2, label = value, fill = value)) +
  geom_tile() +
  geom_text(col = "white") +
  facet_grid(type ~ variable) +
  theme(axis.title = element_blank(),
        axis.ticks = element_blank(),
        axis.line = element_blank()) +
  scale_fill_gradient(high = "#132B43", low = "#56B1F7")
```

### Tajima's $D$

Tajima's $D$ between population was globally low and significantly negative with a mean value of -0.795 and was significantly different between populations (ANOVA, p<2e-16) with *S. globulifera type Regina* having a globally higher value (Fig. \@ref(fig:TajimaDR)). We expect positive selection (or selective sweeps) to give us a negative Tajima’s $D$ in a population that doesn’t have any demographic changes going on (population expansion/contraction, migration, etc). On the other hand with balancing selection, alleles are kept at intermediate frequencies. This produces a positive Tajima’s $D$ because there will be more pairwise differences than segregating sites. Consequently, Tajima's $D$ indicates here that are population are under selection with a stronger selection operating on *S. globulifera type Paracou* and *S. sp1* than *S. globulifera type Regina*. No significant differences existed between SNP types.

```{bash TajimaD, eval=F, echo=T}
types=(neutral hitchhiker functional)
pops=(sp1 globuliferaTypeParacou globuliferaTypeRegina) 
for type in "${types[@]}" ; 
  do 
  for pop in "${pops[@]}" ; 
    do 
    vcftools --vcf paracou3pop.$type.vcf --keep $pop.list --TajimaD 100 --out $type.$pop &
  done
done
```

```{r TajimaDR, fig.cap="Populations Tajima's D distribution estimated by `vcftools` on windows of 100 bp."}
lapply(list.files(file.path(path, "..", "populationGenomics", "diversity"), pattern = ".Tajima.D"), function(file)
  read_tsv(file.path(path, "..", "populationGenomics", "diversity", file)) %>% 
    mutate(file = file)) %>% 
  bind_rows() %>% 
  separate(file, c("type", "Population", "sites", "pi"), sep = "[:punct:]") %>% 
  dplyr::select(-sites, -pi) %>% 
  ggplot(aes(Population, TajimaD, fill = Population)) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "lightgrey") +
  geom_boxplot() +
  facet_wrap(~ type) +
  scale_fill_discrete(guide = "none") +
  theme(axis.text.x = element_text(angle = 45),
        axis.line.x = element_blank())
```

## Site Frequency Structure

[Berlin presentation](https://github.com/speciationgenomics/presentations/blob/master/2019-8-Demographic_modeling_JM.pdf)

```{bash SFS, eval=F, echo=T}
cp symcapture.all.biallelic.snp.filtered.nonmissing.paracouWeighted3pop.bim symcapture.all.biallelic.snp.filtered.nonmissing.paracouWeighted3pop.bim0
awk '{print $1"\t"$1"_snp"$4"\t"$3"\t"$4"\t"$5"\t"$6}' symcapture.all.biallelic.snp.filtered.nonmissing.paracouWeighted3pop.bim0 > symcapture.all.biallelic.snp.filtered.nonmissing.paracouWeighted3pop.bim
rm symcapture.all.biallelic.snp.filtered.nonmissing.paracouWeighted3pop.bim0
module load bioinfo/plink-v1.90b5.3
types=(neutral hitchhiker functional)
for type in "${types[@]}" ; 
  do 
  plink \
    --bfile symcapture.all.biallelic.snp.filtered.nonmissing.paracouWeighted3pop \
    --allow-extra-chr \
    --extract snps.$type \
    --recode vcf-iid \
    -out paracouWeighted3pop.$type &
done
```

```{r vcf2sfs3pop, eval=F, echo=T}
source("https://raw.githubusercontent.com/shenglin-liu/vcf2sfs/master/vcf2sfs.r")
prefix <- file.path(path, "SFS", "paracouWeighted3pop")
pops <- c("sp1", "globuliferaTypeParacou", "globuliferaTypeRegina")
types <- c("neutral", "hitchhiker", "functional")
for(type in types)
  for(pop in pops){
    vcf2dadi(paste0(prefix, ".", type, ".vcf"), 
             paste0(prefix, ".popmap"), 
             paste0(prefix, ".", type, ".", pop, ".dadi"), 
             pop)
    dadi2fsc.1pop(paste0(prefix, ".", type, ".", pop, ".dadi"), 
                  paste0(prefix, ".", type, ".", pop, ".fastsim"),
                  pop, fold = T)
  }
popPairs <- combn(pops,2)
for(type in types)
  for(i in 1:ncol(popPairs)){
  vcf2dadi(paste0(prefix, ".", type, ".vcf"), 
           paste0(prefix, ".popmap"),
           paste0(prefix, ".", type, ".", popPairs[1,i], "_", popPairs[2,i], ".dadi"), 
           popPairs[,i])
  dadi2fsc.2pop(paste0(prefix, ".", type, ".", popPairs[1,i], "_", popPairs[2,i], ".dadi"), 
                paste0(prefix, ".", type, ".", popPairs[1,i], "_", popPairs[2,i], ".fastsim"),
                popPairs[,i], fold = T)
}
```

```{r sfs1pop3pop, fig.cap="Number of alleles per allele count and population."}
prefix <- file.path(path, "SFS", "paracouWeighted3pop")
pops <- c("sp1", "globuliferaTypeParacou", "globuliferaTypeRegina")
types <- c("neutral", "functional")
lapply(types, function(type)
  lapply(pops, function(pop)
    read_tsv(paste0(prefix, ".", type, ".", pop, ".fastsim"), skip = 1) %>% 
      gather() %>% 
      mutate(type = type)
  ) %>% bind_rows()
) %>% bind_rows() %>% 
  separate(key, c("population", "AlleleCount"), convert = T) %>% 
  mutate(population = gsub("d", "", population)) %>% 
  filter(AlleleCount > 0) %>% 
  ggplot(aes(AlleleCount, value, fill = population)) +
  geom_bar(stat = "identity") +
  facet_grid(type ~ population, scales = "free") +
  scale_fill_discrete(guide = "none") +
  xlab("Allele count within population") + ylab("Number of alleles")
```

```{r sfs2pop3pop, fig.cap="Number of alleles per allele count between population."}
prefix <- file.path(path, "SFS", "paracouWeighted3pop")
pops <- c("sp1", "globuliferaTypeParacou", "globuliferaTypeRegina")
types <- c("neutral", "functional")
popPairs <- combn(pops,2)
g <- lapply(types, function(type)
  lapply(1:length(pops), function(i)
    read_tsv(paste0(prefix, ".", type, ".", popPairs[1,i], "_", popPairs[2,i], ".fastsim"), skip = 1) %>% 
      reshape2::melt(value.name = "AlleleCount")
  ) %>% bind_rows() %>% 
    separate(X1, c("population1", "pop1"), convert = T) %>% 
    separate(variable, c("population2", "pop2"), convert = T) %>% 
    mutate(population1 = gsub("d", "", population1)) %>% 
    mutate(population2 = gsub("d", "", population2)) %>% 
    filter(AlleleCount > 0) %>% 
    ggplot(aes(pop1, pop2, fill = log(AlleleCount+1))) +
    geom_tile() +
    scale_fill_gradientn(colours = rev(rainbow(10))) +
    facet_grid(population1 ~ population2) +
    theme(axis.title = element_blank()) +
    ggtitle(type))
names(g) <- types
plotly::ggplotly(g$neutral)
plotly::ggplotly(g$hitchhiker)
plotly::ggplotly(g$functional)
```
