```{r setup_popgenom, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(bayesplot)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
```

```{r trees_popgenom, eval=F}
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Symphonia") %>% 
  filter(CensusYear == 2015) %>% 
  collect()
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou3pop",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop")))
coordinates(trees) <- ~Xutm + Yutm
proj4string(trees) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
trees <- spTransform(trees, CRSobj = crs)
wetness <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "TWI_1m.tif"))
dem <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
trees$wetness <- scale(raster::extract(wetness, trees), center = F, scale = T)
trees@data %>% 
  group_by(pop) %>% 
  dplyr::select(wetness) %>% 
  summarise_all(mean)
```

# Population genomics


We filtered variants for genomics scans based on linkage desequilibrium and minor allele frequency. Then, we performed a genomic scan of populations genomics to identify genepool-specific SNPs.

* __Linkage desequilibrium & Minor alleles frequency__ variants filtering for genomics scans based on linkage desequilibirum and minor allele frequency
* __Genepool-specifc SNPs__ genomic scan to identify genepool-specific SNPs with `bayescan`

## Linkage desequilibrium & Minor alleles frequency

Minor alleles could bias the number of fixed loci increasing false-positives upward in genomic scans [@Foll2008], similarly linkage desequilibrium will increase the number of significant associations thus increasing false-negative when applying false discovery rate. Consequently we eplored linkage desequilibrium distribution and decay and minor allele frequency distribution (Fig. \@ref(fig:LD)). We chose to filter variants with a minor allele frequency above 5% (at least 18 individuals have the minor allele in the population) and with linkage desequilibirum $r^2<0.99$. It reduced the dataset to 70 737 SNPs.

```{bash LDnMAF, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
plink \
  --bfile symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr \
  --r2 with-freqs \
  --ld-window 33137 \
  --indep-pairwise 1000 500 0.99 \
  --out LD
```

```{r LD, fig.cap="Linkage desequilibirum and minor allele frequency distribution in SNPs.", fig.height=10, fig.width=10}
data <- read_delim(file.path(path, "..", "variantCalling", "paracou3pop", "LD.ld"), delim = " ") %>% 
  dplyr::select(-X10) %>%
  dplyr::rename_all(funs(gsub(" ", "", .))) %>%
  mutate_all(funs(gsub(" ", "", .))) %>% 
  mutate(d = sqrt((as.numeric(BP_A) - as.numeric(BP_B))^2)) %>% 
  mutate(R2 =as.numeric(R2)) %>% 
  mutate(MAF_A = as.numeric(MAF_A))
g.ld.decay <- ggplot(data, aes(d, R2, col = MAF_A)) +
  geom_point() +
  ylab(expression(R^2)) +
  xlab("SNP distance (bp)") +
  facet_wrap(~ MAF_A > 0.05, labeller = "label_both", nrow = 2) +
  geom_smooth(method = "glm", formula = y ~ log(x), col = "red")
g.ld.hist <-  ggplot(data, aes(R2, fill = MAF_A > 0.05)) +
  geom_histogram(position = "dodge") +
  geom_vline(xintercept = 0.99, linetype = "dashed",
             col = "red", linewidth = 1.5) +
  xlab(expression(R^2)) + coord_flip() +
  scale_fill_discrete(guide = "none")
g.maf.hist <- ggplot(data, aes(MAF_A)) +
  geom_histogram(position = "dodge") +
  geom_vline(xintercept = 0.05, linetype = "dashed",
             col = "red", linewidth = 1.5) +
  xlab("Minor allele frequency") +
  scale_x_sqrt()
gridExtra::grid.arrange(g.maf.hist, g.ld.decay, g.ld.hist, 
                        widths = c(3, 1), heights = c(1, 4),
                        layout_matrix = rbind(c(1, 1), c(2, 3)))
```

## Genepool-specifc SNPs

We used `bayescan` on filtered SNPs, to detect outlier of genepools-specifc SNP with a q-value < 0.05 (Fig. \@ref(fig:bayescanOutliers)). We found 4 020 SNPs significantly associated to genepools or fixed SNPs representing 5.7% of all SNPs with 309 functional SNPs, 2456 hitchikers, and 1255 neutral. Hitchikers corresponded to 43 106 transcripts isoforms and functional to 2 981 with 2 587 in common. Among the assocaited transcripts isoforms, 89 were differentialy expressed in *S. globulifera* and 32 in *S. sp1* in juveniles (Tysklind et la., in prep). Corresponding ORFs matched 1 603 unique PFAM, 683 unique UniProt, 989 unique KEGG, and  1 260 unique GO terms (top 10 in Table \@ref(tab:bayescanOutliersAnnotation)).

```{bash bayescan, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/PGDSpider_2.1.1.5
module load bioinfo/BayeScan2.1
plink \
  --bfile symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr \
  --extract LD.prune.in \
  --maf 0.05 \
  --make-bed \
  --recode vcf-iid \
  --out ../../populationGenomics/bayescan/paracou3pop
java -Xmx100G -Xms90G -jar $PGD_HOME/PGDSpider2-cli.jar \
	-inputfile paracou3pop.vcf \
	-inputformat VCF \
	-outputfile paracou3pop.bayescan \
	-outputformat GESTE_BAYE_SCAN \
	-spid vcf2bayescan.spid
BayeScan -threads 10 \
	paracou3pop.bayescan \
	> paracou3pop.bayescan.output
# filter(bayescan, qval < 0.05) %>%
#   dplyr::select(snp) %>%
#   write_tsv(file.path(path, "bayescan", "bayescan.snps.list"), col_names = F)
plink  \
  --bfile paracou3pop \
  --allow-extra-chr \
  --extract bayescan.snps.list \
  --recode A \
  --out bayescan.snps
```

```{r bayescanConvergence, fig.cap="Population Fst convergence." }
read_delim(file.path(path, "bayescan", "paracou3pop.baye.sel"), 
                    delim = " ", skip = 1, col_names = c("iter", "LogL", paste0("Fst", 1:3))) %>% 
  mutate(LogL = as.numeric(LogL)) %>%
  dplyr::select(-iter, -LogL) %>%
  mcmc_combo()
```

```{r bayescanOutliers, fig.cap="Genotype-specific SNPs."}
bayescan <- read.table(file.path(path, "bayescan", "paracou3pop.baye_fst.txt")) %>% 
  mutate(snpNb = as.numeric(row.names(.))) %>% 
  dplyr::rename_all(funs(gsub(" ", "", .))) %>% 
  mutate(qval = ifelse(qval <= 0.0001, 0.0001, qval)) %>% 
  left_join(read_tsv(file.path(path, "bayescan", "paracou3pop.bim"), col_names = c("scf", "snp", "posM", "pos", "A1", "A2")) %>% 
  mutate(snpNb = 1:nrow(.))) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")), by = "snp")


ggplot(bayescan, aes(snpNb, -log10(qval), alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ylab(expression(-log[10]("q-value"))) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  ggtitle("4 020 significant SNPs (5.7%)", 
          "309 functional (2.5%), 2456 hitchiker (7.0%), and 1255 neutral (5.4%)")
```

```{r bayescanOutliersAnnotation}
snpHitch <- bayescan %>% 
  filter(qval < 0.05, type == "hitchiker") %>% 
  dplyr::select(snp) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snpsHitch.annotation")))
snpFunc <- bayescan %>% 
  filter(qval < 0.05, type == "functional") %>% 
  dplyr::select(snp) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snpsHitch.annotation")))
trscHitch <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("Transcript") %>% 
  filter(transcript_id %in% local(snpHitch$trsc)) %>% 
  collect()
trscFunc <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("Transcript") %>% 
  filter(transcript_id %in% local(snpFunc$trsc)) %>% 
  collect()
# summary(trscFunc$transcript_id %in% trscHitch$transcript_id)
trscOut <- unique(bind_rows(trscHitch, trscFunc)) ; rm(trscFunc, trscHitch)
deHicth <- read_delim(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "edgeR", "juv_sympho_43k_retained_genes_DecisionTest.txt"),
           delim = " ") %>% 
  dplyr::select(-X) %>% 
  dplyr::rename(trsc = genes, deg = X.0.5.glo_HT..0.5.glo_SF.0.5.sp1_HT.0.5.sp1_SF) %>%
  mutate(deg = ifelse(deg == 1, "Eglo", deg)) %>% 
  mutate(deg = ifelse(deg == -1, "Esp", deg)) %>% 
  mutate(deg = ifelse(deg == 0, "NE", deg)) %>% 
  filter(deg != "NE") %>% 
  filter(trsc %in% trscOut$transcript_id)
# deHicth %>% 
#   group_by(deg) %>% 
#   summarise(n())
annotation <- trscOut %>%
  dplyr::select(annotation) %>% 
  unique() %>% 
  separate(annotation, paste0("X", 1:16), "\t") %>% 
  dplyr::rename(ORF = X3, PFAM = X8, UniProt = X11, KEGG = X12, GO1 = X13, GO2 = X14) %>% 
  dplyr::select(ORF, PFAM, UniProt, KEGG, GO1, GO2)
# annotation %>% 
#   filter(PFAM != ".") %>% 
#   dplyr::select(PFAM) %>% 
#   unique()
# annotation %>% 
#   filter(UniProt != ".") %>% 
#   dplyr::select(UniProt) %>% 
#   unique()
# annotation %>% 
#   filter(KEGG != ".") %>% 
#   dplyr::select(KEGG) %>% 
#   unique()
# annotation %>% 
#   filter(GO2 != ".") %>% 
#   dplyr::select(GO1, GO2) %>% 
#   unique()
bind_rows(dplyr::select(annotation, GO1), 
          dplyr::select(annotation, GO2) %>% 
            dplyr::rename(GO1 = GO2)) %>% 
  filter(GO1 != ".") %>% 
  dplyr::rename(GO = GO1) %>% 
  unique() %>% 
  separate_rows(GO, sep = "`") %>% 
  group_by(GO) %>% 
  summarise(N = n()) %>% 
  arrange(desc(N)) %>% 
  top_n(10) %>% 
  separate(GO, c("ID", "Domain", "Ontology"), "\\^") %>% 
  mutate(ID = paste0("[", ID, "](https://www.ebi.ac.uk/QuickGO/term/", ID, ")")) %>% 
  kable(caption = "Top 10 of GO terms from ORFs on transcipts associated to SNPs significanlty associated to genepools.")
```

