```{r setup_envgenom, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(bayesplot)
library(raster)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
load("./symcapture_save/Env.Rdata")
load(file.path("symcapture_save", 'AnimalModels.Rdata'))
```

# Environmental genomics

We explored and selected environmental descriptors to be used in environmental genomics. We then described the three methods used to explore the link between variable environemental variation and individual SNP and kinship variation with:

* the modele Animal to explore genetic variance associated to the environemntal variable 
* linear mixed models (LMM) used to identify major effect SNPs  associated to the environemntal variable 
* bayesian sparse linear mixed models (BSLMM) used to identify the polygenic architecture  associated to the environemntal variable 

Finally we used the three methods for every selected environmental descriptors, e.g. Topographic Wetness Index (TWI), Neighbor Crowding Index (NCI) and Annual Neighbor Crowding Index (ANCR), investigating selected SNPs, associated annotation, and ontogenetic term (GO) possible enrichment.

* __Descriptors__ environmental descriptors definition and selection based on multicolinearity and eco-evolutionary meaningfulness
* __Methods__ association methods for genetic variance, major effect SNPs and polygenic signal
* __Topographic Wetness Index (TWI)__ relation to individuals kinship and SNP variation
* __Neighbor Crowding Index (NCI)__ relation to individuals kinship and SNP variation
* __Annual Neighbor Crowding Rate (ANCR)__ relation to individuals kinship and SNP variation

## Descriptors

The topographic wetness index (TWI) was selected among several abiotic descriptors as a proxy of water accumulation. Waterlogging and topography have been highlighted as crucial for forest dynamics [@ferry2010higher] and species habitat preference [@Allie2015] at the Paracou study site. TWI was derived from a 1-m resolution digital elevation model using SAGA-GIS [@Conrad2015] based on a LiDAR campaign done in 2015.

Biotic descriptors included the Neighborhood Crowding Index [NCI; @Uriarte2004] and the Annual Crowding Rate, both integrated over time. The Neighborhood Crowding Index $NCI_i$ from tree individual $i$ was calculated with following formula:

$$NCI_i = (\int_{t_0} ^{2015} \sum _{j~|~\delta_{i,j}<20m} ^{J_{i,t}} DBH_{j,t}^2 e^{-\frac14.\delta_{i,j}}.dt).\frac1{\Delta t}$$
with $DBH_j$ the diameter from neighboring tree $j$ and $\delta_{i,j}$ its distance to individual tree $i$. $NCI_i$ is computed for all neighbors at a distance $\delta_{i,j}$ inferior to maximum neighboring distance of 20 meters. The power of neighbors $DBH$ effect was set to 2 to consider neighbors surface. The decrease of neighbors $DBH$ effect with distance was set to 0.25 here to represent trees at 20 meters of the focal trees having 1% of the effect of the same tree at 0  meters. $NCI$ represents biotic asymmetric competition through tree neighborhood over time. 

The Annual Neighborhood Crowding Rate $ANCR_i$ from tree individual $i$ was calculated with following formula:

$$ANCR_i = \int_{t_0+1} ^{2015} \frac{ NCI_{i,t} - NCI_{i,t-1} }{\Delta t}  .dt$$

with the same parameters as $NCI_i$. $ANCR$ represents the dynamic of biotic asymmetric competition through tree neighborhood over time (derivate of $NCI_i$). Both NCI and ANCR were calculated among heterospecific and conspecific neighbors and with all enighbors confounded.

Both correlations and principal component analyses showed non colinear TWI, NCI and ANCR that we selected for further analyses (Fig. \@ref(fig:correlationsEnvGenom) and Fig. \@ref(fig:pca12EnvGenom)).

```{r DataEnvGenom, eval=F}
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Symphonia") %>% 
  filter(CensusYear == 2015) %>% 
  collect()
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou3pop",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop")))
# trees2 <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
#                           "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
#          col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
#   mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
#   mutate(X = gsub("P", "", Ind)) %>% 
#   separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
#   left_join(trees) %>% 
#   left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
#                      col_names = c("IID", "pop")))
treesXY <- trees
coordinates(treesXY) <- ~Xutm + Yutm
proj4string(treesXY) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
treesXY <- spTransform(treesXY, CRSobj = crs)
wetness <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "TWI_1m.tif"))
dem <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
wetness <- projectRaster(wetness, crs = crs)
trees$TWI <- raster::extract(wetness, treesXY)
rm(dem, wetness, treesXY)
cl <- parallel::makeCluster(getOption("cl.cores", 2))
parallel::clusterExport(cl, list("trees"))
NC <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Plot == local(trees$Plot[ind])) %>% 
  filter(idTree != local(trees$idTree[ind])) %>% 
  mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
  filter(dij < 20) %>% 
  mutate(con = ifelse(Genus == local(trees$Genus[ind]) && Species == local(trees$Species[ind]), 1, 0)) %>% 
  mutate(DBH = CircCorr/pi) %>% 
  collect() %>% 
  group_by(CensusYear) %>% 
  summarise(NCIcon = sum(DBH*DBH*exp(-0.25*dij)*con),
            NCIhetero = sum(DBH*DBH*exp(-0.25*dij)*(1-con))) %>% 
  ungroup() %>% 
  arrange(CensusYear) %>%
  mutate(NCI = NCIcon + NCIhetero,
         dt = CensusYear - lag(CensusYear)) %>% 
  mutate(ANCRcon = NCIcon-lag(NCIcon),
         ANCRhetero = NCIhetero-lag(NCIhetero),
         ANCR = NCI-lag(NCI)) %>% 
  summarise(idTree = local(trees$idTree[ind]),
            NCIcon = sum(NCIcon/dt, na.rm = T),
            NCIhetero = sum(NCIcon/dt, na.rm = T),
            NCI = sum(NCI/dt, na.rm = T),
            ANCRcon = sum(ANCRcon/dt, na.rm = T),
            ANCRhetero = sum(ANCRhetero/dt, na.rm = T),
            ANCR = sum(ANCR/dt, na.rm = T))})
parallel::stopCluster(cl) ; rm(cl)
NC <- bind_rows(NC)
trees <- left_join(trees, NC)
rm(NC)
save(trees, file = "./symcapture_save/Env.Rdata")
trees %>%
  dplyr::select(FID, IID, TWI) %>%
  mutate(TWI = RNOmni::rankNorm(TWI)) %>%
  write_tsv(file.path(path, "gemma", "TWI", "TWI.env"), col_names = F)
trees %>%
  dplyr::select(FID, IID, NCI) %>%
    mutate(NCI = RNOmni::rankNorm(NCI)) %>%
  write_tsv(file.path(path, "gemma", "NCI", "NCI.env"), col_names = F)
trees %>%
  dplyr::select(FID, IID, ANCR) %>%
  mutate(ANCR = RNOmni::rankNorm(ANCR)) %>%
  write_tsv(file.path(path, "gemma", "ANCR", "ANCR.env"), col_names = F)
```

```{r correlationsEnvGenom, fig.cap="Environmental descriptors correlations."}
trees %>% 
  dplyr::select(TWI, NCI, NCIcon, NCIhetero, ANCR, ANCRcon, ANCRhetero) %>% 
  na.omit() %>% 
  cor(.) %>% 
  corrplot::corrplot.mixed()
```

```{r pca12EnvGenom, fig.cap="Environmental descriptors principal component analysis."}
factoextra::fviz_pca_var(princomp(~ TWI + NCI + NCIcon + NCIhetero + ANCR + ANCRcon + ANCRhetero, trees, cor = T),
                         axes = c(1, 2), geom = c("arrow", "text"), col.var = "contrib")
```

## Methods

### Genetic variance ($V_G$) with the Animal model

We used between individual kinship and the Animal model [@Wilson2010] to estimate genetic variance associated to individuals global phenotype leaving in a given environment [see environmental association analyses with genome wide assocaition study analyses in @Rellstab2015a]. Animal model is calculated for the environemental values $y$ of the $N$ individuals with following formula:

$$y \sim \mathcal N(\mu + Zu,\sigma_E)$$
$$u \sim \mathcal{MVN_N}(0,\sigma_GK)$$
$$\sigma_P = \sigma_G + \sigma_E$$

where $N$ is the number of individuals, $y$ is the vector of environmental variable values for the $N$ individual, $\mu$ is the intercept representing the mean environmental variable, $Z$ is the loading matrix (computed with Cholesky decomposition of $K$) and $u$ is the vector of random effects corresponding to individuals kinship relations ($u$ thus represents individuals breeding values), $u$ values are centered on 0, covary with known kinship matrix $K$ with an associated genetic variance $\sigma_G$, $\sigma_E$ is the residual variance, and the sum of residual $\sigma_E$ and genetic $\sigma_G$ variances is equal to the phenotypic variance $\sigma_P$.

### Major effect SNPs with Linear Mixed Models (LMM)

We used per SNP linear mixed models implemented in `GEMMA` [Genome-wide Efficient Mixed Model Analysis, @Zhou2012a] to identify major effect SNPs. Linear mixed models are calculated for the environemental values $y$ of the $N$ individuals with following formula:

$$y \sim \mathcal N(\mu + x\beta + Zu, \sigma_E)$$
$$u \sim \mathcal{MVN_N}(0,\sigma_GK)$$
$$\sigma_P = \sigma_G + \sigma_E$$

where $N$ is the number of individuals, $y$ is the vector of environmental variable values for the $N$ individual, $\mu$ is the intercept representing the mean environmental variable, $x$ is a SNP, $\beta$ is the effect size of the SNP, $Z$ is the loading matrix (computed with Cholesky decomposition of $K$) and $u$ is the vector of random effects corresponding to individuals kinship relations ($u$ thus represents individuals breeding values), $u$ values are centered on 0, covary with known kinship matrix $K$ with an associated genetic variance $\sigma_G$, $\sigma_E$ is the residual variance, and the sum of residual $\sigma_E$ and genetic $\sigma_G$ variances is equal to the phenotypic variance $\sigma_P$. $\beta$ significance is calculated with Wald test. And p-values are corrected for multiple testing with false discovery rate, resulting in q-values. 

### Polygenic structure with Bayesian Sparse Linear Mixed Models (BSLMM)

We used Bayesian Sparse Linear Mixed Models [BSLMM, @Zhou2013] implemented in `GEMMA` to identify polygenic structure associated to individuals global phenotype leaving in a given environment [see environmental association analyses with genome wide assocaition study analyses in @Rellstab2015a]. The objective of BSLMM is to merge the hypothesis of Bayesian Variable Sparse Regression (BVSR), which assume few SNPs with large effects, with the hypothesis of Linear Mixed Models (LMM), which assume numerous small effects SNPs, because a priori knowledge of the genomic structure associated to our studied variable is not obvious. Bayesian Sparse Linear Mixed Models is calculated for the environemental values $y$ of the $N$ individuals with following formula:

$$y \sim \mathcal N (\mu + X\widetilde\beta + Zu, \sigma^2_E)$$
$$u \sim \mathcal{MVN_N}(0,\sigma_b^2 K)$$
$$\widetilde\beta_i \sim \pi \mathcal N(0,\sigma_a^2)+(1-\pi)\delta_0$$
$$\sigma^2_P = \sigma^2_a + \sigma^2_b + \sigma^2_E$$
where $N$ is the number of individuals, $y$ is the vector of environmental variable values for the $N$ individual, $\mu$ is the intercept representing the mean environmental variable, $X$ is the matrix of SNPs ($n$ individuals and $p$ markers coded as 0, 1 or 2 copies of a reference allele), $\widetilde\beta$  correspond to SNPs sparse effects, $Z$ is the loading matrix (computed with Cholesky decomposition of $K$) and $u$ is the vector of random effects corresponding to individuals kinship relations ($u$ thus represents individuals breeding values), $u$ values are centered on 0, covary with known kinship matrix $K$ with an associated genetic variance $\sigma^2_b$, $\sigma^2_a$ is the genetic variance associated to sparse effects, $\sigma^2_E$ is the residual variance, the sum of residual $\sigma^2_E$, kinship $\sigma^2_b$ and SNPs $sigma^2_a$ variances is equal to the phenotypic variance $\sigma^2_P$, $\pi$ is the proportion of non-zero $\widetilde \beta$ and $\delta_0$ denotes a point mass at zero (Dirac Delta distribution). In a nutshell:

* $\mu$ and $\sigma^2_E$ control environmental variable mean and residual variance
* $\pi$ controls the proportion of $\widetilde \beta$ values that are non-zero
* $\sigma_a$ control the expected magnitude of non-zero $\widetilde \beta$
* $\sigma_b$ controls the expected magnitude of the random effects $u$

Models inference were done with following priors:

$$\sigma^{-2}_E \sim \Gamma (k_1,k_2)$$
$$\mu|\sigma^2_E \sim \mathcal N(0, \sigma^2_E)$$
$$log(\pi) \sim \mathcal U(-log(p),log(1))$$
$\sigma_a$ and $\sigma_b$ priors specification were done with reparametrization. To this end the Propotion of Variance Explained by the specific sparse effects and random effects terms together (PVE) and the proportion of genetic variance explained by the sparse effects terms (PGE) were defined as follow:

$$PVE(\widetilde\beta,u,\tau) = \frac{V(X\widetilde\beta+u)}{V(X\widetilde\beta+u)+\tau^{-1}}$$
$$PVE(\widetilde\beta,u) = \frac{V(X\widetilde\beta)}{V(X\widetilde\beta+u)}$$
Expectations of PVE and PGE are approximated with $h$ and $rho$, defined as follow:

$$h:=\frac{p \pi s_a \sigma_a^2 + s_b \sigma_b^2}{p \pi s_a \sigma_a^2 + s_b \sigma_b^2+1}$$
$$\rho:=\frac{p \pi s_a \sigma_a^2}{p \pi s_a \sigma_a^2 + s_b \sigma_b^2}$$
where $s_a$ is the average variance of genotypes across markers, and $s_b$ is the mean of diagonal elements in $K$. $\rho$ correspond to the ratio between a models based on few SNPs with large effects ($\rho = 1$ correspond to BVSR) against a model based on many SNPs with small effects ($\rho = 0$ correspond to LMM). Both $h$ and $rho$ were defined a priori following a uniform distribution on $[0,1]$.

To facilitate computation, the model used a binary indicator $\gamma=(\gamma_1,...,\gamma_p)\in[0,1]^p$ indicating wether the corresponding $\widetilde\beta$ are non-zero. $\widetilde\beta$ can thus be written as:
$$\gamma_i \sim \mathcal B(\pi)$$
$$\widetilde\beta_\gamma \sim \mathcal{MVN_{|\gamma|}}(0,\sigma_a^2\tau^{-1}\mathbb I_{|\gamma|})$$
$$\widetilde\beta_{-\gamma}\sim\delta_0$$
where $\widetilde\beta_\gamma$ correspond to the $\widetilde\beta$ with $\gamma_i=1$; $\widetilde\beta_{-\gamma}$ correspond to the $\widetilde\beta$ with $\gamma_i=0$; and $|\gamma|$ or $n_\gamma$ correspond to the number of non-zero $\gamma$.

<style>
div.blue { background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class = "blue">

In summary, to interpret subsequent BSLMM results, we should remember that:

* PVE is the observed variable variance explained by sparse effects $\beta$ and random effects $u$
* PGE is the observed variable genetic variance explained by sparse effects alone
* $\pi$ is the porportion of the SNPs having a polygenic effect on the phenotype
* $n_\gamma$ or $|\gamma|$ is the number of non-zeros SNPs 
* $\beta\gamma$ is the effect size estimate for the additive effect

</div>

## Topographic Wetness Index (TWI)

```{bash TWIgemma, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
cat paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > paracou3pop.fam
pops=(sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  grep $pop paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > $pop.fam ; done
pops=(paracou3pop sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  
  plink --bfile ../../../variantCalling/paracou3pop/symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr --keep $pop.fam \
  --extract ../../../variantCalling/paracou3pop/LD.prune.in --maf 0.05 \
  --pheno TWI.env --out $pop --make-bed &
done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -gk 1 -o $pop.kinship ; done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k output/$pop.kinship.cXX.txt -lmm 1 -o $pop.LMM.TWI & done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k -bslmm 1 -o $pop.BSLMM.TWI & done
cat TWI.LMM.snps | cut -f1 > TWI.snps.list
plink --bfile paracou3pop --allow-extra-chr --extract TWI.snps.list --out TWI.snps --recode A
plink --bfile ../../../variantCalling/paracou/symcapture.all.biallelic.snp.filtered.nonmissing.paracou \
  --allow-extra-chr --extract TWI.snps.list --out TWI.snps.paracou --recode A
```

Topographic wetness index (TWI) is a major environmental factor of genomic variation ($\sigma_G=0.79\pm0.13$, Fig. \@ref(fig:TWIPVE)). TWI was significantly associated only to one major effect SNPs across genepools (Fig. \@ref(fig:TWIassocMLM)), with an advantage of the major allele A in wet areas (Fig. \@ref(fig:TWIoutliersHist)). The polygenic model indicated 0.1% [0-0.45] of the genome to be related to TWI corresponding to 34 [0-146] SNPs, but the model didn't identified specific SNPs in our dataset (no $\gamma > 0.5$).

The TWI-associated SNP is a putative hitchiker located at 305 bp of gene coding for Pentatricopeptide repeat-containing protein At3g22470 implicated in mitochondrial communication.

### Proportion of Variance Explained

```{r TWIPVE, fig.cap="Proportion of Variance Explained (PVE) by genetic with Topographic Wetness Index (TWI)."}
Y <- trees %>%
  # filter(pop == "globuliferaTypeParacou") %>% # to test within populations
  dplyr::select(FID, IID, TWI) %>%
  mutate(TWI = RNOmni::rankNorm(TWI)) %>%
  arrange(IID)
fam <- read_delim(file.path(path, "gemma", "Gmax", "paracou3pop.fam"), delim = " ", 
                  col_names = c("FID", "IID", "PAT", "MAT", "SEX", "PHENOTYPE"))
K <- read_tsv(file.path(path, "gemma", "Gmax", "paracou3pop.kinship.sXX.txt"), 
              col_names = fam$IID) %>% 
  mutate(IID = fam$IID) %>% 
  column_to_rownames(var = "IID") %>% 
  as.matrix()
K <- K - min(K)
# model <- stan_model(paste0("./symcapture_models/Animal.stan"))
# fitsAnimal$TWI <- sampling(model, chains = 2, data = list(J = 1, N = nrow(Y), Y = Y$TWI, K = K[Y$IID,Y$IID]), save_warmup = F)
# save(fitsAnimal, file = file.path("symcapture_save", 'AnimalModels.Rdata'))
load(file.path("symcapture_save", 'AnimalModels.Rdata'))
g.rel <- cbind(reshape2::melt(as.matrix(dist(Y)), id.vars = NULL, value.name = "TWI") %>% 
                 dplyr::select(TWI), 
               reshape2::melt(K[Y$IID,Y$IID], id.vars = NULL, value.name = "Kinship")) %>% 
  ggplot(aes(Kinship, TWI)) + geom_point(alpha = 0.1)
g.sigma <- mcmc_combo(as.array(fitsAnimal$TWI ,  pars = c("sigmaG", "sigmaE")))
# mcmc_pairs(as.array(fitsAnimal$TWI,  pars = c("sigma_G", "sigma_E")))
g.breed <- mcmc_intervals_data(as.array(fitsAnimal$TWI ,  pars = "u_tilde")) %>% 
  mutate(IID = Y$IID) %>% 
  left_join(dplyr::select(trees, IID, pop)) %>%
  ggplot(aes(x = datatools::reorder_within(IID, m, pop), 
             xend = datatools::reorder_within(IID, m, pop)),
         col = "lightgrey") +
  geom_hline(yintercept = 0, linetype = "dashed", col = "lightgrey") +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.2,) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.2) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.2) +
  facet_wrap(~ pop, labeller = label_parsed, scales = "free_x", nrow = 3) +
  ylab("Breeding value") +
  theme(axis.title.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank(), axis.text.x = element_blank())
gridExtra::grid.arrange(g.rel, g.breed, g.sigma,
                        layout_matrix = rbind(c(1,2), c(3,2)))
```

### Major effect SNPs

```{r TWIassocMLM, fig.cap="q-value of major effect SNPs associated to topographic wetness index (TWI). SNP were detected using linear mixed models with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
t <- lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop) 
  read_tsv(file.path(path, "gemma", "TWI", paste0(pop,".LMM.TWI.assoc.txt"))) %>% 
    mutate(pop = pop)) %>% 
  bind_rows() %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  mutate(qval = p.adjust(p_wald, "fdr")) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.))
filter(t , qval < 0.05) %>% 
  mutate(model = "LMM") %>% 
  filter(pop == "paracou3pop") %>% # for the moment
  dplyr::select(rs, pop, model) %>% 
  write_tsv(file.path(path, "gemma", "TWI", "TWI.LMM.snps"), col_names = F)
ggplot(t, aes(snp, -log10(qval), label = snp, alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ggrepel::geom_text_repel(data = t[t$qval < 0.05,]) +
  ylab(expression(-log[10]("q-value"))) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type") +
  facet_wrap(~ pop) 
```

### Polygenic signal

```{r BSLMMTWIConvergence, fig.cap="traceplot of BSLMM hyper-parameters chain."}
lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop) 
  read_tsv(file.path(path, "gemma", "TWI", paste0(pop,".BSLMM.TWI.hyp.txt"))) %>% 
    mutate(Population = pop)) %>% 
  bind_rows()  %>% 
  dplyr::select(Population, pve, pge, pi, n_gamma) %>% 
  dplyr::rename(PVE = pve, PGE = pge, `n[gamma]` = n_gamma) %>% 
  group_by(Population) %>% 
  mutate(iteration = 1:n()) %>% 
  filter(row_number() %% 100 == 0) %>% 
  reshape2::melt(id.vars = c("Population", "iteration"), variable.name = "parameter") %>% 
  ggplot(aes(iteration, value)) +
  geom_line() +
  facet_grid(parameter ~ Population, scales = "free", labeller = "label_parsed") +
  theme(axis.text.x = element_text(angle = 45))
```

```{r BSLMMTWIHypPar}
read_tsv(file.path(path, "gemma", "TWI", paste0("paracou3pop",".BSLMM.TWI.hyp.txt"))) %>% 
  dplyr::select(pve, pge, pi, n_gamma) %>% 
  dplyr::rename(PVE = pve, PGE = pge, `n[gamma]` = n_gamma) %>% 
  broom::tidyMCMC(conf.int = T) %>% 
  kable(caption = "BSLMM hyper-parameters distribution for the whole genus.")
```

```{r TWIAssocBSLMM, fig.cap="Effect of SNPs associated to topographic wetness index (TWI) with a polygenic model. SNP were detected using bayesian sparse linear mixed models (BSLMM) with individual kiniship as random effect. SNP significance was assessed with the $\\gamma$ parameter and highlighted here with SNPs having at least 90% chance of having an effect on the phenotype.", eval=F}
read_tsv(file.path(path, "gemma", "TWI", paste0("paracou3pop",".BSLMM.TWI.param.txt"))) %>% 
  mutate(Population = "paracou3pop") %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  ggplot(aes(snp, beta*gamma, label = snp, col = type, alpha = gamma > 0.5)) +
  geom_point() +
  scale_y_continuous(trans="S_sqrt") +
  ylab(expression(beta*gamma))
```

### Outliers

```{r TWIoutliersHist, fig.cap="Distribution of alleles along topographic wetness index for significantly associated SNPs."}
read_delim(file.path(path, "gemma", "TWI", "TWI.snps.raw"), delim = " ") %>% 
  dplyr::select(-FID, -PAT, -MAT, -SEX) %>% 
  dplyr::rename(TWI = PHENOTYPE) %>% 
  reshape2::melt(id.vars = c("IID", "TWI"), variable.name = "SNP", value.name = "Allele") %>% 
  na.omit(Allele) %>% 
  mutate(Allele = as.factor(Allele)) %>% 
  left_join(trees, by = "IID", suffix = c("", ".trees")) %>% 
  ggplot(aes(pop, TWI, fill = Allele)) +
  geom_boxplot() +
  facet_wrap(~ SNP) +
  coord_flip() +
  ylab("Topographic wetness index") +
  xlab("Population") +
  scale_fill_discrete("Number of copies\nof the major allele")
```

```{r TWIoutliersGO, eval=F}
snp <- read_delim(file.path(path, "gemma", "TWI", "TWI.snps.raw"), delim = " ") %>% 
  dplyr::select(-FID, -PAT, -MAT, -SEX) %>% 
  dplyr::rename(TWI = PHENOTYPE) %>% 
  reshape2::melt(id.vars = c("IID", "TWI"), variable.name = "SNP", value.name = "Allele") %>% 
  na.omit(Allele) %>% 
  mutate(Allele = as.factor(Allele)) %>% 
  left_join(trees, by = "IID", suffix = c("", ".trees"))

trsc <- read_tsv(file.path(path, "..", "annotation", "snpsHitch.annotation")) %>% 
  filter(snp == "Olsson_2017_scf7180005379595_snp1055")
go <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("Transcript") %>% 
  filter(transcript_id %in% local(unique(trsc$trsc))) %>% 
  dplyr::select(gene_id, transcript_id, annotation) %>% 
  dplyr::rename(gene = gene_id, trsc = transcript_id) %>% 
  collect() %>% 
  separate(annotation, paste0("X", 1:16), "\t") %>% 
  dplyr::rename(orf = X3, PFAM = X8, UniProt = X11, KEGG = X12, GO = X13)
```

## Neighbor Crowding Index (NCI)

```{bash NCIgemma, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
cat paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > paracou3pop.fam
pops=(sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  grep $pop paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > $pop.fam ; done
pops=(paracou3pop sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  
  plink --bfile ../../../variantCalling/paracou3pop/symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr --keep $pop.fam \
  --extract ../../../variantCalling/paracou3pop/LD.prune.in --maf 0.05 \
  --pheno NCI.env --out $pop --make-bed &
done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -gk 1 -o $pop.kinship ; done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k output/$pop.kinship.cXX.txt -lmm 1 -o $pop.LMM.NCI & done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k -bslmm 1 -o $pop.BSLMM.NCI & done
cat NCI.LMM.snps TWI.BSLMM.snps | cut -f1 > NCI.snps.list
plink --bfile paracou3pop --allow-extra-chr --extract NCI.snps.list --out NCI.snps --recode A
```

Neighbor crowding index (NCI) is a strong environmental factor of genomic variation ($\sigma_G=0.59\pm0.17$, Fig. \@ref(fig:NCIPVE)). NCI wasn't significantly associated to any  major effect SNPs across genepools (Fig. \@ref(fig:NCIassocMLM)). The polygenic model indicated 0.17% [0-0.61] of the genome to be related to NCI corresponding to 52 [0-192] SNPs, but the model didn't identified specific SNPs in our dataset (no $\gamma > 0.5$).

### Proportion of Variance Explained

```{r NCIPVE, fig.cap="Proportion of Variance Explained (PVE) by genetic with Neighbor Crowding Index (NCI)."}
Y <- trees %>%
  dplyr::select(FID, IID, NCI) %>%
  mutate(NCI = RNOmni::rankNorm(NCI)) %>%
  arrange(IID)
# model <- stan_model(paste0("./symcapture_models/Animal.stan"))
# fitsAnimal$NCI <- sampling(model, chains = 2, data = list(J = 1, N = nrow(Y), Y = Y$NCI, K = K[Y$IID,Y$IID]), save_warmup = F)
# save(fitsAnimal, file = file.path("symcapture_save", 'AnimalModels.Rdata'))
load(file.path("symcapture_save", 'AnimalModels.Rdata'))
g.rel <- cbind(reshape2::melt(as.matrix(dist(Y)), id.vars = NULL, value.name = "NCI") %>% 
                 dplyr::select(NCI), 
               reshape2::melt(K[Y$IID,Y$IID], id.vars = NULL, value.name = "Kinship")) %>% 
  ggplot(aes(Kinship, NCI)) + geom_point(alpha = 0.1)
g.sigma <- mcmc_combo(as.array(fitsAnimal$NCI ,  pars = c("sigmaG", "sigmaE")))
# mcmc_pairs(as.array(fitsAnimal$NCI,  pars = c("sigma_G", "sigma_E")))
g.breed <- mcmc_intervals_data(as.array(fitsAnimal$NCI ,  pars = "u_tilde")) %>% 
            mutate(IID = Y$IID) %>% 
            left_join(dplyr::select(trees, IID, pop)) %>% 
  ggplot(aes(x = datatools::reorder_within(IID, m, pop), 
             xend = datatools::reorder_within(IID, m, pop)),
         col = "lightgrey") +
  geom_hline(yintercept = 0, linetype = "dashed", col = "lightgrey") +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.2,) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.2) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.2) +
  facet_wrap(~ pop, labeller = label_parsed, scales = "free_x", nrow = 3) +
  ylab("Breeding value") +
  theme(axis.title.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank(), axis.text.x = element_blank())
gridExtra::grid.arrange(g.rel, g.breed, g.sigma,
                        layout_matrix = rbind(c(1,2), c(3,2)))
# broom::tidyMCMC(fitsAnimal$NCI, "sigmaG", conf.int = T)
```

### Major effect SNPs

```{r NCIassocMLM, fig.cap="q-value of major effect SNPs associated to neighbor competition index (NCI). SNP were detected using linear mixed models with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop) 
  read_tsv(file.path(path, "gemma", "NCI", paste0(pop,".LMM.NCI.assoc.txt"))) %>% 
    mutate(pop = pop)) %>% 
  bind_rows() %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  mutate(qval = p.adjust(p_wald, "fdr")) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  ggplot(aes(snp, -log10(qval), alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ylab(expression(-log[10]("q-value"))) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type") +
  facet_wrap(~ pop) 
```

### Polygenic signal

```{r BSLMMNCItConvergence, fig.cap="traceplot of BSLMM hyper-parameters chain."}
lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop) 
  read_tsv(file.path(path, "gemma", "NCI", paste0(pop,".BSLMM.NCI.hyp.txt"))) %>% 
    mutate(Population = pop)) %>% 
  bind_rows()  %>% 
  dplyr::select(Population, pve, pge, pi, n_gamma) %>% 
  dplyr::rename(PVE = pve, PGE = pge, `n[gamma]` = n_gamma) %>% 
  group_by(Population) %>% 
  mutate(iteration = 1:n()) %>% 
  filter(row_number() %% 100 == 0) %>% 
  reshape2::melt(id.vars = c("Population", "iteration"), variable.name = "parameter") %>% 
  ggplot(aes(iteration, value)) +
  geom_line() +
  facet_grid(parameter ~ Population, scales = "free", labeller = "label_parsed") +
  theme(axis.text.x = element_text(angle = 45))
```

```{r BSLMMNCIHypPar}
read_tsv(file.path(path, "gemma", "NCI", paste0("paracou3pop",".BSLMM.NCI.hyp.txt"))) %>% 
  dplyr::select(pve, pge, pi, n_gamma) %>% 
  dplyr::rename(PVE = pve, PGE = pge, `n[gamma]` = n_gamma) %>% 
  broom::tidyMCMC(conf.int = T) %>% 
  kable(caption = "BSLMM hyper-parameters distribution for the whole genus.")
```

```{r NCIAssocBSLMM, fig.cap="Effect of SNPs associated to neighbor crowding index (NCI) with a polygenic model. SNP were detected using bayesian sparse linear mixed models (BSLMM) with individual kiniship as random effect. SNP significance was assessed with the $\\gamma$ parameter and highlighted here with SNPs having at least 90% chance of having an effect on the phenotype.", eval=F}
read_tsv(file.path(path, "gemma", "NCI", paste0("paracou3pop",".BSLMM.NCI.param.txt"))) %>% 
  mutate(Population = "paracou3pop") %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  ggplot(aes(snp, beta*gamma, label = snp, col = type, alpha = gamma > 0.5)) +
  geom_point() +
  scale_y_continuous(trans="S_sqrt") +
  ylab(expression(beta*gamma))
```

## Annual Neighbor Crowding Rate (ANCR)

```{bash ANCRgemma, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
cat paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > paracou3pop.fam
pops=(sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  grep $pop paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > $pop.fam ; done
pops=(paracou3pop sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  
  plink --bfile ../../../variantCalling/paracou3pop/symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr --keep $pop.fam \
  --extract ../../../variantCalling/paracou3pop/LD.prune.in --maf 0.05 \
  --pheno ANCR.env --out $pop --make-bed &
done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -gk 1 -o $pop.kinship ; done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k output/$pop.kinship.cXX.txt -lmm 1 -o $pop.LMM.ANCR & done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k -bslmm 1 -o $pop.BSLMM.ANCR ; done
cat ANCR.LMM.snps ANCR.BSLMM.snps | cut -f1 > ANCR.snps.list
plink --bfile paracou3pop --allow-extra-chr --extract ANCR.snps.list --out ANCR.snps --recode A
```

Annual Neighbor Crowding Rate (ANCR) was not significantly related to genetic variation $\sigma_G=0.11\pm0.11$, Fig. \@ref(fig:ANCRPVE)). Thus unsurisingly, ANCR was associated to any major effect SNPs and din't show a polygenic signal.

### Proportion of Variance Explained

```{r ANCRPVE, fig.cap="Proportion of Variance Explained (PVE) by genetic with Annual Neighbor Crowding Rate (ANCR)."}
Y <- trees %>%
  dplyr::select(FID, IID, ANCR) %>%
  mutate(ANCR = RNOmni::rankNorm(ANCR)) %>%
  arrange(IID)
# model <- stan_model(paste0("./symcapture_models/Animal.stan"))
# fitsAnimal$ANCR <- sampling(model, chains = 2, data = list(J = 1, N = nrow(Y), Y = Y$ANCR, K = K[Y$IID,Y$IID]), save_warmup = F)
# save(fitsAnimal, file = file.path("symcapture_save", 'AnimalModels.Rdata'))
load(file.path("symcapture_save", 'AnimalModels.Rdata'))
g.rel <- cbind(reshape2::melt(as.matrix(dist(Y)), id.vars = NULL, value.name = "ANCR") %>% 
                 dplyr::select(ANCR), 
               reshape2::melt(K[Y$IID,Y$IID], id.vars = NULL, value.name = "Kinship")) %>% 
  ggplot(aes(Kinship, ANCR)) + geom_point(alpha = 0.1)
g.sigma <- mcmc_combo(as.array(fitsAnimal$ANCR ,  pars = c("sigmaG", "sigmaE")))
# mcmc_pairs(as.array(fitsAnimal$ANCR,  pars = c("sigma_G", "sigma_E")))
g.breed <- mcmc_intervals_data(as.array(fitsAnimal$ANCR ,  pars = "u_tilde")) %>% 
            mutate(IID = Y$IID) %>% 
            left_join(dplyr::select(trees, IID, pop)) %>% 
  ggplot(aes(x = datatools::reorder_within(IID, m, pop), 
             xend = datatools::reorder_within(IID, m, pop)),
         col = "lightgrey") +
  geom_hline(yintercept = 0, linetype = "dashed", col = "lightgrey") +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.2,) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.2) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.2) +
  facet_wrap(~ pop, labeller = label_parsed, scales = "free_x", nrow = 3) +
  ylab("Breeding value") +
  theme(axis.title.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank(), axis.text.x = element_blank())
gridExtra::grid.arrange(g.rel, g.breed, g.sigma,
                        layout_matrix = rbind(c(1,2), c(3,2)))
```

### Major effect SNPs

```{r ANCRassocMLM, fig.cap="q-value of major effect SNPs associated to annual neighbor crowding rate (ANCR). SNP were detected using linear mixed models with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop) 
  read_tsv(file.path(path, "gemma", "ANCR", paste0(pop,".LMM.ANCR.assoc.txt"))) %>% 
    mutate(pop = pop)) %>% 
  bind_rows() %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  mutate(qval = p.adjust(p_wald, "fdr")) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  ggplot(aes(snp, -log10(qval), alpha = qval < 0.05, col = type, label = snp)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ylab(expression(-log[10]("q-value"))) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type") +
  facet_wrap(~ pop) 
```

### Polygenic signal

```{r BSLMMANCRConvergence, fig.cap="traceplot of BSLMM hyper-parameters chain."}
lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop) 
  read_tsv(file.path(path, "gemma", "ANCR", paste0(pop,".BSLMM.ANCR.hyp.txt"))) %>% 
    mutate(Population = pop)) %>% 
  bind_rows()  %>% 
  dplyr::select(Population, pve, pge, pi, n_gamma) %>% 
  dplyr::rename(PVE = pve, PGE = pge, `n[gamma]` = n_gamma) %>% 
  group_by(Population) %>% 
  mutate(iteration = 1:n()) %>% 
  filter(row_number() %% 100 == 0) %>% 
  reshape2::melt(id.vars = c("Population", "iteration"), variable.name = "parameter") %>% 
  ggplot(aes(iteration, value)) +
  geom_line() +
  facet_grid(parameter ~ Population, scales = "free", labeller = "label_parsed") +
  theme(axis.text.x = element_text(angle = 45))
```

```{r BSLMMANCRHypPar}
read_tsv(file.path(path, "gemma", "ANCR", paste0("paracou3pop",".BSLMM.ANCR.hyp.txt"))) %>% 
  dplyr::select(pve, pge, pi, n_gamma) %>% 
  dplyr::rename(PVE = pve, PGE = pge, `n[gamma]` = n_gamma) %>% 
  broom::tidyMCMC(conf.int = T) %>% 
  kable(caption = "BSLMM hyper-parameters distribution for the whole genus.")
```

```{r ANCRssocBSLMM, fig.cap="Effect of SNPs associated to annual neighbor crowding rate (ANCR) with a polygenic model. SNP were detected using bayesian sparse linear mixed models (BSLMM) with individual kiniship as random effect. SNP significance was assessed with the $\\gamma$ parameter and highlighted here with SNPs having at least 90% chance of having an effect on the phenotype.", eval=F}
read_tsv(file.path(path, "gemma", "ANCR", paste0("paracou3pop",".BSLMM.ANCR.param.txt"))) %>% 
  mutate(Population = "paracou3pop") %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  ggplot(aes(snp, beta*gamma, label = snp, col = type, alpha = gamma > 0.5)) +
  geom_point() +
  scale_y_continuous(trans="S_sqrt") +
  ylab(expression(beta*gamma))
```

## SNP imputation

> Unused, I let the code here in case of need. But I prefer using the ca 40k SNPs without missing information instead of 70k imputed SNPs for the moment.

We used `beagle` to impute missing data in SNPs. `beagle` doesn't work with scaffolds including less than 3 SNPs, wo we removed the 1849 corresponding SNPs.

```{bash beagle, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/Beagle_v5.0
plink \
  --bfile ../../variantCalling/paracou3pop/symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr \
  --extract ../../variantCalling/paracou3pop/LD.prune.in \
  --maf 0.05 \
  --out paracou3pop.raw \
  --make-bed
# read_tsv(file.path(path, "beagle", "paracou3pop.raw.bim"), col_names = F) %>%
#   group_by(X1) %>% filter(n() < 3) %>%  ungroup() %>%  dplyr::select(X2) %>% 
#   write_tsv(file.path(path, "beagle", "snp.lowsnpscf.list"), col_names = F)
plink \
  --bfile paracou3pop.raw \
  --allow-extra-chr \
  --out paracou3pop.raw \
  --recode vcf-iid \
  --exclude snp.lowsnpscf.list
java -Xmx4g -jar $BEAGLE   gt=paracou3pop.raw.vcf out=paracou3pop.imputed nthreads=10
plink \
  --vcf paracou3pop.imputed.vcf.gz \
  --allow-extra-chr \
  --out paracou3pop.imputed \
  --make-bed
```

## Conclusion

> Results among populations, they totally differ within populations.

Whereas topographic wetness index (TWI) and neighbor crowding index (NCI) were identified as major environmental factors of genomic varition ($\sigma_G=0.79\pm0.13$ and $\sigma_G=0.59\pm0.17$ respectively), annual neighbor crowding rate (ANCR) was not significantly related to genetic variation ($\sigma_G=0.11\pm0.11$). Both TWI and ANCR were slightly polygenic with respectively 0.1% [0-0.45] and 0.17% [0-0.61] of the genome related to the environmental variable representing respectively 34 [0-146] and 52 [0-192] SNPs. Finally, only TWI was associated to one major effect SNP, being an hitchhiker SNP with an advantage in wet areas with the allele A, already highlighted as a population specific SNP (see previous chapter).
