```{r setup_envgenom, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(bayesplot)
library(raster)
library(rstan)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
load("./symcapture_save/Env.Rdata")
load(file.path("symcapture_save", 'AnimalModels.Rdata'))
```

# Environmental genomics

We explored and selected environmental descriptors to be used in environmental genomics. We then used the three methods to quantify and explore the link between variable environemental variation and individual SNP and kinship variation with:

* the modele Animal to explore genetic variance associated to the environemntal variable 
* linear mixed models (LMM) used to identify major effect SNPs  associated to the environemntal variable 
* bayesian sparse linear mixed models (BSLMM) used to identify the polygenic architecture  associated to the environemntal variable 

## Descriptors

The topographic wetness index (TWI) was selected among several abiotic descriptors as a proxy of water accumulation. Waterlogging and topography have been highlighted as crucial for forest dynamics [@ferry2010higher] and species habitat preference [@Allie2015] at the Paracou study site. TWI was derived from a 1-m resolution digital elevation model using SAGA-GIS [@Conrad2015] based on a LiDAR campaign done in 2015.

Biotic descriptors included the Neighborhood Crowding Index [NCI; @Uriarte2004] and the Annual Crowding Rate, both integrated over time. The Neighborhood Crowding Index $NCI_i$ from tree individual $i$ was calculated with following formula:

$$NCI_i = (\int_{t_0} ^{2015} \sum _{j~|~\delta_{i,j}<20m} ^{J_{i,t}} DBH_{j,t}^2 e^{-\frac14.\delta_{i,j}}.dt).\frac1{\Delta t}$$
with $DBH_j$ the diameter from neighboring tree $j$ and $\delta_{i,j}$ its distance to individual tree $i$. $NCI_i$ is computed for all neighbors at a distance $\delta_{i,j}$ inferior to maximum neighboring distance of 20 meters. The power of neighbors $DBH$ effect was set to 2 to consider neighbors surface. The decrease of neighbors $DBH$ effect with distance was set to 0.25 here to represent trees at 20 meters of the focal trees having 1% of the effect of the same tree at 0  meters. $NCI$ represents biotic asymmetric competition through tree neighborhood over time. Both correlations and principal component analyses showed non colinear TWI and NCI that we selected for further analyses (Fig. \@ref(fig:correlationsEnvGenom)).

```{r DataEnvGenom, eval=F}
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Symphonia") %>% 
  filter(CensusYear == 2015) %>% 
  collect()
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou3pop",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop")))
# trees2 <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
#                           "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
#          col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
#   mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
#   mutate(X = gsub("P", "", Ind)) %>% 
#   separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
#   left_join(trees) %>% 
#   left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
#                      col_names = c("IID", "pop")))
treesXY <- trees
coordinates(treesXY) <- ~Xutm + Yutm
proj4string(treesXY) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
treesXY <- spTransform(treesXY, CRSobj = crs)
wetness <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "TWI_1m.tif"))
dem <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
wetness <- projectRaster(wetness, crs = crs)
trees$TWI <- raster::extract(wetness, treesXY)
rm(dem, wetness, treesXY)
cl <- parallel::makeCluster(getOption("cl.cores", 2))
parallel::clusterExport(cl, list("trees"))
NC <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Plot == local(trees$Plot[ind])) %>% 
  filter(idTree != local(trees$idTree[ind])) %>% 
  mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
  filter(dij < 20) %>% 
  mutate(con = ifelse(Genus == local(trees$Genus[ind]) && Species == local(trees$Species[ind]), 1, 0)) %>% 
  mutate(DBH = CircCorr/pi) %>% 
  collect() %>% 
  group_by(CensusYear) %>% 
  summarise(NCIcon = sum(DBH*DBH*exp(-0.25*dij)*con),
            NCIhetero = sum(DBH*DBH*exp(-0.25*dij)*(1-con))) %>% 
  ungroup() %>% 
  arrange(CensusYear) %>%
  mutate(NCI = NCIcon + NCIhetero,
         dt = CensusYear - lag(CensusYear)) %>% 
  mutate(ANCRcon = NCIcon-lag(NCIcon),
         ANCRhetero = NCIhetero-lag(NCIhetero),
         ANCR = NCI-lag(NCI)) %>% 
  summarise(idTree = local(trees$idTree[ind]),
            NCIcon = sum(NCIcon/dt, na.rm = T),
            NCIhetero = sum(NCIcon/dt, na.rm = T),
            NCI = sum(NCI/dt, na.rm = T),
            ANCRcon = sum(ANCRcon/dt, na.rm = T),
            ANCRhetero = sum(ANCRhetero/dt, na.rm = T),
            ANCR = sum(ANCR/dt, na.rm = T))})
parallel::stopCluster(cl) ; rm(cl)
NC <- bind_rows(NC)
trees <- left_join(trees, NC)
rm(NC)
save(trees, file = "./symcapture_save/Env.Rdata")
trees %>%
  dplyr::select(FID, IID, TWI) %>%
  mutate(TWI = RNOmni::rankNorm(TWI)) %>%
  write_tsv(file.path(path, "gemma", "TWI", "TWI.env"), col_names = F)
trees %>%
  dplyr::select(FID, IID, NCI) %>%
    mutate(NCI = RNOmni::rankNorm(NCI)) %>%
  write_tsv(file.path(path, "gemma", "NCI", "NCI.env"), col_names = F)
```

```{r correlationsEnvGenom, fig.cap="Environmental descriptors correlations."}
trees %>% 
  dplyr::select(TWI, NCI, NCIcon, NCIhetero, ANCR, ANCRcon, ANCRhetero) %>% 
  na.omit() %>% 
  cor(.) %>% 
  corrplot::corrplot.mixed()
```

```{r Envmdata}
ids <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king.id"))
K <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king"),
         col_names = ids$IID) %>% 
  as.data.frame()
row.names(K) <- ids$IID
K <- as.matrix(K)
trees <- trees %>% 
  filter(IID %in% colnames(K)) %>% 
  mutate(popNum = as.numeric(as.factor(pop))) 
mdata <- function(data, variable) {
  K <- K[trees$IID, trees$IID]
  K[K < 0] <- 0
  K <- K*2
  K <- as.matrix(Matrix::nearPD(K)$mat)
  list(N = nrow(data),
       P = length(unique(data$pop)),
       Y = unlist(data[variable])/sd(unlist(data[variable])),
       population = data$popNum,
       K = K) 
}
```

## Genetic variance

We used between individual kinship and a lognormal Animal model [@Wilson2010] to estimate genetic variance associated to individuals global phenotype leaving in a given environment [see environmental association analyses with genome wide assocaition study analyses in @Rellstab2015a]. Animal model is calculated for the environemental values $y$ of the $N$ individuals with following formula:


$$y \sim \mathcal{logN}(a,\sigma_R) \\ a \sim \mathcal{MVlogN_N}(\mu_{Population},\sigma_GK)$$

where individual is defined as a normal law centered on the individual genetic additive effects $a$ and associated individual remaining variance $\sigma_R$. Additive genetic variance $a$ follow a multivariate lognormal law centered on the population mean $\mu_{Population}$ of covariance $\sigma_G K$.

For both topography (TWI) and neighboring (NCI), most of the variability was not explained either by population or genotype with 41% and 96% of remaining variability respectivelly for TWI and NCI. Thus most of individual environment is determined through plastic response. Nevertheless the variability for topography was also strongly explained by the population (41%) and 18% of the remaining variability was explained by the genotype (Tab. \@ref(tab:EnvAnimalLog) and Fig. \@ref(fig:envKinshipLogG1) and \@ref(fig:envKinshipLogG2)).

```{r EnvAnimalLog, fig.cap="Genetic variance modelling for environmental variables."}
# animal <- stan_model("./symcapture_models/AnimalLog.stan")
# fitEnvKin <- lapply(c("TWI", "NCI"), function(var)
#   sampling(animal, chains = 2, data = mdata(trees, var), save_warmup = F,
#            control = list(adapt_delta = 0.99)))
# names(fitEnvKin) <- c("TWI", "NCI")
# save(fitEnvKin, file = file.path("symcapture_save", 'envkinLog.Rdata'))
load(file.path("symcapture_save", 'envkinLog.Rdata'))
lapply(fitEnvKin, broom::tidyMCMC, c("mu", "Vp", "Vg", "Vr"), 
       droppars = NULL, rhat = T) %>% 
  bind_rows(.id = "Variable") %>% 
  separate(term, c("parameter", "population"), "[:punct:]", convert = T) %>% 
  mutate(population = recode(population, "1" = "S.globuliferaParacou", "2" = "S.globuliferaRegina", "3" = "S.sp1")) %>% 
  mutate(population = ifelse(is.na(population), "", population)) %>% 
  kable(caption = "Summary table of the kinship growth model",
        col.names = c("Variable", "Parameter",  "Population", "Estimate", "$\\sigma$", "$\\hat{R}$"))
```

```{r envKinshipLogG1, fig.cap="Marginal and conditional $R^2$ for environmental variables."}
lapply(fitEnvKin, mcmc_intervals_data, regex_pars = c("R2")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(index = recode(parameter, 
                        "R2m" = "Marginal", "R2c" = "Conditional")) %>%
  ggplot(aes(x = index, xend = index, col = variable, fill = variable)) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  facet_wrap(~ variable, labeller = label_parsed) +
  xlab("") + ylab(expression(R^2)) +
  scale_fill_discrete(guide = "none") +
  scale_color_discrete(guide = "none")
```

```{r envKinshipLogG2, fig.cap="Genetic variance partitionning for environmental variables."}
lapply(fitEnvKin, mcmc_intervals_data, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(parameter = recode(parameter, "Vp" = "Population", "Vg" = "Genotype", "Vr" = "Residual")) %>% 
  group_by(variable) %>%
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = variable, fill = parameter)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  facet_wrap(~ variable, scales = "free") +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2)) 
```

## Major effect SNPs

We used per SNP linear mixed models implemented in `GEMMA` [Genome-wide Efficient Mixed Model Analysis, @Zhou2012a] to identify major effect SNPs. Linear mixed models are calculated for the environemental values $y$ of the $N$ individuals with following formula:

$$y \sim \mathcal N(\mu + x\beta + Zu, \sigma_E)$$
$$u \sim \mathcal{MVN_N}(0,\sigma_GK)$$
$$\sigma_P = \sigma_G + \sigma_E$$

where $N$ is the number of individuals, $y$ is the vector of environmental variable values for the $N$ individual, $\mu$ is the intercept representing the mean environmental variable, $x$ is a SNP, $\beta$ is the effect size of the SNP, $Z$ is the loading matrix (computed with Cholesky decomposition of $K$) and $u$ is the vector of random effects corresponding to individuals kinship relations ($u$ thus represents individuals breeding values), $u$ values are centered on 0, covary with known kinship matrix $K$ with an associated genetic variance $\sigma_G$, $\sigma_E$ is the residual variance, and the sum of residual $\sigma_E$ and genetic $\sigma_G$ variances is equal to the phenotypic variance $\sigma_P$. $\beta$ significance is calculated with Wald test. And p-values are corrected for multiple testing with false discovery rate, resulting in q-values. 

We detected 3 outlier SNPs for ANCR (2 hitchhikers and 1 anonymous) and 15 for TWI (3 genic, 10 hitchhikers and 2 anonymous, 
Fig. \@ref(fig:envMLM)).

```{bash EnvGemma, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
cat paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > paracou3pop.fam
for par in $(seq 1 3) ; do plink \
  --bfile ../../../variantCalling/paracou3pop/symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr --keep paracou3pop.fam \
  --extract ../../../variantCalling/paracou3pop/LD.prune.in \
  --maf 0.05 --pheno env.phenotype --mpheno $par --out paracou3pop.$par --make-bed ; done
gemma -bfile paracou3pop.1 -gk 1 -o kinship
for par in $(seq 1 3) ; do  gemma -bfile paracou3pop.$par -k output/kinship.cXX.txt -lmm 1 -o LMM.$par & done
for par in $(seq 1 3) ; do  gemma -bfile paracou3pop.$par -k -bslmm 1  -o BSLMM.$par & done
cat LMM.snps | cut -f1 > snps.list
plink --bfile paracou3pop.1 --allow-extra-chr --extract snps.list --out outliers --recode A
```

```{r envMLM, fig.cap="q-value of major effect SNPs associated to environmental variables. SNP were detected using linear mixed models (LMM) with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
lmm <- lapply(list.files(file.path(path, "gemma", "env"), pattern = ".assoc.txt$", full.names = T), read_tsv)%>% 
  bind_rows(.id = "parameter") %>% 
  mutate(parameter = recode(parameter, "1" = "TWI",  "2" = "NCI",  "3" = "ANCR")) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  mutate(qval = p.adjust(p_wald, "fdr")) %>%
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  na.omit(qval) %>% 
  mutate(model = "LMM")
filter(lmm , qval < 0.05) %>% 
  dplyr::select(rs, parameter) %>% 
  write_tsv(file.path(path, "gemma", "env", "LMM.snps"), col_names = F)
ggplot(lmm, aes(snp, -log10(qval), label = snp, alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ggrepel::geom_text_repel(data = lmm[lmm$qval < 0.05,]) +
  ylab(expression(-log[10]("q-value"))) +
  facet_wrap(~ parameter) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type")
```

## Polygenic structure

We used Bayesian Sparse Linear Mixed Models [BSLMM, @Zhou2013] implemented in `GEMMA` to identify polygenic structure associated to individuals global phenotype leaving in a given environment [see environmental association analyses with genome wide assocaition study analyses in @Rellstab2015a]. The objective of BSLMM is to merge the hypothesis of Bayesian Variable Sparse Regression (BVSR), which assume few SNPs with large effects, with the hypothesis of Linear Mixed Models (LMM), which assume numerous small effects SNPs, because a priori knowledge of the genomic structure associated to our studied variable is not obvious. Bayesian Sparse Linear Mixed Models is calculated for the environemental values $y$ of the $N$ individuals with following formula:

$$y \sim \mathcal N (\mu + X\widetilde\beta + Zu, \sigma^2_E)$$
$$u \sim \mathcal{MVN_N}(0,\sigma_b^2 K)$$
$$\widetilde\beta_i \sim \pi \mathcal N(0,\sigma_a^2)+(1-\pi)\delta_0$$
$$\sigma^2_P = \sigma^2_a + \sigma^2_b + \sigma^2_E$$
where $N$ is the number of individuals, $y$ is the vector of environmental variable values for the $N$ individual, $\mu$ is the intercept representing the mean environmental variable, $X$ is the matrix of SNPs ($n$ individuals and $p$ markers coded as 0, 1 or 2 copies of a reference allele), $\widetilde\beta$  correspond to SNPs sparse effects, $Z$ is the loading matrix (computed with Cholesky decomposition of $K$) and $u$ is the vector of random effects corresponding to individuals kinship relations ($u$ thus represents individuals breeding values), $u$ values are centered on 0, covary with known kinship matrix $K$ with an associated genetic variance $\sigma^2_b$, $\sigma^2_a$ is the genetic variance associated to sparse effects, $\sigma^2_E$ is the residual variance, the sum of residual $\sigma^2_E$, kinship $\sigma^2_b$ and SNPs $sigma^2_a$ variances is equal to the phenotypic variance $\sigma^2_P$, $\pi$ is the proportion of non-zero $\widetilde \beta$ and $\delta_0$ denotes a point mass at zero (Dirac Delta distribution). In a nutshell:

* $\mu$ and $\sigma^2_E$ control environmental variable mean and residual variance
* $\pi$ controls the proportion of $\widetilde \beta$ values that are non-zero
* $\sigma_a$ control the expected magnitude of non-zero $\widetilde \beta$
* $\sigma_b$ controls the expected magnitude of the random effects $u$

Models inference were done with following priors:

$$\sigma^{-2}_E \sim \Gamma (k_1,k_2)$$
$$\mu|\sigma^2_E \sim \mathcal N(0, \sigma^2_E)$$
$$log(\pi) \sim \mathcal U(-log(p),log(1))$$
$\sigma_a$ and $\sigma_b$ priors specification were done with reparametrization. To this end the Propotion of Variance Explained by the specific sparse effects and random effects terms together (PVE) and the proportion of genetic variance explained by the sparse effects terms (PGE) were defined as follow:

$$PVE(\widetilde\beta,u,\tau) = \frac{V(X\widetilde\beta+u)}{V(X\widetilde\beta+u)+\tau^{-1}}$$
$$PVE(\widetilde\beta,u) = \frac{V(X\widetilde\beta)}{V(X\widetilde\beta+u)}$$
Expectations of PVE and PGE are approximated with $h$ and $rho$, defined as follow:

$$h:=\frac{p \pi s_a \sigma_a^2 + s_b \sigma_b^2}{p \pi s_a \sigma_a^2 + s_b \sigma_b^2+1}$$
$$\rho:=\frac{p \pi s_a \sigma_a^2}{p \pi s_a \sigma_a^2 + s_b \sigma_b^2}$$
where $s_a$ is the average variance of genotypes across markers, and $s_b$ is the mean of diagonal elements in $K$. $\rho$ correspond to the ratio between a models based on few SNPs with large effects ($\rho = 1$ correspond to BVSR) against a model based on many SNPs with small effects ($\rho = 0$ correspond to LMM). Both $h$ and $rho$ were defined a priori following a uniform distribution on $[0,1]$.

To facilitate computation, the model used a binary indicator $\gamma=(\gamma_1,...,\gamma_p)\in[0,1]^p$ indicating wether the corresponding $\widetilde\beta$ are non-zero. $\widetilde\beta$ can thus be written as:
$$\gamma_i \sim \mathcal B(\pi)$$
$$\widetilde\beta_\gamma \sim \mathcal{MVN_{|\gamma|}}(0,\sigma_a^2\tau^{-1}\mathbb I_{|\gamma|})$$
$$\widetilde\beta_{-\gamma}\sim\delta_0$$
where $\widetilde\beta_\gamma$ correspond to the $\widetilde\beta$ with $\gamma_i=1$; $\widetilde\beta_{-\gamma}$ correspond to the $\widetilde\beta$ with $\gamma_i=0$; and $|\gamma|$ or $n_\gamma$ correspond to the number of non-zero $\gamma$.

<style>
div.blue { background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class = "blue">

In summary, to interpret subsequent BSLMM results, we should remember that:

* PVE is the observed variable variance explained by sparse effects $\beta$ and random effects $u$
* PGE is the observed variable genetic variance explained by sparse effects alone
* $\pi$ is the porportion of the SNPs having a polygenic effect on the phenotype
* $n_\gamma$ or $|\gamma|$ is the number of non-zeros SNPs 
* $\beta\gamma$ is the effect size estimate for the additive effect

</div>

We detected only 1 anonymous outlier SNP for NCI (Fig. \@ref(fig:envBSLMM)). Moreover, polygenic modelling with bayesian sparse linear mixed models (BSLMM) didn’t converged due to a small number of individuals (Fig. \@ref(fig:envBSLMMConvergence)).

```{r envBSLMMConvergence, fig.cap="Traceplot of hyper-parameters chain for Baysian Sparse Linear Mixed Model (BSLMM)."}
cowplot::plot_grid(plotlist = lapply(list.files(file.path(path, "gemma", "growth"), pattern = ".hyp.txt$", full.names = T), 
       function(file) read_tsv(file) %>% 
         dplyr::select(pge, n_gamma) %>% 
         dplyr::rename(PGE = pge, `n[gamma]` = n_gamma) %>% 
         mutate(iteration = 1:n()) %>% 
         filter(row_number() %% 100 == 0) %>% 
         dplyr::select(-iteration) %>% 
         mcmc_combo()), labels = c("TWI", "NCI", "ANCR"), nrow = 3)  
```

```{r envBSLMM, fig.cap="Effect of SNPs associated to environmental variables with a polygenic model. SNP were detected using bayesian sparse linear mixed models (BSLMM) with individual kiniship as random effect. SNP significance was assessed with the $\\gamma$ parameter and highlighted here with SNPs having at least 90% chance of having an effect on the phenotype."}
bslmm <- lapply(list.files(file.path(path, "gemma", "growth"), pattern = ".param.txt$", full.names = T), read_tsv) %>% 
  bind_rows(.id = "parameter") %>% 
  mutate(parameter = recode(parameter, "1" = "TWI",  "2" = "NCI",  "3" = "ANCR")) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  mutate(model = "BSLMM")
filter(bslmm , gamma > 0.5) %>% 
  dplyr::select(rs, parameter) %>% 
  write_tsv(file.path(path, "gemma", "env", "BSLMM.snps"), col_names = F)
ggplot(bslmm, aes(snp, beta*gamma, label = snp, col = type, alpha = gamma > 0.5)) +
  geom_point() +
  ggrepel::geom_text_repel(data = bslmm[bslmm$gamma > 0.5,]) +
  scale_y_continuous(trans="S_sqrt") +
  facet_wrap(~ parameter) +
  ylab(expression(beta*gamma))
```

## Outliers

Only one gene associated to outliers SNPs had a significant enrichment of a gene ontology ($qval=0.03244626$), the hitchhiker SNP of ANCR was significantly enriched in a gene implicated in the regulation of flower development (Fig. \@ref(fig:envGO)).

```{r envGO, fig.cap="Term enrichment of Gene Ontology (GO) for outliers genes."}
trsc <- read_tsv(file.path(path, "..", "annotation", "snpsFunc.annotation")) %>% 
              dplyr::select(scf, snp, trsc) %>% 
  bind_rows(read_tsv(file.path(path, "..", "annotation", "snpsHitch.annotation"))) %>% # inclusion to be discussed
  unique() %>% 
  inner_join(dplyr::rename(lmm, snpNum = snp, snp = rs), .)

GO <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("Transcript") %>% 
  filter(transcript_id %in% local(unique(trsc$trsc))) %>% 
  dplyr::select(gene_id, transcript_id, annotation) %>% 
  dplyr::rename(gene = gene_id, trsc = transcript_id) %>% 
  collect() %>% 
  separate(annotation, paste0("X", 1:16), "\t") %>% 
  dplyr::rename(orf = X3, PFAM = X8, UniProt = X11, KEGG = X12, GO = X13) %>% 
  dplyr::select(trsc, gene, orf, GO) %>% 
  filter(GO != ".") %>% 
  left_join(trsc) %>% 
  group_by(model, parameter, gene, GO) %>% 
  summarise(qval = min(qval)) %>% 
  ungroup() %>% 
  dplyr::select(model, parameter, gene, qval, GO) %>% 
  separate_rows(GO, sep = "`") %>% 
  separate(GO, c("GOid", "GOtype", "GOname"), "\\^")

enrichment <- function(x) {
  y <- clusterProfiler::enricher(unique(filter(x, qval < 0.05)$gene),
                                            pvalueCutoff = 0.2, 
                                            pAdjustMethod = "BH",
                                            unique(x$gene),
                                            minGSSize = 10, 
                                            maxGSSize = 500, 
                                            qvalueCutoff = 0.2, 
                                            dplyr::select(x, GOid, gene),
                                            TERM2NAME = dplyr::select(x, GOid, GOname))
  if(is.null(y)){
    data.frame()
  } else {
    y@result
  }
}

group_by(GO, model, parameter) %>% 
  do(enrichment(.)) %>% 
  filter(qvalue < 0.8) %>% 
  ggplot(aes(reorder(Description, qvalue), Count, fill = -log(qvalue))) +
  geom_bar(stat = "identity") +
  coord_flip() +
  facet_wrap(~ model + parameter, labeller = "label_both", nrow = 2, scales = "free") +
  viridis::scale_fill_viridis() +
  theme(axis.title = element_blank())
```
