```{r setup_envgenom, include=FALSE, eval=T}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(bayesplot)
library(raster)
library(rstan)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F, eval=T)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
load("./symcapture_save/Env.Rdata") 
load(file.path("symcapture_save", 'AnimalModels.Rdata'))  
```

# Environmental genomics

We explored and selected environmental descriptors to be used in environmental genomics. We then used the three methods to quantify and explore the link between variable environemental variation and individual SNP and kinship variation with:

* the modele Animal to explore genetic variance associated to the environemntal variable 
* linear mixed models (LMM) used to identify major effect SNPs  associated to the environemntal variable 

## Descriptors

The topographic wetness index (TWI) was selected among several abiotic descriptors as a proxy of water accumulation. Waterlogging and topography have been highlighted as crucial for forest dynamics [@ferry2010higher] and species habitat preference [@Allie2015] at the Paracou study site. TWI was derived from a 1-m resolution digital elevation model using SAGA-GIS [@Conrad2015] based on a LiDAR campaign done in 2015.

Biotic descriptors included the Neighborhood Crowding Index [NCI; @Uriarte2004] and the Annual Crowding Rate, both integrated over time. The Neighborhood Crowding Index $NCI_i$ from tree individual $i$ was calculated with following formula:

\begin{equation} 
  NCI_i = (\int_{t_0} ^{2015} \sum _{j~|~\delta_{i,j}<20m} ^{J_{i,t}} DBH_{j,t}^2 e^{-\frac14.\delta_{i,j}}.dt).\frac1{\Delta t}
  (\#eq:ncienv)
\end{equation} 

with $DBH_j$ the diameter from neighboring tree $j$ and $\delta_{i,j}$ its distance to individual tree $i$. $NCI_i$ is computed for all neighbors at a distance $\delta_{i,j}$ inferior to maximum neighboring distance of 20 meters. The power of neighbors $DBH$ effect was set to 2 to consider neighbors surface. The decrease of neighbors $DBH$ effect with distance was set to 0.25 here to represent trees at 20 meters of the focal trees having 1% of the effect of the same tree at 0  meters. $NCI$ represents biotic asymmetric competition through tree neighborhood over time. Both correlations and principal component analyses showed non colinear TWI and NCI that we selected for further analyses (Fig. \@ref(fig:correlationsEnvGenom)).

```{r DataEnvGenom, eval=F}
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Symphonia") %>% 
  filter(CensusYear == 2015) %>% 
  collect()
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou3pop",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop")))
# trees2 <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
#                           "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
#          col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
#   mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
#   mutate(X = gsub("P", "", Ind)) %>% 
#   separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
#   left_join(trees) %>% 
#   left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
#                      col_names = c("IID", "pop")))
treesXY <- trees
coordinates(treesXY) <- ~Xutm + Yutm
proj4string(treesXY) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
treesXY <- spTransform(treesXY, CRSobj = crs)
wetness <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "TWI_1m.tif"))
dem <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
wetness <- projectRaster(wetness, crs = crs)
trees$TWI <- raster::extract(wetness, treesXY)
rm(dem, wetness, treesXY)
cl <- parallel::makeCluster(getOption("cl.cores", 4))
parallel::clusterExport(cl, list("trees"))
NC <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
    tbl("Paracou") %>% 
    filter(Plot == local(trees$Plot[ind])) %>% 
    filter(idTree != local(trees$idTree[ind])) %>% 
    mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
    filter(dij < 20) %>% 
    mutate(con = ifelse(Genus == local(trees$Genus[ind]) && Species == local(trees$Species[ind]), 1, 0)) %>% 
    mutate(DBH = CircCorr/pi) %>% 
    collect() %>% 
    group_by(CensusYear) %>% 
    summarise(NCI = sum(DBH*DBH*exp(-0.25*dij))) %>% 
    ungroup() %>% 
    summarise(idTree = local(trees$idTree[ind]),
              NCI = mean(NCI))})
parallel::stopCluster(cl) ; rm(cl)
NC <- bind_rows(NC)
trees <- left_join(trees, NC)
rm(NC)
save(trees, file = "./symcapture_save/Env.Rdata")
trees %>%
  dplyr::select(FID, IID, TWI, NCI) %>%
  mutate(TWI = RNOmni::rankNorm(TWI)) %>%
  mutate(NCI = RNOmni::rankNorm(NCI)) %>%
  write_tsv(file.path(path, "gemma", "env", "env.phenotype"), col_names = F)
```

```{r NCI, eval=F}
cl <- parallel::makeCluster(getOption("cl.cores", 4))
parallel::clusterExport(cl, list("trees"))
NC <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
    tbl("Paracou") %>% 
    filter(Plot == local(trees$Plot[ind])) %>% 
    filter(idTree != local(trees$idTree[ind])) %>% 
    mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
    filter(dij < 20) %>% 
    mutate(DBH = CircCorr/pi) %>% 
    group_by(CensusYear) %>% 
    summarise(idTree = local(trees$idTree[ind]),
              NCI = sum(DBH*DBH*exp(-0.25*dij))) %>% 
    ungroup() %>% 
    collect()
})
parallel::stopCluster(cl) ; rm(cl)
NC <- bind_rows(NC)
NC %>% 
  group_by(idTree) %>% 
  arrange(CensusYear) %>% 
  summarise(NCI0 = first(NCI), NCI2017 = last(NCI)) %>% 
  left_join(trees) %>% 
  mutate(DeltaNCI = NCI2017 - NCI0) %>% 
  # filter(DeltaNCI < 0)
  ggplot(aes(DeltaNCI)) + 
  geom_histogram() +
  geom_vline(xintercept = 1, col = "red", linetype = "dashed") +
  ggtitle("Symphonia", "N = 124/372 (33.3%)")
```

```{r NCspCorr, fig.cap="Temporal autocorrelation for NCI."}
# cl <- parallel::makeCluster(getOption("cl.cores", 4))
# parallel::clusterExport(cl, list("trees"))
# NC <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
#   library(tidyverse)
#   src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
#     tbl("Paracou") %>% 
#     filter(Plot == local(trees$Plot[ind])) %>% 
#     filter(idTree != local(trees$idTree[ind])) %>% 
#     mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
#     filter(dij < 20) %>% 
#     mutate(DBH = CircCorr/pi) %>% 
#     collect() %>% 
#     group_by(CensusYear) %>% 
#     summarise(idTree = local(trees$idTree[ind]),
#               NCI = sum(DBH*DBH*exp(-0.25*dij))) %>% 
#     ungroup()})
# parallel::stopCluster(cl) ; rm(cl)
# NC <- bind_rows(NC)
# save(NC, file = "./symcapture_save/NC.Rdata")
load(file = "./symcapture_save/NC.Rdata")
t <- NC %>%
  filter(idTree %in% filter(trees, Plot %in% 1:12)$idTree) %>% 
  reshape2::dcast(CensusYear ~ idTree, value.var = "NCI") %>% 
    reshape2::melt(id.var = "CensusYear", variable.name =  "idTree", value.name = "NCI") %>% 
  group_by(idTree) %>% 
  arrange(CensusYear) %>% 
  mutate(NCI = ifelse(is.na(NCI), (lag(NCI)+lead(NCI))/2, NCI)) %>% 
  # filter(is.na(NCI)) %>% 
  reshape2::dcast(CensusYear ~ idTree, value.var = "NCI") %>% 
  column_to_rownames("CensusYear") %>% 
  as.matrix()
sapply(1:nrow(t)-1, function(n) 
  data.frame(lag = n, r = unlist(sapply((1+n):nrow(t), function(i) cor(t[i,], t[i-n,]), simplify = F))), simplify = F) %>% 
  bind_rows() %>% 
  ggplot(aes(x = lag, y = r)) + geom_boxplot(aes(group = lag)) + geom_smooth() +
  xlab("Lag (year)") + ylab("Correlation between NCI(t) & NCI(t-lag)") +
  geom_vline(xintercept = c(5, 10), col = "lightgrey", linetype = "dashed")
```

```{r Envmdata}
ids <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king.id"))
K <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king"),
         col_names = ids$IID) %>% 
  as.data.frame()
row.names(K) <- ids$IID
K <- as.matrix(K)
trees <- trees %>% 
  filter(IID %in% colnames(K)) %>% 
  mutate(IndNum = 1:nrow(.), popNum = as.numeric(as.factor(pop)))
mdata <- lapply(c("TWI", "NCI"), function(variable) {
                K <- K[trees$IID, trees$IID]
                K[K < 0] <- 0
                K <- K*2
                K <- as.matrix(Matrix::nearPD(K)$mat)
                return(list(N = nrow(trees),
                            P = length(unique(trees$pop)),
                            y = as.vector(scale(trees[variable], center = F)),
                                          population = trees$popNum,
                                          K = K))})
names(mdata) <- c("TWI", "NCI")
save(mdata, file = file.path("symcapture_save", 'dataEnv.Rdata'))   
```

```{r, eval=F}
# quick idea not working
mutate(trees, He = as.numeric(unlist(read_delim(file.path(path, "..", "variantCalling", "growth", "growth.het"), delim = " ")["           F"]))) %>% 
  ggplot(aes(NCI, He)) + geom_point() + geom_smooth() +
  ggpubr::stat_cor(method = "pearson", label.x = 6000, label.y = -0.5)
```

## Genetic variance

We used between individual kinship and a lognormal Animal model [@Wilson2010] to estimate genetic variance associated to individuals global phenotype leaving in a given environment [see environmental association analyses with genome wide assocaition study analyses in @Rellstab2015a]. Animal model is calculated for the environemental values $y$ of the $N$ individuals with following formula:

\begin{equation} 
  y_{p,i} \sim \mathcal{logN}(log(a_{p,i}),\sigma_1) \\
  a_{p,i} \sim \mathcal{MVlogN_N}(log(\mu_p),\sigma_2.K)
  (\#eq:animalenv)
\end{equation} 

where individual is defined as a normal law centered on the individual genetic additive effects $a$ and associated individual remaining variance $\sigma_R$. Additive genetic variance $a$ follow a multivariate lognormal law centered on the population mean $\mu_{Population}$ of covariance $\sigma_G K$.

We fitted the equivalent model with following priors:

\begin{equation} 
  y_{p,i} \sim \mathcal{logN}(log(\mu_p) + \hat{\sigma_2}.A.\epsilon_i, \sigma_1) \\ 
  \epsilon_i \sim \mathcal{N}(0,1) \\ 
  ~ \\
  \mu_p \sim \mathcal{logN}(log(1),1) \\  
  \sigma_1 \sim \mathcal N_T(0,1) \\ 
  \hat{\sigma_2} = \sqrt(V_G)
  ~ \\
  V_Y = Var(log(y)) \\
  V_P = Var(log(\mu_p)) \\
  V_G = V_Y - V_P - V_R \\
  V_R=\sigma_1^2
  (\#eq:animalenvstan)
\end{equation} 

The model converged (Fig. \@ref(fig:envGenoTrace)) well for TWI but had some known diffuclties (see previous chapter) for NCI due to a low residual variation ($V_{R,NCI}=0.04$). We had two constrated results with most of the explained variation mainly attributed to the genotype for NCI (85%) and the majority attributed to population for TWI (42%). Still population explained 12% of NCI and genotype 4% of TWI. Finally, remaining variattion was almost null (3%) for NCI whereas important for TWI (54%) (Tab. \@ref(tab:envGenoTab) and Fig. \@ref(fig:envGenoR2) and \@ref(fig:envGenoVarPart)).

```{bash envGenoCluster, eval=F}
vars=(TWI NCI)
for var in "${vars[@]}" ; do for chain in $(seq 8) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript EnvGeno.R $chain $var" ; done ; done > EnvGeno.sh
sarray -J Env -o out/%j.Env.out -e out/%j.Env.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL EnvGeno.sh
```

```{r envGenoTab, fig.cap="Genetic variance of individual growth potential with a lognormal animal model."}
fitEnv <- list(TWI = list(), NCI = list())
for(var in c("TWI", "NCI")){
  for(sim in list.files("symcapture_save/EnvGeno", 
                        pattern = var, full.names = T)){
    load(sim)
    fitEnv[[var]] <- c(fitEnv[[var]], fit)
  }
}
fitEnv <- lapply(fitEnv, sflist2stanfit)
lapply(fitEnv, broom::tidyMCMC, c("mu", "Vp", "Vg", "Vr"), 
       droppars = NULL, rhat = T) %>% 
  bind_rows(.id = "Variable") %>% 
  separate(term, c("parameter", "population"), "[:punct:]", convert = T) %>% 
  mutate(population = recode(population, "1" = "S. globulifera Paracou",
                             "2" = "S. globulifera Regina", "3" = "S. sp1")) %>% 
  mutate(population = ifelse(is.na(population), "", population)) %>% 
  kable(caption = "Summary table of the kinship growth model",
        col.names = c("Variable", "Parameter",  "Population",  
                      "Estimate", "$\\sigma$", "$\\hat{R}$"))   
```

```{r envGenoTrace, fig.cap="Traceplot for environmental variables."}
cowplot::plot_grid(plotlist = lapply(fitEnv, mcmc_trace, 
                                     regex_pars = c("Vp", "Vg", "Vr"), 
                                     facet_args = list(nrow = 1)),
                   nrow = 2, labels = names(fitEnv))   
```

```{r envgenoR2, fig.cap="R2 for environmental variable"}
lapply(fitEnv, pars = c("Vp", "Vg", "Vr"), as.data.frame) %>% 
  bind_rows(.id = "model") %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vg, Vr))) %>% 
  mutate(Vexp = sum(c(Vp, Vg))) %>% 
  mutate_at(c("Vp", "Vg", "Vexp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = "model") %>% 
  group_by(model, variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, 
                           "Vexp" = "Marginal", "Vg" = "Genotype", 
                           "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +  
  theme(axis.title.y = element_blank()) +
  facet_wrap(~ model, nrow = 2) +
  coord_flip()   
```

```{r envGenoVarPart, fig.cap="Genetic variance partitionning for environmental variables."}
lapply(fitEnv, mcmc_intervals_data, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(parameter = recode(parameter, "Vp" = "Population", "Vg" = "Genotype", "Vr" = "Residual")) %>% 
  group_by(variable) %>%
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = variable, fill = parameter)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  facet_wrap(~ variable, scales = "free") +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2))  
```

```{r aTWInNCI, fig.cap="Multiplicative", fig.height=8, fig.width=8}
t <- lapply(fitEnv, function(fit)
  as.data.frame(fit, "alog") %>% 
    summarise_all(median) %>% 
    reshape2::melt(NULL)) %>% 
  bind_rows(.id = "fit") %>% 
  separate(variable, c("X1", "IndNum", "X2"), convert = T) %>% 
  mutate(value = exp(value)) %>% 
  reshape2::dcast(IndNum ~ fit, value.var = "value") %>% 
  left_join(trees, by = "IndNum", suffix = c(".a", ".var")) %>% 
  mutate(DBH = CircCorr/pi)
g.main <- ggplot(t, aes(NCI.a, TWI.a, size = DBH)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "lm", col = "black") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  xlab("Multiplicative genetic value for NCI") +
  ylab("Multiplicative genetic value for TWI") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.9, label.y = 0.9)
g.NCI <- ggplot(t, aes(NCI.a, NCI.var/10^4, col = pop, size = DBH)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual("", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  ylab("Individual NCI (/10^4)") +
  scale_x_continuous(position = "top") +
  theme(axis.title.x = element_blank(),
        legend.position = c(0.5, 0.9))
g.TWI <- ggplot(t, aes(TWI.var,  TWI.a, col = pop, size = DBH)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous("DBH 2017") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  xlab("TWI") +
  scale_y_continuous(position = "right") +
  theme(axis.title.y = element_blank(),
        legend.position = c(0.9, 0.2))
g.comp <- ggplot(t, aes(TWI.var, NCI.var/10^4, size = DBH)) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  scale_y_continuous(position = "right") +
  scale_x_continuous(position = "top") +
  theme(axis.title = element_blank()) +
  ggpubr::stat_cor(method = "pearson", label.x = 4, label.y = 0.7)
cowplot::plot_grid(g.TWI, g.main, g.comp, g.NCI, 
                   ncol=2, labels = c("A", "B", "C", "D")) 
```

## Association

We used per SNP linear mixed models implemented in `GEMMA` [Genome-wide Efficient Mixed Model Analysis, @Zhou2012a] to identify major effect SNPs. Linear mixed models are calculated for the environemental values $y$ of the $N$ individuals with following formula:

\begin{equation} 
  y_i \sim \mathcal N(\mu + x_i.\beta + Z.u_i, \sigma_E) \\
  u_i \sim \mathcal{MVN_N}(0,\sigma_G.K) \\
  \sigma_P = \sigma_G + \sigma_E
  (\#eq:lmmenv)
\end{equation} 


where $N$ is the number of individuals, $y$ is the vector of environmental variable values for the $N$ individual, $\mu$ is the intercept representing the mean environmental variable, $x$ is a SNP, $\beta$ is the effect size of the SNP, $Z$ is the loading matrix (computed with Cholesky decomposition of $K$) and $u$ is the vector of random effects corresponding to individuals kinship relations ($u$ thus represents individuals breeding values), $u$ values are centered on 0, covary with known kinship matrix $K$ with an associated genetic variance $\sigma_G$, $\sigma_E$ is the residual variance, and the sum of residual $\sigma_E$ and genetic $\sigma_G$ variances is equal to the phenotypic variance $\sigma_P$. $\beta$ significance is calculated with Wald test. And p-values are corrected for multiple testing with false discovery rate, resulting in q-values. 

Within each population, we didb't detect any outliers SNPs both with TWI and NCI (Fig. \@ref(fig:envMLM)).

```{bash EnvGemma, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
pops=(globuliferaTypeParacou globuliferaTypeRegina sp1)
for var in $(seq 2)
  do
  for pop in "${pops[@]}" ; 
    do 
      plink \
        --bfile env \
        --allow-extra-chr \
        --keep  $pop.fam \
        --extract env.prune.in \
        --maf 0.05 \
        --pheno env.phenotype --mpheno $var \
        --out $pop.var$var --make-bed ; 
      gemma -bfile $pop.var$var -gk 1 -o kinship.$pop.var$var ;
      gemma -bfile $pop.var$var -lmm 1 \
        -k output/kinship.$pop.var$var.cXX.txt \
        -o LMM.$pop.var$var ;
  done
done
cat LMM.snps | cut -f1 > snps.list
plink --bfile env --allow-extra-chr --extract snps.list --out outliers --recode A
```

```{r envMLM, fig.cap="q-value of major effect SNPs associated to environmental variables. SNP were detected using linear mixed models (LMM) with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
lmm <- lapply(list.files(file.path(path, "gemma", "env"), 
                         pattern = ".assoc.txt$", full.names = T),
              function(file) read_tsv(file) %>% mutate(file = file)) %>% 
  bind_rows() %>% 
  mutate(file = gsub("/home/sylvain/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics//gemma/env/LMM.", "", file)) %>% 
  mutate(file = gsub(".assoc.txt", "", file)) %>%
  separate(file, c("population", "variable"), sep = ".var") %>% 
  mutate(variable = recode(variable, "1" = "TWI",  "2" = "NCI")) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  mutate(qval = p.adjust(p_wald, "fdr")) %>%
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  na.omit(qval)
filter(lmm , qval < 0.05) %>%
  dplyr::select(rs, population, variable) %>%
  write_tsv(file.path(path, "gemma", "env", "LMM.snps"), col_names = F)
ggplot(lmm, aes(snp, -log10(qval), label = snp, alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ggrepel::geom_text_repel(data = lmm[lmm$qval < 0.05,]) +
  ylab(expression(-log[10]("q-value"))) +
  facet_grid(variable ~ population) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type") +
  theme(legend.position = "bottom")
```


```{r outliersEnv, fig.cap="TWI distribution for outliers genes."}
read_delim(file.path(path, "gemma", "env", "outliers.raw"), delim = " ") %>% 
  left_join(read_tsv(file.path(path, "gemma", "env", "env.phenotype"), 
         col_names = c("FID", "IID", "TWI", "NCI"))) %>% 
  left_join(dplyr::select(trees, IID, pop)) %>% 
  dplyr::select(-FID, -PAT, -MAT, -SEX, -PHENOTYPE) %>% 
  reshape2::melt(id.vars = c("pop", "IID", "TWI"),
                 variable.name = "snp", value.name = "allele") %>% 
  na.omit(allele) %>% 
  mutate(allele = as.factor(allele)) %>% 
  mutate(snp = gsub("_[[:alnum:]]$", "", snp)) %>% 
  right_join(dplyr::select(lmm[lmm$qval < 0.05,], population, rs) %>% 
               dplyr::rename(snp = rs, pop = population)) %>% 
  ggplot(aes(allele, TWI, fill = pop)) +
  geom_boxplot() +
  stat_summary(fun.data = function(y) 
    data.frame(y = max(y)+0.1, label = paste(length(y))),
    geom = "text", hjust = 0.5, vjust = 0.9) + 
  coord_flip() +
  facet_wrap(~ pop + snp, scales = "free") +
  scale_fill_discrete("") + ylab("TWI") + xlab("Major allele copies") +
  theme(strip.text = element_blank(), legend.position = "bottom")
```

```{r envGO, fig.cap="Term enrichment of Gene Ontology (GO) for outliers genes."}
trsc <- read_tsv(file.path(path, "..", "annotation", "snpsFunc.annotation")) %>% 
              dplyr::select(scf, snp, trsc) %>% 
  bind_rows(read_tsv(file.path(path, "..", "annotation", "snpsHitch.annotation"))) %>%
  unique() %>% 
  inner_join(dplyr::rename(lmm, snpNum = snp, snp = rs), .)
GO <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("Transcript") %>% 
  filter(transcript_id %in% local(unique(trsc$trsc))) %>% 
  dplyr::select(gene_id, transcript_id, annotation) %>% 
  dplyr::rename(gene = gene_id, trsc = transcript_id) %>% 
  collect() %>% 
  separate(annotation, paste0("X", 1:16), "\t") %>% 
  dplyr::rename(orf = X3, PFAM = X8, UniProt = X11, KEGG = X12, GO = X13) %>% 
  dplyr::select(trsc, gene, orf, GO) %>% 
  filter(GO != ".") %>% 
  left_join(trsc) %>% 
  group_by(population, gene, GO) %>% 
  summarise(qval = min(qval)) %>% 
  ungroup() %>% 
  dplyr::select(population, gene, qval, GO) %>% 
  separate_rows(GO, sep = "`") %>% 
  separate(GO, c("GOid", "GOtype", "GOname"), "\\^")
enrichment <- function(x) {
  y <- clusterProfiler::enricher(unique(filter(x, qval < 0.05)$gene),
                                            pvalueCutoff = 0.2, 
                                            pAdjustMethod = "BH",
                                            unique(x$gene),
                                            minGSSize = 10, 
                                            maxGSSize = 500, 
                                            qvalueCutoff = 0.2, 
                                            dplyr::select(x, GOid, gene),
                                            TERM2NAME = dplyr::select(x, GOid, GOname))
  if(is.null(y)){
    data.frame()
  } else {
    y@result
  }
}
group_by(GO, population) %>% 
  do(enrichment(.)) %>% 
  filter(qvalue < 0.8) %>% 
  ggplot(aes(reorder(Description, qvalue), Count, fill = -log(qvalue))) +
  geom_bar(stat = "identity") +
  coord_flip() +
  facet_wrap(~ population, labeller = "label_both", nrow = 2, scales = "free") +
  viridis::scale_fill_viridis() +
  theme(axis.title = element_blank())
```

