```{r setup_funcgenom, include=FALSE, eval=T}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(bayesplot)
library(rstan)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F, eval = T)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
```

# Functional genomics

```{r DataFuncGenom}
load("../../functional/functional_save/Individuals.Rdata")
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop"))) %>% 
  left_join(read_tsv(file.path(path, "populations", "paracou.hybridmap")),
            by = "Ind", suffix = c("", ".hybrid")) %>% 
  left_join(Individuals) %>% 
  mutate(LMA = invSLA) %>% 
  filter(!is.na(TWI), !is.na(NCI))
ids <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king.id"))
K <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king"),
         col_names = ids$IID) %>% 
  as.data.frame()
row.names(K) <- ids$IID
K <- as.matrix(K)
model_data <- function(Trait){
  trees <- trees[!is.na(trees[Trait]),]
  trees <- trees %>% 
    filter(IID %in% ids$IID) %>% 
    mutate(popNum = as.numeric(as.factor(pop)),
           plotNum = as.numeric(as.factor(Plot)))
  K <- K[trees$IID, trees$IID]
  K[K < 0] <- 0
  K <- K*2
  K <- as.matrix(Matrix::nearPD(K)$mat)
  list(I = nrow(trees),
       P = length(unique(trees$popNum)),
       Pl = length(unique(trees$plotNum)),
       Trait = as.vector(scale(trees[,Trait], center = F)),
       DBH = trees$DBH,
       TWI = as.vector(scale(trees$TWI, center =F)),
       NCI = as.vector(scale(trees$NCI, center =F)),
       population = trees$popNum,
       plot = trees$plotNum,
       K = K)
}
traits <- c("LMA", "LDMC", "LT", "invLA", "CC")
mdata <- lapply(traits, model_data)
names(mdata) <- traits
save(mdata, file = file.path("symcapture_save", 'dataFT.Rdata')) 
```


## Genotype

We used the following model to asses genetic varaince associated to leaf functional traits once we accounted for ontogenetic variation through DBH:

\begin{equation} 
  Trait_{t,p,plot,i} \sim \mathcal{logN}(log(\alpha_{t,p}.a_{t,i}.\delta_{plot}.\frac{DBH_i}{{\beta_{DBH}}_{t,p} + DBH_i}), \sigma_3) \\
  a_{t,i} \sim \mathcal{MVlogN}(log(1), \sigma_2.K) \\
  \delta_{plot}\sim \mathcal {logN}(log(1),\sigma_3)
  (\#eq:ftgeno)
\end{equation} 

We fitted the equivalent model with following priors:

\begin{equation} 
  Trait_{t,p,plot,i} \sim \mathcal{logN}(log(\alpha_{t,p}.\hat{a_{t,i}}.\hat{\delta_{plot}}.\frac{DBH_i}{{\beta_{DBH}}_{t,p} + DBH_i}), \sigma_3) \\
  \hat{a_{t,i}} = e^{\sigma_2.A.\epsilon_{1,i}} \\
  \hat{\delta_{plot}} = e^{\sigma_3.\epsilon_{2,plot}} \\
  \epsilon_{1,i} \sim \mathcal N(0,1) \\
  \epsilon_{2,plot} \sim \mathcal N(0,1) \\
  ~ \\
  \beta_{DBH} \sim \mathcal{logN}(log(1),1) \\
  (\sigma_1,\sigma_2,\sigma_3) \sim \mathcal{N}_T^3(0,1) \\
  ~ \\
  V_P = Var(log(\alpha_{t,p})) \\
  V_G = \sigma_2^2 \\
  V_{DBH} = Var(log(\frac{DBH_i}{{\beta_{DBH}}_{t,p} + DBH_i})) \\
  V_{plot}= \sigma_3^2 \\
  V_R = \sigma_1^2
  (\#eq:ftgenostan)
\end{equation} 

```{bash ftGenoCluster, eval=F}
traits=(LMA LDMC LT invLA CC)
for trait in "${traits[@]}" ; do for chain in $(seq 8) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript FTGeno.R $chain $trait" ; done ; done > FTGeno.sh
sarray -J FT -o out/%j.FT.out -e out/%j.FT.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL FTGeno.sh
```

```{r ftGenoTab, fig.cap="Genetic variance of individual growth potential with a normal animal model."}
fitFT <- list(LMA = list(), LDMC = list(), LT = list(),
               invLA = list(), CC = list())
for(trait in c("LMA", "LDMC", "LT", "invLA", "CC")){
  for(sim in list.files("symcapture_save/FTGeno", 
                        pattern = trait, full.names = T)){
    load(sim)
    fitFT[[trait]] <- c(fitFT[[trait]], fit)
  }
}
fitFT <- lapply(fitFT, sflist2stanfit)
lapply(fitFT, broom::tidyMCMC, c("alpha", "betaDBH", "Vp", "Vg", "Vr"), 
       droppars = NULL, rhat = T) %>% 
  bind_rows(.id = "Variable") %>% 
  separate(term, c("parameter", "population"), "[:punct:]", convert = T) %>% 
  mutate(population = recode(population, "1" = "S. globulifera Paracou",
                             "2" = "S. globulifera Regina", "3" = "S. sp1")) %>% 
  mutate(population = ifelse(is.na(population), "", population)) %>% 
  kable(caption = "Summary table of the kinship growth model",
        col.names = c("Variable", "Parameter",  "Population", 
                      "Estimate", "$\\sigma$", "$\\hat{R}$"))
```

```{r ftGenoTrace, fig.cap="Traceplot."}
cowplot::plot_grid(plotlist = lapply(fitFT, mcmc_trace, 
                                     regex_pars = c("sigma"), 
                                     facet_args = list(nrow = 1)),
                   labels = names(fitFT), nrow = 5)
```

```{r ftGenoVar, fig.cap="Genetic variance partitionning."}
save(fitFT, file = "../../thesis/rdata/fitFT.Rdata")
lapply(fitFT, mcmc_intervals_data, regex_pars = c("Vp", "Vg", "Vdbh", "Vplot", "Vr")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", "Vdbh" = "DBH",
                           "Vplot" = "Plot", "Vr" = "Residual")) %>% 
  group_by(variable) %>%
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = variable, fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  facet_wrap(~ variable, scales = "free", nrow = 1) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank(),
        legend.position = "bottom") +
  scale_fill_discrete(expression(sigma^2))
```
