```{r setup_growthgenom, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(bayesplot)
library(rstan)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
load("./symcapture_save/Env2.Rdata")  
# load(file.path("symcapture_save", 'AnimalModels.Rdata'))
```

# Growth genomics

We explored and selected ontogenic and environmental descriptors to be used in the annual growth rate model. We then built a growth and environmental model, from which we derived individual growth parameters to be used in association genomics. We then described the three methods used to explore the link between individual growth parameters and individual SNP and kinship with:

* the modele Animal to explore the proportion of phenotypic variance explained by genetic
* linear mixed models (LMM) used to identify major effect SNPs associated to the phenotypic trait
* bayesian sparse linear mixed models (BSLMM) used to identify the polygenic architecture associated to the phenotypic trait

Finally we used the three methods for every selected individual growth parameters, e.g. maximum growth potential (Gmax), optimal growth diameter (NCI), growth kurtosis and slopes of growth response to environmental variables ($\beta_i$), investigating selected SNPs, associated annotation, and ontogenetic term (GO) possible enrichment.

* __Variables__ growth, ontogeny and environmental variables definition and selection based on multicolinearity and eco-evolutionary meaningfulness
* __Growth__ individual growth model explained by ontogeny and environmental variables
* __Methods__ association methods for genetic variance, major effect SNPs and polygenic signal
* __Maximum growth potential (Gmax)__ relation to individuals kinship and SNP variation
* __Optimal growth diameter (Dopt)__ relation to individuals kinship and SNP variation
* __Growth kurtosis (Ks)__ relation to individuals kinship and SNP variation
* __Slopes of growth response to environmental variables ($\beta_i$)__ relation to individuals kinship and SNP variation

## Variables

We explored the role of individual size, abitoic environment, biotic interactions, and individual genotype on individual tree growth in space and time. Individual tree growth was calculated with following formula:

$AGR_{i,t} = \frac{DBH_{i,t} - DBH_{i,t-1}}{\Delta t}$

with the annual growth rate $AGR$ of individual $i$ in year $t$ depended of individual $DBH$ at year $t$ and $t-1$ and the timelaps between the two year $\Delta t$.

The topographic wetness index ($TWI_i$) was selected among several spatial abiotic descriptors as a proxy of water accumulation. Waterlogging and topography have been highlighted as crucial for forest dynamics [@ferry2010higher] and species habitat preference [@Allie2015] at the Paracou study site. TWI was derived from a 1-m resolution digital elevation model using SAGA-GIS [@Conrad2015] based on a LiDAR campaign done in 2015.

The area and the sum of days under the curve for a threshold of 0.4 of the relative extractible water [$REW_t$, @Wagner2011], respectively $AREW$ and $SREW$ were selected among several temporal abiotic descriptors as a major climatic driver of tropical forest dynamics [@Aubry-Kientz2015a].

Biotic descriptors included two spatial and temporal variables: the Neighborhood Crowding Index [NCI; @Uriarte2004] and the Annual Crowding Rate. The Neighborhood Crowding Index $NCI_{i,t}$ from tree individual $i$ in year $t$ was calculated with following formula:

$$NCI_i = \sum _{j~|~\delta_{i,j}<20m} ^{J_{i,t}} DBH_{j,t}^2 e^{-\frac14.\delta_{i,j}}$$

with $DBH_{j,t}$ the diameter from neighboring tree $j$ in year $t$ and $\delta_{i,j}$ its distance to individual tree $i$. $NCI_i$ is computed for all neighbors at a distance $\delta_{i,j}$ inferior to maximum neighboring distance of 20 meters. The power of neighbors $DBH$ effect was set to 2 to consider neighbors surface. The decrease of neighbors $DBH$ effect with distance was set to 0.25 here to represent trees at 20 meters of the focal trees having 1% of the effect of the same tree at 0  meters. $NCI$ represents biotic asymmetric competition through tree neighborhood over time. 

The Annual Neighborhood Crowding Rate $ANCR_{i,t}$ from tree individual $i$ in year $t$ was calculated with following formula:

$ANCR_{i,t} = \frac{NCI_{i,t} - NCI_{i,t-1}}{\Delta t}$

$ANCR$ represents the dynamic of biotic asymmetric competition through tree neighborhood over time (derivate of $NCI_i$). Both NCI and ANCR were calculated among heterospecific and conspecific neighbors and with all neighbors confounded. Edging effect have been corrected by weighing values from individual close to the border by the inverse of the surface outside of the plot in the 20m radius.

Both correlations and principal component analyses showed non colinear DBH, TWI, AUC, NCI and ANCR that we selected for further analyses (Fig. \@ref(fig:correlationsGrowthGenom) and Fig. \@ref(fig:pcaGrowthGenom)).

```{r DataGrowthGenom, eval=F}
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Symphonia") %>% 
  collect()
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop"))) %>% 
  left_join(read_tsv(file.path(path, "populations", "paracou.hybridmap")),
            by = "Ind", suffix = c("", ".hybrid"))
treesXY <- trees
coordinates(treesXY) <- ~Xutm + Yutm
proj4string(treesXY) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
treesXY <- spTransform(treesXY, CRSobj = crs)
wetness <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "TWI_1m.tif"))
dem <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
wetness <- projectRaster(wetness, crs = crs)
trees$TWI <- raster::extract(wetness, treesXY)
rm(dem, wetness, treesXY)
weather <- read_csv(file.path("../../../data/Paracou/", "weather", "REW_1978-2017_Fabien.csv")) %>% 
  dplyr::select(-X1, -day, -month, day_julian) %>% 
  filter(REW < 0.4) %>% 
  mutate(AREW = 0.4-REW) %>% 
  group_by(year) %>% 
  summarise(AREW = sum(AREW), SREW = n()) %>% 
  dplyr::rename(CensusYear = year)
trees <- left_join(trees, weather)
rm(weather)
cl <- parallel::makeCluster(getOption("cl.cores", 4))
parallel::clusterExport(cl, list("trees"))
NCI <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
    tbl("Paracou") %>% 
    filter(CensusYear == local(trees$CensusYear[ind])) %>% 
    filter(Plot == local(trees$Plot[ind])) %>% 
    filter(idTree != local(trees$idTree[ind])) %>% 
    mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
    mutate(dij = ifelse(dij <1, 1, dij)) %>% 
    filter(dij < 20) %>% 
    mutate(con = ifelse(Genus == local(trees$Genus[ind]) && Species == local(trees$Species[ind]), 1, 0)) %>% 
    mutate(DBH = CircCorr/pi) %>% 
    summarise(idTree = local(trees$idTree[ind]),
              CensusYear  = local(trees$CensusYear[ind]),
              NCIcon = sum(DBH*DBH*exp(-0.25*dij)*con),
              NCIhetero = sum(DBH*DBH*exp(-0.25*dij)*(1-con))) %>% 
    collect()})
parallel::stopCluster(cl) ; rm(cl)
NCI <- bind_rows(NCI)
trees <- left_join(trees, NCI)
rm(NCI)
trees <- trees %>% 
  rowwise() %>% 
  mutate(NCI = sum(NCIcon, NCIhetero))
trees <- trees %>% 
  mutate(DBH = CircCorr/pi) %>% 
  group_by(idTree) %>% 
  arrange(CensusYear) %>% 
  mutate(AGR = (DBH-lag(DBH))/(CensusYear-lag(CensusYear)),
         ANCRcon = (NCIcon-lag(NCIcon))/(CensusYear-lag(CensusYear)),
         ANCRhetero = (NCIhetero-lag(NCIhetero))/(CensusYear-lag(CensusYear)),
         ANCR = (NCI-lag(NCI))/(CensusYear-lag(CensusYear))) %>% 
  mutate(AGR = ifelse(AGR < 0, NA, AGR)) %>% 
  ungroup()
save(trees, file = "./symcapture_save/Env2.Rdata")
```

```{r correlationsGrowthGenom, fig.cap="Environmental descriptors correlations."}
trees %>% 
  dplyr::select(DBH, TWI, AREW, SREW, NCI, ANCR) %>% 
  na.omit() %>% 
  cor(.) %>% 
  corrplot::corrplot.mixed()
```

```{r pcaGrowthGenom, fig.cap="Environmental descriptors principal component analysis."}
factoextra::fviz_pca_var(princomp(~ DBH + TWI + AREW + SREW + NCI + ANCR, trees, cor = T),
                         axes = c(1, 2), geom = c("arrow", "text"), col.var = "contrib")
```

```{r mdata}
load("./symcapture_save/Env2.Rdata")  
trees <- trees %>% 
  na.omit(pop) %>% # remove admixed
  filter(!is.na(AGR), !is.na(ANCR)) %>% 
  mutate(IndNum = as.numeric(as.factor(Ind)), popNum = as.numeric(as.factor(pop)))
treesSub <- filter(trees, Plot == 16) %>% # for test purposes
  mutate(IndNum = as.numeric(as.factor(Ind)), popNum = as.numeric(as.factor(pop)))
fam <- read_delim(file.path(path, "gemma", "Gmax0", "paracou3pop.fam"), delim = " ", 
                  col_names = c("FID", "IID", "PAT", "MAT", "SEX", "PHENOTYPE"))
K <- read_tsv(file.path(path, "gemma", "Gmax0", "paracou3pop.kinship.sXX.txt"), 
              col_names = fam$IID) %>% 
  mutate(IID = fam$IID) %>% 
  column_to_rownames(var = "IID") %>% 
  as.matrix()
mdata <- function(data) list(N = nrow(data),
                             I = length(unique(data$Ind)),
                             G = length(unique(data$pop)),
                             AGR = data$AGR,
                             DBH = data$DBH,
                             TWI = as.vector(scale(data$TWI, center = F)),
                             AREW = as.vector(scale(data$AREW, center = F)),
                             NCI = as.vector(scale(data$NCI, center = F)),
                             ind = data$IndNum,
                             gp = data$popNum,
                             indingp = unique(arrange(data, IndNum)[c("IndNum", "popNum")])$popNum,
                             K = dplyr::select(data, IID, IndNum) %>% 
                               unique() %>% arrange(IndNum) %>% 
                               dplyr::select(IID) %>% unlist() %>% K[.,.])
```

## Growth & Ontogeny

We integrated population structure and individual varition in the growth model used in @Herault2011 to explain individual growth with ontogeny. We used the following form of hierachical model to explain the logarithm of Annual Growth Rate $AGR_{t,i,g}$ with Diameter at Breast Height $DBH_{t,i,g}$ at year $t$ for individual $i$ in genepool $gp$:

$$log(AGR_{t,g,i|g}+1) \sim \mathcal N (Gmax_i.exp(-\frac12.[\frac{log(\frac{DBH_{t,g,i}}{Dopt_i})}{Ks_i}]^2), \sigma)$$
$$\begin{bmatrix} Gmax_i \\ Dopt_i \\ Ks_i \end{bmatrix} \sim \mathcal N ^3(\begin{bmatrix} Gmax_g \\ Dopt_g \\ Ks_g \end{bmatrix},\begin{bmatrix} \sigma_{Gmax_i} \\ \sigma_{Dopt_i} \\ \sigma_{Ks_i} \end{bmatrix} )$$
$$\begin{bmatrix} Gmax_g \\ Dopt_g \\ Ks_g \end{bmatrix} \sim \mathcal N ^3(\begin{bmatrix} Gmax \\ Dopt \\ Ks \end{bmatrix},\begin{bmatrix} \sigma_{Gmax_g} \\ \sigma_{Dopt_g} \\ \sigma_{Ks_g} \end{bmatrix} )$$

with $Gmax$, $Dopt$ and $Ks$ the parameters of the growth response to ontgeny representing respectively the maximum growth potential, the optimal growth diameter when this potential is reached, and the kurtosis of the growth curve respectivelly. Ontogenic parameters were defined at the individual level as random effects centered on the genepool level parameters with associated parameters individual variance, and genepool level parameters were themselves defined as random effects on genus level parameters with associated parameters genepool variance. 

At the genus level, *Symphonia* individuals showed a maximum growth of $6.91\pm1.8~mm.yr^{-1}$ at optimal diameter of $28\pm11~cm$ with a kurtosis of $0.7\pm0.1$ (Tab. \@ref(tab:growthOntogenyTab)). Maximum growth and kurtosis didn't significantly varied at the genepool level but optiaml diameter did with a larger optimal diameter for *S. globulifera type Regina* ($Dopt=32~cm$) and a smaller optimal diameter for *S. sp1* ($Dopt=21~cm$). Finally, all parameters showed significant variation at the individual level within genepools ($\pm1.8~mm.yr^{-1}$, $\pm9~cm$, and $\pm0.2$ for maximum growth, optimal diameter, and kurtosis respectively, Fig. \@ref(fig:growthOntogenyG)). The model residual variation is relatively low ($\pm1.5~mm.yr^{-1}$, Fig. \@ref(fig:growthOntogenyGP3)).

```{r growthOntogenyTab}
# growth <- stan_model("./symcapture_models/Growth.stan")
# fitGrowth <- sampling(growth, chains = 2, data = mdata(trees), save_warmup = F)
# save(fitGrowth, file = file.path("symcapture_save", 'growth.Rdata'))
load(file.path("symcapture_save", 'growth.Rdata'))
broom::tidyMCMC(fitGrowth, c("Gmax", "Gmaxg", "sigmaGmaxg", "sigmaGmaxi",
                             "Dopt", "Doptg", "sigmaDoptg", "sigmaDopti",
                             "Ks", "Ksg", "sigmaKsg", "sigmaKsi",
                             "sigma", "lp__"), droppars = NULL, rhat = T) %>% 
  kable(caption = "Summary table of the ontogenic growth model")
```

```{r growthOntogenyG, fig.cap="Growth model predictions."}
pars <- as.data.frame(fitGrowth, pars = c("Gmax", "Dopt", "Ks")) %>% 
  reshape2::melt(variable.name = "parameter") %>% 
  group_by(parameter) %>% 
  summarise(q5 = quantile(value, 0.05), m = mean(value), q95 = quantile(value, 0.95)) %>% 
  mutate(genusNum = 1) %>% 
  reshape2::melt(id.vars = c("parameter", "genusNum")) %>% 
  mutate(parameter = paste0(parameter, "_", variable)) %>% 
  dplyr::select(-variable) %>%
  reshape2::dcast(genusNum ~ parameter)
parsg <- as.data.frame(fitGrowth, pars = c("Gmaxg", "Doptg", "Ksg")) %>% 
  reshape2::melt(id.vars = NULL, variable.name = "parameter") %>% 
  group_by(parameter) %>% 
  summarise(q5 = quantile(value, 0.05), m = mean(value), q95 = quantile(value, 0.95)) %>% 
  mutate(parameter = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(popNum = gsub("([[:alpha:]])", "", parameter)) %>% 
  mutate(parameter = gsub("([[:digit:]])", "", parameter)) %>% 
  reshape2::melt(id.vars = c("parameter", "popNum")) %>% 
  mutate(parameter = paste0(parameter, "_", variable)) %>% 
  dplyr::select(-variable) %>%
  reshape2::dcast(popNum ~ parameter) %>% 
  mutate(popNum = as.numeric(popNum))
parsi <- as.data.frame(fitGrowth, pars = c("Gmaxi", "Dopti", "Ksi")) %>% 
  reshape2::melt(id.vars = NULL, variable.name = "parameter") %>% 
  group_by(parameter) %>% 
  summarise(q5 = quantile(value, 0.05), m = mean(value), q95 = quantile(value, 0.95)) %>% 
  mutate(parameter = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(IndNum = gsub("([[:alpha:]])", "", parameter)) %>% 
  mutate(parameter = gsub("([[:digit:]])", "", parameter)) %>% 
  reshape2::melt(id.vars = c("parameter", "IndNum")) %>% 
  mutate(parameter = paste0(parameter, "_", variable)) %>% 
  dplyr::select(-variable) %>%
  reshape2::dcast(IndNum ~ parameter) %>% 
  mutate(IndNum = as.numeric(IndNum))
data.frame(DBH = rep(1:100, mdata(trees)$I), 
           IndNum = rep(1:mdata(trees)$I, each = 100)) %>% 
  left_join(unique(dplyr::select(trees, IndNum, Ind, popNum, pop))) %>% 
  mutate(genusNum = 1) %>% 
  left_join(pars) %>% 
  left_join(parsg) %>% 
  left_join(parsi) %>% 
  mutate(Gpred = Gmax_m*exp(-0.5*(log(DBH/Dopt_m)/Ks_m)^2),
         Gpred5 = Gmax_q5*exp(-0.5*(log(DBH/Dopt_q5)/Ks_q5)^2),
         Gpred95 = Gmax_q95*exp(-0.5*(log(DBH/Dopt_q95)/Ks_q95)^2)) %>% 
  mutate(Gpredg = Gmaxg_m*exp(-0.5*(log(DBH/Doptg_m)/Ksg_m)^2),
         Gpred5g = Gmaxg_q5*exp(-0.5*(log(DBH/Doptg_q5)/Ksg_q5)^2),
         Gpred95g = Gmaxg_q95*exp(-0.5*(log(DBH/Doptg_q95)/Ksg_q95)^2)) %>% 
  mutate(Gpredi = Gmaxi_m*exp(-0.5*(log(DBH/Dopti_m)/Ksi_m)^2),
         Gpred5i = Gmaxi_q5*exp(-0.5*(log(DBH/Dopti_q5)/Ksi_q5)^2),
         Gpred95i = Gmaxi_q95*exp(-0.5*(log(DBH/Dopti_q95)/Ksi_q95)^2)) %>% 
  ggplot(aes(x = DBH)) +
  geom_ribbon(aes(ymin = Gpred5i, ymax = Gpred95i, group = Ind), alpha = 0.5, fill = "lightgrey") +
   geom_point(aes(y = log(AGR +1), col = Ind), alpha = 0.4, data = trees) +
  geom_line(aes(y = Gpredi, col = Ind), linetype = "dotted") +
  geom_line(aes(y = Gpredg, group = pop), col = "black", linetype = "dashed") +
  geom_line(aes(y = Gpred), col = "black", size = 1.2) +
  scale_color_discrete(guide = "none") +
  scale_fill_discrete(guide = "none") 
```

```{r growthOntogenyGP3, fig.cap="Growth model predictions for Plot 3."}
filter(trees, Plot == 3) %>% 
  group_by(Ind) %>% 
  filter(n() > 1) %>% 
  mutate(genusNum = 1) %>% 
  left_join(pars) %>% 
  left_join(parsg) %>% 
  left_join(parsi) %>% 
  mutate(Gpred = Gmax_m*exp(-0.5*(log(DBH/Dopt_m)/Ks_m)^2),
         Gpred5 = Gmax_q5*exp(-0.5*(log(DBH/Dopt_q5)/Ks_q5)^2),
         Gpred95 = Gmax_q95*exp(-0.5*(log(DBH/Dopt_q95)/Ks_q95)^2)) %>% 
  mutate(Gpredg = Gmaxg_m*exp(-0.5*(log(DBH/Doptg_m)/Ksg_m)^2),
         Gpred5g = Gmaxg_q5*exp(-0.5*(log(DBH/Doptg_q5)/Ksg_q5)^2),
         Gpred95g = Gmaxg_q95*exp(-0.5*(log(DBH/Doptg_q95)/Ksg_q95)^2)) %>% 
  mutate(Gpredi = Gmaxi_m*exp(-0.5*(log(DBH/Dopti_m)/Ksi_m)^2),
         Gpred5i = Gmaxi_q5*exp(-0.5*(log(DBH/Dopti_q5)/Ksi_q5)^2),
         Gpred95i = Gmaxi_q95*exp(-0.5*(log(DBH/Dopti_q95)/Ksi_q95)^2)) %>% 
  ggplot(aes(x = CensusYear)) +
  geom_ribbon(aes(ymin = Gpred5, ymax = Gpred95, fill = "Genus"), alpha = 0.5) +
  geom_ribbon(aes(ymin = Gpred5g, ymax = Gpred95g, fill = "Gene pool"), alpha = 0.5) +
  geom_ribbon(aes(ymin = Gpred5i, ymax = Gpred95i, fill = "Individual"), alpha = 0.5) +
  geom_point(aes(y = log(AGR +1)), col = "black") +
  geom_line(aes(y = Gpredi, col = "Individual"), size = 1.2) +
  geom_line(aes(y = Gpredg, col = "Gene pool"), size = 1.2) +
  geom_line(aes(y = Gpred, col = "Genus"), size = 1.2) +
  scale_color_discrete(guide = "none") +
  scale_fill_discrete("Hierarchical level") +
  facet_wrap(~ Ind, scales = "free") +
  theme(legend.position = "bottom") 
``` 

## Growth & Environment

We then integrated environmental effects in the previous model of individual growth explained with ontogeny, including topography (TWI), climate (AREW), and neighbors interaction (NCI). We used the following form of model to explain the logarithm of Annual Growth Rate $AGR_{t,i,g}$ with Diameter at breast Height $DBH_{t,i,g}$, Topographic Wetness Index $TWI_{i,g}$, Area Under the Curve of Relative Extractible Water Content $AREW_{t}$, and Neighbor Crowding Index $NCI_{t,i,g}$, at year $t$ for individual $i$ in genepool $gp$:

$$log(AGR_{t,g,i|g}+1) \sim \mathcal N (Gmax_i.e^{-\frac12.[\frac{log(\frac{DBH_{t,g,i}}{Dopt_i})}{Ks_i}]^2}.e^{-\beta TWI_g.TWI_{g,i}}.e^{-\beta AREW_i.AREW_{t}}.e^{-\beta NCI_i.NCI_{t,g,i}}, \sigma)$$
$$\begin{bmatrix} Gmax_i \\ Dopt_i \\ Ks_i \\ \beta AREW_i \\ \beta NCI_i\end{bmatrix} \sim \mathcal N ^5(\begin{bmatrix} Gmax_g \\ Dopt_g \\ Ks_g \\ \beta AREW_g \\ \beta NCI_g \end{bmatrix},\begin{bmatrix} \sigma_{Gmax_i} \\ \sigma_{Dopt_i} \\ \sigma_{Ks_i} \\ \sigma_{AREW_i} \\ \sigma_{NCI_i}\end{bmatrix} )$$
$$\begin{bmatrix} Gmax_g \\ Dopt_g \\ Ks_g \\ \beta TWI_g \\ \beta AREW_g \\ \beta NCI_g\end{bmatrix} \sim \mathcal N ^6(\begin{bmatrix} Gmax \\ Dopt \\ Ks \\ \beta TWI \\ \beta AREW \\ \beta NCI \end{bmatrix},\begin{bmatrix} \sigma_{Gmax_g} \\ \sigma_{Dopt_g} \\ \sigma_{Ks_g} \\ \sigma_{TWI_g} \\ \sigma_{AREW_g} \\ \sigma_{NCI_g} \end{bmatrix} )$$

with $Gmax$, $Dopt$ and $Ks$ the parameters of the growth response to ontgeny representing respectively the maximum growth potential, the optimal growth diameter when this potential is reached, and the kurtosis of the growth curve respectivelly; and $\beta TWI$, $\beta AREW$, and $\beta NCI$ the slopes of environmental effects. Ontogenic parameters and environmental slopes were defined at the individual level as random effects centered on the genepool level parameters with associated parameters individual variance (except for $\beta TWI$), and genepool level parameters were themselves defined as random effects on genus level parameters with associated parameters genepool variance. 

> Due to convergence issue I only fitted the model for P16 (33 individuals) for the moment.

Growth response to ontogeny didn't change compared to the previous paragraph, and won't be discussed here. At the genus *Symphonia* level TWI, and NCI had no effect on growth, and AREW had only a small negative but non significant effect, e.g. drier years reduced individual growth. Environmental effects didn't show significant variation at the genepool level ($0 \in \sigma_g$, $0 \in \sigma_i$), except for AREW which significantly varied at the individual level of $\pm0.40$. Consequently TWI and NCI slopes won't be used in further genomic analyses as they don't vary at the individual scale. But AREW could be used in genomic analyses to see the genetic basis behind individual growth response to dry years.

```{r growthEnvTab}
# growthEnv <- stan_model("./symcapture_models/GrowthEnv.stan")
# fitGrowthEnv <- sampling(growthEnv, chains = 2, data = mdataSub, save_warmup = F, control = list(adapt_delta = 0.99))
# save(fitGrowthEnv, file = file.path("symcapture_save", 'growthenv.Rdata'))
load(file.path("symcapture_save", 'growthenv.Rdata'))
broom::tidyMCMC(fitGrowthEnv, c("Gmax", "Gmaxg", "sigmaGmaxg", "sigmaGmaxi",
                             "Dopt", "Doptg", "sigmaDoptg", "sigmaDopti",
                             "Ks", "Ksg", "sigmaKsg", "sigmaKsi",
                             "betaTWI", "betaTWIg", "sigmaKsg",
                             "betaAREW", "betaAREWg", "sigmaAREWg", "sigmaAREWi",
                             "betaNCI", "betaNCIg", "sigmaNCIg", "sigmaNCIi",
                             "sigma", "lp__"), droppars = NULL, rhat = T) %>% 
  kable(caption = "Summary table of the ontogeny & environemnt growth model") 
```

```{r growthEnvG, fig.cap="Environmental effects posterior for the growth model with ontogeny and environment."}
mcmc_dens(as.array(fitGrowthEnv, pars = c("betaTWI", "sigmaTWIg", "betaTWIg",
                                          "betaAREW", "sigmaAREWg", "betaAREWg", "sigmaAREWi", 
                                          "betaNCI",  "sigmaNCIg",  "betaNCIg", "sigmaNCIi"))) 
```

## Growth & Kinship

Model:

$$log(AGR_{y,p,i}+1) \sim \mathcal N (Gmax_i.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{Dopt_i})}{Ks_i}]^2), \sigma^2)$$
$$\mid \forall \theta \in \begin{bmatrix} Gmax \\ Dopt \\ Ks \end{bmatrix}, \begin{align} \theta_i \sim \mathcal N (\theta_p + a_{\theta}, \sigma^2_{R_{\theta}})  \\ \theta_p \sim \mathcal N (\theta, \sigma^2_{P_{\theta}}) \\ a_{\theta} \sim \mathcal{MVN} (0, \sigma^2_{G_{\theta}}.K) \end{align}$$

Derivate:

$$h^2 = \frac{\sigma^2_{G}}{\sigma^2_{P} + \sigma^2_{G} + \sigma^2_{R}}$$

$$h^2_p = \frac{\sigma^2_{P} + \sigma^2_{G}}{\sigma_{P} + \sigma^2_{G} + \sigma^2_{R}}$$

$$Q_{ST} = \frac{\sigma^2_{P}}{\sigma^2_{P} + 2.\sigma^2_{G}}$$

```{r growthKinshipTab}
growth <- stan_model("./symcapture_models/GrowthAnimal.stan")
fitGrowthKin <- sampling(growth, chains = 2, save_warmup = F, include = F, pars = c("a"),
                         control = list(adapt_delta = 0.9, max_treedepth = 12), data = mdata(treesSub))
save(fitGrowthKin, file = file.path("symcapture_save", 'growthkin.Rdata'))
load(file.path("symcapture_save", 'growthkin.Rdata'))
broom::tidyMCMC(fitGrowthKin, c("Gmax", "Dopt", "Ks",
                                "sigmaP", "sigmaG", "sigmaR", 
                                "sigma", "lp__",
                                "h2", "h2p", "Qst"), droppars = NULL, rhat = T) %>% 
  kable(caption = "Summary table of the ontogenic growth model")
mcmc_intervals_data(fitGrowthKin, regex_pars = c("h2", "h2p", "Qst")) %>% 
  separate(parameter, c("index", "parameter"), "[:punct:]") %>% 
  mutate(parameter = recode(parameter, "1" = "Gmax", "2" = "Dopt", "3" = "Ks")) %>% 
  mutate(index = recode(index, "h2" = "h^2 == frac(sigma[G]^2,sigma[P]^2+sigma[G]^2+sigma[R]^2)", 
                        "h2p" = "h[p]^2 == frac(sigma[P]^2+sigma[G]^2,sigma[P]^2+sigma[G]^2+sigma[R]^2)", 
                        "Qst" = "Q[st] == frac(sigma[P]^2,sigma[P]^2+2*sigma[G]^2)")) %>% 
  ggplot(aes(x = parameter, xend = parameter, col = parameter, fill = parameter)) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  coord_flip() +
  facet_wrap(~ index, labeller = label_parsed) +
  xlab("") + ylab("") + 
  scale_fill_discrete(guide = "none") +
  scale_color_discrete(guide = "none")
```

```{r}
load(file.path("symcapture_save", 'growthkin3.Rdata'))
fitGrowthKin@stanmodel
broom::tidyMCMC(fitGrowthKin, c("Gmax", "Dopt", "Ks",
                                "sigmaGmaxg", "sigmaDoptg", "sigmaKsg", 
                                "sigmaG", 
                                "sigmaGmaxi", "sigmaDopti", "sigmaKsi", 
                                "sigma", "lp__"), droppars = NULL, rhat = T) %>% 
  kable(caption = "Summary table of the ontogenic growth model")
```

## Maximum growth potential (Gmax)

```{bash GmaxGemma, eval=F, echo=T}
# left_join(parsi, trees) %>% 
#   dplyr::select(FID, IID, Gmaxi_m) %>%
#   unique() %>%
#   mutate(Gmax = RNOmni::rankNorm(Gmaxi_m)) %>%
#   write_tsv(file.path(path, "gemma", "Gmax", "Gmax.phenotype"), col_names = F)
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
cat paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > paracou3pop.fam
pops=(sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  grep $pop paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > $pop.fam ; done
pops=(paracou3pop sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  
  plink \
  --bfile ../../../variantCalling/paracou3pop/symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr \
  --keep $pop.fam \
  --extract ../../../variantCalling/paracou3pop/LD.prune.in \
  --maf 0.05 \
  --pheno Gmax.phenotype \
  --out $pop \
  --make-bed ;
done  
for pop in "${pops[@]}" ; do  gemma -bfile $pop -gk 1 -o $pop.kinship ; done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k output/$pop.kinship.cXX.txt -lmm 1 -o $pop.LMM.Gmax ; done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k -bslmm 1 -o $pop.BSLMM.Gmax ; done
```

Individual maximum growth potential is mainly explained by genetic variation ($\sigma_G=0.72\pm0.16$, Fig. \@ref(fig:GmaxPVE)). $Gmax$ is thus an ontogenetic factor as it controls the shape of the growth response to ontogeny and its controlled by genetics ! Individual growth potential wasn't significantly associated to any  major effect SNPs (Fig. \@ref(fig:GmaxAssocMLM)). The polygenic model indicated 0.17% [0-0.56] of the genome to be related to Gmax corresponding to 55 [0-177] SNPs, but the model didn’t identified specific SNPs in our dataset (no $\gamma>0.5$).


<!-- ```{r} -->
<!-- left_join(parsi, trees) %>% -->
<!--   dplyr::select(IID, Gmaxi_m, pop, TWI) %>% -->
<!--   unique() %>% -->
<!--   mutate(Gmax = Gmaxi_m/sd(Gmaxi_m)) %>%  -->
<!--   ggplot(aes(TWI, Gmax, col = pop)) + -->
<!--   geom_point() -->
<!-- ``` -->

### Proportion of Variance Explained

#### Animal

$$y_i \sim \mathcal{N}(\mu + u_i, \sigma_R)$$
$$u \sim \mathcal{N}_N(0, \sigma_GK)$$

```{r GmaxAnimal, fig.cap="Proportion of Variance Explained (PVE) by genetic in individual growth potential $Gmax$."}
Y <- left_join(parsi, trees) %>%
  mutate(popNum = as.numeric(as.factor(pop))) %>% 
  dplyr::select(IID, Gmaxi_m, pop, popNum) %>%
  unique() %>%
  mutate(Gmax = Gmaxi_m/sd(Gmaxi_m))
fam <- read_delim(file.path(path, "gemma", "Gmax0", "paracou3pop.fam"), delim = " ", 
                  col_names = c("FID", "IID", "PAT", "MAT", "SEX", "PHENOTYPE"))
K <- read_tsv(file.path(path, "gemma", "Gmax0", "paracou3pop.kinship.sXX.txt"), 
              col_names = fam$IID) %>% 
  mutate(IID = fam$IID) %>% 
  column_to_rownames(var = "IID") %>% 
  as.matrix()
# Yl <- list(paracou3pop = Y, sp1 = filter(Y, pop == "sp1"),
#           globuliferaTypeParacou = filter(Y, pop == "globuliferaTypeParacou"),
#           globuliferaTypeRegina = filter(Y, pop == "globuliferaTypeRegina"))
# model <- stan_model("./symcapture_models/Animal.stan")
# fitsAnimal$Gmax <- lapply(Yl, function(pop)
#   sampling(model, chains = 2, save_warmup = F, include = F, pars = "a",
#            control = list(adapt_delta = 0.99),
#            data = list(N = nrow(pop), Y = pop$Gmax, K = K[pop$IID,pop$IID])))
# save(fitsAnimal, file = file.path("symcapture_save", 'AnimalModels.Rdata'))
load(file.path("symcapture_save", 'AnimalModels.Rdata'))
cowplot::plot_grid(plotlist = lapply(fitsAnimal$Gmax, mcmc_combo, pars = c("sigmaG", "sigmaR")), 
                   labels = names(fitsAnimal$Gmax))
lapply(fitsAnimal$Gmax, broom::tidyMCMC,  pars = c("h2"), conf.int = T, rhat = T, estimate.method = "median") %>% 
  bind_rows(.id = "Population")
```


  #### Animal Pop Mean Var

$$y_{p,i} \sim \mathcal{N}(\mu_p + u_i, \sigma_R)$$
$$u \sim \mathcal{N}_N(0, \sum _p \sigma_{G_p} K_p)$$

```{r GmaxAnimalPopMeanVar, fig.cap="Proportion of Variance Explained (PVE) by genetic in individual growth potential $Gmax$."}
# Karray <- sapply(1:max(Y$popNum), function(p){
#   Kp = K[Y$IID,Y$IID]
#   for(n in 1:nrow(Y)) {
#     if(Y$popNum[n] != p) {
#       Kp[,n] = 10^-6
#       Kp[n,] = 10^-6
#     }
#   }
#   Kp <- as.matrix(Matrix::nearPD(Kp)$mat)
# })
# dim(Karray) <- c(nrow(Y), nrow(Y), max(Y$popNum))
# sigmaY <- group_by(Y, popNum) %>%
#   summarise(sigmaY = var(Gmax)) %>%
#   dplyr::select(sigmaY) %>% unlist()
# model <- stan_model("./symcapture_models/AnimalPopMeanVar.stan")
# fitsAnimal$Gmax <- sampling(model, chains = 2, save_warmup = F, include = F, pars = c("a", "G"),
#                             control = list(adapt_delta = 0.9),
#                             data = list(J = 1, N = nrow(Y), Y = Y$Gmax, sigmaY = sigmaY,
#                                         K = aperm(Karray, c(3,1,2)),
#                                         P = max(Y$popNum), pop = Y$popNum))
# save(fitsAnimal, file = file.path("symcapture_save", 'AnimalPopMeanVarModels.Rdata'))
load(file.path("symcapture_save", 'AnimalPopMeanVarModels.Rdata'))
mcmc_combo(as.array(fitsAnimal$Gmax ,  pars = c("sigmaP", "sigmaG", "sigmaR")))
broom::tidyMCMC(fitsAnimal$Gmax,  pars = c("sigmaP", "h2"), conf.int = T, rhat = T)
```

### Major effect SNPs

```{r GmaxAssocMLM, fig.cap="q-value of major effect SNPs associated to individual growth potential $Gmax$. SNP were detected using linear mixed models with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop) 
  read_tsv(file.path(path, "gemma", "Gmax", paste0(pop,".LMM.Gmax.assoc.txt"))) %>% 
    mutate(pop = pop)) %>% 
  bind_rows() %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  mutate(qval = p.adjust(p_wald, "fdr")) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  na.omit(qval) %>% 
  ggplot(aes(snp, -log10(qval), label = snp, alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ylab(expression(-log[10]("q-value"))) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type") +
  facet_wrap(~ pop) 
```

### Polygenic signal

```{r BSLMMGmaxConvergence, fig.cap="traceplot of BSLMM hyper-parameters chain."}
# lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop)
lapply(c("paracou3pop"), function(pop)
  read_tsv(file.path(path, "gemma", "Gmax", paste0(pop,".BSLMM.Gmax.hyp.txt"))) %>% 
    mutate(Population = pop)) %>% 
  bind_rows()  %>% 
  dplyr::select(Population, pge, pi, n_gamma) %>% 
  dplyr::rename(PGE = pge, `n[gamma]` = n_gamma) %>% 
  group_by(Population) %>% 
  mutate(iteration = 1:n()) %>% 
  filter(row_number() %% 100 == 0) %>% 
  reshape2::melt(id.vars = c("Population", "iteration"), variable.name = "parameter") %>% 
  ggplot(aes(iteration, value)) +
  geom_line() +
  facet_grid(parameter ~ Population, scales = "free", labeller = "label_parsed") +
  theme(axis.text.x = element_text(angle = 45))
```

```{r BSLMMGmaxHypPar}
read_tsv(file.path(path, "gemma", "Gmax", paste0("paracou3pop",".BSLMM.Gmax.hyp.txt"))) %>% 
  dplyr::select(pge, pi, n_gamma) %>% 
  dplyr::rename(PGE = pge, `n[gamma]` = n_gamma) %>% 
  mutate(pi = pi*100) %>% 
  broom::tidyMCMC(conf.int = T) %>% 
  kable(caption = "BSLMM hyper-parameters distribution for the whole genus.")
```

```{r GmaxAssocBSLMM, fig.cap="Effect of SNPs associated to individual growth potential $Gmax$ with a polygenic model. SNP were detected using bayesian sparse linear mixed models (BSLMM) with individual kiniship as random effect. SNP significance was assessed with the $\\gamma$ parameter and highlighted here with SNPs having at least 90% chance of having an effect on the phenotype."}
read_tsv(file.path(path, "gemma", "Gmax", paste0("paracou3pop",".BSLMM.Gmax.param.txt"))) %>% 
  mutate(Population = "paracou3pop") %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  ggplot(aes(snp, beta*gamma, label = snp, col = type, alpha = gamma > 0.5)) +
  geom_point() +
  scale_y_continuous(trans="S_sqrt") +
  ylab(expression(beta*gamma))
```

## Optimal growth diameter (Dopt)

```{bash DoptGemma, eval=F, echo=T}
# left_join(parsi, trees) %>%
#   dplyr::select(FID, IID, Dopti_m) %>%
#   unique() %>%
#   mutate(Dopt = RNOmni::rankNorm(Dopti_m)) %>%
#   write_tsv(file.path(path, "gemma", "Dopt", "Dopt.phenotype"), col_names = F)
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
cat paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > paracou3pop.fam
pops=(sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  grep $pop paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > $pop.fam ; done
pops=(paracou3pop sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  
  plink \
  --bfile ../../../variantCalling/paracou3pop/symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr \
  --keep $pop.fam \
  --extract ../../../variantCalling/paracou3pop/LD.prune.in \
  --maf 0.05 \
  --pheno Dopt.phenotype \
  --out $pop \
  --make-bed ;
done  
for pop in "${pops[@]}" ; do  gemma -bfile $pop -gk 1 -o $pop.kinship ; done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k output/$pop.kinship.cXX.txt -lmm 1 -o $pop.LMM.Dopt ; done
gemma -bfile paracou3pop -k -bslmm 1 -o paracou3pop.BSLMM.Dopt
cat Dopt.BSLMM.snps | cut -f1 > Dopt.snps.list
plink --bfile paracou3pop --allow-extra-chr --extract Dopt.snps.list --out Dopt.snps --recode A
plink --bfile ../../../variantCalling/paracou/symcapture.all.biallelic.snp.filtered.nonmissing.paracou \
  --allow-extra-chr --extract Dopt.snps.list --out Dopt.snps.paracou --recode A
```

Individual optimal diameter is mainly explained by genetic variation ($\sigma_G=0.68\pm0.14$, Fig. \@ref(fig:DoptPVE)). $Dopt$ is thus an ontogenetic factor as it controls the shape of the growth response to ontogeny and its controlled by genetics ! Individual optimal diameter wasn't significantly associated to any  major effect SNPs (Fig. \@ref(fig:DoptAssocMLM)). The polygenic model indicated 0.07% [0-0.42] of the genome to be related to Dopt corresponding to 24 [1-136] SNPs, and identified one neutral SNPs assocaited to individual optimal diameter ($\gamma>0.5$, Fig. \@ref(fig:DoptAssocBSLMM) and Fig. \@ref(fig:DoptOutliersHist)), already highlighted as a population specific SNP (see populations genomic).

### Proportion of Variance Explained

#### Animal

$$y_i \sim \mathcal{N}(\mu + u_i, \sigma_R)$$
$$u \sim \mathcal{N}_N(0, \sigma_GK)$$

```{r DoptAnimal, fig.cap="Proportion of Variance Explained (PVE) by genetic in individual growth potential $Dopt$."}
Y <- left_join(parsi, trees) %>%
  mutate(popNum = as.numeric(as.factor(pop))) %>% 
  dplyr::select(IID, Dopti_m, pop, popNum) %>%
  unique() %>%
  mutate(Dopt = Dopti_m/sd(Dopti_m))
fam <- read_delim(file.path(path, "gemma", "Gmax0", "paracou3pop.fam"), delim = " ", 
                  col_names = c("FID", "IID", "PAT", "MAT", "SEX", "PHENOTYPE"))
K <- read_tsv(file.path(path, "gemma", "Gmax0", "paracou3pop.kinship.sXX.txt"), 
              col_names = fam$IID) %>% 
  mutate(IID = fam$IID) %>% 
  column_to_rownames(var = "IID") %>% 
  as.matrix()
# Yl <- list(paracou3pop = Y, sp1 = filter(Y, pop == "sp1"),
#           globuliferaTypeParacou = filter(Y, pop == "globuliferaTypeParacou"),
#           globuliferaTypeRegina = filter(Y, pop == "globuliferaTypeRegina"))
# model <- stan_model("./symcapture_models/Animal.stan")
# fitsAnimal$Dopt <- lapply(Yl, function(pop)
#   sampling(model, chains = 2, save_warmup = F, include = F, pars = "a",
#            control = list(adapt_delta = 0.99),
#            data = list(N = nrow(pop), Y = pop$Dopt, K = K[pop$IID,pop$IID])))
# save(fitsAnimal, file = file.path("symcapture_save", 'AnimalModels.Rdata'))
load(file.path("symcapture_save", 'AnimalModels.Rdata'))
cowplot::plot_grid(plotlist = lapply(fitsAnimal$Dopt, mcmc_combo, pars = c("sigmaG", "sigmaR")), 
                   labels = names(fitsAnimal$Dopt))
lapply(fitsAnimal$Dopt, broom::tidyMCMC,  pars = c("h2"), conf.int = T, rhat = T, estimate.method = "median") %>% 
  bind_rows(.id = "Population")
```

#### Animal Pop Mean Var

$$y_{p,i} \sim \mathcal{N}(\mu_p + u_i, \sigma_R)$$
$$u \sim \mathcal{N}_N(0, \sum _p \sigma_{G_p} K_p)$$

```{r DoptAnimalPopMeanVar, fig.cap="Proportion of Variance Explained (PVE) by genetic in individual growth potential $Dopt$."}
# Karray <- sapply(1:max(Y$popNum), function(p){
#   Kp = K[Y$IID,Y$IID]
#   for(n in 1:nrow(Y)) {
#     if(Y$popNum[n] != p) {
#       Kp[,n] = 10^-6
#       Kp[n,] = 10^-6
#     }
#   }
#   Kp <- as.matrix(Matrix::nearPD(Kp)$mat)
# })
# dim(Karray) <- c(nrow(Y), nrow(Y), max(Y$popNum))
# sigmaY <- group_by(Y, popNum) %>%
#   summarise(sigmaY = var(Dopt)) %>%
#   dplyr::select(sigmaY) %>% unlist()
# model <- stan_model("./symcapture_models/AnimalPopMeanVar.stan")
# fitsAnimal$Dopt <- sampling(model, chains = 2, save_warmup = F, include = F, pars = c("a", "G"),
#                             control = list(adapt_delta = 0.9),
#                             data = list(J = 1, N = nrow(Y), Y = Y$Dopt, sigmaY = sigmaY,
#                                         K = aperm(Karray, c(3,1,2)),
#                                         P = max(Y$popNum), pop = Y$popNum))
# save(fitsAnimal, file = file.path("symcapture_save", 'AnimalPopMeanVarModels.Rdata'))
load(file.path("symcapture_save", 'AnimalPopMeanVarModels.Rdata'))
mcmc_combo(as.array(fitsAnimal$Dopt ,  pars = c("sigmaP", "sigmaG", "sigmaR")))
broom::tidyMCMC(fitsAnimal$Dopt,  pars = c("sigmaP", "h2"), conf.int = T, rhat = T)
```

### Major effect SNPs

```{r DoptAssocMLM, fig.cap="q-value of major effect SNPs associated to individual optimal diameter $Dopt$. SNP were detected using linear mixed models with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
t <- lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop) 
  read_tsv(file.path(path, "gemma", "Dopt", paste0(pop,".LMM.Dopt.assoc.txt"))) %>% 
    mutate(pop = pop)) %>% 
  bind_rows() %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  mutate(qval = p.adjust(p_wald, "fdr")) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  na.omit(qval)
# filter(t , qval < 0.05) %>%
#   mutate(model = "LMM") %>%
#   dplyr::select(rs, pop, model) %>%
#   write_tsv(file.path(path, "gemma", "Dopt", "Dopt.LMM.snps"), col_names = F)
ggplot(t, aes(snp, -log10(qval), label = snp, alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ylab(expression(-log[10]("q-value"))) +
  ggrepel::geom_text_repel(data = t[t$qval < 0.05,]) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type") +
  facet_wrap(~ pop) 
```

### Polygenic signal

```{r BSLMMDoptConvergence, fig.cap="traceplot of BSLMM hyper-parameters chain."}
lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop)
  read_tsv(file.path(path, "gemma", "Dopt", paste0(pop,".BSLMM.Dopt.hyp.txt"))) %>% 
    mutate(Population = pop)) %>% 
  bind_rows()  %>% 
  dplyr::select(Population, pve, pge, pi, n_gamma) %>% 
  dplyr::rename(PVE = pve, PGE = pge, `n[gamma]` = n_gamma) %>% 
  group_by(Population) %>% 
  mutate(iteration = 1:n()) %>% 
  filter(row_number() %% 100 == 0) %>% 
  reshape2::melt(id.vars = c("Population", "iteration"), variable.name = "parameter") %>% 
  ggplot(aes(iteration, value)) +
  geom_line() +
  facet_grid(parameter ~ Population, scales = "free", labeller = "label_parsed") +
  theme(axis.text.x = element_text(angle = 45))
```

```{r BSLMMDoptHypPar}
read_tsv(file.path(path, "gemma", "Dopt", paste0("paracou3pop",".BSLMM.Dopt.hyp.txt"))) %>% 
  dplyr::select(pge, pi, n_gamma) %>% 
  dplyr::rename(PGE = pge, `n[gamma]` = n_gamma) %>% 
  mutate(pi = pi*100) %>% 
  broom::tidyMCMC(conf.int = T) %>% 
  kable(caption = "BSLMM hyper-parameters distribution for the whole genus.")
```

```{r DoptAssocBSLMM, fig.cap="Effect of SNPs associated to individual growth optimal diameter $Dopt$ with a polygenic model. SNP were detected using bayesian sparse linear mixed models (BSLMM) with individual kiniship as random effect. SNP significance was assessed with the $\\gamma$ parameter and highlighted here with SNPs having at least 90% chance of having an effect on the phenotype."}
t <- read_tsv(file.path(path, "gemma", "Dopt", paste0("paracou3pop",".BSLMM.Dopt.param.txt"))) %>% 
  mutate(Population = "paracou3pop") %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.))
filter(t , gamma > 0.5) %>%
  mutate(model = "BSLMM", pop = "paracou3pop") %>%
  dplyr::select(rs, pop, model) %>%
  write_tsv(file.path(path, "gemma", "Dopt", "Dopt.BSLMM.snps"), col_names = F)
ggplot(t, aes(snp, beta*gamma, label = rs, col = type, alpha = gamma > 0.5)) +
  geom_point() +
  ggrepel::geom_text_repel(data = t[t$gamma > 0.1,]) +
  scale_y_continuous(trans="S_sqrt") +
  ylab(expression(beta*gamma))
```

### Outliers

```{r DoptOutliersHist, fig.cap="Distribution of alleles along individual optiaml diameter $Dopt$ for significantly associated SNPs."}
read_delim(file.path(path, "gemma", "Dopt", "Dopt.snps.raw"), delim = " ") %>% 
  filter(PHENOTYPE != -9) %>% # !! TO SOLVE
  dplyr::select(-FID, -PAT, -MAT, -SEX) %>% 
  dplyr::rename(Dopt = PHENOTYPE) %>% 
  reshape2::melt(id.vars = c("IID", "Dopt"), variable.name = "SNP", value.name = "Allele") %>% 
  na.omit(Allele) %>% 
  mutate(Allele = as.factor(Allele)) %>% 
  left_join(trees, by = "IID", suffix = c("", ".trees")) %>% 
  mutate(SNP = stringr::str_sub(SNP, 1, -3)) %>% 
  ggplot(aes(Allele, Dopt)) +
  geom_boxplot() +
  stat_summary(fun.y=mean, geom="line", col = "red", size = 1.2,
               aes(group=1))  + 
  facet_wrap(~ SNP) +
  coord_flip() +
  theme(strip.text.x = element_text(size = 5)) +
  ylab("Individual optimal diameter") +
  xlab("Number of copies of the major allele") 
```

## Growth kurtosis (Ks)

```{bash KsGemma, eval=F, echo=T}
# left_join(parsi, trees) %>%
#   dplyr::select(FID, IID, Ksi_m) %>%
#   unique() %>%
#   mutate(Ks = RNOmni::rankNorm(Ksi_m)) %>%
#   write_tsv(file.path(path, "gemma", "Ks", "Ks.phenotype"), col_names = F)
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
cat paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > paracou3pop.fam
pops=(sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  grep $pop paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > $pop.fam ; done
pops=(paracou3pop sp1 globuliferaTypeParacou globuliferaTypeRegina)
for pop in "${pops[@]}" ; do  
  plink \
  --bfile ../../../variantCalling/paracou3pop/symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr \
  --keep $pop.fam \
  --extract ../../../variantCalling/paracou3pop/LD.prune.in \
  --maf 0.05 \
  --pheno Ks.phenotype \
  --out $pop \
  --make-bed ;
done  
for pop in "${pops[@]}" ; do  gemma -bfile $pop -gk 1 -o $pop.kinship ; done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k output/$pop.kinship.cXX.txt -lmm 1 -o $pop.LMM.Ks ; done
for pop in "${pops[@]}" ; do  gemma -bfile $pop -k -bslmm 1 -o $pop.BSLMM.Ks & done
```

Individual growth kurtosis is partly explained by genetic variation ($\sigma_G=0.30\pm0.13$, Fig. \@ref(fig:KsPVE)). Individual growth kurtosis wasn't significantly associated to any  major effect SNPs across population (Fig. \@ref(fig:KsAssocMLM)). The polygenic model indicated 0.12% [0-0.74] of the genome to be related to Ks corresponding to 39 [0-243] SNPs, but the model didn’t identified specific SNPs in our dataset (no $\gamma>0.5$).

### Proportion of Variance Explained

#### Animal

$$y_i \sim \mathcal{N}(\mu + u_i, \sigma_R)$$
$$u \sim \mathcal{N}_N(0, \sigma_GK)$$

```{r KsAnimal, fig.cap="Proportion of Variance Explained (PVE) by genetic in individual growth potential $Ks$."}
Y <- left_join(parsi, trees) %>%
  mutate(popNum = as.numeric(as.factor(pop))) %>% 
  dplyr::select(IID, Ksi_m, pop, popNum) %>%
  unique() %>%
  mutate(Ks = Ksi_m/sd(Ksi_m))
fam <- read_delim(file.path(path, "gemma", "Gmax0", "paracou3pop.fam"), delim = " ", 
                  col_names = c("FID", "IID", "PAT", "MAT", "SEX", "PHENOTYPE"))
K <- read_tsv(file.path(path, "gemma", "Gmax0", "paracou3pop.kinship.sXX.txt"), 
              col_names = fam$IID) %>% 
  mutate(IID = fam$IID) %>% 
  column_to_rownames(var = "IID") %>% 
  as.matrix()
# Yl <- list(paracou3pop = Y, sp1 = filter(Y, pop == "sp1"),
#           globuliferaTypeParacou = filter(Y, pop == "globuliferaTypeParacou"),
#           globuliferaTypeRegina = filter(Y, pop == "globuliferaTypeRegina"))
# model <- stan_model("./symcapture_models/Animal.stan")
# fitsAnimal$Ks <- lapply(Yl, function(pop)
#   sampling(model, chains = 2, save_warmup = F, include = F, pars = "a",
#            control = list(adapt_delta = 0.99),
#            data = list(N = nrow(pop), Y = pop$Ks, K = K[pop$IID,pop$IID])))
# save(fitsAnimal, file = file.path("symcapture_save", 'AnimalModels.Rdata'))
load(file.path("symcapture_save", 'AnimalModels.Rdata'))
cowplot::plot_grid(plotlist = lapply(fitsAnimal$Ks, mcmc_combo, pars = c("sigmaG", "sigmaR")), 
                   labels = names(fitsAnimal$Ks))
lapply(fitsAnimal$Ks, broom::tidyMCMC,  pars = c("h2"), conf.int = T, rhat = T, estimate.method = "median") %>% 
  bind_rows(.id = "Population")
```

#### Animal Pop Mean Var

$$y_{p,i} \sim \mathcal{N}(\mu_p + u_i, \sigma_R)$$
$$u \sim \mathcal{N}_N(0, \sum _p \sigma_{G_p} K_p)$$

```{r KsAnimalPopMeanVar, fig.cap="Proportion of Variance Explained (PVE) by genetic in individual growth potential $Ks$."}
# Karray <- sapply(1:max(Y$popNum), function(p){
#   Kp = K[Y$IID,Y$IID]
#   for(n in 1:nrow(Y)) {
#     if(Y$popNum[n] != p) {
#       Kp[,n] = 10^-6
#       Kp[n,] = 10^-6
#     }
#   }
#   Kp <- as.matrix(Matrix::nearPD(Kp)$mat)
# })
# dim(Karray) <- c(nrow(Y), nrow(Y), max(Y$popNum))
# sigmaY <- group_by(Y, popNum) %>%
#   summarise(sigmaY = var(Ks)) %>%
#   dplyr::select(sigmaY) %>% unlist()
# model <- stan_model("./symcapture_models/AnimalPopMeanVar.stan")
# fitsAnimal$Ks <- sampling(model, chains = 2, save_warmup = F, include = F, pars = c("a", "G"),
#                             control = list(adapt_delta = 0.9),
#                             data = list(J = 1, N = nrow(Y), Y = Y$Ks, sigmaY = sigmaY,
#                                         K = aperm(Karray, c(3,1,2)),
#                                         P = max(Y$popNum), pop = Y$popNum))
# save(fitsAnimal, file = file.path("symcapture_save", 'AnimalPopMeanVarModels.Rdata'))
load(file.path("symcapture_save", 'AnimalPopMeanVarModels.Rdata'))
mcmc_combo(as.array(fitsAnimal$Ks ,  pars = c("sigmaP", "sigmaG", "sigmaR")))
broom::tidyMCMC(fitsAnimal$Ks,  pars = c("sigmaP", "h2"), conf.int = T, rhat = T)
```

### Major effect SNPs

```{r KsAssocMLM, fig.cap="q-value of major effect SNPs associated to individual kurtosis $Ks$. SNP were detected using linear mixed models with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
t <- lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop) 
  read_tsv(file.path(path, "gemma", "Ks", paste0(pop,".LMM.Ks.assoc.txt"))) %>% 
    mutate(pop = pop)) %>% 
  bind_rows() %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  mutate(qval = p.adjust(p_wald, "fdr")) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  na.omit(qval)
ggplot(t, aes(snp, -log10(qval), label = snp, alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ylab(expression(-log[10]("q-value"))) +
  ggrepel::geom_text_repel(data = t[t$qval < 0.05,]) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type") +
  facet_wrap(~ pop) 
```

### Polygenic signal

```{r BSLMMKsConvergence, fig.cap="traceplot of BSLMM hyper-parameters chain."}
# lapply(c("paracou3pop", "sp1", "globuliferaTypeParacou", "globuliferaTypeRegina"), function(pop) 
lapply(c("paracou3pop"), function(pop) 
  read_tsv(file.path(path, "gemma", "Ks", paste0(pop,".BSLMM.Ks.hyp.txt"))) %>% 
    mutate(Population = pop)) %>% 
  bind_rows()  %>% 
  dplyr::select(Population, pve, pge, pi, n_gamma) %>% 
  dplyr::rename(PVE = pve, PGE = pge, `n[gamma]` = n_gamma) %>% 
  group_by(Population) %>% 
  mutate(iteration = 1:n()) %>% 
  filter(row_number() %% 100 == 0) %>% 
  reshape2::melt(id.vars = c("Population", "iteration"), variable.name = "parameter") %>% 
  ggplot(aes(iteration, value)) +
  geom_line() +
  facet_grid(parameter ~ Population, scales = "free", labeller = "label_parsed") +
  theme(axis.text.x = element_text(angle = 45))
```

```{r BSLMMKsHypPar}
read_tsv(file.path(path, "gemma", "Ks", paste0("paracou3pop",".BSLMM.Ks.hyp.txt"))) %>% 
  dplyr::select(pge, pi, n_gamma) %>% 
  dplyr::rename(PGE = pge, `n[gamma]` = n_gamma) %>% 
  mutate(pi = pi*100) %>% 
  broom::tidyMCMC(conf.int = T) %>% 
  kable(caption = "BSLMM hyper-parameters distribution for the whole genus.")
```

```{r KsAssocBSLMM, fig.cap="Effect of SNPs associated to individual kurtosis $Ks$ with a polygenic model. SNP were detected using bayesian sparse linear mixed models (BSLMM) with individual kiniship as random effect. SNP significance was assessed with the $\\gamma$ parameter and highlighted here with SNPs having at least 90% chance of having an effect on the phenotype.", eval=F}
read_tsv(file.path(path, "gemma", "Ks", paste0("paracou3pop",".polygenic.Ks.param.txt"))) %>% 
  mutate(Population = "paracou3pop") %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  ggplot(aes(snp, beta*gamma, label = snp, col = type, alpha = gamma > 0.5)) +
  geom_point() +
  scale_y_continuous(trans="S_sqrt") +
  ylab(expression(beta*gamma))
```

## Conclusion

Growth individual parameters related to ontogeny, i.e. maximum growth potential $Gmax$, optimal diameter $Dopt$ and growth kurtosis $Ks$, are mainly explained by genetic variation ($\sigma_G$ equal $0.72\pm0.16$, $0.68\pm0.14$, and $0.30\pm0.13$, respectively for $Gmax$, $Dopt$, and $Ks$). They are consequently ontegenetic parameters as they define the growth response to individual development and they are mainly geneticaly determined. $Ks$ is less explained by genetics, as it express the species plasticity in growth responce to development, and might thus be also linked to light access. Any ontogenetic parameters were related to major effect SNPs in LMM models. All ontogenetic parameters had a low to null polygenic signature (less than 0.17% [0-0.7] of the genome representing 55 [0-243] SNPs). Finally, only Dopt was associated to one neutral SNP with strong effect, already highlighted as a population specific SNP (see populations genomic).
