```{r setup_growthgenom, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(bayesplot)
library(rstan)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
load("./symcapture_save/Env2.Rdata")
load(file.path("symcapture_save", 'GrowthModels.Rdata'))
```

# Growth genomics

We will ... :

* __xxx__ ...

## Variables

We explored the role of individual size, abitoic environment, biotic interactions, and individual genotype on individual tree growth in space and time. Individual tree growth was calculated with following formula:

$AGR_{i,t} = \frac{DBH_{i,t} - DBH_{i,t-1}}{\Delta t}$

with the annual growth rate $AGR$ of individual $i$ in year $t$ depended of individual $DBH$ at year $t$ and $t-1$ and the timelaps between the two year $\Delta t$.

The topographic wetness index ($TWI_i$) was selected among several spatial abiotic descriptors as a proxy of water accumulation. Waterlogging and topography have been highlighted as crucial for forest dynamics [@ferry2010higher] and species habitat preference [@Allie2015] at the Paracou study site. TWI was derived from a 1-m resolution digital elevation model using SAGA-GIS [@Conrad2015] based on a LiDAR campaign done in 2015.

The area under the curve (AUC) of the relative extractible water [$REW_t$, @Wagner2011] for a threshold of 0.4 was selected among several temporal abiotic descriptors as a major climatic driver of tropical forest dynamics [@Aubry-Kientz2015a]. AUC was **...**

Biotic descriptors included two spatial and temporal variables: the Neighborhood Crowding Index [NCI; @Uriarte2004] and the Annual Crowding Rate. The Neighborhood Crowding Index $NCI_{i,t}$ from tree individual $i$ in year $t$ was calculated with following formula:

$$NCI_i = \sum _{j~|~\delta_{i,j}<20m} ^{J_{i,t}} DBH_{j,t}^2 e^{-\frac14.\delta_{i,j}}$$

with $DBH_{j,t}$ the diameter from neighboring tree $j$ in year $t$ and $\delta_{i,j}$ its distance to individual tree $i$. $NCI_i$ is computed for all neighbors at a distance $\delta_{i,j}$ inferior to maximum neighboring distance of 20 meters. The power of neighbors $DBH$ effect was set to 2 to consider neighbors surface. The decrease of neighbors $DBH$ effect with distance was set to 0.25 here to represent trees at 20 meters of the focal trees having 1% of the effect of the same tree at 0  meters. $NCI$ represents biotic asymmetric competition through tree neighborhood over time. 

The Annual Neighborhood Crowding Rate $ANCR_{i,t}$ from tree individual $i$ in year $t$ was calculated with following formula:

$ANCR_{i,t} = \frac{NCI_{i,t} - NCI_{i,t-1}}{\Delta t}$

$ANCR$ represents the dynamic of biotic asymmetric competition through tree neighborhood over time (derivate of $NCI_i$). Both NCI and ANCR were calculated among heterospecific and conspecific neighbors and with all neighbors confounded. Edging effect have been corrected by weighing values from individual close to the border by the inverse of the surface outside of the plot in the 20m radius.

Both correlations and principal component analyses showed non colinear DBH, TWI, AUC, NCI and ANCR that we selected for further analyses (Fig. \@ref(fig:correlationsGrowthGenom) and Fig. \@ref(fig:pcaGrowthGenom)).

```{r DataGrowthGenom, eval=F}
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Symphonia") %>% 
  filter(CensusYear < 2016) %>% 
  collect()
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou3pop",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop")))
treesXY <- trees
coordinates(treesXY) <- ~Xutm + Yutm
proj4string(treesXY) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
treesXY <- spTransform(treesXY, CRSobj = crs)
wetness <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "TWI_1m.tif"))
dem <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
wetness <- projectRaster(wetness, crs = crs)
trees$TWI <- raster::extract(wetness, treesXY)
rm(dem, wetness, treesXY)
weather <- read_csv(file.path("../../../data/Paracou/", "weather", "REW_1978-2017_Fabien.csv")) %>% 
  dplyr::select(-X1, -day, -month, day_julian) %>% 
  mutate(AUC = ifelse(REW > 0.4, 0, 0.4-REW)) %>% 
  group_by(year) %>% 
  summarise_all(mean) %>% 
  dplyr::rename(CensusYear = year)
trees <- left_join(trees, weather)
rm(weather)
cl <- parallel::makeCluster(getOption("cl.cores", 4))
parallel::clusterExport(cl, list("trees"))
NCI <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
    tbl("Paracou") %>% 
    filter(CensusYear == local(trees$CensusYear[ind])) %>% 
    filter(Plot == local(trees$Plot[ind])) %>% 
    filter(idTree != local(trees$idTree[ind])) %>% 
    mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
    filter(dij < 20) %>% 
    mutate(con = ifelse(Genus == local(trees$Genus[ind]) && Species == local(trees$Species[ind]), 1, 0)) %>% 
    mutate(DBH = CircCorr/pi) %>% 
    summarise(idTree = local(trees$idTree[ind]),
              CensusYear  == local(trees$CensusYear[ind],
              NCIcon = sum(DBH*DBH*exp(-0.25*dij)*con),
              NCIhetero = sum(DBH*DBH*exp(-0.25*dij)*(1-con))) %>% 
    collect()})
parallel::stopCluster(cl) ; rm(cl)
NCI <- bind_rows(NCI)
trees <- left_join(trees, NCI)
rm(NCI)
trees <- trees %>% 
  rowwise() %>% 
  mutate(NCI = sum(NCIcon, NCIhetero))
trees <- trees %>% 
  mutate(DBH = CircCorr/pi) %>% 
  group_by(idTree) %>% 
  arrange(CensusYear) %>% 
  mutate(AGR = (DBH-lag(DBH))/(CensusYear-lag(CensusYear)),
         ANCRcon = (NCIcon-lag(NCIcon))/(CensusYear-lag(CensusYear)),
         ANCRhetero = (NCIhetero-lag(NCIhetero))/(CensusYear-lag(CensusYear)),
         ANCR = (NCI-lag(NCI))/(CensusYear-lag(CensusYear))) %>% 
  mutate(AGR = ifelse(AGR < 0, NA, AGR)) %>% 
  ungroup()
      load("../../functional/functional_save/CompetitionMatrix.Rdata")
trees <- trees %>% 
  left_join(dplyr::select(Competition, idTree, AreaOutside20) %>% 
              unique()) %>% 
  mutate_at(c("NCIcon", "NCIhetero", "NCI", "ANCRcon", "ANCRhetero", "ANCR"), funs(ifelse(AreaOutside20 != 0, ./AreaOutside20, .)))
rm(Competition)
trees <- trees %>% 
  group_by(idTree) %>% 
  arrange(CensusYear) %>% 
  mutate(logNCI = log(NCI), 
         logANCR = (log(NCI)-lag(log(NCI)))/(CensusYear-lag(CensusYear))) %>% 
  ungroup()
trees %>% 
  ungroup() %>% 
  filter(CensusYear == 2015) %>%  # for the moment for tests with gemma 
  dplyr::select(logDBH, TWI) %>% 
  mutate_all(scale) %>% 
  write_tsv(file.path(path, "gemma", "AGR", "AGR.covariates"), col_names = F)
trees %>% 
  ungroup() %>% 
  filter(CensusYear == 2015) %>%  # for the moment for tests with gemma 
  mutate(AGR = log(AGR+1)) %>% 
  dplyr::select(FID, IID, AGR) %>% 
  write_tsv(file.path(path, "gemma", "AGR", "AGR.phenotype"), col_names = F)
save(trees, file = "./symcapture_save/Env2.Rdata")
```

```{r correlationsGrowthGenom, fig.cap="Environmental descriptors correlations."}
trees %>% 
  dplyr::select(DBH, TWI, AUC, logNCI, logANCR) %>% 
  na.omit() %>% 
  cor(.) %>% 
  corrplot::corrplot.mixed()
```

```{r pcaGrowthGenom, fig.cap="Environmental descriptors principal component analysis."}
factoextra::fviz_pca_var(princomp(~ DBH + TWI + AUC + logNCI + logANCR, trees, cor = T),
                         axes = c(1, 2), geom = c("arrow", "text"), col.var = "contrib")
```

```{r mdata}
trees <- trees %>% 
  filter(!is.na(AGR), !is.na(ANCR)) %>% 
  mutate(IndNum = as.numeric(as.factor(Ind)), popNum = as.numeric(as.factor(pop)))
fits <- list()
mdata <- list(
  N = nrow(trees),
  I = length(unique(trees$Ind)),
  G = length(unique(trees$pop)),
  AGR = trees$AGR,
  DBH = trees$DBH,
  TWI = trees$TWI,
  AUC = trees$AUC,
  logNCI = trees$logNCI,
  logANCR = trees$logANCR,
  ind = trees$IndNum,
  gp = trees$popNum,
  indingp = unique(arrange(trees, IndNum)[c("IndNum", "popNum")])$popNum
)
# lapply(mdata, summary)
```
## Growth & Ontogeny

We explored different way of integrating gene pools and individuals structure in the Gompertz model used in @Herault2011 to explain individual annual growth rate (AGR) with individual diameter at breast height (DBH). I used three forms of models:

$$GrowthNull:log(AGR_{t,i,g}+1) \sim \mathcal N (Gmax.exp(\frac12.[\frac{log(\frac{DBH_{t,i,g}}{Dopt})}{Ks}]^2), \sigma)$$
$$GrowthGenePool:log(AGR_{t,i,g}+1) \sim \mathcal N (Gmax_g.exp(\frac12.[\frac{log(\frac{DBH_{t,i,g}}{Dopt_g})}{Ks_g}]^2), \sigma)$$
$$GrowthIndividual:log(AGR_{t,i,g}+1) \sim \mathcal N (Gmax_i.exp(\frac12.[\frac{log(\frac{DBH_{t,i,g}}{Dopt_i})}{Ks_i}]^2), \sigma) \\ \begin{bmatrix} Gmax_i \\ Dopt_i \\ Ks_i \end{bmatrix} \sim \mathcal N ^3(\begin{bmatrix} Gmax_g \\ Dopt_g \\ Ks_g \end{bmatrix},\begin{bmatrix} \sigma_{Gmax} \\ \sigma_{Dopt} \\ \sigma_{Ks} \end{bmatrix} )$$
with ...

```{r growthOntogenyTab}
# models <- lapply(1:3, function(i) stan_model(paste0("./symcapture_models/Growth", i, ".stan")))
# fits <- lapply(models, function(model) sampling(model, chains = 2, data = mdata, save_warmup = F))
# names(fits) <- paste0("growth", 0:2)
# save(fits, file = file.path("symcapture_save", 'GrowthModels.Rdata'))
load(file.path("symcapture_save", 'GrowthModels.Rdata'))
# lapply(fits, function(fit) mcmc_trace(as.array(fit), pars = c("Gmax[1]", "Dopt[1]", "Ks[1]", "sigma"),
#                                       facet_args = list(labeller = label_parsed)))
bind_rows(broom::tidyMCMC(fits$growth0, fits$growth0@model_pars, droppars = NULL, rhat = T) %>% 
            mutate(model = "GrowthNull"),
          broom::tidyMCMC(fits$growth1, fits$growth1@model_pars, droppars = NULL, rhat = T) %>% 
            mutate(model = "GrowthGenePool"),
          broom::tidyMCMC(fits$growth2, fits$growth1@model_pars, droppars = NULL, rhat = T) %>% 
            mutate(model = "GrowthIndividual")) %>% 
  reshape2::dcast(term ~ model, value.var = "estimate") %>% 
  mutate(term = gsub("([[:punct:]])", "", term)) %>% 
  mutate(popNum = gsub("([[:alpha:]])", "", term)) %>% 
  mutate(term = gsub("([[:digit:]])", "", term)) %>% 
  mutate(popNum = as.numeric(popNum)) %>% 
  left_join(unique(dplyr::select(trees, popNum, pop))) %>% 
  dplyr::select(term, pop, GrowthNull, GrowthGenePool, GrowthIndividual) %>% 
  kable(caption = "Summary table of the models")
```

```{r growthOntogenyG, fig.cap="Growth models predictions."}
load(file.path("symcapture_save", 'GrowthModels.Rdata'))
pars0 <- as.data.frame(fits$growth0)
pars1 <- as.data.frame(fits$growth1, pars = c("Gmax", "Dopt", "Ks")) %>% 
  reshape2::melt(variable.name = "parameter") %>% 
  group_by(parameter) %>% 
  summarise(q5 = quantile(value, 0.05), m = mean(value), q95 = quantile(value, 0.95)) %>% 
  mutate(parameter = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(popNum = gsub("([[:alpha:]])", "", parameter)) %>% 
  mutate(parameter = gsub("([[:digit:]])", "", parameter)) %>% 
  reshape2::melt(id.vars = c("parameter", "popNum")) %>% 
  mutate(parameter = paste0(parameter, "_", variable)) %>% 
  dplyr::select(-variable) %>%
  reshape2::dcast(popNum ~ parameter) %>% 
  mutate(popNum = as.numeric(popNum))
pars2 <- as.data.frame(fits$growth2, pars = c("Gmax", "Dopt", "Ks")) %>% 
  reshape2::melt(variable.name = "parameter") %>% 
  group_by(parameter) %>% 
  summarise(q5 = quantile(value, 0.05), m = mean(value), q95 = quantile(value, 0.95)) %>% 
  mutate(parameter = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(popNum = gsub("([[:alpha:]])", "", parameter)) %>% 
  mutate(parameter = gsub("([[:digit:]])", "", parameter)) %>% 
  reshape2::melt(id.vars = c("parameter", "popNum")) %>% 
  mutate(parameter = paste0(parameter, "_", variable)) %>% 
  dplyr::select(-variable) %>%
  reshape2::dcast(popNum ~ parameter) %>% 
  mutate(popNum = as.numeric(popNum))
pars2i <- as.data.frame(fits$growth2, pars = c("Gmaxi", "Dopti", "Ksi")) %>% 
  reshape2::melt(variable.name = "parameter") %>% 
  group_by(parameter) %>% 
  summarise(q5 = quantile(value, 0.05), m = mean(value), q95 = quantile(value, 0.95)) %>% 
  mutate(parameter = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(IndNum = gsub("([[:alpha:]])", "", parameter)) %>% 
  mutate(parameter = gsub("([[:digit:]])", "", parameter)) %>% 
  reshape2::melt(id.vars = c("parameter", "IndNum")) %>% 
  mutate(parameter = paste0(parameter, "_", variable)) %>% 
  dplyr::select(-variable) %>%
  reshape2::dcast(IndNum ~ parameter) %>% 
  mutate(IndNum = as.numeric(IndNum))
bind_rows(data.frame(DBH = rep(1:100, mdata$I), 
                     IndNum = rep(1:mdata$I, each = 100)) %>% 
            left_join(unique(dplyr::select(trees, IndNum, Ind, popNum, pop))) %>% 
            mutate(Gpred = mean(pars0$Gmax)*exp(-0.5*(log(DBH/mean(pars0$Dopt))/mean(pars0$Ks))^2)) %>% 
            mutate(Gpred5 = quantile(pars0$Gmax, 0.05)*exp(-0.5*(log(DBH/mean(pars0$Dopt, 0.05))/mean(pars0$Ks, 0.05))^2)) %>% 
            mutate(Gpred95 = quantile(pars0$Gmax, 0.95)*exp(-0.5*(log(DBH/mean(pars0$Dopt, 0.95))/mean(pars0$Ks, 0.95))^2)) %>% 
            mutate(model = "GrowthNull"),
          data.frame(DBH = rep(1:100, mdata$I), 
                     IndNum = rep(1:mdata$I, each = 100)) %>% 
            left_join(unique(dplyr::select(trees, IndNum, Ind, popNum, pop))) %>% 
            left_join(pars1) %>% 
            mutate(Gpred = Gmax_m*exp(-0.5*(log(DBH/Dopt_m)/Ks_m)^2)) %>%
            mutate(Gpred5 = Gmax_q5*exp(-0.5*(log(DBH/Dopt_q5)/Ks_q5)^2)) %>%
            mutate(Gpred95 = Gmax_q95*exp(-0.5*(log(DBH/Dopt_q95)/Ks_q95)^2)) %>%
            mutate(model = "GrowthGenePool"),
          data.frame(DBH = rep(1:100, mdata$I), 
                     IndNum = rep(1:mdata$I, each = 100)) %>% 
            left_join(unique(dplyr::select(trees, IndNum, Ind, popNum, pop))) %>% 
            left_join(pars2) %>% 
            left_join(pars2i) %>% 
            mutate(Gpred = Gmaxi_m*exp(-0.5*(log(DBH/Dopti_m)/Ksi_m)^2)) %>%
            mutate(Gpred5 = Gmax_q5*exp(-0.5*(log(DBH/Dopt_q5)/Ks_q5)^2)) %>% 
            mutate(Gpred95 = Gmax_q95*exp(-0.5*(log(DBH/Dopt_q95)/Ks_q95)^2)) %>% 
            mutate(model = "GrowthIndividual")) %>% 
  mutate(model = factor(model, levels = c("GrowthNull", "GrowthGenePool", "GrowthIndividual"))) %>% 
  ggplot(aes(x = DBH)) +
  geom_ribbon(aes(ymin = Gpred5, ymax = Gpred95, fill = pop), alpha = 0.5) +
  geom_point(aes(y = log(AGR +1), col = Ind), alpha = 0.4, data = trees) +
  geom_line(aes(y = Gpred, col = Ind)) +
  scale_color_discrete(guide = "none") +
  scale_fill_discrete(guide = "none") +
  facet_grid(model ~ pop)
```

## Growth & Environment

### Visual exploration

```{r growthEnvRel, fig.cap="Environmental variables relations with Annual Growth Rates (AGR) and Individual growth model parameters (Gmax, Dopt and ks)"}
load(file.path("symcapture_save", 'GrowthModels.Rdata'))
cowplot::plot_grid(trees %>% 
                     dplyr::select(CensusYear, pop, Ind, AGR, TWI, AUC, logNCI, logANCR) %>% 
                     reshape2::melt(id.vars = c("CensusYear", "pop", "Ind", "AGR")) %>% 
                     ggplot(aes(value, log(AGR+1), group = Ind)) +
                     geom_line(alpha = 0.1) +
                     facet_grid(pop ~ variable, scales = "free"),
                   as.data.frame(fits$growth2, pars = c("Gmaxi", "Dopti", "Ksi")) %>% 
                     reshape2::melt(variable.name = "parameter") %>% 
                     group_by(parameter) %>% 
                     summarise(value = mean(value)) %>% 
                     mutate(parameter = gsub("([[:punct:]])", "", parameter)) %>% 
                     mutate(parameter = gsub("i", "", parameter)) %>% 
                     mutate(IndNum = gsub("([[:alpha:]])", "", parameter)) %>% 
                     mutate(parameter = gsub("([[:digit:]])", "", parameter)) %>% 
                     mutate(IndNum = as.numeric(IndNum)) %>% 
                     left_join(unique(dplyr::select(trees, IndNum, Ind, popNum, pop, TWI))) %>% 
                     ggplot(aes(TWI, value)) +
                     geom_point() +
                     facet_grid(parameter ~ pop, scales = "free") +
                     geom_smooth(method = "lm"))
```
  
### *Brutal* linear model

```{r LMgrowthEnv, fig.cap="Result of the linear model between individual growth and environmental variables."}
trees %>% 
  group_by(pop) %>%
  do(lm = lm(log(AGR+1) ~ log(DBH) + TWI + AUC + logNCI + logANCR, data = .)) %>% 
  # broom::glance(lm)
  broom::tidy(lm) %>%
  filter(term != "(Intercept)") %>% 
  ggplot(aes(x = term, xend = term, col = term, alpha = p.value < 0.05)) +
  geom_hline(yintercept = 0, linetype = "dashed", col = "lightgrey") +
  geom_point(aes(y = estimate)) +
  geom_segment(aes(y = estimate - std.error, yend = estimate + std.error),
               size = 1, show.legend = F) +
  coord_flip() +
  facet_wrap(~ pop, nrow = 3) +
  scale_y_continuous(trans="S_sqrt") +
  ggtitle("LM: log(AGR+1) ~ log(DBH) + TWI + AUC + logNCI + logANCR", "Rsquared in [0.019,0.053]") +
  theme(axis.title = element_blank()) +
  scale_color_discrete("")
```

### GrowthEnv0


```{r growthOntogenyTab}
# fitsEnv <- list()
# model <- stan_model(paste0("./symcapture_models/GrowthEnv", 0, ".stan"))
# fitsEnv$growthEnv0 <- sampling(model, chains = 2, data = mdata, save_warmup = F)
# save(fitsEnv, file = file.path("symcapture_save", 'GrowthEnvModels.Rdata'))
load(file.path("symcapture_save", 'GrowthEnvModels.Rdata'))
mcmc_trace(as.array(fitsEnv$growthEnv0, pars = fitsEnv$growthEnv0@model_pars),
           facet_args = list(labeller = label_parsed))
broom::tidyMCMC(fitsEnv$growthEnv0, fitsEnv$growthEnv0@model_pars, droppars = NULL, rhat = T) %>% 
  kable(caption = "Summary table of the models")
mcmc_dens(as.array(fitsEnv$growthEnv0, pars = "beta"))
pars <- as.data.frame(fitsEnv$growthEnv0, pars = c("Gmax", "Dopt", "Ks", "beta")) %>% 
  reshape2::melt(variable.name = "parameter") %>% 
  group_by(parameter) %>% 
  summarise(q5 = quantile(value, 0.05), m = mean(value), q95 = quantile(value, 0.95)) %>% 
  mutate(parameter = gsub("([[:punct:]])", "", parameter)) %>% 
  reshape2::melt(id.vars = "parameter") %>% 
  mutate(parameter = paste0(parameter, "_", variable)) %>% 
  dplyr::select(-variable) %>%
  mutate(join = 1) %>% 
  reshape2::dcast(join ~ parameter)
data.frame(DBH = rep(1:100, mdata$I), 
           TWI = mean(trees$TWI),
           AUC = mean(trees$AUC),
           logNCI = mean(trees$logNCI),
           logANCR = mean(trees$logANCR),
           IndNum = rep(1:mdata$I, each = 100)) %>% 
  left_join(unique(dplyr::select(trees, IndNum, Ind, popNum, pop))) %>% 
  mutate(join = 1) %>% 
  left_join(pars) %>% 
  mutate(Gpred = Gmax_m*exp(-0.5*(log(DBH/Dopt_m)/Ks_m)^2)*exp(-beta1_m*TWI)*exp(-beta2_m*AUC)*exp(-beta3_m*logNCI)*exp(-beta4_m*logANCR)) %>%
  mutate(Gpred5 = Gmax_q5*exp(-0.5*(log(DBH/Dopt_q5)/Ks_q5)^2)*exp(-beta1_q5*TWI)*exp(-beta2_q5*AUC)*exp(-beta3_q5*logNCI)*exp(-beta4_q5*logANCR)) %>%
  mutate(Gpred95 = Gmax_q95*exp(-0.5*(log(DBH/Dopt_q95)/Ks_q95)^2)*exp(-beta1_q95*TWI)*exp(-beta2_q95*AUC)*exp(-beta3_q95*logNCI)*exp(-beta4_q95*logANCR)) %>%
  ggplot(aes(x = DBH)) +
  geom_ribbon(aes(ymin = Gpred5, ymax = Gpred95, fill = pop), alpha = 0.5) +
  geom_point(aes(y = log(AGR +1), col = Ind), alpha = 0.4, data = trees) +
  geom_line(aes(y = Gpred, col = Ind)) +
  scale_color_discrete(guide = "none") +
  scale_fill_discrete(guide = "none") +
  facet_wrap(~ pop)
data.frame(DBH = mean(trees$DBH), 
           TWI = rep(seq(min(trees$TWI), max(trees$TWI), length.out = 100), mdata$I),
           AUC = mean(trees$AUC),
           logNCI = mean(trees$logNCI),
           logANCR = mean(trees$logANCR),
           IndNum = rep(1:mdata$I, each = 100)) %>% 
  left_join(unique(dplyr::select(trees, IndNum, Ind, popNum, pop))) %>% 
  mutate(join = 1) %>% 
  left_join(pars) %>% 
  mutate(Gpred = Gmax_m*exp(-0.5*(log(DBH/Dopt_m)/Ks_m)^2)*exp(-beta1_m*TWI)*exp(-beta2_m*AUC)*exp(-beta3_m*logNCI)*exp(-beta4_m*logANCR)) %>%
  mutate(Gpred5 = Gmax_q5*exp(-0.5*(log(DBH/Dopt_q5)/Ks_q5)^2)*exp(-beta1_q5*TWI)*exp(-beta2_q5*AUC)*exp(-beta3_q5*logNCI)*exp(-beta4_q5*logANCR)) %>%
  mutate(Gpred95 = Gmax_q95*exp(-0.5*(log(DBH/Dopt_q95)/Ks_q95)^2)*exp(-beta1_q95*TWI)*exp(-beta2_q95*AUC)*exp(-beta3_q95*logNCI)*exp(-beta4_q95*logANCR)) %>%
  ggplot(aes(x = TWI)) +
  geom_ribbon(aes(ymin = Gpred5, ymax = Gpred95, fill = pop), alpha = 0.5) +
  geom_point(aes(y = log(AGR +1), col = Ind), alpha = 0.4, data = trees) +
  geom_line(aes(y = Gpred, col = Ind)) +
  scale_color_discrete(guide = "none") +
  scale_fill_discrete(guide = "none") +
  facet_wrap(~ pop)

trees %>%  
  mutate(join = 1) %>% 
  left_join(pars) %>% 
  mutate(Gpred = Gmax_m*exp(-0.5*(log(DBH/Dopt_m)/Ks_m)^2)*exp(-beta1_m*TWI)*exp(-beta2_m*AUC)*exp(-beta3_m*logNCI)*exp(-beta4_m*logANCR)) %>%
  mutate(Gpred5 = Gmax_q5*exp(-0.5*(log(DBH/Dopt_q5)/Ks_q5)^2)*exp(-beta1_q5*TWI)*exp(-beta2_q5*AUC)*exp(-beta3_q5*logNCI)*exp(-beta4_q5*logANCR)) %>%
  mutate(Gpred95 = Gmax_q95*exp(-0.5*(log(DBH/Dopt_q95)/Ks_q95)^2)*exp(-beta1_q95*TWI)*exp(-beta2_q95*AUC)*exp(-beta3_q95*logNCI)*exp(-beta4_q95*logANCR)) %>%
  dplyr::select(AGR, Ind, pop, Gpred, Gpred5, Gpred95, DBH, TWI, AUC, logNCI, logANCR) %>% 
  reshape2::melt(id.vars = c("Ind", "pop", "AGR", "Gpred", "Gpred5", "Gpred95")) %>% 
  ggplot(aes(x = value)) +
  geom_ribbon(aes(ymin = Gpred5, ymax = Gpred95, fill = pop), alpha = 0.5) +
  geom_point(aes(y = log(AGR +1), col = Ind), alpha = 0.4) +
  geom_line(aes(y = Gpred, col = Ind)) +
  scale_color_discrete(guide = "none") +
  scale_fill_discrete(guide = "none") +
  facet_grid(pop ~ variable, scales = "free")

data.frame(DBH = mean(trees$DBH), 
           TWI = rep(seq(min(trees$TWI), max(trees$TWI), length.out = 100), mdata$I),
           AUC = mean(trees$AUC),
           logNCI = mean(trees$logNCI),
           logANCR = mean(trees$logANCR),
           IndNum = rep(1:mdata$I, each = 100))
```


## Growth & genomics

```{bash AGRgemma, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
plink \
  --bfile ../../../variantCalling/paracou3pop/symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr \
  --extract ../../../variantCalling/paracou3pop/LD.prune.in \
  --maf 0.05 \
  --pheno AGR.phenotype \
  --out paracou3pop \
  --make-bed
gemma -bfile paracou3pop -gk 1 -o paracou3pop.kinship
gemma -bfile paracou3pop -k output/paracou3pop.kinship.cXX.txt -c AGR.covariates -lmm 1 -o paracou3pop.AGR
gemma -bfile paracou3pop -k -bslmm 1 -o paracou3pop.polygenic.AGR # BSLMM doesn't support covariates
cat AGR.LMM.snps | cut -f1 > AGR.snps.list
plink \
  --bfile paracou3pop \
  --allow-extra-chr \
  --extract AGR.snps.list \
  --out AGR.snps \
  --recode A
```

### Major effect SNPs

We detected 19 outliers SNPs with TWI independently of population structure (2 were neutral, 3 functional and 14 hitchhikers, Fig. \@ref(fig:TWIassocMLM)). The 14 hitchhiker SNPs (Fig. \@ref(fig:TWIoutliersHist)) corresponded to 4 transcripts enriched in *S. globulifera* and 2 transcripts enriched in *S. sp1* juveniles in common garden (Tysklind et al., in prep). **Gene ontology ...**

```{r AGRassocMLM, fig.cap="q-value of major effect SNPs associated to topographic wetness index (TWI). SNP were detected using linear mixed models with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
t <- lapply(c("paracou3pop"), function(pop) 
  read_tsv(file.path(path, "gemma", "AGR", paste0(pop,".AGR.assoc.txt"))) %>% 
    mutate(pop = pop)) %>% 
  bind_rows() %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  mutate(qval = p.adjust(p_wald, "fdr")) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.))
filter(t , qval < 0.05) %>%
  mutate(model = "LMM") %>%
  dplyr::select(rs, pop, model) %>%
  write_tsv(file.path(path, "gemma", "AGR", "AGR.LMM.snps"), col_names = F)
ggplot(t, aes(snp, -log10(qval), label = snp, alpha = qval < 0.05, col = type)) +
# ggplot(t, aes(snp, beta, label = snp, alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ggrepel::geom_text_repel(data = t[t$qval < 0.05,]) +
  ylab(expression(-log[10]("q-value"))) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  # scale_y_continuous(trans="S_sqrt") +
  xlab("SNP number") +
  scale_color_discrete("SNP type") +
  facet_wrap(~ pop) 
```

### Polygenic signal

A total of 31 804 SNPs explained approximately 59% of environmental variation (i.e. PVE=0.5915). Among these 31 804 SNPs, a small proportion of them (1% or 300 SNPs) have relatively large effects and explain a total of 50% PVE (i.e. PGE=0.4989).

```{r BSLMMAGRConvergence, fig.cap="BSLMM parameters distribution." }
lapply(c("paracou3pop"), function(pop) 
  read_tsv(file.path(path, "gemma", "AGR", paste0(pop,".polygenic.AGR.hyp.txt"))) %>% 
    mutate(Population = pop)) %>% 
  bind_rows() %>% 
  dplyr::select(pve, pge, pi, n_gamma, Population) %>% 
  dplyr::rename(PVE = pve, PGE = pge, `n[gamma]` = n_gamma) %>% 
  reshape2::melt(id.vars = "Population") %>% 
  ggplot(aes(value, fill = Population, col = Population)) +
  geom_density(alpha = 0.15) +
  facet_wrap(~ variable, scales = "free", labeller = "label_parsed") +
  theme(legend.position = "bottom",
        axis.title = element_blank(), axis.text.y = element_blank(), axis.ticks.y = element_blank())
```

```{r AGRassocBSLMM, eval=T}
t <- lapply(c("paracou3pop"), function(pop) 
  read_tsv(file.path(path, "gemma", "AGR", paste0(pop,".polygenic.AGR.param.txt"))) %>% 
    mutate(Population = pop)) %>% 
  bind_rows() %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.))
filter(t , gamma > 0.1) %>% 
  mutate(model = "BSLMM") %>% 
  dplyr::select(rs, Population, model) %>% 
  write_tsv(file.path(path, "gemma", "TWI", "TWI.BSLMM.snps"), col_names = F)
ggplot(t, aes(snp, beta*gamma, label = snp, col = type, alpha = gamma > 0.1)) +
  geom_point() +
  ggrepel::geom_text_repel(data = t[t$gamma > 0.1,]) +
  scale_y_continuous(trans="S_sqrt") +
  ylab(expression(beta*gamma)) +
  facet_wrap(~ Population, nrow = 3) 
```

### Outliers

```{r AGRoutliersHist, fig.cap="Distribution of alleles along topographic wetness index for significantly associated SNPs."}
read_delim(file.path(path, "gemma", "AGR", "AGR.snps.raw"), delim = " ") %>% 
  dplyr::select(-FID, -PAT, -MAT, -SEX) %>% 
  dplyr::rename(AGR = PHENOTYPE) %>% 
  reshape2::melt(id.vars = c("IID", "AGR"), variable.name = "SNP", value.name = "Allele") %>% 
  na.omit(Allele) %>% 
  mutate(Allele = as.factor(Allele)) %>% 
  left_join(trees, by = "IID", suffix = c("", ".trees")) %>% 
  mutate(SNP = stringr::str_sub(SNP, 1, -3)) %>% 
  ggplot(aes(pop, AGR, fill = Allele)) +
  geom_boxplot() +
  facet_wrap(~ SNP) +
  coord_flip() +
  theme(strip.text.x = element_text(size = 5)) +
  ylab("Topographic wetness index") +
  xlab("Population") +
  scale_fill_discrete("Number of copies\nof the major allele")
```

```{r AGRGO, fig.cap="Term enrichment of Gene Ontology (GO) for TWI-outlier genes.", eval=F}
outliers <- read_delim(file.path(path, "gemma", "AGR", "AGR.snps.raw"), delim = " ") %>% 
                           dplyr::select(-FID, -PAT, -MAT, -SEX) %>% 
                           dplyr::rename(TWI = PHENOTYPE) %>% 
                           reshape2::melt(id.vars = c("IID", "TWI"), variable.name = "snp", value.name = "Allele") %>% 
                           mutate(snp = stringr::str_sub(snp, 1, -3)) %>% 
                           dplyr::select(snp) %>% 
                           unique() %>% unlist()

trsc <- read_tsv(file.path(path, "..", "annotation", "snpsFunc.annotation")) %>% 
              dplyr::select(scf, snp, trsc) %>% 
  unique() %>% 
  mutate(outlier = ifelse(snp %in% outliers, 1, 0))

GO <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("Transcript") %>% 
  filter(transcript_id %in% local(unique(trsc$trsc))) %>% 
  dplyr::select(gene_id, transcript_id, annotation) %>% 
  dplyr::rename(gene = gene_id, trsc = transcript_id) %>% 
  collect() %>% 
  separate(annotation, paste0("X", 1:16), "\t") %>% 
  dplyr::rename(orf = X3, PFAM = X8, UniProt = X11, KEGG = X12, GO = X13) %>% 
  dplyr::select(trsc, gene, orf, GO) %>% 
  filter(GO != ".") %>% 
  left_join(trsc) %>% 
  group_by(gene, GO) %>% 
  summarise(outlier = max(outlier)) %>% 
  ungroup() %>% 
  dplyr::select(gene, outlier, GO) %>% 
  separate_rows(GO, sep = "`") %>% 
  separate(GO, c("GOid", "GOtype", "GOname"), "\\^")
  
clusterProfiler::enricher(unique(filter(GO, outlier == 1)$gene),
                               pvalueCutoff = 0.2, 
                               pAdjustMethod = "BH",
                               unique(GO$gene),
                               minGSSize = 10, 
                               maxGSSize = 500, 
                               qvalueCutoff = 0.2, 
                               dplyr::select(GO, GOid, gene),
                               TERM2NAME = dplyr::select(GO, GOid, GOname))@result %>% 
  ggplot(aes(reorder(Description, qvalue), Count, fill = -log(qvalue))) +
  geom_bar(stat = "identity") +
  coord_flip() +
  viridis::scale_fill_viridis() +
  theme(axis.title = element_blank())
```
