```{r setup_growthgenom, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(bayesplot)
library(rstan)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
pathCluster <- "~/Remotes/genotoul/work/PhD/documents/genomics/symcapture"
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
pairs_stan <- function(chain, stan_model, pars) {
  energy <- as.matrix(sapply(get_sampler_params(stan_model, inc_warmup = F), 
                             function(x) x[,"energy__"]))
  pars <- rstan::extract(stan_model, pars = pars, permuted = F)
  df <- data.frame(energy[,chain], pars[,chain,])
  names(df)[1] <- "energy"
  GGally::ggpairs(df, title = paste0("Chain", chain), 
                  lower = list(continuous = GGally::wrap("points", alpha = 0.2)))
}
```

# Growth genomics

We sampled growth and ontogeny models developped in the chapter "Growth & Animal models". Then, we derived growth parameters to be used in association genomics for mono- and poly-genic signals. 

```{r DataGrowthGenom}
load("./symcapture_save/Env.Rdata")
env <- trees
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Symphonia") %>% 
  mutate(DBH = CircCorr/pi) %>% 
  filter(!(CodeMeas %in% c(4))) %>% 
  collect()
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop"))) %>% 
  left_join(read_tsv(file.path(path, "populations", "paracou.hybridmap")),
            by = "Ind", suffix = c("", ".hybrid"))
trees0 <- trees
trees <- trees %>% 
  group_by(Ind) %>% 
  mutate(Y0 = dplyr::first(CensusYear), DBH0 = dplyr::first(DBH), 
         DBHtoday = dplyr::last(DBH), N = n()) %>%  
  ungroup() %>% 
  dplyr::select(Ind, IID, pop, Plot, SubPlot, TreeFieldNum, Y0, DBH0, DBHtoday, N) %>% 
  unique() %>% 
  mutate(DBHtoday = ifelse(DBHtoday == DBH0, DBHtoday + 0.1, DBHtoday))
trees <- trees %>%
  left_join(env)
rm(env)
save(trees, file = "./symcapture_save/Env2.Rdata")
```

```{r mdata}
trees <- trees %>% 
  na.omit(pop) %>% # remove admixed
  mutate(IndNum = as.numeric(as.factor(Ind)), popNum = as.numeric(as.factor(pop)))
# write_tsv(dplyr::select(trees, IID),
#           file.path(path, "populations", "growth.ind"), col_names = F)
# awk '{print "0\t"$1"\t0\t0\t0\t-9"}' growth.ind > growth.fam
# fam <- read_delim(file.path(path, "gemma", "Gmax0", "paracou3pop.fam"), delim = " ", 
#                   col_names = c("FID", "IID", "PAT", "MAT", "SEX", "PHENOTYPE"))
# K <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
#                         "out.relatedness2"))  %>% 
#   reshape2::dcast(INDV1 ~ INDV2, value.var = "RELATEDNESS_PHI") %>% 
#   column_to_rownames("INDV1") %>% 
#   as.matrix()
# fam <- read_delim(file.path(path,"..", "variantCalling", "growth", "growth.fam"), delim = " ", 
#                   col_names = c("FID", "IID", "PAT", "MAT", "SEX", "PHENOTYPE"))
# K <- read_tsv(file.path(path, "..", "variantCalling", "growth",
#                         "out.relatedness2"))  %>% 
#   reshape2::dcast(INDV1 ~ INDV2, value.var = "RELATEDNESS_PHI") %>% 
#   column_to_rownames("INDV1") %>% 
#   as.matrix()
# plink2 --bfile growth --allow-extra-chr --make-king square
ids <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king.id"))
K <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king"),
         col_names = ids$IID) %>% 
  as.data.frame()
row.names(K) <- ids$IID
K <- as.matrix(K)
model_data <- function(data) {
  K <- K[trees$IID, trees$IID]
  K[K < 0] <- 0
  K <- K*2
  K <- as.matrix(Matrix::nearPD(K)$mat)
  list(I = nrow(data),
       Y = 2017 - min(data$Y0) + 1,
       P = length(unique(data$popNum)),
       years = min(data$Y0):2017,
       DBH0 = data$DBH0,
       Y0 = data$Y0,
       DBHtoday = data$DBHtoday + 10^-6,
       TWI = as.vector(scale(data$TWI, center = F)),
       NCI = as.vector(scale(data$NCI, center = F)),
       ind = data$IndNum,
       pop = data$popNum,
       K = K)
}
mdata <- model_data(trees)
save(mdata, file = file.path("symcapture_save", "growthPar.Rdata"))
```

## Models

### Gmax

We used the model $Gmax$ to estimate individual growth potential per population:

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_p})}{\theta3_p}]^2)), \sigma^2) \\ \theta1_i \sim \mathcal{logN}(log(\theta1_p), \sigma^2_{R1}) $$

Results bla ...

```{bash fitGmaxParCluster, eval=F}
for chain in $(seq 12) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GmaxPar.R $chain" ; done > GmaxPar.sh
sarray -J Gmax -o out/%j.Gmax.out -e out/%j.Gmax.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GmaxPar.sh
```

```{r gmaxParTab}
fitGmax <- list()
for(chain in 1:12){
  load(paste0("./symcapture_save/gmaxPar/gmaxPar", chain, ".Rdata"))
  fitGmax[[chain]] <- fit
}
fitGmax <- sflist2stanfit(fitGmax[c(1:10)])
broom::tidyMCMC(fitGmax, pars = c("theta", "sigmaR", "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r gmaxParTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmax, pars = c("theta", "sigmaR", "sigma", "lp__")),
           np = nuts_params(fitGmax)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r gmaxParPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(6, fitGmax, c("sigmaR", "sigma"))
```

```{r gmaxParEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmax))
```

```{r gmaxParPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGmax, "theta[1,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "theta[1,2]"))))/
                 median(as.array(fitGmax, "theta[1,3]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGmax, "theta[2,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "theta[2,2]"))))/
                 median(as.array(fitGmax, "theta[2,3]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGmax, "theta[3,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "theta[3,2]"))))/
                 median(as.array(fitGmax, "theta[3,3]")))^2))
```

### Gmax1

We used the model $Gmax1$ to estimate individual growth potential per population:

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_p})}{\theta3_p}]^2)), \sigma^2) \\ \theta1_i \sim \mathcal{logN}(log(\theta1_p), \sigma^2_{R1}) \\ \begin{pmatrix} \theta2_p \\ \theta3_p \end{pmatrix} \sim \mathcal{logN}(\begin{pmatrix} \theta2 \\ \theta3 \end{pmatrix}, \begin{pmatrix} \sigma^2_{P2} \\ \sigma^2_{P3} \end{pmatrix})$$

```{bash fitGmax1ParCluster, eval=F}
for chain in $(seq 12) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript Gmax1Par.R $chain" ; done > Gmax1Par.sh
sarray -J Gm1 -o out/%j.Gm1.out -e out/%j.Gm1.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL Gmax1Par.sh
```

```{r gmax1ParTab}
fitGmax1 <- list()
for(sim in list.files("symcapture_save/gmax1Par", full.names = T)){
  load(sim)
  fitGmax1 <- c(fitGmax1, fit)
}
fitGmax1 <- sflist2stanfit(fitGmax1)
# as.data.frame(fitGmax1, "thetai1") %>%
#   summarise_all(median) %>%
#   reshape2::melt(NULL) %>%
#   separate(variable, c("theta", "IndNum"), convert = T) %>%
#   dplyr::select(-theta) %>%
#   left_join(dplyr::select(trees, IndNum, IID)) %>%
#   dplyr::select(-IndNum) %>%
#   mutate(FID = 0) %>%
#   dplyr::select(FID, IID, value) %>%
#   write_tsv(file.path(path, "gemma", "gmax", "gmax.phenotype"),
#             col_names = F) # for genomic scans
broom::tidyMCMC(fitGmax1, pars = c("thetap1", "theta2", "theta3",
                                   "sigmaR", "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r gmax1ParTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmax1, pars = c("thetap1", "theta2", "theta3",
                                       "sigmaR", "sigma", "lp__")),
           np = nuts_params(fitGmax1)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r gmax1ParPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmax1, c("sigmaR", "sigma"))
```

```{r gmax1ParEnergy, fig.cap="Energy of the model.", eval=F}
mcmc_nuts_energy(nuts_params(fitGmax1))
```

```{r gmax1ParPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGmax1, "thetap1[1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax1, "thetap2[1]"))))/
                 median(as.array(fitGmax1, "thetap3[1]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGmax1, "thetap1[2]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax1, "thetap2[2]"))))/
                 median(as.array(fitGmax1, "thetap3[2]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGmax1, "thetap1[3]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax1, "thetap2[3]"))))/
                 median(as.array(fitGmax1, "thetap3[3]")))^2))
```

### Growth

We used the model $Growth$ to estimate individual growth potential, optimal diameter and kurtosis per population:

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_i})}{\theta3_i}]^2)), \sigma^2) \\ \forall \begin{pmatrix} \theta & \sigma^2_R \end{pmatrix} \in \begin{pmatrix} \theta1 & \sigma^2_{R1} \\ \theta2 & \sigma^2_{R2} \\ \theta3 & \sigma^2_{R3} \end{pmatrix}, \theta_i \sim \mathcal{logN}(log(\theta_p), \sigma^2_{R}) $$

Results bla ...

```{bash fitGrowthParCluster, eval=F}
for chain in $(seq 12) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GrowthPar.R $chain" ; done > GrowthPar.sh
sarray -J Growth -o out/%j.Growth.out -e out/%j.Growth.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GrowthPar.sh
```

```{r growthParTab}
fitGrowth <- list()
for(chain in 1:12){
  load(paste0("./symcapture_save/growthPar/growthPar", chain, ".Rdata"))
  fitGrowth[[chain]] <- fit
}
fitGrowth <- sflist2stanfit(fitGrowth)
# as.data.frame(fitGrowth, "thetai") %>% 
#   summarise_all(median) %>% 
#   reshape2::melt(NULL) %>% 
#   separate(variable, c("theta", "IndNum", "parameter"), convert = T) %>% 
#   dplyr::select(-theta) %>% 
#   left_join(dplyr::select(trees, IndNum, IID)) %>% 
#   dplyr::select(-IndNum) %>% 
#   mutate(FID = 0) %>% 
#   reshape2::dcast(FID + IID ~ parameter) %>% 
#   write_tsv(file.path(path, "gemma", "growth", "growth.phenotype"),
#             col_names = F) # for genomic scans
broom::tidyMCMC(fitGrowth, pars = c("theta", "sigmaR", "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r growthParTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGrowth, pars = c("theta", "sigmaR", "sigma", "lp__")),
           np = nuts_params(fitGrowth)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r growthParPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGrowth, c("sigmaR", "sigma"))
```

```{r growthParEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGrowth))
```

```{r growthParPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGrowth, "theta[1,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGrowth, "theta[1,2]"))))/
                 median(as.array(fitGrowth, "theta[1,3]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGrowth, "theta[2,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGrowth, "theta[2,2]"))))/
                 median(as.array(fitGrowth, "theta[2,3]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGrowth, "theta[3,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGrowth, "theta[3,2]"))))/
                 median(as.array(fitGrowth, "theta[3,3]")))^2))
```

### GmaxGeno

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_p})}{\theta3_p}]^2)), \sigma^2) \\  \theta1_i \sim \mathcal{logN}(log(\theta1_p + a1_i), \sigma^2_{R1}) \\ a1_i \sim \mathcal{MVlogN}(0, \sigma^2_{G1}.K)$$

```{bash fitGmaxGenoParCluster, eval=F}
for chain in $(seq 12) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GmaxGenoPar.R $chain" ; done > GmaxGenoPar.sh
sarray -J GmaxGeno -o out/%j.GmaxGeno.out -e out/%j.GmaxGeno.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GmaxGenoPar.sh
```

```{r gmaxgenoParTab}
fitGmaxGeno <- list()
sims <- c(1:25)
for(chain in 1:12){
  load(paste0("./symcapture_save/gmaxgenoPar/gmaxgenoPar", chain, ".Rdata"))
  fitGmaxGeno[[chain]] <- fit
}
fitGmaxGeno <- sflist2stanfit(fitGmaxGeno)
broom::tidyMCMC(fitGmaxGeno, pars = c("theta", "sigmaR", "sigmaG", "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r gmaxgenoParTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmaxGeno, pars = c("theta", "sigmaR", "sigmaG", "sigma", "lp__")),
           np = nuts_params(fitGrowth)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r gmaxgenoParPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmaxGeno, c("sigmaR", "sigma"))
```

```{r gmaxgenoParEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmaxGeno))
```

```{r gmaxgenoParPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGmaxGeno, "theta[1,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGeno, "theta[1,2]"))))/
                 median(as.array(fitGmaxGeno, "theta[1,3]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGmaxGeno, "theta[2,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGeno, "theta[2,2]"))))/
                 median(as.array(fitGmaxGeno, "theta[2,3]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGmaxGeno, "theta[3,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGeno, "theta[3,2]"))))/
                 median(as.array(fitGmaxGeno, "theta[3,3]")))^2))
```

### GmaxGeno1

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_p})}{\theta3_p}]^2)), \sigma^2) \\  \theta1_i \sim \mathcal{logN}(log(\theta1_p + a1_i), \sigma^2_{R1}) \\ a1_i \sim \mathcal{MVlogN}(0, \sigma^2_{G1}.K) \\ \begin{pmatrix} \theta2_p \\ \theta3_p \end{pmatrix} \sim \mathcal{logN}(\begin{pmatrix} \theta2 \\ \theta3 \end{pmatrix}, \begin{pmatrix} \sigma^2_{P2} \\ \sigma^2_{P3} \end{pmatrix})$$

```{bash fitGmaxGeno1ParCluster, eval=F}
for chain in $(seq 12) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GmaxGeno1Par.R $chain" ; done > GmaxGeno1Par.sh
sarray -J GmG1 -o out/%j.GmG1.out -e out/%j.GmG1.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GmaxGeno1Par.sh
```

```{r gmaxgeno1ParTab}
fitGmaxGeno1 <- list()
for(sim in list.files("symcapture_save/gmaxgeno1Par", full.names = T)){
  load(sim)
  fitGmaxGeno1 <- c(fitGmaxGeno1, fit)
}
fitGmaxGeno1 <- sflist2stanfit(fitGmaxGeno1)
broom::tidyMCMC(fitGmaxGeno1, pars = c("thetap1", "theta2", "theta3",
                                      "sigmaR", "sigmaG", "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r gmaxgeno1ParTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmaxGeno1, pars = c("thetap1", "theta2", "theta3",
                                          "sigmaR", "sigmaG", "sigma", "lp__")),
           np = nuts_params(fitGmaxGeno1)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r gmaxgeno1ParPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmaxGeno1, c("sigmaR", "sigma"))
```

```{r gmaxgeno1ParEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmaxGeno1))
```

```{r gmaxgeno1ParPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGmaxGeno1, "thetap1[1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGeno1, "thetap2[1]"))))/
                 median(as.array(fitGmaxGeno1, "thetap3[1]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGmaxGeno1, "thetap1[2]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGeno1, "thetap2[2]"))))/
                 median(as.array(fitGmaxGeno1, "thetap3[2]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGmaxGeno1, "thetap1[3]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGeno1, "thetap2[3]"))))/
                 median(as.array(fitGmaxGeno1, "thetap3[3]")))^2))
```

### GmaxGenoEnv

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_p})}{\theta3_p}]^2)), \sigma^2) \\  \theta1_i \sim \mathcal{logN}(log(\theta1_p + a1_i + \beta_{TWI}.TWI+\beta_{NCI}.NCI), \sigma^2_{R1}) \\ a1_i \sim \mathcal{MVlogN}(0, \sigma^2_{G1}.K) \\ \begin{pmatrix} \theta2_p \\ \theta3_p \end{pmatrix} \sim \mathcal{logN}(\begin{pmatrix} \theta2 \\ \theta3 \end{pmatrix}, \begin{pmatrix} \sigma^2_{P2} \\ \sigma^2_{P3} \end{pmatrix})$$

```{bash fitGmaxGenoEnvCluster, eval=F}
for chain in $(seq 12) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GmaxGenoEnv.R $chain" ; done > GmaxGenoEnv.sh
sarray -J GmGE -o out/%j.GmGE.out -e out/%j.GmGE.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GmaxGenoEnv.sh
```

```{r gmaxgenoenvTab}
fitGmaxGenoEnv <- list()
for(sim in list.files("symcapture_save/gmaxgenoenv", full.names = T)){
  load(sim)
  fitGmaxGenoEnv <- c(fitGmaxGenoEnv, fit)
}
fitGmaxGenoEnv <- sflist2stanfit(fitGmaxGenoEnv)
broom::tidyMCMC(fitGmaxGenoEnv, pars = c("thetap1", "betaTWI", "betaNCI",
                                         "theta2", "theta3",
                                      "sigmaR", "sigmaG", "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r gmaxgenoenvTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmaxGenoEnv, pars = c("thetap1", "betaTWI", "betaNCI",
                                           "theta2", "theta3",
                                          "sigmaR", "sigmaG", "sigma", "lp__")),
           np = nuts_params(fitGmaxGenoEnv)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r gmaxgenoenvPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmaxGenoEnv, c("sigmaR", "sigma"))
```

```{r gmaxgenoenvEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmaxGenoEnv))
```

```{r gmaxgenoenvPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGmaxGenoEnv, "thetap1[1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGenoEnv, "thetap2[1]"))))/
                 median(as.array(fitGmaxGenoEnv, "thetap3[1]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGmaxGenoEnv, "thetap1[2]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGenoEnv, "thetap2[2]"))))/
                 median(as.array(fitGmaxGenoEnv, "thetap3[2]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGmaxGenoEnv, "thetap1[3]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGenoEnv, "thetap2[3]"))))/
                 median(as.array(fitGmaxGenoEnv, "thetap3[3]")))^2))
```


```{r gmaxgenoenvVarTab}
broom::tidyMCMC(fitGmaxGenoEnv, pars = c("Vp", "Vg", "Vnci", "Vtwi", "Vr"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r gmaxgenoenvVar, fig.cap="Genetic variance partitionning for ontogenetic parameters."}
mcmc_intervals_data(fitGmaxGenoEnv, regex_pars = c("Vp", "Vg", "Vnci", "Vtwi", "Vr")) %>% 
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", 
                           "Vnci" = "NCI", "Vtwi" = "TWI", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2)) 
```

### GmaxGenoEnv2

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_p})}{\theta3_p}]^2)), \sigma^2) \\  \theta1_i \sim \mathcal{logN}(log(\theta1_p + a1_i + \beta_{TWI_p}.TWI+\beta_{NCI_p}.NCI), \sigma^2_{R1}) \\ a1_i \sim \mathcal{MVlogN}(0, \sigma^2_{G1}.K) \\ \begin{pmatrix} \theta2_p \\ \theta3_p \end{pmatrix} \sim \mathcal{logN}(\begin{pmatrix} \theta2 \\ \theta3 \end{pmatrix}, \begin{pmatrix} \sigma^2_{P2} \\ \sigma^2_{P3} \end{pmatrix})$$

```{bash fitGmaxGenoEnv2Cluster, eval=F}
for chain in $(seq 12) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GmaxGenoEnv2.R $chain" ; done > GmaxGenoEnv2.sh
sarray -J GmGE2 -o out/%j.GmGE2.out -e out/%j.GmGE2.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GmaxGenoEnv2.sh
```

```{r gmaxgenoenv2Tab}
fitGmaxGenoEnv2 <- list()
for(sim in list.files("symcapture_save/gmaxgenoenv2", full.names = T)){
  load(sim)
  fitGmaxGenoEnv2 <- c(fitGmaxGenoEnv2, fit)
}
fitGmaxGenoEnv2 <- sflist2stanfit(fitGmaxGenoEnv2)
```

```{r gmaxgenoenv2Trace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmaxGenoEnv2, pars = c("thetap1", "betaTWIp", "betaNCIp",
                                           "theta2", "theta3",
                                          "sigmaR", "sigmaG", "sigma", "lp__")),
           np = nuts_params(fitGmaxGenoEnv2)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r gmaxgenoenv2Pairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmaxGenoEnv2, c("sigmaR", "sigma"))
```

```{r gmaxgenoenvEnergy2, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmaxGenoEnv2))
```

```{r gmaxgenoenv2Pred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGmaxGenoEnv2, "thetap1[1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGenoEnv2, "thetap2[1]"))))/
                 median(as.array(fitGmaxGenoEnv2, "thetap3[1]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGmaxGenoEnv2, "thetap1[2]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGenoEnv2, "thetap2[2]"))))/
                 median(as.array(fitGmaxGenoEnv2, "thetap3[2]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGmaxGenoEnv2, "thetap1[3]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGenoEnv2, "thetap2[3]"))))/
                 median(as.array(fitGmaxGenoEnv2, "thetap3[3]")))^2))
```

```{r gmaxgenoenv2Var, fig.cap="Genetic variance partitionning for ontogenetic parameters."}
as.data.frame(fitGmaxGenoEnv2, 
              pars = c("Vp", "Vg", "Vnci", "Vtwi", "Vr")) %>% 
  na.omit() %>% 
  mcmc_intervals_data() %>% 
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", 
                           "Vnci" = "NCI", "Vtwi" = "TWI", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2)) 

```

### GrowthGeno

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_i})}{\theta3_i}]^2)), \sigma^2) \\ \forall \begin{pmatrix} \theta & \sigma^2_R & a\end{pmatrix} \in \begin{pmatrix} \theta1 & \sigma^2_{R1} & a1 \\ \theta2 & \sigma^2_{R2} & a2 \\ \theta3 & \sigma^2_{R3} & a3 \end{pmatrix}, \theta_i \sim \mathcal{logN}(log(\theta_p + a_i), \sigma^2_{R}) \\ \forall \begin{pmatrix} a & \sigma^2_G \end{pmatrix} \in \begin{pmatrix} a1 & \sigma^2_{G1} \\ a2 & \sigma^2_{G2} \\ a3 & \sigma^2_{G3} \end{pmatrix}, a_i \sim \mathcal{MVlogN}(0, \sigma^2_{G}.K)$$

```{bash fitGrowthGenoParCluster, eval=F}
for chain in $(seq 12) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GrowthGenoPar.R $chain" ; done > GrowthGenoPar.sh
sarray -J GrGP -o out/%j.GrGP.out -e out/%j.GrGP.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GrowthGenoPar.sh
```

```{r growthgenoParTab}
fitGrowthGeno <- list()
for(sim in list.files("symcapture_save/growthgenoPar", full.names = T)){
  load(sim)
  fitGrowthGeno <- c(fitGrowthGeno, fit)
}
fitGrowthGeno <- sflist2stanfit(fitGrowthGeno)
broom::tidyMCMC(fitGrowthGeno, pars = c("theta", "sigmaR", "sigmaG", "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r growthgenoParTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGrowthGeno, pars = c("theta", "sigmaR", "sigmaG", "sigma", "lp__")),
           np = nuts_params(fitGrowthGeno)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r growthgenoParPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGrowthGeno, c("sigmaR", "sigma"))
```

```{r growthgenoParEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGrowthGeno))
```

```{r growthgenoParPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGrowthGeno, "theta[1,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGrowthGeno, "theta[1,2]"))))/
                 median(as.array(fitGrowthGeno, "theta[1,3]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGrowthGeno, "theta[2,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGrowthGeno, "theta[2,2]"))))/
                 median(as.array(fitGrowthGeno, "theta[2,3]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGrowthGeno, "theta[3,1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGrowthGeno, "theta[3,2]"))))/
                 median(as.array(fitGrowthGeno, "theta[3,3]")))^2))
```

## Variance partitionning

```{r growthgenoR2, fig.cap="Marginal and conditional $R^2$ for ontogenetic parameters."}
mcmc_intervals_data(fitGmaxGeno1, regex_pars = c("R2")) %>% 
  mutate(index = recode(parameter, 
                        "R2m" = "Marginal", "R2c" = "Conditional")) %>%
  ggplot(aes(x = index, xend = index)) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  xlab("") + ylab(expression(R^2)) +
  scale_fill_discrete(guide = "none") +
  scale_color_discrete(guide = "none")
```

```{r growthgenoVar, fig.cap="Genetic variance partitionning for ontogenetic parameters."}
mcmc_intervals_data(fitGmaxGeno1, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2)) 
```

## Associations

SNP association to growth potential $Gmax$ has been assessed per individual with mono- and poly-genic modelling (respectivelly LMM and BSLMM, see methods in environmental genomics) for every populations. 

```{bash Gemma, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
pops=(globuliferaTypeParacou globuliferaTypeRegina sp1)
for pop in "${pops[@]}" ; 
  do 
    plink \
      --bfile growth \
      --allow-extra-chr \
      --keep  $pop.fam \
      --extract growth.prune.in \
      --maf 0.05 \
      --pheno gmax.phenotype \
      --out $pop --make-bed ; 
    gemma -bfile $pop -gk 1 -o kinship.$pop ;
    gemma -bfile $pop -lmm 1 \
      -k output/kinship.$pop.cXX.txt \
      -o LMM.$pop ;
    gemma -bfile $pop -bslmm 1 -o BSLMM.$pop ;
done
cat LMM.snps | cut -f1 > snps.list
plink --bfile growth --allow-extra-chr --extract snps.list --out outliers --recode A
```

```{r growthLMM, fig.cap="q-value of major effect SNPs associated to individual growth parameters. SNP were detected using linear mixed models (LMM) with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
lmm <- lapply(list.files(file.path(path, "gemma", "gmax"), 
                  pattern = ".assoc.txt$", full.names = T), 
       function(x) read_tsv(x) %>% mutate(file = x))%>% 
  bind_rows() %>% 
  mutate(file = gsub("/home/sylvain/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics//gemma/gmax/LMM.", "", file)) %>% 
  mutate(file = gsub(".assoc.txt", "", file)) %>% 
  dplyr::rename(population = file) %>% 
  mutate(rs = paste0(chr, "_snp", ps)) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
  dplyr::rename(rs = snp)) %>%
  mutate(qval = p.adjust(p_wald, "fdr")) %>%
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  na.omit(qval)
filter(lmm , qval < 0.05) %>%
  dplyr::select(rs, population) %>%
  write_tsv(file.path(path, "gemma", "gmax", "LMM.snps"), col_names = F)
ggplot(lmm, aes(snp, -log10(qval), label = snp, alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ggrepel::geom_text_repel(data = lmm[lmm$qval < 0.05,]) +
  ylab(expression(-log[10]("q-value"))) +
  facet_wrap(~ population) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type") +
  theme(legend.position = "bottom")
```

```{r growthBSLMMConvergence, fig.cap="Traceplot of hyper-parameters chain for Baysian Sparse Linear Mixed Model (BSLMM).", fig.height=10, fig.width=10, eval=F}
cowplot::plot_grid(plotlist = lapply(list.files(file.path(path, "gemma", "gmax"), pattern = ".hyp.txt$", full.names = T), 
       function(file) read_tsv(file) %>% 
         dplyr::select(pge, n_gamma) %>% 
         dplyr::rename(PGE = pge, `n[gamma]` = n_gamma) %>% 
         mutate(iteration = 1:n()) %>% 
         filter(row_number() %% 100 == 0) %>% 
         dplyr::select(-iteration) %>% 
         mcmc_combo()), labels = c("Paracou", "Regina", "Sp1"), nrow = 3)
```

```{r growthBSLMM, fig.cap="Effect of SNPs associated to individual growth parameters with a polygenic model. SNP were detected using bayesian sparse linear mixed models (BSLMM) with individual kiniship as random effect. SNP significance was assessed with the $\\gamma$ parameter and highlighted here with SNPs having at least 90% chance of having an effect on the phenotype.", eval=F}
bslmm <- lapply(list.files(file.path(path, "gemma", "gmax"), pattern = ".param.txt$", full.names = T), function(x) read_tsv(x) %>% mutate(file = x)) %>% 
  bind_rows() %>% 
  mutate(file = gsub("/home/sylvain/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics//gemma/gmax/BSLMM.", "", file)) %>% 
  mutate(file = gsub(".param.txt", "", file)) %>% 
  separate(file, c("population", "parameter")) %>% 
  mutate(parameter = recode(parameter, "1" = "Gmax",  "2" = "Dopt",  "3" = "Ks")) %>% 
  mutate(rs = paste0(chr, "_snp", ps)) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.))
ggplot(bslmm, aes(snp, beta*gamma, label = snp, col = type, alpha = gamma > 0.5)) +
  geom_point() +
  ggrepel::geom_text_repel(data = bslmm[bslmm$gamma > 0.5,]) +
  scale_y_continuous(trans="S_sqrt") +
  facet_wrap(~ population) +
  ylab(expression(beta*gamma))
```

```{r outliersGrowth, fig.cap="Growth parameters distribution for outliers genes."}
read_delim(file.path(path, "gemma", "gmax", "outliers.raw"), delim = " ") %>% 
  left_join(read_tsv(file.path(path, "gemma", "gmax", "gmax.phenotype"), 
         col_names = c("FID", "IID", "Gmax"))) %>% 
  left_join(dplyr::select(trees, IID, pop)) %>% 
  dplyr::select(-FID, -PAT, -MAT, -SEX, -PHENOTYPE) %>% 
  reshape2::melt(id.vars = c("pop", "IID", "Gmax"),
                 variable.name = "snp", value.name = "allele") %>% 
  na.omit(allele) %>% 
  mutate(allele = as.factor(allele)) %>% 
  mutate(snp = gsub("_[[:alnum:]]$", "", snp)) %>% 
  right_join(dplyr::select(lmm[lmm$qval < 0.05,], population, rs) %>% 
               dplyr::rename(snp = rs, pop = population)) %>% 
  ggplot(aes(allele, Gmax, fill = pop)) +
  geom_boxplot() +
  stat_summary(fun.data = function(y) 
    data.frame(y = max(y)+0.1, label = paste(length(y))),
    geom = "text", hjust = 0.5, vjust = 0.9) + 
  coord_flip() +
  facet_wrap(~ pop + snp, scales = "free") +
  scale_fill_discrete("") + ylab("") + xlab("Major allele copies") +
  theme(strip.text = element_blank(), legend.position = "bottom")
```

```{r growthGO, fig.cap="Term enrichment of Gene Ontology (GO) for outliers genes.", eval=F}
trsc <- read_tsv(file.path(path, "..", "annotation", "snpsFunc.annotation")) %>% 
              dplyr::select(scf, snp, trsc) %>% 
  bind_rows(read_tsv(file.path(path, "..", "annotation", "snpsHitch.annotation"))) %>%
  unique() %>% 
  inner_join(dplyr::rename(lmm, snpNum = snp, snp = rs), .)
GO <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("Transcript") %>% 
  filter(transcript_id %in% local(unique(trsc$trsc))) %>% 
  dplyr::select(gene_id, transcript_id, annotation) %>% 
  dplyr::rename(gene = gene_id, trsc = transcript_id) %>% 
  collect() %>% 
  separate(annotation, paste0("X", 1:16), "\t") %>% 
  dplyr::rename(orf = X3, PFAM = X8, UniProt = X11, KEGG = X12, GO = X13) %>% 
  dplyr::select(trsc, gene, orf, GO) %>% 
  filter(GO != ".") %>% 
  left_join(trsc) %>% 
  group_by(population, gene, GO) %>% 
  summarise(qval = min(qval)) %>% 
  ungroup() %>% 
  dplyr::select(population, gene, qval, GO) %>% 
  separate_rows(GO, sep = "`") %>% 
  separate(GO, c("GOid", "GOtype", "GOname"), "\\^")
enrichment <- function(x) {
  y <- clusterProfiler::enricher(unique(filter(x, qval < 0.05)$gene),
                                            pvalueCutoff = 0.2, 
                                            pAdjustMethod = "BH",
                                            unique(x$gene),
                                            minGSSize = 10, 
                                            maxGSSize = 500, 
                                            qvalueCutoff = 0.2, 
                                            dplyr::select(x, GOid, gene),
                                            TERM2NAME = dplyr::select(x, GOid, GOname))
  if(is.null(y)){
    data.frame()
  } else {
    y@result
  }
}
group_by(GO, population) %>% 
  do(enrichment(.)) %>% 
  filter(qvalue < 0.8) %>% 
  ggplot(aes(reorder(Description, qvalue), Count, fill = -log(qvalue))) +
  geom_bar(stat = "identity") +
  coord_flip() +
  facet_wrap(~ population, labeller = "label_both", nrow = 2, scales = "free") +
  viridis::scale_fill_viridis() +
  theme(axis.title = element_blank())
```
