```{r setup_growthgenom, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(bayesplot)
library(rstan)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
pathCluster <- "~/Remotes/genotoul/work/PhD/documents/genomics/symcapture"
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
```

# Growth genomics

We explored and selected ontogenic and environmental descriptors to be used in the annual growth rate model. We then built four growth model with growing complexity including ontogeny, environment, kinship and population effects. We finally compared growth models and derived growth parameters to be used in association genomics for mono- and poly-genic signals. 

## Variables

```{r DataGrowthGenom}
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Symphonia") %>% 
  mutate(DBH = CircCorr/pi) %>% 
  collect()
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop"))) %>% 
  left_join(read_tsv(file.path(path, "populations", "paracou.hybridmap")),
            by = "Ind", suffix = c("", ".hybrid"))
trees <- trees %>% 
  group_by(Ind) %>% 
  mutate(Y0 = dplyr::first(CensusYear), DBH0 = dplyr::first(DBH), 
         DBHtoday = dplyr::last(DBH)) %>%  
  ungroup() %>% 
  dplyr::select(Ind, IID, pop, Plot, SubPlot, TreeFieldNum, Y0, DBH0, DBHtoday) %>% 
  unique()
save(trees, file = "./symcapture_save/Env2.Rdata")
```

```{r mdata}
trees <- trees %>% 
  na.omit(pop) %>% # remove admixed
  mutate(IndNum = as.numeric(as.factor(Ind)), popNum = as.numeric(as.factor(pop)))
fam <- read_delim(file.path(path, "gemma", "Gmax0", "paracou3pop.fam"), delim = " ", 
                  col_names = c("FID", "IID", "PAT", "MAT", "SEX", "PHENOTYPE"))
K <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                        "out.relatedness2"))  %>% 
  reshape2::dcast(INDV1 ~ INDV2, value.var = "RELATEDNESS_PHI") %>% 
  column_to_rownames("INDV1") %>% 
  as.matrix()
mdata <- function(data) {
  K <- K[trees$IID, trees$IID]
  K[K < 0] <- 0
  K <- K*2
  K <- as.matrix(nearPD(K)$mat)
  list(I = nrow(data),
       Y = 2017 - min(data$Y0) + 1,
       P = length(unique(data$popNum)),
       years = min(data$Y0):2017,
       DBH0 = data$DBH0,
       Y0 = data$Y0,
       DBHtoday = data$DBHtoday,
       ind = data$IndNum,
       pop = data$popNum,
       K = K)
}
```

## Growth & Ontogeny

We used the following form of hierachical model to model DBH today $DBH_{y=today}$ with Diameter at Breast Height $DBH_{y,p,i}$ at year $y\in[y_0,today]$ for individual $i$ in population $p$:

$$DBH_{y=today,p,i} \sim \mathcal{logN} (DBH_0 + \sum _{y=y_0} ^{y=today} Gmax_i.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{Dopt_i})}{Ks_i}]^2), \sigma)$$
$$\mid \forall \theta \in [Gmax, Dopt,Ks], \theta_i \sim \mathcal N (\theta_p, \sigma^2_{R_{\theta}})$$

with $Gmax$, $Dopt$ and $Ks$ the parameters of the growth response to ontgeny representing respectively the maximum growth potential, the optimal growth diameter when this potential is reached, and the kurtosis of the growth curve respectivelly. Ontogenic parameters were defined at the individual level $i$ as random effects centered on the population level $p$ parameters with associated individual variance $\sigma^2_R$.

At the genus level, *Symphonia* individuals showed a maximum growth of $6.91\pm1.8~mm.yr^{-1}$ at optimal diameter of $28\pm11~cm$ with a kurtosis of $0.7\pm0.1$ (Tab. \@ref(tab:growthOntogenyTab)). Maximum growth and kurtosis didn't significantly varied at the population level but optmal diameter did with a larger optimal diameter for *S. globulifera type Regina* ($Dopt=32~cm$) and a smaller optimal diameter for *S. sp1* ($Dopt=21~cm$). Finally, all parameters showed significant variation at the individual level within populations ($\pm1.8~mm.yr^{-1}$, $\pm9~cm$, and $\pm0.2$ for maximum growth, optimal diameter, and kurtosis respectively, Fig. \@ref(fig:growthOntogenyG)). The model residual variation is relatively low ($\pm1.5~mm.yr^{-1}$, Fig. \@ref(fig:growthOntogenyGP3)).

```{r growthOntogenyTab, eval=F}
mdata <- mdata(trees)
save(mdata, file = file.path("symcapture_save", "growth.Rdata"))
file.copy("./symcapture_save/growth.Rdata",
          file.path(pathCluster, "symcapture_save"), overwrite = T)
file.copy("./symcapture_models/Growth5.stan",
          file.path(pathCluster, "symcapture_models"), overwrite = T)
load(file = file.path("..", "symcapture_save", "growth.Rdata"))  # Begin cluster
growth <- stan_model("../symcapture_models/Growth5.stan")
fitGrowth <- sampling(growth, chains = 2, data = mdata, save_warmup = F,
                      include = F, pars = c("DBH", "Gmaxi", "Dopti", "Ksi"))
save(mdata, fitGrowth, file = file.path("..", "symcapture_save", "growth.Rdata"))  # End cluster
file.copy(file.path(pathCluster, "symcapture_save"),
          "./symcapture_save/growth.Rdata", overwrite = T)
load(file.path("symcapture_save", 'growth5.Rdata'))
broom::tidyMCMC(fitGrowth, droppars = NULL, rhat = T) %>% 
  mutate(term = recode(term, "sigmaR[1]" = "$\\sigma_{R_{Gmax}}$",
                       "sigmaR[2]" = "$\\sigma_{R_{Dopt}}$",
                       "sigmaR[3]" = "$\\sigma_{R_{Ks}}$",
                       "sigma" = "$\\sigma$",
                      "lp__" = "loglokelihood")) %>%
  kable(caption = "Summary table of the ontogenic growth model",
        col.names = c("Parameter", "Estimate", "$\\sigma$", "$\\hat{R}$"))
# mcmc_trace(fitGrowth)
```

## Growth & Kinship

The kinship matrix express individuals genetic covariation an can help assess the role of genetic variation in individual response to growth. We used the following form of hierachical model to model DBH today $DBH_{y=today}$ with Diameter at Breast Height $DBH_{y,p,i}$  and individuals Kinsip matrix $K$ at year $y\in[y_0,today]$ for individual $i$ in population $p$:

$$DBH_{y=today,p,i} \sim \mathcal{logN} (DBH_0 + \sum _{y=y_0} ^{y=today} Gmax_i.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{Dopt_i})}{Ks_i}]^2), \sigma)$$
$$\mid \forall \theta \in [Gmax, Dopt,Ks], \theta_i \sim \mathcal N (\theta_p+A.a_\theta, \sigma^2_{R_{\theta}}),a_\theta\sim\mathcal N(0,\sigma_{G_\theta})$$

with $Gmax$, $Dopt$ and $Ks$ the parameters of the growth response to ontgeny representing respectively the maximum growth potential, the optimal growth diameter when this potential is reached, and the kurtosis of the growth curve respectivelly. Ontogenic parameters were defined at the individual level $i$ as random effects centered on the sum of population level $p$ parameters and additive gentic variance $a_{theta}$ with associated individual variance $\sigma_R$. Parameter additive gentic variance $a_{theta}$ follow a multivariate normal law centered on 0 of covariance $\sigma_G K$.

Growth response to ontogeny didn't change compared to the previous paragraph, and won't be discussed here. For the three ontogenetic parameters ($Gmax$, $Dopt$, and $Ks$), most of the variability was explained by the population (ca 50%) and around 30% of the remaining variability was explained by the genotype (Tab. \@ref(tab:growthKinshipTab) and Fig. \@ref(fig:growthKinshipG1) and \@ref(fig:growthKinshipG1)). Growth parameters were more differentiated between population than random genetic variation ($Qst >> Fst$). These results highlights a strong local adptation with *Symphonia*.

```{r growthKinshipTab}
# growth <- stan_model("./symcapture_models/GrowthAnimal.stan")
# fitGrowthKin <- sampling(growth, chains = 2, save_warmup = F,
#                          control = list(adapt_delta = 0.9, max_treedepth = 12), data = mdata(trees))
# save(fitGrowthKin, file = file.path("symcapture_save", 'growthkin2.Rdata'))
# as.data.frame(fitGrowthKin, pars = c("Gmaxi", "Dopti", "Ksi")) %>% 
#   summarise_all(mean) %>%
#   reshape2::melt(NULL) %>% 
#   separate(variable, c("parameter", "IndNum"), "[:punct:]", convert = T) %>% 
#   reshape2::dcast(IndNum ~ parameter) %>%
#     left_join(dplyr::select(trees, IndNum, FID, IID) %>% unique()) %>% 
#   dplyr::select(FID, IID, Gmaxi, Dopti, Ksi) %>%
#   write_tsv(file.path(path, "gemma", "growth", "growth.phenotype"), col_names = F)
load(file.path("symcapture_save", 'growthkin.Rdata'))
fitGrowthKin0 <- fitGrowthKin
load(file.path("symcapture_save", 'growthkin2.Rdata'))
broom::tidyMCMC(fitGrowthKin, c("sigmaP", "sigmaG", "sigmaR", 
                                "sigma", "lp__"), droppars = NULL, rhat = T) %>% 
  separate(term, c("sigma", "parameter"), "[:punct:]", convert = T) %>% 
  mutate(parameter = recode(parameter, "1" = "Gmax",  "2" = "Dopt", "3" = "Ks")) %>% 
  mutate(parameter = ifelse(is.na(parameter), "", parameter)) %>% 
  mutate(sigma = recode(sigma, "sigmaP" = "$\\sigma_P$", "sigmaG" = "$\\sigma_G$", "sigmaR" = "$\\sigma_R$", 
                            "sigma" = "$\\sigma$", "lp" = "loglikelihood")) %>% 
  kable(caption = "Summary table of the kinship growth model", 
        col.names = c("Parameter", "$\\theta$", "Estimate", "$\\sigma$", "$\\hat{R}$"))
```

```{r}
load(file.path("symcapture_save", 'growthkin2.Rdata'))
mcmc_trace(as.array(fitGrowthKin, pars = c("Gmaxp", "Doptp", "Ksp", "sigmaG", "sigmaR", "sigma")))
# mcmc_pairs(as.array(fitGrowthKin, pars = c("Gmaxg", "Doptg", "Ksg")))
```

```{r growthKinshipG1, fig.cap="Heritability $h^2$, broad heritability $h_p^2$ and trait differentiation $Qst$ for ontogenetic parameters."}
mcmc_intervals_data(fitGrowthKin, regex_pars = c("h2", "h2p", "Qst")) %>% 
  separate(parameter, c("index", "parameter"), "[:punct:]") %>% 
  mutate(parameter = recode(parameter, "1" = "Gmax", "2" = "Dopt", "3" = "Ks")) %>% 
  mutate(index = recode(index, "h2" = "h^2 == frac(sigma[G]^2,sigma[P]^2+sigma[G]^2+sigma[R]^2)", 
                        "h2p" = "h[p]^2 == frac(sigma[P]^2+sigma[G]^2,sigma[P]^2+sigma[G]^2+sigma[R]^2)", 
                        "Qst" = "Q[st] == frac(sigma[P]^2,sigma[P]^2+2*sigma[G]^2)")) %>% 
  ggplot(aes(x = parameter, xend = parameter, col = parameter, fill = parameter)) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  coord_flip() +
  facet_wrap(~ index, labeller = label_parsed) +
  xlab("") + ylab("") + 
  scale_fill_discrete(guide = "none") +
  scale_color_discrete(guide = "none")
```

```{r growthKinshipG2, fig.cap="Genetic variance partitionning for ontogenetic parameters."}
mcmc_intervals_data(fitGrowthKin, regex_pars = c("sigmaP", "sigmaG", "sigmaR")) %>%
  separate(parameter, c("sigma", "parameter"), "[:punct:]") %>% 
  mutate(parameter = recode(parameter, "1" = "Gmax", "2" = "Dopt", "3" = "Ks")) %>% 
  mutate(sigma = recode(sigma, "sigmaP" = "Population", "sigmaG" = "Genotype", "sigmaR" = "Residual")) %>% 
  mutate(sigma = factor(sigma, levels = c("Population", "Genotype", "Residual"))) %>% 
  group_by(parameter) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>% 
  ggplot(aes(x = parameter, fill = sigma)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  facet_wrap(~ parameter, scales = "free") +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2))
```

## Associations

SNP association to growth parameters have been assessed on growth parametrs per individual ($Gmax_i$, $Dopt_i$, $Ks_i$) with mono- and poly-genic modelling (respectivelly LMM and BSLMM, see methods in environmental genomics). Both mono- and poly-genic modelling didn't detected any significant SNP (Fig. \@ref(fig:growthMLM) and \@ref(fig:growthBSLMM)), at the exception of one neutral SNPs for $Dopt$. Moreover, polygenic modelling with bayesian sparse linear mixed models (BSLMM) didn't converged due to a small number of individuals (Fig. \@ref(fig:growthBSLMMConvergence)).

```{bash Gemma, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
cat paracou3pop.popmap | awk '{print"0\t"$1"\t0\t0\t0\t-9"}' > paracou3pop.fam
for par in $(seq 1 3) ; do plink \
  --bfile ../../../variantCalling/paracou3pop/symcapture.all.biallelic.snp.filtered.nonmissing.paracou3pop \
  --allow-extra-chr --keep paracou3pop.fam \
  --extract ../../../variantCalling/paracou3pop/LD.prune.in \
  --maf 0.05 --pheno growth.phenotype --mpheno $par --out paracou3pop.$par --make-bed ; done
gemma -bfile paracou3pop.1 -gk 1 -o kinship
for par in $(seq 1 3) ; do  gemma -bfile paracou3pop.$par -k output/kinship.cXX.txt -lmm 1 -o LMM.$par ; done
for par in $(seq 1 3) ; do  gemma -bfile paracou3pop.$par -k -bslmm 1  -o BSLMM.$par & done
```

```{r growthMLM, fig.cap="q-value of major effect SNPs associated to individual growth parameters. SNP were detected using linear mixed models (LMM) with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
lapply(list.files(file.path(path, "gemma", "growth"), pattern = ".assoc.txt$", full.names = T), read_tsv)%>% 
  bind_rows(.id = "parameter") %>% 
  mutate(parameter = recode(parameter, "1" = "Gmax",  "2" = "Dopt",  "3" = "Ks")) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  mutate(qval = p.adjust(p_wald, "fdr")) %>%
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  na.omit(qval) %>% 
  ggplot(aes(snp, -log10(qval), label = snp, alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ylab(expression(-log[10]("q-value"))) +
  facet_wrap(~ parameter) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type")
```

```{r growthBSLMMConvergence, fig.cap="Traceplot of hyper-parameters chain for Baysian Sparse Linear Mixed Model (BSLMM)."}
cowplot::plot_grid(plotlist = lapply(list.files(file.path(path, "gemma", "growth"), pattern = ".hyp.txt$", full.names = T), 
       function(file) read_tsv(file) %>% 
         dplyr::select(pge, n_gamma) %>% 
         dplyr::rename(PGE = pge, `n[gamma]` = n_gamma) %>% 
         mutate(iteration = 1:n()) %>% 
         filter(row_number() %% 100 == 0) %>% 
         dplyr::select(-iteration) %>% 
         mcmc_combo()), labels = c("Gmax", "Dopt", "Ks"), nrow = 3)  
```

```{r growthBSLMM, fig.cap="Effect of SNPs associated to individual growth parameters with a polygenic model. SNP were detected using bayesian sparse linear mixed models (BSLMM) with individual kiniship as random effect. SNP significance was assessed with the $\\gamma$ parameter and highlighted here with SNPs having at least 90% chance of having an effect on the phenotype."}
lapply(list.files(file.path(path, "gemma", "growth"), pattern = ".param.txt$", full.names = T), read_tsv) %>% 
  bind_rows(.id = "parameter") %>% 
  mutate(parameter = recode(parameter, "1" = "Gmax",  "2" = "Dopt",  "3" = "Ks")) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
              dplyr::rename(rs = snp)) %>% 
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  ggplot(aes(snp, beta*gamma, label = snp, col = type, alpha = gamma > 0.5)) +
  geom_point() +
  scale_y_continuous(trans="S_sqrt") +
  facet_wrap(~ parameter) +
  ylab(expression(beta*gamma))  
```
