```{r setup_growthgenom, include=FALSE, eval=T}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(bayesplot)
library(rstan)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F, eval=T)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
pathCluster <- "~/Remotes/genotoul/work/PhD/documents/genomics/symcapture"
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
pairs_stan <- function(chain, stan_model, pars) {
  energy <- as.matrix(sapply(get_sampler_params(stan_model, inc_warmup = F), 
                             function(x) x[,"energy__"]))
  pars <- rstan::extract(stan_model, pars = pars, permuted = F)
  df <- data.frame(energy[,chain], pars[,chain,])
  names(df)[1] <- "energy"
  GGally::ggpairs(df, title = paste0("Chain", chain), 
                  lower = list(continuous = GGally::wrap("points", alpha = 0.2)))
}
crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
```

# Growth genomics

We sampled growth and ontogeny models developped in the chapter "Growth & Animal models". Then, we derived growth parameters to be used in association genomics for mono- and poly-genic signals. 

```{r DataGrowthGenom}
load("./symcapture_save/Env.Rdata")
env <- trees
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Symphonia") %>% 
  mutate(DBH = CircCorr/pi) %>% 
  filter(!(CodeMeas %in% c(4))) %>% 
  collect()
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop"))) %>% 
  left_join(read_tsv(file.path(path, "populations", "paracou.hybridmap")),
            by = "Ind", suffix = c("", ".hybrid"))
trees0 <- trees
trees <- trees %>% 
  group_by(Ind) %>% 
  mutate(Y0 = dplyr::first(CensusYear), DBH0 = dplyr::first(DBH), 
         DBHtoday = dplyr::last(DBH), N = n()) %>%  
  ungroup() %>% 
  dplyr::select(Ind, IID, pop, Plot, SubPlot, TreeFieldNum, Y0, DBH0, DBHtoday, N) %>% 
  unique() %>% 
  mutate(DBHtoday = ifelse(DBHtoday == DBH0, DBHtoday + 0.1, DBHtoday))
trees <- trees %>%
  left_join(env)
rm(env)
save(trees, file = "./symcapture_save/Env2.Rdata") 
```

```{r P16Env, eval=F}
p16 <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Plot == 1) %>% 
  filter(CensusYear == 2015) %>% 
  filter(Xfield > 20, Xfield < 250-20, Yfield > 20, Yfield < 250-20) %>% 
  collect() # ! individuals on edge to be removed
p16XY <- p16
coordinates(p16XY) <- ~Xutm + Yutm
proj4string(p16XY) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
p16XY <- spTransform(p16XY, CRSobj = crs)
wetness <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "TWI_1m.tif"))
dem <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
wetness <- projectRaster(wetness, crs = crs)
p16$TWI <- raster::extract(wetness, p16XY)
rm(dem, wetness, p16XY)
p16$TWI[p16$TWI < 0] <- 0
cl <- parallel::makeCluster(getOption("cl.cores", 4))
parallel::clusterExport(cl, list("p16"))
NC <- parallel::parLapply(cl, 1:nrow(p16), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
    tbl("Paracou") %>% 
    filter(Plot == local(p16$Plot[ind])) %>% 
    filter(idTree != local(p16$idTree[ind])) %>% 
    mutate(dij = sqrt((local(p16$Xutm[ind]) - Xutm)^2+(local(p16$Yutm[ind]) - Yutm)^2)) %>% 
    filter(dij < 20) %>% 
    mutate(con = ifelse(Genus == local(p16$Genus[ind]) && Species == local(p16$Species[ind]), 1, 0)) %>% 
    mutate(DBH = CircCorr/pi) %>% 
    collect() %>% 
    group_by(CensusYear) %>% 
    summarise(NCI = sum(DBH*DBH*exp(-0.25*dij))) %>% 
    ungroup() %>% 
    summarise(idTree = local(p16$idTree[ind]),
              NCI = mean(NCI))})
parallel::stopCluster(cl) ; rm(cl)
NC <- bind_rows(NC)
p16 <- left_join(p16, NC)
rm(NC)
DBH <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Plot == 1) %>% 
  filter(Xfield > 20, Xfield < 250-20, Yfield > 20, Yfield < 250-20) %>% 
  mutate(DBH = CircCorr/pi) %>% 
  collect() %>% 
  group_by(idTree) %>% 
  mutate(Y0 = dplyr::first(CensusYear), DBH0 = dplyr::first(DBH), 
         DBHtoday = dplyr::last(DBH)) %>%  
  ungroup() %>% 
  dplyr::select(idTree, Y0, DBH0, DBHtoday) %>% 
  unique() %>% 
  mutate(DBHtoday = ifelse(DBHtoday == DBH0, DBHtoday + 0.1, DBHtoday))
p16 <- left_join(p16, DBH)
rm(DBH)
save(p16, file = "./symcapture_save/p16.Rdata")
```

```{r mdata}
trees <- trees %>% 
  na.omit(pop) %>% # remove admixed
  mutate(IndNum = 1:nrow(.), popNum = as.numeric(as.factor(pop)))
# write_tsv(dplyr::select(trees, IID),
#           file.path(path, "populations", "growth.ind"), col_names = F)
# awk '{print "0\t"$1"\t0\t0\t0\t-9"}' growth.ind > growth.fam
# fam <- read_delim(file.path(path, "gemma", "Gmax0", "paracou3pop.fam"), delim = " ", 
#                   col_names = c("FID", "IID", "PAT", "MAT", "SEX", "PHENOTYPE"))
# K <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
#                         "out.relatedness2"))  %>% 
#   reshape2::dcast(INDV1 ~ INDV2, value.var = "RELATEDNESS_PHI") %>% 
#   column_to_rownames("INDV1") %>% 
#   as.matrix()
# fam <- read_delim(file.path(path,"..", "variantCalling", "growth", "growth.fam"), delim = " ", 
#                   col_names = c("FID", "IID", "PAT", "MAT", "SEX", "PHENOTYPE"))
# K <- read_tsv(file.path(path, "..", "variantCalling", "growth",
#                         "out.relatedness2"))  %>% 
#   reshape2::dcast(INDV1 ~ INDV2, value.var = "RELATEDNESS_PHI") %>% 
#   column_to_rownames("INDV1") %>% 
#   as.matrix()
# plink2 --bfile growth --allow-extra-chr --make-king square
ids <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king.id"))
K <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king"),
         col_names = ids$IID) %>% 
  as.data.frame()
row.names(K) <- ids$IID
K <- as.matrix(K)
model_data <- function(data) {
  K <- K[trees$IID, trees$IID]
  K[K < 0] <- 0
  K <- K*2
  K <- as.matrix(Matrix::nearPD(K)$mat)
  list(I = nrow(data),
       Y = 2017 - min(data$Y0) + 1,
       P = length(unique(data$popNum)),
       years = min(data$Y0):2017,
       DBH0 = data$DBH0,
       Y0 = data$Y0,
       DBHtoday = data$DBHtoday + 10^-6,
       TWI = as.vector(scale(data$TWI, center = F)),
       NCI = as.vector(scale(data$NCI, center = F)),
       ind = data$IndNum,
       pop = data$popNum,
       K = K)
}
mdata <- model_data(trees)
save(mdata, file = file.path("symcapture_save", "dataGrowthPar.Rdata")) 
```

## Gmax

We used the following growth model with a lognormal distribution to estimate individual growth potential:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \theta_{1,p,i}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\theta_{2,p}})}{\theta_{3,p}}]^2)), \sigma_1) \\ 
  \theta_{1,p,i} \sim \mathcal {logN}(log(\theta_{1,p}), \sigma_2) \\ 
  \theta_{2,p} \sim \mathcal {logN}(log(\theta_2),\sigma_3) \\ 
  \theta_{3,p} \sim \mathcal {logN}(log(\theta_3),\sigma_4)
  (\#eq:gmax)
\end{equation} 

We fitted the equivalent model with following priors:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \hat{\theta_{1,p,i}}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\hat{\theta_{2,p}}})}{\hat{\theta_{3,p}}}]^2)), \sigma_1) \\ 
  \hat{\theta_{1,p,i}} = e^{log(\theta_{1,p}) + \sigma_2.\epsilon_{1,i}} \\ 
  \hat{\theta_{2,p}}  = e^{log(\theta_2) + \sigma_3.\epsilon_{2,p}} \\ 
  \hat{\theta_{3,p}}  = e^{log(\theta_3) + \sigma_4.\epsilon_{3,p}} \\  
  \epsilon_{1,i} \sim \mathcal{N}(0,1) \\
  \epsilon_{2,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{3,p} \sim \mathcal{N}(0,1) \\
  ~ \\
  (\theta_{1,p}, \theta_2, \theta_3) \sim \mathcal{logN}^3(log(1),1) \\ 
  (\sigma_1, \sigma_2, \sigma_3, \sigma_4) \sim \mathcal{N}^4_T(0,1) \\ 
    ~ \\
  V_P = Var(log(\mu_p)) \\ 
  V_R=\sigma_2^2
  (\#eq:gmaxstan)
\end{equation} 

```{bash fitGmaxParCluster, eval=F}
for chain in $(seq 8) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript Gmax.R $chain" ; done > Gmax.sh
sarray -J Gmax -o out/%j.Gmax.out -e out/%j.Gmax.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL Gmax.sh
```

```{r gmaxParTab}
fitGmax <- list()
for(sim in list.files("symcapture_save/Gmax", full.names = T)){
  load(sim)
  fitGmax <- c(fitGmax, fit)
}
fitGmax <- sflist2stanfit(fitGmax)
fitGmaxSympho <- fitGmax
save(fitGmaxSympho, file = "../../thesis/rdata/fitGmaxSympho.Rdata")
rm(fitGmaxSympho)
# as.data.frame(fitGmax, "thetai1") %>%
#   summarise_all(median) %>%
#   reshape2::melt(NULL) %>%
#   separate(variable, c("theta", "IndNum"), convert = T) %>%
#   dplyr::select(-theta) %>%
#   left_join(dplyr::select(trees, IndNum, IID)) %>%
#   dplyr::select(-IndNum) %>%
#   mutate(FID = 0) %>%
#   dplyr::select(FID, IID, value) %>%
#   write_tsv(file.path(path, "gemma", "gmax", "gmax.phenotype"),
#             col_names = F) # for genomic scans
broom::tidyMCMC(fitGmax, pars = c("thetap1", "theta2", "theta3",
                                   "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r gmaxParTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmax, pars = c("thetap1", "theta2", "theta3",
                                      "sigma", "lp__")),
           np = nuts_params(fitGmax)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r gmaxParPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmax, c("sigma"))
```

```{r gmaxParEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmax))
```

```{r gmaxParPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[1]"))))/
                 median(as.array(fitGmax, "thetap3[1]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[2]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[2]"))))/
                 median(as.array(fitGmax, "thetap3[2]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGmax, "thetap1[3]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmax, "thetap2[3]"))))/
                 median(as.array(fitGmax, "thetap3[3]")))^2))
```

```{r GmaxR2, fig.cap="R2 for Gmax."}
as.data.frame(fitGmax, c("Vp", "Vr")) %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vr))) %>% 
  mutate_at(c("Vp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = NULL) %>% 
  group_by(variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +
  theme(axis.title.y = element_blank()) +
  coord_flip()
```

```{r GmaxVarPart, fig.cap="Genetic variance partitionning for Gmax."}
 mcmc_intervals_data(fitGmax, regex_pars = c("Vp", "Vr")) %>% 
  mutate(parameter = recode(parameter, "Vp" = "Population", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = parameter)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2)) 
```

## Gmax & Genotype

We used the following growth model with a lognormal distribution to estimate individual growth potential and associated genotypic variation:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \theta_{1,p,i}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\theta_{2,p}})}{\theta_{3,p}}]^2)), \sigma_1) \\ 
  \theta_{1,p,i} \sim \mathcal {logN}(log(\theta_{1,p}.a_{1,i}), \sigma_2) \\ 
  \theta_{2,p} \sim \mathcal {logN}(log(\theta_2),\sigma_3) \\ 
  \theta_{3,p} \sim \mathcal {logN}(log(\theta_3),\sigma_4) \\
  a_{1,i} \sim \mathcal{MVlogN}(log(1), \sigma_5.K)
  (\#eq:gmaxgeno)
\end{equation} 

We fitted the equivalent model with following priors:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \hat{\theta_{1,p,i}}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\hat{\theta_{2,p}}})}{\hat{\theta_{3,p}}}]^2)), \sigma_1) \\ 
  \hat{\theta_{1,p,i}} = e^{log(\theta_{1,p}.\hat{a_{1,i}}) + \sigma_2.\epsilon_{1,i}} \\ 
  \hat{\theta_{2,p}}  = e^{log(\theta_2) + \sigma_3.\epsilon_{2,p}} \\ 
  \hat{\theta_{3,p}}  = e^{log(\theta_3) + \sigma_4.\epsilon_{3,p}} \\  
  \hat{a_{1,i}} = e^{\sigma_5.A.\epsilon_{4,i}} \\
  \epsilon_{1,i} \sim \mathcal{N}(0,1) \\
  \epsilon_{2,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{3,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{4,i} \sim \mathcal{N}(0,1) \\
  ~ \\
  (\theta_{1,p}, \theta_2, \theta_3) \sim \mathcal{logN}^3(log(1),1) \\ 
  (\sigma_1, \sigma_2, \sigma_3, \sigma_4, \sigma_5) \sim \mathcal{N}^5_T(0,1) \\ 
    ~ \\ 
    V_P = Var(log(\mu_p)) \\
    V_G=\sigma_5^2\\
    V_R=\sigma_2^2
  (\#eq:gmaxgenostan)
\end{equation} 

```{bash fitGmaxGenoParCluster, eval=F}
for chain in $(seq 8) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GmaxGeno.R $chain" ; done > GmaxGeno.sh
sarray -J GmaxGeno -o out/%j.GmaxGeno.out -e out/%j.GmaxGeno.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GmaxGeno.sh
```

```{r gmaxgenoParTab}
fitGmaxGeno <- list()
for(sim in list.files("symcapture_save/GmaxGeno", full.names = T)){
  load(sim)
  fitGmaxGeno <- c(fitGmaxGeno, fit)
}
fitGmaxGeno <- sflist2stanfit(fitGmaxGeno)
fitGmaxGenoSympho <- fitGmaxGeno
save(fitGmaxGenoSympho, file = "../../thesis/rdata/fitGmaxGenoSympho.Rdata")
rm(fitGmaxGenoSympho)
# broom.mixed::(fitGmaxGeno, pars = c("thetap1", "theta2", "theta3", "sigma", "lp__"),
#                 droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
#   dplyr::select(term, estimate, std.error, rhat, ess) %>% 
#   kable(caption = "Individual growth potential model.",
#         col.names = c("Parameter", "Estimate", "Standard error", 
#                       "$\\hat R$", "$N_{eff}$"))
```

```{r gmaxgenoParTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmaxGeno, 
                    pars = c("thetap1", "theta2", "theta3", "sigma", "lp__")),
           np = nuts_params(fitGmaxGeno)) +
  viridis::scale_color_viridis(discrete = T)
```

```{r gmaxgenoParPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmaxGeno, c("sigma"))
```

```{r gmaxgenoParEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmaxGeno))
```

```{r gmaxgeno1ParPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGmaxGeno, "thetap1[1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGeno, "thetap2[1]"))))/
                 median(as.array(fitGmaxGeno, "thetap3[1]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGmaxGeno, "thetap1[2]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGeno, "thetap2[2]"))))/
                 median(as.array(fitGmaxGeno, "thetap3[2]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGmaxGeno, "thetap1[3]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGeno, "thetap2[3]"))))/
                 median(as.array(fitGmaxGeno, "thetap3[3]")))^2))
```

```{r GmaxGenoR2, fig.cap="R2 for Gmax."}
as.data.frame(fitGmaxGeno, c("Vp", "Vg", "Vr")) %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vg, Vr))) %>% 
  mutate(Vexp = sum(c(Vp, Vg))) %>% 
  mutate_at(c("Vp", "Vg", "Vexp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = NULL) %>% 
  group_by(variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, 
                           "Vexp" = "Marginal", "Vg" = "Genotype", 
                           "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +
  theme(axis.title.y = element_blank()) +
  coord_flip()
```

```{r GmaxGenoVarPart, fig.cap="Genetic variance partitionning for Gmax."}
 mcmc_intervals_data(fitGmaxGeno, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  mutate(parameter = recode(parameter, "Vp" = "Population", 
                            "Vg" = "Genotype", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = parameter)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2)) 
```

```{r VarVsGenetic, fig.cap="Long legend...", fig.height=8, fig.width=8}
fitNCI <- list()
for(sim in list.files("symcapture_save/EnvGeno", 
                      pattern = "NCI", full.names = T)){
  load(sim)
  fitNCI <- c(fitNCI, fit)
}
fitNCI <- sflist2stanfit(fitNCI)
fitTWI <- list()
for(sim in list.files("symcapture_save/EnvGeno", 
                      pattern = "TWI", full.names = T)){
  load(sim)
  fitTWI <- c(fitTWI, fit)
}
fitTWI <- sflist2stanfit(fitTWI)
t <- lapply(c(fitNCI, fitTWI, fitGmaxGeno), function(fit)
  as.data.frame(fit, "alog") %>%
  summarise_all(median) %>%
  reshape2::melt(NULL)) %>%
  bind_rows(.id = "fit") %>% 
  mutate(fit = recode(fit, "1" = "aNCI", "2" = "aTWI", "3" = "aGmax")) %>% 
  separate(variable, c("X1", "IndNum", "X2"), convert = T) %>% 
  mutate(value = exp(value)) %>% 
  reshape2::dcast(IndNum ~ fit, value.var = "value") %>% 
    left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "gemma", "gmax", "gmax.phenotype"),
                     col_names = c("FID", "IID", "Gmax")))
## Up
g.a.NCI.Gmax <- ggplot(t, aes(aGmax, aNCI, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "lm", col = "black") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank()) +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = 0.85)
g.a.TWI.Gmax <- ggplot(t, aes(aGmax, aTWI, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "lm", col = "darkgrey") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank()) +  ggtitle("Gmax") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = 0.8, col = "darkgrey")
g.a.TWI.NCI <- ggplot(t, aes(aNCI, aTWI, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "lm", col = "darkgrey") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank()) +  ggtitle("NCI") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.8, label.y = 0.85, col = "darkgrey")
## Diag
g.TWI <- ggplot(t, aes(aTWI, TWI, col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  theme(axis.title.x = element_blank()) +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  ylab("TWI") + ggtitle("TWI") +
  annotate("text", x = 0.9, y = 7, label = expression(R[m]^2==0.35)) +
  annotate("text", x = 0.9, y = 6, label = expression(R[c]^2==0.44))
g.NCI <- ggplot(t, aes(aNCI, NCI/10^4, col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
    theme(axis.title = element_blank()) +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
    annotate("text", x = 0.9, y = 0.7, label = expression(R[m]^2==0.06)) +
  annotate("text", x = 0.9, y = 0.6, label = expression(R[c]^2==0.24))
g.gmax <- ggplot(t, aes(aGmax, (DBHtoday - DBH0)/(2017-Y0), 
                        col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
    theme(axis.title = element_blank()) +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
      annotate("text", x = 0.5, y = 1.1, label = expression(R[m]^2==0.10)) +
  annotate("text", x = 0.5, y = 0.9, label = expression(R[c]^2==0.53))
## Down
load(file = "./symcapture_save/p16.Rdata")
g.var.DBH.NCI <- ggplot(t, aes(NCI/10^4, (DBHtoday - DBH0)/(2017-Y0), 
                               size = DBHtoday)) +
  geom_hex(data = p16, fill = "lightgrey", alpha = 0.6, bins = 50) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank())
g.var.DBH.TWI <- ggplot(t, aes(TWI, (DBHtoday - DBH0)/(2017-Y0), 
                        size = DBHtoday)) +
  geom_hex(data = p16, fill = "lightgrey", alpha = 0.6) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title.x = element_blank()) +
  ylab("Gmax")
g.var.NCI.TWI <- ggplot(t, aes(TWI, NCI/10^4, size = DBHtoday)) +
  geom_hex(data = p16, fill = "lightgrey", alpha = 0.6) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title.x = element_blank()) +
  ylab("NCI")
# all
cowplot::plot_grid(g.TWI, g.a.TWI.NCI, g.a.TWI.Gmax,
                   g.var.NCI.TWI, g.NCI, g.a.NCI.Gmax,
                   g.var.DBH.TWI, g.var.DBH.NCI, g.gmax,
                   ncol=3, labels = LETTERS[1:9])
# summary(MASS::rlm(aGmax ~ aNCI, data = t, psi = "psi.huber"))
# summary(robustbase::lmrob(aGmax ~ aNCI, data = t))
```
```{r, eval=F}
ggplot(t, aes(x = Gmax, fill = pop)) + 
  geom_density(alpha = 0.5)
  geom_histogram(aes(y = ..density..), position = "dodge", bins = 10)
```

```{r }
t %>% 
  arrange(DBHtoday) %>% 
  mutate(DBHclass = rep(1:5, each = 75)[1:371]) %>% 
  ggplot(aes(aGmax, aNCI, size = DBHtoday)) +
  geom_point() +
  facet_wrap(~ DBHclass, scales = "free") +
  geom_smooth(method = "lm", col = "black") +
  scale_size_continuous(guide = "none") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = 0.9)
```

```{r spcorrelo, fig.cap="Spatial autocorrelogram (Moran\'s I) of variables and associated genetic multiplicative values."}
cor <- lapply(c(TWI = "TWI", NCI = "NCI", Gmax = "Gmax", 
                gTWI = "aTWI", gNCI = "aNCI", gGmax = "aGmax"), function(var) 
  pgirmess::correlog(data.frame(t$Xutm, t$Yutm), t[,var], method = "Moran", nbclass = 100) %>% 
    as.data.frame()) %>% 
  bind_rows(.id = "variable")
cor %>% 
  filter(dist.class < 10^3) %>% 
  ggplot(aes(x = dist.class, y = coef, col = variable)) + geom_hline(yintercept = 0) +
  geom_point(aes(alpha = p.value < 0.01)) + geom_line() +
  scale_alpha_manual("p < 0.01", values = c(0.3, 1)) + scale_x_log10() +
  geom_vline(xintercept = c(27, 53), col = "lightgrey", linetype = "dashed") +
  geom_vline(xintercept = 150, col = "lightgrey", linetype = "dotted") +
  xlab("Distance (m)") + ylab("Moran\'s I") + scale_color_discrete("Variable")
```

```{r, eval=F}
load("../../functional/functional_save/Individuals.Rdata")
left_join(t, dplyr::select(Individuals, idTree, SLA, LDMC, LT, LA, CC)) %>% 
  filter(DBHtoday > 14, DBHtoday < 46) %>% 
  mutate(logLA = log(LA)) %>% 
  dplyr::select(pop, idTree, aNCI, aGmax, SLA, LDMC, LT, logLA, CC, DBHtoday) %>% 
  reshape2::melt(c("pop", "idTree", "aNCI", "aGmax"), variable.name = "trait") %>% 
  group_by(trait) %>% 
  mutate(value = as.vector(scale(value))) %>% 
  reshape2::melt(c("pop", "idTree", "trait", "value"),
                 value.name = "geneticValue") %>% 
  na.omit() %>% 
  ggplot(aes(geneticValue, value)) +
  geom_point(alpha = 0.3) +
  facet_grid(trait ~ variable, scales = "free") +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(method = "pearson", label.x = 1, label.y = -3) +
  xlab("Genotypic value") +
  ylab("Standardized trait value") +
  theme(legend.position = "bottomright")
```

```{r, eval=F}
ggplot(t, aes(DBHtoday, Gmax)) + 
  geom_point() + 
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(method = "pearson", label.x = 40, label.y = 0)
ggplot(t, aes(DBHtoday, aGmax)) + 
  geom_point() + 
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(method = "pearson", label.x = 40, label.y = 0.5)
```


```{r outlines, eval=F}
save(fitGmaxGeno, fitNCI, t, cor, file = "symcapture_save/outlines.Rdata")
```

## Gmax, Genotype & Environment

We used the following growth model with a lognormal distribution to estimate individual growth potential and associated genotypic variation:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \theta_{1,p,i}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\theta_{2,p}})}{\theta_{3,p}}]^2)), \sigma_1) \\ 
  \theta_{1,p,i} \sim \mathcal {logN}(log(\theta_{1,p}.a_{1,i}. \beta_1.TWI_i.\beta_2.NCI_i), \sigma_2) \\ 
  \theta_{2,p} \sim \mathcal {logN}(log(\theta_2),\sigma_3) \\ 
  \theta_{3,p} \sim \mathcal {logN}(log(\theta_3),\sigma_4) \\
  a_{1,i} \sim \mathcal{MVlogN}(log(1), \sigma_5.K)
  (\#eq:gmaxgenoenv)
\end{equation} 

We fitted the equivalent model with following priors:

\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \hat{\theta_{1,p,i}}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\hat{\theta_{2,p}}})}{\hat{\theta_{3,p}}}]^2)), \sigma_1) \\ 
  \hat{\theta_{1,p,i}} = e^{log(\theta_{1,p}.\hat{a_{1,i}}. \beta_1.TWI_i.\beta_2.NCI_i) + \sigma_2.\epsilon_{1,i}} \\ 
  \hat{\theta_{2,p}}  = e^{log(\theta_2) + \sigma_3.\epsilon_{2,p}} \\ 
  \hat{\theta_{3,p}}  = e^{log(\theta_3) + \sigma_4.\epsilon_{3,p}} \\  
  \hat{a_{1,i}} = e^{\sigma_5.A.\epsilon_{4,i}} \\
  \epsilon_{1,i} \sim \mathcal{N}(0,1) \\
  \epsilon_{2,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{3,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{4,i} \sim \mathcal{N}(0,1) \\
  ~ \\
  (\theta_{1,p}, \theta_2, \theta_3) \sim \mathcal{logN}^3(log(1),1) \\ 
  (\sigma_1, \sigma_2, \sigma_3, \sigma_4, \sigma_5) \sim \mathcal{N}^5_T(0,1) \\ 
    ~ \\ 
    V_P = Var(log(\mu_p)) \\
    V_G=\sigma_5^2\\
    V_{TWI} = Var(log(\beta_1.TWI_i)) \\
    V_{NCI} = Var(log(\beta_2.NCI_i)) \\
    V_R=\sigma_2^2
  (\#eq:gmaxgenoenvstan)
\end{equation} 

```{bash fitGmaxGenoEnvCluster, eval=F}
for chain in $(seq 8) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GmaxGenoEnv.R $chain" ; done > GmaxGenoEnv.sh
sarray -J GmGE -o out/%j.GmGE.out -e out/%j.GmGE.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GmaxGenoEnv.sh
```

```{r gmaxgenoenvTab}
fitGmaxGenoEnv <- list()
for(sim in list.files("symcapture_save/GmaxGenoEnv", full.names = T)){
  load(sim)
  fitGmaxGenoEnv <- c(fitGmaxGenoEnv, fit)
}
fitGmaxGenoEnv <- sflist2stanfit(fitGmaxGenoEnv)
broom::tidyMCMC(fitGmaxGenoEnv, pars = c("thetap1", "beta", 
                                         "theta2", "theta3",
                                         "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$")) 
```

```{r gmaxgenoenvTrace, fig.cap="Traceplot of model parameters."}
mcmc_trace(as.array(fitGmaxGenoEnv, pars = c("thetap1", "beta",
                                           "theta2", "theta3",
                                          "sigma", "lp__")),
           np = nuts_params(fitGmaxGenoEnv)) +
  viridis::scale_color_viridis(discrete = T) 
```

```{r gmaxgenoenvPairs, message=FALSE, fig.cap="Pairs of model parameters."}
pairs_stan(1, fitGmaxGenoEnv, c("sigma")) 
```

```{r gmaxgenoenvEnergy, fig.cap="Energy of the model."}
mcmc_nuts_energy(nuts_params(fitGmaxGenoEnv)) 
```

```{r gmaxgenoenvPred, fig.cap="Species predicted growth curve."}
ggplot(data.frame(DBH = 0:200, AGR = 2), aes(x = DBH, y = AGR)) +
  geom_vline(xintercept = mdata$DBH0, alpha = 0.1) +
  stat_function(aes(col = "S. globulifera Paracou"), fun = function(.x) 
    median(as.array(fitGmaxGenoEnv, "thetap1[1]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGenoEnv, "thetap2[1]"))))/
                 median(as.array(fitGmaxGenoEnv, "thetap3[1]")))^2)) +
    stat_function(aes(col = "S. globulifera Regina"), fun = function(.x) 
    median(as.array(fitGmaxGenoEnv, "thetap1[2]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGenoEnv, "thetap2[2]"))))/
                 median(as.array(fitGmaxGenoEnv, "thetap3[2]")))^2)) +
    stat_function(aes(col = "S. sp1"), fun = function(.x) 
    median(as.array(fitGmaxGenoEnv, "thetap1[3]"))*
      exp(-.5*(log(.x/(100*median(as.array(fitGmaxGenoEnv, "thetap2[3]"))))/
                 median(as.array(fitGmaxGenoEnv, "thetap3[3]")))^2)) 
```

```{r GmaxGenoEnvR2, fig.cap="R2 for Gmax."}
as.data.frame(fitGmaxGenoEnv, c("Vp", "Vg", "Vnci", "Vtwi", "Vr")) %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vg, Vnci, Vtwi, Vr))) %>% 
  mutate(Vexp = sum(c(Vp, Vg, Vnci, Vtwi))) %>% 
  mutate_at(c("Vp", "Vg", "Vnci", "Vtwi", "Vexp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = NULL) %>% 
  group_by(variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, 
                           "Vexp" = "Marginal", "Vg" = "Genotype", 
                           "Vnci" = "NCI", "Vtwi" = "TWI",
                           "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +
  theme(axis.title.y = element_blank()) +
  coord_flip() 
```

```{r gmaxgenoenvVar, fig.cap="Genetic variance partitionning for ontogenetic parameters."}
mcmc_intervals_data(fitGmaxGenoEnv, regex_pars = c("Vp", "Vg", "Vnci", "Vtwi", "Vr")) %>% 
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", 
                           "Vnci" = "NCI", "Vtwi" = "TWI", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2))  
```

## Associations

SNP association to growth potential $Gmax$ has been assessed per individual with mono- and poly-genic modelling (respectivelly LMM and BSLMM, see methods in environmental genomics) for every populations. 

```{bash Gemma, eval=F, echo=T}
module load bioinfo/plink-v1.90b5.3
module load bioinfo/gemma-v0.97
pops=(globuliferaTypeParacou globuliferaTypeRegina sp1)
for pop in "${pops[@]}" ; 
  do 
    plink \
      --bfile growth \
      --allow-extra-chr \
      --keep  $pop.fam \
      --extract growth.prune.in \
      --maf 0.05 \
      --pheno gmax.phenotype \
      --out $pop --make-bed ; 
    gemma -bfile $pop -gk 1 -o kinship.$pop ;
    gemma -bfile $pop -lmm 1 \
      -k output/kinship.$pop.cXX.txt \
      -o LMM.$pop ;
done
cat LMM.snps | cut -f1 > snps.list
plink --bfile growth --allow-extra-chr --extract snps.list --out outliers --recode A
```

```{r growthLMM, fig.cap="q-value of major effect SNPs associated to individual growth parameters. SNP were detected using linear mixed models (LMM) with individual kiniship as random effect. SNP significance was assessed with Wald test. q-value was obtained correcting mutlitple testing with false discovery rate."}
lmm <- lapply(list.files(file.path(path, "gemma", "gmax"), 
                  pattern = ".assoc.txt$", full.names = T), 
       function(x) read_tsv(x) %>% mutate(file = x))%>% 
  bind_rows() %>% 
  mutate(file = gsub("/home/sylvain/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics//gemma/gmax/LMM.", "", file)) %>% 
  mutate(file = gsub(".assoc.txt", "", file)) %>% 
  dplyr::rename(population = file) %>% 
  mutate(rs = paste0(chr, "_snp", ps)) %>% 
  left_join(read_tsv(file.path(path, "..", "annotation", "snps.annotation")) %>%
  dplyr::rename(rs = snp)) %>%
  mutate(qval = p.adjust(p_wald, "fdr")) %>%
  arrange(chr, ps) %>% 
  mutate(snp = 1:nrow(.)) %>% 
  na.omit(qval)
filter(lmm , qval < 0.05) %>%
  dplyr::select(rs, population) %>%
  write_tsv(file.path(path, "gemma", "gmax", "LMM.snps"), col_names = F)
ggplot(lmm, aes(snp, -log10(qval), label = snp, alpha = qval < 0.05, col = type)) +
  geom_hline(yintercept = 2, linetype = "dashed", col = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "red") +
  geom_point() +
  ggrepel::geom_text_repel(data = lmm[lmm$qval < 0.05,]) +
  ylab(expression(-log[10]("q-value"))) +
  facet_wrap(~ population) +
  scale_alpha_discrete(guide = "none") +
  scale_y_sqrt() +
  xlab("SNP number") +
  scale_color_discrete("SNP type") +
  theme(legend.position = "bottom")
```

```{r outliersGrowth, fig.cap="Growth parameters distribution for outliers genes."}
read_delim(file.path(path, "gemma", "gmax", "outliers.raw"), delim = " ") %>% 
  left_join(read_tsv(file.path(path, "gemma", "gmax", "gmax.phenotype"), 
         col_names = c("FID", "IID", "Gmax"))) %>% 
  left_join(dplyr::select(trees, IID, pop)) %>% 
  dplyr::select(-FID, -PAT, -MAT, -SEX, -PHENOTYPE) %>% 
  reshape2::melt(id.vars = c("pop", "IID", "Gmax"),
                 variable.name = "snp", value.name = "allele") %>% 
  na.omit(allele) %>% 
  mutate(allele = as.factor(allele)) %>% 
  mutate(snp = gsub("_[[:alnum:]]$", "", snp)) %>% 
  right_join(dplyr::select(lmm[lmm$qval < 0.05,], population, rs) %>% 
               dplyr::rename(snp = rs, pop = population)) %>% 
  ggplot(aes(allele, Gmax, fill = pop)) +
  geom_boxplot() +
  stat_summary(fun.data = function(y) 
    data.frame(y = max(y)+0.1, label = paste(length(y))),
    geom = "text", hjust = 0.5, vjust = 0.9) + 
  coord_flip() +
  facet_wrap(~ pop + snp, scales = "free") +
  scale_fill_discrete("") + ylab("") + xlab("Major allele copies") +
  theme(strip.text = element_blank(), legend.position = "bottom")
```

```{r growthGO, fig.cap="Term enrichment of Gene Ontology (GO) for outliers genes.", eval=F}
trsc <- read_tsv(file.path(path, "..", "annotation", "snpsFunc.annotation")) %>% 
              dplyr::select(scf, snp, trsc) %>% 
  bind_rows(read_tsv(file.path(path, "..", "annotation", "snpsHitch.annotation"))) %>%
  unique() %>% 
  inner_join(dplyr::rename(lmm, snpNum = snp, snp = rs), .)
GO <- src_sqlite(file.path("..", "..", "..", "data", "Symphonia_Genomic", "Niklas_transcripts",
                     "Trinotate", "symphonia.trinity500.trinotate.sqlite")) %>% 
  tbl("Transcript") %>% 
  filter(transcript_id %in% local(unique(trsc$trsc))) %>% 
  dplyr::select(gene_id, transcript_id, annotation) %>% 
  dplyr::rename(gene = gene_id, trsc = transcript_id) %>% 
  collect() %>% 
  separate(annotation, paste0("X", 1:16), "\t") %>% 
  dplyr::rename(orf = X3, PFAM = X8, UniProt = X11, KEGG = X12, GO = X13) %>% 
  dplyr::select(trsc, gene, orf, GO) %>% 
  filter(GO != ".") %>% 
  left_join(trsc) %>% 
  group_by(population, gene, GO) %>% 
  summarise(qval = min(qval)) %>% 
  ungroup() %>% 
  dplyr::select(population, gene, qval, GO) %>% 
  separate_rows(GO, sep = "`") %>% 
  separate(GO, c("GOid", "GOtype", "GOname"), "\\^")
enrichment <- function(x) {
  y <- clusterProfiler::enricher(unique(filter(x, qval < 0.05)$gene),
                                            pvalueCutoff = 0.2, 
                                            pAdjustMethod = "BH",
                                            unique(x$gene),
                                            minGSSize = 10, 
                                            maxGSSize = 500, 
                                            qvalueCutoff = 0.2, 
                                            dplyr::select(x, GOid, gene),
                                            TERM2NAME = dplyr::select(x, GOid, GOname))
  if(is.null(y)){
    data.frame()
  } else {
    y@result
  }
}
group_by(GO, population) %>% 
  do(enrichment(.)) %>% 
  filter(qvalue < 0.8) %>% 
  ggplot(aes(reorder(Description, qvalue), Count, fill = -log(qvalue))) +
  geom_bar(stat = "identity") +
  coord_flip() +
  facet_wrap(~ population, labeller = "label_both", nrow = 2, scales = "free") +
  viridis::scale_fill_viridis() +
  theme(axis.title = element_blank())
```
