```{r setup_introgression, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(bayesplot)
library(rstan)
library(rstanarm)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
S_sqrt_trans <- function() scales::trans_new("S_sqrt",
                                             function(x) sign(x)*sqrt(abs(x)),
                                             function(x) x^2*sign(x))
load("./symcapture_save/introgression.Rdata")
```

```{r Dataintrogression, eval=F}
trees  <- read_tsv(file.path(path, "populations", "paracou.hybridmap")) %>% 
  left_join(src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                                 "trees", "Paracou.sqlite")) %>% 
              tbl("Paracou") %>% 
              filter(Genus == "Symphonia") %>% 
              filter(CensusYear == 2015) %>% 
              mutate(Ind = paste0("P", Plot, "-", SubPlot, "-", TreeFieldNum)) %>% 
              filter(Ind %in% local(trees$Ind)) %>% 
              collect())
treesXY <- trees
coordinates(treesXY) <- ~Xutm + Yutm
proj4string(treesXY) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
treesXY <- spTransform(treesXY, CRSobj = crs)
wetness <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "TWI_1m.tif"))
dem <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
wetness <- projectRaster(wetness, crs = crs)
trees$TWI <- raster::extract(wetness, treesXY)
rm(dem, wetness, treesXY)
cl <- parallel::makeCluster(getOption("cl.cores", 2))
parallel::clusterExport(cl, list("trees"))
NC <- parallel::parLapply(cl, 1:nrow(trees), function(ind){
  library(tidyverse)
  src_sqlite(file.path("../../../data/Paracou/", "trees", "Paracou.sqlite")) %>% 
    tbl("Paracou") %>% 
    filter(Plot == local(trees$Plot[ind])) %>% 
    filter(idTree != local(trees$idTree[ind])) %>% 
    mutate(dij = sqrt((local(trees$Xutm[ind]) - Xutm)^2+(local(trees$Yutm[ind]) - Yutm)^2)) %>% 
    filter(dij < 20) %>% 
    mutate(DBH = CircCorr/pi) %>% 
    collect() %>% 
    group_by(CensusYear) %>% 
    summarise(NCI = sum(DBH*DBH*exp(-0.25*dij))) %>% 
    ungroup() %>% 
    arrange(CensusYear) %>%
    mutate(dt = CensusYear - lag(CensusYear)) %>% 
    summarise(NCI = sum(NCI/dt, na.rm = T))})
parallel::stopCluster(cl) ; rm(cl)
NC <- bind_rows(NC)
trees$NCI <- NC$NCI
rm(NC)
trees <-  trees %>% 
  dplyr::rename(pop.hybrid = pop) %>% 
  mutate(pop = recode(pop.hybrid, 
                      "S. globulifera Paracou pure" = "S. globulifera Paracou", 
                      "S. sp1 pure" = "S. sp1", 
                      "S. globulifera Regina pure" = "S. gobulifera Regina", 
                      "S. sp1 introgressed with S. globulifera Regina" = "S. sp1",  
                      "S. globulifera Paracou introgressed with S. sp1" = "S. globulifera Paracou", 
                      "Admixed S. globulifera - S. sp1" = "Admixed", 
                      "Admixed S. sp1 - S. globulifera Regina" = "Admixed"))
trees <- trees %>% 
  left_join(src_sqlite(file.path("/home/sylvain/Documents/BIOGECO/PhD/data/Paracou/",
                                 "trees", "Paracou.sqlite")) %>% 
              tbl("Paracou") %>% 
              filter(Genus == "Symphonia") %>% 
              mutate(Ind = paste0("P", Plot, "-", SubPlot, "-", TreeFieldNum)) %>% 
              filter(Ind %in% local(trees$Ind)) %>% 
              mutate(DBH = CircCorr/pi) %>% 
              collect() %>% 
              group_by(Ind) %>% 
              mutate(Y0 = dplyr::first(CensusYear), DBH0 = dplyr::first(DBH), 
                     DBHtoday = dplyr::last(DBH), N = n()) %>%  
              ungroup() %>% 
              dplyr::select(Ind, Plot, SubPlot, TreeFieldNum, Y0, DBH0, DBHtoday, N) %>% 
              unique() %>% 
              mutate(DBHtoday = ifelse(DBHtoday == DBH0, DBHtoday + 0.1, DBHtoday)))
Gmax <- function(Y0, DBH0, DBHtoday, Dopt = 0.2540835, Ks = 0.7008939){
  DBH <- DBH0
  for(y in Y0:2016) 
    DBH = DBH + exp(-0.5*(log(DBH / (100*Dopt)) / Ks)^2)
  return((DBHtoday - DBH0)/(DBH - DBH0))
}
trees <- mutate(trees, Gmax = Gmax(Y0, DBH0, DBHtoday))
save(trees, file = "./symcapture_save/introgression.Rdata")
```

# Introgressions

Genomics structure analyses (population, envionmental and growth genomics) have been done with *pure* populations, defined with `admixture` as individuals with more than 90% of its genotype related to the genepool, rejecting admixed and introgressed individuals. In order to explore the importance of gene flow in the *Symphonia* species complex, we then explored admixed and introgressed individuals by identifying theim (`introgress`), scanning loci genomci clines (`bgc`), and exploring their distribution along environmental variables ($TWI$ and $NCI$) and individual growth potential ($Gmax$).

## Methodology

> Another demographic process that can lead to idiosyncratic effects at individual loci is hybridization and introgression from related taxa [@Hoban2016]

This justify the realization of all genomics scan without admixed individuals and the inclusion of admixed individuals a posteriori.

## Identification

We used the method developed by @Gompert2009 implemented in `introgress` [@Gompert2010] to map admixture between Paracou genepools. We used individuals with more than 90% of the genotype belonging to the genepool to define parental allele frequencies and mapped admixture between the two pairs of *S. sp1* - *S. globulifera Paracou* and *S. sp1* - *S. globulifera Regina* as the remaining pair didn't show admixture sign with the `admixture` software. We furthered classified individuals as (i) *pure-bred* with a hybrid index $h$ above 0.9, (ii) introgressed with $h \in [0.6-0.9]$, and (iii) admixed with $h \in [0.5-0.6]$.

We obtained relatilevely low levels of admixture (Fig. \@ref(fig:introgress)) with 222 *S. sp1* pure-bred, 108 *S. globulifera Paracou* pure-bred, and 30 *S. globulifera Regina* pure-bred. Only 5 individuals were admixed (2 *S. sp1* - *S. globulifera Regina* and 3 *S. sp1* - *S. globulifera Paracou*). Nevertheless *S. sp1* showed 13(6%) individuals introgressed with *S. globulifera Regina* and *S. globulifera Paracou* showed 7(6%) individuals introgressed with *S. sp1*.

```{r introgressCluster, eval=F, echo=F}
# module load bioinfo/plink-v1.90b5.3
# plink --bfile symcapture.all.biallelic.snp.filtered.nonmissing.paracou --allow-extra-chr \
#   --extract LD.prune.in --maf 0.05 --geno 0 --out paracou.filtered --make-bed --recode structure
paracou <- data.table::fread(file.path(path, "..", "variantCalling", "paracou", "paracou.filtered.recode.strct_in"), skip = 2) %>% as.tbl()
inds <- gsub(".g.vcf", "", paracou$V1)
fam <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                                              "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID))
symcapture.admix <- pophelper::readQ(list.files(file.path(path, "admixture", "paracou"), 
                                     full.names = T, pattern = ".Q"), indlabfromfile=F)
symcapture.admix <- lapply(symcapture.admix, "rownames<-", fam$Ind)
symcapture.admix <- pophelper::alignK(symcapture.admix)
pops <- symcapture.admix[[3]] %>% 
  rownames_to_column(var = "Ind") %>%
  mutate(type = "admixed") %>% 
  mutate(type = ifelse(Cluster1 > 0.9, "Ssp1", type)) %>% 
  mutate(type = ifelse(Cluster2 > 0.9, "SgParacou", type)) %>% 
  mutate(type = ifelse(Cluster3 > 0.9, "SgRegina", type)) %>% 
  mutate(type = ifelse(type == "admixed" & Cluster3 < 0.1, "admixedSP", type)) %>% 
  mutate(type = ifelse(type == "admixed", "admixedSR", type))
paracou <- paracou[-c(1:2)]
cl <- parallel::makeCluster(getOption("cl.cores", 4))
parallel::clusterExport(cl, list("paracou"))
admix <- parallel::parLapply(cl, seq(1:(ncol(paracou)/2)), 
                             function(i) paste(unlist(paracou[,i]), unlist(paracou[,(i+1)]), sep = "/"))
parallel::stopCluster(cl) ; rm(cl)
names(admix) <- 1:length(admix)
admix <- bind_rows(admix)
loci <- readr::read_tsv(file.path(path, "..", "variantCalling", "paracou", "paracou.filtered.bim"),
         col_names = c("contig", "snp", "posCenti", "pos", "A1", "A2")) %>%
  dplyr::mutate(snp = 1:nrow(.)) %>%
  dplyr::mutate(locus = paste0(contig, "_snp", snp, "_pos", pos)) %>%
  dplyr::mutate(type = "C") %>%
  dplyr::select(locus, type, contig, pos) %>%
  as.matrix()
count.matrix <- list(
  sp = prepare.data(admix.gen = t(admix[which(inds %in% filter(pops, type != "SgRegina", type != "admixedSR")$Ind),]), 
                    parental1 = t(admix[which(inds %in% filter(pops, type == "Ssp1")$Ind),]),
                    parental2 = t(admix[which(inds %in% filter(pops, type == "SgParacou")$Ind),]),
                    loci.data = loci, pop.id = F, ind.id = F, fixed = F, sep.rows = F),
  sr = prepare.data(admix.gen = t(admix[which(inds %in% filter(pops, type != "SgParacou", type != "admixedSP")$Ind),]), 
                    parental1 = t(admix[which(inds %in% filter(pops, type == "Ssp1")$Ind),]),
                    parental2 = t(admix[which(inds %in% filter(pops, type == "SgRegina")$Ind),]),
                    loci.data = loci, pop.id = F, ind.id = F, fixed = F, sep.rows = F),
  pr = prepare.data(admix.gen = t(admix[which(inds %in% filter(pops, type != "Ssp1", type != "admixedSP", type != "admixedSR")$Ind),]), 
                    parental1 = t(admix[which(inds %in% filter(pops, type == "SgParacou")$Ind),]),
                    parental2 = t(admix[which(inds %in% filter(pops, type == "SgRegina")$Ind),]),
                    loci.data = loci, pop.id = F, ind.id = F, fixed = F, sep.rows = F))
cl <- parallel::makeCluster(getOption("cl.cores", 3))
parallel::clusterExport(cl, list("count.matrix", "loci"))
hi.index <- parallel::parLapply(cl, count.matrix, function(m)
  introgress::est.h(introgress.data = m, loci.data = loci, fixed = F))
parallel::stopCluster(cl) ; rm(cl)
hi.index$sp$Ind <- inds[which(inds %in% filter(pops, type != "SgRegina", type != "admixedSR")$Ind)]
hi.index$sr$Ind <- inds[which(inds %in% filter(pops, type != "SgParacou", type != "admixedSP")$Ind)]
hi.index$pr$Ind <- inds[which(inds %in% filter(pops, type != "Ssp1", type != "admixedSP", type != "admixedSR")$Ind)]
hi.index <- bind_rows(hi.index, .id = "pair")
save(symcapture.admix, hi.index, file = file.path("symcapture_save", "introgress.Rdata"))
cl <- parallel::makeCluster(getOption("cl.cores", 3))
parallel::clusterExport(cl, list("count.matrix", "loci", "hi.index"))
clines <- parallel::parLapply(cl, names(count.matrix), function(pair){
  library(tidyverse)
  introgress::genomic.clines(introgress.data = count.matrix[[pair]], 
                             hi.index = hi.index %>% 
                               filter(pair == pair) %>% 
                               dplyr::select(lower, h, upper),
                             loci.data = loci,
                             sig.test = T, method = "permutation")
})
parallel::stopCluster(cl) ; rm(cl)
names(clines) <- names(count.matrix)
clines <- clines$sp$Summary.data
save(symcapture.admix, hi.index, clines,
     file = file.path("symcapture_save", "introgress.Rdata"))
```

```{r introgressT}
load(file = file.path("symcapture_save", "introgress.Rdata"))
rm(clines)
hi.index %>% 
  dplyr::select(Ind, pair, h) %>% 
  reshape2::dcast(Ind ~ pair) %>% 
  mutate(pop = NA) %>% 
  mutate(pop = ifelse(sr < 0.1 & sp < 0.1, "S. sp1 pure", pop)) %>% 
  mutate(pop = ifelse(is.na(pop) & sp > 0.9, "S. globulifera Paracou pure", pop)) %>% 
  mutate(pop = ifelse(is.na(pop) & sr > 0.9, "S. globulifera Regina pure", pop)) %>%
  mutate(pop = ifelse(is.na(pop) & !is.na(sr) & sr > 0.3, "Admixed S. sp1 - S. globulifera Regina", pop)) %>% 
  mutate(pop = ifelse(is.na(pop) & !is.na(sr) & sr < 0.3 & sr > sp, "S. sp1 introgressed with S. globulifera Regina", pop)) %>% 
  mutate(pop = ifelse(is.na(pop) & sp > 0.7, "S. globulifera Paracou introgressed with S. sp1", pop)) %>% 
  mutate(pop = ifelse(is.na(pop) & sp > 0.1 & sp < 0.7, "Admixed S. globulifera - S. sp1", pop)) %>% 
  mutate(pop = ifelse(is.na(pop), "S. sp1 pure", pop)) %>% 
  dplyr::select(-sr, -sp) %>% 
  write_tsv(file.path(path, "populations", "paracou.hybridmap"))
```

```{r introgress, fig.cap="Population structure and fraction of the genome inherited from S. sp1 for each individual (hybrid index or admixture coefficient). Population structure assessed with ADMIXTURE is represented with the color bar for each individual, with the percentage of membership to S. sp1 genepools represented by the bar height. The hybdrid index and it's confidence interval is represented by the black line and the white area. The white dashed line indicates levels used to define previous genepools and parental alleles frequencies."}
hi.index %>% 
  mutate_at(c("h", "lower", "upper"), ~ifelse(pair == "pr", 1-., .)) %>% 
  mutate(pair = recode(pair, 
                       sp = "S. sp1 - S. globulifera type Paracou", 
                       sr = "S. sp1 - S. globulifera type Regina",
                       pr = "S. globulifera type Paracou - S. globulifera type Regina")) %>% 
  left_join(symcapture.admix[[3]] %>% 
              rownames_to_column(var = "Ind") %>%
              reshape2::melt(id.vars = "Ind", variable.name = "genepool") %>%
              mutate(genepool = recode(genepool, Cluster1 = "S. sp1", 
                                       Cluster2 = "S. globulifera type Paracou", 
                                       Cluster3 = "S. globulifera type Regina"))) %>%
  arrange(pair, desc(h)) %>%
  mutate(order = 1:nrow(.)) %>% 
  ggplot(aes(reorder(Ind, order), group = NA)) +
  geom_col(aes(y = value, fill = genepool, col = genepool), position = position_stack(reverse = TRUE)) +
  geom_hline(yintercept = c(0.1, 0.9), linetype = "dashed", col = "white") +
  geom_ribbon(aes(ymin = 1 - lower, ymax = 1 - upper), alpha = 0.5, fill = "white") +
  geom_line(aes(y = 1 - h)) +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        axis.title.x = element_blank(), axis.line.x = element_blank()) +
  scale_fill_manual("Genepool", values =  c("#CD2626", "#1E90FF", "#0000EE")) +
  scale_color_manual("Genepool", values =  c("#CD2626", "#1E90FF", "#0000EE")) +
  ylab("H index") +
  facet_wrap(~ pair, nrow = 3) +
  theme(legend.position = "bottom") 
```

```{r kinshipintrogressed, fig.cap="Kinship relationships between introgressed individuals."}
ids <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king.id"))
K <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king"),
         col_names = ids$IID) %>% 
  as.data.frame() %>% 
  mutate(Ind1 = gsub(".g.vcf", "", ids$IID)) %>% 
  reshape2::melt(id.vars = "Ind1", variable.name = "Ind2", value.name = "kinship") %>% 
  mutate(Ind2 = gsub(".g.vcf", "", Ind2))
lapply(unique(trees$pop.hybrid)[4:5], function(p)
  filter(K, 
         Ind1 %in% filter(trees, pop.hybrid %in% p)$Ind,
         Ind2 %in% filter(trees, pop.hybrid %in% p)$Ind) %>% 
    ggplot(aes(Ind1, Ind2, fill = kinship)) +
    geom_tile() +
    coord_equal() +
    viridis::scale_fill_viridis()) %>% 
  cowplot::plot_grid(plotlist = ., nrow = 2,
                     labels = unique(trees$pop.hybrid)[4:5])
```

## Loci

[@Burgarella2019; @Racimo2016a; @Pfeifer2019; @Pfeifer2019b]

**Use `bgc`**

```{bash bgcSHdataPrep, eval=F, echo=T}
plink=~/Tools/plink_high_contig/plink 
$plink \
  --bfile ../paracou/symcapture.all.biallelic.snp.filtered.nonmissing.paracou \
  --allow-extra-chr \
  --set-missing-var-ids @_snp# \
  --mind 0.9 \
  --geno 0 \
  --recode A-transpose \
  --make-bed \
  --out bgc
```


```{r bgcRdataPrep, eval=F}
snps <- vroom::vroom(file.path(path, "..", "variantCalling", "bgc", "bgc.traw")) %>% 
  # sample_n(10^3) %>%
  mutate(SNP = paste("locus", 1:nrow(.))) %>% 
  reshape2::melt(id.vars = c("CHR", "SNP", "(C)M", "POS", "COUNTED", "ALT"), 
                 variable.name = "IID", value.name = "MA1") %>% 
  dplyr::rename(REF = COUNTED) %>% 
  dplyr::select(CHR, POS, SNP, ALT, REF, IID, MA1) %>% 
  mutate(IID = gsub("0_", "", IID)) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(MA2 = 2-MA1)
for(p in unique(trees$pop.hybrid)[1:3]){
  t <- filter(snps, Ind %in% filter(trees, pop.hybrid == p)$Ind) %>% 
    group_by(SNP) %>% 
    summarise(MA1 = sum(MA1), MA2 = sum(MA2))
  t2 <- data.frame(X1 = rep(NA, 2*nrow(t)), 
                   X2 = rep(NA, 2*nrow(t)))
  for(i in 1:nrow(t)){
    t2[(i*2-1),1] <- t$SNP[i]
    t2[(i*2-1),2] <- ""
    t2[(i*2),1] <- as.character(t$MA1[i])
    t2[(i*2),2] <- as.character(t$MA2[i])
  }
  write_tsv(t2, file.path(path, "..", "variantCalling", "bgc",
                          paste0(gsub("\\s", "", p), ".txt")),
            col_names = F)
}
inds <- list(sp = filter(trees, pop.hybrid %in% unique(trees$pop.hybrid)[5:6])$Ind,
             sr = filter(trees, pop.hybrid %in% unique(trees$pop.hybrid)[c(4,7)])$Ind)
for(p in names(inds)){                                
  t <- filter(snps, Ind %in% inds[[p]]) %>% 
    dplyr::select(SNP, MA1, MA2) %>% 
    arrange(SNP)
  lapply(unique(snps$SNP), function(s)
    rbind(data.frame(MA1 = c(s, "pop0"), MA2 = c("", ""),
                     stringsAsFactors = F),
          filter(t, SNP == s)[c("MA1", "MA2")])) %>% 
    bind_rows() %>% 
    write_tsv(file.path(path, "..", "variantCalling", "bgc",
                            paste0("admixed.", p, ".txt")),
              col_names = F)
}
rm(snps) ; gc()
```

> No linkage map as contigs are too shot !

```{bash bgcSHrun, eval=F, echo=T}
bgc=~/Tools/bgcdist/bgc
$bgc \
  -a S.sp1pure.txt \
  -b S.globuliferaParacoupure.txt \
  -h admixed.sp.txt \
  -F admixed.sp \
  -p 2 \
  -q 1
$bgc \
  -a S.sp1pure.txt \
  -b S.globuliferaReginapure.txt \
  -h admixed.sr.txt \
  -F admixed.sr \
  -p 2 \
  -q 1
```

```{r bgclp, fig.cap="bgc loglikelihood."}
lapply(c("sp", "sr"), function(folder)
  vroom::vroom(file.path(path, "..", "variantCalling", "bgc", folder, "LnL.txt"), 
             delim = " ", col_names = folder)) %>% 
  bind_cols() %>% 
  mcmc_trace(facet_args = list(nrow = 3)) +
  ggtitle("Loglikelihood")
```

```{r bgcparameters, fig.cap="bgc parameters.", fig.cap="bgc $\\alpha$ and $\\beta$ parameters."}
pars <- lapply(c("sp", "sr"), function(pair) {
  lapply(c("alpha", "beta"), function(par) {
    vroom::vroom(file.path(path, "..", "variantCalling", "bgc", pair, paste0(par, ".txt")),
             col_names = as.character(1:1000)) %>% 
        mutate(SNP = 1:nrow(.)) %>% 
      reshape2::melt(id.var= "SNP", variable.name = "iteration", value.name = par)
  }) %>% 
    plyr::join_all() %>% 
    reshape2::melt(c("SNP", "iteration"))
}) %>% 
  bind_rows(.id = "pair")
pars %>% 
  group_by(pair, variable, SNP) %>% 
  summarise(l = quantile(value, 0.05), m = median(value), 
            h = quantile(value, 0.95)) %>%  
  ggplot(aes(x = SNP)) +
  geom_ribbon(aes(ymin = l, ymax = h), alpha = 0.5, col = "lightgrey") +
  geom_line(aes(y = m), alpha = 1) +
  facet_grid(variable ~ pair, scales = "free")
```

```{r pfeiferCode, eval=F}
# plink=~/Tools/plink_high_contig/plink 
# $plink --bfile symcapture.all.biallelic.snp.filtered.nonmissing --allow-extra-chr --set-missing-var-ids @_snp#  --recode vcf-iid --out all --make-bed
# bgzip all.vcf 
# tabix -p vcf all.vcf.gz
library(PopGenome)
bim <- vroom::vroom(file.path(path, "..", "variantCalling", "nonmissing", "all.bim"),
             col_names = c("CHR", "SNP", "CM", "POS", "A1", "A2"))
fam <- vroom::vroom(file.path(path, "..", "variantCalling", "nonmissing", "all.fam"),
             col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  left_join(trees) %>% 
  mutate(pop = ifelse(is.na(pop), "outgroup", pop))
pops.paracou <- lapply(unique(fam$pop)[-c(1,4)], 
       function(i) filter(fam, pop == i)$IID)
genome <- lapply(unique(bim$CHR), function(chr)
  readVCF(file.path("/home/sylvain/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/variantCalling/nonmissing/all.vcf.gz"), 
          numcols = 10^5,
          tid = chr,
          include.unknown = T,
          frompos = 1, topos = 10^6))
genome <- concatenate.classes(genome)
genome@region.names <- unique(bim$CHR)
genome <- concatenate.regions(genome)
# get.sum.data(genome)
genome <- set.populations(genome, pops.paracou, diploid = T)
genome <- set.outgroup(genome, filter(fam, Ind %in% c("F023", "FO24"))$IID, 
                       diploid = T)
# genome.sl <- sliding.window.transform(genome)
genome.sl <- sliding.window.transform(genome, type = 2,
                                      width = 100, jump = 10)
genome.sl <- introgression.stats(genome.sl, l.smooth = F)
genome.sl <- weighted.jackknife(genome.sl, per.region = F)
df <- data.frame(region.names = genome.sl@region.names,
           df = as.numeric(genome.sl@df)) %>% 
  na.omit() %>% 
  mutate(Z = (df - mean(genome.sl@df, na.rm = T)) / genome.sl@df.SE) %>% 
  mutate(P = 2*(pnorm(-abs(Z)))) %>% 
  mutate(adjP = p.adjust(P, "BH")) %>% 
  rowwise() %>% 
  mutate(pos = mean(as.numeric(strsplit(as.character(region.names)," ")[[1]][c(1,3)])))
save(genome, df, file = "symcapture_save/df.Rdata")
```

```{r pfeiferFig, eval=F}
load("symcapture_save/df.Rdata")
ggplot(df, aes(pos, df)) +
  geom_hline(yintercept = 0, col = "lightgrey", type = "dashed") +
  geom_line() +
  geom_point(aes(col = (adjP)<0.01, alpha = (adjP)<0.01)) +
  xlab("Genome position (bp)") +
  ylab(expression(d[f])) +
  scale_color_discrete("P < 0.01") +
  scale_alpha_discrete("P < 0.01")
```

## Environment

Introgression of *S. sp1* within *S. globulifera Paracou*, allowed introgressed *S. globulifera Paracou* individuals to reach drier habitats (TWI < 3) of *S. sp1*, but not the introgression of *S. globulifera Regina* in *S. sp1* (Fig. \@ref(fig:introgressTWIBox)). Admixed didn't show any specific patterns, and the pattern is similar but less clear for competition (NCI). Nevertheless, scatter plots (Fig. \@ref(fig:introgressTWIScatter)) suggested a positive relation between *S. sp1* ancestry and tolerance to lower TWI, i.e. *S. sp1* individuals introgressed with either *S. globulifera Paracou* or *S. globulifera Regina* seems to grow preferentially in wetter habitat.

```{r introgressTWIBox, fig.cap="TWI and NCI distributions between pure-bred and admixed individuals per population."}
trees %>% 
  mutate(pop.hybrid = fct_reorder(pop.hybrid, dplyr::desc(TWI))) %>%
  dplyr::select(Ind, pop, pop.hybrid, TWI, NCI) %>% 
  reshape2::melt(id.vars = c("Ind", "pop", "pop.hybrid")) %>% 
  ggplot(aes(pop.hybrid, value, fill = pop, col = pop)) +
  geom_hline(aes(yintercept = y), linetype = "dashed", color = "grey",
             data = data.frame(y = c(3,NA), variable = c("TWI", "NCI"))) +
  geom_boxplot(alpha = 0.7) +
  coord_flip() +
  scale_fill_manual("", values =  c("lightgrey", "#0000EE", "#1E90FF", "#CD2626"),
                    guide = guide_legend(nrow = 2)) +
  scale_color_manual("", values =  c("lightgrey", "#0000EE", "#1E90FF", "#CD2626")) +
  theme(axis.title.y = element_blank(), legend.position = "bottom") +
  scale_y_sqrt() + ylab("") + 
  facet_wrap(~ variable, scales = "free_x")
```

```{r introgressTWIScatter, fig.cap="TWI and NCI vs hybrid index (S. sp1 ancestry) between population-pairs."}
hi.index %>% 
  filter(pair != "pr") %>% 
  mutate(pair = recode_factor(pair,
                              "sp" = "S. sp1 - S. globulifera Paracou",
                              "sr" = "S. sp1 - S. globulifera Regina")) %>% 
  left_join(trees) %>% 
  filter(pop != "Admixed") %>% 
  mutate_at(c("h", "lower", "upper"), ~ifelse(pop == "S. sp1", 1-., .)) %>% 
  dplyr::select(Ind, pop, pop.hybrid, TWI, NCI, pair, lower, h, upper, Gmax) %>% 
  reshape2::melt(id.vars = c("Ind", "pop", "pop.hybrid", "Gmax",
                             "pair", "lower", "h", "upper")) %>% 
  group_by(variable, pop) %>% 
  mutate(diffmeanpop = mean(value) - value) %>% 
  ggplot(aes(x = diffmeanpop, col = Gmax)) +
  geom_point(aes(y = 1 - h)) +
  geom_errorbar(aes(ymin = 1 - lower, ymax = 1 - upper), alpha = 0.5) +
  facet_grid(pop ~ variable, scales = "free") +
  ylab("Hybrid index") +
  xlab("Difference with population mean") +
  viridis::scale_color_viridis()
```

## Growth

Admixed individuals between *S. globulifera Paracou* and *S. sp1* seems to have an increased growth potential, but with a sample size of 3 it's hard to say (Fig. \@ref(fig:introgressGmaxBox)). introgressed didn't show any specific patterns. Scatter plots (Fig. \@ref(fig:introgressGmaxScatter)) didn't suggested anything specific.

```{r introgressGmaxBox, fig.cap="Individual growth potential (Gmax) distribution between pure-bred and admixed individuals per population."}
trees %>% 
  mutate(pop.hybrid = fct_reorder(pop.hybrid, dplyr::desc(Gmax))) %>%
  ggplot(aes(pop.hybrid, log(Gmax), fill = pop, col = pop)) +
  geom_boxplot(alpha = 0.7) +
  coord_flip() +
  scale_fill_manual("Genepool", values =  c("lightgrey", "#0000EE", "#1E90FF", "#CD2626"),
                    guide = guide_legend(nrow = 2)) +
  scale_color_manual("Genepool", values =  c("lightgrey", "#0000EE", "#1E90FF", "#CD2626")) +
  theme(axis.title.y = element_blank(), legend.position = "bottom") +
  ylab("Logarithm of individual growth potential (Gmax)") +
  stat_summary(fun.data = function(y) 
    return(data.frame(y = 0.95 * 5, label = length(y))), 
    geom = "text", hjust = 0.5, vjust = 0.9) +
  ggtitle(paste("ANOVA p-val = ",
                round(summary(aov(log(Gmax) ~ pop.hybrid, data = trees))[[1]]$`Pr(>F)`[1], 5)))
```

```{r introgressGmaxScatter, fig.cap="Individual growth potential (Gmax) and hybrid index (S. sp1 ancestry) between population-pairs."}
hi.index %>% 
    filter(pair != "pr") %>% 
  mutate(pair = recode_factor(pair,
                              "sp" = "S. sp1 - S. globulifera Paracou",
                              "sr" = "S. sp1 - S. globulifera Regina")) %>% 
  left_join(trees) %>% 
  filter(pop != "Admixed") %>% 
  mutate_at(c("h", "lower", "upper"), ~ifelse(pop == "S. sp1", 1-., .)) %>% 
  mutate(diffmeanpop = mean(log(Gmax)) - log(Gmax)) %>%
  ggplot(aes(x = diffmeanpop, col = DBHtoday)) +
  geom_point(aes(y = 1 - h)) +
  geom_errorbar(aes(ymin = 1 - lower, ymax = 1 - upper), alpha = 0.5) +
  facet_wrap(~ pop, scales = "free") +
  xlab("Difference with mean population logarithm of individual growth potential") +
  ylab("Hybrid index") +
  viridis::scale_color_viridis() +
  theme(legend.position = "bottom")
```
