```{r setup_methcomp, include=FALSE, eval = T}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(bayesplot)
library(rstan)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F, eval = T)
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = T)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
```

# Methods comparisons

```{r DataMetComp}
trees <- src_sqlite(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                              "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(Genus == "Symphonia") %>% 
  mutate(DBH = CircCorr/pi) %>% 
  filter(!(CodeMeas %in% c(4))) %>% 
  collect()
trees <- read_tsv(file.path(path, "..", "variantCalling", "paracou",
                          "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
         col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype")) %>% 
  mutate(Ind = gsub(".g.vcf", "", IID)) %>% 
  mutate(X = gsub("P", "", Ind)) %>% 
  separate(X, c("Plot", "SubPlot", "TreeFieldNum"), convert = T) %>% 
  left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "bayescenv", "paracou3pop.popmap"),
                     col_names = c("IID", "pop"))) %>% 
  left_join(read_tsv(file.path(path, "populations", "paracou.hybridmap")),
            by = "Ind", suffix = c("", ".hybrid"))
trees0 <- trees
trees <- trees %>% 
  group_by(Ind) %>% 
  mutate(Y0 = dplyr::first(CensusYear), DBH0 = dplyr::first(DBH), 
         DBHtoday = dplyr::last(DBH), N = n()) %>%  
  ungroup() %>% 
  dplyr::select(Ind, IID, pop, Plot, SubPlot, TreeFieldNum, Y0, DBH0, DBHtoday, N) %>% 
  unique() %>% 
  mutate(DBHtoday = ifelse(DBHtoday == DBH0, DBHtoday + 0.1, DBHtoday))
trees <- trees %>% 
  na.omit(pop) %>% # remove admixed
  mutate(IndNum = as.numeric(as.factor(Ind)), popNum = as.numeric(as.factor(pop)))
ids <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king.id"))
K <- read_tsv(file.path(path, "..", "variantCalling", "growth", "plink2.king"),
         col_names = ids$IID) %>% 
  as.data.frame()
row.names(K) <- ids$IID
K <- as.matrix(K)
K <- K[trees$IID, trees$IID]
K[K < 0] <- 0
K <- K*2
K <- as.matrix(Matrix::nearPD(K)$mat)
```

## Gmax

We used the model $Gmax$ to estimate individual growth potential per population:

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_p})}{\theta3_p}]^2)), \sigma^2) \\ \theta1_i \sim \mathcal{logN}(log(\theta1_p), \sigma^2_{R1}) \\ \begin{pmatrix} \theta2_p \\ \theta3_p \end{pmatrix} \sim \mathcal{logN}(\begin{pmatrix} \theta2 \\ \theta3 \end{pmatrix}, \begin{pmatrix} \sigma^2_{P2} \\ \sigma^2_{P3} \end{pmatrix})$$

```{r gmax1ParTabMetComp}
fitGmax1 <- list()
for(sim in list.files("symcapture_save/gmax1Par", full.names = T)){
  load(sim)
  fitGmax1 <- c(fitGmax1, fit)
}
fitGmax1 <- sflist2stanfit(fitGmax1)
Gmax <- as.data.frame(fitGmax1, "thetai1") %>%
  summarise_all(median) %>%
  reshape2::melt(NULL, value.name = "Gmax") %>%
  separate(variable, c("theta", "IndNum"), convert = T) %>%
  dplyr::select(-theta) %>%
  left_join(trees)
ggplot(Gmax, aes(Gmax)) +
  geom_histogram() +
  facet_wrap(~ pop, scales = "free")
```

## Animal Normal

We first estimated genetic variance with a normal animal model with log-transformed Gmax, as follow:

$$log(Gmax_i) \sim \mathcal{N}(a,\sigma_1) \\ a \sim \mathcal{MVN_N}(log(\mu_p),\sigma_2.K)$$

We fitted the equivalent model with following priors:

$$log(Gmax_{p,i}) \sim \mathcal{N}(log(\mu_p) + \sigma_2.A.\epsilon_i, \sigma_1) \\ \epsilon \sim \mathcal{N}(0,1) \\ \mu_p \sim \mathcal{logN}(0,1) \\ (\sigma_1,\sigma_2) \sim \mathcal{N}(0,1) \\ V_P = Var(\mu_p) \\V_G=\sigma_2^2\\V_R=\sigma_1^2$$

```{bash AnNormCluster, eval=F}
# mdata = list(N = nrow(Gmax),
#              P = length(unique(Gmax$pop)),
#              y = log(Gmax$Gmax),
#              population = Gmax$popNum,
#              K = K)
# save(mdata, file = file.path("symcapture_save", 'dataCompAnimal.Rdata'))
for chain in $(seq 4) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript CompAnimal.R $chain" ; done > CompAnimal.sh
sarray -J An -o out/%j.An.out -e out/%j.An.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL CompAnimal.sh
watch 'tail -n 1 out/*.out | grep Iteration'
```

```{r AnNormTab, fig.cap="Genetic variance of individual growth potential with a normal animal model."}
fitAn <- list()
for(sim in list.files("symcapture_save/CompAnimal", full.names = T)){
  load(sim)
  fitAn <- c(fitAn, fit)
}
fitAn <- sflist2stanfit(fitAn)
broom::tidyMCMC(fitAn, c("mu", "Vp", "Vg", "Vr", "R2m", "R2c", "lp__"), 
       droppars = NULL, rhat = T) %>% 
  separate(term, c("parameter", "population"), "[:punct:]", convert = T) %>% 
  mutate(population = recode(population, "1" = "S. globulifera Paracou", 
                             "2" = "S. globulifera Regina", "3" = "S.sp1")) %>% 
  mutate(population = ifelse(is.na(population), "", population)) %>% 
  kable(caption = "Summary table of the kinship growth model",
        col.names = c("Parameter",  "Population", "Estimate", 
                      "$\\sigma$", "$\\hat{R}$"))
```

```{r AnNormTrace, fig.cap="Traceplot."}
mcmc_trace(fitAn, regex_pars = c("mu", "sigma", "lp__"), 
           np = nuts_params(fitAn))
```

```{r AnNormR2, fig.cap="Marginal and conditional $R^2$."}
mcmc_intervals_data(fitAn, regex_pars = c("R2")) %>% 
  mutate(index = recode(parameter, 
                        "R2m" = "Marginal", "R2c" = "Conditional")) %>%
  ggplot(aes(x = index, xend = index)) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  xlab("") + ylab(expression(R^2)) +
  scale_fill_discrete(guide = "none") +
  scale_color_discrete(guide = "none")
```

```{r AnNormVar, fig.cap="Genetic variance partitionning."}
mcmc_intervals_data(fitAn, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", 
                           "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2)) 
```

## Animal Lognormal

Second, we estimated genetic variance with a lognormal animal model as follow:

$$Gmax_i \sim \mathcal{logN}(log(a),\sigma_1) \\ a \sim \mathcal{MVlogN_N}(log(\mu_p),\sigma_2.K)$$

We fitted the equivalent model with following priors:

$$Gmax_{p,i} \sim \mathcal{logN}(log(\mu_p) + \sigma_2.A.\epsilon_i, \sigma_1) \\ \epsilon \sim \mathcal{N}(0,1) \\ \mu_p \sim \mathcal{logN}(0,1) \\  (\sigma_1,\sigma_2) \sim \mathcal N(0,1) \\V_P = Var(\mu_p) \\V_G=\sigma_2^2\\V_R=\sigma_1^2$$

```{bash AnLogormCluster, eval=F}
# mdata = list(N = nrow(Gmax),
#              P = length(unique(Gmax$pop)),
#              y = Gmax$Gmax,
#              population = Gmax$popNum,
#              K = K)
# save(mdata, file = file.path("symcapture_save", 'dataCompAnimalLog.Rdata'))
for chain in $(seq 4) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript CompAnimalLog.R $chain" ; done > CompAnimalLog.sh
sarray -J AnLog -o out/%j.AnLog.out -e out/%j.AnLog.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL CompAnimalLog.sh
```

```{r AnLognormTab, fig.cap="Genetic variance of individual growth potential with a lognormal animal model."}
fitAnLog <- list()
for(sim in list.files("symcapture_save/CompAnimalLog", full.names = T)){
  load(sim)
  fitAnLog <- c(fitAnLog, fit)
}
fitAnLog <- sflist2stanfit(fitAnLog)
broom::tidyMCMC(fit, c("mu", "Vp", "Vg", "Vr", "R2m", "R2c", "lp__"), 
       droppars = NULL, rhat = T) %>% 
  separate(term, c("parameter", "population"), "[:punct:]", convert = T) %>% 
  mutate(population = recode(population, "1" = "S. globulifera Paracou", 
                             "2" = "S. globulifera Regina", "3" = "S.sp1")) %>% 
  mutate(population = ifelse(is.na(population), "", population)) %>% 
  kable(caption = "Summary table of the kinship growth model",
        col.names = c("Parameter",  "Population", "Estimate", 
                      "$\\sigma$", "$\\hat{R}$"))
```

```{r AnLognormTrace, fig.cap="Traceplot."}
mcmc_trace(fitAnLog, regex_pars = c("mu", "sigma", "lp__"), 
           np = nuts_params(fitAnLog))
```

```{r AnLognormR2, fig.cap="Marginal and conditional $R^2$."}
mcmc_intervals_data(fit, regex_pars = c("R2")) %>% 
  mutate(index = recode(parameter, 
                        "R2m" = "Marginal", "R2c" = "Conditional")) %>%
  ggplot(aes(x = index, xend = index)) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  xlab("") + ylab(expression(R^2)) +
  scale_fill_discrete(guide = "none") +
  scale_color_discrete(guide = "none")
```

```{r AnLognormVar, fig.cap="Genetic variance partitionning."}
mcmc_intervals_data(fit, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", 
                           "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2)) 
```

## Hierarchical Lognormal

Finally, we estimated individual growth potential and associated genetic variance simultanously in a hierarchical model with a lognormal distribution as follow:

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_p})}{\theta3_p}]^2)), \sigma_1) \\  \theta1_i \sim \mathcal{logN}(log(a1_i), \sigma_2) \\ \begin{pmatrix} \theta2_p \\ \theta3_p \end{pmatrix} \sim \mathcal{logN}(\begin{pmatrix} \theta2 \\ \theta3 \end{pmatrix}, \begin{pmatrix} \sigma_3 \\ \sigma_4 \end{pmatrix}) \\ a1_i \sim \mathcal{MVlogN}(log(\theta1_p), \sigma_5.K)$$

We fitted the equivalent model with following priors:

$$DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \mathcal{logN} ( log(\sum _{y=y_0} ^{y=today} \theta1_i . exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{\theta2_p})}{\theta3_p}]^2)), \sigma_1) \\  \theta1_i = e^{log(a1_i) + \sigma_2.\epsilon1_i} \\ \theta2_p  = e^{log(\theta2) + \sigma_3.\epsilon2_i} \\ \theta3_p  = e^{log(\theta3) + \sigma_4.\epsilon3_i} \\ \\ a1_i = e^{log(\theta1_p) + \sigma_5.A.\epsilon4_i} \\ (\epsilon1, \epsilon2, \epsilon3, \epsilon4) \sim \mathcal{N}(0,1) \\ (\theta1_p, \theta2, \theta3) \sim \mathcal{logN}(0,1) \\ (\sigma_1, \sigma_2, \sigma_3, \sigma_4, \sigma_5) \sim \mathcal{N}(0,1) \\ V_P = Var(\mu_p) \\V_G=\sigma_5^2\\V_R=\sigma_2^2$$

```{bash fitGmaxGenoParCluster, eval=F}
for chain in $(seq 8) ; do echo "module purge ; module load compiler/gcc-7.2.0 ; module load system/R-3.5.3 ; R_LIBS_USER=\" \" Rscript GmaxGenoPar.R $chain" ; done > GmaxGenoPar.sh
sarray -J GmaxGenoPar -o out/%j.GmaxGenoPar.out -e out/%j.GmaxGenoPar.err -t 48:00:00 --constraint=broadwell --cpus-per-task=1 --mail-type=BEGIN,END,FAIL GmaxGenoPar.sh
```

```{r HierarchicalTab}
fitGmaxGeno <- list()
for(sim in list.files("symcapture_save/gmaxgenoPar", full.names = T)){
  load(sim)
  fitGmaxGeno <- c(fitGmaxGeno, fit)
}
fitGmaxGeno <- sflist2stanfit(fitGmaxGeno)
broom::tidyMCMC(fitGmaxGeno, c("thetap1", "theta2", "theta3", "sigma", "lp__"), 
                droppars = NULL, ess = T, rhat = T, estimate.method = "median") %>% 
  dplyr::select(term, estimate, std.error, rhat, ess) %>% 
  kable(caption = "Individual growth potential model.",
        col.names = c("Parameter", "Estimate", "Standard error", "$\\hat R$", "$N_{eff}$"))
```

```{r HierarchicalTrace, fig.cap="Traceplot."}
mcmc_trace(as.array(fitGmaxGeno, 
                    pars = c("thetap1", "theta2", "theta3", "sigma", "lp__")),
           np = nuts_params(fitGmaxGeno))
```

```{r  HierarchicalR2, fig.cap="Marginal and conditional $R^2$."}
mcmc_intervals_data(fitGmaxGeno, regex_pars = c("R2")) %>% 
  mutate(index = recode(parameter, 
                        "R2m" = "Marginal", "R2c" = "Conditional")) %>%
  ggplot(aes(x = index, xend = index)) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  xlab("") + ylab(expression(R^2)) +
  scale_fill_discrete(guide = "none") +
  scale_color_discrete(guide = "none")
```

```{r  HierarchicalVar, fig.cap="Genetic variance partitionning."}
mcmc_intervals_data(fitGmaxGeno, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", 
                           "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  scale_fill_discrete(expression(sigma^2)) 
```
