---
title: Pervasive local-scale species niche differentiation with topography within widespread Neotropical tree species complexes
date: '`r Sys.Date()`'
output:
  bookdown::word_document2:
    reference_docx: ./template/template.docx
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
  bookdown::html_document2:
    number_sections: false  
    toc: true
    toc_float: yes
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/joe.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(broom)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, 
               fig.height = 8, fig.width = 8,
               cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Paracou/"
```

```{r complexes}
Complexes <- bind_rows(
  data.frame(Complex = "Chartacea", Genus = "Eschweilera",
             Species = c("simiorum", "congestiflora")),
  data.frame(Complex = "Parvifolia", Genus = "Eschweilera",
             Species = c("pedicellata", "coriacea", "decolorans", "sagotiana",
                         "wachenheimii", "grandiflora_form2")),
  data.frame(Complex = "Licania", Genus = "Licania",
             Species = c("alba", "membranacea", "canescens", "micrantha",
                         "ovalifolia", "sprucei", "densiflora",
                         "laxiflora", "parvifructa")),
  data.frame(Complex = "Iryanthera", Genus = "Iryanthera",
             Species = c("hostmannii", "sagotiana")),
  data.frame(Complex = "Talisia", Genus = "Talisia",
             Species = c("hexaphylla", "praealta", "simaboides")),
  data.frame(Complex = "Symphonia", Genus = "Symphonia",
             Species = c("globulifera", "sp.1")))
```

```{r data}
d <- 20
trees <- src_sqlite(file.path(path, "trees", "Paracou.sqlite")) %>% 
  tbl("Paracou") %>% 
  filter(CensusYear == 2015) %>%
  filter(Xfield > d, Xfield < 250-d, Yfield > d, Yfield < 250-d) %>% 
  filter(Species != "Indet.") %>% 
  mutate(DBH = CircCorr/pi) %>% 
  collect()
load("../distribution_save/env.Rdata")
```

```{r predComplex}
load("../distribution_save/Competition.Rdata")
data <- trees %>% 
  left_join(env) %>% 
  left_join(Complexes) %>% 
  filter(!is.na(Plot))
complexes <- unique(Complexes$Complex)
data <- mutate(data, Complex = ifelse(is.na(Complex), "no", Complex))
mdata <- lapply(complexes, function(complex) {
  data_complex <- left_join(data,
                            Competition %>% 
                              left_join(Complexes, by = c("Genusj" = "Genus", "Speciesj" = "Species")) %>% 
                              mutate(Complex = ifelse(is.na(Complex), "no", Complex)) %>% 
                              mutate(dij = ifelse(dij <1, 1, dij)) %>% 
                              mutate(strata = as.numeric(Complex == complex)) %>% 
                              group_by(idTree) %>% 
                              summarise(NCIother = log(sum(DBHj^2*exp(-0.25*dij)*abs(strata-1))+1),
                                        NCIcomplex = log(sum(DBHj^2*exp(-0.25*dij)*strata)+1)))
  list(N = nrow(data_complex),
       K = 3,
       Y = as.numeric(data_complex$Complex == complex),
       X = dplyr::select(data_complex, TWI, NCIother, NCIcomplex) %>%
         mutate_all(funs(scale)) %>%
         as.matrix(),
       w = ifelse(data_complex$Complex == complex,
                  1-sum(data_complex$Complex == complex)/length(data_complex$Complex),
                  sum(data_complex$Complex == complex)/length(data_complex$Complex)))
})
names(mdata) <- complexes
load("../distribution_save/complexes0.Rdata")
names(fits)[1:2] <- c("Chartacea", "Parvifolia")
fitsComplex <- fits
predComplex <- lapply(names(fits), function(complex){

  TWI <- seq(from = min(mdata[[complex]]$X[,1]), 
             to = max(mdata[[complex]]$X[,1]), length.out = 100)
  NCIother <- seq(from = min(mdata[[complex]]$X[,2]), 
             to = max(mdata[[complex]]$X[,2]), length.out = 100)
  NCIcomplex <- seq(from = min(mdata[[complex]]$X[,3]), 
             to = max(mdata[[complex]]$X[,3]), length.out = 100)

  alpha <- as.matrix(fits[[complex]], pars = "alpha")
  beta <- as.matrix(fits[[complex]], pars = "beta")
  gamma <- as.matrix(fits[[complex]], pars = "gamma")

  Ytwi <- do.call("rbind", sapply(1: nrow(alpha), function(n) (arm::invlogit(alpha[n] + beta[n,1]*TWI + beta[n,2]*mean(NCIother) + beta[n,3]*mean(NCIcomplex) + gamma[n,1]*TWI^2 + gamma[n,2]*mean(NCIother)^2 + gamma[n,3]*mean(NCIcomplex)^2)), simplify = F))
  Ynciother <- do.call("rbind", sapply(1: nrow(alpha), function(n) (arm::invlogit(alpha[n] + beta[n,1]*mean(TWI) + beta[n,2]*NCIother + beta[n,3]*mean(NCIcomplex) + gamma[n,1]*mean(TWI)^2 + gamma[n,2]*NCIother^2 + gamma[n,3]*mean(NCIcomplex)^2)), simplify = F))
  Yncicomplex <- do.call("rbind", sapply(1: nrow(alpha), function(n) (arm::invlogit(alpha[n] + beta[n,1]*mean(TWI) + beta[n,2]*mean(NCIother) + beta[n,3]*NCIcomplex + gamma[n,1]*mean(TWI)^2 + gamma[n,2]*mean(NCIother)^2 + gamma[n,3]*NCIcomplex^2)), simplify = F))

  bind_rows(
    data.frame(
      Complex = complex,
      variable = "TWI",
      value = TWI,
      Y = apply(Ytwi, 2, mean),
      Y5 = apply(Ytwi, 2, quantile, probs = 0.05),
      Y95 = apply(Ytwi, 2, quantile, probs = 0.95)),
    data.frame(
      Complex = complex,
      variable = "NCIother",
      value = NCIother,
      Y = apply(Ynciother, 2, mean),
      Y5 = apply(Ynciother, 2, quantile, probs = 0.05),
      Y95 = apply(Ynciother, 2, quantile, probs = 0.95)),
    data.frame(
      Complex = complex,
      variable = "NCIcomplex",
      value = NCIcomplex,
      Y = apply(Yncicomplex, 2, mean),
      Y5 = apply(Yncicomplex, 2, quantile, probs = 0.05),
      Y95 = apply(Yncicomplex, 2, quantile, probs = 0.95)))
}) %>% bind_rows()
```

```{r predSpecies}
load("../distribution_save/Competition.Rdata")
Competition <- Competition %>% 
  mutate(dij = ifelse(dij <1, 1, dij)) %>% 
  left_join(dplyr::select(trees, idTree, Genus, Species)) %>% 
  mutate(strata = as.numeric(Genusj == Genus, Speciesj == Species)) %>% 
  group_by(idTree) %>% 
  summarise(NCIhetero = log(sum(DBHj^2*exp(-0.25*dij)*abs(strata-1))+1),
            NCIcon = log(sum(DBHj^2*exp(-0.25*dij)*strata)+1))
data <- trees %>% 
  left_join(env) %>% 
  left_join(Complexes) %>% 
  left_join(Competition) %>% 
  filter(!is.na(Plot)) %>% 
  filter(!is.na(Complex)) %>% 
  group_by(Complex, Species) %>% 
  filter(n() > 10) %>%
  ungroup()
complexes <- unique(Complexes$Complex)
mdata <- lapply(complexes, function(complex) {
  data<- filter(data, Complex == complex)
  list(N = nrow(data),
       S = length(unique(data$Species)),
       K = 3,
       Y = sapply(levels(as.factor(data$Species)),
                  function(sp) as.numeric(data$Species == sp)),
       X = dplyr::select(data, TWI, NCIhetero, NCIcon) %>%
         mutate_all(funs(scale)) %>%
         as.matrix())
})
names(mdata) <- complexes
load("../distribution_save/species.Rdata")
predSpecies <- lapply(complexes, function(complex){

  N <- 25
  TWI <- seq(from = min(mdata[[complex]]$X[,1]), 
             to = max(mdata[[complex]]$X[,1]), length.out = N)
  BA <- seq(from = min(mdata[[complex]]$X[,2]), 
             to = max(mdata[[complex]]$X[,2]), length.out = N)
  BAgenus <- seq(from = min(mdata[[complex]]$X[,3]), 
             to = max(mdata[[complex]]$X[,3]), length.out = N)
  
  S <- mdata[[complex]]$S
  K <- mdata[[complex]]$K
  alpha <- as.matrix(fits[[complex]], pars = "alpha")
  beta0 <- as.matrix(fits[[complex]], pars = "beta")
  gamma0 <- as.matrix(fits[[complex]], pars = "gamma")
  beta <- array(dim = c(nrow(alpha), S, K))
  gamma <- array(dim = c(nrow(alpha), S, K))
  for(k in 1:K){
    beta[,,k] <- beta0[,seq.int(1, 3*S, S)[k]:(seq.int(1, 3*S, S)+(S-1))[k]]
    gamma[,,k] <- gamma0[,seq.int(1, 3*S, S)[k]:(seq.int(1, 3*S, S)+(S-1))[k]]
  }
    
  Yp <- abind::abind(sapply(1:length(TWI), function(i) 
    sapply(1: nrow(alpha), function(n)
      tsensembler:::softmax(alpha[n,] + beta[n,,1] * TWI[i] + beta[n,,2]*BA[i] + beta[n,,3]*BAgenus[i] + gamma[n,,1]*TWI[i]^2 + gamma[n,,2]*BA[i]^2 + gamma[n,,3]*BAgenus[i]^2)
    ), simplify = F), along = 3)
  Ytwi <- abind::abind(sapply(1:length(TWI), function(i) 
    sapply(1: nrow(alpha), function(n)
      tsensembler:::softmax(alpha[n,] + beta[n,,1]*TWI[i] + beta[n,,2]*mean(BA) + beta[n,,3]*mean(BAgenus) + gamma[n,,1]*TWI[i]^2 + gamma[n,,2]*mean(BA)^2 + gamma[n,,3]*mean(BAgenus)^2)
    ), simplify = F), along = 3)
  Yba <- abind::abind(sapply(1:length(TWI), function(i) 
    sapply(1: nrow(alpha), function(n)
      tsensembler:::softmax(alpha[n,] + beta[n,,1]*mean(TWI) + beta[n,,2]*BA[i] + beta[n,,3]*mean(BAgenus) + gamma[n,,1]*mean(TWI)^2 + gamma[n,,2]*BA[i]^2 + gamma[n,,3]*mean(BAgenus)^2)
    ), simplify = F), along = 3)
  Ybagenus <- abind::abind(sapply(1:length(TWI), function(i) 
    sapply(1: nrow(alpha), function(n)
      tsensembler:::softmax(alpha[n,] + beta[n,,1]*mean(TWI) + beta[n,,2]*mean(BA) + beta[n,,3]*BAgenus[i] + gamma[n,,1]*mean(TWI)^2 + gamma[n,,2]*mean(BA)^2 + gamma[n,,3]*BAgenus[i]^2)
    ), simplify = F), along = 3)

  bind_rows(
    lapply(1:S, function(sp)
      data.frame(
        complex = complex,
        species = colnames(mdata[[complex]]$Y)[sp],
        variable = "TWI",
        value = TWI,
        Yp = apply(Yp[sp,,], 2, mean),
        Y = apply(Ytwi[sp,,], 2, mean),
        Y5 = apply(Ytwi[sp,,], 2, quantile, probs = 0.05),
        Y95 = apply(Ytwi[sp,,], 2, quantile, probs = 0.95))) %>% 
      bind_rows(),
    lapply(1:S, function(sp)
      data.frame(
        complex = complex,
        species = colnames(mdata[[complex]]$Y)[sp],
        variable = "NCIhetero",
        value = BA,
        Yp = apply(Yp[sp,,], 2, mean),
        Y = apply(Yba[sp,,], 2, mean),
        Y5 = apply(Yba[sp,,], 2, quantile, probs = 0.05),
        Y95 = apply(Yba[sp,,], 2, quantile, probs = 0.95))) %>% 
      bind_rows(),
    lapply(1:S, function(sp)
      data.frame(
        complex = complex,
        species = colnames(mdata[[complex]]$Y)[sp],
        variable = "NCIcon",
        value = BAgenus,
        Yp = apply(Yp[sp,,], 2, mean),
        Y = apply(Ybagenus[sp,,], 2, mean),
        Y5 = apply(Ybagenus[sp,,], 2, quantile, probs = 0.05),
        Y95 = apply(Ybagenus[sp,,], 2, quantile, probs = 0.95))) %>% 
      bind_rows())
}) %>% bind_rows()
```

1. Sylvain Schmitt
    - Corresponding author
    - sylvain.m.schmitt@gmail.com
    - orcid.org/0000-0001-7759-7106
    - Univ. Bordeaux, INRAE, BIOGECO, 69 route d’Arcachon, 33610 Cestas France
1. Niklas Tysklind
    - Niklas.Tysklind@ecofog.gf
    - INRA, UMR EcoFoG (Agroparistech, CNRS, Cirad, Université des Antilles, Université de la Guyane), Campus Agronomique, 97310 Kourou, French Guiana
1. Géraldine Derroire
    - geraldine.derroire@cirad.fr
    - orcid.org/0000-0001-7239-2881
    - Cirad, UMR EcoFoG (Agroparistech, CNRS, INRA, Université des Antilles, Université de la Guyane), Campus Agronomique, 97310 Kourou, French Guiana
1. Myriam Heuertz
    - myriam.heuertz@inrae.fr
    - orcid.org/0000-0002-6322-3645
    - Univ. Bordeaux, INRAE, BIOGECO, 69 route d’Arcachon, 33610 Cestas France
1. Bruno Hérault
    - bruno.herault@cirad.fr
    - orcid.org/0000-0002-6950-7286
    - Cirad, UR Forests and Societies, Univ Montpellier, Montpellier, France
    - Institut National Polytechnique Félix Houphouët-Boigny (INP-HB), BP 1093, Yamoussoukro, Ivory Coast

\newpage 

# Abstract

Lowland Amazonia includes around five thousands described tree species belonging to more than height hundred genera, resulting in many species-rich genera. Numerous species-rich tree genera share a large amount of genetic variation due to a recent common ancestor and/or hybridization, forming species complexes. Despite species complexes' key role to understand Neotropical diversification, and the importance of ecological niches exploitation in species complexes establishment, little is known about local distribution of Neotropical species complexes in sympatry.

By taking advantage of a long-term study site in a hyperdiverse tropical forest of the Guiana Shield, we explored sister species local distribution between and within species complexes along a diversity of micro-habitats varying both in abiotic environment and biotic interactions. Combining tree inventories and LiDAR-derived topographic data, we used a Bayesian modelling framework to test the role of topography and tree neighborhood at different taxonomic scales on the distribution of species complexes and the relative distribution of sister species within complexes.

Species complexes of Neotropical trees were widely spread across habitats at local-scale. But sister species showed pervasive niche differentiation along topographic position, water accumulation, and competition gradients within species complexes. If supported by reproductive isolation of species and adaptive introgression at the complex level, our results suggest that both habitat specialisation of sister species and widespread distribution of species complexes might explain the success of these Neotropical species complex at the regional scale [@TerSteege2013].

## Keywords

species distribution model; syngameon; habitat specialisation; ...

\newpage 

# Introduction 

Tropical forests and especially the Neotropics shelter the highest level of biodiversity worldwide [@Gaston2000], as already highlighted by @connell_diversity_1978 more than 40 years ago. Lowland Amazonia include around five thousand described tree species belonging to 810 genera [@TerSteege2013], resulting in many species-rich genera. Among Neotropical species-rich genera, many have been shown to include a higher level of genetic polymorphism than species-poor genera, which is likely related to introgression between sister species [@Caron2019]. Indeed, interspecific hybrids occurs in 16% of genera, and more often in species-rich taxa [@Whitney2010]. Species sharing a large amount of genetic variation due to a recent common ancestor and/or hybridization are defined as species complexes. In addition, species forming a group of otherwise distinct species connected by occasional interspecific gene flow are called syngameon [@Suarez-Gonzalez2018]. Despite species complexes' key role to understand Neotropical diversification [@Pinheiro2018], little is known about Neotropical species complexes [@baraloto_using_2012-1; @Levi2019a; @TerSteege2013].

Simulations suggest that low hybridization success promotes coexistence of species in the community [@Cannon2015b]. Syngamea might have a selective advantage, decreasing the overall risk of genus extinction while increasing local adaptation of their distinct species [@Cannon2015b]. Consequently, syngamea are not necessarily a transitional or incipient phase of a process towards complete speciation [@Cannon2019a]. Both syngamea and species complexes can result from gene flow due to hybridization between related species. Hybrids are often sterile or less fertile than their parents, although some survive and reproduce allowing the transfer of adaptive variance across species boundaries [@Runemark2019]. Similarly, species complexes can result from adaptive radiation due to combination of genetic features between species in hybrid swarms [@Seehausen2004]. The establishment of a hybrid or derived species requires some degree of reproductive isolation from its progenitors. The exploitation of novel ecological niches helps gaining extrinsic isolation barriers [@Runemark2019]. 

Neotropical congeneric tree species have been shown to present pervasive species-habitat associations with topography [@Allie2015], soil type [@Fine2004; @Esteves2013], fertility [@Fine2004], and light access [@Esteves2013]. Similarly, in Europe, white oaks represent a well known species complex with ecological preferences of constituent species, including drought and cold tolerance and adaptation to alkaline soils [@Cannon2019a]. Nevertheless, little is known about the ecological distribution of Neotropical species complexes and the constituent species' ecological niches.

In the present study, we explored sister species distribution between and within species complexes along abiotic environment and biotic interactions. We took advantage of a long-term study site with more than 75000 censused and geolocalized individuals in a highly diverse tropical forest site located within the Guiana Shield in the Amazon Basin. The site encompasses a diversity of micro-habitats through topographic variation ranging from seasonally flooded bottomlands to drier plateaus [@Allie2015] and displays variation in tree crowding including canopy gaps and closed-canopy forest patches [@ferry2010higher]. Combining tree inventories and LiDAR-derived topographic data, we used Bayesian modelling to address the following questions: (1) how do biotic and abiotic environment influence the distribution of species complexes at local scale ? (2) how do biotic and abiotic environment influence the relative distribution of sister species within species complexes ? Based on the local abundance of species-rich genera [@Allie2016], we expected species complexes to be widespread at local scale spanning all topographic positions. We hypothesized that both abiotic environment and tree neighbours shaped sister species relative distribution within species complexes, due to the pervasive habitat specialization along topography of several congeneric taxa [@Allie2015].

# Material and Methods

## Study site

The study was conducted in the northernmost part of the Guiana Plateau region, at the Paracou field station. The site is characterized by an average of 3041 mm annual rainfall and mean air temperature of 25.71 °C. Old tropical forest with an exceptional richness (over 750 woody species) develops in this area acress a succession of small hills rising to 10–40 m a.s.l. [@Gourlet-Fleury2004].

The site contains 16 permanent inventory plots (fifteen plots of 6.25 ha and one of 25 ha) which have been censused (diameter at breast height > 10 cm) every 1-2 years for more than 35 years. Nine of the plots were logged and subjected to human-induced disturbance in 1986.

## Species complexes

We extracted from the Paracou database species complexes defined on evidence for low phylogenetic resolution or plastid DNA sharing in the clade [@baraloto_using_2012-1; @Gonzalez2009]: *Eschweilera* with clades Parvifolia and Chartacea (Lecythidaceae; Chave *et al*, unpublished; Heuertz *et al*. unpublished); *Licania* [Chrysobalanaceae; @Bardon2016]; *Iryanthera* (Myristicaceae); *Symphonia* (Clusiaceae); *Talisia* (Sapindaceae). We further removed species with less than 10 individuals in Paracou in 2015, resulting in 6 species complexes including 2 to 9 species (Table \@ref(tab:ComplexTab)).

## Environmental variables

Three little correlated (Pearson's r < 0.13, see supplementary material S1) environmental descriptors were chosen to depict the distribution of species and species complexes. Topographic wetness index ($TWI$) was selected between various abiotic descriptors available and highlighted water accumulation areas, which is crucial information at local scale when it comes to tropical ecosystems [@ferry2010higher].

A neighbor crowding index [$NCI$; @Uriarte2004a] was used as a descriptor of biotic interactions through tree neighborhood, implying that other biotic effects, such as herbivory and pathogen effects, were not directly taken into account. The neighborhood crowding index $NCI_i$ for each tree individual $i$ was calculated with the following formula:
$$NCI_i = \sum _{j|\delta_{i,j}<20m} ^{J_i} DBH_j ^2 e^{-\frac{1}{4}\delta_{i,j}}$$
where $DBH_j$ is the diameter of neighboring tree $j$ and $\delta_{i,j}$ its distance to individual tree $i$. $NCI_i$ is computed for all neighbors at a distance $\delta_{i,j}$ inferior to 20m because NCI showed very little effect beyond 20m in preliminary analysis. Size effect of neighbors through their diameter was set to 2 to represent neighbors through their basal area. Distance effect of neighbors was set to -0.25 corresponding to neighbors beyond 20m having less than 1% effect compared to the effect of neighbors at 0m. The neighbor crowding index was further split between different taxa categories: (i) $NCI_{con}$ represented conspecific neighbors, (ii) $NCI_{complex}$ represented neighbors belonging to the same species complex, (iii) $NCI_{hetero}$ represented heterospecific neighbors and (iv) $NCI_{other}$ represented neighbors not belonging to the species complex.

Descriptors used were derived from field censuses ($NCI$) or derived from a 1-m resolution digital elevation model built using LiDAR campaign done in 2015 ($TWI$) using SAGA-GIS [@Conrad2015].

## Analysis

### Distribution of complexes

Distribution of species complex was inferred by considering occurrences of all species belonging to the complex as presence, and all occurrences of trees belonging to other species as pseudo-absences. The presence of each species complex $Presence_{species~complex}$ was inferred with a logistic regression following a Bernoulli distribution (best model form tested, see supplementary material S2 and S3):
$$Presence_{species~complex} \sim Bernoulli[logit^{-1}(\alpha + \beta*X+\gamma*X^2)]$$
were $X$ is the matrix of environmental descriptors, $\alpha$ is the intercept, $\beta$ is a vector representing the slope of environmental descriptors and $\gamma$ is a vector representing the quadratic form of environment variable effects.

### Distribution of species

Joint-distributions of species were inferred within species complexes with softmax regression following a conjugated Dirichlet Process and Multinomial distribution:
$$Presence_{species|species~complex} \sim Dirichlet~Multinomial(softmax(\alpha + \beta*X + \gamma*X^2))$$
were ${Presence_{species|species~complex}}_i$ representing the species for individual $i$ as a simplex of 0 and 1 (0 for all other species from the species complex and 1 for the species), $X$ is the matrix of environmental descriptors, $\alpha$ is the vector of species intercepts, $\beta$ is a matrix representing the slope of environmental descriptors and $\gamma$ is a matrix representing the quadratic form of environment variable effects for each species.

Environmental descriptors were all scaled for the model inference, in order to ease model inference and later compare strength of effects between distributions. A Bayesian method was used to infer parameters of the model regarding each complex and species using `stan` language [@Carpenter2017] and `rstan` package [@Team2018] in the R environment [@RCoreTeam2018].

# Results

The topographic wetness index ($TWI$) and the neighbor crowding index ($NCI$) had weak effects on the relative abundance of species complexes (Fig. \@ref(fig:ComplexesDistribution)). Nevertheless, all relative abundances of species complexes increased with $NCI_{complex}$ (except for *Talisia*), increased neighboring of trees belonging to the same species complex increased the probability to find a tree belonging to the complex. And *Symphonia* and *Iryanthera* showed a small preference for wet habitats (positive $beta$ for $TWI$) whereas *Licania* showed a small preference for drier habitats (negative $beta$ for $TWI$).

On the other hand, the topographic wetness index ($TWI$) strongly influenced species distribution within complexes, with a gradient of response from positive to negative effect within each complex (Fig. \@ref(fig:betaTWI)). For instance, *S. globulifera*, *E. coriacea*, and *I. hostmanii* species relative abundance increased with water accumulation when it decreased for *S. sp1*, *E. sagotiana*, and *I. sagotiana* within *Symphonia*, *Parvifolia* and *Iryanthera* complexes respectively (Fig. \@ref(fig:SpeciesDistribution)). Still, the $TWI$ influence was null or low for *Chartacea* and *Talisia* complexes.

Finally, $NCI_{hetero}$ and $NCI_{con}$ had no or weak effects on species relative abundance within species complexes. Still conspecific interactions ($NCI_{con}$) had an effect on species from *Parvifolia* and *Symphonia* complexes with a change of dominance within the complex with increasing congeneric interactions (*E. sagotiana* and *E. decolorans* replacing *E. coriacea* and *E. wachenheimii*, *S. globulifera* replacing *S. sp1* within *Parvifolia* and  *Symphonia* complexes respectively). And heterospecific interactions ($NCI_{hetero}$) had an effect on species from *Symphonia* complex with a change of dominance within the complex with increasing heterospecific interactions (*S. sp1* replacing *S. globulifera*).

# Discussion

Despite species complexes key role to understand Neotropical diversification [@Pinheiro2018], and the importance of ecological niches exploitation in species complexes establishment [@Runemark2019], little is known about Neotropical species complexes distribution [@baraloto_using_2012-1; @Levi2019a; @TerSteege2013]. Here, we show that species complexes of Neotropical trees are widely spread across habitats at local-scale. Moreover, sister species show pervasive niche differentiation along topographic position, water accumulation, and competition gradients within species complexes. Consequently, if supported by ecological reproductive isolation of species and adaptive introgression at the complex level, our results suggest that both habitat specialisation of sister species and widespread distribution of species complexes might explain the success of these Neotropical species complex at the regional scale [@TerSteege2013].

## Species complexes are widely spread across habitats

Biotic and abiotic environment null or weak effect on species complexes distribution suggest that species complexes are widely spread across habitats at local-scale. Topographic wetness index (TWI) reflects topographic position where water accumulates. Whereas, neighbor crowding index of trees not belonging to the species complex (NCIother) represents competition with neighbors and thus access to water, nutrients and light. Consequently, TWI and NCIother represent the combined effects of abiotic and biotic environment in defining local scale habitats for species and species complexes distributions. In our study, TWI and NCIother had no effect on species complexes distribution, at the exception of a weak effect of TWI on *Symphonia*, *Iryanthera*  and *Licania* abundances, highlighting the widespread distribution of species complexes across habitats.

*Symphonia* and *Iryanthera* advantage in wet habitats was already evidenced experimentally [@Ferry2007], and might be due to a more constraining habitat with anoxy requiring species specific adaptations. In addition, neighbor crowding index of trees belonging to the species complex (NCIcomplex) had a positive effect on relative abundances of all species complexes, highlighting trees of the same species complexes to be aggregated in a radius of 20 meters. Species aggregated distribution is a widely observed phenomenon in forests pan-tropically [@Zhang2013; @PLOTKIN2000; @Wang2012; @condit_spatial_2000; @Forget1999]. Aggregation can be explained by dispersal limitation or habitat association. Consequently, due to the weak to null effect of abiotic and biotic environment variables on species complexes distribution, we assumed species complexes to be distributed in aggregates due to the sole effect of dispersal limitation of their species.

## Species show pervasive niche differentiation within species complexes

The repeated opposed response of sister species relative distribution within species complexes to biotic and abiotic environment highlighted a pervasive niche differentiation of species within four locally-abundant species complexes. Indeed, sister species within *Iryanthera*, *Licania*, *Symphonia*, and *Eschweilera* clade *Parvifolia* species complexes are segregated along a gradient from wet habitats with low heterospecific neighboring (conspecific neighboring for *Eschweilera*)  to drier habitats with increased heterospecific neighboring (conspecific neighboring for *Eschweilera*). The sister species segregation along a topographic, wet, and neighboring gradient could be linked to the increased dynamic of bottomlands compared to slopes and plateaus in Paracou [@ferry2010higher], with seasonally flooded and lower topographic position suffering from an increased tree mortality resulting in diminished standing biomass. Our results are in agreement with previous evidence of species-habitat associations with topography and seasonal flooding between sister species at local [@Allie2015; @Ferry2007] and regional [@Guitet2015; @Fine2014] scales. 

## Conclusion

We found Neotropical tree species complexes to be widespread across habitats at local scale, while the sister species composing them showed pervasive habitat differentiation along topography and competition gradients. The pervasive niche differentiation between sister species might be related to repeated functional strategy adaptation to local environment [Schmitt *et al., under review*; @Ferry2007] with several diversification events across taxa (*e.g.* @Fine2014 for Protiae). Habitat specialisation increase isolation barriers between species, allowing reproductive isolation for establishment of sister species [@Runemark2019]. Meanwhile, porous genomes allow for the transfer of adaptive genes through introgression that may benefit one or several sister species, allowing coevolution at the species complex level [@Cannon2015b]. Consequently, if habitat specialisation allow reproductive isolation and porous genomes allow adaptive introgression, our results suggest that both habitat specialisation of sister species and widespread distribution of species complexes might explain the success of these Neotropical species complex at the regional scale [@TerSteege2013]. Further studies both about the effect of habitat specialisation on reproductive isolation and adaptive introgression are needed to correctly assess this hypothesis. Ongoing genetic data production on individuals from *Symphonia* and *Eschweilera* clade *Parvifolia* species complexes will hopefully allow us to explore this question in future contributions.

# Acknowledgements

We thank the University of Bordeaux for a PhD grant to Sylvain Schmitt. We are grateful to Pascal Petronelli and the CIRAD inventory team for their work on tree inventories and botanical identification.

# Authors’ contributions

NT and SS conceived the ideas; SS, BH and NT designed methodology; SS and BH analysed model outputs; SS, NT and MH led the writing of the manuscript. All authors contributed critically to the drafts and gave final approval for publication.

# Data accessibility

TWI and spatial positions of individuals were extracted from the Paracou Station database, for which access is modulated by the scientific director of the station (https://paracou.cirad.fr).

# References

<div id="refs"></div>

\newpage 

## Table captions

__Table \@ref(tab:ComplexTab)__ Identified species complexes with species including more than 10 individuals in Paracou.


## Figure captions

__Figure \@ref(fig:ComplexesDistribution)__ Relative abundance of species complexes. Subplot **A** represent $\beta$ parameter values for each descriptor and species complex, with the circle representing the mean value of the parameter posterior, thick lines the 50% confidence interval and thin lines the 95% confidence interval. Subplot **B**, represent species complex predicted relative abundance, with dashed line indicating observed relative abundance of species complex in Paracou, and solid line and area representing respectively the mean and the 95% confidence interval of projected relative abundance of species complexes depending on descriptors. The color indicates the species complex.

__Figure \@ref(fig:betaTWI)__ TWI and NCI heterospecific and conspecific effects on species relative distribution within species complexes. $\beta_{TWI}$, $\beta_{NCIhetero}$, $\beta_{TWIcon}$ posterior are represented for each species complex and species with mean (circle), 50% credibility interval (thick segment) and 95% confidence interval (thin segment).

__Figure \@ref(fig:SpeciesDistribution)__ Relative abundance of species within species complexes. Predicted relative abundance of species within species complexes with dashed line indicating observed relative abundance of species within species complex in Paracou, and solid line and area representing respectively the mean and the 95% confidence interval of projected relative abundance of species complexes depending on descriptors. The color indicates the species complex.

\newpage 

<!-- tables -->

```{r ComplexTab}
trees %>% 
  left_join(Complexes, by = c("Genus", "Species")) %>% 
  mutate(Species = ifelse(Species == "grandiflora_form2", "grandiflora", Species)) %>% 
  filter(!is.na(Complex)) %>% 
  group_by(Complex, Genus, Species) %>% 
  summarise(N = n()) %>% 
  arrange(Complex, desc(N)) %>% 
  kable(caption = "Identified species complexes with species including more than 10 individuals in Paracou.", format = "pandoc") %>% 
  kable_styling(full_width = F)
```

<!-- figures -->

```{r ComplexesDistribution, fig.cap="Relative abundance of species complexes. Subplot **A** represent $\\beta$ parameter values for each descriptor and species complex, with the circle representing the mean value of the parameter posterior, thick lines the 50% confidence interval and thin lines the 95% confidence interval. Subplot **B**, represent species complex predicted relative abundance and solid line and area representing respectively the mean and the 95% confidence interval of projected relative abundance of species complexes depending on descriptors. The color indicates the species complex."}
g1 <-lapply(fitsComplex, function(fit) 
  bayesplot::mcmc_intervals_data(as.array(fit, pars = "beta"))) %>% 
  bind_rows(.id = "complex") %>% 
  mutate(descriptor = gsub("([[:alpha:]])", "", parameter)) %>% 
  mutate(descriptor = gsub("([[:punct:]])", "", descriptor)) %>% 
  mutate(descriptor = recode_factor(descriptor, `1` = "TWI", 
                                    `2` = "NCIother", `3` = "NCIcomplex")) %>% 
  mutate(descriptor = factor(descriptor, levels = c("NCIcomplex",
                                                    "NCIother", "TWI"))) %>% 
  mutate(parameter = gsub("([[:digit:]])", "", parameter)) %>% 
  mutate(parameter = gsub("([[:punct:]])", "", parameter)) %>%
  dplyr::select(-outer_width, -inner_width, -point_est) %>% 
  dplyr::select(complex, parameter, descriptor, ll, l, m, h, hh) %>% 
  ggplot(aes(x = complex, xend = complex, col = complex, fill = complex)) +
  geom_hline(yintercept = 0, color = "gray90", size = 0.5) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  coord_flip() +
  facet_wrap(~ descriptor, labeller = label_parsed, scales = "free_y") +
  ylab("") + xlab("Species\ncomplex") + 
  scale_fill_discrete(guide = "none") +
  scale_color_discrete(guide = "none") +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank())
g2 <-predComplex %>% 
  left_join(group_by(data, Complex) %>% 
              summarise(Yo = n()/nrow(.))) %>% 
  ggplot(aes(x = value, col = Complex)) +
  geom_ribbon(aes(ymin = Y5, ymax = Y95), alpha = 0.2) +
  geom_line(aes(y = Y)) +
  # geom_hline(aes(yintercept = Yo, col = Complex), lty = "dashed") +
  facet_wrap(~ variable, nrow = 1, scales = "free") + 
  xlab("") + ylab("Species complex relative abundance") +
  scale_y_sqrt() +
  theme(legend.position = "bottom",
        strip.background = element_blank(),
        strip.text.x = element_blank())
cowplot::plot_grid(g1, g2, nrow = 2, rel_heights = c(1,2), 
                   labels = LETTERS[1:2], label_x = 0.1) 
```

```{r betaTWI, fig.cap="TWI and NCI heterospecific and conspecific effects on species relative distribution within species complexes. $\\beta_{TWI}$, $\\beta_{NCIhetero}$, $\\beta_{TWIcon}$ posterior are represented for each species complex and species with mean (circle), 50% credibility interval (thick segment) and 95% confidence interval (thin segment).", fig.height=10} 
species <- data %>%
  dplyr::select(Complex, Genus, Species) %>% 
  unique() %>% 
  mutate(Species = ifelse(Species == "grandiflora_form2", "grandiflora", Species)) %>% 
  group_by(Complex) %>% 
  arrange(Complex, Species) %>% 
  mutate(species = row_number()) %>% 
  mutate(SpeciesLong = paste(substr(Genus, 1, 1), Species))
lapply(fits, function(fit) 
  bayesplot::mcmc_intervals_data(as.array(fit, pars = "beta"))) %>% 
  bind_rows(.id = "Complex") %>% 
  mutate(descriptor = gsub("([[:alpha:]])", "", parameter)) %>% 
  mutate(descriptor = gsub("([[:punct:]])", "", descriptor)) %>% 
  mutate(parameter = gsub("([[:digit:]])", "", parameter)) %>% 
  mutate(parameter = gsub("([[:punct:]])", "", parameter)) %>% 
  mutate(species = as.numeric(substr(descriptor, 1, 1))) %>% 
  left_join(species) %>% 
  mutate(descriptor = substr(descriptor, 2, 2)) %>% 
  mutate(descriptor = recode_factor(descriptor, `1` = "TWI", 
                                    `2` = "NCIhetero", `3` = "NCIcon")) %>% 
  dplyr::select(Complex, parameter, descriptor, SpeciesLong, ll, l, m, h, hh) %>% 
  mutate(parameter = ifelse(parameter != "alpha",
                            paste0(parameter, "[", descriptor, "]"),
                            parameter)) %>% 
  mutate(Complex = gsub("E. ", "", Complex)) %>% 
  ggplot(aes(x = datatools::reorder_within(SpeciesLong, m, Complex), 
             xend = datatools::reorder_within(SpeciesLong, m, Complex), 
             col = SpeciesLong, fill = SpeciesLong)) +
  geom_hline(yintercept = 0, color = "gray90", size = 0.5) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  coord_flip() +
  facet_grid(Complex ~ parameter, labeller = label_parsed, scales = "free") +
  datatools::scale_x_reordered() +
  xlab("") + ylab("") + 
  scale_fill_discrete(guide = "none") +
  scale_color_discrete(guide = "none")
```

```{r SpeciesDistribution, fig.cap="Relative abundance of species within species complexes. Predicted relative abundance of species within species complexes with dashed line indicating observed relative abundance of species within species complex in Paracou, and solid line and area representing respectively the mean and the 95% confidence interval of projected relative abundance of species complexes depending on descriptors. The color indicates the species complex.", fig.height=10}
predSpecies %>%
    group_by(complex) %>% 
  left_join(data %>% 
              group_by(Complex) %>% 
              mutate(N_complex = n()) %>% 
              group_by(Complex, N_complex, Genus, Species) %>% 
              summarise(N_species = n()) %>% 
              mutate(RelAbund = N_species / N_complex),
            by = c("complex" = "Complex", "species" = "Species")) %>% 
  mutate(species = ifelse(species == "grandiflora_form2", "grandiflora", species)) %>%
  mutate(Species = paste0(substr(Genus, 1,1), ". ", species)) %>% 
  do(g = ggplot(data = ., aes(x = value, col = Species)) + 
       geom_ribbon(aes(ymin = Y5, ymax = Y95), alpha = 0.2) +
       geom_line(aes(y = Y), lwd = 1.3) +
       geom_hline(aes(yintercept = RelAbund, col = Species), linetype = "dashed") +
       facet_grid(complex ~ variable, scales = "free_x") +
       guides(col = guide_legend(nrow=3)) +
       theme(legend.position = "bottom") +
       ylab("Species relative abundance") + xlab("Standardized value") +
       scale_y_sqrt()) %>% 
  gridExtra::grid.arrange(grobs = .$g)
```
