```{r setup_chloro, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
# library(seqinr)
library(Biostrings)
library(rentrez)
library(tidyverse)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Genomes/"
```

# Appendix 2: Chloroplast data

For cloroplastic sequences (see figure\@ref(fig:seqTree)) we have different sequences available:

* The assembly of *Symphonia globulifera* chloroplaste by Rocio Bautista & M. Gonzalo Claros
* Available plastid sequences from American Symphonia available on NCBI (https://www.ncbi.nlm.nih.gov/nuccore/?term=Symphonia+globulifera)

The aim is to select genes from the African assembly that are relativelly stable checking their correspondance to American sequences found on NCBI. We will assess correspondance by blasting American sequences from NCBI on Rocio Bautista & M. Gonzalo Claros assembly from Africa.

## MIA assemblage

Rocio Bautista & M. Gonzalo Claros already anotated chloroplastic gene from *Symphonia globulifera* from African samples (figure \@ref(fig:chloroRef)) that we will use as reference for the blast:

```{r chloroRef, fig.cap="Result from anotation of 14 scaffolds from chloroplastic genom from *Symphonia globulifera* witg cpGAVAS (Rocio Bautista & M. Gonzalo Claros)"}
include_graphics("/home/sylvain/Documents/BIOGECO/PhD/data/Symphonia_Genomes/Chloroplast_Africa/Final_Result/Annotacion/annotation_Chloroplast_Symphonia_scaffolds.png")
```

## NCBI data

We searched for nucleotide sequence from *Symphonia globulifera*, either as an organism or in all field, excluding microsatellites, restricting research to genomic DNA and RNA on genbank, and only in chloroplast. It resulted in following query : 

> ("Symphonia globulifera"[Organism] OR ("Symphonia globulifera"[Organism] OR symphonia globulifera[All Fields])) NOT microsatellite[All Fields] AND (plants[filter] AND biomol_genomic[PROP] AND ddbj_embl_genbank[filter] AND chloroplast[filter])

We then explored each population set (PopSet field) and corresponding article to filter sequences coming only from Central and South America (excluding pantropical datasets). We obtained the 6 following datasets :

* PopSet = 261864032 [@Kress2009]
* PopSet = 1139718125 (Caron,H., Molino,J.-F., Sabatier,D., Leger,P., Chaumeil,P., Scotti Saintagne,C., Frigerio,J.-M., Scotti,I., Franc,A. and Petit,R.J. Pervasive introgression in a tropical tree community)
* PopSet = 261279143 [@Kress2009]
* PopSet = 224986824 [@Gonzalez2009]
* PopSet = 263042640 [@Gonzalez2009]
* PopSet = 194410525 [@Dick2008]

We extracted gene bank ID for all *Symphonia globulifera* sequences from those population sets and retrieve corresponding fasta sequences from NCBI.

```{r sequences}
seq <- read_table(file.path(path, "NCBI", "sequence.txt"),
                  col_names = F) %>%
  rename(GeneBankId = X1, Name = X2) %>% 
  mutate(GeneBankId = gsub("Symphonia", "", GeneBankId)) %>% 
  mutate(Name = paste("Symphonia", Name))
seqFa <- entrez_fetch(db = "Nucleotide", id = seq$GeneBankId, rettype = "fasta")
write_file(seqFa, file.path(path, "NCBI", "sequence.fa"))
```

## NCBI data  blast on MIA assembly

We then built blast data base from MIA assembly with `makeblastdb` and blasted NCBI sequences on it with `blastn`.

```{bash blastChloro, eval=F, echo=T}
cd ~/Documents/BIOGECO/PhD/data/Symphonia_Genomes/NCBI
mkdir chloroplast_db
cp ~/Documents/BIOGECO/PhD/data/Symphonia_Genomes/Chloroplast_Africa/Final_Result/Annotacion/annot_fasta/genes.fasta chloroplast_db/chloroplast.fa
makeblastdb -in chloroplast_db/chloroplast.fa -parse_seqids -dbtype nucl
blastn -query sequence.fa -db chloroplast_db/chloroplast.fa -outfmt 6 -num_alignments 1 -out blast_result.txt
```

Extracted sequence from NCBI matched 3 genes from the MIA assembly (figure \@ref(fig:blastResult)). Almost all sequences (150 on 170) matched the **psbA_662_1723** gene.

```{r blastChloroResult, fig.cap="Number of sequences matching reference gene (total of 170 sequences blasted).", fig.height=4}
blast <- read_tsv(file.path(path, "NCBI", "blast_result.txt"),
                  col_names = F)
names(blast) <- c("Seq", "Gene", "Perc_Ident", "Alignment_length", "Mismatches",
                  "Gap_openings", "R_start", "R_end", "S_start", "S_end", "E", "Bits")
blast %>% 
  group_by(Gene) %>% 
  summarise(n = n()) %>% 
  ggplot(aes(Gene, n)) +
  geom_bar(stat = "identity")
```

```{r blastChloroResult2}
db <- readDNAStringSet(file.path(path, "NCBI", "chloroplast_db", "chloroplast.fa"))
db[unique(blast$Gene),] %>% 
  data.frame(Gene = names(.), Width = width(.)) %>% 
  select(Gene, Width) %>% 
  kable(row.names = F, caption = "Selected genes.")
```

