---
title: Title
date: '`r Sys.Date()`'
output:
  bookdown::word_document2:
    reference_docx: ./template/template.docx
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
  bookdown::html_document2:
    number_sections: false  
    toc: true
    toc_float: yes
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/mee.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
library(tidyverse)
library(bayesplot)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, 
               fig.height = 8, fig.width = 8,
               cache = T, cache.lazy = F)
```

1. Sylvain Schmitt
    - Corresponding author
    - sylvain.m.schmitt@gmail.com
    - orcid.org/0000-0001-7759-7106
    - Univ. Bordeaux, INRAE, BIOGECO, 69 route d’Arcachon, 33610 Cestas France
1. Myriam Heuertz
    - heuertzm@gmail.com
    - orcid.org/0000-0002-6322-3645
    - Univ. Bordeaux, INRAE, BIOGECO, 69 route d’Arcachon, 33610 Cestas France
1. Niklas Tysklind
    - Niklas.Tysklind@ecofog.gf
    - INRAE, UMR EcoFoG (Agroparistech, CNRS, Cirad, Université des Antilles, Université de la Guyane), Campus Agronomique, 97310 Kourou, French Guiana
1. Bruno Hérault
    - bruno.herault@cirad.fr
    - orcid.org/0000-0002-6950-7286
    - CIRAD, UPR Forêts et Sociétés, Yamoussoukro, Côte d’Ivoire
    - Forêts et Sociétés, Univ Montpellier, CIRAD, Montpellier, France
    - Institut National Polytechnique Félix Houphouët-Boigny, INP-HB, Yamoussoukro, Côte d’Ivoire

\newpage 

# Abstract

**To be written**

## Keywords

XX ;

\newpage 

# Introduction 

<!-- * Hook on Neotropical diversity ? -->
<!--     * highest biodiveristy [@Gaston2000] -->
<!--     * long date [@connell_diversity_1978] -->
<!--     * numbers Amazonia [@TerSteege2013] -->
<!-- * Ecological point of view for Neotropical diversity - coexistence -->
<!-- * Transition intraspecific variability neglected => population genomics -->
<!-- * Evolutionary point of view for Neotropical diversity -->
<!--     * Hyrbidization interest [@Cannon2015b] -->
<!--     * Syngameon success ? [@Cannon2019a] -->
<!--     * Species complex key role ? [@Pinheiro2018] -->
<!-- * Eco-evolutionary study to unravel Neotropical diveristy ? -->
<!-- * Species complex and syngameons as a study model ? -->
<!-- * __previous point as an introduction of article, PhD thesis, or both ?__ -->

<!-- Points needed for discussion : -->

<!-- * Species complex & syngameons concepts & evolutionary consequences -->
<!--     * Species complex definition [@Pernes1984] -->
<!--     * syngameon definition [@Suarez-Gonzalez2018] -->
<!--     * Species complex origin [@Seehausen2004] -->
<!--     * Genera sharing of genetic diversity [@Caron2019] -->
<!--     * Hybridization frequency [@Whitney2010] -->
<!--     * Hybrid genomes [@Runemark2019] -->
<!--     * gene flow and adaptation [@Tigano2016] -->
<!--     * oak example [@Petit2002] -->
<!--     * oak adaptation [@Leroy2019] -->
<!-- * Ecology on species habitat preferences, especially among congeneric and closely-related species -->
<!--     * [@Fine2004] -->
<!--     * [@Esteves2013] -->
<!--     * [@Crystal2003] -->
<!--     * [@Allie2015] -->
<!--     * [@Lan2016] -->
<!--     * [@Yamasaki2013] -->
<!-- * Ecological genomics -->
<!-- * Local adaptation -->
<!-- * Ecological speciation litterature ? -->
<!-- * Species habtitat specialization and speciation among sister species within species complex -->
<!-- * Growth and performance link toward fitness and local adaptation / speciation (several points in one) -->
<!-- * Syngameon eco-evolutionary consequences -->
<!-- * Putative role of syngameons in Neotropical diversification -->

<!-- Maybe *Symphonia* details can be kept for Mat&Met or subtly introduced to illustrate  -->

<!-- * *Symphonia* current taxonomy -->
<!-- * *Symphonia* morphotypes in Paracou -->
<!-- * *Symphonia* distribution in Paracou -->
<!-- * *Symphonia* functional traits in Paracou -->
<!-- * *Symphonia* growth patterns in Paracou -->

...  to address the following questions: 
(1) how is genetic diversity structured in a Neotropical forest?
(2) how does abiotic and biotic habitat affect population genetic diversity and structure?
(3) what are the consequences of genetic diversity and structure for individual tree growth?

# Material and Methods

## Study site

The study was conducted in the northernmost part of the Guiana Plateau region, at the Paracou field station. 
The site was characterized by an average of 3,041 mm annual rainfall and a mean air temperature of 25.71 °C. 
Old tropical forest with an exceptional richness (over 750 woody species) has developed among the succession of small hills of this area, rising to 10–40 m a.s.l. [@Gourlet-Fleury2004].
The site is made of 16 permanent plots (fifteen 6.25 ha plus one 25 ha) which have been censused (DBH>10) every 1-2 years for more than 35 years. Nine of the plots were logged and subjected to human-induced disturbance in 1986.

## Plant material

402 individuals from *Symphonia globulifera* (Clusiaceae), an Amazonian hyperdominant tree species [@TerSteege2013], were sampled in 2017 during the dry season (from September to December) in Paracou.
*Symphonia globulifera* L.f (Clusiaceae) was previously recognized as composed of two morphotypes in French Guiana [@Sabatier1997].
*S. globulifera sensu stricto* and *Symphonia sp.1* occur in sympatry but in differentiated habitats, with *S. globulifera* preferentially growing in valley bottoms and *S. sp1* preferentially exploiting a variety of drier habitats [@Allie2015; Schmitt et al., in prep].
The genus *Symphonia* has been highlighted as a species complexe with low phylogenetic resolution and high levels of plastid DNA sharing among sister species [@baraloto_using_2012-1; @Caron2019; @Gonzalez2009; @Torroba-Balmori2017]. 
In addition, outgroup were built with 13 individuals of *Symphonia globulifera* from Africa (Sao Tome, Gabon, Cameroun, Congo, Benin, Liberia, Ivory Coast, and Gana), 7 *Symphonia globulifera* from South America (Brazil, Costa Rica and Panama), 2 *Symphonia nectarifera* from Madagascar, 2 *Symphonia urophylla* from Madagascar, 5 *Pentadesma butyracea* from Benin and Cameroun and 1 *Pentadesma grandifolia* from Cameroun.
Leaves were collected from the 432 (402 + 30) individuals and desicated using silica gel (Fig. \@ref(fig:map)). 

## Design of sequence capture 

We designed in *silico* a set of 20,000 80-mer probes sequences for the genome of *Symphonia globulifera* (Supplementary Material). 
We used a published low-coverage draft genome of an African *Symphonia globulifera* [@Olsson2017], an unpublished draft genome of a *Symhpnoia globulifera* from French Guiana [Scotti et al., in prep], an unpublished transcriptome from 20 juveniles of *Symphonia globulifera* from French Guiana [Tysklind et al., in prep], and genomic reads of individuals from French Guiana [Torroba-Balmori et al., unpublished]. 

We aligned genomic reads on the two genome drafts with `bwa` [@Li2009]. 
We kept scaffolds from the two genome drafts with a length superior to 1 kbp and at least one matching alignment with a read with a single match on the genome, and merged the two filtered genome drafts with `quickmerge` [@Chakraborty2016]. 
We aligned transcripts on the new filtered genome draft with `BLAT` [@Kent2002] and selected 533 scaffolds without transcript-match, *i.e.* anonymous scaffolds. 
We finally masked repetitive regions with `RepeatMasker` [@Smit2015] and selected 533 1-kbp loci within the 533 previous scaffolds.

Similarly, we filtered transcripts from the 20 juveniles of *Symphonia globulifera* from French Guiana [Tysklind et al., in prep] based on SNP quality, type and frequency. 
We further detected  open reading frames (ORFs) using `transdecoder` [@Haas2013], and selected transcripts with non-overlapping  ORFs including a start codon (Methionine). 
We kept ORFs with an alignment on scaffolds from the aforementioned genome draft, using `BLAT` [@Kent2002], and masked repetitive regions with `RepeatMasker` [@Smit2015]. 
We finally selected 1,150 loci of 500-bp to 1-kbp, from 100 bp before the start to a maximum of 900 bp after the end of the ORFs, resulting in 1-Mbp of transcriptomic loci. 

## Genomic data production

*Libraries preparations*

Genomic DNA was extracted from 5 mg of dried leaf tissue with a CTAB protocol [@Doyle1987]. 
Libraries were digested with 'Ultra II FS Enzyme Mix' (new England Biolabs Inc, MA, USA) for a target size of 150 bp, and built with the 'NEBNext Ultra II FS DNA Library Prep kit for Illumina'(new England Biolabs Inc, MA, USA).
We amplified and tagged libraries using 5 $\mu L$ of adaptor-ligated DNA, 8.3 $\mu L$ of 'NEBNext Ultra II Q5 Master Mix' (new England Biolabs Inc, MA, USA), 2x 1.6 $\mu L$ of Index Primer i5 and i7 from 'NEBNext Multiplex Oligos for Illumina (Dual Index Primers Set 1 and Set 2)' (new England Biolabs Inc, MA, USA). 
Initial denaturation (98°C for 30 s) was followed by 8 cycles (98°C for 10 s and 65°C for 1 min 30 s) and a final extension (65°C for 5 min). 
We pooled libraries in 4 equimolar multiplex and captured sequences by hybridization for 80 hours.
We used the 20,000 80-mer probes sequences designed for the capture using myBaits Custom 1-20K (Arbor Biosciences, MI, USA) and corresponding myBaits V4 protocol. 
We finally pooled and sequenced the previous 4 multiplexes in 2 lanes using Illumina HiSeq 4000 with pair-end for 2x150bp reads.

*SNP call and filtering*

We assessed quality from raw reads `multiqc` [@Ewels2016] and trimmed them with `trimmomatic` [@Andrews2010]. 
We kept only pair-end reads without adaptors and a phred score above 15 in a sliding window of 4.
70% of trimmed reads mapped off-targets using `bwa` [@Li2009]. 
We thus mapped trimmed reads on the hybrid reference built for target design of sequence capture using `bwa` [@Li2009], `picard` [@BroadInstitute2018], `samtools` [@Li2009a] and `bedtools` [@Quinlan2010].
We called variants for each individual using `HaplotypeCaller`, aggregated variants using `GenomicsDBImport` and jointly-genotyped individuals using `GenotypeGVCFs` all in `GATK4` software [@Freksa1993]. 
We filtered biallelic SNPs with a quality above 30, a quality by depth above 2, a Fisher strand bias below 60 and strand odd ratio above 3 using `GATK4` [@Freksa1993]. 
Finally, we filtered individuals and SNPs for missing data with a maximum of 95% and 15% of missing data per individual and SNP, respectively, using `plink2` [@Chen2019].
We obtained 454,262 biallelic SNPs over 406 individuals.

*Annotation*

We used the previous alignments (see design of sequence capture) from genome drafts [@Olsson2017; Scotti et al., in prep] with the transcriptome [Tysklind et al., in prep] to classify called SNPs into (i) anonymous SNPs (on scaffolds matching no transcripts), (ii) putatively-hitchhiker SNPs (close to a transcript or within an intron), and (iii) genic SNP (within an exon).
We used transcript annotation of gene ontology to further annotate called SNPs. 
We used long ORFs detection with `TransDecoder` [@Haas2013], their alignment with `blastp` on the `Uniprot` database [@Bateman2019; @Bias1991] and with `hmmscan` on `Pfam` database [@Johnson2010; El-Gebali2019] in addition to called SNPs position into `SNPdat` [@Doran2013] to assess SNP synonymy.

*SNP filtering for quantitative genomics*

This first dataset including 454,262 SNPs was used for population genetic and demographic analyses. 
But, minor alleles and linkage disequilibrium bias the number of fixed loci, increasing false-positives upward in genomic scans [@Foll2008].
So we built a second dataset for quantitative genomics, filtering variants with a minor allele frequency above 5% (18 individuals) and with linkage disequilibrium $r^2<0.99$. 
We further removed admixed individuals from genomic scans (see population genetic analyses for admixed individuals definitions). 
We obtained 70,737 biallelic SNPs over 372 individuals.

## Analyses

### Populations

We investigated population structure using `admixture` [@Alexander2011], with 10 repetitions of K genetic groups varying from 1 to 10. 
We assessed the number of gene-pools with cross validation.
We defined as admixed individuals with a membership to gene-pools below 90% and the remaining individuals as *pure-bred*. 
We further investigated admixture with `introgress` R package [@Gompert2010], using *pure-bred* individuals as parental populations and all individuals as hybrid population.
We further removed from subsequent analyses admixed individuals. 
We then measured following diversity metrics of populations: (i) the nucleotide diversity $\pi$, (ii) the ratio of non-synonymous over synonymous SNPs $\frac{\pi_a}{\pi_s}$, (iii) pairwise fixation index $F_{ST}$, and (iv) Tajima's $D$, using `VCFTools` [@Danecek2011].
All metrics were measured per population and SNP annotation (anonymous, putatively-hitchhiker, and genic) on a sliding window of 100 bp. 
We additionally built per population site frequency spectrum (SFS) to investigate demography patterns.

We used `treemix` [@Pickrell2012] on a subset of individuals per Paracou population and outgroups, pooled as Madagascar, Africa, Sao Tome, Brazil and Costa Rica, to build a pantropical phylogeny of *Symphonia* populations.
We allowed from 0 to 10 migration events and built a tree per SNP annotation (anonymous, putatively-hitchhiker, and genic).

We used `bayescan` [@Foll2008] to detect fixed SNPs between the different populations using the 70,737 SNPs filtered for quantitative genomics (see genomic data production).
We considered SNPs with p-value corrected for multiple testing by false discovery rate below 5% as significant outliers.
We tested genic SNP which were outliers for enrichment in gene ontology with the R package `clusterProfiler` [@DeLaCruz2014].

### Environments

We did environmental association analyses [@Rellstab2015a] using general linear models from genome wide association studies (GWAS). 
Counterintuitively, we used the environment as the response variable and genetic structure (populations) and relatedness (kinship matrix) as explanatory variables. 
We assumed that the environment where individuals have grown above 10-cm DBH was strongly correlated to the individual heritable phenotypes [e.g. @Eckert2010]. 

We used two uncorrelated variables (Pearson'$r=-0.16$). 
We selected the topographic wetness index (TWI) among several abiotic descriptors as a proxy of water accumulation. 
Waterlogging and topography have been highlighted as crucial for forest dynamics [@ferry2010higher], species-habitats [@Allie2015, Schmitt et al., submitted], and *Symphonia* phenotypic variation [Schmitt et al., submitted]. 
TWI was derived from a 1-m resolution digital elevation model using SAGA-GIS [@Conrad2015] based on a LiDAR campaign done in 2015.
The neighborhood crowding index [$NCI$; @Uriarte2004] integrated over time was used to represent biotic asymmetric competition through tree neighborhood over time.
The Neighborhood Crowding Index $NCI_i$ from tree individual $i$ was calculated with following formula:

\begin{equation}
    NCI_i = (\sum_{t_0} ^{2015} \sum _{j~|~\delta_{i,j}<20m} ^{J_{i,t}} DBH_{j,t}^2 e^{-\frac14.\delta_{i,j}}.dt).\frac1{\Delta t}
   (\#eq:NCI)
\end{equation}

with $DBH_j$ the diameter from neighboring tree $j$ and $\delta_{i,j}$ its distance to individual tree $i$. 
$NCI_i$ is computed for all neighbors at a distance $\delta_{i,j}$ inferior to the maximum neighboring distance of 20 meters. 
The power of neighbors $DBH$ effect was set to 2 to consider the neighbors surface. 
The decrease of neighbors $DBH$ effect with distance was set to 0.25 to represent trees at 20 meters of the focal trees having 1% of the effect of the same tree at 0  meters.

We used population and individual kinship in a lognormal Animal model [@Wilson2010] to estimate genetic variance associated to the environment where individuals grow.
We defined populations based on the gene-pools identified with `admixture`, removing admixed individuals.
We inferred individual kinship using KING [@Manichaikul2010], as the method is robust to population structure. 
We set to null negative values of kinship as they were confounding with population structure, and we further ensured that the matrix was positive-definitive using `nearPD`  function from the R package `Matrix`.
The environment $y_{p,i}$ where the individual $i$ in population $p$ grow was inferred with a lognormal distribution with following formula:

\begin{equation}
   \begin{aligned}
      y_{p,i} \sim logN(log(\mu_p.a_{i}),\sigma_1) \\
      a_{i} \sim MVlogN_N(log(1),\sigma_2.K)
   \end{aligned}
   (\#eq:AnimalEnv)
\end{equation}

where $u_p$ is the mean environment of population $p$, $a_i$ is the breeding value of the individual $i$ and $\sigma_1$ is the shape parameter of the lognormal. 
Individual breeding values $a_i$ are defined following a multivariate lognormal law $\mathcal{MVlogN}$ of co-shape matrix defined as the product of the kinship matrix $K$ with estimated individual genotypic variation $\sigma^2_G$.
To estimate variances on a normal-scale, we log-transformed population fixed effect, genetic additive values, and we calculated conditional and marginal $R^2$ [@Nakagawa2013].
A Bayesian method was used to infer parameters using `stan` language [@Carpenter2017, see supplementary material] and `rstan` package [@Team2018] in the R environment [@RCoreTeam2020] using No-U-Turn Sampler [NUTS, @Hoffman2014] algorithm, performing better for estimating genetic parameters and breeding values [@Nishio2019].

We scanned the 70,737 SNPs filtered for quantitative genomics for association to environmental variables, controlling for kinship relations, using `GEMMA` [Genome-wide Efficient Mixed Model Analysis, @Zhou2012a]. 
We used linear mixed models (LMM) and Wald-test to define individual SNP significance ($p-value$).
We corrected $p-values$ by false discovery rate defining $q-values$ and considered SNPs as significant outliers with $q-values < 0.05$.
We tested genic SNP which were outliers for enrichment in gene ontology with the R package `clusterProfiler` [@DeLaCruz2014].

### Growth

We investigated effects of ecological and evolutionary processes on individual growth, using populations, kinship, topographic wetnes index and neighbor crowding index.
The individual growth of individual $i$ in population $p$ between individual recruitment $y_0$ and 2017, correspond to the difference of DBH between the two years, and is defined with a hierarchical model in a lognormal distribution as follow:

\begin{equation}
    DBH_{y=2017,p,i}-DBH_{y=y_0,p,i} \sim logN(log[\sum_{y=y_0}^{y=2017}AGR(DBH_{y,p,i})],\sigma_1)
   (\#eq:Growth)
\end{equation}

where the difference of DBH $DBH_{y=2017,p,i}-DBH_{y=y_0,p,i}$ is defined with a lognormal distribution located on the logarithm of the sum of annual growth rates $AGR$ during the period $y_0-2017$ and of shape $\sigma_1$. 
The annual growth rates $AGR$ for individual $i$ in population $p$ at year $y$ with a diameter of $DBH_{y,p,i}$ is defined following a Gompertz model [@Gompertz1825] already identified as the best model for growth-trajectories in Paracou [@Herault2011]:

\begin{equation}
   \begin{aligned}
      AGR(DBH_{y,p,i})=Gmax_i.exp(-\frac12[\frac{log\frac{DBH_{y,p,i}}{Dopt_p}}{Ks_p}]^2)
   \end{aligned}
   (\#eq:AGR)
\end{equation}

where $Gmax_i$ is the maximum growth potential (maximal AGR during individual life) for individual $i$, 
$Dopt_p$ is the population optimal diameter at which the individual reach its maximum growth potential, 
and $Ks_p$ is the population kurtosis defining the width of the bell-shaped growth-trajectory [see figure 1 in @Herault2011]. 
To ease model inference population optimal diameter $Dopt_p$ and kurtosis $Ks_p$ were defined as random population effect centered on a global $Dopt$ and $Ks$ with corresponding variances $\sigma^2_{P,Dopt}$ and $\sigma^2_{P,Ks}$. 
Individual $i$ maximum growth potential $Gmax_i$ was defined in a nested Animal model with a lognormal distribution as follow:

\begin{equation}
   \begin{aligned}
      Gmax_i \sim logN(log(Gmax_p.a_i.\beta_1.TWI.\beta_2.NCI), \sigma_{R,Gmax}) \\ 
      a_i \sim MVlogN(log(1), \sigma_{G,Gmax}.K)
   \end{aligned}
   (\#eq:AnimalGmax)
\end{equation}

where $Gmax_p$ is the mean $Gmax$ of population $p$, $a_i$ is the breeding value of individual $i$, $\beta_{TWI}$ and $\beta_{NCI}$ are the effects of TWI and NCI on $Gmax_i$, and $\sigma_{R,Gmax}$ is the shape of the lognormal distribution.
Individual breeding values $a_i$ are defined following a multivariate lognormal law $MVlogN$ with a co-shape matrix defined as the product of the kinship matrix $K$ and the genotypic variation $\sigma_{G,Gmax}$. 
$\beta_{TWI}$ and $\beta_{NCI}$ are fixed effects assumed constant across population (besides tested between population, see supplementary material SX). 
To estimate variances on a normal-scale, we log-transformed population fixed effect, genetic additive values, TWI and NCI fixed effects, and we calculated conditional and marginal $R^2$ [@Nakagawa2013].
We used Bayesian inference with No-U-Turn Sampler [NUTS, @Hoffman2014] using `stan` language [@Carpenter2017, see supplementary material] and previously introduced tools.
<!-- The complex geometry of the model resulted in a low Bayesian fraction of missing information (BFMI). -->
<!-- We tested the model with simulated data corresponding in 25 simulations were $\sigma^2_{R,Gmax}$ and $\sigma^2_{G,Gmax}$ varied from 0.1 to 1, which validated the robustness of the  model for a number of observations corresponding to our dataset (see supplementary material SX). -->

We also used linear mixed models [LMM, @Zhou2012a] and test of gene-ontology enrichment [@DeLaCruz2014] to scan the dataset of the 70,737 SNPs filtered for quantitative genomics for association to individual growth potential (see environmental genomics).

<!-- * Gienapp et al. 2017 Genomic quantitative genetics to study evolution in the wild -->
<!-- * Nichio and Arakawa 2019 Performance of Hamiltonian Monte Carlo and No-U-Turn Sampler for estimating genetic parameters and breeding values -->
<!-- * Muff, S., Niskanen, A. K., Saatoglu, D., Keller, L. F., & Jensen, H. (2019). Animal models with group-specific additive genetic variances: extending genetic group models. Genetics Selection Evolution, 1–16. https://doi.org/10.1186/s12711-019-0449-7 -->
<!-- * Bonnet, T., Morrissey, M. B., & Kruuk, L. E. B. (2019). Estimation of Genetic Variance in Fitness, and Inference of Adaptation, When Fitness Follows a Log-Normal Distribution. Journal of Heredity, 110(4), 393–395. https://doi.org/10.1093/jhered/esz018 -->
<!-- * Wan, Y., De Andrade, M., Yu, L., Cohen, J., & Amos, G. I. (1998). Genetic linkage analysis using lognormal variance components. Annals of Human Genetics, 62(6), 521–530. https://doi.org/10.1017/S0003480099007228 -->
<!-- * Wilson, A. J., Réale, D., Clements, M. N., Morrissey, M. M., Postma, E., Walling, C. A., … Nussey, D. H. (2010). An ecologist’s guide to the animal model. Journal of Animal Ecology, 79(1), 13–26. https://doi.org/10.1111/j.1365-2656.2009.01639.x -->

# Results

### Populations

*Symphonia globulifera* was previously recognized as composed of two morphotypes in French Guiana [@Sabatier1997]. 
But cross validation of admixture analyses for different levels of clustering K advocated for 3 different gene-pools (Fig. \@ref(fig:admixture), see supplementary material). 
The three gene-pools corresponded to the previously identified two morphotypes (70-80% of match) *S. globulifera* and *S. sp1*, with *S. globulifera* morphotype itself structured in two gene-pools. 
The two gene-pools composing the *S. globulifera* morphotype corresponded to two locally defined sub-morphotypes called *S. globulifera type Paracou* and *S. globulifera type Régina* (Saint-Omer Cazal, pers. com.). 
We conducted a second blind-identification of every individual in November 2019 to investigate mismatches between gene-pools and morphotypes and the existence of the two morphotypes *S. globulifera type Paracou* and *S. globulifera type Régina*. 
The blind-identification correctly assigned the difference between all *S. globulifera type Paracou* and *S. globulifera type Régina* and reclassified correctly 87% of previous mismatches between gene-pools and previous identification.
Consequently, we defined three populations for the subsequent analyses based on gene-pools, morphotypes and ecotypes characteristics (Fig.\@ref(fig:map) and Fig.\@ref(fig:admixture)).
*S. sp1* grows preferentially in drier plateaux and slopes (Schmitt et al., submitted) and has a light grey thin and smooth bark associated with smaller leaves, flowers and fruits  (Fig.\@ref(fig:admixture); @Sabatier1997; Schmitt et al., submitted). 
*S. globulifera type Paracou* grows in wet habitats of bottomlands, but drier than the last population, and has a dark and intermediate thin and smooth bark compared to the last population (Fig.\@ref(fig:admixture); Schmitt, pers. com.). 
*S. globulifera type Regina* grows in the wettest bottomlands of Paracou and has a thick and lashed bark (Fig.\@ref(fig:admixture); Schmitt, pers. com.).
Among the three populations, 25 individuals (7%) showed patterns of introgression, 15 between *S. globulifera type Regina* and *S. sp1* and 10 between *S. globulifera type Paracou* and *S. sp1* (confirmed with `introgress` R package [@Gompert2010], see supplementary material). 
Admixed individuals were removed from subsequent analyses.

The three populations of *Symphonia* in Paracou showed a mean nucleotidic diversity of 0.05140 across populations, and was significantly different between populations (ANOVA, p<2e-16) with *S. globulifera type Regina* being more diverse. 
The ratio of non-synonymous over synonymous SNPs was overall positive indicating putative selection, but not different between populations.
Pairwise fixation index ($F_{st}$) between populations was low with a mean value of 0.15, with *S. globulifera type Regina* being more differentiated with the two oher populations (ANOVA, p<2e-16).
Genic SNP were significantly less differentiated than anonymous SNPs (mean $F_{st}$ of 0.07 against 0.15). 
Tajima's $D$ between populations was low and significantly negative with a mean value of -0.795, and significantly different between populations (ANOVA, p<2e-16) with *S. globulifera type Regina* having a globally higher value. 
Site frequency spectrum revealed population in expansion with an excess of rare alleles in each population (see supplementary material). 
And phylogeny with `treemix` revealed a high likelihood of at least one migration event between *S. sp1* and the two *S. globulifera* populations in Paracou.
The phylogeny confirmed previously published topology [ref] with an ancestral population in Madagascar, and Sao Tome being the link between African and American populations (see supplementary material SX). 
Interestingly *S. sp1* was closer to the population of Brazil and the two *S. globulifera* populations were closer to the population of Costa Rica. 
Genic SNPs showed less drift than anonymous SNPs.

The genomic scan revealed 4,020 outliers SNPs (5.7% of SNPs) significantly associated to populations, *i.e.* fixed, with 309 genic SNPs, 2,456 putatively-hitchhiker SNPs, and 1,255 anonymous SNPs. 
Putatively-hitchhikers and genic SNPs corresponded to 43,106 and 2,981 transcripts isoforms respectively, with 121 transcripts isoforms differentially expressed between *S. globulifera* and *S. sp1* juveniles (Tysklind et la., in prep). 
Genes corresponding to previous transcripts isoforms correspond to 1,133 unique terms of gene ontology (GO) with no significantly enriched GO term in outliers genes after false discovery rate correction, besides "response to water deprivation" had a p-value of 0.01106483.

### Environments

Variance partitioning (Fig. \@ref(fig:Epart) and supplementary material) showed contrasted patterns between topographic wetness index ($TWI$) and neighbor crowding index ($NCI$). $TWI$ was mainly explained by population ($35\% \pm 4\%$) and a little by genotype ($9\% \pm 7\%$) with an important residual variation ($56\% \pm 10\%$). On the other hand, $NCI$ was largely explained by genotype ($84\% \pm 10\%$) and a little by population ($9\% \pm 3\%$) with little residual variation ($7\% \pm 7\%$).
Genomic scans did not reveal major effect SNPs.

### Growth

The growth modelling showed a high variation in individual growth potential with differences at the population level (Supplementary material).
*S. globulifera type Regina* had the higher growth potential ($7.8~mm.yr^{-1}$) and *S. sp1* had the smallest growth potential ($5.2~mm.yr^{-1}$). 
The variation in growth potential between populations corresponded to variation in maximum observed diameter in Paracou ($65.3~cm$ for *S. globulifera type Regina* against $48.9~cm$ for *S. sp1* within randomly-sampled individuals).
Both $TWI$ and $NCI$ had a significant and positive effect on individual growth with similar strength, standardized $\beta$ of 1.3172 and 1.3085 respectively for $TWI$ and $NCI$.
Environment and genetic effects explained together the majority of the total variation in individual growth potential with a residual variation of only 28% ($\pm17\%$). 
Strongest effects were genotype ($30\%\pm20\%$) and $NCI$ ($26\pm0\%$), with an important effect of $TWI$ ($14\pm0\%$) and little to no effect of population  ($1\pm1\%$). 
Nevertheless, population covary with $TWI$ as shown in environmental genomics. 
In a growth model without environmental variables (supplementary material), population effect had a variation of 10% ($\pm6\%$).
Genomic scans revealed 3 significant major effect SNP for individual growth potential (2 for *S. globulifera type Regina* and 1 for *S. sp1*, supplementary material) but no significant enrichment in gene ontology.

# Discussion

1. three populations of *Symphonia globulifera* coexist in sympatry with low level of hybridization between 2 species pairs
1. populations are adapted to topography
1. genotypes within populations are adapted to competition
1. growth is mainly driven by genotype and competition, a little by topography but not by populations
1. genotypic adaptations to competition and growth are not correlated

> Take home message ?

## Three populations of *Symphonia* adapted to topography coexist in sympatry

<!-- * Species complex & syngameons concepts & evolutionary consequences -->
<!-- * Species habtitat specialization and speciation among sister species within species complex -->
<!-- * *Symphonia* current taxonomy -->
<!-- * *Symphonia* morphotypes in Paracou -->
<!-- * *Symphonia* distribution in Paracou -->
<!-- * *Symphonia* functional traits in Paracou -->
<!-- * *Symphonia* growth patterns in Paracou -->
<!-- * Ecology on species habitat preferences, especially among congeneric and closely-related species -->
<!-- * Ecological genomics -->
<!-- * Local adaptation -->
<!-- * Ecological speciation litterature ? -->
<!-- * Species habtitat specialization and speciation among sister species within species complex -->
<!-- * *Symphonia* distribution in Paracou -->
<!-- * Arguments for distinct species advocating for broader study at regional scale -->

## Genotype but not population is a strong predictor of individual competition and growth

<!-- * Tropical tree growth -->
<!-- * Topography effects on growth -->
<!-- * Competition effects on growth -->
<!-- * Population and genetic variance on growth / growth heritability -->

## Is the association of population adaptation to topography and individual adaptation to competition and growth improving species coexistence within species complexes ? / Are genetic adaptations to niches among population and individual fitness within populations the key for species coexistence within species complexes ?

<!-- * Growth and performance link toward fitness and local adaptation / speciation (several points in one) -->
<!-- * Syngameon eco-evolutionary consequences -->
<!-- * Putative role of syngameons in Neotropical diversification -->

# Acknowledgements

We thank the University of Bordeaux for a PhD grant to Sylvain Schmitt. We are grateful to Pascal Petronelli and the CIRAD inventory team for their work on tree inventories and botanical identification. This study was partially funded by an Investissement d’Avenir grant of the ANR: CEBA (ANR-10-LABEX-0025).

# Authors’ contributions

SS and XX conceived the ideas; SS and XX designed methodology; SS and XX analysed outputs; SS and XX led the writing of the manuscript. All authors contributed critically to the drafts and gave final approval for publication.

# References

<div id="refs"></div>

\newpage 

## Table captions

<!-- __Table \@ref(tab:)__ -->

## Figure captions

__Figure \@ref(fig:map)__ Map of sampled Symphonia individuals. Every individual is represented by a pie chart representing population admixture with a radius proportional to the individual circonference in 2015. Grey level represents topographic wetness index.

__Figure \@ref(fig:admixture)__ Admixture plot of Symphonia individuals for K=3 with population distribution along topographic wetness index and associated trunk morphology. The three morphotypes are identified with their bark with *S. sp1* having a light grey thin and smooth bark, the *S. globulifera type Paracou* having a dark and intermediate thin and smooth bark compared to the thick and lashed bark of *S. globulifera type Regina*.

__Figure \@ref(fig:Epart)__ Posteriors of conditional and marginal $R^2$ (**A**) and variance partitioning (**B**) for environmental variables. Variations of individuals environments (**B**) have been partitioned into between-population (green), between-genotype (blue) and residual (grey) variation for Neighbor Crowding Index ($NCI$, top) and Topographic Wetness Index ($TWI$, bottom). Variations have been used to compute corresponding $R^2$ (**A**) for population (green), genotype (blue) and the two as marginal  $R^2$ (red). Variance partitioning is based on the median of parameters estimate, whereas circles represent the median estimate, thick lines the 50% confidence interval, and thin lines the 90% confidence interval for $R^2$.

__Figure \@ref(fig:Ppart)__ Posteriors of conditional and marginal $R^2$ (**A**) and variance partitioning (**B**) for individual maximum growth potential (Gmax). Variation of individual maximum growth (**B**) has been partitioned into between-population (orange), between-genotype (purple),  neighbor crowding (green) and topographic wetness effects (blue), and residual (grey) variation for Neighbor Crowding Index ($NCI$, top) and Topographic Wetness Index ($TWI$, bottom). Variations have been used to compute corresponding $R^2$ (**A**) between-population (orange), between-genotype (purple),  neighbor crowding (green) and topographic wetness effects (blue), and the four as marginal  $R^2$ (red). Variance partitioning is based on the median of parameters estimate, whereas circles represent the median estimate, thick lines the 50% confidence interval, and thin lines the 90% confidence interval for $R^2$.


```{r map, fig.cap="Map of sampled Symphonia individuals. Every individual is represented by a pie chart representing population admixture with a radius proportional to the individual circonference in 2015. Grey level represents topographic wetness index."}
include_graphics("figures/Fig1.png")
```

```{r admixture, fig.cap="Admixture plot of Symphonia individuals for K=3 with population distribution along topographic wetness index and associated trunk morphology. The three morphotypes are identified with their bark with *S. sp1* having a light grey thin and smooth bark, the *S. globulifera type Paracou* having a dark and intermediate thin and smooth bark compared to the thick and lashed bark of *S. globulifera type Regina*."}
include_graphics("figures/Fig2.png")
```

```{r Epart, fig.cap="Posteriors of conditional and marginal $R^2$ (**A**) and variance partitioning (**B**) for environmental variables. Variations of individuals environments (**B**) have been partitioned into between-population (orange), between-genotype (purple) and residual (grey) variation for Neighbor Crowding Index ($NCI$, top) and Topographic Wetness Index ($TWI$, bottom). Variations have been used to compute corresponding $R^2$ (**A**) for population (orange), genotype (purple) and the two as marginal  $R^2$ (red). Variance partitioning is based on the median of parameters estimate, whereas circles represent the median estimate, thick lines the 50% confidence interval, and thin lines the 90% confidence interval for $R^2$"}
fitEnv <- list(TWI = list(), NCI = list())
for(var in c("TWI", "NCI")){
  for(sim in list.files("../symcapture_save/EnvGeno", 
                        pattern = var, full.names = T)){
    load(sim)
    fitEnv[[var]] <- c(fitEnv[[var]], fit)
  }
}
fitEnv <- lapply(fitEnv, rstan::sflist2stanfit)
g1 <- lapply(fitEnv, pars = c("Vp", "Vg", "Vr"), as.data.frame) %>% 
  bind_rows(.id = "model") %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vg, Vr))) %>% 
  mutate(Vexp = sum(c(Vp, Vg))) %>% 
  mutate_at(c("Vp", "Vg", "Vexp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = "model") %>% 
  group_by(model, variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, 
                           "Vexp" = "Marginal", "Vg" = "Genotype", 
                           "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +
  theme(axis.title.y = element_blank()) +
  facet_wrap(~ model, nrow = 2) +
  scale_color_manual(guide = "none", 
                     values = RColorBrewer::brewer.pal(5, "Set1")[c(1,4:5)]) +
  coord_flip()
g2 <- lapply(fitEnv, mcmc_intervals_data, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(parameter = recode(parameter, "Vp" = "Population", "Vg" = "Genotype", "Vr" = "Residual")) %>% 
  group_by(variable) %>%
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = variable, fill = parameter)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  facet_wrap(~ variable, scales = "free", nrow = 2) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
    ylab(expression(sigma^2)) +
  scale_fill_manual("", 
                    values = c(RColorBrewer::brewer.pal(5, "Set1")[c(5,4)],
                               "grey")) 
cowplot::plot_grid(g1, g2, labels = LETTERS[1:2])
```

```{r Ppart, fig.cap="Posteriors of conditional and marginal $R^2$ (**A**) and variance partitioning (**B**) for individual maximum growth potential (Gmax). Variation of individual maximum growth (**B**) has been partitioned into between-population (orange), between-genotype (purple),  neighbor crowding (green) and topographic wetness effects (blue), and residual (grey) variation. Variations have been used to compute corresponding $R^2$ (**A**) between-population (orange), between-genotype (purple),  neighbor crowding (green) and topographic wetness effects (blue), and the four as marginal  $R^2$ (red). Variance partitioning is based on the median of parameters estimate, whereas circles represent the median estimate, thick lines the 50% confidence interval, and thin lines the 90% confidence interval for $R^2$.", fig.height=4}
fitGmaxGenoEnv <- list()
for(sim in list.files("../symcapture_save/GmaxGenoEnv", full.names = T)){
  load(sim)
  fitGmaxGenoEnv <- c(fitGmaxGenoEnv, fit)
}
fitGmaxGenoEnv <- rstan::sflist2stanfit(fitGmaxGenoEnv)
g1 <- as.data.frame(fitGmaxGenoEnv, pars = c("Vp", "Vg", "Vnci", "Vtwi", "Vr")) %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vg, Vnci, Vtwi, Vr))) %>% 
  mutate(Vexp = sum(c(Vp, Vg, Vnci, Vtwi))) %>% 
  mutate_at(c("Vp", "Vg", "Vnci", "Vtwi", "Vexp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = NULL) %>% 
  group_by(variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, 
                           "Vexp" = "Marginal", "Vtwi" = "TWI", "Vnci" = "NCI",
                           "Vg" = "Genotype", "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +
  scale_color_manual(guide = "none", values = RColorBrewer::brewer.pal(5, "Set1")) +
  theme(axis.title.y = element_blank()) +
  coord_flip()
g2 <- mcmc_intervals_data(fitGmaxGenoEnv, regex_pars = c("Vp", "Vg", "Vnci", "Vtwi", "Vr")) %>%
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", 
                           "Vnci" = "NCI", "Vtwi" = "TWI", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  ylab(expression(sigma^2)) +
  scale_fill_manual("", 
                    values = c(RColorBrewer::brewer.pal(5, "Set1")[c(5,4,3,2)],
                               "grey")) 
cowplot::plot_grid(g1, g2, labels = LETTERS[1:2])
```
