---
title: "Genetic adaptations to topography and competition are strong predictors of tree growth in a Neotropical complex"
date: '`r Sys.Date()`'
output:
  bookdown::html_document2:
    number_sections: false  
    toc: true
    toc_float: yes
  bookdown::word_document2:
    reference_docx: ./template/template.docx
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/mee.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
opts_chunk$set(echo = F, message = F, warning = F, 
               fig.height = 8, fig.width = 8,
               cache = T, cache.lazy = F)
```

1. Sylvain Schmitt
    - Corresponding author
    - sylvain.m.schmitt@gmail.com
    - orcid.org/0000-0001-7759-7106
    - UMR BIOGECO, INRA - Université de Bordeaux, 69 route d’Arcachon, F-33612 Cestas cedex France
1. Myriam Heuertz
    - heuertzm@gmail.com
    - orcid.org/0000-0002-6322-3645
    - UMR BIOGECO, INRA - Université de Bordeaux, 69 route d’Arcachon, F-33612 Cestas cedex France
1. Niklas Tysklind
    - Niklas.Tysklind@ecofog.gf
    - INRA, UMR EcoFoG (Agroparistech, CNRS, Cirad, Université des Antilles, Université de la Guyane), Campus Agronomique, 97310 Kourou, French Guiana
1. Bruno Hérault
    - bruno.herault@cirad.fr
    - orcid.org/0000-0002-6950-7286
    - Cirad, UR Forests and Societies, Univ Montpellier, Montpellier, France
    - Institut National Polytechnique Félix Houphouët-Boigny (INP-HB), BP 1093, Yamoussoukro, Ivory Coast

\newpage 

# Abstract

> To be written

## Keywords

key ;

\newpage 

# Introduction 

...  to address the following questions: 
(1) how is genetic diversity structured in a Neotropical forest?
(2) how does abiotic and biotic habitat affect population genetic diversity and structure?
(3) what are the consequences of genetic diversity and structure for individual tree growth?

# Material and Methods

## Study site

The study was conducted in the northernmost part of the Guiana Plateau region, at the Paracou field station. 
The site was characterized by an average of 3041 mm annual rainfall and a mean air temperature of 25.71 °C. 
Old tropical forest with an exceptional richness (over 750 woody species) has developed among the succession of small hills of this area, rising to 10–40 m a.s.l. [@Gourlet-Fleury2004].
The site is made of 16 permanent plots (fifteen 6.25 ha plus one 25 ha) which have been censused (DBH>10) every 1-2 years for more than 35 years. Nine of the plots were logged and subjected to human-induced disturbance in 1986.

## Plant material

402 individuals from *Symphonia globulifera* (Clusiaceae), an Amazonian hyperdominant tree species [@TerSteege2013], were sampled in 2017 during the dry season (from September to December) in Paracou.
*Symphonia globulifera* L.f (Clusiaceae) was previoulsy recognized as composed of two morphotypes in French Guiana [@Sabatier1997], *S. globulifera sensu stricto* and *Symphonia sp.1*, occurring in sympatry but in differentiated habitats, with *S. globulifera* preferentially growing in valley bottoms and *S. sp1* preferentially exploiting a variety of drier habitats [@Allie2015; Schmitt et al., in prep]. 
The genus *Symphonia* has been highlighted as a species complexe with low phylogenetic resolution and high levels of plastid DNA sharing among sister species [@baraloto_using_2012-1; @Caron2019; @Gonzalez2009; @Torroba-Balmori2017]. 
In addition, outgroup were built with 13 individuals of *Symphonia globulifera* from Africa (Sao Tome, Gabon, Cameroun, Congo, Benin, Liberia, Ivory Coast, and Gana), 7 from South America (Brazil, Costa Rica and Panama), 2 *Symphonia nectarifera* from Madagascar, 2 *Symphonia urophylla* from Madagascar, 5 *Pentadesma butyracea* from Benin and Cameroun and 1 *Pentadesma grandifolia* from Cameroun.
Leaves were collected from the 432 (402 + 30) individuals and desicated using silica gel. 

## Design of sequence capture 

We designed in *silico* a set of 20,000 80-mer probes sequences for *Symphonia globulifera* (Supplementary Material S1). We used a published low-coverage draft genome of an African *Symphonia globulifera* [@Olsson2017], an unpublished draft genome of a *Symhpnoia globulifera* from French Guiana [Scotti et al., in prep], an unpublished transcriptome from the 20 juvenile of *Symphonia globulifera* from French Guiana [Tysklind et al., in prep], and genomic reads from genome-by-sequencing data from individuals from French Guiana [Torroba-Balmori et al., unpublished]. 

We aligned genomic reads on the two genome drafts with `bwa` [@Li2009]. We then filtered scaffolds from the two genome drafts with a length superior to 1 kbp and at least one matching alignment with a read matching on a unique position, and we merged the two filtered genome drafts with `quickmerge` [@Chakraborty2016]. We aligned transcripts on the new filtered genome draft with `BLAT` [@Kent2002] and selected 533 anonymous scaffolds without any transcript matching. Finally, we masked repetitive regions with `RepeatMasker` [@Smit2015] and selected 533 1-kbp loci within the 533 selected scaffolds.

In parrallel, we filtered transcripts from the 20 juvenile of *Symphonia globulifera* from French Guiana [Tysklind et al., in prep] based on SNP quality, type and frequency. We further selected transcripts including non-overlapping open reading frame (ORFs) with a start codon (Methyonin) detected using `transdecoder` [@Haas2013]. We further filtered ORFs aligning with `BLAT` [@Kent2002] on the previous new filtered genome draft and masked repetitive regions with `RepeatMasker` [@Smit2015]. We finally selected 1,150 loci of 500-bp to 1-kbp starting 100 bp before ORFs start and a maximum of 900 bp after, resulting in 1-Mbp of transcriptomic loci including ORFs. 

## Genomic data production

Genomic DNA was extracted from 5 mg of dried leaf tissue with a CTAB protocol described by @Doyle1987 and libraries were digested with 'Ultra II FS Enzyme Mix' (new England Biolabs Inc, MA, USA) for a target size of 150 bp  and prepared using the 'NEBNext Ultra II FS DNA Library Prep kit for Illumina'(new England Biolabs Inc, MA, USA).

Libraries were amplified and tagged using 5 $\mu L$ of adaptor-ligated DNA together with 8.3 $\mu L$ of 'NEBNext Ultra II Q5 Master Mix' (new England Biolabs Inc, MA, USA), and 1.6 $\mu L$ of diluted Index Primer i5 and 1.6 $\mu L$ of diluted Index Primer i7 from 'NEBNext Multiplex Oligos for Illumina (Dual Index Primers Set 1 and Set 2)' (new England Biolabs Inc, MA, USA). Initial denaturation (98°C for 30 s) was followed by 8 cycles (98°C for 10 s and 65°C for 1 min 30 s) and a final extension (65°C for 5 min).
 
 before pooling in equimolar amounts
 
 to be captured by hybridization with the 20,000 80-mer designed probes sequences using myBaits Custom 1-20K (Arbor Biosciences, MI, USA). 



A set of 20,000 -mer probes sequences targeting XX distinct loci 

## Analysis

### Q1: Admixture, pi, ...

### Q2: Genotype by Environment association (no ANCR)

### Q3: Main focus on the growth model describing correctly and friendly the “hierarchy of models” 

### Outlier tests, GO-terms, major effects, polygenicity

# Results

# Discussion

# Acknowledgements

We thanks the University of Bordeaux for a PhD grant to Sylvain Schmitt. We are grateful to Pascal Petronelli and the CIRAD inventory team for their work on tree inventories and botanical identification.

# Authors’ contributions

SS...

# References

<div id="refs"></div>

\newpage 

## Table captions

__Table \@ref(tab:)__

## Figure captions

__Figure \@ref(fig:map)__ Map of sampled Symphonia individuals. Every individual is represented by a pie chart representing population admixture with a radius proportional to the individual circonference in 2015. Grey level represent topographic wetness index.

__Figure \@ref(fig:admixture)__ Admixture plot of Symphonia individuals for K=3 with population distribution along topographic wetness index and associated trunk morphology. The three morphotypes are identified with their bark with S. sp1 having a light grey thin and smooth bark, the S. globulifera type Paracou having a dark and intermediate thin and smooth bark compared to the thick and lashed bark of S. globulifera type Regina.

__Figure \@ref(fig:Epart)__ Variance partitioning for environmental variables. Individuals environmental variations have been partitioned into between-population, between-genotype (within-population) and residual variation for Neighbor Crowding Index (NCI) and Topographic Wetness Index (TWI).

__Figure \@ref(fig:Ppart)__ Variance partitioning for parameters of growth ontogenetic-trajectory. Individuals variations for parameters of growth ontogenetic-trajectory have been partitioned into between-population, between-genotype (within-population) and residual variation for optimal diameter (Dopt), maximum growth potential (Gmax)  and growth kurtosis (Ks).

```{r map, fig.cap="Map of sampled Symphonia individuals. Every individual is represented by a pie chart representing population admixture with a radius proportional to the individual circonference in 2015. Grey level represent topographic wetness index."}
include_graphics("figures/Fig1.png")
```

```{r admixture, fig.cap="Admixture plot of Symphonia individuals for K=3 with population distribution along topographic wetness index and associated trunk morphology. The three morphotypes are identified with their bark with S. sp1 having a light grey thin and smooth bark, the S. globulifera type Paracou having a dark and intermediate thin and smooth bark compared to the thick and lashed bark of S. globulifera type Regina."}
include_graphics("figures/Fig2.png")
```

```{r Epart, fig.cap="Variance partitioning for environmental variables. Individuals environmental variations have been partitioned into between-population, between-genotype (within-population) and residual variation for Neighbor Crowding Index (NCI) and Topographic Wetness Index (TWI)."}
include_graphics("figures/Fig3.png")
```

```{r Ppart, fig.cap="Variance partitioning for parameters of growth ontogenetic-trajectory. Individuals variations for parameters of growth ontogenetic-trajectory have been partitioned into between-population, between-genotype (within-population) and residual variation for optimal diameter (Dopt), maximum growth potential (Gmax)  and growth kurtosis (Ks)."}
include_graphics("figures/Fig4.png")
```

