---
title: An eco-evolutionary perspective on local coexistence within Neotropical species complexes
date: '`r Sys.Date()`'
output:
  bookdown::word_document2:
    reference_docx: ./template/template.docx
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
  bookdown::html_document2:
    number_sections: false  
    toc: true
    toc_float: yes
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/mee.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

# Outlines 1

Biodiversity maintenance is a long standing issue in both ecology [@Hutchinson1941] and evolution [@Darwin1909]. Species coexistence in hyperdiverse ecosystems, such as tropical forests and coral reefs [@connell_diversity_1978], remains partly unexplained and subject to debate [@wright2002plant]. However, biodiversity is in crisis and increasing human-induced disturbances threaten many species worldwide [@Cardinale2012]. Consequently, there is an increased need to understand and predict biodiversity dynamic in response to global changes.

Prediction of biodiviersity dynamic and lost rest entirely on ecophysiology, phenology, and species distribution [@Aleixo2019]. Nevertheless, eco-evolutionary processes have been documented for a while [@Tutt1896] and can play an important role on the dynamic of both species and communities [@Bailey2009]. Assumption that evolution and ecology play at very different time-scales might have been an impediment to understand the role of eco-evolutionary processes in biodiviersity maintenance [@Pelletier2009]. Evidences exist for eco-evolutionary processes at microgeographic scale, e.g. within the dispersal neighborhood of the organism [@Richardson2014].

Local eco-evolutionary processes are biologically organized among genes, individuals and populations in the community over different time-scales [@Morris2011]. Sampling all individuals within the community is often unrealistic. Thus, we need to adopt either a top-down approach, with numerous populations including few individuals [@Caron2019], or a bottom-up approach with numerous individuals among few populations. In particular, communities include species complexes composed of closely-related species sharing large amount of genetic variation due to a recent common ancestor and/or hybridization [@Pernes1984]. Species complexes can lead to adaptive radiation and species seggregation along environmental gradients [@Seehausen2004]. Consequently, species complexes are a good model to investigate the role of eco-evolutionary processes in individual and populations local coexistence [@Pinheiro2018].

Individual fitness arise from local gene-environment interactions. Individual fitness thus links ecology and evolution at the individual scale. Individual fitness is determined by individual ability to grow, survive and reproduce in their environment. And evolution ultimately increases individual fitness in their environment with selection on genetic diversity. Individuals reproducing preferentially with each other form local populations [@Mayr1996]. The sum of individual fitness within the population determine population dynamic with the local environment and ultimately the population ecological niche. And genetic structure fix genetic adaptation conferring population adaptation to its ecological niche.

In forest, trees interact with their environment through water and nutrient uptake and light access in order to grow and survive. Water, nutrient and light availability are determined both by abiotic environment and biotic interactions. Topography is an important driver of water and nutrient availability [@Ferry2010], and drives tropical tree species distribution [@Allie2015]. Neighboring trees modulate trees access to light, nutrient and water in a given abiotic environment. Thus competition from neighboring trees is a major biotic interactions driving tree growth and surival [@Uriarte2004a].

Topography drives the spatial genetic structure and microhabitat adaptations in tropical tree species [@Torroba-Balmori2017; Tysklind et al., in prep]. Less is know about genetic adaptation to competition in topical tree species, despite modelling approach suggested it may play an important role in the genetic diversity of tree populations [@CostaeSilva2013]. Indeed, competiton plays a major role in tree ecology [@VanBreugel2012] but the strong dynamic of competition vary from one generation to the next and even over an individual's lifetime. Moreover, woody species harbour more genetic variation within populations but less among populations than non woody species [@Hamrick1992b]. Finally, genetic diveristy and structure effect on growth is important but remains unexplored in non-model organisms and natural populations such as trees from Neotropical forests [@Gonzalez-Martinez2006; @Grattapaglia2009].

In the present study, we assessed genetic diversity and structure among trees of two Neotropical species complexes, and addressed their relation with abiotic environment and competition and their joint effect on individual growth. We hypothesized both abiotic and biotic environmental heterogeneity to be major factors of genetic diveristy and structure within and among populations. We hypothesized topography to determine populations ecological niches through fixed adaptation in the genetic structure, due to evidences of topography determining populations ecological niches [Schmitt et al., in prep]. But interspecific gene flow within the species complex could reduce species genetic adaptations to topography [@Tigano2016]. Similarly, we hypothesized competition to drive genetic diversity due to its strong impact on tree ecology [@VanBreugel2012]. But the short term dynamic of competition could favor plasticity over genetic adaptations [@BenitoGarzon2019].

*Symphonia globulifera* was locally composed of three sympatric populations corresponding to three different morphotypes. Abiotic environment, through topography and wetness, drove individual survival among populations ($TWI$,  $R_m^2=0.35$) resulting in intermediate genetic differentiation ($F_{ST}=0.15$) with fixed markers among populations (5.7% of SNPs) and an enrichment of gene related to "response to water deprivation" ($p=0.01106483$). Individuals within population were adapted to a trade-off (Pearson $R=-0.3$, $p=4.5 10^{-9}$) between genotypes with high growth rate associated with survival under low competition and genotypes with low growth rate associated with survival under high competition. Genetic diversity thus explained individual tolerance to neighbourhood crowding ($NCI$, $R_m^2-R_c^2=0.18$) and individual growth potential ($Gmax$, $R_m^2-R_c^2=0.43$).  

The interaction between abiotic environment and genes shapes the microgeogrpahic genetic structure of the three sympatric populations of *Symphonia globulifera*. Population of *Symphonia globulifera* are locally adapted to fine-scale topography and wetness, along a gradient opposing flooded botomlands to drier slopes and plateaus. Adaptation to topography allow populations local coexistence within the species complex besides the inter-populations gene flow. On the other hand, the interaction between competition and genes shapes genetic diversity of individuals within populations. Resulting genotypic adaptation opposed "opportunist" strategy, with fast growth under low competition, to "struggler" strategy, with slow growth under high competition, among individuals within population. The genotypic trade-off between "opportunist" and "struggler" allow individual to coexist locally and fully occupy the abiotic niche from their population. The abiotic environment is stable over several generations, while competition varies from one generation to the next and even over an individual's lifetime. Genetic diversity within populations allows thus to unfold rapid response to competition dynamic through genotypic adaptations to growth and competitive ability; whereas genetic structure fix adaptations to the stable abiotic environment. Consequently, a fine-scale eco-evolutionary dynamics drives the coexistence of individuals and populations within Neotropical species complexes.

# Figures

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
library(tidyverse)
library(bayesplot)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, 
               fig.height = 8, fig.width = 8,
               cache = T, cache.lazy = F)
```

```{r map, fig.cap="Map of sampled Symphonia individuals. Every individual is represented by a pie chart representing population admixture with a radius proportional to the individual circonference in 2015. Grey level represents topographic wetness index."}
include_graphics("figures/Fig1.png")
```

```{r admixture, fig.cap="Admixture plot of Symphonia individuals for K=3 with population distribution along topographic wetness index and associated trunk morphology. The three morphotypes are identified with their bark with *S. sp1* having a light grey thin and smooth bark, the *S. globulifera type Paracou* having a dark and intermediate thin and smooth bark compared to the thick and lashed bark of *S. globulifera type Regina*."}
include_graphics("figures/Fig2.png")
```

```{r rel, fig.cap="Relations between explained variables and corresponding genetic multiplicative values. Subplots in the diagonal (**A**,**E**,**I**) represent the relation between the multiplicative genetic values on the x-axis and  the explained variable on the y-axis. Subplots above the diagonal (**B**, **C**, **F**) represent the pairwise relations among the multiplicative genetic values. Subplots below the diagonal (**D**, **G**, **H**) represent the pairwise relations among the explained variables. In the row and column orders, explained variables and multiplicative genetic values correspond to topographic wetness index ($TWI$), neighbor crowding index ($NCI$) mean annual growth rate ($AGR$) and growth potential ($Gmax$). Dote size indicates individual diameter at breast height. Red dot represents *S. sp1*, light blue dots *S. globulifera type Paracou* and dark blue dots *S. globulifera type Regina*. Annotations in the diagonal subplots indicate the marginal and the conditional goodness-of-fit [@Nakagawa2013], respectively $R_m^2$ and $R_c^2$, of corresponding genetic models. Annotations in the subplots above the diagonal indicate the Pearson's $R$ correlation coefficient and associated $p$-value between the pair of multiplicative genetic values, also indicated by the regression line. Grey annotation and regression line indicate a non-significant correlation; whereas black annotation and regression line indicate a significant correlation. Grey areas in the subplots below the diagonal represent space of explained variables for every tree in the plot 16 from Paracou.", fig.height=8, fig.width=8}
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
load(file = "../symcapture_save/Env2.Rdata") 
trees <- trees %>% 
  na.omit(pop) %>% # remove admixed
  mutate(IndNum = 1:nrow(.), popNum = as.numeric(as.factor(pop)))
fitGmaxGeno <- list()
for(sim in list.files("../symcapture_save/GmaxGeno", full.names = T)){
  load(sim)
  fitGmaxGeno <- c(fitGmaxGeno, fit)
}
fitGmaxGeno <- rstan::sflist2stanfit(fitGmaxGeno)
fitNCI <- list()
for(sim in list.files("../symcapture_save/EnvGeno", 
                      pattern = "NCI", full.names = T)){
  load(sim)
  fitNCI <- c(fitNCI, fit)
}
fitNCI <- rstan::sflist2stanfit(fitNCI)
fitTWI <- list()
for(sim in list.files("../symcapture_save/EnvGeno", 
                      pattern = "TWI", full.names = T)){
  load(sim)
  fitTWI <- c(fitTWI, fit)
}
fitTWI <- rstan::sflist2stanfit(fitTWI)
t <- lapply(c(fitNCI, fitTWI, fitGmaxGeno), function(fit)
  as.data.frame(fit, "alog") %>% 
  summarise_all(median) %>% 
  reshape2::melt(NULL)) %>% 
  bind_rows(.id = "fit") %>% 
  mutate(fit = recode(fit, "1" = "aNCI", "2" = "aTWI", "3" = "aGmax")) %>% 
  separate(variable, c("X1", "IndNum", "X2"), convert = T) %>% 
  mutate(value = exp(value)) %>% 
  reshape2::dcast(IndNum ~ fit, value.var = "value") %>% 
    left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "gemma", "gmax", "gmax.phenotype"),
                     col_names = c("FID", "IID", "Gmax")))
## Up
# ggplot(filter(t, aNCI > 0.9), aes(aGmax, aNCI, size = DBHtoday)) + # there is no outliers driving the relation
g.a.NCI.Gmax <- ggplot(t, aes(aGmax, aNCI, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "loess", col = "black") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank()) +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = 0.85)
g.a.TWI.Gmax <- ggplot(t, aes(aGmax, aTWI, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "loess", col = "darkgrey") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank()) +  ggtitle("Gmax") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = 0.8, col = "darkgrey")
g.a.TWI.NCI <- ggplot(t, aes(aNCI, aTWI, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "loess", col = "darkgrey") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank()) +  ggtitle("NCI") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.8, label.y = 0.85, col = "darkgrey")
## Diag
g.TWI <- ggplot(t, aes(aTWI, TWI, col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  theme(axis.title.x = element_blank()) +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  ylab("TWI") + ggtitle("TWI") +
  annotate("text", x = 0.9, y = 7, label = expression(R[m]^2==0.35)) +
  annotate("text", x = 0.9, y = 6, label = expression(R[c]^2==0.44))
g.NCI <- ggplot(t, aes(aNCI, NCI/10^4, col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
    theme(axis.title = element_blank()) +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
    annotate("text", x = 0.9, y = 0.7, label = expression(R[m]^2==0.06)) +
  annotate("text", x = 0.9, y = 0.6, label = expression(R[c]^2==0.24))
g.gmax <- ggplot(t, aes(aGmax, (DBHtoday - DBH0)/(2017-Y0), 
                        col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
    theme(axis.title = element_blank()) +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
      annotate("text", x = 0.5, y = 1.1, label = expression(R[m]^2==0.10)) +
  annotate("text", x = 0.5, y = 0.9, label = expression(R[c]^2==0.53))
## Down
load(file = "../symcapture_save/p16.Rdata")
g.var.DBH.NCI <- ggplot(t, aes(NCI/10^4, (DBHtoday - DBH0)/(2017-Y0), 
                               size = DBHtoday)) +
  geom_hex(data = p16, fill = "lightgrey", alpha = 0.6, bins = 50) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank())
g.var.DBH.TWI <- ggplot(t, aes(TWI, (DBHtoday - DBH0)/(2017-Y0), 
                        size = DBHtoday)) +
  geom_hex(data = p16, fill = "lightgrey", alpha = 0.6) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title.x = element_blank()) +
  ylab("AGR")
  # ylab(expression(frac(DBH[2017]-DBH[Y0],2017-Y0)))
g.var.NCI.TWI <- ggplot(t, aes(TWI, NCI/10^4, size = DBHtoday)) +
  geom_hex(data = p16, fill = "lightgrey", alpha = 0.6) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title.x = element_blank()) +
  ylab("NCI")
# all
cowplot::plot_grid(g.TWI, g.a.TWI.NCI, g.a.TWI.Gmax,
                   g.var.NCI.TWI, g.NCI, g.a.NCI.Gmax,
                   g.var.DBH.TWI, g.var.DBH.NCI, g.gmax,
                   ncol=3, labels = LETTERS[1:9])
rm(list = ls()); invisible(gc())
```

```{r Ppart, fig.cap="Posteriors of conditional and marginal $R^2$ (**A**) and variance partitioning (**B**) for individual maximum growth potential (Gmax). Variation of individual maximum growth (**B**) has been partitioned into between-population (orange), between-genotype (purple),  neighbor crowding (green) and topographic wetness effects (blue), and residual (grey) variation. Variations have been used to compute corresponding $R^2$ (**A**) between-population (orange), between-genotype (purple),  neighbor crowding (green) and topographic wetness effects (blue), and the four as marginal  $R^2$ (red). Variance partitioning is based on the median of parameters estimate, whereas circles represent the median estimate, thick lines the 50% confidence interval, and thin lines the 90% confidence interval for $R^2$.", fig.height=4}
fitGmaxGenoEnv <- list()
for(sim in list.files("../symcapture_save/GmaxGenoEnv", full.names = T)){
  load(sim)
  fitGmaxGenoEnv <- c(fitGmaxGenoEnv, fit)
}
fitGmaxGenoEnv <- rstan::sflist2stanfit(fitGmaxGenoEnv)
g1 <- as.data.frame(fitGmaxGenoEnv, pars = c("Vp", "Vg", "Vnci", "Vtwi", "Vr")) %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vg, Vnci, Vtwi, Vr))) %>% 
  mutate(Vexp = sum(c(Vp, Vg, Vnci, Vtwi))) %>% 
  mutate_at(c("Vp", "Vg", "Vnci", "Vtwi", "Vexp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = NULL) %>% 
  group_by(variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, 
                           "Vexp" = "Marginal", "Vtwi" = "TWI", "Vnci" = "NCI",
                           "Vg" = "Genotype", "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +
  scale_color_manual(guide = "none", values = RColorBrewer::brewer.pal(5, "Set1")) +
  theme(axis.title.y = element_blank()) +
  coord_flip()
g2 <- mcmc_intervals_data(fitGmaxGenoEnv, regex_pars = c("Vp", "Vg", "Vnci", "Vtwi", "Vr")) %>%
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", 
                           "Vnci" = "NCI", "Vtwi" = "TWI", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  ylab(expression(sigma^2)) +
  scale_fill_manual("", 
                    values = c(RColorBrewer::brewer.pal(5, "Set1")[c(5,4,3,2)],
                               "grey")) 
cowplot::plot_grid(g1, g2, labels = LETTERS[1:2])
rm(list = ls()); invisible(gc())
```

# References

