---
title: An eco-evolutionary perspective on local coexistence within Neotropical species complexes
date: '`r Sys.Date()`'
output:
  bookdown::html_document2:
    number_sections: false  
    toc: true
    toc_float: yes
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
  bookdown::word_document2:
    reference_docx: ./template/template.docx
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/mee.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

# Take-home message

Genotypic adaptations within species decrease the risk of a stochastic local extinction due to wide successional niches to respond to fine spatio-temporal dynamics of forest gaps; while species adaptations stabilize local coexistence with differentiated species’s topographic niches within species complexes.

# Outlines

Tropical forests tremendous biodiversity [@Gaston2000] have intrigued biologists for decades [@connell_diversity_1978]. 
Lowland Amazonia alone is estimated to shelter around sixteen thousand tree species [@TerSteege2013]. 
But the five thousand species of trees described in Amazonia belong to only height hundred and ten genera [@TerSteege2013], including thus many species rich genera. 
Even at the hectare-scale, tropical forests shelter up to several hundreds tree species [@Gentry1988] including species rich genera with closely-related species coexisting in sympatry [@Caron2019]. 
Species-rich genera have a higher level of genetic polymorphisms [@Caron2019] and hybridize more often between congeners [@Whitney2010] than species-poor genera.
Species-rich genera may thus result from species complexes, composed of morphologically similar species and/or species sharing large amounts of genetic variation due to recent common ancestry and/or hybridization [@Pernes1984].
Specifically, when species are connected by limited, but recurrent, interspecific gene flow, they form a syngameon [@Suarez-Gonzalez2018].
However, closely-related species are expected to share similar niche and functional strategies due to phylogenetic constraints [@Wiens2010], especially with the low genetic differentiation found within species complexes.
Niche and functional similarities is expected to lead to increase competition between closely-related species and eventually to local competitive exclusion of one species by the other. 
Despite the local abundance and regional success of closely-related species growing in sympatry within species complexes [@Gentry1988; @TerSteege2013; @Pinheiro2018], little is known of eco-evolutionary forces driving the local coexistence within Neotropical species complexes.

Niche theory explains species local coexistence based on ecological niche differences limiting competitive exclusion [@lortie_rethinking_2004-1; @weiher_assembly_1995].
The heterogeneity of ressources distribution in space and time contribute to define fine-scale habitat where species can coexist [@lortie_rethinking_2004-1; @weiher_assembly_1995]. 
For instance, topography is spatially driving water and nutrient distribution in tropical forest [@Ferry2010], leading to pervasive habitat differentiation among species [@Allie2015; Schmitt et al., in prep].
Additionally, forest gap dynamics is a stong driver of above- and below-ground competition for ressources in space and time [@Hubbell1999; @Molino2001].
During succession, pioneer species grow first in light-gaps, whereas late-successional species grow later under closed-canopy [@Craven2015].
Oppositely, neutral theory explains the local coexistence of functionally equivalent species through stochastic life, death, reproduction and dispersal dynamics [@Hubbell2001].
Despite the role of intraspecific variability in community [@Messier2010; @Siefert2015a], niche and neutral theories often ignores individual diversity within species' local populations [@chave_neutral_2004].
Individuals within species' local populations varies functionaly with the fine-scale habitat variations [@Umana2019; Schmitt et al., in prep], allowing the species to exploit a wider diversity of ecological niches. 

But "nothing in biology makes sense except in the light of evolution" [@Dobzhansky1973].
The genetic proximity of species within species complexes question the role taken by evolution in their local coexistence.
The assumption that evolution and ecology play at very different time-scales might have been an impediment to understand the role of eco-evolutionary processes in species coexistence [@Pelletier2009]. 
Despite, eco-evolutionary processes have been documented for a while [@Tutt1896] and can play an important role on the dynamic of both species and communities [@Bailey2009]. 
Evidences exist for eco-evolutionary processes at microgeographic scale, e.g. within the dispersal neighborhood of the organism [@Richardson2014].

Closely-related species growing in sympatry in differentiated ecological niches form an adaptive radiation, such as Darwin's finches in the Galapagos [@Seehausen2004]. 
Evolutive history behind adaptive radiations fall within a continuum from sympatric ecological speciation to seconday contacts of species ecologically specialised in allopatry or parapatry [@rundell_adaptive_2009].
Few studies evidenced adaptive radiation driven by topography or other abiotic habitats for tropical trees [@Paun2016; @Pillon2014].
Despite the fact that sympatric species grow differently depending on access to light [@Yamasaki2013], to our knowledge, no studies have demonstrated adaptive radiation for tropical trees along forest successional gradients.

Moreover, ecological preferences of populations vary even within species through local adaptation [@Lascoux2016; @Savolainen2013]. 
Local adaptation can even occur at microgeographic scale [@Richardson2014] and under gene-flow [@Tigano2016], further termed as microgeographic adaptation to avoid ambiguity. 
Microgeographic adaptation and phenotypic plasiticity are the local drivers of intraspecific variability [@BenitoGarzon2019].
Topography and abiotic habitat have been evidenced to promote intraspecies divergences and microgeographic adaptations in Neotropical tree species [@Brousseau2015; @Team2013]. 
Despite the strong impact of competition and access to light on tree ecology [@VanBreugel2012], to our knowledge, no studies have explored the possible role of competition on microgeographic adaptation within populations.

In the present study, we assessed species delineation and genotypic diversity of closely-related sympatric tree species belonging to two Neotropical species complexes.
We addressed species delineation and microgeographic adaptation along topographic and competition gradients and with individual growth. 
Combining tree inventories, LiDAR-derived topographic data, and single nucleotide polymorphisms (SNPs), we used population genomics, environmental association analyses, genome wide association studies and growth modelling to adress the following questions:

1. How is genetic diversity structured among and within species of Neotropical species complexes ?
1. Are species delineated and are individuals adapted to the topographic and/or competitive gradients?
1. What are the consequences of species delineation and mircogeographic adaptation on individual growth along topographic and competitive gradients ?

We hypothesized species to be delineated into genepools corresonding to already described taxonomic morphotypes but to include interspecific gene flow and misidentifications due to blur morphological differences [@Pinheiro2018].
But molecular data could reveal cryptic species or merge taxon  [@Ewedje2020].
We hypothesized species to be delineated along topographic but not competitive gradients, with fixed neutral and adaptive variants, following previous evidences of topography determining species' ecological niches [Schmitt et al., in prep].
But interspecific gene flow within the species complex could reduce species adaptations to topography and blur evolutive history [@Tigano2016], while competition could also fix adaptive variants among species [@Yamasaki2013]. 
Similarly, we hypothesized competition to drive microgeographic adaptive evolution due to the strong impact of competition on tree growth and survival [@VanBreugel2012]. 
But the short term dynamics of competition could favor plasticity over genetic adaptations [@BenitoGarzon2019].
Finally, we hypothesized individual growth to differ mainly between delineated species in relation to topography following evidences of differences between species of potential growth [@Herault2011] and ecological niches [Schmitt et al., in prep]. 
But microgeographic adaptations wihtin species and competition could be the main driver behind the pattern observed among species at the community level [@Read2016].

We sampled leaf from 402 individuals previously morpholigcally identified as *Symphonia globulifera* and *Symphonia sp. 1* and 432 individuals belonging to the clade *Parvifolia* of the genus *Eschweilera* in the Paracou field station, where individuals have been censused for more than 30 years. 
We designed and conducted a targeted sequence capture of genomic regions for each individual. 
We called, filtered and annotated single nucleotide polymorphisms (SNPs) of sequenced libraries against the references used to build the targeted sequences. 
We analyzed population genetic structure, genetic diversity, and demographic history. 
We then measured species and genotypes associtation to environmental variables and individual growth potential with Bayesian inferences of Animal and growth models. 
Finally, we scanned genomes for SNPs association to population structure, environmental variables and individual growth potential, and tested the functional enrichment of detected outliers. 

*Symphonia globulifera* was locally composed of three sympatric species corresponding to three different morphotypes. 
The spatial distribution of species suggested that abiotic environment, through topography and wetness, drove individual survival among populations ($TWI$, $R_m^2=0.35$) resulting in intermediate genetic differentiation ($F_{ST}=0.15$) with fixed markers among populations (5.7% of SNPs) and an enrichment of genes related to “response to water deprivation” ($p=0.01106483$) revealing adaptive markers among the markers fixed among species. 
Within species, genotypes explained a significant part of individual tolerance to neighbourhood crowding ($NCI$, $R_m^2−R_c^2=0.18$) and individual growth potential ($Gmax$, $R_m^2−R_c^2=0.43$), spatially structured up to 47m (Moran's~I, $p-value < 0.01$). 
Moreover, a negative correlation  (Pearson $R=−0.3$, $p=4.5.10^{−9}$) was observed between genotypic multiplicative values of individual tolerance to neighbor crowding and individual growth potential. 
The negative correlation evidenced a trade-off between genotypes adapted to high growth rate and survival under low competition opposed to genotypes adapted to low growth rate and survival under high competition. 

Within forest tree species, individual tree genotypes are differentially adapted to regenerate and thrive in response to the fine spatio-temporal dynamic of forest gaps. A genetic trade-off exists between intra-specific adaptive strategies for competition vs. for growth. Therefore, species can grow in a diversity of competition environments, i.e., competition niches, thanks to these genotypic adaptations to forest gap dynamics. Within tree species complexes, closely-related species have different realized optima for topographic niches with species genetic adaptations. Consequently, genotypic adaptations within species decrease the risk of a stochastic local extinction while species adaptations stabilize local coexistence.

# Figures

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
library(tidyverse)
library(bayesplot)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, 
               fig.height = 8, fig.width = 8,
               cache = T, cache.lazy = F)
```

```{r map, fig.cap="Map of sampled Symphonia individuals. Every individual is represented by a pie chart representing population admixture with a radius proportional to the individual circonference in 2015. Grey level represents topographic wetness index."}
include_graphics("figures/Fig1.png")
```

```{r admixture, fig.cap="Admixture plot of Symphonia individuals for K=3 with population distribution along topographic wetness index and associated trunk morphology. The three morphotypes are identified with their bark with *S. sp1* having a light grey thin and smooth bark, the *S. globulifera type Paracou* having a dark and intermediate thin and smooth bark compared to the thick and lashed bark of *S. globulifera type Regina*."}
include_graphics("figures/Fig2.png")
```

```{r rel, fig.cap="Relations between explained variables and corresponding genetic multiplicative values. Subplots in the diagonal (**A**,**E**,**I**) represent the relation between the multiplicative genetic values on the x-axis and  the explained variable on the y-axis. Subplots above the diagonal (**B**, **C**, **F**) represent the pairwise relations among the multiplicative genetic values. Subplots below the diagonal (**D**, **G**, **H**) represent the pairwise relations among the explained variables. In the row and column orders, explained variables and multiplicative genetic values correspond to topographic wetness index ($TWI$), neighbor crowding index ($NCI$) mean annual growth rate ($AGR$) and growth potential ($Gmax$). Dote size indicates individual diameter at breast height. Red dot represents *S. sp1*, light blue dots *S. globulifera type Paracou* and dark blue dots *S. globulifera type Regina*. Annotations in the diagonal subplots indicate the marginal and the conditional goodness-of-fit [@Nakagawa2013], respectively $R_m^2$ and $R_c^2$, of corresponding genetic models. Annotations in the subplots above the diagonal indicate the Pearson's $R$ correlation coefficient and associated $p$-value between the pair of multiplicative genetic values, also indicated by the regression line. Grey annotation and regression line indicate a non-significant correlation; whereas black annotation and regression line indicate a significant correlation. Grey areas in the subplots below the diagonal represent space of explained variables for every tree in the plot 16 from Paracou.", fig.height=8, fig.width=8}
path <- "~/Documents/BIOGECO/PhD/data/Symphonia_Paracou/Sequences/populationGenomics/"
load(file = "../symcapture_save/Env2.Rdata") 
trees <- trees %>% 
  na.omit(pop) %>% # remove admixed
  mutate(IndNum = 1:nrow(.), popNum = as.numeric(as.factor(pop)))
fitGmaxGeno <- list()
for(sim in list.files("../symcapture_save/GmaxGeno", full.names = T)){
  load(sim)
  fitGmaxGeno <- c(fitGmaxGeno, fit)
}
fitGmaxGeno <- rstan::sflist2stanfit(fitGmaxGeno)
fitNCI <- list()
for(sim in list.files("../symcapture_save/EnvGeno", 
                      pattern = "NCI", full.names = T)){
  load(sim)
  fitNCI <- c(fitNCI, fit)
}
fitNCI <- rstan::sflist2stanfit(fitNCI)
fitTWI <- list()
for(sim in list.files("../symcapture_save/EnvGeno", 
                      pattern = "TWI", full.names = T)){
  load(sim)
  fitTWI <- c(fitTWI, fit)
}
fitTWI <- rstan::sflist2stanfit(fitTWI)
t <- lapply(c(fitNCI, fitTWI, fitGmaxGeno), function(fit)
  as.data.frame(fit, "alog") %>% 
  summarise_all(median) %>% 
  reshape2::melt(NULL)) %>% 
  bind_rows(.id = "fit") %>% 
  mutate(fit = recode(fit, "1" = "aNCI", "2" = "aTWI", "3" = "aGmax")) %>% 
  separate(variable, c("X1", "IndNum", "X2"), convert = T) %>% 
  mutate(value = exp(value)) %>% 
  reshape2::dcast(IndNum ~ fit, value.var = "value") %>% 
    left_join(trees) %>% 
  left_join(read_tsv(file.path(path, "gemma", "gmax", "gmax.phenotype"),
                     col_names = c("FID", "IID", "Gmax")))
## Up
# ggplot(filter(t, aNCI > 0.9), aes(aGmax, aNCI, size = DBHtoday)) + # there is no outliers driving the relation
g.a.NCI.Gmax <- ggplot(t, aes(aGmax, aNCI, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "loess", col = "black") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank()) +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = 0.85)
g.a.TWI.Gmax <- ggplot(t, aes(aGmax, aTWI, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "loess", col = "darkgrey") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank()) +  ggtitle("Gmax") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = 0.8, col = "darkgrey")
g.a.TWI.NCI <- ggplot(t, aes(aNCI, aTWI, size = DBHtoday)) +
  geom_hline(yintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_smooth(method = "loess", col = "darkgrey") +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank()) +  ggtitle("NCI") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.8, label.y = 0.85, col = "darkgrey")
## Diag
g.TWI <- ggplot(t, aes(aTWI, TWI, col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  theme(axis.title.x = element_blank()) +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  ylab("TWI") + ggtitle("TWI") +
  annotate("text", x = 0.9, y = 7, label = expression(R[m]^2==0.35)) +
  annotate("text", x = 0.9, y = 6, label = expression(R[c]^2==0.44))
g.NCI <- ggplot(t, aes(aNCI, NCI/10^4, col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
    theme(axis.title = element_blank()) +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
    annotate("text", x = 0.9, y = 0.7, label = expression(R[m]^2==0.06)) +
  annotate("text", x = 0.9, y = 0.6, label = expression(R[c]^2==0.24))
g.gmax <- ggplot(t, aes(aGmax, (DBHtoday - DBH0)/(2017-Y0), 
                        col = pop, size = DBHtoday)) +
  geom_vline(xintercept = 1, linetype = "dashed", col = "lightgrey") +
  geom_point(alpha = 0.4) +
  scale_size_continuous(guide = "none") +
    theme(axis.title = element_blank()) +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
      annotate("text", x = 0.5, y = 1.1, label = expression(R[m]^2==0.10)) +
  annotate("text", x = 0.5, y = 0.9, label = expression(R[c]^2==0.53))
## Down
load(file = "../symcapture_save/p16.Rdata")
g.var.DBH.NCI <- ggplot(t, aes(NCI/10^4, (DBHtoday - DBH0)/(2017-Y0), 
                               size = DBHtoday)) +
  geom_hex(data = p16, fill = "lightgrey", alpha = 0.6, bins = 50) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title = element_blank())
g.var.DBH.TWI <- ggplot(t, aes(TWI, (DBHtoday - DBH0)/(2017-Y0), 
                        size = DBHtoday)) +
  geom_hex(data = p16, fill = "lightgrey", alpha = 0.6) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title.x = element_blank()) +
  ylab("AGR")
  # ylab(expression(frac(DBH[2017]-DBH[Y0],2017-Y0)))
g.var.NCI.TWI <- ggplot(t, aes(TWI, NCI/10^4, size = DBHtoday)) +
  geom_hex(data = p16, fill = "lightgrey", alpha = 0.6) +
  geom_point(aes(col = pop), alpha = 0.4) +
  scale_size_continuous(guide = "none") +
  scale_color_manual(guide = "none", values = c("#1E90FF", "#0000EE", "#CD2626")) +
  theme(axis.title.x = element_blank()) +
  ylab("NCI")
# all
cowplot::plot_grid(g.TWI, g.a.TWI.NCI, g.a.TWI.Gmax,
                   g.var.NCI.TWI, g.NCI, g.a.NCI.Gmax,
                   g.var.DBH.TWI, g.var.DBH.NCI, g.gmax,
                   ncol=3, labels = LETTERS[1:9])
rm(list = ls()); invisible(gc())
```

```{r Ppart, fig.cap="Posteriors of conditional and marginal $R^2$ (**A**) and variance partitioning (**B**) for individual maximum growth potential (Gmax). Variation of individual maximum growth (**B**) has been partitioned into between-population (orange), between-genotype (purple),  neighbor crowding (green) and topographic wetness effects (blue), and residual (grey) variation. Variations have been used to compute corresponding $R^2$ (**A**) between-population (orange), between-genotype (purple),  neighbor crowding (green) and topographic wetness effects (blue), and the four as marginal  $R^2$ (red). Variance partitioning is based on the median of parameters estimate, whereas circles represent the median estimate, thick lines the 50% confidence interval, and thin lines the 90% confidence interval for $R^2$.", fig.height=4}
fitGmaxGenoEnv <- list()
for(sim in list.files("../symcapture_save/GmaxGenoEnv", full.names = T)){
  load(sim)
  fitGmaxGenoEnv <- c(fitGmaxGenoEnv, fit)
}
fitGmaxGenoEnv <- rstan::sflist2stanfit(fitGmaxGenoEnv)
g1 <- as.data.frame(fitGmaxGenoEnv, pars = c("Vp", "Vg", "Vnci", "Vtwi", "Vr")) %>% 
  rowwise() %>% 
  mutate(Vtot = sum(c(Vp, Vg, Vnci, Vtwi, Vr))) %>% 
  mutate(Vexp = sum(c(Vp, Vg, Vnci, Vtwi))) %>% 
  mutate_at(c("Vp", "Vg", "Vnci", "Vtwi", "Vexp"), funs(./Vtot)) %>% 
  dplyr::select(-Vtot, -Vr) %>% 
  reshape2::melt(id.vars = NULL) %>% 
  group_by(variable) %>% 
  summarise(q5 = quantile(value, 0.05),
            q25 = quantile(value, 0.25),
            mean = mean(value),
            median = median(value),
            sd = sd(value),
            q75 = quantile(value, 0.75),
            q95 = quantile(value, 0.95)) %>% 
    mutate(variable = recode_factor(variable, 
                           "Vexp" = "Marginal", "Vtwi" = "TWI", "Vnci" = "NCI",
                           "Vg" = "Genotype", "Vp" = "Population")) %>% 
  ggplot(aes(x = variable, xend = variable, col = variable)) +
  geom_point(aes(y = median), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = q5, yend = q95),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = q25, yend = q75), size = 2, alpha = 0.5) +
  ylab(expression(R^2)) +
  scale_color_manual(guide = "none", values = RColorBrewer::brewer.pal(5, "Set1")) +
  theme(axis.title.y = element_blank()) +
  coord_flip()
g2 <- mcmc_intervals_data(fitGmaxGenoEnv, regex_pars = c("Vp", "Vg", "Vnci", "Vtwi", "Vr")) %>%
  mutate(variance = recode_factor(parameter, 
                           "Vp" = "Population", "Vg" = "Genotype", 
                           "Vnci" = "NCI", "Vtwi" = "TWI", "Vr" = "Residual")) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  ggplot(aes(x = "Gmax", fill = variance)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", 
            position = position_stack(vjust = .5)) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  ylab(expression(sigma^2)) +
  scale_fill_manual("", 
                    values = c(RColorBrewer::brewer.pal(5, "Set1")[c(5,4,3,2)],
                               "grey")) 
cowplot::plot_grid(g1, g2, labels = LETTERS[1:2])
rm(list = ls()); invisible(gc())
```

# References

