---
title: Topography drives microgeographic adaptations among closely-related species of two tropical tree species complexes
date: '`r Sys.Date()`'
output:
  bookdown::word_document2:
    reference_docx: ./template/template.docx
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
  bookdown::html_document2:
    number_sections: false  
    toc: true
    toc_float: yes
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/mee.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

# Outlines 4

Tropical forests tremendous biodiversity [@Gaston2000] have intrigued biologists for decades [@connell_diversity_1978]. 
Lowland Amazonia alone is estimated to shelter around sixteen thousand tree species [@TerSteege2013]. 
But the five thousand species of trees described in Amazonia belong to only height hundred ten genera [@TerSteege2013], including thus many species rich genera. 
Even at the hectare-scale, tropical forests shelter up to several hundred tree species [@Gentry1988] including species-rich genera with closely-related species coexisting in sympatry [@Caron2019]. 
Species in species-rich genera have a higher level of genetic polymorphism [@Caron2019] and hybridize more often between congeners [@Whitney2010] than those in species-poor genera.
Species-rich genera may thus result from species complexes, composed of morphologically similar species and/or species sharing large amounts of genetic variation due to recent common ancestry and/or hybridization [@Pernes1984].
Specifically, when species are connected by limited, but recurrent, interspecific gene flow, they form what is called a syngameon [@Suarez-Gonzalez2018].
However, closely-related species are expected to share similar niche and functional strategies due to phylogenetic constraints [@Wiens2010], especially with the low genetic differentiation found within species complexes.
Niche and functional similarities is expected to lead to increase competition between closely-related species and ultimately to local competitive exclusion of one species by the other. 
Despite the local abundance and regional success of closely-related species growing in sympatry within species complexes [@Gentry1988; @TerSteege2013; @Pinheiro2018], little is known of eco-evolutionary forces driving the local coexistence within Neotropical species complexes.

Niche theory explains species local coexistence based on ecological niche differences limiting competitive exclusion [@lortie_rethinking_2004-1; @weiher_assembly_1995].
The heterogeneity of ressources distribution in space and time define fine-scale habitat where species can coexist [@lortie_rethinking_2004-1; @weiher_assembly_1995]. 
For instance, topography is spatially driving water and nutrient distribution in tropical forests [@ferry2010higher].
Topography drives pervasive habitat preference among species [@Allie2015; Schmitt et al., in prep] and functional responses among and within species [@Schmitt2020].
Additionally, forest gap dynamics is a stong driver of above- and below-ground competition for ressources in space and time [@Hubbell1999; @Molino2001].

But "nothing in biology makes sense except in the light of evolution" [@Dobzhansky1973].
The genetic proximity of species within species complexes questions the role taken by evolution in their local coexistence.
The assumption that evolution and ecology play at very different time-scales might have been an impediment to understand the role of eco-evolutionary processes in species coexistence [@Pelletier2009]. 
However, eco-evolutionary processes have been documented for a while [@Tutt1896] and can play an important role on the dynamic of both species and communities [@Bailey2009]. 
Evidences exist for eco-evolutionary processes at microgeographic scale, e.g. within the dispersal neighborhood of the organism [@Richardson2014].

Closely-related species growing in sympatry in differentiated ecological niches form an adaptive radiation, such as Darwin's finches in the Galapagos [@Seehausen2004]. 
Evolutive history behind adaptive radiations fall within a continuum from sympatric ecological speciation to seconday contacts of species ecologically specialised in allopatry or parapatry [@rundell_adaptive_2009].
Few studies evidenced adaptive radiation driven by topography or other abiotic habitats for tropical trees [@Paun2016; @Pillon2014].

In the present study, we assessed genetic species delimitation and genotypic diversity of closely-related sympatric tree species belonging to two Neotropical species complexes.
We addressed fine-scale spatial and temporal adaptation along topography. 
Combining tree inventories, LiDAR-derived topographic data, and single nucleotide polymorphisms (SNPs), we used population genomics, environmental association analyses, genome wide association studies to adress the following questions:

1. How is genetic diversity structured among and within species of Neotropical tree species complexes ?
1. Are tree species and individuals adapted to the topographic gradient?
1. What are the putative differences between topographic determinant of species niches?

We hypothesized species to be delineated into gene pools corresonding to already described taxonomic morphotypes but to include interspecific gene flow and misidentifications due to blured morphological differences [@Pinheiro2018].
But molecular data could reveal cryptic species or merge taxon  [@Ewedje2020].
We hypothesized species to be delineated along topographic with fixed neutral and adaptive variants, following previous evidences of topography determining species' ecological niches [Schmitt et al., in prep].
But interspecific gene flow within the species complex could reduce species adaptations to topography and blur evolutive history [@Tigano2016].

We sampled leaf from 402 individuals previously morpholigcally identified as *Symphonia globulifera* and *Symphonia sp. 1* and 432 individuals belonging to the clade *Parvifolia* of the genus *Eschweilera* in the Paracou field station, where individuals have been censused for more than 30 years. 
We designed and conducted a targeted sequence capture of genomic regions for each individual. 
We called, filtered and annotated single nucleotide polymorphisms (SNPs) of sequenced libraries against the references used to build the targeted sequences. 
We analyzed population genetic structure, genetic diversity, and demographic history. 
We then measured species and genotypes association to topography with Bayesian inferences of the animal model [@Wilson2010]. 
Finally, we scanned genomes for SNPs association wih population structure and topography, and tested the functional enrichment of detected outliers. 

*Symphonia globulifera* was locally composed of three sympatric species corresponding to three different morphotypes (Fig. \@ref(fig:Sadmixture)). 
*Eschweilera* clade Parvifolia was locally composed of three sympatric species corresponding to botanical species.
The spatial distribution of *Symphonia* species suggested that wetness drove individual survival among populations ($TWI$, $R_m^2=0.35$) resulting in intermediate genetic differentiation ($F_{ST}=0.15$) with fixed markers among populations (5.7% of SNPs) and an enrichment of genes related to “response to water deprivation” ($p=0.01106483$) revealing adaptive markers among the markers fixed among species. 
The spatial distribution of *Eschweilera* species suggested that relative elevation drove individual survival among populations ($RE$, $R_m^2=0.13$) resulting in intermediate genetic differentiation ($F_{ST}=0.X$) with fixed markers among populations (X% of SNPs). 
Topographic wetness index ($TWI$) was related to hydrology (ANOVA with water table depth, $R^2=0.32$, $p<10^{-151}$), wheras relative elevation ($RE$) was related to soil chemistry due to hydromorphy (ANOVA with soil carbon, nitrogen, organic matter, exchagable cation and soil color due to hyrdomorphy with $0.23<R^2<0.28$, $p<10^{-30}$).

Within tree species complexes, closely-related species have different realized optima for topographic niches with species genetic adaptations. 
*Symphonia* bla.
*Eschweilera* bla.
Consequently, species adaptations to different characteristics of topgraphy stabilize local coexistence between closely-related species within species complexes.

# Figures

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
library(tidyverse)
library(raster)
library(bayesplot)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, 
               fig.height = 6, fig.width = 6,
               cache = T, cache.lazy = F)
# crs <- '+proj=longlat +datum=NAD83 +no_defs +ellps=GRS80 +towgs84=0,0,0'
crs <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
spcol <- c("#1b9e77", "#d95f02", "#7570b3", "#1E90FF", "#0000EE", "#CD2626")
names(spcol) <- c("E. decolorans", "E. sagotiana", "E. coriacea", "S. globulifera Paracou", "S. globulifera Regina", "S. sp1")
```

```{r maps, fig.cap="Map of sampled individuals for species complexes *Symphonia* and *Echweilera* clade Parvifolia. Color represent species determined with genetic structure and not orignial botanical identification. Grey level represents hillshading to visualize topography.", fig.width=6, fig.height=5}
load("../symcapture_save/Env.Rdata")
symphonia <- trees
load("../../parvicapture/parvicapture_save//Env.Rdata")
eschweilera <- trees
trees <- bind_rows(
  mutate(symphonia, genus = "Symphonia") %>% 
    dplyr::select("Ind", "Xutm", "Yutm", "genus", "pop") %>% 
    dplyr::rename(species = pop) %>% 
    mutate(species = gsub("Type", " ", species)) %>% 
    mutate(species = paste0("S. ", species)),
  mutate(eschweilera, genus = "Eschweilera") %>% 
    dplyr::select("Library", "Xutm", "Yutm", "genus", "cluster") %>% 
    dplyr::rename(Ind = Library, species = cluster) %>% 
    mutate(species = recode(species, "3" =  "E. decolorans", 
                            "2" = "E. sagotiana",
                            "1" = "E. coriacea")) 
)
coordinates(trees) <- ~Xutm + Yutm
proj4string(trees) <- '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0'
trees <- spTransform(trees, CRSobj = crs)
hills <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                          "topography", "hillshade_1m.tif"))
hills <- crop(hills, trees)
hills <- projectRaster(hills, crs = crs)
hills <- as(aggregate(hills, 4), "SpatialPixelsDataFrame")
hills <- as.data.frame(hills)
colnames(hills) <- c("hills", "Xutm", "Yutm")
hills <- lapply(c("Symphonia", "Eschweilera"), function(g) mutate(hills, genus = g)) %>% bind_rows()
ggplot(hills, aes(x = Xutm, y = Yutm)) +
  geom_tile(aes(fill = hills),  col = NA, alpha = 0.6, data = ) +
  scale_fill_distiller(guide = "none", palette = "Greys", direction = 1) +
  geom_point(data = as.data.frame(trees), aes(col = species)) +
  scale_color_manual("", values = spcol[c(3, 1, 2, 4, 5, 6)]) +
  facet_wrap(~ genus) +
  coord_equal() +
  cowplot::theme_map() +
  ggspatial::annotation_scale(location = "br", width_hint = 0.5, plot_unit = "m", 
                              data = data.frame(genus = "Symphonia")) +
  ggspatial::annotation_north_arrow(location = "tl", height = unit(1, "cm"), width = unit(1, "cm"),
                                    data = data.frame(genus = "Eschweilera")) +
  theme(axis.title = element_blank(),
        legend.position = "bottom", legend.text = element_text(face = "italic"),
        strip.background = element_blank(), strip.text.x = element_blank()) 
```

```{r symphospecies, fig.cap="Admixture plot of *Symphonia* individuals for the best K = 3.", fig.width=6, fig.height=3}
pophelper::readQ(file.path("../../../../data/Symphonia_Paracou/Sequences/populationGenomics/",
                           "admixture", "paracou", "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.3.Q"))[[1]] %>%
  dplyr::rename(Ssp1 = Cluster1, SgParacou = Cluster2, SgRegina = Cluster3) %>%
  mutate(Ind = gsub(".g.vcf", "", read_tsv(file.path("../../../../data/Symphonia_Paracou/Sequences", "variantCalling", "paracou",
                                                     "symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam"),
                                           col_names = c("FID", "IID", "FIID", "MIID", "sex", "phenotype"))$IID)) %>%
  arrange(desc(Ssp1), desc(SgParacou), desc(SgRegina)) %>%
  mutate(order = 1:nrow(.)) %>%
  reshape2::melt(id.vars = c("order", "Ind"), variable.name = "pop") %>%
  ggplot(aes(reorder(Ind, order), value, fill = pop, col = pop)) +
  geom_col() +
  scale_fill_manual("", values = c("#CD2626", "#1E90FF", "#0000EE"),
                    labels = c("S. sp1", "S. globulifera Paracou", "S. globulifera Regina")) +
  scale_color_manual("", values = c("#CD2626", "#1E90FF", "#0000EE"),
                     labels = c("S. sp1", "S. globulifera Paracou", "S. globulifera Regina")) +
  ylab("Admixture coefficient") +
  theme(axis.title.x = element_blank(), axis.line.x = element_blank(),
        axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        legend.text = element_text(face = "italic"), legend.position = "top")
```

```{r symphomorphotypes, fig.cap="*Symphonia* trunk morphology. The three morphotypes are identified with their bark with *S. sp1* (left, red) having a light grey thin and smooth bark, the *S. globulifera type Paracou* (center, light blue) having a dark and intermediate thin and smooth bark compared to the thick and lashed bark of *S. globulifera type Regina* (right, dark blue)."}
include_graphics("figures/symphoniamorphotypes.png")
```

```{r parvispecies, fig.cap="Principal component analysis (PCA) of single nucleotide polymorphisms (SNP) from *Eschweilera* clade Parvifolia. The colors represent the cluster detected with Kmeans for the best K = 3.", fig.height=5, fig.width=5}
vars <- read_delim(file.path("../../../../data/Eschweilera_Paracou/Sequences/", "variants", "final", "filtered.eigenval"), 
               delim = " ", col_names = "var")
read_delim(file.path("../../../../data/Eschweilera_Paracou/Sequences/", "variants", "final", "filtered.eigenvec"),
           delim = " ", col_names = c("X1", "Sample", paste0("PCA", 1:20))) %>% 
  separate(Sample, paste0("X", 1:7), "[:punct:]", remove = F) %>% 
  dplyr::rename(IdGenetic = X4, Lane = X5) %>% 
  left_join(read_tsv(file.path("../../../../data/Eschweilera_Paracou/Sequences/", "variants", "final", "filtered.kmeans"))) %>% 
  mutate(species = recode(cluster, "3" =  "E. decolorans", 
                          "2" = "E. sagotiana",
                          "1" = "E. coriacea")) %>% 
  ggplot(aes(x = PCA1, y = PCA2, col = species)) +
  geom_point(size=2, alpha = 0.5) +
  theme(legend.position = "bottom", legend.text = element_text(face = "italic")) +
  xlab(paste0("PCA 1 - ", vars$var[1], "%")) +
  ylab(paste0("PCA 2 - ", vars$var[2], "%")) +
  scale_color_manual("", values = spcol[c(3,1,2)])
```

```{r SpDistVar, fig.cap="Species distribution (**A**) and individual variance partitionning (**B**) along topography. In subfigure **A**, individuals density per species identified is given along the tropographic wetness (TWI) and the logarithm of relative elevation (log(RE+1)). In subfigure **B**, individual variation along the tropographic wetness index (TWI) and the relative elevation (RE) is partitionned with the Animal model into between-species (red), between-genotypes (green) and residual (blue).", fig.width=10, fig.height=6}
wetness <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "TWI_1m.tif"))
relele <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/", 
                            "topography", "RelativeElevation_1m.tif"))
dem <- raster(file.path("/home/sylvain//Documents/BIOGECO/PhD/data/Paracou/",
                        "topography", "DEM_1m_2015.tif")) # for CRS
projection(wetness) <- projection(dem)
projection(relele) <- projection(dem)
wetness <- projectRaster(wetness, crs = crs)
relele <- projectRaster(relele, crs = crs)
trees$TWI <- raster::extract(wetness, trees)
trees$RE <- raster::extract(relele, trees)
gl <- trees@data %>% 
  mutate(RE = log(RE+1)) %>% 
  reshape2::melt(id.vars = c("Ind", "genus", "species")) %>% 
  mutate(variable = recode(variable, "TWI" = "Topographic wetness index",
                           "RE" = "log(Relative elevation + 1)")) %>% 
  ggplot(aes(value, fill = species, color = species)) +
  geom_density(alpha = 0.1) +
  facet_wrap(~ variable, scales = "free", nrow = 2) +
  scale_color_manual("", values = spcol[c(3, 1, 2, 4, 5, 6)]) +
  scale_fill_manual("", values = spcol[c(3, 1, 2, 4, 5, 6)]) +
  theme(legend.position = "bottom", legend.text = element_text(face = "italic"))
fitSympho <- list(TWI = list(), RE = list())
for(var in c("TWI", "RE")){
  for(sim in list.files("../symcapture_save/EnvGeno", 
                        pattern = var, full.names = T)){
    load(sim)
    fitSympho[[var]] <- c(fitSympho[[var]], fit)
  }
}
fitSympho <- lapply(fitSympho, rstan::sflist2stanfit)
names(fitSympho) <- paste0("Symphonia.", names(fitSympho))
fitParvi <- list(TWI = list(), RE = list())
for(var in c("TWI", "RE")){
  for(sim in list.files("../../parvicapture/parvicapture_save/all/EnvGeno2/", 
                        pattern = var, full.names = T)){
    load(sim)
    fitParvi[[var]] <- c(fitParvi[[var]], fit)
  }
}
fitParvi <- lapply(fitParvi, rstan::sflist2stanfit)
names(fitParvi) <- paste0("Eschweilera.", names(fitParvi))
fitEnv <- c(fitSympho, fitParvi)
rm(fitSympho, fitParvi)
gr <- lapply(fitEnv, mcmc_intervals_data, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(parameter = recode(parameter, "Vp" = "Species", "Vg" = "Genotype", "Vr" = "Residual")) %>% 
  group_by(variable) %>%
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>%
  separate(variable, c("genus", "variable")) %>% 
  group_by(genus, variable) %>%
  mutate(m2 = m/sum(m)) %>% 
  ungroup() %>% 
  mutate(variable = recode_factor(variable, "TWI" = "Topographic wetness index",
                           "RE" = "Relative elevation")) %>% 
  ggplot(aes(x = genus, fill = parameter)) +
  geom_col(aes(y = m2)) +
  geom_text(aes(y = m2, label = pct), col = "white", position = position_stack(vjust = .5)) +
  facet_wrap(~ variable, scales = "free", nrow = 2) +
  coord_flip() +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), 
        axis.ticks.x = element_blank(), axis.line.y = element_blank(),
        legend.position = "bottom", axis.text.y = element_text(face = "italic")) +
  scale_fill_discrete(expression(sigma^2))
cowplot::plot_grid(gl, gr, labels = LETTERS[1:2])
```

# References

