---
title: "**Forest gap dynamics: an underexplored factor that drives divergent adaptive growth strategies within tropical tree species**"
# author: "*Bruno Hérault, Myriam Heuertz, Sylvain Schmitt, Niklas Tysklind*"
# date: '`r Sys.Date()`'
output: 
  bookdown::html_document2:
    number_sections: false  
    toc: true
    toc_float: yes
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
    keep_tex: yes
  bookdown::word_document2:
    reference_docx: ./template/template.docx
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/pnas.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

**PNAS: 4323/~4000 words (6 pages with references) and /50 references, 150/250 words abstract, 117/120 words significance statement**

# Title Page

**BIOLOGICAL SCIENCES:** Ecology & Evolution *(Dual classifications are permitted in exceptional cases, subject to Editorial Board approval, within a major category.)*

Sylvain Schmitt$^a,2$, 
Niklas Tysklind$^b$, 
Myriam Heuertz$^{a,1}$,
Bruno Hérault$^{c,d,e,1}$

*$^a$ Univ. Bordeaux, INRAE, BIOGECO, 69 route d’Arcachon, 33610 Cestas France;* 
*$^b$ INRAE, UMR EcoFoG (Agroparistech, CNRS, Cirad, Université des Antilles, Université de la Guyane), Campus Agronomique, 97310 Kourou, French Guiana;*
*$^c$ CIRAD, UPR Forêts et Sociétés, Yamoussoukro, Côte d'Ivoire;*
*$^d$ Forêts et Sociétés, Univ Montpellier, CIRAD, Montpellier, France;*
*$^e$ Institut National Polytechnique Félix Houphouët-Boigny, INP-HB, Yamoussoukro, Côte d'Ivoire;*
*$^{1}$ M.H. contributed equally to this work with B.H.*
*$^{2}$ To whom correspondence should be addressed. E-mail: sylvain.m.schmitt@gmail.com*

**Keywords:** genotypic adaptation | neighbor crowding index | *Symphonia globulifera*

# Abstract

**In tropical forests, natural gap dynamics is a primary driver of ecosystem functioning by triggering a wide variety of ecological growth and survival strategies of trees. These strategies have long been studied among species, neglecting individual variation in demographic responses within species. Variation in demographic response is facilitated by overlapping generations which maintain the genetic diversity necessary for temporally-variable but strong selection to act on specific life stages. Tropical trees are long-lived and produce many seedlings whose growth and survival is shaped by forest gap dynamics. Here, we provide evidence that genotypic adaptive growth strategies allow forest tree species to grow in a mosaic of light and competition environments, or successional niches, shaped by time since the last tree fall. Consequently, the successional niche selects individuals with the most suitable adaptive growth strategy to reach the canopy, with fast-growing *hares* in forest gaps and slow-growing *turtles* in shaded closed-canopy patches.**

# Significance Statement

**The natural gap dynamics from falling trees is one of the main drivers of ecosystem functioning in tropical forests. Trees respond to forest gap dynamics by a wide variety of ecological strategies, but these strategies have long been studied among species, neglecting genetic variability within species. Here we provide genetic evidence for diverse adaptive growth strategies of individuals within mature forest tree species that allow them to grow in a diversity of light and competition environments that vary with time since the last tree fall. We show that the fine spatio-temporal dynamics of forest gaps is a previously neglected factor that, along with others, contributes to maintaining tropical tree within-species genetic diversity, the raw material of evolution.**

# Introduction

The heterogeneity of resource distribution in space and time defines the fine-scale habitat structure where species and individuals can grow and coexist [@Lortie2004a; @weiher_assembly_1995]. 
For instance, the spatial distribution of water and nutrients varies strongly along topographic gradients in tropical forests [@Ferry2010]. 
Therefore, topography drives pervasive differentiation in habitat preference among tree species [@Engelbrecht2007]
and in functional responses among and within species [@Schmitt2020].
In natural tropical forests, competition for light due to forest gap dynamics [@Hubbell1999; @Molino2001]
is often greater than the effect of competition for water and nutrients, even in early successional stages [@VanBreugel2012].
Forest gap dynamics is characterised by a mosaic of light and competition environments, 
or successional niches, shaped by time since the last treefall. 
Forest succession contrasts a bright environment with reduced competition from adult trees after a treefall with a variety of more shaded environments in which competition from tall trees increases as they fill with vegetation.
Successional niches trigger a wide variety of demographic responses associated with ecological strategies of individuals and species [@Herault2010; @Ruger2009].
Pioneer species with fast growth, thanks to productive leaf tissue and light wood, rapidly colonize treefall gaps whereas late successional species with slower growth and more conservative tissue progressively establish in more shaded environments [@Craven2015].

But “nothing in biology makes sense except in the light of evolution” [@Dobzhansky1973].
Spatial habitat heterogeneity has been shown to drive differential adaptations of tree species, 
in the case of adaptive radiations [@Paun2016; @Pillon2014],
and within tree species, with microgeographic adaptations within the dispersal neighbourhood of an individual [@Brousseau2015; @Richardson2014; @Team2013]. 
However, the role of temporal habitat heterogeneity as a driver of adaptive evolution has been much less explored. 
Forest gap dynamics is a spatio-temporal process that has a strong impact on tree ecology [@VanBreugel2012],
however, to our knowledge, no studies have explored its role as a driver of adaptation within species.

The assumption that evolution and ecology operate at very different time scales may have impeded such studies [@Pelletier2009],
however, evidence is accumulating showing that evolutionary processes can be fast enough to impact ecological processes on a contemporary time scale [@Hairston2005; @Rudman2017]. 
Theoretical work demonstrated that biological systems with overlapping generations can maintain genetic variation in long-lived stages in which selection is relaxed and which provide diversity for specific, young stages, on which strong temporally variable selection can act [@Ellner1994]. 
For instance, overlapping generations with oscillation between short high-recruitment periods and long population-decline periods explain the spatial genetic structure of the Neotropical tree *Jacaranda copaia* [@Jones2006].

In tropical forests, the dynamics of forest gaps strongly determines the survival and recruitment of non-mature trees through competition for access to light.
Once in the canopy, adult trees with full access to light are free from the light-driven selection of forest gap dynamics. 
Adult trees can therefore act as a reservoir of genotypes against the hazards of fine spatio-temporal dynamics of forest gaps.
Generations of tropical trees broadly overlap, for instance late-successional tree species start reproducing around 20cm of diameter at breast height (DBH),
while they may grow up to 1m DBH [@Hardy2006] and live more than a century [@OBrien1995].

Tropical trees thus disperse many seedlings over time into a mosaic of light conditions that fluctuates spatially and temporally due to the dynamics of forest gaps.
Individual offspring encounter mainly shady and sometimes bright environments in which they face competition and must grow to survive.
Since forest gap dynamics opposes rapid growth in a bright environment to slow growth in a shaded environment, 
we hypothesize that individual genotypes within species would be adapted to divergent growth strategies between bright and shaded environments. 
Adaptation to slow growth in shade would allow individuals to persist with conservative tissues;
whereas adaptation to fast growth in a bright environment would allow individuals to outgrow other seedlings and maintain access to light through an acquisition strategy with cheaper tissues but with less defence against herbivores and pathogens [@Fine2006].
Consequently, we hypothesize that forest gap dynamics is a major driver of the spatial and temporal structuring of genotypic diversity within tropical tree species.

Following our hypothesis, we expect that the divergent adaptive growth strategies of individuals would be spatially structured according to the mosaic of successional niches created by the past fine-scale gap dynamics.
We also expect the divergent adaptive growth strategies to be structured in time. 
Because of strong light and competition-mediated selection on sapling stages [@Lewis2000],
genotypic adaptations to growth in different light conditions are expected to be more pronounced in intermediate life stages.
Seedlings may be mostly affected by very stochastic survival. 
Conversely, older stages maintain diversity as they have been recruited from diverse past successional niches and, 
once in the canopy, they are free from the light-driven selection of forest gap dynamics.

In the present study, we assessed genotypic diversity within closely-related sympatric tree species belonging to the widespread tropical tree species complex *Symphonia globulifera*.
We addressed the fine-scale spatial and temporal genetic adaptations of individuals through differential growth strategies in response to forest gap dynamics. 
We finally compared the breadth of successional niches encountered by *Symphonia* species to other locally abundant species. 
Combining tree diameter censuses, indirect measures of light environment of the recent past and present, and single nucleotide polymorphisms (SNPs), 
we used population genomics, environmental association analyses, genome wide association and growth modelling to address the following questions:

* Are individual genotypes structured by the mosaic of light and competition environments resulting from forest gap dynamics?
* Is the growth of individuals determined by genotypes?
* Is there an association between genotypic adaptations to gap dynamics and to growth?
* How are genotypic adaptations to gap dynamics and to growth structured in  time, i.e., across life stages?
* Are breadths of successional niches for *Symphonia* species wider than those of other locally abundant species?

# Results

*Symphonia globulifera* was locally composed of three sympatric species corresponding to three different gene pools with associated morphotypes and differentiated topographic niches (Schmitt et al., in prep). 
Individual tree genotypes explained a significant variation in individual neighbourhood crowding, an indirect measure of competition and light access during the last three decades, (18%) 
and in individual growth potential (43%, Fig. \@ref(fig:VarPart)).
This indicates an adaptive genetic response of *Symphonia* genotypes to successional niches and an adaptive component to growth potential. 
Genotypic adaptations to successional niches and individual growth potential were spatially structured on a similar scale, with higher than average genotypic similarity found up to 47m for both adaptations ($Moran’s~I, p-value<0.01$, Fig. \@ref(fig:SpCor)).
The adaptive responses to successional niches and to growth potential were significantly correlated, and, 
as predicted based on our hypothesis, the correlation was strongest for intermediate life stages: 
a significant negative correlation (Pearson's $R=-0.3$, $p=4.5.10^{-9}$) was observed between genotypic values for neighbourhood crowding and those for individual growth potential across diameter classes (Fig. \@ref(fig:gCor)),
with a stronger and a more significant signal for individuals in intermediate diameter classes, from 14 to 46cm DBH, than for individuals with smaller or larger DBH. 
These results suggest that individual genotypes within *Symphonia* species are adapted to successional niches that trigger adaptive growth strategies: 
a fast growth potential in response to a preference for low competition and bright environments, 
or a slower growth potential in response to a preference for higher competition and more shaded environments, 
structured in space and in diameter classes. 
Nevertheless, breadths of successional niches for *Symphonia* species (92 and 103 for *S. globulifera* and *S. sp1*, respectively) 
are within the distribution of those of locally abundant species ($97 \pm 10$, Fig. \@ref(fig:ncibreadth)).

# Discussion

In agreement with our hypothesis, we found that individual tree genotypes are differentially adapted to regenerate and thrive in response to the fine spatio-temporal dynamics of forest gaps.
A genetic trade-off exists between divergent adaptive growth strategies under different light conditions. 
The genetic trade-off is spatially structured into patches the size of a typical gap opened by a treefall.
The genetic trade-off is also structured across life stages with a stronger effect in the intermediate stages, 
than either young stages under diverse selection pressures (Tysklind et al. 2020)
or in older stages recruited in diverse past environments and currently free from the light-driven selection of forest gap dynamics.
Therefore, a species uses genetically differentiated growth strategies that allow its individuals to grow in a diversity of successional niches, 
thanks to these divergent adaptive growth strategies to forest gap dynamics, 
decreasing overall the risk of a stochastic local extinction. 
Consequently, the effect of forest gap dynamics on genetic adaptations among tropical tree species [@Herault2010; @Ruger2019]
is maintained within species in the species complex *Symphonia globulifera*.

Our results observed in a complex containing three species reveal the influence of forest gap dynamics on genotypes on tree species of tropical forests.
Breadths of successional niches for *Symphonia* species are similar to those of most of the locally abundant species (Fig. \@ref(fig:ncibreadth)). 
As the process driving genotypic adaptations is not specific to *Symphonia* nor to species complexes, 
we can assume it would be present in tree species of tropical mature forests.
Indeed, locally abundant species could only adapt to such broad successional niches through plasticity, 
if not through genotypic adaptations.
But the strength of competition for light due to forest gap dynamics [@VanBreugel2012]
advocates more for genotypic adaptations than a single genotype superior across a wide variety of successional niches [@Ellner1994], 
likely due to physiological or adaptive constraints.

Forest gap dynamics of natural [@Scotti2015] or anthropogenic [@Leclerc2015] origin was evidenced as a driver of genetic diversity and structure with varying directions depending on the disturbance intensity. 
However, to our knowledge, no studies have investigated genotypic adaptations to forest gap dynamics until now. 
Temporal dynamics and spatial stochasticity may have been an impediment to the hypothesis of selection based on forest gap dynamics,
but patches with varying successional stages and steady-state are abundant, while disturbance events are episodic [@Chambers2013]. 
Our study, linking genotype distribution to forest gap dynamics, was only possible thanks to a long-term and large-sized study site that allows us to study the eco-evolutionary dynamics of a complex phenotype such as the individual growth trajectory. 
Combined with results of Chambers et al. [@Chambers2013], 
the relative temporal stability of forest patches and the strength of forest gap dynamics on ecosystem functioning probably make it a major driver of diversity distribution in tropical forests.

The existence of a few taller individuals growing all the way to the canopy, i.e. the *winners*, 
and relegating a group of *losers* to the understory has already been demonstrated in tropical forests [@Farrior2016].
But our analyses revealed that the *winners* are not determined entirely at random but may result from a subtle match between the light environment produced by the forest gap stage and the relevance of individual genotypic adaptations to such an environment.
Therefore, the *winners* are not necessarily the fastest growing individuals, i.e. with the strongest vigor sensu Aubry-Kientz et al. @Aubry-Kientz2015, 
but slow-growing individuals might also represent *winners* in shaded closed-canopy patches. 
To conclude, genetic adequacy with forest gap dynamics determines successful individuals reaching the canopy through divergent growth strategies, 
with *hares* in forest gaps and *turtles* in more shaded closed-canopy patches.

# Material and methods

**Study design.**
The study was conducted in the Guiana Plateau region, at the Paracou field station. 
The site is made of 16 permanent plots which have been censused (diameter at breast height over 10cm) every 1-2 years for more than 35 years.
Nine of the plots were logged and subjected to human-induced disturbance in 1986. 
We sampled leaf tissue from 402 individuals previously morphologically identified as *Symphonia globulifera* and *Symphonia sp. 1* (*SI Materials and Methods*).

**Sequence capture.**
We designed in silico 20,000 80-mer probes for sequence capture on *Symphonia globulifera*.
The genetic resources used for the design consisted of two draft genomes, a transcriptome, and reduced-representation genomic sequence reads. 
We prepared and pooled libraries for each individual with a few repeats, 
captured sequences by hybridization with probes, 
and sequenced them in two lanes of an Illumina HiSeq 4000 instrument, following standard protocols.
We called, filtered and annotated single nucleotide polymorphisms (SNPs) of sequenced libraries against the references used to build the experiment with standards used in quantitative genomics (*SI Materials and Methods*).

**Genetic analyses.**
We investigated population genetic structure using `admixture` [@Alexander2011]
and the `introgress` R package [@Gompert2010]. 
We validated species delimitation with a second blind-identification of every collected individual in November 2019. 
Population genetics is further described in Schmitt et al. (in prep) and *SI Materials and Methods*. 
We inferred individual kinship using `KING` [@Manichaikul2010], 
as the method is robust to population structure. 
We set negative kinship values to null as they were confounding with population structure, 
and we further ensured that the matrix was positive-definite using the `nearPD` function from the R package `Matrix`.

**Neighbour crowding index.**
We used mean neighbourhood crowding over the last 30 years as an indirect measurement of access to light and forest gap dynamics, defined as follow:

$$NCI_i=\overline{\sum_{j|\delta_{i,j}<20m}DBH^2_{j,t}.e^{-\frac14\delta_{i,j}}}$$

with $DBH_{j,t}$ the diameter of the neighbouring tree $j$ in year $t$ and $\delta_{i,j}$ its distance to the individual tree $i$.
$NCI_i$ is computed for all neighbours at a distance $\delta_{i,j}$ inferior to the maximum neighbouring distance of 20 meters.
The power of neighbours $DBH_{j,t}$ effect was set to 2 to represent a surface. 
The decrease of neighbours diameter effect with distance was set to -0.25 to represent trees at 20 meters of the focal trees having 1% of the effect of the same tree at 0 meters. 
$NCI_i$ is computed as the mean of yearly $NCI_{i,t}$ over the last 30 years denoted by the overline.

**Individual maximum growth potential.**
The individual growth of individual $i$ in population $p$ between individual recruitment $y_0$ and 2017, correspond to the difference of diameter at breast height $DBH$ between the two years, and is defined with a hierarchical model in a lognormal distribution as follow:
$$DBH_{y=2017,p,i} - DBH_{y=y0,p,i} \sim logN(log[\sum_{y=y0}^{y=2017}AGR(DBH_{y,p,i})], \sigma^2_1)$$
where the difference of $DBH_{y=2017,p,i}-DBH_{y=y0,p,i}$ is defined with a lognormal distribution located on the logarithm of the sum of annual growth rates $AGR$ during the period $y_0$ to 2017 and of shape $\sigma^2_1$. 
The annual growth rates $AGR$ for individual $i$ in population $p$ at year $y$ with a diameter of $DBH_{y,p,i}$ is defined following a Gompertz model [@Gompertz1825], already identified as the best model for growth-trajectories in Paracou [@Herault2011]:
$$AGR(DBH_{y,p,i}) = Gmax_i.exp(-\frac12[\frac{log(\frac{DBH_{y,p,i}}{Doptp})}{Ksp}]^2)$$
where $Gmax_i$ is the maximum growth potential (maximal $AGR$ during individual life) for individual $i$ further used in association analyses, $Dopt_p$ is the population optimal diameter at which the individual reaches its maximum growth potential, and $Ks_p$ is the population kurtosis defining the width of the bell-shaped growth-trajectory [see figure 1 in @Herault2011].
To ease model inference population optimal diameter $Dopt_p$ and kurtosis $Ks_p$ were defined as a random population effect centered on a genus $Dopt$ and $Ks$ with corresponding variances $\sigma^2_{Dopt}$ and $\sigma^2_{Ks}$.

**Genotypic values estimate and associations.**
We used population and individual kinship in an homemade bayesian Animal model with a lognormal distribution to estimate genetic variance associated to the neighbourhood crowding where individual grow and individual maximum growth potential.
The Animal model was directly inferred with neighbourhood crowding, while the Animal model was nested in a hierarchical model to describe individual maximum growth potential $Gmax_i$ in the previous growth model  (*SI Materials and Methods*). 
To estimate variances on a normal-scale, we log-transformed population fixed effect, genetic additive values, and we calculated conditional and marginal $R^2$ [@Nakagawa2013] to partition observed variance.
We investigated the relationships between genotypic associations to neighbourhood crowding and individual maximum growth and their structure in space and time with respectively spatial autocorrelogram, using the `pgirmess` R package [@Giraudoux2018], and decomposed relations over diameter classes.

# Acknowledgments

We thank the University of Bordeaux for a PhD grant to Sylvain Schmitt. 
We are grateful to Pascal Petronelli and the CIRAD inventory team for their work on tree inventories and botanical identification, 
and to Christophe Plomion for comments on an early version of the manuscript. 
This study was funded through an Investissement d’Avenir grant of the ANR: CEBA (ANR-10-LABEX-0025), and the “projet innovant” LOCOCAP.

# Authors’ contributions

All authors conceived the ideas and designed methodology;
SS, MH, and NT sampled individuals;
SS and MH realized the gene capture experiment;
SS, MH and BH  analysed outputs; 
SS and BH led the writing of the manuscript. 
All authors contributed critically to the drafts and gave final approval for publication.

# References

<div id="refs"></div>

\newpage 

<!-- # Figures -->

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc())
library(knitr)
library(tidyverse)
library(bayesplot)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, 
               fig.height = 4, fig.width = 6,
               cache = T, cache.lazy = F)
load(file = "../symcapture/symcapture_save/outlines.Rdata")
```

```{r VarPart, fig.cap="Variance partitioning for neighbourhood crowding index (NCI), an indirect measurement of access to light, and for individual maximum growth potential (Gmax). Variation of each variable has been partitioned into among-species (red), among-genotype (green), and residual (blue) variation."}
lapply(list(NCI = fitNCI, Gmax = fitGmaxGeno), 
       mcmc_intervals_data, regex_pars = c("Vp", "Vg", "Vr")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(parameter = recode(parameter, "Vp" = "Species", 
                            "Vg" = "Genotype", "Vr" = "Residual")) %>%
  mutate(variable = factor(variable, levels = c("NCI", "Gmax"))) %>%
  group_by(variable) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>% 
  ggplot(aes(x = variable, fill = parameter)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  facet_wrap(~ variable, scales = "free") +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), axis.ticks.x = element_blank(),
        legend.position = "bottom") +
  scale_fill_discrete(expression(sigma^2)) 
```

```{r VarPart2, fig.cap="Variance partitioning for neighbourhood crowding index (NCI), an indirect measurement of access to light, and for individual maximum growth potential (Gmax). Variation of each variable has been partitioned into among-species (red), among-genotype (green), and residual (blue) variation."}
lapply(list(NCI = fitNCI, Gmax = fitGmaxGenoEnv), 
       mcmc_intervals_data, regex_pars = c("Vp", "Vg", "Vtwi", "Vnci", "Vr")) %>% 
  bind_rows(.id = "variable") %>% 
  mutate(parameter = recode(parameter, "Vp" = "Species", 
                            "Vg" = "Genotype", 
                            "Vtwi" = "TWI",
                            "Vnci" = "NCI",
                            "Vr" = "Residual")) %>%
  mutate(variable = factor(variable, levels = c("NCI", "Gmax"))) %>%
  mutate(parameter = factor(parameter, levels = c("Species", "Genotype", "NCI", "TWI", "Residual"))) %>%
  group_by(variable) %>% 
  mutate(pct = paste0(round(m / sum(m) * 100), "%")) %>% 
  ggplot(aes(x = variable, fill = parameter)) +
  geom_col(aes(y = m)) +
  geom_text(aes(y = m, label = pct), col = "white", position = position_stack(vjust = .5)) +
  facet_wrap(~ variable, scales = "free") +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
        axis.text.x = element_blank(), axis.line.x = element_blank(), axis.ticks.x = element_blank(),
        legend.position = "bottom") +
  scale_fill_discrete(expression(sigma^2)) 
```

\newpage 

```{r SpCor, fig.cap="Spatial autocorrelogram for raw variable and genotypic values of neighbourhood crowding index (NCI), an indirect measurement of access to light, and individual maximum growth potential (Gmax). The spatial autocorrelogram shows Moran’s I value for different distance classes with significant values represented by filled points against transparent non-significant values. Colours of lines and points represent neighbourhood crowding index (blue) and individual maximum growth potential (red). Linetype represents raw variable (dashed) and associated genotypic values (continuous)."}
cor %>% 
  filter(dist.class < 10^3) %>% 
  filter(variable %in% c("Gmax", "NCI", "gGmax", "gNCI")) %>% 
  mutate(type = ifelse(substr(variable, 1, 1) == "g", "genotype", "variable")) %>% 
  mutate(variable = gsub("^g", "", variable)) %>% 
  ggplot(aes(x = dist.class, y = coef, col = variable, linetype = type)) + 
  geom_hline(yintercept = 0) +
  geom_point(aes(alpha = p.value < 0.01)) + 
  geom_line() +
  scale_alpha_manual(guide = "none", values = c(0.3, 1)) + 
  scale_linetype("") +
  scale_color_discrete("") +
  scale_x_log10() +
  geom_vline(xintercept = c(27, 53), col = "lightgrey", linetype = "dashed") +
  geom_vline(xintercept = 150, col = "lightgrey", linetype = "dotted") +
  xlab("Distance (m)") + ylab("Moran\'s I") +
  theme(legend.position = "bottom")
```

```{r SpCor2, fig.cap="Spatial autocorrelogram for raw variable and genotypic values of neighbourhood crowding index (NCI), an indirect measurement of access to light, and individual maximum growth potential (Gmax). The spatial autocorrelogram shows Moran’s I value for different distance classes with significant values represented by filled points against transparent non-significant values. Colours of lines and points represent neighbourhood crowding index (blue) and individual maximum growth potential (red). Linetype represents raw variable (dashed) and associated genotypic values (continuous)."}
cor2 %>% 
  filter(dist.class < 10^3) %>% 
  filter(variable %in% c("Gmax", "NCI", "gGmax", "gNCI")) %>% 
  mutate(type = ifelse(substr(variable, 1, 1) == "g", "genotype", "variable")) %>% 
  mutate(variable = gsub("^g", "", variable)) %>% 
  ggplot(aes(x = dist.class, y = coef, col = variable, linetype = type)) + 
  geom_hline(yintercept = 0) +
  geom_point(aes(alpha = p.value < 0.01)) + 
  geom_line() +
  scale_alpha_manual(guide = "none", values = c(0.3, 1)) + 
  scale_linetype("") +
  scale_color_discrete("") +
  scale_x_log10() +
  geom_vline(xintercept = c(27, 53), col = "lightgrey", linetype = "dashed") +
  geom_vline(xintercept = 150, col = "lightgrey", linetype = "dotted") +
  xlab("Distance (m)") + ylab("Moran\'s I") +
  theme(legend.position = "bottom")
```

\newpage 

```{r gCor, fig.cap="Relation between genotypic values for individual growth potential (Gmax) and neighbourhood crowding index (NCI), an indirect measurement of access to light, for different classes of diameters. Regression lines represent a linear model of form y ~ x. Annotations give for each diameter class the Pearson’s R correlation coefficient and the associated p-value.", fig.width=8}
n <- 6
t %>% 
  arrange(DBHtoday) %>% 
  mutate(DBHclass = rep(1:n, each = ceiling(nrow(.)/n))[1:nrow(.)]) %>%
  mutate(significance = ifelse(DBHclass %in% c(1,6), "0", "1")) %>% 
  group_by(DBHclass) %>% 
  mutate(DBHclassCm = paste0(round(min(DBHtoday)), " - ", round(max(DBHtoday)), " cm")) %>% 
  ggplot(aes(aGmax, aNCI)) +
  geom_point(alpha = 0.5) +
  facet_wrap(~ DBHclassCm, scales = "free") +
  geom_smooth(method = "lm", aes(col = significance)) +
  scale_size_continuous(guide = "none") +
  ggpubr::stat_cor(method = "pearson", label.x = 0.5, label.y = 0.85, ) +
  xlab("Genetic additive values for individual growth potential (Gmax)") +
  ylab("Genetic additive values for \n neighbourhood crowding index (NCI)") +
  scale_color_manual(guide = "none", values = c("darkgrey", "black"))
```

\newpage 

```{r ncibreadth, fig.cap="Niche breadth in neighbourhood crowding index (NCI) for species abundant in Paracou. Niche breadth is defined by the difference between the 95th and 5th quantiles of neighbourhood crowding index (NCI) for individuals with a diameter at breast height between 10 and 20 cm. Niche breadth has been calculated for species with at least fifty individuals meeting previous criteria. The analyses used only control plots, and not human disturbed plots. Dashed lines represent niche breadth for Symphonia globulifera (red) and Symphonia sp. 1 (blues)."}
load("../symcapture/symcapture_save/NCI2017all.Rdata")
sp.3 <- read_delim("../../../data/Symphonia_Paracou/Sequences/populationGenomics/populations/paracou3pop.popmap",
                   col_names = c("library", "species"), delim = " ") %>%
  filter(species == "globuliferaTypeRegina") %>% 
  dplyr::select(library) %>% 
  mutate(library = gsub("P", "", library)) %>% 
  separate(library, c("Plot", "SubPlot", "TreeFieldNum"), convert = T)
t3 <- trees %>% 
  filter(Species != "Indet.") %>% 
  anti_join(sp.3) %>% 
  filter(DBH >= 10 & DBH <= 20) %>% 
  group_by(Genus, Species) %>% 
  mutate(N = n()) %>%
  mutate(N = ifelse(Genus == "Symphonia" & Species == "globulifera", 50, N)) %>% 
  filter(N >= 50) %>%
  group_by(Genus, Species, N) %>% 
  summarise(NCI95 = quantile(NCI, 0.95), NCI05 = quantile(NCI, 0.05)) %>% 
  mutate(NCIbreadth = NCI95 - NCI05) %>% 
  mutate(Species = recode(Species, "globulifera" = "sp.2"))
ggplot(t3, aes(NCIbreadth)) + 
  geom_histogram(fill = "lightgrey", bins = 30) + 
  geom_vline(aes(xintercept = NCIbreadth,
                 col = paste0(substr(Genus, 1, 1), ". ", Species)),
             linetype = "dashed",
             data = filter(t3, Genus %in% c("Symphonia"))) +
  ylab("Number of species") + xlab("Niche breadth in neighbourhood crowding index (NCI)") +
  scale_color_manual("Studied species", values = c("#CD2626", "#1E90FF")) +
  theme(legend.position = "bottom", legend.text = element_text(face = "italic"))
```

```{r thesis, eval=F, echo=T}
save(list = ls(), file = "../../thesis/rdata/Ch4Msc.Rdata")
```
