---
title: Topography and forest gaps dynamics jointly control adaptation and populations structure in a tropical tree species, insights from a virtual experiment 
output:
  bookdown::pdf_document2:
    includes:
      before_body: ./template/doc_prefix.tex
      in_header: ./template/preamble.tex
    number_sections: false
    toc: false
  bookdown::word_document2: 
    number_sections: false
  bookdown::html_document2:
    number_sections: false
    toc: true
    toc_float: yes
    include:
      in_header: ../html/hypothesis.html
      before_body: ../html/open_review_block.html
linestretch: 1.5
csl: /home/sylvain/Documents/Bibliography/csl/oikos.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

```{r config, message=FALSE, warning=FALSE, include=FALSE}
rm(list = ls()); invisible(gc()) ; set.seed(42)
library(knitr)
library(tidyverse)
library(bayesplot)
theme_set(bayesplot::theme_default())
opts_chunk$set(echo = F, message = F, warning = F, cache = T, cache.lazy = F)
```

1. Sylvain Schmitt
    - Corresponding author
    - sylvain.m.schmitt@gmail.com
    - orcid.org/0000-0001-7759-7106
    - CNRS, UMR EcoFoG (Agroparistech, Cirad, INRAE, Université des Antilles, Université de la Guyane), Campus Agronomique, 97310 Kourou, French Guiana
    - Univ. Bordeaux, INRAE, BIOGECO, 69 route d’Arcachon, 33610 Cestas France
1. Niklas Tysklind
    - Niklas.Tysklind@ecofog.gf
    - INRAE, UMR EcoFoG (Agroparistech, CNRS, Cirad, Université des Antilles, Université de la Guyane), Campus Agronomique, 97310 Kourou, French Guiana
1. Myriam Heuertz
    - heuertzm@gmail.com
    - orcid.org/0000-0002-6322-3645
    - Univ. Bordeaux, INRAE, BIOGECO, 69 route d’Arcachon, 33610 Cestas France
1. Bruno Hérault
    - bruno.herault@cirad.fr
    - orcid.org/0000-0002-6950-7286
    - CIRAD, UPR Forêts et Sociétés, Yamoussoukro, Côte d’Ivoire
    - Forêts et Sociétés, Univ Montpellier, CIRAD, Montpellier, France
    - Institut National Polytechnique Félix Houphouët-Boigny, INP-HB, Yamoussoukro, Côte d’Ivoire

\newpage 

# Abstract

*To be writen depending on manuscript review and aimed journal.*


## Keywords

xx ; xx ; xx

\newpage 

# Introduction

The maintenance of biodiversity is a long-standing issue in ecology [@Hutchinson1941] and evolution [@Darwin1909]. Biodiversity is characterized by nested levels, ranging from genes to ecosystems. The Earth has a large number of terrestrial and marine ecosystems, however, the exceptional biodiversity of tropical rainforests has always fascinated biologists [@connell_diversity_1978]. Tropical forests are home to the greatest diversity of species in the world [@Gaston2000],  a fact that remains partly unexplained and whose origin is subject to debate [@wright2002plant]. The five thousand described tree species in the Amazon belong to only eight hundred and ten genera [@TerSteege2013]. Even at the hectare scale, tropical forests harbor species-rich genera with closely related species coexisting in sympatry [@Caron2019]. Closely related species are expected to share similar niche and functional strategies due to phylogenetic constraints [@Wiens2010]. Similarities in niche and function are generally expected to lead to increased competition between closely related species and ultimately to the local competitive exclusion of one species by the other. Despite the local abundance and regional success of closely related species that thrive in sympatry [@Gentry1988; @Pinheiro2018; @TerSteege2013], little is known about the eco-evolutionary forces driving the local coexistence of closely related species [@baraloto_using_2012-1; @Cannon2019a; @Levi2019; @TerSteege2013].


At the species level, different ecological theories have been developed to explain the coexistence of species and thus the persistence of biodiversity. Niche theory explains local coexistence of species by ecological niche differences that limit competitive exclusion [@Lortie2004a; @weiher_assembly_1995]. Heterogeneity in resource distribution across space and time defines a small-scale habitat where species can coexist. For example, topography has a pervasive effect in differentiating habitat preferences among species [@gunatilleke_specieshabitat_2006; @kraft_functional_2008; @Allie2015]. In addition, forest gap dynamics is an important driver of competition between above- and below-ground resources over space and time [@Hubbell1999; @VanBreugel2012], and has been recognized as a driver of coexistence of species closely related by habitat differentiation [@Yamasaki2013]. Conversely, neutral theory explains the local coexistence of functionally equivalent species through stochastic dynamics of life, death, reproduction, and dispersal [@Hubbell2001].  Emerging similarity theory reconciles niche and neutrality theories by suggesting the coexistence of distinct groups of species that are functionally similar within groups [@Scheffer2006; @Herault2007]. Ecological theories have also explored biotic interactions to explain species coexistence, such as the Janzen-Connell hypothesis, which explains species coexistence and maintenance of tropical diversity through interactions between dispersal distance and density-dependent offspring survival [@Clark1984; @Hyatt2003]. Beyond density-dependence, theories have explored the particular and primary role of tree fall and succession in natural tropical forests. After a tree fall, pioneer species first develop in light gaps, while late-successional species later develop under a closed canopy [@Craven2015], defining successional niches among species [@Herault2010]. Therefore, forest gap dynamics underscore the importance of considering temporal and spatial dynamics to understand species coexistence [@Soininen2010].

Traditionally, ecological theories focus on species, ignoring genotypic and phenotypic variation within species [@mcgill_rebuilding_2006], and therefore they ignore the evolutionary forces that determine the past and future of biodiversity. But "nothing in biology makes sense except in the light of evolution" [@Dobzhansky1973]. The assumption that evolution and ecology operate at very different spatial and temporal scales may have been an obstacle to understanding the role of eco-evolutionary processes in the coexistence of species [@Pelletier2009]. However, eco-evolutionary processes have been documented for some time [@Tutt1896] and may play an important role in species and community dynamics [@Bailey2009]. Individual adaptation in trees has traditionally been studied among populations using common gardens representing provenances sampled over broad ecological and spatial gradients [e.g. @Dewoody2015]. But fine environmental variation at the spatio-temporal scale can occur locally and lead to individual adaptations as well. There is evidence for eco-evolutionary processes at the microgeographic scale, for example in the vicinity of organism dispersal [@Richardson2014], and under gene flow [@Tigano2016; @Savolainen2007a]. Topography and forest gap dynamics have been shown to promote intraspecific divergence and microgeographic adaptations in neotropical tree species [@Brousseau2015; @Team2013; @Schmitt2020].  Other studies have found microgeographic changes in genetic structure and diversity of neotropical trees without identifying adaptations [@Jones2006; @Audigeos2013], including the effect of topography [@Torroba-Balmori2017], logging [@Degen2006; @Leclerc2015], and forest gaps [@Scotti2015]. Few studies have identified traits underlying genetic adaptations to local factors, such as serotiny adaptations to fire in Mediterranean pine [@Budde2014]. The *Symphonia globulifera* species complex particularly showed signatures of local adaptation to topography among species and adaptive growth strategies to forest gap dynamics within species [@Schmitt2020]. Nevertheless, to our knowledge, no study has examined the joint effect of topography and forest gap dynamics on the eco-evolutionary dynamics of tropical trees.

The long-term temporal dynamics of forest gaps [@Chambers2013]
and tree eco-evolution [@Petit2006] is obviously an impediment to such studies.


* evolution simulator
* ecological simulator
* eco-evo-simulator with changing environment
* simulator with spatial processes
* overlapping generations
* ours = first with all needed for tropical forests
* questions
  * What are the effects of species biological parameters on the establishment of genotypic adaptations to forest gap dynamics and topography in the forest community?
  * What are the effects of species biological parameters on the population structure and divergence with forest gap dynamics and topography in the forest community?
* hypotheses

# Material and Methods

## Simulator

Trees are simulated on a grid of size $grid×grid$.
Each cell of the grid is defined with two environmental values,
one for topography and one for forest gap dynamics.
Mature canopy trees are initialized in each cell.
Then, simulations follow three basic steps for each tree in each cell of the  grid over $Nt$ time-steps (Fig. \@ref(fig:ecoevosim)):
(1) death, (2) reproduction and recruitment, and (3) selection.
The source code is written as a R package in R and C++.
Version 1.0.0 used in this study is available at https://github.com/sylvainschmitt/ecoevosimulator .

```{r ecoevosim, fig.cap="Representation of life cyle for individual trees in a spatially explicit grid in the simulator. Trees risk dying at each time step with a year-probability drawn in a binomial distribution (top). Dead trees (right) are replaced by recruitment from the reproduction of parent trees (bottom). A single individual is selected in each cell (left) from all seedlings based on their viability. See material and methods for further details."}
include_graphics("figures/EcoEvoSimulator.png")
```

### Environments

The grid is defined with two environmental variables:
(1) a topographic wetness index (TWI) 
representing abiotic environment through topography [@Schmitt2020],
and (2) a neighbourhood crowding index (NCI),
representing biotic environment through forest gap dynamics [@Schmitt2020a].

The topographic wetness index  (TWI) is a proxy of water accumulation. 
The TWI is defined by the watershed of the cell divided by the local slope, 
and thus represents a relative measure of soil moisture availability [@Kopecky2010],
where a higher TWI represents greater soil moisture availability.
Water access and topography have been identified as crucial to forest dynamics [@ferry2010higher] and species distribution [@Allie2015; @Schmitt2020].

The neighbourhood crowding index [NCI; @Uriarte2004a] was used as a proxy of forest gap dynamics [@Schmitt2020a].
The neighbourhood crowding index $NCI_i$ for each tree individual $i$  was defined with the following formula:

\begin{equation} 
  NCI_i = \sum^J_{j|\delta_{i,j}<20m} DBH_j^2e^{-\frac14\delta_{i,j}}
  (\#eq:nci)
\end{equation} 

where $DBH_j$ is the diameter at breast height of neighbouring tree $j$ 
and $\delta_{i,j}$ its distance to individual tree $i$.
$NCI_i$ is defined for neighbours at a distance $\delta_{i,j}$ of up to 20m.
The size effect of neighbours was taken as their squared diameter, 
and hence proportional to their basal area. 
The distance effect of neighbours was set to $-\frac{1}{4}$,
corresponding to neighbours beyond 20m having less than 1% effect compared to the effect of neighbours at 0m.
We simulated NCI over time using four distributions:

1. a normal distribution to initialize NCI values in each cells
1. a bernoulli distribution to decide if the cell observed an increase in NCI at the next time-step, i.e. growing neighbours, or a decrease in NCI at the next time-step, i.e. a treefall
1. a lognormal distribution for increasing NCI, i.e. growing neighbours
1. an exponential distribution  for decreasing NCI, i.e. a treefall

We used the following formula to simulate NCI over time $t$:

\begin{equation} 
  NCI_{i,t=0} \sim \mathcal N(\mu_{NCI}, \sigma_{NCI}) \\
  NCI_{i,t} = NCI_{i,t-1} + \delta_{NCI_{i,t}} \\
  P(\delta_{NCI_{i,t}} \leq 0) \sim \mathcal B(logit^{-1}(\alpha + \beta.NCI_{i,t-1})) \\
  \delta_{NCI_{i,t}}\sim \left\{
    \begin{array}{ll}
        \mathcal {logN} (\mu_{\delta_{NCI} > 0}, \sigma_{\delta_{NCI} > 0}) & \mbox{if } \delta NCI > 0 \\
        \mathcal - {Exp} (\lambda_{\delta_{NCI} \leq 0}) & \mbox{ifelse}
    \end{array}
\right.
  (\#eq:ncisim)
\end{equation} 

We aggregated the simulated NCI at 3m to represent spatial auto-correlation.

### Initialization 

The grid is initialized with mature canopy trees.
Trees genotypes $g_i$ are drawn in a normal distribution centered on environmental mean $\mu_E$ of genetic variance $\sigma^2_G$. 
Trees phenotypes $p_i$ are drawn from a normal distribution centered on trees genotypes $g_i$ with a plastic variance $\sigma^2_P$:

\begin{equation} 
  p_i ∼ \mathcal N(g_i, \sigma^2_P) \\
  g_i ∼ \mathcal N(\mu_E,\sigma^2_G)
  (\#eq:init)
\end{equation} 

Note that two genotypes and phenotypes are defined for trees, 
one for topography and one for forest gap dynamics.
We further defined heritability as the genetic variance $\sigma^2_G$ divided by total variance, to be used as a unique biological parameter:

\begin{equation} 
  h^2 = \frac{\sigma^2_G}{\sigma^2_G+\sigma^2_P}
  (\#eq:h2)
\end{equation} 

### Death

Trees risk dying with a random draw in a binomial distribution: 

\begin{equation} 
  Death∼B(timestep,P_{death})
  (\#eq:death)
\end{equation} 

where $timestep$ (year) is the number of Bernoulli trials and $P_{death}$ is the annual death probability. 
Dead trees are replaced with reproduction and recruitment steps. 
Remaining trees form the overlapping generations.

### Reproduction and recruitment

Following tree death, $N_s$ seedlings are recruited in each empty cell.
Seedlings parents are drawn using all trees from the previous time-step, 
i.e. including tree dying at the current time-step. 
The mother tree is selected at a distance $R_{dispersion}$ from the empty cell.
The father tree is selected at a distance $R_{pollination}$ from the mother tree.
Dispersal kernels are discrete circles with uniform probabilities and include the empty cell, i.e. allowing auto-fecundation. 
Seedlings genotypes $g_s$ are drawn from a normal distribution centered on parental mean genotype with variance of half of genetic variance $\frac{\sigma^2_G}{2}$ [following @Vincenzi2014].
Seedlings phenotypes $p_s$ are drawn from a normal distribution centered on seedlings genotypes $g_s$ with plastic variance $\sigma^2_P$:

\begin{equation} 
  p_s ∼ \mathcal N (g_s, \sigma^2_P) \\
  g_s ∼ \mathcal N (\frac{g_{mother}+g_{father}}{2}, \frac{\sigma^2_G}{2})
  (\#eq:repro)
\end{equation} 

### Selection

Finally, the seedling with the highest viability is selected to replace the dead tree as a mature tree in canopy among the $N_s$ seedlings.
Viability is defined as the inverse of the Euclidean distance between seedlings phenotype $p_s$ and the environment $e_s$ reduced by environment variance $\sigma^2_E$:

\begin{equation} 
  viability_s = [\sqrt{(\frac{p_s−e_s}{\sigma^2_E})^2}]^{-1}
  (\#eq:viability)
\end{equation}

Note that the Euclidean distance used for viability is calculated in two dimensions composed by topography and forest gap dynamics.

## Simulation experiment

We varied four biological parameters of interest (Tab. \@ref(tab:params), see SI? for the choice of parameters):

* heritability associated with topography, $h^2_{TWI}$, from 0.1 to 0.9
* heritability associated with forest gap dynamics, $h^2_{NCI}$, from 0.1 to 0.9
* number of seedlings, $N_s$, from 4 to 128
* dispersion radius, $R_{dispersion}$, from 1 to 10 cells

We used a grid of 50x50 over 60 time-steps of length 30 years (estimated time to reach the canopy for *Symphonia globulifera* trees since recruitment at 10cm DBH, SI?).
The TWI was directly derived at 1m resolution from a 1m resolution digital elevation model using SAGA-GIS [@Conrad2015] based on a LiDAR campaign carried out in 2015 at the Paracou field station (5°18′N, 52°53′W) in French Guiana.
TWI was standardized to the unit.
Values of NCI parameters and death probability were inferred from NCI values and death events observed in the Paracou field station across all censuses (SI?).
Pollination radius was set to one to focus on the variation of dispersion radius as they are symmetric (SI?).
Simulation stochasticity was low (SI?).
We thus ran one simulation per parameter set, 
resulting in 6,000 simulations 
($10~h^2_{TWI}\times10~h^2_{NCI}\times6~N_s\times10~R_{dispersion}$).


```{r params}
data.frame(
  Parameter = c(
    "$grid$", "$N_t$", "$timestep$",
    "$\\mu_{TWI}$", "$\\sigma^2_{P,TWI}$",
    "$\\mu_{NCI}$", "$\\sigma^2_{P,NCI}$",
    "$\\alpha$", "$\\beta$",
    "$\\mu_{\\delta_{NCI} > 0}$", "$\\sigma_{\\delta_{NCI} > 0}$",
    "$\\lambda_{\\delta_{NCI} \\leq 0}$",
    "$h^2_{TWI}$", "$h^2_{NCI}$",
    "$P_{death}$", "$N_s$", 
    "$R_{pollination}$", "$R_{dispersion}$"
  ),
  Equation = c(
    "", "", "\\@ref(eq:death)",
    "\\@ref(eq:init)", "\\@ref(eq:init)", 
    "\\@ref(eq:ncisim), \\@ref(eq:init)", "\\@ref(eq:ncisim), \\@ref(eq:init)",
    "\\@ref(eq:ncisim)", "\\@ref(eq:ncisim)",
    "\\@ref(eq:ncisim)", "\\@ref(eq:ncisim)",
    "\\@ref(eq:ncisim)",
    "\\@ref(eq:h2)", "\\@ref(eq:h2)",
    "\\@ref(eq:death)", "\\@ref(eq:repro)",
    "\\@ref(eq:repro)", "\\@ref(eq:repro)"
  ),
  Values = c(
    "50", "60", "30",
    "0", "1", 
    "123", "26",
    "-1.32", "0.003",
    "0.749", "2.651",
    "0.31",
    "$[1:9].10^{-1}$", "$[1:9].10^{-1}$",
    "0.013", "$2^{[2:7]}$",
    "1", "$[1:10]$"
  ),
  Description = c(
    "Grid size", 
    "Number of time-steps (ca 3m)",
    "Length of a time-step (years)",
    "TWI mean (normal distribution)",
    "TWI plastic variance",
    "NCI mean",
    "NCI plastic variance",
    "Odd of negative NCI variation",
    "Effect of NCI from previous year on the odd of negative NCI variation",
    "Mode of positive NCI variation",
    "Scale of positive NCI variation",
    "Rate of negative NCI variation",
    "Heritability associated with TWI",
    "Heritability associated with NCI",
    "Death probability per year",
    "Number of seedlings",
    "Pollination radius (number of cells)",
    "Dispersion radius (number of cells)"
  )
) %>% kable(caption = "Model parameters with their name, referenced equation, values in the experiment, and description.")
```

## Analysis

### Local adaptation

We defined local adaptation as the inverse of the Euclidean distance between tree genotype $g_i$ and tree environment $e_i$ reduced by environment variance $\sigma^2_E$. 
We used the mean value $D_G$ across the grid as a metric of local adaptation at time-step $t$:

\begin{equation} 
  D_G(t) = \overline{[\sqrt{(\frac{g_i(t)−e_i(t)}{\sigma^2_E})^2}]^{-1}}
  (\#eq:Dg)
\end{equation}

A $D_G(t)$ close to zero indicates an absence of local adaptation,
whereas high $D_G(t)$ indicates strong local adaptation.

### Population structure

We used k-means to test for the existence of at least two genetically distinct groups in the tree community, using the localization of genotypes along environmental gradients.
We defined a $WSS_{score}$ as an index of population structure based on the weighted sum of squares of every individual distance for a k-means of respectively 1 and 2 groups:

\begin{equation} 
  WSS_{score}(t) = \frac{WSS_{k=1}(t)}{WSS_{k=2}(t)} \frac12
  (\#eq:WSSscore)
\end{equation}

The weighted sum of squares ($WSS$) represent the group total variation along environmental gradients.
A mean $WSS$ of the two groups equal to half the $WSS$ of the whole community indicates adjacent groups, revealing an absence of population structure.
On the opposite, a mean $WSS$ of the two groups lower than half the $WSS$ of the whole community indicates distant groups, revealing population structure.
In a nutshell, $WSS_{score}$ close to one indicates an absence of population structure, whereas high $WSS_{score}$ indicates at least two structured populations.

### Statistical analysis

We tested the effect of the four biological parameters and their first order interaction on the variation of local adaptation $D_G$ and population structure $WSS_{score}$ for $TWI$ and $NCI$ after 60 time-steps (ca 1,800 years): heritabilites associated with TWI ($h^2_{TWI}$ and NCI ($h^2_{NCI}$) number of seedlings ($N_s$),  and dispersion radius ($R_{dispersion}$) as follows:

$$
log(Y) \sim \mathcal N(\mu + \theta_{topo} \times h^2_{TWI} + \theta_{gaps} \times h^2_{NCI} + \\ \theta_{seed} \times N_s + \theta_{disp} \times R_{dispersion} +  \theta_{topo:gaps} \times h^2_{TWI} \times h^2_{NCI} + \\\theta_{topo:seed} \times h^2_{TWI} \times N_s + \theta_{topo:disp} \times h^2_{TWI} \times R_{dispersion} + \\ \\\theta_{gaps:seed} \times h^2_{NCI} \times N_s + \theta_{gaps:disp} \times h^2_{NCI} \times R_{dispersion} +\\ \theta_{seed:disp} \times N_s \times R_{dispersion}, \sigma^2)
$$

where all $\theta$ model the effect of each biological parameters and their first order interactions on $log(Y)$ under a normal law of variance $\sigma^2$.
The analysis was conducted four times for each metric (local adaptation $D_G$ and population structure $WSS_{score}$) and each environmental variable (topography $TWI$ and forest gap dynamics $NCI$).
Response variable $Y$ was log-transformed to improve the normality of the residuals.

# Results

At the exception of population structure $WSS_{score}$ with $NCI$, the four biological parameters and their paired interactions explained most of the variation (Fig. \@ref(fig:parmetersfig) and Tab. \@ref(tab:parmeterstab)) in local adaptations $D_G$ ($R^2>0.9$) and population structure $WSS_{score}$ ($R^2=0.82$ for $TWI$).
Strong local adaptation to temporally-fixed topography promote populations structure, 
whereas reduced local adaptation to temporally-variable forest gap dynamics does not promote populations structure.

The heritability associated with topography $h^2_{TWI}$ favoured local adaptation and population structure along topography ($D_G$ and $WSS_{score}$ for $TWI$), 
while the heritability associated to forest gap dynamics $h^2_{NCI}$ limited local adaptation to topography ($D_G$ for $TWI$) but favoured local adaptation to forest gap dynamics ($D_G$ for $NCI$).
Moreover, the interaction between heritabilities associated with topography and forest gap dynamics ($h^2_{TWI}\times h^2_{NCI}$) favoured local adaptations to topography and forest gap dynamics ($D_G$  for $TWI$ and $NCI$, Fig. \@ref(fig:interaction)), and population structure along topography ($WSS_{score}$ for $TWI$).
Local adaptation and population structure along topography increased with increasing heritability to topography and forest gap dynamics, but decreased if heritability to forest gap dynamics became stronger than heritability to topography, i.e. genotypes got more locally adapted to forest gap dynamics (Fig. \@ref(fig:interaction)).

The number of seedlings $N_s$ alone limited local adaptation to topography ($D_G$ for $TWI$),
and its interaction with heritabilities associated with topography $h^2_{TWI} \times N_s$ or forest gap dynamics $h^2_{NCI} \times N_s$ reinforced the effect of heritiabilites on the system.
The number of seedlings, related to the strength selection, 
increased the effect of heritability on local adaptation and population structure.

The radius of dispersion $R_{dispersion}$ alone limited local adaptation and population structure along topography ($D_G$ and $WSS_{score}$ for $TWI$),
and its interaction with heritabilities associated with topography $h^2_{TWI} \times N_s$ and forest gap dynamics $h^2_{NCI} \times N_s$ limited local adaptation to topography ($D_G$ for $TWI$) and forest gap dynamics ($D_G$ for $NCI$), respectively. 
The radius of dispersion, related to the distance of the gene flow, 
decreased local adaptation and population structure.

```{r R2, eval=F}
vroom::vroom("../modelling_save/simulations.tsv") %>% 
  filter(timestep == 60) %>% 
  dplyr::select(-timestep) %>% 
  mutate(parameter.value = round(parameter.value, 1)) %>% 
  reshape2::dcast(variable + metric + metric.value ~ parameter) %>% 
  group_by(metric, variable) %>% 
  do(lm = lm(metric.value ~ Htopo + Hnci + Ns + Rdispersion + 
               Htopo:Hnci + Htopo:Ns + Htopo:Rdispersion + 
               Hnci:Ns + Hnci:Rdispersion + Ns:Rdispersion, data = .) %>% 
       broom::glance()) %>% 
  unnest(lm) %>% 
  dplyr::select(metric, variable, r.squared) %>% 
  mutate(variable = recode(variable, "nci" = "NCI", "topo" = "TWI")) %>% 
  mutate(metric = recode(metric, "Dg" = "$D_G$", "wss score" = "$WSS_{score}$")) %>%
  reshape2::dcast(metric ~ variable) %>% 
  kable(caption = "Goodness of fit ($R^2$) for the four linear regressions. $D_G$ and $WSS_{score}$ represent local adaptation and population structure, respectively. $TWI$ and $NCI$ represent topography and forest gap dynamics, respectively.", 
        col.names = c("Metric", "NCI", "TWI"), digits = 2)
```

```{r parmetersfig, fig.cap="Effects of biological parameters and their paired interactions on local adaptation ($D_G$) and population structure ($WSS_{score}$) for topography ($TWI$) and forest gap dynamics ($NCI$). Parameters values have been scaled to ease their comparison. topo, gaps, seed, and disp are refereing to the heritiability associated with topography ($h^2_{TWI}$), the heritability associated with forest gap dynamics ($h^2_{NCI}$), the number of seedlings ($N_s$), and the radius of dispersion ($R_{dispersion}$), respectively. Significant ($p < 0.05$) and strong ($|standardized~effect| > 0.05$) effects are opaque, whereas weak or non-significant effects are transparent. Dashed grey lines indicate a null effect, whereas dotted grey lines indicate an absolute standardized effect of 0.05."}
vroom::vroom("../modelling_save/simulations.tsv") %>% 
  filter(timestep == 60) %>% 
  dplyr::select(-timestep) %>% 
  mutate(metric.value = ifelse(metric == "Dg", 1/metric.value, metric.value)) %>% 
  group_by(parameter) %>%
  mutate(parameter.value = as.vector(scale(parameter.value, center = F))) %>%
  reshape2::dcast(variable + metric + metric.value ~ parameter) %>% 
  group_by(metric, variable) %>% 
  do(lm = lm(log(metric.value) ~ Htopo + Hnci + Ns + Rdispersion +
               Htopo:Hnci + Htopo:Ns + Htopo:Rdispersion + Hnci:Ns + 
               Hnci:Rdispersion + Ns:Rdispersion, data = .) %>%
       broom::tidy()) %>%
  unnest(lm) %>% 
  filter(term != "(Intercept)") %>% 
  dplyr::select(metric, variable, term, estimate, std.error) %>% 
  mutate(variable = recode(variable, "nci" = "NCI", "topo" = "TWI")) %>% 
  mutate(metric = recode(metric, "Dg" = "log(D[G])", 
                         "wss score" = "log(WSS[score])")) %>%
  mutate(term = recode_factor(term, 
                              "Ns:Rdispersion" = "seed:disp",
                              "Hnci:Rdispersion" = "gaps:disp",
                              "Hnci:Ns" = "gaps:seed",
                              "Htopo:Rdispersion" = "topo:disp",
                              "Htopo:Ns" = "topo:seed",
                              "Htopo:Hnci" = "topo:gaps",
                              "Rdispersion" = "disp",
                              "Ns" = "seed", 
                              "Hnci" = "gaps",
                              "Htopo" = "topo"
  )) %>%
  ggplot(aes(term, col = term)) +
  geom_hline(yintercept = 0, col = "lightgrey", linetype = "dashed") +
  geom_hline(yintercept = c(0.05, -0.05), col = "lightgrey", linetype = "dotted") +
  geom_point(aes(y = estimate, alpha = (abs(estimate) > 0.05))) +
  geom_segment(aes(xend = term, y = estimate - std.error, 
                   yend = estimate + std.error)) +
  coord_flip() +
  facet_grid(metric ~ variable, labeller = "label_parsed") +
  xlab("") + ylab("Standardized effect") +
  scale_alpha_manual(guide = "none", values = c(0.2, 1)) +
  scale_color_manual(guide = "none", values = rcartocolor::carto_pal(10, "Safe")) +
  theme(axis.text.y = element_text(colour = rcartocolor::carto_pal(10, "Safe"))) +
  geom_text(aes(label = R2), col = "black", x = "gaps:disp", y = 0.4,
            data = data.frame(R2 = paste0("R² = ", c(0.93, 0.91, 0.18, 0.82)),
                              metric = c("log(D[G])", "log(D[G])",
                                         "log(WSS[score])", "log(WSS[score])"),
                              variable = c("NCI", "TWI", "NCI", "TWI")))
```

```{r parmeterstab}
vroom::vroom("../modelling_save/simulations.tsv") %>% 
  filter(timestep == 60) %>% 
  dplyr::select(-timestep) %>% 
  mutate(metric.value = ifelse(metric == "Dg", 1/metric.value, metric.value)) %>% 
  group_by(parameter) %>%
  mutate(parameter.value = as.vector(scale(parameter.value, center = F))) %>%
  reshape2::dcast(variable + metric + metric.value ~ parameter) %>% 
  group_by(metric, variable) %>% 
  do(lm = lm(log(metric.value) ~ Htopo + Hnci + Ns + Rdispersion +
               Htopo:Hnci + Htopo:Ns + Htopo:Rdispersion + 
               Hnci:Ns + Hnci:Rdispersion + Ns:Rdispersion, data = .) %>%
       broom::tidy()) %>% 
  unnest(lm) %>% 
  filter(term != "(Intercept)") %>% 
  filter(p.value < 0.05) %>% 
  filter(abs(estimate) > 0.05) %>% 
  mutate(est = paste0(round(estimate, 3), " (",
                      round(std.error, 3), ")")) %>% 
  dplyr::select(metric, variable, term, est) %>%
  reshape2::dcast(term ~ metric + variable) %>% 
  mutate_all(funs(ifelse(is.na(.), "", .))) %>% 
  mutate(term = recode(term,
                       "Htopo" = "$h^2_{TWI}$",
                       "Hnci" = "$h^2_{NCI}$",
                       "Ns" = "$N_s$",
                       "Rdispersion" = "$R_{dispersion}$",
                       "Htopo:Hnci" = "$h^2_{TWI} \\times h^2_{NCI}$",
                       "Htopo:Ns" = "$h^2_{TWI} \\times N_s$",
                       "Htopo:Ns" = "$h^2_{TWI} \\times N_s$",
                       "Htopo:Rdispersion" = "$h^2_{TWI} \\times R_{dispersion}$",
                       "Hnci:Ns" = "$h^2_{NCI} \\times N_s$",
                       "Hnci:Rdispersion" = "$h^2_{NCI} \\times R_{dispersion}$",
                       "Ns:Rdispersion" = "$N_s \\times R_{dispersion}$",
  )) %>%
  kable(caption = "Effects of biological parameters and their paired interactions on local adaptation ($D_G$) and population structure ($WSS_{score}$) for topography ($TWI$) and forest gap dynamics ($NCI$). Parameters values have been scaled to ease their comparison. Biological parameters include the heritiability associated with topography ($h^2_{TWI}$), the heritability associated with forest gap dynamics ($h^2_{NCI}$), the number of seedlings ($N_s$), and the radius of dispersion ($R_{dispersion}$). The table show parameters effect and standard deviation in bracket for parameters with a significant ($p < 0.05$) and strong ($|standardized~effect| > 0.05$) effect.",
        col.names = c("Parameter", "$D_G$ for $NCI$", "$D_G$ for $TWI$",
                      "$WSS_{score}$ for $TWI$"))
```


```{r interaction, fig.cap="Projection of the effects of the heritiability associated with topography ($h^2_{TWI}$), the heritability associated with forest gap dynamics ($h^2_{NCI}$), and their interaction ($h^2_{TWI} \\times h^2_{NCI}$) on local adaptation to topography ($D_G$ for $TWI$). The x-axis shows the variation in the heritiability associated with topography ($h^2_{TWI}$). The colour scale shows the varaition in the heritability associated with forest gap dynamics ($h^2_{NCI}$) from low in blue to high in red. The y-axis shows the projected mean local adaptation to topography ($D_G$ for $TWI$)."}
t1 <- vroom::vroom("../modelling_save/simulations.tsv") %>% 
  filter(timestep == 60) %>% 
  dplyr::select(-timestep) %>% 
  mutate(metric.value = ifelse(metric == "Dg", 1/metric.value, metric.value)) %>% 
  group_by(parameter) %>%
  reshape2::dcast(variable + metric + metric.value ~ parameter) %>% 
  group_by(metric, variable) %>% 
  filter(metric == "Dg", variable == "topo")
m1  <- lm(log(metric.value) ~ Htopo + Hnci + Ns + Rdispersion +
               Htopo:Hnci + Htopo:Ns + Htopo:Rdispersion + 
               Hnci:Ns + Hnci:Rdispersion + Ns:Rdispersion, data = t1)
Inter.HandPick1 <- effects::effect('Htopo*Hnci', m1,
                                  xlevels=list(Htopo = seq(0, 1, 0.01),
                                               Hnci = seq(0, 1, 0.01))) %>% 
  as.data.frame()
ggplot(data=Inter.HandPick1, aes(x = Htopo, y = fit, group = Hnci))+
  geom_line(size=2, aes(color=Hnci))+
  ylab(expression(D[G]~TWI)) +
  xlab(expression(h[TWI]^2)) +
  scale_colour_gradient2(expression(h[NCI]^2), midpoint = 0.5,
                        low = scales::muted("blue"),
                        high = scales::muted("red"))
```

```{r interaction2, fig.cap="Projection of the effects of the heritiability associated with topography ($h^2_{TWI}$), the heritability associated with forest gap dynamics ($h^2_{NCI}$), and their interaction ($h^2_{TWI} \\times h^2_{NCI}$) on local adaptation to topography ($D_G$ for $TWI$). The x-axis shows the variation in the heritiability associated with topography ($h^2_{TWI}$). The colour scale shows the varaition in the heritability associated with forest gap dynamics ($h^2_{NCI}$) from low in blue to high in red. The y-axis shows the projected mean local adaptation to topography ($D_G$ for $TWI$)."}
sims <- vroom::vroom("../modelling_save/simulations.tsv") %>% 
  filter(timestep == 60) %>% 
  dplyr::select(-timestep) %>% 
  mutate(metric.value = ifelse(metric == "Dg", 1/metric.value, metric.value)) %>% 
  group_by(parameter) %>%
  reshape2::dcast(variable + metric + metric.value ~ parameter) %>% 
  group_by(metric, variable)
preds <- lapply(list(c("Dg", "topo"),
            c("Dg", "nci"),
            c("wss score", "topo"),
            c("wss score", "nci")),
  function(p){
  fit  <- lm(log(metric.value) ~ Htopo + Hnci + Ns + Rdispersion +
               Htopo:Hnci + Htopo:Ns + Htopo:Rdispersion + 
               Hnci:Ns + Hnci:Rdispersion + Ns:Rdispersion, 
             data = filter(sims, metric == p[1], variable == p[2])  )
  effects::effect('Htopo*Hnci', fit,
                  xlevels=list(Htopo = seq(0, 1, 0.01),
                               Hnci = seq(0, 1, 0.01))) %>% 
    as.data.frame() %>% 
    mutate(metric = p[1], variable = p[2])
}) %>% bind_rows() %>% 
  mutate(variable = recode(variable, "nci" = "NCI", "topo" = "TWI")) %>% 
  mutate(metric = recode(metric, "Dg" = "log(D[G])", 
                         "wss score" = "log(WSS[score])"))
ggplot(data = preds, aes(x = Htopo, y = Hnci)) +
  metR::geom_contour_fill(aes(z = fit), bins = 25) +
  xlab(expression(h[TWI]^2)) +
  ylab(expression(h[NCI]^2)) +
  viridis::scale_fill_viridis(expression(log(metric)),  option = "magma", discrete = F) +
  facet_grid(metric ~ variable, labeller = "label_parsed")

ggplot(data = filter(preds, metric == "log(D[G])"),
       aes(x = Htopo, y = Hnci)) +
  metR::geom_contour_fill(aes(z = fit), bins = 25) +
  xlab(expression(h[TWI]^2)) +
  ylab(expression(h[NCI]^2)) +
  viridis::scale_fill_viridis(expression(log(metric)),  option = "magma", discrete = F) +
  facet_grid(metric ~ variable, labeller = "label_parsed")

ggplot(data = filter(preds, metric == "log(D[G])", variable == "NCI"),
       aes(x = Htopo, y = Hnci)) +
  metR::geom_contour_fill(aes(z = fit), bins = 25) +
  xlab(expression(h[TWI]^2)) +
  ylab(expression(h[NCI]^2)) +
  viridis::scale_fill_viridis(expression(log(metric)),  option = "magma", discrete = F) +
  facet_grid(metric ~ variable, labeller = "label_parsed")

filter(preds, metric == "log(D[G])") %>% 
  group_by(Htopo, Hnci) %>% 
  summarise(m = mean(fit)) %>% 
  ggplot(aes(x = Htopo, y = Hnci)) +
  metR::geom_contour_fill(aes(z = m), bins = 200) +
  xlab(expression(h[TWI]^2)) +
  ylab(expression(h[NCI]^2)) +
  viridis::scale_fill_viridis(expression(log(D[G])),  option = "magma", discrete = F) +
  xlim(0, 0.25) + ylim(0, 0.25)
```

**To be compared with species environmental niches across NCI and TWI (mean position = niche mean + bars = niche width).**

# Discussion

> The topography and dynamics of forest gaps jointly favoured or limited the local adaptation and structure of the populations of a tropical tree species, depending on the interaction between four biological parameters of the species.

# Acknowledgements

We thank the University of Bordeaux for a PhD grant to Sylvain Schmitt. 
XXX.
This study was partially funded by an Investissement d’Avenir grant of the ANR: CEBA (ANR-10-LABEX-0025).

# Authors’ contributions

# Data accessibility

\newpage 

# References

<div id="refs"></div>

\newpage 

## Table captions

__Table \@ref(tab:headcount)__ 

## Figure captions

__Figure \@ref(fig:PCA)__ 
